<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ad0000; } /* Alert */
code span.an { color: #5e5e5e; } /* Annotation */
code span.at { color: #20794d; } /* Attribute */
code span.bn { color: #ad0000; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007ba5; } /* ControlFlow */
code span.ch { color: #20794d; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #5e5e5e; } /* Comment */
code span.cv { color: #5e5e5e; font-style: italic; } /* CommentVar */
code span.do { color: #5e5e5e; font-style: italic; } /* Documentation */
code span.dt { color: #ad0000; } /* DataType */
code span.dv { color: #ad0000; } /* DecVal */
code span.er { color: #ad0000; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #ad0000; } /* Float */
code span.fu { color: #4758ab; } /* Function */
code span.im { } /* Import */
code span.in { color: #5e5e5e; } /* Information */
code span.kw { color: #007ba5; } /* Keyword */
code span.op { color: #5e5e5e; } /* Operator */
code span.ot { color: #007ba5; } /* Other */
code span.pp { color: #ad0000; } /* Preprocessor */
code span.sc { color: #20794d; } /* SpecialChar */
code span.ss { color: #20794d; } /* SpecialString */
code span.st { color: #20794d; } /* String */
code span.va { color: #111111; } /* Variable */
code span.vs { color: #20794d; } /* VerbatimString */
code span.wa { color: #5e5e5e; font-style: italic; } /* Warning */
</style>

  <!--radix_placeholder_meta_tags-->
  <title>C086: Multi-sample comparisons of the Kinkel (C086) Smchd1 LSK mini-bulk data set</title>

  <meta property="description" itemprop="description" content="Read-based analysis"/>


  <!--  https://schema.org/Article -->
  <meta property="article:published" itemprop="datePublished" content="2021-10-06"/>
  <meta property="article:created" itemprop="dateCreated" content="2021-10-06"/>
  <meta name="article:author" content="Peter Hickey"/>

  <!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
  <meta property="og:title" content="C086: Multi-sample comparisons of the Kinkel (C086) Smchd1 LSK mini-bulk data set"/>
  <meta property="og:type" content="article"/>
  <meta property="og:description" content="Read-based analysis"/>
  <meta property="og:locale" content="en_US"/>
  <meta property="og:site_name" content="C086"/>

  <!--  https://dev.twitter.com/cards/types/summary -->
  <meta property="twitter:card" content="summary"/>
  <meta property="twitter:title" content="C086: Multi-sample comparisons of the Kinkel (C086) Smchd1 LSK mini-bulk data set"/>
  <meta property="twitter:description" content="Read-based analysis"/>

  <!--/radix_placeholder_meta_tags-->
  
  <!--radix_placeholder_rmarkdown_metadata-->

  <script type="text/json" id="radix-rmarkdown-metadata">
  {"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","description","author","date","editor_options","bibliography"]}},"value":[{"type":"character","attributes":{},"value":["Multi-sample comparisons of the Kinkel (C086) Smchd1 LSK mini-bulk data set"]},{"type":"character","attributes":{},"value":["Read-based analysis"]},{"type":"list","attributes":{},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","url","affiliation","affiliation_url"]}},"value":[{"type":"character","attributes":{},"value":["Peter Hickey"]},{"type":"character","attributes":{},"value":["https://peterhickey.org"]},{"type":"character","attributes":{},"value":["WEHI SCORE"]},{"type":"character","attributes":{},"value":["https://www.wehi.edu.au/people/shalin-naik/3310/score"]}]}]},{"type":"character","attributes":{},"value":["2021-10-06"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["chunk_output_type"]}},"value":[{"type":"character","attributes":{},"value":["console"]}]},{"type":"character","attributes":{},"value":["ref.bib"]}]}
  </script>
  <!--/radix_placeholder_rmarkdown_metadata-->
  <!--radix_placeholder_navigation_in_header-->
  <meta name="distill:offset" content=""/>

  <script type="application/javascript">

    window.headroom_prevent_pin = false;

    window.document.addEventListener("DOMContentLoaded", function (event) {

      // initialize headroom for banner
      var header = $('header').get(0);
      var headerHeight = header.offsetHeight;
      var headroom = new Headroom(header, {
        tolerance: 5,
        onPin : function() {
          if (window.headroom_prevent_pin) {
            window.headroom_prevent_pin = false;
            headroom.unpin();
          }
        }
      });
      headroom.init();
      if(window.location.hash)
        headroom.unpin();
      $(header).addClass('headroom--transition');

      // offset scroll location for banner on hash change
      // (see: https://github.com/WickyNilliams/headroom.js/issues/38)
      window.addEventListener("hashchange", function(event) {
        window.scrollTo(0, window.pageYOffset - (headerHeight + 25));
      });

      // responsive menu
      $('.distill-site-header').each(function(i, val) {
        var topnav = $(this);
        var toggle = topnav.find('.nav-toggle');
        toggle.on('click', function() {
          topnav.toggleClass('responsive');
        });
      });

      // nav dropdowns
      $('.nav-dropbtn').click(function(e) {
        $(this).next('.nav-dropdown-content').toggleClass('nav-dropdown-active');
        $(this).parent().siblings('.nav-dropdown')
           .children('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $("body").click(function(e){
        $('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $(".nav-dropdown").click(function(e){
        e.stopPropagation();
      });
    });
  </script>

  <style type="text/css">

  /* Theme (user-documented overrideables for nav appearance) */

  .distill-site-nav {
    color: rgba(255, 255, 255, 0.8);
    background-color: #0F2E3D;
    font-size: 15px;
    font-weight: 300;
  }

  .distill-site-nav a {
    color: inherit;
    text-decoration: none;
  }

  .distill-site-nav a:hover {
    color: white;
  }

  @media print {
    .distill-site-nav {
      display: none;
    }
  }

  .distill-site-header {

  }

  .distill-site-footer {

  }


  /* Site Header */

  .distill-site-header {
    width: 100%;
    box-sizing: border-box;
    z-index: 3;
  }

  .distill-site-header .nav-left {
    display: inline-block;
    margin-left: 8px;
  }

  @media screen and (max-width: 768px) {
    .distill-site-header .nav-left {
      margin-left: 0;
    }
  }


  .distill-site-header .nav-right {
    float: right;
    margin-right: 8px;
  }

  .distill-site-header a,
  .distill-site-header .title {
    display: inline-block;
    text-align: center;
    padding: 14px 10px 14px 10px;
  }

  .distill-site-header .title {
    font-size: 18px;
    min-width: 150px;
  }

  .distill-site-header .logo {
    padding: 0;
  }

  .distill-site-header .logo img {
    display: none;
    max-height: 20px;
    width: auto;
    margin-bottom: -4px;
  }

  .distill-site-header .nav-image img {
    max-height: 18px;
    width: auto;
    display: inline-block;
    margin-bottom: -3px;
  }



  @media screen and (min-width: 1000px) {
    .distill-site-header .logo img {
      display: inline-block;
    }
    .distill-site-header .nav-left {
      margin-left: 20px;
    }
    .distill-site-header .nav-right {
      margin-right: 20px;
    }
    .distill-site-header .title {
      padding-left: 12px;
    }
  }


  .distill-site-header .nav-toggle {
    display: none;
  }

  .nav-dropdown {
    display: inline-block;
    position: relative;
  }

  .nav-dropdown .nav-dropbtn {
    border: none;
    outline: none;
    color: rgba(255, 255, 255, 0.8);
    padding: 16px 10px;
    background-color: transparent;
    font-family: inherit;
    font-size: inherit;
    font-weight: inherit;
    margin: 0;
    margin-top: 1px;
    z-index: 2;
  }

  .nav-dropdown-content {
    display: none;
    position: absolute;
    background-color: white;
    min-width: 200px;
    border: 1px solid rgba(0,0,0,0.15);
    border-radius: 4px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.1);
    z-index: 1;
    margin-top: 2px;
    white-space: nowrap;
    padding-top: 4px;
    padding-bottom: 4px;
  }

  .nav-dropdown-content hr {
    margin-top: 4px;
    margin-bottom: 4px;
    border: none;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .nav-dropdown-active {
    display: block;
  }

  .nav-dropdown-content a, .nav-dropdown-content .nav-dropdown-header {
    color: black;
    padding: 6px 24px;
    text-decoration: none;
    display: block;
    text-align: left;
  }

  .nav-dropdown-content .nav-dropdown-header {
    display: block;
    padding: 5px 24px;
    padding-bottom: 0;
    text-transform: uppercase;
    font-size: 14px;
    color: #999999;
    white-space: nowrap;
  }

  .nav-dropdown:hover .nav-dropbtn {
    color: white;
  }

  .nav-dropdown-content a:hover {
    background-color: #ddd;
    color: black;
  }

  .nav-right .nav-dropdown-content {
    margin-left: -45%;
    right: 0;
  }

  @media screen and (max-width: 768px) {
    .distill-site-header a, .distill-site-header .nav-dropdown  {display: none;}
    .distill-site-header a.nav-toggle {
      float: right;
      display: block;
    }
    .distill-site-header .title {
      margin-left: 0;
    }
    .distill-site-header .nav-right {
      margin-right: 0;
    }
    .distill-site-header {
      overflow: hidden;
    }
    .nav-right .nav-dropdown-content {
      margin-left: 0;
    }
  }


  @media screen and (max-width: 768px) {
    .distill-site-header.responsive {position: relative; min-height: 500px; }
    .distill-site-header.responsive a.nav-toggle {
      position: absolute;
      right: 0;
      top: 0;
    }
    .distill-site-header.responsive a,
    .distill-site-header.responsive .nav-dropdown {
      display: block;
      text-align: left;
    }
    .distill-site-header.responsive .nav-left,
    .distill-site-header.responsive .nav-right {
      width: 100%;
    }
    .distill-site-header.responsive .nav-dropdown {float: none;}
    .distill-site-header.responsive .nav-dropdown-content {position: relative;}
    .distill-site-header.responsive .nav-dropdown .nav-dropbtn {
      display: block;
      width: 100%;
      text-align: left;
    }
  }

  /* Site Footer */

  .distill-site-footer {
    width: 100%;
    overflow: hidden;
    box-sizing: border-box;
    z-index: 3;
    margin-top: 30px;
    padding-top: 30px;
    padding-bottom: 30px;
    text-align: center;
  }

  /* Headroom */

  d-title {
    padding-top: 6rem;
  }

  @media print {
    d-title {
      padding-top: 4rem;
    }
  }

  .headroom {
    z-index: 1000;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
  }

  .headroom--transition {
    transition: all .4s ease-in-out;
  }

  .headroom--unpinned {
    top: -100px;
  }

  .headroom--pinned {
    top: 0;
  }

  /* adjust viewport for navbar height */
  /* helps vertically center bootstrap (non-distill) content */
  .min-vh-100 {
    min-height: calc(100vh - 100px) !important;
  }

  </style>

  <script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
  <link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet"/>
  <link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet"/>
  <script src="site_libs/headroom-0.9.4/headroom.min.js"></script>
  <script src="site_libs/autocomplete-0.37.1/autocomplete.min.js"></script>
  <script src="site_libs/fuse-6.4.1/fuse.min.js"></script>

  <script type="application/javascript">

  function getMeta(metaName) {
    var metas = document.getElementsByTagName('meta');
    for (let i = 0; i < metas.length; i++) {
      if (metas[i].getAttribute('name') === metaName) {
        return metas[i].getAttribute('content');
      }
    }
    return '';
  }

  function offsetURL(url) {
    var offset = getMeta('distill:offset');
    return offset ? offset + '/' + url : url;
  }

  function createFuseIndex() {

    // create fuse index
    var options = {
      keys: [
        { name: 'title', weight: 20 },
        { name: 'categories', weight: 15 },
        { name: 'description', weight: 10 },
        { name: 'contents', weight: 5 },
      ],
      ignoreLocation: true,
      threshold: 0
    };
    var fuse = new window.Fuse([], options);

    // fetch the main search.json
    return fetch(offsetURL('search.json'))
      .then(function(response) {
        if (response.status == 200) {
          return response.json().then(function(json) {
            // index main articles
            json.articles.forEach(function(article) {
              fuse.add(article);
            });
            // download collections and index their articles
            return Promise.all(json.collections.map(function(collection) {
              return fetch(offsetURL(collection)).then(function(response) {
                if (response.status === 200) {
                  return response.json().then(function(articles) {
                    articles.forEach(function(article) {
                      fuse.add(article);
                    });
                  })
                } else {
                  return Promise.reject(
                    new Error('Unexpected status from search index request: ' +
                              response.status)
                  );
                }
              });
            })).then(function() {
              return fuse;
            });
          });

        } else {
          return Promise.reject(
            new Error('Unexpected status from search index request: ' +
                        response.status)
          );
        }
      });
  }

  window.document.addEventListener("DOMContentLoaded", function (event) {

    // get search element (bail if we don't have one)
    var searchEl = window.document.getElementById('distill-search');
    if (!searchEl)
      return;

    createFuseIndex()
      .then(function(fuse) {

        // make search box visible
        searchEl.classList.remove('hidden');

        // initialize autocomplete
        var options = {
          autoselect: true,
          hint: false,
          minLength: 2,
        };
        window.autocomplete(searchEl, options, [{
          source: function(query, callback) {
            const searchOptions = {
              isCaseSensitive: false,
              shouldSort: true,
              minMatchCharLength: 2,
              limit: 10,
            };
            var results = fuse.search(query, searchOptions);
            callback(results
              .map(function(result) { return result.item; })
            );
          },
          templates: {
            suggestion: function(suggestion) {
              var img = suggestion.preview && Object.keys(suggestion.preview).length > 0
                ? `<img src="${offsetURL(suggestion.preview)}"</img>`
                : '';
              var html = `
                <div class="search-item">
                  <h3>${suggestion.title}</h3>
                  <div class="search-item-description">
                    ${suggestion.description || ''}
                  </div>
                  <div class="search-item-preview">
                    ${img}
                  </div>
                </div>
              `;
              return html;
            }
          }
        }]).on('autocomplete:selected', function(event, suggestion) {
          window.location.href = offsetURL(suggestion.path);
        });
        // remove inline display style on autocompleter (we want to
        // manage responsive display via css)
        $('.algolia-autocomplete').css("display", "");
      })
      .catch(function(error) {
        console.log(error);
      });

  });

  </script>

  <style type="text/css">

  .nav-search {
    font-size: x-small;
  }

  /* Algolioa Autocomplete */

  .algolia-autocomplete {
    display: inline-block;
    margin-left: 10px;
    vertical-align: sub;
    background-color: white;
    color: black;
    padding: 6px;
    padding-top: 8px;
    padding-bottom: 0;
    border-radius: 6px;
    border: 1px #0F2E3D solid;
    width: 180px;
  }


  @media screen and (max-width: 768px) {
    .distill-site-nav .algolia-autocomplete {
      display: none;
      visibility: hidden;
    }
    .distill-site-nav.responsive .algolia-autocomplete {
      display: inline-block;
      visibility: visible;
    }
    .distill-site-nav.responsive .algolia-autocomplete .aa-dropdown-menu {
      margin-left: 0;
      width: 400px;
      max-height: 400px;
    }
  }

  .algolia-autocomplete .aa-input, .algolia-autocomplete .aa-hint {
    width: 90%;
    outline: none;
    border: none;
  }

  .algolia-autocomplete .aa-hint {
    color: #999;
  }
  .algolia-autocomplete .aa-dropdown-menu {
    width: 550px;
    max-height: 70vh;
    overflow-x: visible;
    overflow-y: scroll;
    padding: 5px;
    margin-top: 3px;
    margin-left: -150px;
    background-color: #fff;
    border-radius: 5px;
    border: 1px solid #999;
    border-top: none;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion {
    cursor: pointer;
    padding: 5px 4px;
    border-bottom: 1px solid #eee;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion:last-of-type {
    border-bottom: none;
    margin-bottom: 2px;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item {
    overflow: hidden;
    font-size: 0.8em;
    line-height: 1.4em;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item h3 {
    font-size: 1rem;
    margin-block-start: 0;
    margin-block-end: 5px;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-description {
    display: inline-block;
    overflow: hidden;
    height: 2.8em;
    width: 80%;
    margin-right: 4%;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview {
    display: inline-block;
    width: 15%;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview img {
    height: 3em;
    width: auto;
    display: none;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview img[src] {
    display: initial;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion.aa-cursor {
    background-color: #eee;
  }
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion em {
    font-weight: bold;
    font-style: normal;
  }

  </style>


  <!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->

  <style type="text/css">

  body {
    background-color: white;
  }

  .pandoc-table {
    width: 100%;
  }

  .pandoc-table>caption {
    margin-bottom: 10px;
  }

  .pandoc-table th:not([align]) {
    text-align: left;
  }

  .pagedtable-footer {
    font-size: 15px;
  }

  d-byline .byline {
    grid-template-columns: 2fr 2fr;
  }

  d-byline .byline h3 {
    margin-block-start: 1.5em;
  }

  d-byline .byline .authors-affiliations h3 {
    margin-block-start: 0.5em;
  }

  .authors-affiliations .orcid-id {
    width: 16px;
    height:16px;
    margin-left: 4px;
    margin-right: 4px;
    vertical-align: middle;
    padding-bottom: 2px;
  }

  d-title .dt-tags {
    margin-top: 1em;
    grid-column: text;
  }

  .dt-tags .dt-tag {
    text-decoration: none;
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0em 0.4em;
    margin-right: 0.5em;
    margin-bottom: 0.4em;
    font-size: 70%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  d-article table.gt_table td,
  d-article table.gt_table th {
    border-bottom: none;
  }

  .html-widget {
    margin-bottom: 2.0em;
  }

  .l-screen-inset {
    padding-right: 16px;
  }

  .l-screen .caption {
    margin-left: 10px;
  }

  .shaded {
    background: rgb(247, 247, 247);
    padding-top: 20px;
    padding-bottom: 20px;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .html-widget {
    margin-bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .shaded-content {
    background: white;
  }

  .text-output {
    margin-top: 0;
    line-height: 1.5em;
  }

  .hidden {
    display: none !important;
  }

  d-article {
    padding-top: 2.5rem;
    padding-bottom: 30px;
  }

  d-appendix {
    padding-top: 30px;
  }

  d-article>p>img {
    width: 100%;
  }

  d-article h2 {
    margin: 1rem 0 1.5rem 0;
  }

  d-article h3 {
    margin-top: 1.5rem;
  }

  d-article iframe {
    border: 1px solid rgba(0, 0, 0, 0.1);
    margin-bottom: 2.0em;
    width: 100%;
  }

  /* Tweak code blocks */

  d-article div.sourceCode code,
  d-article pre code {
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
  }

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: auto;
  }

  d-article div.sourceCode {
    background-color: white;
  }

  d-article div.sourceCode pre {
    padding-left: 10px;
    font-size: 12px;
    border-left: 2px solid rgba(0,0,0,0.1);
  }

  d-article pre {
    font-size: 12px;
    color: black;
    background: none;
    margin-top: 0;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  d-article pre a {
    border-bottom: none;
  }

  d-article pre a:hover {
    border-bottom: none;
    text-decoration: underline;
  }

  d-article details {
    grid-column: text;
    margin-bottom: 0.8em;
  }

  @media(min-width: 768px) {

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: visible !important;
  }

  d-article div.sourceCode pre {
    padding-left: 18px;
    font-size: 14px;
  }

  d-article pre {
    font-size: 14px;
  }

  }

  figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  /* CSS for d-contents */

  .d-contents {
    grid-column: text;
    color: rgba(0,0,0,0.8);
    font-size: 0.9em;
    padding-bottom: 1em;
    margin-bottom: 1em;
    padding-bottom: 0.5em;
    margin-bottom: 1em;
    padding-left: 0.25em;
    justify-self: start;
  }

  @media(min-width: 1000px) {
    .d-contents.d-contents-float {
      height: 0;
      grid-column-start: 1;
      grid-column-end: 4;
      justify-self: center;
      padding-right: 3em;
      padding-left: 2em;
    }
  }

  .d-contents nav h3 {
    font-size: 18px;
    margin-top: 0;
    margin-bottom: 1em;
  }

  .d-contents li {
    list-style-type: none
  }

  .d-contents nav > ul {
    padding-left: 0;
  }

  .d-contents ul {
    padding-left: 1em
  }

  .d-contents nav ul li {
    margin-top: 0.6em;
    margin-bottom: 0.2em;
  }

  .d-contents nav a {
    font-size: 13px;
    border-bottom: none;
    text-decoration: none
    color: rgba(0, 0, 0, 0.8);
  }

  .d-contents nav a:hover {
    text-decoration: underline solid rgba(0, 0, 0, 0.6)
  }

  .d-contents nav > ul > li > a {
    font-weight: 600;
  }

  .d-contents nav > ul > li > ul {
    font-weight: inherit;
  }

  .d-contents nav > ul > li > ul > li {
    margin-top: 0.2em;
  }


  .d-contents nav ul {
    margin-top: 0;
    margin-bottom: 0.25em;
  }

  .d-article-with-toc h2:nth-child(2) {
    margin-top: 0;
  }


  /* Figure */

  .figure {
    position: relative;
    margin-bottom: 2.5em;
    margin-top: 1.5em;
  }

  .figure img {
    width: 100%;
  }

  .figure .caption {
    color: rgba(0, 0, 0, 0.6);
    font-size: 12px;
    line-height: 1.5em;
  }

  .figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  .figure .caption a {
    color: rgba(0, 0, 0, 0.6);
  }

  .figure .caption b,
  .figure .caption strong, {
    font-weight: 600;
    color: rgba(0, 0, 0, 1.0);
  }

  /* Citations */

  d-article .citation {
    color: inherit;
    cursor: inherit;
  }

  div.hanging-indent{
    margin-left: 1em; text-indent: -1em;
  }

  /* Citation hover box */

  .tippy-box[data-theme~=light-border] {
    background-color: rgba(250, 250, 250, 0.95);
  }

  .tippy-content > p {
    margin-bottom: 0;
    padding: 2px;
  }


  /* Tweak 1000px media break to show more text */

  @media(min-width: 1000px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 16px;
    }

    .grid {
      grid-column-gap: 16px;
    }

    d-article {
      font-size: 1.06rem;
      line-height: 1.7em;
    }
    figure .caption, .figure .caption, figure figcaption {
      font-size: 13px;
    }
  }

  @media(min-width: 1180px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 32px;
    }

    .grid {
      grid-column-gap: 32px;
    }
  }


  /* Get the citation styles for the appendix (not auto-injected on render since
     we do our own rendering of the citation appendix) */

  d-appendix .citation-appendix,
  .d-appendix .citation-appendix {
    font-size: 11px;
    line-height: 15px;
    border-left: 1px solid rgba(0, 0, 0, 0.1);
    padding-left: 18px;
    border: 1px solid rgba(0,0,0,0.1);
    background: rgba(0, 0, 0, 0.02);
    padding: 10px 18px;
    border-radius: 3px;
    color: rgba(150, 150, 150, 1);
    overflow: hidden;
    margin-top: -12px;
    white-space: pre-wrap;
    word-wrap: break-word;
  }

  /* Include appendix styles here so they can be overridden */

  d-appendix {
    contain: layout style;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-top: 60px;
    margin-bottom: 0;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    color: rgba(0,0,0,0.5);
    padding-top: 60px;
    padding-bottom: 48px;
  }

  d-appendix h3 {
    grid-column: page-start / text-start;
    font-size: 15px;
    font-weight: 500;
    margin-top: 1em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.65);
  }

  d-appendix h3 + * {
    margin-top: 1em;
  }

  d-appendix ol {
    padding: 0 0 0 15px;
  }

  @media (min-width: 768px) {
    d-appendix ol {
      padding: 0 0 0 30px;
      margin-left: -30px;
    }
  }

  d-appendix li {
    margin-bottom: 1em;
  }

  d-appendix a {
    color: rgba(0, 0, 0, 0.6);
  }

  d-appendix > * {
    grid-column: text;
  }

  d-appendix > d-footnote-list,
  d-appendix > d-citation-list,
  d-appendix > distill-appendix {
    grid-column: screen;
  }

  /* Include footnote styles here so they can be overridden */

  d-footnote-list {
    contain: layout style;
  }

  d-footnote-list > * {
    grid-column: text;
  }

  d-footnote-list a.footnote-backlink {
    color: rgba(0,0,0,0.3);
    padding-left: 0.5em;
  }



  /* Anchor.js */

  .anchorjs-link {
    /*transition: all .25s linear; */
    text-decoration: none;
    border-bottom: none;
  }
  *:hover > .anchorjs-link {
    margin-left: -1.125em !important;
    text-decoration: none;
    border-bottom: none;
  }

  /* Social footer */

  .social_footer {
    margin-top: 30px;
    margin-bottom: 0;
    color: rgba(0,0,0,0.67);
  }

  .disqus-comments {
    margin-right: 30px;
  }

  .disqus-comment-count {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    cursor: pointer;
  }

  #disqus_thread {
    margin-top: 30px;
  }

  .article-sharing a {
    border-bottom: none;
    margin-right: 8px;
  }

  .article-sharing a:hover {
    border-bottom: none;
  }

  .sidebar-section.subscribe {
    font-size: 12px;
    line-height: 1.6em;
  }

  .subscribe p {
    margin-bottom: 0.5em;
  }


  .article-footer .subscribe {
    font-size: 15px;
    margin-top: 45px;
  }


  .sidebar-section.custom {
    font-size: 12px;
    line-height: 1.6em;
  }

  .custom p {
    margin-bottom: 0.5em;
  }

  /* Styles for listing layout (hide title) */
  .layout-listing d-title, .layout-listing .d-title {
    display: none;
  }

  /* Styles for posts lists (not auto-injected) */


  .posts-with-sidebar {
    padding-left: 45px;
    padding-right: 45px;
  }

  .posts-list .description h2,
  .posts-list .description p {
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
  }

  .posts-list .description h2 {
    font-weight: 700;
    border-bottom: none;
    padding-bottom: 0;
  }

  .posts-list h2.post-tag {
    border-bottom: 1px solid rgba(0, 0, 0, 0.2);
    padding-bottom: 12px;
  }
  .posts-list {
    margin-top: 60px;
    margin-bottom: 24px;
  }

  .posts-list .post-preview {
    text-decoration: none;
    overflow: hidden;
    display: block;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
    padding: 24px 0;
  }

  .post-preview-last {
    border-bottom: none !important;
  }

  .posts-list .posts-list-caption {
    grid-column: screen;
    font-weight: 400;
  }

  .posts-list .post-preview h2 {
    margin: 0 0 6px 0;
    line-height: 1.2em;
    font-style: normal;
    font-size: 24px;
  }

  .posts-list .post-preview p {
    margin: 0 0 12px 0;
    line-height: 1.4em;
    font-size: 16px;
  }

  .posts-list .post-preview .thumbnail {
    box-sizing: border-box;
    margin-bottom: 24px;
    position: relative;
    max-width: 500px;
  }
  .posts-list .post-preview img {
    width: 100%;
    display: block;
  }

  .posts-list .metadata {
    font-size: 12px;
    line-height: 1.4em;
    margin-bottom: 18px;
  }

  .posts-list .metadata > * {
    display: inline-block;
  }

  .posts-list .metadata .publishedDate {
    margin-right: 2em;
  }

  .posts-list .metadata .dt-authors {
    display: block;
    margin-top: 0.3em;
    margin-right: 2em;
  }

  .posts-list .dt-tags {
    display: block;
    line-height: 1em;
  }

  .posts-list .dt-tags .dt-tag {
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0.3em 0.4em;
    margin-right: 0.2em;
    margin-bottom: 0.4em;
    font-size: 60%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  .posts-list img {
    opacity: 1;
  }

  .posts-list img[data-src] {
    opacity: 0;
  }

  .posts-more {
    clear: both;
  }


  .posts-sidebar {
    font-size: 16px;
  }

  .posts-sidebar h3 {
    font-size: 16px;
    margin-top: 0;
    margin-bottom: 0.5em;
    font-weight: 400;
    text-transform: uppercase;
  }

  .sidebar-section {
    margin-bottom: 30px;
  }

  .categories ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
  }

  .categories li {
    color: rgba(0, 0, 0, 0.8);
    margin-bottom: 0;
  }

  .categories li>a {
    border-bottom: none;
  }

  .categories li>a:hover {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
  }

  .categories .active {
    font-weight: 600;
  }

  .categories .category-count {
    color: rgba(0, 0, 0, 0.4);
  }


  @media(min-width: 768px) {
    .posts-list .post-preview h2 {
      font-size: 26px;
    }
    .posts-list .post-preview .thumbnail {
      float: right;
      width: 30%;
      margin-bottom: 0;
    }
    .posts-list .post-preview .description {
      float: left;
      width: 45%;
    }
    .posts-list .post-preview .metadata {
      float: left;
      width: 20%;
      margin-top: 8px;
    }
    .posts-list .post-preview p {
      margin: 0 0 12px 0;
      line-height: 1.5em;
      font-size: 16px;
    }
    .posts-with-sidebar .posts-list {
      float: left;
      width: 75%;
    }
    .posts-with-sidebar .posts-sidebar {
      float: right;
      width: 20%;
      margin-top: 60px;
      padding-top: 24px;
      padding-bottom: 24px;
    }
  }


  /* Improve display for browsers without grid (IE/Edge <= 15) */

  .downlevel {
    line-height: 1.6em;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
    margin: 0;
  }

  .downlevel .d-title {
    padding-top: 6rem;
    padding-bottom: 1.5rem;
  }

  .downlevel .d-title h1 {
    font-size: 50px;
    font-weight: 700;
    line-height: 1.1em;
    margin: 0 0 0.5rem;
  }

  .downlevel .d-title p {
    font-weight: 300;
    font-size: 1.2rem;
    line-height: 1.55em;
    margin-top: 0;
  }

  .downlevel .d-byline {
    padding-top: 0.8em;
    padding-bottom: 0.8em;
    font-size: 0.8rem;
    line-height: 1.8em;
  }

  .downlevel .section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
  }

  .downlevel .d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
    padding-top: 1rem;
    padding-bottom: 2rem;
  }


  .downlevel .d-appendix {
    padding-left: 0;
    padding-right: 0;
    max-width: none;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.5);
    padding-top: 40px;
    padding-bottom: 48px;
  }

  .downlevel .footnotes ol {
    padding-left: 13px;
  }

  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
    padding-left: 40px;
    padding-right: 40px;
  }

  @media(min-width: 768px) {
    .downlevel .base-grid,
    .downlevel .distill-header,
    .downlevel .d-title,
    .downlevel .d-abstract,
    .downlevel .d-article,
    .downlevel .d-appendix,
    .downlevel .distill-appendix,
    .downlevel .d-byline,
    .downlevel .d-footnote-list,
    .downlevel .d-citation-list,
    .downlevel .distill-footer,
    .downlevel .appendix-bottom,
    .downlevel .posts-container {
    padding-left: 150px;
    padding-right: 150px;
    max-width: 900px;
  }
  }

  .downlevel pre code {
    display: block;
    border-left: 2px solid rgba(0, 0, 0, .1);
    padding: 0 0 0 20px;
    font-size: 14px;
  }

  .downlevel code, .downlevel pre {
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  .downlevel .posts-list .post-preview {
    color: inherit;
  }



  </style>

  <script type="application/javascript">

  function is_downlevel_browser() {
    if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                   window.navigator.userAgent)) {
      return true;
    } else {
      return window.load_distill_framework === undefined;
    }
  }

  // show body when load is complete
  function on_load_complete() {

    // add anchors
    if (window.anchors) {
      window.anchors.options.placement = 'left';
      window.anchors.add('d-article > h2, d-article > h3, d-article > h4, d-article > h5');
    }


    // set body to visible
    document.body.style.visibility = 'visible';

    // force redraw for leaflet widgets
    if (window.HTMLWidgets) {
      var maps = window.HTMLWidgets.findAll(".leaflet");
      $.each(maps, function(i, el) {
        var map = this.getMap();
        map.invalidateSize();
        map.eachLayer(function(layer) {
          if (layer instanceof L.TileLayer)
            layer.redraw();
        });
      });
    }

    // trigger 'shown' so htmlwidgets resize
    $('d-article').trigger('shown');
  }

  function init_distill() {

    init_common();

    // create front matter
    var front_matter = $('<d-front-matter></d-front-matter>');
    $('#distill-front-matter').wrap(front_matter);

    // create d-title
    $('.d-title').changeElementType('d-title');

    // create d-byline
    var byline = $('<d-byline></d-byline>');
    $('.d-byline').replaceWith(byline);

    // create d-article
    var article = $('<d-article></d-article>');
    $('.d-article').wrap(article).children().unwrap();

    // move posts container into article
    $('.posts-container').appendTo($('d-article'));

    // create d-appendix
    $('.d-appendix').changeElementType('d-appendix');

    // flag indicating that we have appendix items
    var appendix = $('.appendix-bottom').children('h3').length > 0;

    // replace footnotes with <d-footnote>
    $('.footnote-ref').each(function(i, val) {
      appendix = true;
      var href = $(this).attr('href');
      var id = href.replace('#', '');
      var fn = $('#' + id);
      var fn_p = $('#' + id + '>p');
      fn_p.find('.footnote-back').remove();
      var text = fn_p.html();
      var dtfn = $('<d-footnote></d-footnote>');
      dtfn.html(text);
      $(this).replaceWith(dtfn);
    });
    // remove footnotes
    $('.footnotes').remove();

    // move refs into #references-listing
    $('#references-listing').replaceWith($('#refs'));

    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      var id = $(this).attr('id');
      $('.d-contents a[href="#' + id + '"]').parent().remove();
      appendix = true;
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
    });

    // show d-appendix if we have appendix content
    $("d-appendix").css('display', appendix ? 'grid' : 'none');

    // localize layout chunks to just output
    $('.layout-chunk').each(function(i, val) {

      // capture layout
      var layout = $(this).attr('data-layout');

      // apply layout to markdown level block elements
      var elements = $(this).children().not('details, div.sourceCode, pre, script');
      elements.each(function(i, el) {
        var layout_div = $('<div class="' + layout + '"></div>');
        if (layout_div.hasClass('shaded')) {
          var shaded_content = $('<div class="shaded-content"></div>');
          $(this).wrap(shaded_content);
          $(this).parent().wrap(layout_div);
        } else {
          $(this).wrap(layout_div);
        }
      });


      // unwrap the layout-chunk div
      $(this).children().unwrap();
    });

    // remove code block used to force  highlighting css
    $('.distill-force-highlighting-css').parent().remove();

    // remove empty line numbers inserted by pandoc when using a
    // custom syntax highlighting theme
    $('code.sourceCode a:empty').remove();

    // load distill framework
    load_distill_framework();

    // wait for window.distillRunlevel == 4 to do post processing
    function distill_post_process() {

      if (!window.distillRunlevel || window.distillRunlevel < 4)
        return;

      // hide author/affiliations entirely if we have no authors
      var front_matter = JSON.parse($("#distill-front-matter").html());
      var have_authors = front_matter.authors && front_matter.authors.length > 0;
      if (!have_authors)
        $('d-byline').addClass('hidden');

      // article with toc class
      $('.d-contents').parent().addClass('d-article-with-toc');

      // strip links that point to #
      $('.authors-affiliations').find('a[href="#"]').removeAttr('href');

      // add orcid ids
      $('.authors-affiliations').find('.author').each(function(i, el) {
        var orcid_id = front_matter.authors[i].orcidID;
        if (orcid_id) {
          var a = $('<a></a>');
          a.attr('href', 'https://orcid.org/' + orcid_id);
          var img = $('<img></img>');
          img.addClass('orcid-id');
          img.attr('alt', 'ORCID ID');
          img.attr('src','data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==');
          a.append(img);
          $(this).append(a);
        }
      });

      // hide elements of author/affiliations grid that have no value
      function hide_byline_column(caption) {
        $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
      }

      // affiliations
      var have_affiliations = false;
      for (var i = 0; i<front_matter.authors.length; ++i) {
        var author = front_matter.authors[i];
        if (author.affiliation !== "&nbsp;") {
          have_affiliations = true;
          break;
        }
      }
      if (!have_affiliations)
        $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');

      // published date
      if (!front_matter.publishedDate)
        hide_byline_column("Published");

      // document object identifier
      var doi = $('d-byline').find('h3:contains("DOI")');
      var doi_p = doi.next().empty();
      if (!front_matter.doi) {
        // if we have a citation and valid citationText then link to that
        if ($('#citation').length > 0 && front_matter.citationText) {
          doi.html('Citation');
          $('<a href="#citation"></a>')
            .text(front_matter.citationText)
            .appendTo(doi_p);
        } else {
          hide_byline_column("DOI");
        }
      } else {
        $('<a></a>')
           .attr('href', "https://doi.org/" + front_matter.doi)
           .html(front_matter.doi)
           .appendTo(doi_p);
      }

       // change plural form of authors/affiliations
      if (front_matter.authors.length === 1) {
        var grid = $('.authors-affiliations');
        grid.children('h3:contains("Authors")').text('Author');
        grid.children('h3:contains("Affiliations")').text('Affiliation');
      }

      // remove d-appendix and d-footnote-list local styles
      $('d-appendix > style:first-child').remove();
      $('d-footnote-list > style:first-child').remove();

      // move appendix-bottom entries to the bottom
      $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
      $('.appendix-bottom').remove();

      // hoverable references
      $('span.citation[data-cites]').each(function() {
        var refHtml = $('#ref-' + $(this).attr('data-cites')).html();
        window.tippy(this, {
          allowHTML: true,
          content: refHtml,
          maxWidth: 500,
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start'
        });
      });

      // clear polling timer
      clearInterval(tid);

      // show body now that everything is ready
      on_load_complete();
    }

    var tid = setInterval(distill_post_process, 50);
    distill_post_process();

  }

  function init_downlevel() {

    init_common();

     // insert hr after d-title
    $('.d-title').after($('<hr class="section-separator"/>'));

    // check if we have authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;

    // manage byline/border
    if (!have_authors)
      $('.d-byline').remove();
    $('.d-byline').after($('<hr class="section-separator"/>'));
    $('.d-byline a').remove();

    // remove toc
    $('.d-contents').remove();

    // move appendix elements
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
    });


    // inject headers into references and footnotes
    var refs_header = $('<h3></h3>');
    refs_header.text('References');
    $('#refs').prepend(refs_header);

    var footnotes_header = $('<h3></h3');
    footnotes_header.text('Footnotes');
    $('.footnotes').children('hr').first().replaceWith(footnotes_header);

    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
    $('.appendix-bottom').remove();

    // remove appendix if it's empty
    if ($('.d-appendix').children().length === 0)
      $('.d-appendix').remove();

    // prepend separator above appendix
    $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));

    // trim code
    $('pre>code').each(function(i, val) {
      $(this).html($.trim($(this).html()));
    });

    // move posts-container right before article
    $('.posts-container').insertBefore($('.d-article'));

    $('body').addClass('downlevel');

    on_load_complete();
  }


  function init_common() {

    // jquery plugin to change element types
    (function($) {
      $.fn.changeElementType = function(newType) {
        var attrs = {};

        $.each(this[0].attributes, function(idx, attr) {
          attrs[attr.nodeName] = attr.nodeValue;
        });

        this.replaceWith(function() {
          return $("<" + newType + "/>", attrs).append($(this).contents());
        });
      };
    })(jQuery);

    // prevent underline for linked images
    $('a > img').parent().css({'border-bottom' : 'none'});

    // mark non-body figures created by knitr chunks as 100% width
    $('.layout-chunk').each(function(i, val) {
      var figures = $(this).find('img, .html-widget');
      if ($(this).attr('data-layout') !== "l-body") {
        figures.css('width', '100%');
      } else {
        figures.css('max-width', '100%');
        figures.filter("[width]").each(function(i, val) {
          var fig = $(this);
          fig.css('width', fig.attr('width') + 'px');
        });

      }
    });

    // auto-append index.html to post-preview links in file: protocol
    // and in rstudio ide preview
    $('.post-preview').each(function(i, val) {
      if (window.location.protocol === "file:")
        $(this).attr('href', $(this).attr('href') + "index.html");
    });

    // get rid of index.html references in header
    if (window.location.protocol !== "file:") {
      $('.distill-site-header a[href]').each(function(i,val) {
        $(this).attr('href', $(this).attr('href').replace("index.html", "./"));
      });
    }

    // add class to pandoc style tables
    $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
    $('.kable-table').children('table').addClass('pandoc-table');

    // add figcaption style to table captions
    $('caption').parent('table').addClass("figcaption");

    // initialize posts list
    if (window.init_posts_list)
      window.init_posts_list();

    // implmement disqus comment link
    $('.disqus-comment-count').click(function() {
      window.headroom_prevent_pin = true;
      $('#disqus_thread').toggleClass('hidden');
      if (!$('#disqus_thread').hasClass('hidden')) {
        var offset = $(this).offset();
        $(window).resize();
        $('html, body').animate({
          scrollTop: offset.top - 35
        });
      }
    });
  }

  document.addEventListener('DOMContentLoaded', function() {
    if (is_downlevel_browser())
      init_downlevel();
    else
      window.addEventListener('WebComponentsReady', init_distill);
  });

  </script>

  <!--/radix_placeholder_distill-->
  <script src="site_libs/header-attrs-2.7/header-attrs.js"></script>
  <script src="site_libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
  <script src="site_libs/jquery-3.5.1/jquery.min.js"></script>
  <link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
  <script src="site_libs/datatables-binding-0.17/datatables.js"></script>
  <link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
  <link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
  <script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
  <link href="site_libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
  <script src="site_libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
  <script src="site_libs/popper-2.6.0/popper.min.js"></script>
  <link href="site_libs/tippy-6.2.7/tippy.css" rel="stylesheet" />
  <link href="site_libs/tippy-6.2.7/tippy-light-border.css" rel="stylesheet" />
  <script src="site_libs/tippy-6.2.7/tippy.umd.min.js"></script>
  <script src="site_libs/anchor-4.2.2/anchor.min.js"></script>
  <script src="site_libs/bowser-1.9.3/bowser.min.js"></script>
  <script src="site_libs/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="site_libs/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
  <!--/radix_placeholder_site_in_header-->


</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"Multi-sample comparisons of the Kinkel (C086) Smchd1 LSK mini-bulk data set","description":"Read-based analysis","authors":[{"author":"Peter Hickey","authorURL":"https://peterhickey.org","affiliation":"WEHI SCORE","affiliationURL":"https://www.wehi.edu.au/people/shalin-naik/3310/score","orcidID":""}],"publishedDate":"2021-10-06T00:00:00.000+11:00","citationText":"Hickey, 2021"}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<header class="header header--fixed" role="banner">
<nav class="distill-site-nav distill-site-header">
<div class="nav-left">
<a href="index.html" class="title">C086</a>
<a href="index.html">Introduction</a>
<a href="C086_Kinkel.preprocess.html">Preprocessing</a>
<a href="C086_Kinkel.multi-sample_comparisons.html">Differential analyses</a>
<div class="nav-dropdown">
<button class="nav-dropbtn">
Supplementary files
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<a href="../output/scPipe/">scPipe reports</a>
<a href="../output/DEGs/">Differentially expressed genes</a>
</div>
</div>
</div>
<div class="nav-right">
<a href="https://github.com/WEHISCORE/C086_Kinkel">
<i class="fa fa-github" aria-hidden="true"></i>
</a>
<a href="javascript:void(0);" class="nav-toggle">&#9776;</a>
</div>
</nav>
</header>
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>Multi-sample comparisons of the Kinkel (C086) Smchd1 LSK mini-bulk data set</h1>
<!--radix_placeholder_categories-->
<!--/radix_placeholder_categories-->
<p>Read-based analysis</p>
</div>

<div class="d-byline">
  Peter Hickey <a href="https://peterhickey.org" class="uri">https://peterhickey.org</a> (WEHI SCORE)<a href="https://www.wehi.edu.au/people/shalin-naik/3310/score" class="uri">https://www.wehi.edu.au/people/shalin-naik/3310/score</a>
  
<br/>2021-10-06
</div>

<div class="d-article">
<div class="d-contents d-contents-float">
<nav class="l-text toc figcaption" id="TOC">
<h3>Contents</h3>
<ul>
<li><a href="#summary">Summary</a></li>
<li><a href="#setup">Setup</a>
<ul>
<li><a href="#load-data">Load data</a></li>
<li><a href="#gene-filtering">Gene filtering</a></li>
<li><a href="#sample-filtering">Sample filtering</a></li>
<li><a href="#create-dgelist-objects">Create DGEList objects</a>
<ul>
<li><a href="#gene-filtering-1">Gene filtering</a></li>
<li><a href="#normalization">Normalization</a></li>
</ul></li>
<li><a href="#aggregation-of-technical-replicates">Aggregation of technical replicates</a></li>
<li><a href="#further-normalization">Further normalization</a></li>
<li><a href="#gene-sets">Gene sets</a></li>
</ul></li>
<li><a href="#analysis-of-aggregated-technical-replicates">Analysis of aggregated technical replicates</a>
<ul>
<li><a href="#mds">MDS</a></li>
<li><a href="#de-analysis">DE analysis</a></li>
<li><a href="#gene-set-tests">Gene set tests</a>
<ul>
<li><a href="#goana">goana</a></li>
<li><a href="#kegga">kegga</a></li>
<li><a href="#camera-with-msigdb-gene-sets">camera with MSigDB gene sets</a></li>
<li><a href="#sarahs-gene-lists">Sarah’s gene lists</a></li>
</ul></li>
</ul></li>
<li><a href="#male-only-analysis-of-aggregated-technical-replicates">Male-only: analysis of aggregated technical replicates</a>
<ul>
<li><a href="#mds-1">MDS</a></li>
<li><a href="#de-analysis-1">DE analysis</a></li>
<li><a href="#gene-set-tests-1">Gene set tests</a>
<ul>
<li><a href="#goana-1">goana</a></li>
<li><a href="#kegga-1">kegga</a></li>
<li><a href="#camera-with-msigdb-gene-sets-1">camera with MSigDB gene sets</a></li>
<li><a href="#sarahs-gene-lists-1">Sarah’s gene lists</a></li>
</ul></li>
</ul></li>
<li><a href="#female-only-analysis-of-aggregated-technical-replicates">Female-only: analysis of aggregated technical replicates</a>
<ul>
<li><a href="#mds-2">MDS</a></li>
<li><a href="#de-analysis-2">DE analysis</a></li>
<li><a href="#gene-set-tests-2">Gene set tests</a>
<ul>
<li><a href="#goana-2">goana</a></li>
<li><a href="#kegga-2">kegga</a></li>
<li><a href="#camera-with-msigdb-gene-sets-2">camera with MSigDB gene sets</a></li>
<li><a href="#sarahs-gene-lists-2">Sarah’s gene lists</a></li>
</ul></li>
</ul></li>
<li><a href="#session-info">Session info</a></li>
</ul>
</nav>
</div>
<h1 id="summary">Summary</h1>
<p>This report analyses the read counts of the <code>Del</code> and <code>Het</code> samples only. The report begins with some general <a href="#setup">Setup</a> and is followed by various analyses denoted by a prefix (<code>y</code>, <code>ym</code>, <code>yf</code>):</p>
<ul>
<li><code>y</code>: <a href="#analysis-of-aggregated-technical-replicates">Analysis of aggregated technical replicates</a> with Remove Unwanted Variation (RUV) normalization</li>
<li><code>ym</code>: <a href="#male-only-analysis-of-aggregated-technical-replicates">Male-only: analysis of aggregated technical replicates</a> using RUV normalization</li>
<li><code>yf</code>: <a href="#female-only-analysis-of-aggregated-technical-replicates">Female-only: analysis of aggregated technical replicates</a> using RUV normalization</li>
</ul>
<h1 id="setup">Setup</h1>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'>SingleCellExperiment</span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://here.r-lib.org/'>here</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='http://bioinf.wehi.edu.au/edgeR'>edgeR</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://ggplot2.tidyverse.org'>ggplot2</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://wilkelab.org/cowplot/'>cowplot</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://patchwork.data-imaginist.com'>patchwork</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/slowkow/ggrepel'>ggrepel</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://magrittr.tidyverse.org'>magrittr</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/hasaru-k/GlimmaV2'>Glimma</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/drisso/EDASeq'>EDASeq</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/drisso/RUVSeq'>RUVSeq</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'>pheatmap</span><span class='op'>)</span>

<span class='kw'><a href='https://rdrr.io/r/base/source.html'>source</a></span><span class='op'>(</span><span class='fu'><a href='https://here.r-lib.org//reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"code/helper_functions.R"</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<h2 id="load-data">Load data</h2>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>sce</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/readRDS.html'>readRDS</a></span><span class='op'>(</span><span class='fu'><a href='https://here.r-lib.org//reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"data"</span>, <span class='st'>"SCEs"</span>, <span class='st'>"C086_Kinkel.preprocessed.SCE.rds"</span><span class='op'>)</span><span class='op'>)</span>
<span class='co'># NOTE: Reorder samples for plots (e.g., RLE plots).</span>
<span class='va'>sce</span> <span class='op'>&lt;-</span> <span class='va'>sce</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/order.html'>order</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>smchd1_genotype_updated</span>, <span class='va'>sce</span><span class='op'>$</span><span class='va'>sex</span>, <span class='va'>sce</span><span class='op'>$</span><span class='va'>mouse_number</span><span class='op'>)</span><span class='op'>]</span>

<span class='co'># Restrict analysis to read counts</span>
<span class='fu'>assays</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>counts <span class='op'>=</span> <span class='fu'>assay</span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"read_counts"</span><span class='op'>)</span><span class='op'>)</span>

<span class='va'>outdir</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://here.r-lib.org//reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"output"</span>, <span class='st'>"DEGs"</span>, <span class='st'>"reads"</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/files2.html'>dir.create</a></span><span class='op'>(</span><span class='va'>outdir</span>, recursive <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<h2 id="gene-filtering">Gene filtering</h2>
<p>We filter the genes to only retain protein coding genes.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>sce</span> <span class='op'>&lt;-</span> <span class='va'>sce</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/any.html'>any</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/grep.html'>grepl</a></span><span class='op'>(</span><span class='st'>"protein_coding"</span>, <span class='fu'>rowData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>$</span><span class='va'>ENSEMBL.GENEBIOTYPE</span><span class='op'>)</span><span class='op'>)</span>, <span class='op'>]</span>
</code></pre>
</div>
</details>
</div>
<h2 id="sample-filtering">Sample filtering</h2>
<p>We filter out the <code>WT</code> samples as these are not useful for DE analysis since there is only 1 biological replicate.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>sce</span> <span class='op'>&lt;-</span> <span class='va'>sce</span><span class='op'>[</span>, <span class='va'>sce</span><span class='op'>$</span><span class='va'>smchd1_genotype_updated</span> <span class='op'>!=</span> <span class='st'>"WT"</span><span class='op'>]</span>
</code></pre>
</div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/droplevels.html'>droplevels</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>)</span>
<span class='co'># Some useful colours</span>
<span class='va'>sex_colours</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/setNames.html'>setNames</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/base/unique.html'>unique</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>sex_colours</span><span class='op'>)</span>,
  <span class='fu'><a href='https://rdrr.io/r/base/unique.html'>unique</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>sex_colours</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>smchd1_genotype_updated_colours</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/setNames.html'>setNames</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/base/unique.html'>unique</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>smchd1_genotype_updated_colours</span><span class='op'>)</span>,
  <span class='fu'><a href='https://rdrr.io/r/base/unique.html'>unique</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>smchd1_genotype_updated_colours</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>mouse_number_colours</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/setNames.html'>setNames</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/base/unique.html'>unique</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>mouse_number_colours</span><span class='op'>)</span>,
  <span class='fu'><a href='https://rdrr.io/r/base/unique.html'>unique</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>mouse_number_colours</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<h2 id="create-dgelist-objects">Create DGEList objects</h2>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>x</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/DGEList.html'>DGEList</a></span><span class='op'>(</span>
  counts <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/matrix.html'>as.matrix</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/SingleCellExperiment/man/assays.html'>counts</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>)</span>,
  samples <span class='op'>=</span> <span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span>,
  group <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>smchd1_genotype_updated</span><span class='op'>)</span>,
  genes <span class='op'>=</span> <span class='fu'>flattenDF</span><span class='op'>(</span><span class='fu'>rowData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>x</span> <span class='op'>&lt;-</span> <span class='va'>x</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/colSums.html'>rowSums</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>$</span><span class='va'>counts</span><span class='op'>)</span> <span class='op'>&gt;</span> <span class='fl'>0</span>, <span class='op'>]</span>
</code></pre>
</div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>y</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/sumTechReps.html'>sumTechReps</a></span><span class='op'>(</span><span class='va'>x</span>, <span class='va'>x</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>genotype.mouse</span><span class='op'>)</span>
<span class='va'>ym</span> <span class='op'>&lt;-</span> <span class='va'>y</span><span class='op'>[</span>, <span class='va'>y</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>sex</span> <span class='op'>==</span> <span class='st'>"M"</span><span class='op'>]</span>
<span class='va'>yf</span> <span class='op'>&lt;-</span> <span class='va'>y</span><span class='op'>[</span>, <span class='va'>y</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>sex</span> <span class='op'>==</span> <span class='st'>"F"</span><span class='op'>]</span>
</code></pre>
</div>
</details>
</div>
<h3 id="gene-filtering-1">Gene filtering</h3>
<p>For each analysis, shown are the number of genes that have sufficient counts to test for DE (<code>TRUE</code>) or not (<code>FALSE</code>).</p>
<h4 id="y"><code>y</code></h4>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>y_keep_exprs</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/filterByExpr.html'>filterByExpr</a></span><span class='op'>(</span><span class='va'>y</span>, group <span class='op'>=</span> <span class='va'>y</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span>, min.count <span class='op'>=</span> <span class='fl'>5</span><span class='op'>)</span>
<span class='va'>y</span> <span class='op'>&lt;-</span> <span class='va'>y</span><span class='op'>[</span><span class='va'>y_keep_exprs</span>, , keep.lib.sizes <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>]</span>
<span class='fu'><a href='https://rdrr.io/r/base/table.html'>table</a></span><span class='op'>(</span><span class='va'>y_keep_exprs</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<pre><code>y_keep_exprs
FALSE  TRUE 
 3040 11450 </code></pre>
</div>
<h4 id="ym"><code>ym</code></h4>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>ym_keep_exprs</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/filterByExpr.html'>filterByExpr</a></span><span class='op'>(</span><span class='va'>ym</span>, group <span class='op'>=</span> <span class='va'>ym</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span>, min.count <span class='op'>=</span> <span class='fl'>5</span><span class='op'>)</span>
<span class='va'>ym</span> <span class='op'>&lt;-</span> <span class='va'>ym</span><span class='op'>[</span><span class='va'>ym_keep_exprs</span>, , keep.lib.sizes <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>]</span>
<span class='fu'><a href='https://rdrr.io/r/base/table.html'>table</a></span><span class='op'>(</span><span class='va'>ym_keep_exprs</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<pre><code>ym_keep_exprs
FALSE  TRUE 
 2891 11599 </code></pre>
</div>
<h4 id="yf"><code>yf</code></h4>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>yf_keep_exprs</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/filterByExpr.html'>filterByExpr</a></span><span class='op'>(</span><span class='va'>yf</span>, group <span class='op'>=</span> <span class='va'>yf</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span>, min.count <span class='op'>=</span> <span class='fl'>5</span><span class='op'>)</span>
<span class='va'>yf</span> <span class='op'>&lt;-</span> <span class='va'>yf</span><span class='op'>[</span><span class='va'>yf_keep_exprs</span>, , keep.lib.sizes <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>]</span>
<span class='fu'><a href='https://rdrr.io/r/base/table.html'>table</a></span><span class='op'>(</span><span class='va'>yf_keep_exprs</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<pre><code>yf_keep_exprs
FALSE  TRUE 
 2977 11513 </code></pre>
</div>
<h3 id="normalization">Normalization</h3>
<p>Normalization with upper-quartile (UQ) normalization.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>x</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/calcNormFactors.html'>calcNormFactors</a></span><span class='op'>(</span><span class='va'>x</span>, method <span class='op'>=</span> <span class='st'>"upperquartile"</span><span class='op'>)</span>
<span class='va'>y</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/calcNormFactors.html'>calcNormFactors</a></span><span class='op'>(</span><span class='va'>y</span>, method <span class='op'>=</span> <span class='st'>"upperquartile"</span><span class='op'>)</span>
<span class='va'>ym</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/calcNormFactors.html'>calcNormFactors</a></span><span class='op'>(</span><span class='va'>ym</span>, method <span class='op'>=</span> <span class='st'>"upperquartile"</span><span class='op'>)</span>
<span class='va'>yf</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/calcNormFactors.html'>calcNormFactors</a></span><span class='op'>(</span><span class='va'>yf</span>, method <span class='op'>=</span> <span class='st'>"upperquartile"</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<h2 id="aggregation-of-technical-replicates">Aggregation of technical replicates</h2>
<p>We have up to 3 technical replicates from each biological replicate. Ideally, the technical replicates are very similar to one another when view on an MDS plot. However, Figure <a href="#fig:mds">1</a> shows that:</p>
<ul>
<li>Some of the technical replicates are more variable than we would like</li>
<li>Those ‘outlier’ samples have smaller library sizes</li>
</ul>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>x_mds</span> <span class='op'>&lt;-</span> <span class='fu'>plotMDS</span><span class='op'>(</span><span class='va'>x</span>, plot <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>

<span class='va'>x_df</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cbind.html'>cbind</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x_mds</span><span class='op'>$</span><span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>x_mds</span><span class='op'>$</span><span class='va'>y</span><span class='op'>)</span>, <span class='va'>x</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>x_df</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span>

<span class='va'>p1</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span>, colour <span class='op'>=</span> <span class='va'>smchd1_genotype_updated</span>, label <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>x_df</span><span class='op'>)</span><span class='op'>)</span>,
  data <span class='op'>=</span> <span class='va'>x_df</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html'>geom_text_repel</a></span><span class='op'>(</span>size <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 1"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 2"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>smchd1_genotype_updated_colours</span><span class='op'>)</span>
<span class='va'>p2</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span>, colour <span class='op'>=</span> <span class='va'>sex</span>, label <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>x_df</span><span class='op'>)</span><span class='op'>)</span>,
  data <span class='op'>=</span> <span class='va'>x_df</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html'>geom_text_repel</a></span><span class='op'>(</span>size <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 1"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 2"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>sex_colours</span><span class='op'>)</span>
<span class='va'>p3</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span>, colour <span class='op'>=</span> <span class='va'>mouse_number</span>, label <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>x_df</span><span class='op'>)</span><span class='op'>)</span>,
  data <span class='op'>=</span> <span class='va'>x_df</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html'>geom_text_repel</a></span><span class='op'>(</span>size <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 1"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 2"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>mouse_number_colours</span><span class='op'>)</span>
<span class='va'>p4</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span>, colour <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log10</a></span><span class='op'>(</span><span class='va'>lib.size</span><span class='op'>)</span>, label <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>x_df</span><span class='op'>)</span><span class='op'>)</span>,
  data <span class='op'>=</span> <span class='va'>x_df</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html'>geom_text_repel</a></span><span class='op'>(</span>size <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_viridis.html'>scale_colour_viridis_c</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 1"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 2"</span><span class='op'>)</span>

<span class='va'>p1</span> <span class='op'>+</span> <span class='va'>p2</span> <span class='op'>+</span> <span class='va'>p3</span> <span class='op'>+</span> <span class='va'>p4</span> <span class='op'>+</span> <span class='fu'><a href='https://patchwork.data-imaginist.com/reference/plot_layout.html'>plot_layout</a></span><span class='op'>(</span>ncol <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:mds"></span>
<img src="C086_Kinkel.multi-sample_comparisons_files/figure-html5/mds-1.png" alt="MDS plots of individual technical replicates coloured by various experimental factors." width="624" />
<p class="caption">
Figure 1: MDS plots of individual technical replicates coloured by various experimental factors.
</p>
</div>
</div>
<p>Although we could analyse the data at the level of technical replicates, this is more complicated. It will make our life easier by aggregating these technical replicates so that we have 1 sample per biological replicate. To aggregate technical replicates we simply sum their counts.</p>
<p>Figure <a href="#fig:mds-y">2</a> shows the MDS plot after aggregating the technical replicates. We still observe that:</p>
<ul>
<li>Samples cluster more strongly by sex than by genotype</li>
<li>Some of the variation is associated with library size</li>
</ul>
<p>These observations (and others) prompt us to consider further normalization of the data.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>y_mds</span> <span class='op'>&lt;-</span> <span class='fu'>plotMDS</span><span class='op'>(</span><span class='va'>y</span>, plot <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>

<span class='va'>y_df</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cbind.html'>cbind</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>y_mds</span><span class='op'>$</span><span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y_mds</span><span class='op'>$</span><span class='va'>y</span><span class='op'>)</span>, <span class='va'>y</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>y_df</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>)</span>

<span class='va'>p1</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span>, colour <span class='op'>=</span> <span class='va'>smchd1_genotype_updated</span>, label <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>y_df</span><span class='op'>)</span><span class='op'>)</span>,
  data <span class='op'>=</span> <span class='va'>y_df</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html'>geom_text_repel</a></span><span class='op'>(</span>size <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 1"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 2"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>smchd1_genotype_updated_colours</span><span class='op'>)</span>
<span class='va'>p2</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span>, colour <span class='op'>=</span> <span class='va'>sex</span>, label <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>y_df</span><span class='op'>)</span><span class='op'>)</span>,
  data <span class='op'>=</span> <span class='va'>y_df</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html'>geom_text_repel</a></span><span class='op'>(</span>size <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 1"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 2"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>sex_colours</span><span class='op'>)</span>
<span class='va'>p3</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span>, colour <span class='op'>=</span> <span class='va'>mouse_number</span>, label <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>y_df</span><span class='op'>)</span><span class='op'>)</span>,
  data <span class='op'>=</span> <span class='va'>y_df</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html'>geom_text_repel</a></span><span class='op'>(</span>size <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 1"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 2"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>mouse_number_colours</span><span class='op'>)</span>
<span class='va'>p4</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span>, colour <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log10</a></span><span class='op'>(</span><span class='va'>lib.size</span><span class='op'>)</span>, label <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>y_df</span><span class='op'>)</span><span class='op'>)</span>,
  data <span class='op'>=</span> <span class='va'>y_df</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html'>geom_text_repel</a></span><span class='op'>(</span>size <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_viridis.html'>scale_colour_viridis_c</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 1"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 2"</span><span class='op'>)</span>

<span class='va'>p1</span> <span class='op'>+</span> <span class='va'>p2</span> <span class='op'>+</span> <span class='va'>p3</span> <span class='op'>+</span> <span class='va'>p4</span> <span class='op'>+</span> <span class='fu'><a href='https://patchwork.data-imaginist.com/reference/plot_layout.html'>plot_layout</a></span><span class='op'>(</span>ncol <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:mds-y"></span>
<img src="C086_Kinkel.multi-sample_comparisons_files/figure-html5/mds-y-1.png" alt="MDS plots of aggregated technical replicates coloured by various experimental factors." width="624" />
<p class="caption">
Figure 2: MDS plots of aggregated technical replicates coloured by various experimental factors.
</p>
</div>
</div>
<h2 id="further-normalization">Further normalization</h2>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>design</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/model.matrix.html'>model.matrix</a></span><span class='op'>(</span>
  <span class='op'>~</span><span class='fl'>0</span> <span class='op'>+</span> <span class='va'>smchd1_genotype_updated</span> <span class='op'>+</span> <span class='va'>sex</span>,
  <span class='va'>y</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>design</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/grep.html'>sub</a></span><span class='op'>(</span>
  <span class='st'>"smchd1_genotype_updated|sex"</span>,
  <span class='st'>""</span>,
  <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>design</span><span class='op'>)</span><span class='op'>)</span>

<span class='va'>y</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/estimateDisp.html'>estimateDisp</a></span><span class='op'>(</span><span class='va'>y</span>, <span class='va'>design</span><span class='op'>)</span>

<span class='va'>dgeglm</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/glmfit.html'>glmFit</a></span><span class='op'>(</span><span class='va'>y</span>, <span class='va'>design</span><span class='op'>)</span>
<span class='va'>contrasts</span> <span class='op'>&lt;-</span> <span class='fu'>makeContrasts</span><span class='op'>(</span><span class='va'>Del</span> <span class='op'>-</span> <span class='va'>Het</span>, levels <span class='op'>=</span> <span class='va'>design</span><span class='op'>)</span>
<span class='va'>dgelrt</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/glmfit.html'>glmLRT</a></span><span class='op'>(</span><span class='va'>dgeglm</span>, contrast <span class='op'>=</span> <span class='va'>contrasts</span><span class='op'>)</span>
<span class='va'>tt</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/topTags.html'>topTags</a></span><span class='op'>(</span><span class='va'>dgelrt</span>, n <span class='op'>=</span> <span class='cn'>Inf</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/summary.html'>summary</a></span><span class='op'>(</span><span class='fu'>decideTests</span><span class='op'>(</span><span class='va'>dgelrt</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>The MDS plots in Figure <a href="#fig:mds-y">2</a> strongly suggest that we will not detect many differentially expressed genes (DEGs) using a standard analysis. This is borne out in Figure <a href="#fig:vanilla-de">3</a>, which shows the results of a standard <em><a href="https://bioconductor.org/packages/3.12/edgeR">edgeR</a></em> analysis of the data (there is 3 DEG (<span class="math inline">\(FDR &lt; 0.05\)</span>)).</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/r/graphics/par.html'>par</a></span><span class='op'>(</span>mfrow <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>2</span><span class='op'>)</span><span class='op'>)</span>
<span class='fu'>plotMD</span><span class='op'>(</span><span class='va'>dgelrt</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/graphics/abline.html'>abline</a></span><span class='op'>(</span>h <span class='op'>=</span> <span class='fl'>0</span>, lty <span class='op'>=</span> <span class='fl'>2</span>, col <span class='op'>=</span> <span class='st'>"dodgerBlue"</span><span class='op'>)</span>
<span class='fu'>plotMD</span><span class='op'>(</span>
  <span class='va'>dgelrt</span>,
  status <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/ifelse.html'>ifelse</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>dgelrt</span><span class='op'>)</span> <span class='op'>==</span> <span class='st'>"Smchd1"</span>, <span class='st'>"Smchd1"</span>, <span class='st'>"Other"</span><span class='op'>)</span>,
  hl.col <span class='op'>=</span> <span class='st'>"orange"</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/graphics/abline.html'>abline</a></span><span class='op'>(</span>h <span class='op'>=</span> <span class='fl'>0</span>, lty <span class='op'>=</span> <span class='fl'>2</span>, col <span class='op'>=</span> <span class='st'>"dodgerBlue"</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:vanilla-de"></span>
<img src="C086_Kinkel.multi-sample_comparisons_files/figure-html5/vanilla-de-1.png" alt="MD plot highlighting DE genes (left) and Smchd1 (right)." width="624" />
<p class="caption">
Figure 3: MD plot highlighting DE genes (left) and Smchd1 (right).
</p>
</div>
</div>
<p>To further normalize the data, we use <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4404308/"><strong>RUVs</strong></a>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>differences</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/RUVSeq/man/makeGroups.html'>makeGroups</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/base/interaction.html'>interaction</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>smchd1_genotype_updated</span>, <span class='va'>y</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>sex</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>ruv</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/RUVSeq/man/RUVs.html'>RUVs</a></span><span class='op'>(</span>
    x <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/EDASeq/man/betweenLaneNormalization-methods.html'>betweenLaneNormalization</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>$</span><span class='va'>counts</span>, <span class='st'>"upper"</span><span class='op'>)</span>,
    cIdx <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>)</span>,
    k <span class='op'>=</span> <span class='fl'>3</span>,
    scIdx <span class='op'>=</span> <span class='va'>differences</span><span class='op'>)</span>

<span class='va'>design</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/model.matrix.html'>model.matrix</a></span><span class='op'>(</span>
  <span class='op'>~</span><span class='fl'>0</span> <span class='op'>+</span> <span class='va'>smchd1_genotype_updated</span> <span class='op'>+</span> <span class='va'>sex</span> <span class='op'>+</span> <span class='va'>ruv</span><span class='op'>$</span><span class='va'>W</span>,
  <span class='va'>y</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>design</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/grep.html'>sub</a></span><span class='op'>(</span>
  <span class='st'>"ruv\\$|smchd1_genotype_updated|sex"</span>,
  <span class='st'>""</span>,
  <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>design</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>Figure <a href="#fig:ruv-mds">4</a> shows that after applying RUVs samples cluster by combination of genotype and sex, which is what we want to see.</p>
<p><strong>Therefore, we opt to further normalize the data using RUVs in each DE analysis<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a></strong></p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>y_mds</span> <span class='op'>&lt;-</span> <span class='fu'>plotMDS</span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/SingleCellExperiment/man/assays.html'>cpm</a></span><span class='op'>(</span><span class='va'>ruv</span><span class='op'>$</span><span class='va'>normalizedCounts</span>, log <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>, plot <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>

<span class='va'>y_df</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cbind.html'>cbind</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>y_mds</span><span class='op'>$</span><span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y_mds</span><span class='op'>$</span><span class='va'>y</span><span class='op'>)</span>, <span class='va'>y</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>y_df</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>)</span>

<span class='va'>p1</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span>, colour <span class='op'>=</span> <span class='va'>smchd1_genotype_updated</span>, label <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>y_df</span><span class='op'>)</span><span class='op'>)</span>,
  data <span class='op'>=</span> <span class='va'>y_df</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html'>geom_text_repel</a></span><span class='op'>(</span>size <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 1"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 2"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>smchd1_genotype_updated_colours</span><span class='op'>)</span>
<span class='va'>p2</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span>, colour <span class='op'>=</span> <span class='va'>sex</span>, label <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>y_df</span><span class='op'>)</span><span class='op'>)</span>,
  data <span class='op'>=</span> <span class='va'>y_df</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html'>geom_text_repel</a></span><span class='op'>(</span>size <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 1"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 2"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>sex_colours</span><span class='op'>)</span>
<span class='va'>p3</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span>, colour <span class='op'>=</span> <span class='va'>mouse_number</span>, label <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>y_df</span><span class='op'>)</span><span class='op'>)</span>,
  data <span class='op'>=</span> <span class='va'>y_df</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html'>geom_text_repel</a></span><span class='op'>(</span>size <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 1"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 2"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>mouse_number_colours</span><span class='op'>)</span>
<span class='va'>p4</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span>, colour <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log10</a></span><span class='op'>(</span><span class='va'>lib.size</span><span class='op'>)</span>, label <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>y_df</span><span class='op'>)</span><span class='op'>)</span>,
  data <span class='op'>=</span> <span class='va'>y_df</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html'>geom_text_repel</a></span><span class='op'>(</span>size <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_viridis.html'>scale_colour_viridis_c</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 1"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 2"</span><span class='op'>)</span>

<span class='va'>p1</span> <span class='op'>+</span> <span class='va'>p2</span> <span class='op'>+</span> <span class='va'>p3</span> <span class='op'>+</span> <span class='va'>p4</span> <span class='op'>+</span> <span class='fu'><a href='https://patchwork.data-imaginist.com/reference/plot_layout.html'>plot_layout</a></span><span class='op'>(</span>ncol <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:ruv-mds"></span>
<img src="C086_Kinkel.multi-sample_comparisons_files/figure-html5/ruv-mds-1.png" alt="MDS plots of aggregated technical replicates following RUV coloured by various experimental factors." width="624" />
<p class="caption">
Figure 4: MDS plots of aggregated technical replicates following RUV coloured by various experimental factors.
</p>
</div>
</div>
<h2 id="gene-sets">Gene sets</h2>
<p>Marnie and Sarah interested in the following gene sets:</p>
<aside>
These are available as Excel files in <a href="../data/sarahs_gene_lists"><code>data/sarahs_gene_lists/</code></a>. These genes are matched to the present dataset using the supplied gene symbol.
</aside>
<ul>
<li><strong>Non-directional gene sets</strong> (i.e. no logFC available from previous data)
<ul>
<li><code>X-linked</code>: Anything with <code>ENSEMBL.SEQNAME</code> set as ‘<code>X</code>’).</li>
<li><code>Protocadherin</code>: Genes with symbol of the form <em>Pcdh#</em>.</li>
<li><code>Prader-Willi</code>: Imprinted genes from the Prader-Willi cluster (<em>Ndn</em>, <em>Mkrn3</em>, <em>Peg12</em>)</li>
</ul></li>
<li><strong>Directional gene sets</strong> (i.e. logFC available from previous data).
<ul>
<li><code>Male: Het vs. Del</code>: DEGs from a comparison of male Smchd1-het and Smchd1-del B cells from aged mice (&gt;12 months old)<a href="#fn2" class="footnote-ref" id="fnref2" role="doc-noteref"><sup>2</sup></a></li>
<li><code>Female: Het vs. Del</code>: DEGs from a comparison of female Smchd1-het and Smchd1-del B cells from aged mice (&gt;12 months old)<a href="#fn3" class="footnote-ref" id="fnref3" role="doc-noteref"><sup>3</sup></a>.</li>
<li><code>Sun (2014)</code>: DEGs from a comparison of HSCs from young (2mo) and aged (24mo) male mice <span class="citation" data-cites="sun2014epigenomic">(<a href="#ref-sun2014epigenomic" role="doc-biblioref">Sun et al. 2014</a>)</span>.</li>
<li><code>Chambers (2007)</code>: Upregulated DEGs from a comparison of HSCs to an average of all other cell types analysed <span class="citation" data-cites="chambers2007hematopoietic">(<a href="#ref-chambers2007hematopoietic" role="doc-biblioref">Chambers et al. 2007</a>)</span>.</li>
<li><code>Gazit (2013)</code>: Genes with “enriched expression in HSCs”<a href="#fn4" class="footnote-ref" id="fnref4" role="doc-noteref"><sup>4</sup></a> compared with all other hematopoietic cell types <span class="citation" data-cites="gazit2013transcriptome">(<a href="#ref-gazit2013transcriptome" role="doc-biblioref">Gazit et al. 2013</a>)</span>.
<ul>
<li>Although no logFC is available, by definition these genes have a logFC &gt; 0.</li>
</ul></li>
<li><code>Chambers (2007) &amp; Gazit (2013)</code>: The union of <code>Chambers (2007)</code> and <code>Gazit (2013)</code> with the logFC preferentially taken from <code>Chambers (2007)</code>.
<ul>
<li>This is in the spirit of <span class="citation" data-cites="vizan2020polycomb"><a href="#ref-vizan2020polycomb" role="doc-biblioref">Vizán et al.</a> (<a href="#ref-vizan2020polycomb" role="doc-biblioref">2020</a>)</span> who do a gene set enrichment analysis using “a merge of differentially expressed genes provided by <code>Gazit (2013)</code> and <code>Chambers (2007)</code>” but do not explain how they ‘merge’ these gene lists.</li>
</ul></li>
<li><code>Proliferation Group Venezia (2004)</code>: Genes that have their maximum expression at peak-cycling times (<strong>T</strong>ime <strong>O</strong>f <strong>M</strong>ax = TOM = 2, 3, and 6 days post-treatment) in a time-course experiment designed to stimulate adult HSCs to proliferate <span class="citation" data-cites="venezia2004molecular">(<a href="#ref-venezia2004molecular" role="doc-biblioref">Venezia et al. 2004</a>)</span>.
<ul>
<li>The p-value (ANOVA) is based on about 1500 genes that changed significantly over the time course while the logFC is a separate comparison between FL HSCs (naturally proliferative) and adult HSCs (naturally quiescent).</li>
</ul></li>
<li><code>Quiescence Group Venezia (2004)</code>: Genes that have their maximum expression at non-cycling times (<strong>T</strong>ime <strong>O</strong>f <strong>M</strong>ax = TOM = 0, 1, 10, and 30 days post-treatment) in a time-course experiment designed to stimulate adult HSCs to proliferate <span class="citation" data-cites="venezia2004molecular">(<a href="#ref-venezia2004molecular" role="doc-biblioref">Venezia et al. 2004</a>)</span>.
<ul>
<li>The p-value (ANOVA) is based on about 1500 genes that changed significantly over the time course while the logFC is a separate comparison between FL HSCs (naturally proliferative) and adult HSCs (naturally quiescent).</li>
</ul></li>
<li><code>Smchd1 female B cells (chrX)</code>: logFCs for X chromosome genes from a comparison of Smchd1-del versus Smchd1-het B cells from female mice
<ul>
<li>Sarah performed this analysis in SeqMonk and exported the complete DE results as a CSV file.</li>
<li>These data are from the same experiment as the <code>Female: Het vs. Del</code> gene list.</li>
</ul></li>
</ul></li>
</ul>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='co'># NOTE: The `as.data.frame()` calls are needed because read_excel() returns a </span>
<span class='co'>#       tibble but `limma::fry()` does not support tibbles.</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://readxl.tidyverse.org'>readxl</a></span><span class='op'>)</span>
<span class='va'>male_het_vs_del</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://readxl.tidyverse.org/reference/read_excel.html'>read_excel</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://here.r-lib.org//reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"data"</span>, <span class='st'>"sarahs_gene_lists"</span>, <span class='st'>"MaleHetvDel_EdgeR.xlsx"</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>male_het_vs_del</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='va'>male_het_vs_del</span><span class='op'>)</span>
<span class='va'>female_het_vs_del</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://readxl.tidyverse.org/reference/read_excel.html'>read_excel</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://here.r-lib.org//reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"data"</span>, <span class='st'>"sarahs_gene_lists"</span>, <span class='st'>"Bcell_female_HetvDel_EdgeR_mRNA.xlsx"</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>female_het_vs_del</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='va'>female_het_vs_del</span><span class='op'>)</span>
<span class='va'>sun_2014</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://readxl.tidyverse.org/reference/read_excel.html'>read_excel</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://here.r-lib.org//reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"data"</span>, <span class='st'>"sarahs_gene_lists"</span>, <span class='st'>"Sun_2014_CellStemCell_DE_HSC_aging.xlsx"</span><span class='op'>)</span>,
  sheet <span class='op'>=</span> <span class='st'>"Differentially expressed genes"</span><span class='op'>)</span>
<span class='va'>sun_2014</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='va'>sun_2014</span><span class='op'>)</span>
<span class='va'>chambers_2007</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://readxl.tidyverse.org/reference/read_excel.html'>read_excel</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://here.r-lib.org//reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"data"</span>, <span class='st'>"sarahs_gene_lists"</span>, <span class='st'>"HSC_signature_genes.xlsx"</span><span class='op'>)</span>,
  sheet <span class='op'>=</span> <span class='st'>"Chambers_2007_CellStemCell"</span>,
  na <span class='op'>=</span> <span class='st'>"NA"</span><span class='op'>)</span>
<span class='va'>chambers_2007</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='va'>chambers_2007</span><span class='op'>)</span>
<span class='va'>gazit_2013</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://readxl.tidyverse.org/reference/read_excel.html'>read_excel</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://here.r-lib.org//reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"data"</span>, <span class='st'>"sarahs_gene_lists"</span>, <span class='st'>"HSC_signature_genes.xlsx"</span><span class='op'>)</span>,
  sheet <span class='op'>=</span> <span class='st'>"Gazit_2013_StemCellRep"</span>,
  range <span class='op'>=</span> <span class='st'>"A3:C325"</span><span class='op'>)</span>
<span class='va'>gazit_2013</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='va'>gazit_2013</span><span class='op'>)</span>
<span class='co'># NOTE: File manually downloaded from   </span>
<span class='co'>#       https://doi.org/10.1371/journal.pbio.0020301.st002 and converted to a </span>
<span class='co'>#       CSV file using LibreOffice.</span>
<span class='va'>venezia_proliferation</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/utils/read.table.html'>read.csv</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://here.r-lib.org//reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"data"</span>, <span class='st'>"sarahs_gene_lists"</span>, <span class='st'>"pbio.0020301.st002.csv"</span><span class='op'>)</span>,
  header <span class='op'>=</span> <span class='cn'>TRUE</span>,
  skip <span class='op'>=</span> <span class='fl'>1</span>,
  nrow <span class='op'>=</span> <span class='fl'>680</span><span class='op'>)</span>
<span class='co'># NOTE: File manually downloaded from   </span>
<span class='co'>#       https://doi.org/10.1371/journal.pbio.0020301.st004 and converted to a </span>
<span class='co'>#       CSV file using LibreOffice.</span>
<span class='va'>venezia_quiescence</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/utils/read.table.html'>read.csv</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://here.r-lib.org//reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"data"</span>, <span class='st'>"sarahs_gene_lists"</span>, <span class='st'>"pbio.0020301.st004.csv"</span><span class='op'>)</span>,
  header <span class='op'>=</span> <span class='cn'>TRUE</span>,
  skip <span class='op'>=</span> <span class='fl'>1</span>,
  nrow <span class='op'>=</span> <span class='fl'>808</span><span class='op'>)</span>
<span class='va'>smchd1_femBcells</span> <span class='op'>&lt;-</span> <span class='fu'>data.table</span><span class='fu'>::</span><span class='fu'><a href='https://Rdatatable.gitlab.io/data.table/reference/fread.html'>fread</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://here.r-lib.org//reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"data"</span>, <span class='st'>"sarahs_gene_lists"</span>, <span class='st'>"210923_EdgeR stats p1.0 log2.txt"</span><span class='op'>)</span>,
  data.table <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<h1 id="analysis-of-aggregated-technical-replicates">Analysis of aggregated technical replicates</h1>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>y_differences</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/RUVSeq/man/makeGroups.html'>makeGroups</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/base/interaction.html'>interaction</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>smchd1_genotype_updated</span>, <span class='va'>y</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>sex</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>y_ruv</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/RUVSeq/man/RUVs.html'>RUVs</a></span><span class='op'>(</span>
    x <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/EDASeq/man/betweenLaneNormalization-methods.html'>betweenLaneNormalization</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>$</span><span class='va'>counts</span>, <span class='st'>"upper"</span><span class='op'>)</span>,
    cIdx <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>)</span>,
    k <span class='op'>=</span> <span class='fl'>3</span>,
    scIdx <span class='op'>=</span> <span class='va'>y_differences</span><span class='op'>)</span>

<span class='va'>y_design</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/model.matrix.html'>model.matrix</a></span><span class='op'>(</span>
  <span class='op'>~</span><span class='fl'>0</span> <span class='op'>+</span> <span class='va'>smchd1_genotype_updated</span> <span class='op'>+</span> <span class='va'>sex</span> <span class='op'>+</span> <span class='va'>y_ruv</span><span class='op'>$</span><span class='va'>W</span>,
  <span class='va'>y</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>y_design</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/grep.html'>sub</a></span><span class='op'>(</span>
  <span class='st'>"ruv\\$|smchd1_genotype_updated|sex"</span>,
  <span class='st'>""</span>,
  <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>y_design</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<h2 id="mds">MDS</h2>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>y_mds</span> <span class='op'>&lt;-</span> <span class='fu'>plotMDS</span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/SingleCellExperiment/man/assays.html'>cpm</a></span><span class='op'>(</span><span class='va'>y_ruv</span><span class='op'>$</span><span class='va'>normalizedCounts</span>, log <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>, plot <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>

<span class='va'>y_df</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cbind.html'>cbind</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>y_mds</span><span class='op'>$</span><span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y_mds</span><span class='op'>$</span><span class='va'>y</span><span class='op'>)</span>, <span class='va'>y</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>y_df</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>)</span>

<span class='va'>p1</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span>, colour <span class='op'>=</span> <span class='va'>smchd1_genotype_updated</span>, label <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>y_df</span><span class='op'>)</span><span class='op'>)</span>,
  data <span class='op'>=</span> <span class='va'>y_df</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html'>geom_text_repel</a></span><span class='op'>(</span>size <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 1"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 2"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>smchd1_genotype_updated_colours</span><span class='op'>)</span>
<span class='va'>p2</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span>, colour <span class='op'>=</span> <span class='va'>sex</span>, label <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>y_df</span><span class='op'>)</span><span class='op'>)</span>,
  data <span class='op'>=</span> <span class='va'>y_df</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html'>geom_text_repel</a></span><span class='op'>(</span>size <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 1"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 2"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>sex_colours</span><span class='op'>)</span>
<span class='va'>p3</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span>, colour <span class='op'>=</span> <span class='va'>mouse_number</span>, label <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>y_df</span><span class='op'>)</span><span class='op'>)</span>,
  data <span class='op'>=</span> <span class='va'>y_df</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html'>geom_text_repel</a></span><span class='op'>(</span>size <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 1"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 2"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>mouse_number_colours</span><span class='op'>)</span>
<span class='va'>p4</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span>, colour <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log10</a></span><span class='op'>(</span><span class='va'>lib.size</span><span class='op'>)</span>, label <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>y_df</span><span class='op'>)</span><span class='op'>)</span>,
  data <span class='op'>=</span> <span class='va'>y_df</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html'>geom_text_repel</a></span><span class='op'>(</span>size <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_viridis.html'>scale_colour_viridis_c</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 1"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 2"</span><span class='op'>)</span>

<span class='va'>p1</span> <span class='op'>+</span> <span class='va'>p2</span> <span class='op'>+</span> <span class='va'>p3</span> <span class='op'>+</span> <span class='va'>p4</span> <span class='op'>+</span> <span class='fu'><a href='https://patchwork.data-imaginist.com/reference/plot_layout.html'>plot_layout</a></span><span class='op'>(</span>ncol <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:unnamed-chunk-16"></span>
<img src="C086_Kinkel.multi-sample_comparisons_files/figure-html5/unnamed-chunk-16-1.png" alt="MDS plots of aggregated technical replicates following RUV coloured by various experimental factors." width="624" />
<p class="caption">
Figure 5: MDS plots of aggregated technical replicates following RUV coloured by various experimental factors.
</p>
</div>
</div>
<h2 id="de-analysis">DE analysis</h2>
<p>Fit a model using <em><a href="https://bioconductor.org/packages/3.12/edgeR">edgeR</a></em> with a term for <code>smchd1_genotype_updated</code> and <code>sex</code>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>y</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/estimateDisp.html'>estimateDisp</a></span><span class='op'>(</span><span class='va'>y</span>, <span class='va'>y_design</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/graphics/par.html'>par</a></span><span class='op'>(</span>mfrow <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/plotBCV.html'>plotBCV</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>y_dgeglm</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/glmfit.html'>glmFit</a></span><span class='op'>(</span><span class='va'>y</span>, <span class='va'>y_design</span><span class='op'>)</span>
<span class='va'>y_contrast</span> <span class='op'>&lt;-</span> <span class='fu'>makeContrasts</span><span class='op'>(</span><span class='va'>Del</span> <span class='op'>-</span> <span class='va'>Het</span>, levels <span class='op'>=</span> <span class='va'>y_design</span><span class='op'>)</span>
<span class='va'>y_dgelrt</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/glmfit.html'>glmLRT</a></span><span class='op'>(</span><span class='va'>y_dgeglm</span>, contrast <span class='op'>=</span> <span class='va'>y_contrast</span><span class='op'>)</span>
<span class='va'>y_dt</span> <span class='op'>&lt;-</span> <span class='fu'>decideTests</span><span class='op'>(</span><span class='va'>y_dgelrt</span><span class='op'>)</span>

<span class='fu'><a href='https://rdrr.io/r/base/summary.html'>summary</a></span><span class='op'>(</span><span class='va'>y_dt</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span>caption <span class='op'>=</span> <span class='st'>"Number of DEGs (FDR &lt; 0.05)"</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<table>
<caption><span id="tab:unnamed-chunk-18">Table 1: </span>Number of DEGs (FDR &lt; 0.05)</caption>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">1<em>Del -1</em>Het</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Down</td>
<td style="text-align: right;">78</td>
</tr>
<tr class="even">
<td style="text-align: left;">NotSig</td>
<td style="text-align: right;">11362</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Up</td>
<td style="text-align: right;">10</td>
</tr>
</tbody>
</table>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/r/graphics/par.html'>par</a></span><span class='op'>(</span>mfrow <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span>
<span class='fu'>plotMD</span><span class='op'>(</span><span class='va'>y_dgelrt</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:unnamed-chunk-19"></span>
<img src="C086_Kinkel.multi-sample_comparisons_files/figure-html5/unnamed-chunk-19-1.png" alt="MD plot highlighting DEGs (red)." width="624" />
<p class="caption">
Figure 6: MD plot highlighting DEGs (red).
</p>
</div>
</div>
<p>An interactive <em><a href="https://bioconductor.org/packages/3.12/Glimma">Glimma</a></em> MD plot of the differential expression results is available from <a href="../output/DEGs/reads/glimma-plots/aggregated_tech_reps.md-plot.html"><code>output/DEGs/reads/glimma-plots/aggregated_tech_reps.md-plot.html</code></a>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/pkg/Glimma/man/glMDPlot.html'>glMDPlot</a></span><span class='op'>(</span>
  x <span class='op'>=</span> <span class='va'>y_dgelrt</span>,
  counts <span class='op'>=</span> <span class='va'>y</span>,
  anno <span class='op'>=</span> <span class='va'>y</span><span class='op'>$</span><span class='va'>genes</span>,
  display.columns <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"ENSEMBL.SYMBOL"</span><span class='op'>)</span>,
  groups <span class='op'>=</span> <span class='va'>y</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span>,
  samples <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>)</span>,
  status <span class='op'>=</span> <span class='va'>y_dt</span>,
  transform <span class='op'>=</span> <span class='cn'>TRUE</span>,
  sample.cols <span class='op'>=</span> <span class='va'>y</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>smchd1_genotype_updated_colours</span>,
  path <span class='op'>=</span> <span class='va'>outdir</span>,
  html <span class='op'>=</span> <span class='st'>"aggregated_tech_reps.md-plot"</span>,
  launch <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>y_tt</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/topTags.html'>topTags</a></span><span class='op'>(</span><span class='va'>y_dgelrt</span>, p.value <span class='op'>=</span> <span class='fl'>0.05</span>, n <span class='op'>=</span> <span class='cn'>Inf</span><span class='op'>)</span>
<span class='kw'>if</span> <span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/nrow.html'>nrow</a></span><span class='op'>(</span><span class='va'>y_tt</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>{</span>
  <span class='va'>y_tt</span><span class='op'>$</span><span class='va'>table</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"logFC"</span>, <span class='st'>"logCPM"</span>, <span class='st'>"LR"</span>, <span class='st'>"PValue"</span>, <span class='st'>"FDR"</span><span class='op'>)</span><span class='op'>]</span> <span class='op'>%&gt;%</span>
    <span class='fu'>DT</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/DT/man/datatable.html'>datatable</a></span><span class='op'>(</span>caption <span class='op'>=</span> <span class='st'>"DEGs (FDR &lt; 0.05)"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
    <span class='fu'>DT</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/DT/man/formatCurrency.html'>formatSignif</a></span><span class='op'>(</span>
      <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"logFC"</span>, <span class='st'>"logCPM"</span>, <span class='st'>"LR"</span>, <span class='st'>"PValue"</span>, <span class='st'>"FDR"</span><span class='op'>)</span>,
      digits <span class='op'>=</span> <span class='fl'>3</span><span class='op'>)</span>
<span class='op'>}</span>
</code></pre>
</div>
</details>
<div id="htmlwidget-cfdf39670e2a6930ce86" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-cfdf39670e2a6930ce86">{"x":{"filter":"none","caption":"<caption>DEGs (FDR &lt; 0.05)<\/caption>","data":[["Gm10401","Homer2","Mmp16","Acvr1","Ehhadh","Atp10a","Pld2","Sema4a","Csprs","Stat2","Hdx","Smtnl1","Pde4b","D630039A03Rik","Nr1d2","Kcnk6","Gm7120","1810055G02Rik","Acot11","Zswim4","Plcb1","Itsn1","Nwd1","Dennd2c","Tmem44","Gfi1","Spic","Anxa3","Itga9","Bbs12","Tm9sf1","Fbxo16","Ncoa2","4930404N11Rik","Il18rap","Nr6a1","Klhl24","Atg9a","Kbtbd12","Utp3","Notch1","Mrpl3","Mcmdc2","Syn3","Zc4h2","Frs3","Tesk2","Ndrg4","Stard9","Arrdc4","Ets1","Klc3","Tnk1","Gm26529","Trim62","Tceanc","Smtnl2","Smchd1","Ppm1e","Cap1","Lpcat2","Veph1","Dync2h1","Gpr64","Slc25a53","Aven","Prpf38b","Plekhf2","Rcbtb1","Rab2b","Pcdhb17","Rnf150","Fez1","Met","Mbd5","Tctex1d2","Dpf3","Mfn1","Kcnj5","Medag","Cyc1","Tsc2","Stxbp1","1810022K09Rik","Fam114a1","Hist1h2bc","Orc5","Ralgps1"],[-7.84808082993086,-5.75684646056494,-6.31196516297321,-2.11746148975432,-6.39154891273014,-2.09739121945655,-7.10337056951102,0.931363638826828,5.12192484580367,-1.37288423436865,-4.02835572823555,-8.13640616874141,-0.814150924869048,-1.95584625649233,-1.78829403633264,-5.21281937444203,-1.21542976094887,-6.25786519406955,-2.2209430270725,-1.37822132000364,-3.78096846522169,-0.814445058245925,-0.832749588385807,-3.71951700715647,-5.04269411434463,-3.92173528847513,-4.77384930296946,-5.71414166585496,-5.21749606269008,-2.64774870395839,-5.72465766169812,-5.06419694894095,-0.815483378554858,-6.09651900026999,-3.26378577854021,-4.37067677371054,-0.604164361959971,-6.74490465456455,-1.92405800620215,0.839370638736923,-1.10931051287529,0.876213192231672,-4.72951268171027,-4.75110211129152,-2.41497801633033,-4.69174061502301,-4.00249198315395,-5.48073303586583,-6.71466805035337,-1.01368498202527,-1.99827117916017,-6.1297343907904,-4.553429472103,-6.24143965686269,-3.68164596457823,1.39209718275963,-5.93899557054786,-0.878115928320397,-3.25134439564286,-2.66156014646254,-5.44751411252183,-5.70404705752142,-1.59913548233815,-2.77142001131969,-2.24985707423792,-1.13887005834666,-0.54030515416344,-1.1236087987457,-2.3410888716749,-1.25501288195402,-4.97936050410376,-1.75951836934637,-3.03450648640412,-5.77517351517917,-1.09196322250628,1.02266752370567,-1.91565825231355,-1.70773897618107,-5.48736424771184,-5.95487930522397,0.70239695145888,-1.05356880711304,-1.0953376159652,0.640929633574322,-3.00646860686495,0.796860514212153,-1.15249234906087,1.48017136051297],[0.955536269164961,2.73234490039862,3.00818225880876,5.32477031360424,1.82623100517402,4.45313049646912,1.4586425525055,2.59483441640004,1.21820715030448,5.43630017080721,1.03376027846901,2.54760415463225,6.75600407018609,4.93902120353162,6.28836664440434,1.69241669648006,5.70383214742746,3.15698180218629,3.91774989590973,5.6753625695574,4.28993104846008,7.97225120149731,7.02493844903246,2.64989922576958,1.59918975894339,4.54300412756556,1.36834277522146,1.40426804598299,1.15257925462096,3.57472361843296,1.45469093418362,2.30019342649539,6.60530951444459,0.184046108401979,3.39838105411134,1.0518237548917,7.83184700183316,0.223687253529935,1.47943136240065,7.01082461312712,5.42216575664466,6.18239632070984,1.2125992686978,1.07496035604063,4.03893272844761,2.90411295145399,3.07444144835565,1.50873637051011,0.864332284185668,5.75118454566345,4.63403250355178,0.883725633207962,1.77531833654171,1.32521364162011,2.61207354129052,4.49279062958832,0.716453865618732,6.65273470544343,1.83667429737215,5.39260442919168,2.31335256663006,1.89216536116327,4.90809333639143,3.27221798702064,3.70155486698859,5.31402246093214,8.19778546532528,5.07063541595254,4.76560057336016,5.80088797908267,1.78215408529576,4.29238911660093,1.34858986458794,1.48433657000651,5.75801335473454,5.79610644803866,0.874553506683678,4.24892733903507,2.34321089378287,0.91459268398026,6.5087120790676,6.36246014572765,5.44383206772236,6.81470321802475,3.02235781438988,6.58378739965889,6.00079909916802,5.39493920748861],[28.6797910914129,27.8179857972552,26.7424975058021,25.0270443880627,24.43502675497,22.1340282406098,21.7528611086239,21.1377611000853,20.6652407161872,19.4577351679763,19.4140181397995,19.1460682488001,19.0153678300333,18.9815895541808,18.4242179287567,18.2332749776932,18.0132368492803,17.7896179750694,17.1842296895429,16.9489680732052,16.8191744523018,16.790301262219,16.7140477663004,16.7099054176351,16.7016717327254,16.6238191305885,16.6117140424701,16.4512010932674,16.2123674728784,16.1251377944161,16.1098656903648,16.0897817146201,16.0771097961882,15.9724948277211,15.913914954373,15.7749035666755,15.7651046133494,15.6135893826959,15.5540032130318,15.4728041628672,15.419267290694,15.3702315791383,15.3632236804992,15.3202676653257,15.1535038683109,15.0370710316049,14.9387607073591,14.9312136916456,14.896726057859,14.8555270585378,14.7110723792883,14.6973609667619,14.6131680890866,14.6061729433232,14.5718648902713,14.526451090611,14.4902772301146,14.4137650800778,14.1673915961321,14.1264296005503,14.0762881768295,14.0620391047418,14.0043497499663,13.9536561747495,13.890561022534,13.8020884942726,13.6951584264818,13.6860948174008,13.5761358682687,13.5744333460446,13.5713196171801,13.4901801168956,13.3593469197423,13.2746961253909,13.25820828138,13.1908946699609,13.0787796415238,13.0663742512348,13.0567600894486,13.0418358658913,13.0242287502755,12.911404728551,12.8591457724143,12.8517448221625,12.8248814357995,12.7031319908704,12.6909871537815,12.631389483656],[8.53901912656334e-08,1.3328130302983e-07,2.32452493439739e-07,5.65317938974555e-07,7.68585297156902e-07,2.54262550443676e-06,3.10127019851178e-06,4.27422236563646e-06,5.47000359880467e-06,1.02850467390276e-05,1.05231806494794e-05,1.21086670600266e-05,1.29669894155265e-05,1.31985830660866e-05,1.76796674204009e-05,1.95434488726593e-05,2.19374263271845e-05,2.46725861140446e-05,3.39240925779484e-05,3.83981861995409e-05,4.11157601621574e-05,4.17461232357212e-05,4.34580117609637e-05,4.35530029420372e-05,4.37424357811724e-05,4.55750528549045e-05,4.58668562083733e-05,4.99185451455025e-05,5.66232448247034e-05,5.92917935048897e-05,5.97718476688456e-05,6.04091110589875e-05,6.08147016201785e-05,6.42694960631901e-05,6.62895612362446e-05,7.13425945230242e-05,7.17130717085138e-05,7.76942102630921e-05,8.01816860128138e-05,8.3700924539566e-05,8.61058704290186e-05,8.8369583084256e-05,8.869795418389e-05,9.07377582884934e-05,9.91142077213826e-05,0.000105419933626103,0.000111057661805683,0.000111502796991261,0.000113559897910232,0.000116067454703916,0.000125308300135564,0.000126223028641797,0.000131989036394148,0.000132479873328144,0.000134913948682027,0.000138205345195236,0.000140884829844357,0.000146725869920342,0.0001672439392215,0.000170925284400761,0.000175542904781567,0.000176877911664609,0.000182388214347381,0.000187372884572289,0.000193769157027579,0.000203110323397182,0.000215008070083852,0.000216048273324782,0.000229079325019216,0.000229287195490913,0.000229667859906787,0.000239815174703961,0.000257137835394806,0.000269012487777264,0.000271388914790751,0.000281312567689754,0.000298659968464125,0.000300644539705175,0.000302191720492804,0.00030460935343818,0.000307486686801021,0.000326586102297332,0.00033583386063532,0.00033716471275841,0.000342040181769677,0.000365043659526808,0.000367422249102371,0.000379323124556278],[0.000763035459845774,0.000763035459845774,0.000887193683295006,0.00161822260031466,0.00176006033048931,0.00485217700430015,0.0050727919675657,0.00611748076081718,0.00695906013403483,0.0107945554361923,0.0107945554361923,0.0107945554361923,0.0107945554361923,0.0107945554361923,0.0134954794642393,0.0139857805994968,0.0147755018497802,0.0156945061669895,0.0194509445772546,0.0194509445772546,0.0194509445772546,0.0194509445772546,0.0194509445772546,0.0194509445772546,0.0194509445772546,0.0194509445772546,0.0194509445772546,0.020413119354143,0.0211008585924559,0.0211008585924559,0.0211008585924559,0.0211008585924559,0.0211008585924559,0.0216436979389273,0.0216861564615715,0.0221922884070941,0.0221922884070941,0.023410492302958,0.0235405206370953,0.0236124393728011,0.0236124393728011,0.0236124393728011,0.0236124393728011,0.0236124393728011,0.0252190595202185,0.0262403965221495,0.0265359353280031,0.0265359353280031,0.0265359353280031,0.0265794471271967,0.0277933399605494,0.0277933399605494,0.0280866311347129,0.0280866311347129,0.0280866311347129,0.0282580571872402,0.0283005491529453,0.0289657105273778,0.0324566627811216,0.0326182417731453,0.0326653562670931,0.0326653562670931,0.0331483341948812,0.033522180130511,0.0341331822763966,0.0352365636802687,0.0363787166113052,0.0363787166113052,0.0370379858582072,0.0370379858582072,0.0370379858582072,0.0381372743105605,0.0403318933598702,0.0414320409913879,0.0414320409913879,0.0423819592111537,0.0434657106650826,0.0434657106650826,0.0434657106650826,0.0434657106650826,0.0434657106650826,0.0456025716012738,0.0459587614414737,0.0459587614414737,0.0460748244854447,0.0483561465772661,0.0483561465772661,0.0493551110928339]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>logFC<\/th>\n      <th>logCPM<\/th>\n      <th>LR<\/th>\n      <th>PValue<\/th>\n      <th>FDR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"targets":1,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 3, 3, \",\", \".\");\n  }"},{"targets":2,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 3, 3, \",\", \".\");\n  }"},{"targets":3,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 3, 3, \",\", \".\");\n  }"},{"targets":4,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 3, 3, \",\", \".\");\n  }"},{"targets":5,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 3, 3, \",\", \".\");\n  }"},{"className":"dt-right","targets":[1,2,3,4,5]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":["options.columnDefs.0.render","options.columnDefs.1.render","options.columnDefs.2.render","options.columnDefs.3.render","options.columnDefs.4.render"],"jsHooks":[]}</script>
</div>
<p>CSV file of results available from <a href="../output/DEGs/reads/"><code>output/DEGs/reads/</code></a>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>gzout</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/connections.html'>gzfile</a></span><span class='op'>(</span>
  description <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span><span class='va'>outdir</span>, <span class='st'>"aggregated_tech_reps.DEGs.csv.gz"</span><span class='op'>)</span>,
  open <span class='op'>=</span> <span class='st'>"wb"</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/utils/write.table.html'>write.csv</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/topTags.html'>topTags</a></span><span class='op'>(</span><span class='va'>y_dgelrt</span>, n <span class='op'>=</span> <span class='cn'>Inf</span><span class='op'>)</span>,
  <span class='va'>gzout</span>,
  <span class='co'># NOTE: quote = TRUE needed because some fields contain commas.</span>
  quote <span class='op'>=</span> <span class='cn'>TRUE</span>,
  row.names <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/connections.html'>close</a></span><span class='op'>(</span><span class='va'>gzout</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>Figure <a href="#fig:heatmap-y">7</a> is a heatmap showing the expression of the DEGs.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/pkg/pheatmap/man/pheatmap.html'>pheatmap</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/pkg/SingleCellExperiment/man/assays.html'>cpm</a></span><span class='op'>(</span><span class='va'>y</span>, log <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>y_tt</span><span class='op'>)</span>, <span class='op'>]</span>,
  cluster_rows <span class='op'>=</span> <span class='cn'>TRUE</span>,
  cluster_cols <span class='op'>=</span> <span class='cn'>FALSE</span>,
  color <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/grDevices/palettes.html'>hcl.colors</a></span><span class='op'>(</span><span class='fl'>101</span>, <span class='st'>"Blue-Red 3"</span><span class='op'>)</span>,
  breaks <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span><span class='op'>(</span><span class='op'>-</span><span class='fl'>3</span>, <span class='fl'>3</span>, length.out <span class='op'>=</span> <span class='fl'>101</span><span class='op'>)</span>,
  scale <span class='op'>=</span> <span class='st'>"row"</span>,
  annotation_col <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>
    genotype <span class='op'>=</span> <span class='va'>y</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>smchd1_genotype_updated</span>,
    sex <span class='op'>=</span> <span class='va'>y</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>sex</span>,
    row.names <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>)</span><span class='op'>)</span>,
  annotation_colors <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>
    genotype <span class='op'>=</span> <span class='va'>smchd1_genotype_updated_colours</span>,
    sex <span class='op'>=</span> <span class='va'>sex_colours</span><span class='op'>)</span>,
  main <span class='op'>=</span> <span class='st'>"Male and female samples"</span>,
  fontsize <span class='op'>=</span> <span class='fl'>9</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:heatmap-y"></span>
<img src="C086_Kinkel.multi-sample_comparisons_files/figure-html5/heatmap-y-1.png" alt="Heatmap of upper-quartile-normalized logCPM values for the DEGs." width="624" />
<p class="caption">
Figure 7: Heatmap of upper-quartile-normalized logCPM values for the DEGs.
</p>
</div>
</div>
<h2 id="gene-set-tests">Gene set tests</h2>
<p>Gene set testing is commonly used to interpret the differential expression results in a biological context. Here we apply various gene set tests to the DEG lists.</p>
<h3 id="goana">goana</h3>
<p>We use the <code>goana()</code> function from the <em><a href="https://bioconductor.org/packages/3.12/limma">limma</a></em> R/Bioconductor package to test for over-representation of gene ontology (GO) terms in each DEG list.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>list_of_go</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/lapply.html'>lapply</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>y_contrast</span><span class='op'>)</span>, <span class='kw'>function</span><span class='op'>(</span><span class='va'>j</span><span class='op'>)</span> <span class='op'>{</span>
  <span class='va'>dgelrt</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/glmfit.html'>glmLRT</a></span><span class='op'>(</span><span class='va'>y_dgeglm</span>, contrast <span class='op'>=</span> <span class='va'>y_contrast</span><span class='op'>[</span>, <span class='va'>j</span><span class='op'>]</span><span class='op'>)</span>
  
  <span class='va'>go</span> <span class='op'>&lt;-</span> <span class='fu'>goana</span><span class='op'>(</span>
    de <span class='op'>=</span> <span class='va'>dgelrt</span>,
    geneid <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/lapply.html'>sapply</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/strsplit.html'>strsplit</a></span><span class='op'>(</span><span class='va'>dgelrt</span><span class='op'>$</span><span class='va'>genes</span><span class='op'>$</span><span class='va'>NCBI.ENTREZID</span>, <span class='st'>"; "</span><span class='op'>)</span>, <span class='st'>"[["</span>, <span class='fl'>1</span><span class='op'>)</span>,
    species <span class='op'>=</span> <span class='st'>"Mm"</span><span class='op'>)</span>
  <span class='va'>gzout</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/connections.html'>gzfile</a></span><span class='op'>(</span>
    description <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span><span class='va'>outdir</span>, <span class='st'>"aggregated_tech_reps.goana.csv.gz"</span><span class='op'>)</span>,
    open <span class='op'>=</span> <span class='st'>"wb"</span><span class='op'>)</span>
  <span class='fu'><a href='https://rdrr.io/r/utils/write.table.html'>write.csv</a></span><span class='op'>(</span>
    <span class='fu'>topGO</span><span class='op'>(</span><span class='va'>go</span>, number <span class='op'>=</span> <span class='cn'>Inf</span><span class='op'>)</span>,
    <span class='va'>gzout</span>,
    row.names <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
  <span class='fu'><a href='https://rdrr.io/r/base/connections.html'>close</a></span><span class='op'>(</span><span class='va'>gzout</span><span class='op'>)</span>
  <span class='va'>go</span>
<span class='op'>}</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>list_of_go</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>y_contrast</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>CSV files of the results are available from <a href="../output/DEGs/reads/"><code>output/DEGs/reads/</code></a>.</p>
<p>An example of the results are shown below.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'>topGO</span><span class='op'>(</span><span class='va'>list_of_go</span><span class='op'>[[</span><span class='fl'>1</span><span class='op'>]</span><span class='op'>]</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span>
    caption <span class='op'>=</span> <span class='st'>'`goana()` produces a table with a row for each GO term and the following columns: **Term** (GO term); **Ont** (ontology that the GO term belongs to. Possible values are "BP", "CC" and "MF"); **N** (number of genes in the GO term); **Up** (number of up-regulated differentially expressed genes); **Down** (number of down-regulated differentially expressed genes); **P.Up** (p-value for over-representation of GO term in up-regulated genes); **P.Down** (p-value for over-representation of GO term in down-regulated genes)'</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<table>
<caption><span id="tab:unnamed-chunk-24">Table 2: </span><code>goana()</code> produces a table with a row for each GO term and the following columns: <strong>Term</strong> (GO term); <strong>Ont</strong> (ontology that the GO term belongs to. Possible values are “BP,” “CC” and “MF”); <strong>N</strong> (number of genes in the GO term); <strong>Up</strong> (number of up-regulated differentially expressed genes); <strong>Down</strong> (number of down-regulated differentially expressed genes); <strong>P.Up</strong> (p-value for over-representation of GO term in up-regulated genes); <strong>P.Down</strong> (p-value for over-representation of GO term in down-regulated genes)</caption>
<colgroup>
<col style="width: 7%" />
<col style="width: 67%" />
<col style="width: 2%" />
<col style="width: 3%" />
<col style="width: 2%" />
<col style="width: 3%" />
<col style="width: 6%" />
<col style="width: 6%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;">Term</th>
<th style="text-align: left;">Ont</th>
<th style="text-align: right;">N</th>
<th style="text-align: right;">Up</th>
<th style="text-align: right;">Down</th>
<th style="text-align: right;">P.Up</th>
<th style="text-align: right;">P.Down</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"><a href="GO:0007610" class="uri">GO:0007610</a></td>
<td style="text-align: left;">behavior</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">330</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0000743</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="GO:0005886" class="uri">GO:0005886</a></td>
<td style="text-align: left;">plasma membrane</td>
<td style="text-align: left;">CC</td>
<td style="text-align: right;">2383</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">31</td>
<td style="text-align: right;">0.6545710</td>
<td style="text-align: right;">0.0000942</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="GO:0008277" class="uri">GO:0008277</a></td>
<td style="text-align: left;">regulation of G protein-coupled receptor signaling pathway</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">73</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0001331</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="GO:0071944" class="uri">GO:0071944</a></td>
<td style="text-align: left;">cell periphery</td>
<td style="text-align: left;">CC</td>
<td style="text-align: right;">2469</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">31</td>
<td style="text-align: right;">0.6758572</td>
<td style="text-align: right;">0.0001873</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="GO:0006928" class="uri">GO:0006928</a></td>
<td style="text-align: left;">movement of cell or subcellular component</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">1156</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">19</td>
<td style="text-align: right;">0.2716901</td>
<td style="text-align: right;">0.0002006</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="GO:0060956" class="uri">GO:0060956</a></td>
<td style="text-align: left;">endocardial cell differentiation</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0002710</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="GO:0060046" class="uri">GO:0060046</a></td>
<td style="text-align: left;">regulation of acrosome reaction</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0002710</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="GO:0003348" class="uri">GO:0003348</a></td>
<td style="text-align: left;">cardiac endothelial cell differentiation</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0004497</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="GO:0061343" class="uri">GO:0061343</a></td>
<td style="text-align: left;">cell adhesion involved in heart morphogenesis</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0004497</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="GO:0045597" class="uri">GO:0045597</a></td>
<td style="text-align: left;">positive regulation of cell differentiation</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">670</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">13</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0005275</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="GO:0072148" class="uri">GO:0072148</a></td>
<td style="text-align: left;">epithelial cell fate commitment</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0006717</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="GO:0062043" class="uri">GO:0062043</a></td>
<td style="text-align: left;">positive regulation of cardiac epithelial to mesenchymal transition</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0006717</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="GO:0062042" class="uri">GO:0062042</a></td>
<td style="text-align: left;">regulation of cardiac epithelial to mesenchymal transition</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0006717</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="GO:0010717" class="uri">GO:0010717</a></td>
<td style="text-align: left;">regulation of epithelial to mesenchymal transition</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">59</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0006757</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="GO:0051239" class="uri">GO:0051239</a></td>
<td style="text-align: left;">regulation of multicellular organismal process</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">1844</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">24</td>
<td style="text-align: right;">0.8308213</td>
<td style="text-align: right;">0.0008459</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="GO:0045153" class="uri">GO:0045153</a></td>
<td style="text-align: left;">electron transporter, transferring electrons within CoQH2-cytochrome c reductase complex activity</td>
<td style="text-align: left;">MF</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0008825</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="GO:0005887" class="uri">GO:0005887</a></td>
<td style="text-align: left;">integral component of plasma membrane</td>
<td style="text-align: left;">CC</td>
<td style="text-align: right;">452</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">0.3344946</td>
<td style="text-align: right;">0.0009219</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="GO:0060973" class="uri">GO:0060973</a></td>
<td style="text-align: left;">cell migration involved in heart development</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">7</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0009362</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="GO:0003198" class="uri">GO:0003198</a></td>
<td style="text-align: left;">epithelial to mesenchymal transition involved in endocardial cushion formation</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">7</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0009362</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="GO:0005523" class="uri">GO:0005523</a></td>
<td style="text-align: left;">tropomyosin binding</td>
<td style="text-align: left;">MF</td>
<td style="text-align: right;">7</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0009362</td>
</tr>
</tbody>
</table>
</div>
<h3 id="kegga">kegga</h3>
<p>We use the <code>kegga()</code> function from the <em><a href="https://bioconductor.org/packages/3.12/limma">limma</a></em> R/Bioconductor package to test for over-representation of KEGG pathways in each DEG list.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>list_of_kegg</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/lapply.html'>lapply</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>y_contrast</span><span class='op'>)</span>, <span class='kw'>function</span><span class='op'>(</span><span class='va'>j</span><span class='op'>)</span> <span class='op'>{</span>
  <span class='va'>dgelrt</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/glmfit.html'>glmLRT</a></span><span class='op'>(</span><span class='va'>y_dgeglm</span>, contrast <span class='op'>=</span> <span class='va'>y_contrast</span><span class='op'>[</span>, <span class='va'>j</span><span class='op'>]</span><span class='op'>)</span>
  
  <span class='va'>kegg</span> <span class='op'>&lt;-</span> <span class='fu'>kegga</span><span class='op'>(</span>
    de <span class='op'>=</span> <span class='va'>dgelrt</span>,
    geneid <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/lapply.html'>sapply</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/strsplit.html'>strsplit</a></span><span class='op'>(</span><span class='va'>dgelrt</span><span class='op'>$</span><span class='va'>genes</span><span class='op'>$</span><span class='va'>NCBI.ENTREZID</span>, <span class='st'>"; "</span><span class='op'>)</span>, <span class='st'>"[["</span>, <span class='fl'>1</span><span class='op'>)</span>,
    species <span class='op'>=</span> <span class='st'>"Mm"</span><span class='op'>)</span>
  <span class='va'>gzout</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/connections.html'>gzfile</a></span><span class='op'>(</span>
    description <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span><span class='va'>outdir</span>, <span class='st'>"aggregated_tech_reps.kegga.csv.gz"</span><span class='op'>)</span>,
    open <span class='op'>=</span> <span class='st'>"wb"</span><span class='op'>)</span>
  <span class='fu'><a href='https://rdrr.io/r/utils/write.table.html'>write.csv</a></span><span class='op'>(</span>
    <span class='fu'>topKEGG</span><span class='op'>(</span><span class='va'>kegg</span>, number <span class='op'>=</span> <span class='cn'>Inf</span><span class='op'>)</span>,
    <span class='va'>gzout</span>,
    row.names <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
  <span class='fu'><a href='https://rdrr.io/r/base/connections.html'>close</a></span><span class='op'>(</span><span class='va'>gzout</span><span class='op'>)</span>
  <span class='va'>kegg</span>
<span class='op'>}</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>list_of_kegg</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>y_contrast</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>CSV files of the results are available from <a href="../output/DEGs/reads/"><code>output/DEGs/reads/</code></a>.</p>
<p>An example of the results are shown below.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'>topKEGG</span><span class='op'>(</span><span class='va'>list_of_kegg</span><span class='op'>[[</span><span class='fl'>1</span><span class='op'>]</span><span class='op'>]</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span>
    caption <span class='op'>=</span> <span class='st'>'`kegga()` produces a table with a row for each KEGG pathway ID and the following columns: **Pathway** (KEGG pathway); **N** (number of genes in the GO term); **Up** (number of up-regulated differentially expressed genes); **Down** (number of down-regulated differentially expressed genes); **P.Up** (p-value for over-representation of KEGG pathway in up-regulated genes); **P.Down** (p-value for over-representation of KEGG pathway in down-regulated genes)'</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<table style="width:100%;">
<caption><span id="tab:unnamed-chunk-26">Table 3: </span><code>kegga()</code> produces a table with a row for each KEGG pathway ID and the following columns: <strong>Pathway</strong> (KEGG pathway); <strong>N</strong> (number of genes in the GO term); <strong>Up</strong> (number of up-regulated differentially expressed genes); <strong>Down</strong> (number of down-regulated differentially expressed genes); <strong>P.Up</strong> (p-value for over-representation of KEGG pathway in up-regulated genes); <strong>P.Down</strong> (p-value for over-representation of KEGG pathway in down-regulated genes)</caption>
<colgroup>
<col style="width: 14%" />
<col style="width: 53%" />
<col style="width: 4%" />
<col style="width: 3%" />
<col style="width: 5%" />
<col style="width: 10%" />
<col style="width: 10%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;">Pathway</th>
<th style="text-align: right;">N</th>
<th style="text-align: right;">Up</th>
<th style="text-align: right;">Down</th>
<th style="text-align: right;">P.Up</th>
<th style="text-align: right;">P.Down</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">path:mmu04928</td>
<td style="text-align: left;">Parathyroid hormone synthesis, secretion and action</td>
<td style="text-align: right;">72</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0014987</td>
</tr>
<tr class="even">
<td style="text-align: left;">path:mmu04919</td>
<td style="text-align: left;">Thyroid hormone signaling pathway</td>
<td style="text-align: right;">95</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0041173</td>
</tr>
<tr class="odd">
<td style="text-align: left;">path:mmu04724</td>
<td style="text-align: left;">Glutamatergic synapse</td>
<td style="text-align: right;">58</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0073758</td>
</tr>
<tr class="even">
<td style="text-align: left;">path:mmu00565</td>
<td style="text-align: left;">Ether lipid metabolism</td>
<td style="text-align: right;">24</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0117017</td>
</tr>
<tr class="odd">
<td style="text-align: left;">path:mmu04915</td>
<td style="text-align: left;">Estrogen signaling pathway</td>
<td style="text-align: right;">80</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0176377</td>
</tr>
<tr class="even">
<td style="text-align: left;">path:mmu04929</td>
<td style="text-align: left;">GnRH secretion</td>
<td style="text-align: right;">37</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0266607</td>
</tr>
<tr class="odd">
<td style="text-align: left;">path:mmu04072</td>
<td style="text-align: left;">Phospholipase D signaling pathway</td>
<td style="text-align: right;">101</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0323083</td>
</tr>
<tr class="even">
<td style="text-align: left;">path:mmu05032</td>
<td style="text-align: left;">Morphine addiction</td>
<td style="text-align: right;">42</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0337172</td>
</tr>
<tr class="odd">
<td style="text-align: left;">path:mmu05200</td>
<td style="text-align: left;">Pathways in cancer</td>
<td style="text-align: right;">352</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0340336</td>
</tr>
<tr class="even">
<td style="text-align: left;">path:mmu04260</td>
<td style="text-align: left;">Cardiac muscle contraction</td>
<td style="text-align: right;">47</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0407222</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="odd">
<td style="text-align: left;">path:mmu04726</td>
<td style="text-align: left;">Serotonergic synapse</td>
<td style="text-align: right;">52</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0497017</td>
</tr>
<tr class="even">
<td style="text-align: left;">path:mmu04713</td>
<td style="text-align: left;">Circadian entrainment</td>
<td style="text-align: right;">53</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0514257</td>
</tr>
<tr class="odd">
<td style="text-align: left;">path:mmu04621</td>
<td style="text-align: left;">NOD-like receptor signaling pathway</td>
<td style="text-align: right;">123</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0527650</td>
</tr>
<tr class="even">
<td style="text-align: left;">path:mmu04925</td>
<td style="text-align: left;">Aldosterone synthesis and secretion</td>
<td style="text-align: right;">55</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0549374</td>
</tr>
<tr class="odd">
<td style="text-align: left;">path:mmu05206</td>
<td style="text-align: left;">MicroRNAs in cancer</td>
<td style="text-align: right;">129</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0592099</td>
</tr>
<tr class="even">
<td style="text-align: left;">path:mmu05322</td>
<td style="text-align: left;">Systemic lupus erythematosus</td>
<td style="text-align: right;">75</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0642668</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="odd">
<td style="text-align: left;">path:mmu05165</td>
<td style="text-align: left;">Human papillomavirus infection</td>
<td style="text-align: right;">221</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0657841</td>
</tr>
<tr class="even">
<td style="text-align: left;">path:mmu04014</td>
<td style="text-align: left;">Ras signaling pathway</td>
<td style="text-align: right;">136</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0671756</td>
</tr>
<tr class="odd">
<td style="text-align: left;">path:mmu05211</td>
<td style="text-align: left;">Renal cell carcinoma</td>
<td style="text-align: right;">62</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0678581</td>
</tr>
<tr class="even">
<td style="text-align: left;">path:mmu00564</td>
<td style="text-align: left;">Glycerophospholipid metabolism</td>
<td style="text-align: right;">65</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0736747</td>
</tr>
</tbody>
</table>
</div>
<h3 id="camera-with-msigdb-gene-sets">camera with MSigDB gene sets</h3>
<p>We use the <code>camera()</code> function from the <em><a href="https://bioconductor.org/packages/3.12/limma">limma</a></em> R/Bioconductor package to test whether a set of genes is highly ranked relative to other genes in terms of differential expression, accounting for inter-gene correlation. Specifically, we test using gene sets from <a href="https://www.gsea-msigdb.org/gsea/msigdb">MSigDB</a>, namely:</p>
<aside>
We download these gene sets from <a href="http://bioinf.wehi.edu.au/MSigDB/index.html">http://bioinf.wehi.edu.au/MSigDB/index.html</a>
</aside>
<ul>
<li><strong>H</strong>: <strong>hallmark gene sets</strong> are coherently expressed signatures derived by aggregating many MSigDB gene sets to represent well-defined biological states or processes.</li>
<li><strong>C2</strong>: <strong>curated gene sets</strong> from online pathway databases, publications in PubMed, and knowledge of domain experts.</li>
</ul>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='co'># NOTE: Using BiocFileCache to avoid re-downloading these gene sets everytime </span>
<span class='co'>#       the report is rendered.</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'>BiocFileCache</span><span class='op'>)</span>
<span class='va'>bfc</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html'>BiocFileCache</a></span><span class='op'>(</span><span class='op'>)</span>
<span class='co'># NOTE: Creating list of gene sets in this slightly convoluted way so that each </span>
<span class='co'>#       gene set name is prepended by its origin (e.g. H, C2, or C7).</span>
<span class='va'>msigdb</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/do.call.html'>do.call</a></span><span class='op'>(</span>
  <span class='va'>c</span>, 
  <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>
    H <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/readRDS.html'>readRDS</a></span><span class='op'>(</span>
      <span class='fu'><a href='https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html'>bfcrpath</a></span><span class='op'>(</span>
        <span class='va'>bfc</span>,
        <span class='st'>"http://bioinf.wehi.edu.au/MSigDB/v7.1/Mm.h.all.v7.1.entrez.rds"</span><span class='op'>)</span><span class='op'>)</span>,
    C2 <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/readRDS.html'>readRDS</a></span><span class='op'>(</span>
      <span class='fu'><a href='https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html'>bfcrpath</a></span><span class='op'>(</span>
        <span class='va'>bfc</span>, 
        <span class='st'>"http://bioinf.wehi.edu.au/MSigDB/v7.1/Mm.c2.all.v7.1.entrez.rds"</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>

<span class='va'>y_idx</span> <span class='op'>&lt;-</span> <span class='fu'>ids2indices</span><span class='op'>(</span>
  <span class='va'>msigdb</span>, 
  id <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/lapply.html'>sapply</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/strsplit.html'>strsplit</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>$</span><span class='va'>genes</span><span class='op'>$</span><span class='va'>NCBI.ENTREZID</span>, <span class='st'>"; "</span><span class='op'>)</span>, <span class='st'>"[["</span>, <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>list_of_camera</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/lapply.html'>lapply</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>y_contrast</span><span class='op'>)</span>, <span class='kw'>function</span><span class='op'>(</span><span class='va'>j</span><span class='op'>)</span> <span class='op'>{</span>
  <span class='va'>cam</span> <span class='op'>&lt;-</span> <span class='fu'>camera</span><span class='op'>(</span>
    y <span class='op'>=</span> <span class='va'>y</span>, 
    index <span class='op'>=</span> <span class='va'>y_idx</span>,
    design <span class='op'>=</span> <span class='va'>y_design</span>, 
    contrast <span class='op'>=</span> <span class='va'>y_contrast</span><span class='op'>[</span>, <span class='va'>j</span><span class='op'>]</span><span class='op'>)</span>
  <span class='va'>gzout</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/connections.html'>gzfile</a></span><span class='op'>(</span>
    description <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span><span class='va'>outdir</span>, <span class='st'>"aggregated_tech_reps.camera.csv.gz"</span><span class='op'>)</span>,
    open <span class='op'>=</span> <span class='st'>"wb"</span><span class='op'>)</span>
  <span class='fu'><a href='https://rdrr.io/r/utils/write.table.html'>write.csv</a></span><span class='op'>(</span>
    <span class='va'>cam</span>,
    <span class='va'>gzout</span>,
    row.names <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
  <span class='fu'><a href='https://rdrr.io/r/base/connections.html'>close</a></span><span class='op'>(</span><span class='va'>gzout</span><span class='op'>)</span>
  <span class='va'>cam</span>
<span class='op'>}</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>list_of_camera</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>y_contrast</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>CSV files of the results are available from <a href="../output/DEGs/reads/"><code>output/DEGs/reads</code></a>.</p>
<p>An example of the results are shown below.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span><span class='op'>(</span><span class='va'>list_of_camera</span><span class='op'>[[</span><span class='fl'>1</span><span class='op'>]</span><span class='op'>]</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span>
    caption <span class='op'>=</span> <span class='st'>'`camera()` produces a table with a row for each gene set (prepended by which MSigDB collection it comes from) and the following columns: **NGenes** (number of genes in set); **Direction** (direction of change); **PValue** (two-tailed p-value); **FDR** (Benjamini and Hochberg FDR adjusted p-value).'</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<table>
<caption><span id="tab:unnamed-chunk-28">Table 4: </span><code>camera()</code> produces a table with a row for each gene set (prepended by which MSigDB collection it comes from) and the following columns: <strong>NGenes</strong> (number of genes in set); <strong>Direction</strong> (direction of change); <strong>PValue</strong> (two-tailed p-value); <strong>FDR</strong> (Benjamini and Hochberg FDR adjusted p-value).</caption>
<colgroup>
<col style="width: 79%" />
<col style="width: 4%" />
<col style="width: 6%" />
<col style="width: 4%" />
<col style="width: 5%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">NGenes</th>
<th style="text-align: left;">Direction</th>
<th style="text-align: right;">PValue</th>
<th style="text-align: right;">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">C2.KEGG_RIBOSOME</td>
<td style="text-align: right;">102</td>
<td style="text-align: left;">Up</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">6.0e-07</td>
</tr>
<tr class="even">
<td style="text-align: left;">C2.REACTOME_EUKARYOTIC_TRANSLATION_ELONGATION</td>
<td style="text-align: right;">107</td>
<td style="text-align: left;">Up</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2.3e-06</td>
</tr>
<tr class="odd">
<td style="text-align: left;">C2.REACTOME_SRP_DEPENDENT_COTRANSLATIONAL_PROTEIN_TARGETING_TO_MEMBRANE</td>
<td style="text-align: right;">130</td>
<td style="text-align: left;">Up</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">4.4e-06</td>
</tr>
<tr class="even">
<td style="text-align: left;">C2.REACTOME_RESPIRATORY_ELECTRON_TRANSPORT_ATP_SYNTHESIS_BY_CHEMIOSMOTIC_COUPLING_AND_HEAT_PRODUCTION_BY_UNCOUPLING_PROTEINS</td>
<td style="text-align: right;">120</td>
<td style="text-align: left;">Up</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">4.4e-06</td>
</tr>
<tr class="odd">
<td style="text-align: left;">C2.REACTOME_RESPONSE_OF_EIF2AK4_GCN2_TO_AMINO_ACID_DEFICIENCY</td>
<td style="text-align: right;">115</td>
<td style="text-align: left;">Up</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">4.4e-06</td>
</tr>
<tr class="even">
<td style="text-align: left;">C2.REACTOME_EUKARYOTIC_TRANSLATION_INITIATION</td>
<td style="text-align: right;">139</td>
<td style="text-align: left;">Up</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">4.4e-06</td>
</tr>
</tbody>
</table>
</div>
<h3 id="sarahs-gene-lists">Sarah’s gene lists</h3>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>y_X</span> <span class='op'>&lt;-</span> <span class='va'>y</span><span class='op'>$</span><span class='va'>genes</span><span class='op'>$</span><span class='va'>ENSEMBL.SEQNAME</span> <span class='op'>==</span> <span class='st'>"X"</span>
<span class='va'>y_protocadherin</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/grep.html'>grepl</a></span><span class='op'>(</span><span class='st'>"Pcdh"</span>, <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>y_pw</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>)</span> <span class='op'>%in%</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Ndn"</span>, <span class='st'>"Mkrn3"</span>, <span class='st'>"Peg12"</span><span class='op'>)</span>
<span class='va'>y_male</span> <span class='op'>&lt;-</span> <span class='va'>male_het_vs_del</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Probe"</span>, <span class='st'>"Differential expression"</span><span class='op'>)</span><span class='op'>]</span>
<span class='va'>y_female</span> <span class='op'>&lt;-</span> <span class='va'>female_het_vs_del</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Probe"</span>, <span class='st'>"Differential expression"</span><span class='op'>)</span><span class='op'>]</span>
<span class='va'>y_sun</span> <span class='op'>&lt;-</span> <span class='va'>sun_2014</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"geneSymbol"</span>, <span class='st'>"log2FC"</span><span class='op'>)</span><span class='op'>]</span>
<span class='va'>y_chambers</span> <span class='op'>&lt;-</span> <span class='va'>chambers_2007</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Gene Symbol"</span>, <span class='st'>"Fold Difference"</span><span class='op'>)</span><span class='op'>]</span>
<span class='va'>y_chambers</span><span class='op'>$</span><span class='va'>logFC</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log2</a></span><span class='op'>(</span><span class='va'>y_chambers</span><span class='op'>$</span><span class='va'>`Fold Difference`</span><span class='op'>)</span>
<span class='va'>y_chambers</span><span class='op'>$</span><span class='va'>`Fold Difference`</span> <span class='op'>&lt;-</span> <span class='cn'>NULL</span>
<span class='va'>y_chambers</span> <span class='op'>&lt;-</span> <span class='va'>y_chambers</span><span class='op'>[</span><span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/NA.html'>is.na</a></span><span class='op'>(</span><span class='va'>y_chambers</span><span class='op'>$</span><span class='va'>`Gene Symbol`</span><span class='op'>)</span>, <span class='op'>]</span>
<span class='va'>y_chambers</span> <span class='op'>&lt;-</span> <span class='va'>y_chambers</span><span class='op'>[</span><span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/duplicated.html'>duplicated</a></span><span class='op'>(</span><span class='va'>y_chambers</span><span class='op'>$</span><span class='va'>`Gene Symbol`</span><span class='op'>)</span>, <span class='op'>]</span>
<span class='va'>y_gazit</span> <span class='op'>&lt;-</span> <span class='va'>gazit_2013</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Symbol"</span><span class='op'>)</span>, drop <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>]</span>
<span class='va'>y_gazit</span><span class='op'>$</span><span class='va'>Symbol</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/base/substr.html'>substring</a></span><span class='op'>(</span><span class='va'>y_gazit</span><span class='op'>$</span><span class='va'>Symbol</span>, <span class='fl'>1</span>, <span class='fl'>1</span><span class='op'>)</span>, 
  <span class='fu'><a href='https://rdrr.io/r/base/chartr.html'>tolower</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/substr.html'>substring</a></span><span class='op'>(</span><span class='va'>y_gazit</span><span class='op'>$</span><span class='va'>Symbol</span>, <span class='fl'>2</span><span class='op'>)</span><span class='op'>)</span>,
  sep <span class='op'>=</span> <span class='st'>""</span><span class='op'>)</span>
<span class='va'>y_gazit</span><span class='op'>$</span><span class='va'>logFC</span> <span class='op'>&lt;-</span> <span class='fl'>1</span>
<span class='va'>y_chambers_gazit</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>
  Symbol <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='va'>y_chambers</span><span class='op'>$</span><span class='va'>`Gene Symbol`</span>, <span class='va'>y_gazit</span><span class='op'>$</span><span class='va'>Symbol</span><span class='op'>)</span>,
  logFC <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='va'>y_chambers</span><span class='op'>$</span><span class='va'>logFC</span>, <span class='va'>y_gazit</span><span class='op'>$</span><span class='va'>logFC</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>y_chambers_gazit</span> <span class='op'>&lt;-</span> <span class='va'>y_chambers_gazit</span><span class='op'>[</span><span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/duplicated.html'>duplicated</a></span><span class='op'>(</span><span class='va'>y_chambers_gazit</span><span class='op'>$</span><span class='va'>Symbol</span><span class='op'>)</span>, <span class='op'>]</span>
<span class='va'>y_venezia_proliferation</span> <span class='op'>&lt;-</span> <span class='va'>venezia_proliferation</span><span class='op'>[</span>
  , <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Gene.Symbol"</span>, <span class='st'>"Log2.Fold.Change..FL.HSC.vs.Adult.HSC.."</span><span class='op'>)</span><span class='op'>]</span>
<span class='va'>y_venezia_proliferation</span> <span class='op'>&lt;-</span> <span class='va'>y_venezia_proliferation</span><span class='op'>[</span>
  <span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/duplicated.html'>duplicated</a></span><span class='op'>(</span><span class='va'>y_venezia_proliferation</span><span class='op'>$</span><span class='va'>`Gene.Symbol`</span><span class='op'>)</span>, <span class='op'>]</span>
<span class='va'>y_venezia_quiescence</span> <span class='op'>&lt;-</span> <span class='va'>venezia_quiescence</span><span class='op'>[</span>
  , <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Gene.Symbol"</span>, <span class='st'>"Log2.Fold.Change..FL.HSC.vs.Adult.HSC.."</span><span class='op'>)</span><span class='op'>]</span>
<span class='va'>y_venezia_quiescence</span> <span class='op'>&lt;-</span> <span class='va'>y_venezia_quiescence</span><span class='op'>[</span>
  <span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/duplicated.html'>duplicated</a></span><span class='op'>(</span><span class='va'>y_venezia_quiescence</span><span class='op'>$</span><span class='va'>`Gene.Symbol`</span><span class='op'>)</span>, <span class='op'>]</span>
<span class='co'># TODO: Clarify how this gene set is to be used.</span>
<span class='co'># y_smchd1_femBcells &lt;- smchd1_femBcells[</span>
<span class='co'>#   smchd1_femBcells$`FDR (EdgeR stats p&lt;1.0 after correction)` &lt; 0.05, </span>
<span class='co'>#   c("Feature", "Log2 Fold Change (EdgeR stats p&lt;1.0 after correction)")]</span>
<span class='co'># y_smchd1_femBcells &lt;- y_smchd1_femBcells[</span>
<span class='co'>#   !duplicated(y_smchd1_femBcells$Feature), ]</span>
<span class='va'>y_smchd1_femBcells</span> <span class='op'>&lt;-</span> <span class='va'>smchd1_femBcells</span><span class='op'>[</span>
    <span class='co'># smchd1_femBcells$Chromosome == "X", </span>
  <span class='va'>smchd1_femBcells</span><span class='op'>$</span><span class='va'>Chromosome</span> <span class='op'>==</span> <span class='st'>"X"</span> <span class='op'>&amp;</span>
    <span class='va'>smchd1_femBcells</span><span class='op'>$</span><span class='va'>`FDR (EdgeR stats p&lt;1.0 after correction)`</span> <span class='op'>&lt;</span> <span class='fl'>0.05</span>,
  <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Feature"</span>, <span class='st'>"Log2 Fold Change (EdgeR stats p&lt;1.0 after correction)"</span><span class='op'>)</span><span class='op'>]</span>
<span class='va'>y_smchd1_femBcells</span> <span class='op'>&lt;-</span> <span class='va'>y_smchd1_femBcells</span><span class='op'>[</span>
  <span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/duplicated.html'>duplicated</a></span><span class='op'>(</span><span class='va'>y_smchd1_femBcells</span><span class='op'>$</span><span class='va'>Feature</span><span class='op'>)</span>, <span class='op'>]</span>


<span class='va'>y_index</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>
  `X-linked` <span class='op'>=</span> <span class='va'>y_X</span>,
  Protocadherin <span class='op'>=</span> <span class='va'>y_protocadherin</span>,
  `Prader-Willi` <span class='op'>=</span> <span class='va'>y_pw</span>,
  `Male: Het vs. Del` <span class='op'>=</span> <span class='va'>y_male</span>,
  `Female: Het vs. Del` <span class='op'>=</span> <span class='va'>y_female</span>,
  `Sun (2014)` <span class='op'>=</span> <span class='va'>y_sun</span>,
  `Chambers (2007)` <span class='op'>=</span> <span class='va'>y_chambers</span>,
  `Gazit (2013)` <span class='op'>=</span> <span class='va'>y_gazit</span>,
  `Chambers (2007) &amp; Gazit (2013)` <span class='op'>=</span> <span class='va'>y_chambers_gazit</span>,
  `Proliferation Group Venezia (2004)` <span class='op'>=</span> <span class='va'>y_venezia_proliferation</span>,
  `Quiescence Group Venezia (2004)` <span class='op'>=</span> <span class='va'>y_venezia_quiescence</span>,
  `Smchd1 female B cells (chrX)` <span class='op'>=</span> <span class='va'>y_smchd1_femBcells</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>Sarah requested testing a number of gene sets (see <a href="#gene-sets">Gene sets</a>).</p>
<p>We use the <code>fry()</code> function from the <em><a href="https://bioconductor.org/packages/3.12/edgeR">edgeR</a></em> R/Bioconductor package to perform a self-contained gene set test against the null hypothesis that none of the genes in the set are differentially expressed.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'>fry</span><span class='op'>(</span>
  <span class='va'>y</span>,
  index <span class='op'>=</span> <span class='va'>y_index</span>,
  design <span class='op'>=</span> <span class='va'>y_design</span>,
  contrast <span class='op'>=</span> <span class='va'>y_contrast</span>, 
  sort <span class='op'>=</span> <span class='st'>"none"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
    <span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span>
      caption <span class='op'>=</span> <span class='st'>"Results of applying `fry()` to the RNA-seq differential expression results using the supplied gene sets. A significant 'Up' P-value means that the differential expression results found in the RNA-seq data are positively correlated with the expression signature from the corresponding gene set. Conversely, a significant 'Down' P-value means that the differential expression log-fold-changes are negatively correlated with the expression signature from the corresponding gene set. A significant 'Mixed' P-value means that the genes in the set tend to be differentially expressed without regard for direction."</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<table style="width:100%;">
<caption><span id="tab:unnamed-chunk-30">Table 5: </span>Results of applying <code>fry()</code> to the RNA-seq differential expression results using the supplied gene sets. A significant ‘Up’ P-value means that the differential expression results found in the RNA-seq data are positively correlated with the expression signature from the corresponding gene set. Conversely, a significant ‘Down’ P-value means that the differential expression log-fold-changes are negatively correlated with the expression signature from the corresponding gene set. A significant ‘Mixed’ P-value means that the genes in the set tend to be differentially expressed without regard for direction.</caption>
<colgroup>
<col style="width: 36%" />
<col style="width: 7%" />
<col style="width: 10%" />
<col style="width: 10%" />
<col style="width: 10%" />
<col style="width: 13%" />
<col style="width: 10%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">NGenes</th>
<th style="text-align: left;">Direction</th>
<th style="text-align: right;">PValue</th>
<th style="text-align: right;">FDR</th>
<th style="text-align: right;">PValue.Mixed</th>
<th style="text-align: right;">FDR.Mixed</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">X-linked</td>
<td style="text-align: right;">388</td>
<td style="text-align: left;">Down</td>
<td style="text-align: right;">0.4732142</td>
<td style="text-align: right;">0.5115083</td>
<td style="text-align: right;">0.0000240</td>
<td style="text-align: right;">0.0002881</td>
</tr>
<tr class="even">
<td style="text-align: left;">Protocadherin</td>
<td style="text-align: right;">4</td>
<td style="text-align: left;">Down</td>
<td style="text-align: right;">0.1818142</td>
<td style="text-align: right;">0.3636284</td>
<td style="text-align: right;">0.0359534</td>
<td style="text-align: right;">0.0616345</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Prader-Willi</td>
<td style="text-align: right;">3</td>
<td style="text-align: left;">Up</td>
<td style="text-align: right;">0.2856524</td>
<td style="text-align: right;">0.4896898</td>
<td style="text-align: right;">0.4549877</td>
<td style="text-align: right;">0.4870635</td>
</tr>
<tr class="even">
<td style="text-align: left;">Male: Het vs. Del</td>
<td style="text-align: right;">16</td>
<td style="text-align: left;">Up</td>
<td style="text-align: right;">0.3944671</td>
<td style="text-align: right;">0.5115083</td>
<td style="text-align: right;">0.1453178</td>
<td style="text-align: right;">0.1937570</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Female: Het vs. Del</td>
<td style="text-align: right;">45</td>
<td style="text-align: left;">Up</td>
<td style="text-align: right;">0.4297682</td>
<td style="text-align: right;">0.5115083</td>
<td style="text-align: right;">0.2342053</td>
<td style="text-align: right;">0.2810464</td>
</tr>
<tr class="even">
<td style="text-align: left;">Sun (2014)</td>
<td style="text-align: right;">1997</td>
<td style="text-align: left;">Down</td>
<td style="text-align: right;">0.3728438</td>
<td style="text-align: right;">0.5115083</td>
<td style="text-align: right;">0.0004929</td>
<td style="text-align: right;">0.0011829</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Chambers (2007)</td>
<td style="text-align: right;">208</td>
<td style="text-align: left;">Down</td>
<td style="text-align: right;">0.0036999</td>
<td style="text-align: right;">0.0266860</td>
<td style="text-align: right;">0.0001395</td>
<td style="text-align: right;">0.0004367</td>
</tr>
<tr class="even">
<td style="text-align: left;">Gazit (2013)</td>
<td style="text-align: right;">249</td>
<td style="text-align: left;">Down</td>
<td style="text-align: right;">0.0430404</td>
<td style="text-align: right;">0.1291213</td>
<td style="text-align: right;">0.0094767</td>
<td style="text-align: right;">0.0189534</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Chambers (2007) &amp; Gazit (2013)</td>
<td style="text-align: right;">404</td>
<td style="text-align: left;">Down</td>
<td style="text-align: right;">0.0066715</td>
<td style="text-align: right;">0.0266860</td>
<td style="text-align: right;">0.0001456</td>
<td style="text-align: right;">0.0004367</td>
</tr>
<tr class="even">
<td style="text-align: left;">Proliferation Group Venezia (2004)</td>
<td style="text-align: right;">365</td>
<td style="text-align: left;">Up</td>
<td style="text-align: right;">0.0053007</td>
<td style="text-align: right;">0.0266860</td>
<td style="text-align: right;">0.0001244</td>
<td style="text-align: right;">0.0004367</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Quiescence Group Venezia (2004)</td>
<td style="text-align: right;">382</td>
<td style="text-align: left;">Up</td>
<td style="text-align: right;">0.0705588</td>
<td style="text-align: right;">0.1693412</td>
<td style="text-align: right;">0.0475334</td>
<td style="text-align: right;">0.0713001</td>
</tr>
<tr class="even">
<td style="text-align: left;">Smchd1 female B cells (chrX)</td>
<td style="text-align: right;">6</td>
<td style="text-align: left;">Up</td>
<td style="text-align: right;">0.5115083</td>
<td style="text-align: right;">0.5115083</td>
<td style="text-align: right;">0.4870635</td>
<td style="text-align: right;">0.4870635</td>
</tr>
</tbody>
</table>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/r/graphics/par.html'>par</a></span><span class='op'>(</span>mfrow <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span><span class='op'>(</span><span class='va'>y_index</span><span class='op'>)</span>, <span class='fl'>2</span><span class='op'>)</span><span class='op'>)</span>
<span class='kw'>for</span> <span class='op'>(</span><span class='va'>n</span> <span class='kw'>in</span> <span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>y_index</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>{</span>
  <span class='kw'>if</span> <span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>is.data.frame</a></span><span class='op'>(</span><span class='va'>y_index</span><span class='op'>[[</span><span class='va'>n</span><span class='op'>]</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>{</span>
    <span class='va'>y_status</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/rep.html'>rep</a></span><span class='op'>(</span><span class='fl'>0</span>, <span class='fu'><a href='https://rdrr.io/r/base/nrow.html'>nrow</a></span><span class='op'>(</span><span class='va'>y_dgelrt</span><span class='op'>)</span><span class='op'>)</span>
    <span class='va'>y_status</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>)</span> <span class='op'>%in%</span> <span class='va'>y_index</span><span class='op'>[[</span><span class='va'>n</span><span class='op'>]</span><span class='op'>]</span><span class='op'>[</span>, <span class='fl'>1</span><span class='op'>]</span><span class='op'>]</span> <span class='op'>&lt;-</span>
      <span class='fu'><a href='https://rdrr.io/r/base/sign.html'>sign</a></span><span class='op'>(</span><span class='va'>y_index</span><span class='op'>[[</span><span class='va'>n</span><span class='op'>]</span><span class='op'>]</span><span class='op'>[</span><span class='va'>y_index</span><span class='op'>[[</span><span class='va'>n</span><span class='op'>]</span><span class='op'>]</span><span class='op'>[</span>, <span class='fl'>1</span><span class='op'>]</span> <span class='op'>%in%</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>)</span>, <span class='fl'>2</span><span class='op'>]</span><span class='op'>)</span>
    <span class='fu'>plotMD</span><span class='op'>(</span>
      <span class='va'>y_dgelrt</span>,
      status <span class='op'>=</span> <span class='va'>y_status</span>,
      hl.col <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"red"</span>, <span class='st'>"blue"</span><span class='op'>)</span>,
      legend <span class='op'>=</span> <span class='st'>"bottomright"</span>,
      main <span class='op'>=</span> <span class='va'>n</span><span class='op'>)</span>
    <span class='fu'><a href='https://rdrr.io/r/graphics/abline.html'>abline</a></span><span class='op'>(</span>h <span class='op'>=</span> <span class='fl'>0</span>, col <span class='op'>=</span> <span class='st'>"darkgrey"</span><span class='op'>)</span>
  <span class='op'>}</span> <span class='kw'>else</span> <span class='op'>{</span>
    <span class='va'>y_status</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/rep.html'>rep</a></span><span class='op'>(</span><span class='st'>"Other"</span>, <span class='fu'><a href='https://rdrr.io/r/base/nrow.html'>nrow</a></span><span class='op'>(</span><span class='va'>y_dgelrt</span><span class='op'>)</span><span class='op'>)</span>
    <span class='va'>y_status</span><span class='op'>[</span><span class='va'>y_index</span><span class='op'>[[</span><span class='va'>n</span><span class='op'>]</span><span class='op'>]</span><span class='op'>]</span> <span class='op'>&lt;-</span> <span class='va'>n</span>
    <span class='fu'>plotMD</span><span class='op'>(</span>
      <span class='va'>y_dgelrt</span>,
      status <span class='op'>=</span> <span class='va'>y_status</span>,
      values <span class='op'>=</span> <span class='va'>n</span>,
      hl.col <span class='op'>=</span> <span class='st'>"red"</span>,
      legend <span class='op'>=</span> <span class='st'>"bottomright"</span>,
      main <span class='op'>=</span> <span class='va'>n</span><span class='op'>)</span>
    <span class='fu'><a href='https://rdrr.io/r/graphics/abline.html'>abline</a></span><span class='op'>(</span>h <span class='op'>=</span> <span class='fl'>0</span>, col <span class='op'>=</span> <span class='st'>"darkgrey"</span><span class='op'>)</span>
  <span class='op'>}</span>
  
  <span class='kw'>if</span> <span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>is.data.frame</a></span><span class='op'>(</span><span class='va'>y_index</span><span class='op'>[[</span><span class='va'>n</span><span class='op'>]</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>{</span>
    <span class='fu'>barcodeplot</span><span class='op'>(</span>
      <span class='va'>y_dgelrt</span><span class='op'>$</span><span class='va'>table</span><span class='op'>$</span><span class='va'>logFC</span>,
      <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>)</span> <span class='op'>%in%</span> <span class='va'>y_index</span><span class='op'>[[</span><span class='va'>n</span><span class='op'>]</span><span class='op'>]</span><span class='op'>[</span>, <span class='fl'>1</span><span class='op'>]</span>,
      gene.weights <span class='op'>=</span> <span class='va'>y_index</span><span class='op'>[[</span><span class='va'>n</span><span class='op'>]</span><span class='op'>]</span><span class='op'>[</span>
        <span class='va'>y_index</span><span class='op'>[[</span><span class='va'>n</span><span class='op'>]</span><span class='op'>]</span><span class='op'>[</span>, <span class='fl'>1</span><span class='op'>]</span> <span class='op'>%in%</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>)</span>, <span class='fl'>2</span><span class='op'>]</span><span class='op'>)</span>
  <span class='op'>}</span> <span class='kw'>else</span> <span class='op'>{</span>
    <span class='fu'>barcodeplot</span><span class='op'>(</span>
      <span class='va'>y_dgelrt</span><span class='op'>$</span><span class='va'>table</span><span class='op'>$</span><span class='va'>logFC</span>,
      <span class='va'>y_index</span><span class='op'>[[</span><span class='va'>n</span><span class='op'>]</span><span class='op'>]</span><span class='op'>)</span>
  <span class='op'>}</span>
<span class='op'>}</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:unnamed-chunk-31"></span>
<img src="C086_Kinkel.multi-sample_comparisons_files/figure-html5/unnamed-chunk-31-1.png" alt="MD-plot and barcode plot of DEGs in supplied gene sets. Directional gene sets have (1) MD plot points coloured according to the statistical significance of the gene **in the gene set** (2) barcode plot 'weights' given by the logFC of the gene **in the gene set**. For the barcode plot, genes are represented by bars and are ranked from left to right by increasing log-fold change. This forms the barcode-like pattern. The line (or *worm*) above the barcode shows the relative local enrichment of the vertical bars in each part of the plot. The dotted horizontal line indicates neutral enrichment; the worm above the dotted line shows enrichment while the worm below the dotted line shows depletion." width="624" />
<p class="caption">
Figure 8: MD-plot and barcode plot of DEGs in supplied gene sets. Directional gene sets have (1) MD plot points coloured according to the statistical significance of the gene <strong>in the gene set</strong> (2) barcode plot ‘weights’ given by the logFC of the gene <strong>in the gene set</strong>. For the barcode plot, genes are represented by bars and are ranked from left to right by increasing log-fold change. This forms the barcode-like pattern. The line (or <em>worm</em>) above the barcode shows the relative local enrichment of the vertical bars in each part of the plot. The dotted horizontal line indicates neutral enrichment; the worm above the dotted line shows enrichment while the worm below the dotted line shows depletion.
</p>
</div>
</div>
<h1 id="male-only-analysis-of-aggregated-technical-replicates">Male-only: analysis of aggregated technical replicates</h1>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>ym_differences</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/RUVSeq/man/makeGroups.html'>makeGroups</a></span><span class='op'>(</span><span class='va'>ym</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>smchd1_genotype_updated</span><span class='op'>)</span>
<span class='va'>ym_ruv</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/RUVSeq/man/RUVs.html'>RUVs</a></span><span class='op'>(</span>
    x <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/EDASeq/man/betweenLaneNormalization-methods.html'>betweenLaneNormalization</a></span><span class='op'>(</span><span class='va'>ym</span><span class='op'>$</span><span class='va'>counts</span>, <span class='st'>"upper"</span><span class='op'>)</span>,
    cIdx <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>ym</span><span class='op'>)</span>,
    k <span class='op'>=</span> <span class='fl'>1</span>,
    scIdx <span class='op'>=</span> <span class='va'>ym_differences</span><span class='op'>)</span>

<span class='va'>ym_design</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/model.matrix.html'>model.matrix</a></span><span class='op'>(</span>
  <span class='op'>~</span><span class='fl'>0</span> <span class='op'>+</span> <span class='va'>smchd1_genotype_updated</span> <span class='op'>+</span> <span class='va'>ym_ruv</span><span class='op'>$</span><span class='va'>W</span>,
  <span class='va'>ym</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>ym_design</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/grep.html'>sub</a></span><span class='op'>(</span>
  <span class='st'>"ruv\\$|smchd1_genotype_updated"</span>,
  <span class='st'>""</span>,
  <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>ym_design</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<h2 id="mds-1">MDS</h2>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>ym_mds</span> <span class='op'>&lt;-</span> <span class='fu'>plotMDS</span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/SingleCellExperiment/man/assays.html'>cpm</a></span><span class='op'>(</span><span class='va'>ym_ruv</span><span class='op'>$</span><span class='va'>normalizedCounts</span>, log <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>, plot <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>

<span class='va'>ym_df</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cbind.html'>cbind</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>ym_mds</span><span class='op'>$</span><span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>ym_mds</span><span class='op'>$</span><span class='va'>y</span><span class='op'>)</span>, <span class='va'>ym</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>ym_df</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>ym</span><span class='op'>)</span>

<span class='va'>p1</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span>, colour <span class='op'>=</span> <span class='va'>smchd1_genotype_updated</span>, label <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>ym_df</span><span class='op'>)</span><span class='op'>)</span>,
  data <span class='op'>=</span> <span class='va'>ym_df</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html'>geom_text_repel</a></span><span class='op'>(</span>size <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 1"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 2"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>smchd1_genotype_updated_colours</span><span class='op'>)</span>
<span class='va'>p2</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span>, colour <span class='op'>=</span> <span class='va'>sex</span>, label <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>ym_df</span><span class='op'>)</span><span class='op'>)</span>,
  data <span class='op'>=</span> <span class='va'>ym_df</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html'>geom_text_repel</a></span><span class='op'>(</span>size <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 1"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 2"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>sex_colours</span><span class='op'>)</span>
<span class='va'>p3</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span>, colour <span class='op'>=</span> <span class='va'>mouse_number</span>, label <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>ym_df</span><span class='op'>)</span><span class='op'>)</span>,
  data <span class='op'>=</span> <span class='va'>ym_df</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html'>geom_text_repel</a></span><span class='op'>(</span>size <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 1"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 2"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>mouse_number_colours</span><span class='op'>)</span>
<span class='va'>p4</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span>, colour <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log10</a></span><span class='op'>(</span><span class='va'>lib.size</span><span class='op'>)</span>, label <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>ym_df</span><span class='op'>)</span><span class='op'>)</span>,
  data <span class='op'>=</span> <span class='va'>ym_df</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html'>geom_text_repel</a></span><span class='op'>(</span>size <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_viridis.html'>scale_colour_viridis_c</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 1"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 2"</span><span class='op'>)</span>

<span class='va'>p1</span> <span class='op'>+</span> <span class='va'>p2</span> <span class='op'>+</span> <span class='va'>p3</span> <span class='op'>+</span> <span class='va'>p4</span> <span class='op'>+</span> <span class='fu'><a href='https://patchwork.data-imaginist.com/reference/plot_layout.html'>plot_layout</a></span><span class='op'>(</span>ncol <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:unnamed-chunk-33"></span>
<img src="C086_Kinkel.multi-sample_comparisons_files/figure-html5/unnamed-chunk-33-1.png" alt="MDS plots of aggregated technical replicates following RUV coloured by various experimental factors." width="624" />
<p class="caption">
Figure 9: MDS plots of aggregated technical replicates following RUV coloured by various experimental factors.
</p>
</div>
</div>
<h2 id="de-analysis-1">DE analysis</h2>
<p>Fit a model using <em><a href="https://bioconductor.org/packages/3.12/edgeR">edgeR</a></em> with a term for <code>smchd1_genotype_updated</code>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>ym</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/estimateDisp.html'>estimateDisp</a></span><span class='op'>(</span><span class='va'>ym</span>, <span class='va'>ym_design</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/graphics/par.html'>par</a></span><span class='op'>(</span>mfrow <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/plotBCV.html'>plotBCV</a></span><span class='op'>(</span><span class='va'>ym</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>ym_dgeglm</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/glmfit.html'>glmFit</a></span><span class='op'>(</span><span class='va'>ym</span>, <span class='va'>ym_design</span><span class='op'>)</span>
<span class='va'>ym_contrast</span> <span class='op'>&lt;-</span> <span class='fu'>makeContrasts</span><span class='op'>(</span><span class='va'>Del</span> <span class='op'>-</span> <span class='va'>Het</span>, levels <span class='op'>=</span> <span class='va'>ym_design</span><span class='op'>)</span>
<span class='va'>ym_dgelrt</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/glmfit.html'>glmLRT</a></span><span class='op'>(</span><span class='va'>ym_dgeglm</span>, contrast <span class='op'>=</span> <span class='va'>ym_contrast</span><span class='op'>)</span>
<span class='va'>ym_dt</span> <span class='op'>&lt;-</span> <span class='fu'>decideTests</span><span class='op'>(</span><span class='va'>ym_dgelrt</span><span class='op'>)</span>

<span class='fu'><a href='https://rdrr.io/r/base/summary.html'>summary</a></span><span class='op'>(</span><span class='va'>ym_dt</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span>caption <span class='op'>=</span> <span class='st'>"Number of DEGs (FDR &lt; 0.05)"</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<table>
<caption><span id="tab:unnamed-chunk-35">Table 6: </span>Number of DEGs (FDR &lt; 0.05)</caption>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">1<em>Del -1</em>Het</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Down</td>
<td style="text-align: right;">44</td>
</tr>
<tr class="even">
<td style="text-align: left;">NotSig</td>
<td style="text-align: right;">11543</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Up</td>
<td style="text-align: right;">12</td>
</tr>
</tbody>
</table>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/r/graphics/par.html'>par</a></span><span class='op'>(</span>mfrow <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span>
<span class='fu'>plotMD</span><span class='op'>(</span><span class='va'>ym_dgelrt</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:unnamed-chunk-36"></span>
<img src="C086_Kinkel.multi-sample_comparisons_files/figure-html5/unnamed-chunk-36-1.png" alt="MD plot highlighting DEGs (red)." width="624" />
<p class="caption">
Figure 10: MD plot highlighting DEGs (red).
</p>
</div>
</div>
<p>An interactive <em><a href="https://bioconductor.org/packages/3.12/Glimma">Glimma</a></em> MD plot of the differential expression results is available from <a href="../output/DEGs/reads/glimma-plots/aggregated_tech_reps.males.md-plot.html"><code>output/DEGs/reads/glimma-plots/aggregated_tech_reps.males.md-plot.html</code></a>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>ym_tt</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/topTags.html'>topTags</a></span><span class='op'>(</span><span class='va'>ym_dgelrt</span>, n <span class='op'>=</span> <span class='cn'>Inf</span>, sort.by <span class='op'>=</span> <span class='st'>"none"</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/pkg/Glimma/man/glMDPlot.html'>glMDPlot</a></span><span class='op'>(</span>
  x <span class='op'>=</span> <span class='va'>ym_dgelrt</span>,
  counts <span class='op'>=</span> <span class='va'>ym</span>,
  anno <span class='op'>=</span> <span class='va'>ym</span><span class='op'>$</span><span class='va'>genes</span>,
  display.columns <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"ENSEMBL.SYMBOL"</span><span class='op'>)</span>,
  groups <span class='op'>=</span> <span class='va'>ym</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span>,
  samples <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>ym</span><span class='op'>)</span>,
  status <span class='op'>=</span> <span class='va'>ym_dt</span>,
  transform <span class='op'>=</span> <span class='cn'>TRUE</span>,
  sample.cols <span class='op'>=</span> <span class='va'>ym</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>smchd1_genotype_updated_colours</span>,
  path <span class='op'>=</span> <span class='va'>outdir</span>,
  html <span class='op'>=</span> <span class='st'>"aggregated_tech_reps.males.md-plot"</span>,
  launch <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>ym_tt</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/topTags.html'>topTags</a></span><span class='op'>(</span><span class='va'>ym_dgelrt</span>, p.value <span class='op'>=</span> <span class='fl'>0.05</span>, n <span class='op'>=</span> <span class='cn'>Inf</span><span class='op'>)</span>
<span class='kw'>if</span> <span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/nrow.html'>nrow</a></span><span class='op'>(</span><span class='va'>ym_tt</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>{</span>
  <span class='va'>ym_tt</span><span class='op'>$</span><span class='va'>table</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"logFC"</span>, <span class='st'>"logCPM"</span>, <span class='st'>"LR"</span>, <span class='st'>"PValue"</span>, <span class='st'>"FDR"</span><span class='op'>)</span><span class='op'>]</span> <span class='op'>%&gt;%</span>
    <span class='fu'>DT</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/DT/man/datatable.html'>datatable</a></span><span class='op'>(</span>caption <span class='op'>=</span> <span class='st'>"DEGs (FDR &lt; 0.05)"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
    <span class='fu'>DT</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/DT/man/formatCurrency.html'>formatSignif</a></span><span class='op'>(</span>
      <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"logFC"</span>, <span class='st'>"logCPM"</span>, <span class='st'>"LR"</span>, <span class='st'>"PValue"</span>, <span class='st'>"FDR"</span><span class='op'>)</span>,
      digits <span class='op'>=</span> <span class='fl'>3</span><span class='op'>)</span>
<span class='op'>}</span>
</code></pre>
</div>
</details>
<div id="htmlwidget-846bc22a95d0cb74eed7" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-846bc22a95d0cb74eed7">{"x":{"filter":"none","caption":"<caption>DEGs (FDR &lt; 0.05)<\/caption>","data":[["Krt80","Stx11","Ccsap","Chmp5","Sptan1","Timp3","Traf3ip1","Rassf8","Dnajc25","Prkaa2","Caap1","Hs3st1","Unc79","Fgl2","Gstm2","Cpeb4","Ccdc82","Usp44","Tmem60","Gpatch2l","Grrp1","Pwwp2a","Ptafr","Lgr4","Panx2","Cmpk2","Zfp9","Dync2h1","Aim1","Zc3h14","Gipc1","Gfpt2","Nnt","Fbxo16","Ippk","Zfp142","1700125H20Rik","Nuf2","Stx3","Veph1","Trpm6","Sv2a","Lpl","Il1rl2","Ldhd","Ggta1","Fn3k","Ctnnd1","Mettl24","Pif1","Cenpe","Tmf1","Taf13","Kbtbd7","Cnr2","Ppa1"],[10.1360701417494,-7.84063595438943,-7.00591243235241,-1.99294294797262,-1.35733309564449,-2.91758329033185,-4.01785965479608,-7.60689235032542,-3.8397489954201,2.36206134339408,-3.00874621467529,4.27915717811345,9.12937576870831,-2.7166403596259,3.93842283357552,-2.96988466526866,-2.18291781409505,8.80549859342335,-1.63438223055823,-3.27626985715026,8.4738676941509,-1.7244141777308,-6.31172439981735,-10.7403836327535,5.60092646383058,-1.57598878501468,-11.799597606702,-2.72333488881891,-6.01055894993104,-1.1387565133864,-2.19611087645905,-15.5638080206674,-9.39630975617573,-10.1759178712196,-2.54697836629903,-3.14707435861432,8.30807425415516,-2.28999524712172,-4.24802653656932,-9.03281228350475,8.08637773159277,-9.15916026728246,2.32039470601702,-10.7333143258968,-3.3565833991286,-3.57452694047389,-9.44406520476531,-1.27771844126287,-9.36272452721074,-8.95121869296607,-2.01421731391993,-1.55281495980771,-3.44031584016049,-1.89634091959627,-12.5056067579613,0.97677262812963],[3.09387720487023,3.70994009536117,4.26794252550756,6.24420405655555,7.61422179387396,5.91852394296344,4.28091839885259,3.28545028500475,4.37345776493545,5.16160680445383,4.90824554219929,3.79896749302116,2.10681321271457,5.0979442409568,4.07777702368088,4.78284304891313,6.44799523051061,1.6273029071549,6.40960352589336,4.52405170936802,1.71464437177075,5.99700700159667,3.51343764848676,3.58119591013925,2.45865355782674,6.2930364875174,2.52264049028219,4.74553422859729,2.95079805904034,7.4929446851535,6.16804331942651,2.27309015856203,2.78541411197553,2.94026320214778,5.59344362560036,4.46963694037294,1.29995878676916,5.96495749731964,3.69165546161761,2.4074337194513,1.91570912202135,2.46472911401626,5.47672220968812,3.53927717246354,4.44905832239271,4.23067480198919,2.51348210022845,6.79191288364689,2.42907139860677,2.24168160995697,5.64166717687187,6.27275179073011,4.37534830514309,7.02839654061435,2.42715249554722,7.98092202561569],[32.2799717174376,26.7247428041143,25.5211036574999,23.7907325484983,20.8115556408187,20.4997293516627,20.3433951158457,20.2267866978036,19.6447536531715,19.0911137420041,18.9712004445149,18.946302185376,18.9295417778835,18.2845909223669,17.5820741756571,17.4890822861092,17.3651160888293,16.8802871003874,16.8682773348118,16.7828120158231,16.3470797390005,16.2555219498648,15.9864584930019,15.8696110936032,15.7604386819368,15.6471942056782,15.480653339689,15.4089516603865,15.3980493175763,15.2912899511965,15.2603783783641,15.102697702216,15.0924424829594,15.0211392330289,14.9757800907229,14.8908528689308,14.8629997200936,14.7774395189645,14.7281912243682,14.6645779026573,14.5675180228041,14.5494551758458,14.4730121248231,14.4154463054334,14.4050810744724,14.4042774891443,14.1394631028031,14.0734924723454,14.0166915647128,14.0031038370679,13.9328880735795,13.8926310041384,13.793267637758,13.7924267991537,13.6855644212898,13.6143743368558],[1.3348213709871e-08,2.3459816991867e-07,4.37570491206787e-07,1.07398314980919e-06,5.06762766133293e-06,5.96396837687959e-06,6.47160853158402e-06,6.87830274443637e-06,9.32590175699387e-06,1.24623803331043e-05,1.32706457224663e-05,1.34449615280894e-05,1.35635964054421e-05,1.9023978109977e-05,2.75170012279549e-05,2.88962310035589e-05,3.08435292437588e-05,3.9812919765755e-05,4.00656335040162e-05,4.19112103599384e-05,5.27373616671302e-05,5.53480482222932e-05,6.37971830036458e-05,6.78596463910599e-05,7.18901600063218e-05,7.63257376266399e-05,8.33540551596737e-05,8.6577190377247e-05,8.70781389785625e-05,9.21403884646417e-05,9.36606853426841e-05,0.00010181768244055,0.000102372316253974,0.000106313611312531,0.00010889996615464,0.000113914006385923,0.00011560853259471,0.000120974188034019,0.000124175612538875,0.000128437427384819,0.000135225545550665,0.000136528126297766,0.000142182076982406,0.000146594938833539,0.000147404043385189,0.00014746695834283,0.000169745166429613,0.000175804033874033,0.000181195004456684,0.000182509108502637,0.00018945441946092,0.000193555862970486,0.000204066175070276,0.000204157526853847,0.000216109302105295,0.000224460277440071],[0.000154825930820794,0.00136055208644333,0.00169179337583584,0.00311428263865919,0.00997267919158969,0.00997267919158969,0.00997267919158969,0.00997267919158969,0.0120190149421524,0.0121018580543633,0.0121018580543633,0.0121018580543633,0.0121018580543633,0.015761365864116,0.0209479614631425,0.0209479614631425,0.0210443585704917,0.0243064064482463,0.0243064064482463,0.0243064064482463,0.0291286027608116,0.0291810005150172,0.0321731967677951,0.032796001603746,0.0333541586365331,0.0340500857973614,0.0348282528969775,0.0348282528969775,0.0348282528969775,0.0350442028803159,0.0350442028803159,0.0359823180675712,0.0359823180675712,0.0360894487836475,0.0360894487836475,0.036241712690974,0.036241712690974,0.0369257791317521,0.0369311007650873,0.0371841141264888,0.0371841141264888,0.0371841141264888,0.0371841141264888,0.0371841141264888,0.0371841141264888,0.0371841141264888,0.041890940115257,0.0423384629904417,0.0423384629904417,0.0423384629904417,0.0430878786534748,0.0431741241268204,0.043852280629218,0.043852280629218,0.0455754871839877,0.0464913349647747]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>logFC<\/th>\n      <th>logCPM<\/th>\n      <th>LR<\/th>\n      <th>PValue<\/th>\n      <th>FDR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"targets":1,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 3, 3, \",\", \".\");\n  }"},{"targets":2,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 3, 3, \",\", \".\");\n  }"},{"targets":3,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 3, 3, \",\", \".\");\n  }"},{"targets":4,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 3, 3, \",\", \".\");\n  }"},{"targets":5,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 3, 3, \",\", \".\");\n  }"},{"className":"dt-right","targets":[1,2,3,4,5]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":["options.columnDefs.0.render","options.columnDefs.1.render","options.columnDefs.2.render","options.columnDefs.3.render","options.columnDefs.4.render"],"jsHooks":[]}</script>
</div>
<p>CSV file of results available from <a href="../output/DEGs/reads/aggregated_tech_reps.males.DEGs.csv.gz"><code>output/DEGs/reads/aggregated_tech_reps.males.DEGs.csv.gz</code></a>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>gzout</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/connections.html'>gzfile</a></span><span class='op'>(</span>
  description <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span><span class='va'>outdir</span>, <span class='st'>"aggregated_tech_reps.males.DEGs.csv.gz"</span><span class='op'>)</span>,
  open <span class='op'>=</span> <span class='st'>"wb"</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/utils/write.table.html'>write.csv</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/topTags.html'>topTags</a></span><span class='op'>(</span><span class='va'>ym_dgelrt</span>, n <span class='op'>=</span> <span class='cn'>Inf</span><span class='op'>)</span>,
  <span class='va'>gzout</span>,
  <span class='co'># NOTE: quote = TRUE needed because some fields contain commas.</span>
  quote <span class='op'>=</span> <span class='cn'>TRUE</span>,
  row.names <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/connections.html'>close</a></span><span class='op'>(</span><span class='va'>gzout</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>Figure <a href="#fig:heatmap-ym">11</a> is a heatmap showing the expression of the DEGs.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/pkg/pheatmap/man/pheatmap.html'>pheatmap</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/pkg/SingleCellExperiment/man/assays.html'>cpm</a></span><span class='op'>(</span><span class='va'>ym</span>, log <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>ym_tt</span><span class='op'>)</span>, <span class='op'>]</span>,
  cluster_rows <span class='op'>=</span> <span class='cn'>TRUE</span>,
  cluster_cols <span class='op'>=</span> <span class='cn'>FALSE</span>,
  color <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/grDevices/palettes.html'>hcl.colors</a></span><span class='op'>(</span><span class='fl'>101</span>, <span class='st'>"Blue-Red 3"</span><span class='op'>)</span>,
  breaks <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span><span class='op'>(</span><span class='op'>-</span><span class='fl'>3</span>, <span class='fl'>3</span>, length.out <span class='op'>=</span> <span class='fl'>101</span><span class='op'>)</span>,
  scale <span class='op'>=</span> <span class='st'>"row"</span>,
  annotation_col <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>
    genotype <span class='op'>=</span> <span class='va'>ym</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>smchd1_genotype_updated</span>,
    sex <span class='op'>=</span> <span class='va'>ym</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>sex</span>,
    row.names <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>ym</span><span class='op'>)</span><span class='op'>)</span>,
  annotation_colors <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>
    genotype <span class='op'>=</span> <span class='va'>smchd1_genotype_updated_colours</span>,
    sex <span class='op'>=</span> <span class='va'>sex_colours</span><span class='op'>)</span>,
  main <span class='op'>=</span> <span class='st'>"Male samples"</span>,
  fontsize <span class='op'>=</span> <span class='fl'>9</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:heatmap-ym"></span>
<img src="C086_Kinkel.multi-sample_comparisons_files/figure-html5/heatmap-ym-1.png" alt="Heatmap of upper-quartile-normalzied logCPM values for the DEGs." width="624" />
<p class="caption">
Figure 11: Heatmap of upper-quartile-normalzied logCPM values for the DEGs.
</p>
</div>
</div>
<h2 id="gene-set-tests-1">Gene set tests</h2>
<p>Gene set testing is commonly used to interpret the differential expression results in a biological context. Here we apply various gene set tests to the DEG lists.</p>
<h3 id="goana-1">goana</h3>
<p>We use the <code>goana()</code> function from the <em><a href="https://bioconductor.org/packages/3.12/limma">limma</a></em> R/Bioconductor package to test for over-representation of gene ontology (GO) terms in each DEG list.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>list_of_go</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/lapply.html'>lapply</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>ym_contrast</span><span class='op'>)</span>, <span class='kw'>function</span><span class='op'>(</span><span class='va'>j</span><span class='op'>)</span> <span class='op'>{</span>
  <span class='va'>dgelrt</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/glmfit.html'>glmLRT</a></span><span class='op'>(</span><span class='va'>ym_dgeglm</span>, contrast <span class='op'>=</span> <span class='va'>ym_contrast</span><span class='op'>[</span>, <span class='va'>j</span><span class='op'>]</span><span class='op'>)</span>
  
  <span class='va'>go</span> <span class='op'>&lt;-</span> <span class='fu'>goana</span><span class='op'>(</span>
    de <span class='op'>=</span> <span class='va'>dgelrt</span>,
    geneid <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/lapply.html'>sapply</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/strsplit.html'>strsplit</a></span><span class='op'>(</span><span class='va'>dgelrt</span><span class='op'>$</span><span class='va'>genes</span><span class='op'>$</span><span class='va'>NCBI.ENTREZID</span>, <span class='st'>"; "</span><span class='op'>)</span>, <span class='st'>"[["</span>, <span class='fl'>1</span><span class='op'>)</span>,
    species <span class='op'>=</span> <span class='st'>"Mm"</span><span class='op'>)</span>
  <span class='va'>gzout</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/connections.html'>gzfile</a></span><span class='op'>(</span>
    description <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span><span class='va'>outdir</span>, <span class='st'>"aggregated_tech_reps.males.goana.csv.gz"</span><span class='op'>)</span>,
    open <span class='op'>=</span> <span class='st'>"wb"</span><span class='op'>)</span>
  <span class='fu'><a href='https://rdrr.io/r/utils/write.table.html'>write.csv</a></span><span class='op'>(</span>
    <span class='fu'>topGO</span><span class='op'>(</span><span class='va'>go</span>, number <span class='op'>=</span> <span class='cn'>Inf</span><span class='op'>)</span>,
    <span class='va'>gzout</span>,
    row.names <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
  <span class='fu'><a href='https://rdrr.io/r/base/connections.html'>close</a></span><span class='op'>(</span><span class='va'>gzout</span><span class='op'>)</span>
  <span class='va'>go</span>
<span class='op'>}</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>list_of_go</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>ym_contrast</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>CSV files of the results are available from <a href="../output/DEGs/reads/"><code>output/DEGs/reads/</code></a>.</p>
<p>An example of the results are shown below.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'>topGO</span><span class='op'>(</span><span class='va'>list_of_go</span><span class='op'>[[</span><span class='fl'>1</span><span class='op'>]</span><span class='op'>]</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span>
    caption <span class='op'>=</span> <span class='st'>'`goana()` produces a table with a row for each GO term and the following columns: **Term** (GO term); **Ont** (ontology that the GO term belongs to. Possible values are "BP", "CC" and "MF"); **N** (number of genes in the GO term); **Up** (number of up-regulated differentially expressed genes); **Down** (number of down-regulated differentially expressed genes); **P.Up** (p-value for over-representation of GO term in up-regulated genes); **P.Down** (p-value for over-representation of GO term in down-regulated genes)'</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<table style="width:100%;">
<caption><span id="tab:unnamed-chunk-41">Table 7: </span><code>goana()</code> produces a table with a row for each GO term and the following columns: <strong>Term</strong> (GO term); <strong>Ont</strong> (ontology that the GO term belongs to. Possible values are “BP,” “CC” and “MF”); <strong>N</strong> (number of genes in the GO term); <strong>Up</strong> (number of up-regulated differentially expressed genes); <strong>Down</strong> (number of down-regulated differentially expressed genes); <strong>P.Up</strong> (p-value for over-representation of GO term in up-regulated genes); <strong>P.Down</strong> (p-value for over-representation of GO term in down-regulated genes)</caption>
<colgroup>
<col style="width: 11%" />
<col style="width: 51%" />
<col style="width: 4%" />
<col style="width: 4%" />
<col style="width: 3%" />
<col style="width: 5%" />
<col style="width: 10%" />
<col style="width: 10%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;">Term</th>
<th style="text-align: left;">Ont</th>
<th style="text-align: right;">N</th>
<th style="text-align: right;">Up</th>
<th style="text-align: right;">Down</th>
<th style="text-align: right;">P.Up</th>
<th style="text-align: right;">P.Down</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"><a href="GO:0048786" class="uri">GO:0048786</a></td>
<td style="text-align: left;">presynaptic active zone</td>
<td style="text-align: left;">CC</td>
<td style="text-align: right;">48</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0000323</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="GO:0006633" class="uri">GO:0006633</a></td>
<td style="text-align: left;">fatty acid biosynthetic process</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">92</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0001039</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="GO:0016079" class="uri">GO:0016079</a></td>
<td style="text-align: left;">synaptic vesicle exocytosis</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">70</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0001431</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="GO:0099568" class="uri">GO:0099568</a></td>
<td style="text-align: left;">cytoplasmic region</td>
<td style="text-align: left;">CC</td>
<td style="text-align: right;">134</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0001514</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="GO:0072330" class="uri">GO:0072330</a></td>
<td style="text-align: left;">monocarboxylic acid biosynthetic process</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">113</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0001914</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="GO:0071398" class="uri">GO:0071398</a></td>
<td style="text-align: left;">cellular response to fatty acid</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">21</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0002080</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="GO:0045055" class="uri">GO:0045055</a></td>
<td style="text-align: left;">regulated exocytosis</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">147</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0002334</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="GO:0043207" class="uri">GO:0043207</a></td>
<td style="text-align: left;">response to external biotic stimulus</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">715</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">0.5378743</td>
<td style="text-align: right;">0.0002948</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="GO:0051707" class="uri">GO:0051707</a></td>
<td style="text-align: left;">response to other organism</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">715</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">0.5378743</td>
<td style="text-align: right;">0.0002948</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="GO:0009607" class="uri">GO:0009607</a></td>
<td style="text-align: left;">response to biotic stimulus</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">739</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">0.5500911</td>
<td style="text-align: right;">0.0003843</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="GO:0007269" class="uri">GO:0007269</a></td>
<td style="text-align: left;">neurotransmitter secretion</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">94</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0004451</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="GO:0099643" class="uri">GO:0099643</a></td>
<td style="text-align: left;">signal release from synapse</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">94</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0004451</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="GO:0070542" class="uri">GO:0070542</a></td>
<td style="text-align: left;">response to fatty acid</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">31</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0004578</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="GO:0007080" class="uri">GO:0007080</a></td>
<td style="text-align: left;">mitotic metaphase plate congression</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">40</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0004699</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="GO:0032838" class="uri">GO:0032838</a></td>
<td style="text-align: left;">plasma membrane bounded cell projection cytoplasm</td>
<td style="text-align: left;">CC</td>
<td style="text-align: right;">102</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0006066</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="GO:0007052" class="uri">GO:0007052</a></td>
<td style="text-align: left;">mitotic spindle organization</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">103</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0006294</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="GO:0043312" class="uri">GO:0043312</a></td>
<td style="text-align: left;">neutrophil degranulation</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0006334</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="GO:0030425" class="uri">GO:0030425</a></td>
<td style="text-align: left;">dendrite</td>
<td style="text-align: left;">CC</td>
<td style="text-align: right;">395</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">7</td>
<td style="text-align: right;">0.3431457</td>
<td style="text-align: right;">0.0006892</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="GO:0097447" class="uri">GO:0097447</a></td>
<td style="text-align: left;">dendritic tree</td>
<td style="text-align: left;">CC</td>
<td style="text-align: right;">396</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">7</td>
<td style="text-align: right;">0.3438566</td>
<td style="text-align: right;">0.0006996</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="GO:0046394" class="uri">GO:0046394</a></td>
<td style="text-align: left;">carboxylic acid biosynthetic process</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">184</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0008007</td>
<td style="text-align: right;">1.0000000</td>
</tr>
</tbody>
</table>
</div>
<h3 id="kegga-1">kegga</h3>
<p>We use the <code>kegga()</code> function from the <em><a href="https://bioconductor.org/packages/3.12/limma">limma</a></em> R/Bioconductor package to test for over-representation of KEGG pathways in each DEG list.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>list_of_kegg</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/lapply.html'>lapply</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>ym_contrast</span><span class='op'>)</span>, <span class='kw'>function</span><span class='op'>(</span><span class='va'>j</span><span class='op'>)</span> <span class='op'>{</span>
  <span class='va'>dgelrt</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/glmfit.html'>glmLRT</a></span><span class='op'>(</span><span class='va'>ym_dgeglm</span>, contrast <span class='op'>=</span> <span class='va'>ym_contrast</span><span class='op'>[</span>, <span class='va'>j</span><span class='op'>]</span><span class='op'>)</span>
  
  <span class='va'>kegg</span> <span class='op'>&lt;-</span> <span class='fu'>kegga</span><span class='op'>(</span>
    de <span class='op'>=</span> <span class='va'>dgelrt</span>,
    geneid <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/lapply.html'>sapply</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/strsplit.html'>strsplit</a></span><span class='op'>(</span><span class='va'>dgelrt</span><span class='op'>$</span><span class='va'>genes</span><span class='op'>$</span><span class='va'>NCBI.ENTREZID</span>, <span class='st'>"; "</span><span class='op'>)</span>, <span class='st'>"[["</span>, <span class='fl'>1</span><span class='op'>)</span>,
    species <span class='op'>=</span> <span class='st'>"Mm"</span><span class='op'>)</span>
  <span class='va'>gzout</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/connections.html'>gzfile</a></span><span class='op'>(</span>
    description <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span><span class='va'>outdir</span>, <span class='st'>"aggregated_tech_reps.males.kegga.csv.gz"</span><span class='op'>)</span>,
    open <span class='op'>=</span> <span class='st'>"wb"</span><span class='op'>)</span>
  <span class='fu'><a href='https://rdrr.io/r/utils/write.table.html'>write.csv</a></span><span class='op'>(</span>
    <span class='fu'>topKEGG</span><span class='op'>(</span><span class='va'>kegg</span>, number <span class='op'>=</span> <span class='cn'>Inf</span><span class='op'>)</span>,
    <span class='va'>gzout</span>,
    row.names <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
  <span class='fu'><a href='https://rdrr.io/r/base/connections.html'>close</a></span><span class='op'>(</span><span class='va'>gzout</span><span class='op'>)</span>
  <span class='va'>kegg</span>
<span class='op'>}</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>list_of_kegg</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>ym_contrast</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>CSV files of the results are available from <a href="../output/DEGs/reads/"><code>output/DEGs/reads/</code></a>.</p>
<p>An example of the results are shown below.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'>topKEGG</span><span class='op'>(</span><span class='va'>list_of_kegg</span><span class='op'>[[</span><span class='fl'>1</span><span class='op'>]</span><span class='op'>]</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span>
    caption <span class='op'>=</span> <span class='st'>'`kegga()` produces a table with a row for each KEGG pathway ID and the following columns: **Pathway** (KEGG pathway); **N** (number of genes in the GO term); **Up** (number of up-regulated differentially expressed genes); **Down** (number of down-regulated differentially expressed genes); **P.Up** (p-value for over-representation of KEGG pathway in up-regulated genes); **P.Down** (p-value for over-representation of KEGG pathway in down-regulated genes)'</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<table>
<caption><span id="tab:unnamed-chunk-43">Table 8: </span><code>kegga()</code> produces a table with a row for each KEGG pathway ID and the following columns: <strong>Pathway</strong> (KEGG pathway); <strong>N</strong> (number of genes in the GO term); <strong>Up</strong> (number of up-regulated differentially expressed genes); <strong>Down</strong> (number of down-regulated differentially expressed genes); <strong>P.Up</strong> (p-value for over-representation of KEGG pathway in up-regulated genes); <strong>P.Down</strong> (p-value for over-representation of KEGG pathway in down-regulated genes)</caption>
<colgroup>
<col style="width: 13%" />
<col style="width: 56%" />
<col style="width: 3%" />
<col style="width: 2%" />
<col style="width: 4%" />
<col style="width: 9%" />
<col style="width: 9%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;">Pathway</th>
<th style="text-align: right;">N</th>
<th style="text-align: right;">Up</th>
<th style="text-align: right;">Down</th>
<th style="text-align: right;">P.Up</th>
<th style="text-align: right;">P.Down</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">path:mmu05418</td>
<td style="text-align: left;">Fluid shear stress and atherosclerosis</td>
<td style="text-align: right;">106</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0052450</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="even">
<td style="text-align: left;">path:mmu04130</td>
<td style="text-align: left;">SNARE interactions in vesicular transport</td>
<td style="text-align: right;">31</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0062180</td>
</tr>
<tr class="odd">
<td style="text-align: left;">path:mmu00534</td>
<td style="text-align: left;">Glycosaminoglycan biosynthesis - heparan sulfate / heparin</td>
<td style="text-align: right;">15</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0155706</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="even">
<td style="text-align: left;">path:mmu04080</td>
<td style="text-align: left;">Neuroactive ligand-receptor interaction</td>
<td style="text-align: right;">51</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0162443</td>
</tr>
<tr class="odd">
<td style="text-align: left;">path:mmu00982</td>
<td style="text-align: left;">Drug metabolism - cytochrome P450</td>
<td style="text-align: right;">22</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0227606</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="even">
<td style="text-align: left;">path:mmu05204</td>
<td style="text-align: left;">Chemical carcinogenesis - DNA adducts</td>
<td style="text-align: right;">23</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0237838</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="odd">
<td style="text-align: left;">path:mmu00980</td>
<td style="text-align: left;">Metabolism of xenobiotics by cytochrome P450</td>
<td style="text-align: right;">25</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0258272</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="even">
<td style="text-align: left;">path:mmu04979</td>
<td style="text-align: left;">Cholesterol metabolism</td>
<td style="text-align: right;">28</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0288851</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="odd">
<td style="text-align: left;">path:mmu04710</td>
<td style="text-align: left;">Circadian rhythm</td>
<td style="text-align: right;">28</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0288851</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="even">
<td style="text-align: left;">path:mmu04978</td>
<td style="text-align: left;">Mineral absorption</td>
<td style="text-align: right;">28</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0288851</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="odd">
<td style="text-align: left;">path:mmu00561</td>
<td style="text-align: left;">Glycerolipid metabolism</td>
<td style="text-align: right;">40</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0410286</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="even">
<td style="text-align: left;">path:mmu03320</td>
<td style="text-align: left;">PPAR signaling pathway</td>
<td style="text-align: right;">41</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0420342</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="odd">
<td style="text-align: left;">path:mmu05410</td>
<td style="text-align: left;">Hypertrophic cardiomyopathy</td>
<td style="text-align: right;">44</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0450454</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="even">
<td style="text-align: left;">path:mmu00601</td>
<td style="text-align: left;">Glycosphingolipid biosynthesis - lacto and neolacto series</td>
<td style="text-align: right;">12</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0450454</td>
</tr>
<tr class="odd">
<td style="text-align: left;">path:mmu00983</td>
<td style="text-align: left;">Drug metabolism - other enzymes</td>
<td style="text-align: right;">46</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0470480</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="even">
<td style="text-align: left;">path:mmu04213</td>
<td style="text-align: left;">Longevity regulating pathway - multiple species</td>
<td style="text-align: right;">48</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0490468</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="odd">
<td style="text-align: left;">path:mmu04920</td>
<td style="text-align: left;">Adipocytokine signaling pathway</td>
<td style="text-align: right;">49</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0500447</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="even">
<td style="text-align: left;">path:mmu00480</td>
<td style="text-align: left;">Glutathione metabolism</td>
<td style="text-align: right;">50</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0510417</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="odd">
<td style="text-align: left;">path:mmu01524</td>
<td style="text-align: left;">Platinum drug resistance</td>
<td style="text-align: right;">65</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0658817</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="even">
<td style="text-align: left;">path:mmu04211</td>
<td style="text-align: left;">Longevity regulating pathway</td>
<td style="text-align: right;">71</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0717580</td>
<td style="text-align: right;">1.0000000</td>
</tr>
</tbody>
</table>
</div>
<h3 id="camera-with-msigdb-gene-sets-1">camera with MSigDB gene sets</h3>
<p>We use the <code>camera()</code> function from the <em><a href="https://bioconductor.org/packages/3.12/limma">limma</a></em> R/Bioconductor package to test whether a set of genes is highly ranked relative to other genes in terms of differential expression, accounting for inter-gene correlation. Specifically, we test using gene sets from <a href="https://www.gsea-msigdb.org/gsea/msigdb">MSigDB</a>, namely:</p>
<aside>
We download these gene sets from <a href="http://bioinf.wehi.edu.au/MSigDB/index.html">http://bioinf.wehi.edu.au/MSigDB/index.html</a>
</aside>
<ul>
<li><strong>H</strong>: <strong>hallmark gene sets</strong> are coherently expressed signatures derived by aggregating many MSigDB gene sets to represent well-defined biological states or processes.</li>
<li><strong>C2</strong>: <strong>curated gene sets</strong> from online pathway databases, publications in PubMed, and knowledge of domain experts.</li>
</ul>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>ym_idx</span> <span class='op'>&lt;-</span> <span class='fu'>ids2indices</span><span class='op'>(</span>
  <span class='va'>msigdb</span>, 
  id <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/lapply.html'>sapply</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/strsplit.html'>strsplit</a></span><span class='op'>(</span><span class='va'>ym</span><span class='op'>$</span><span class='va'>genes</span><span class='op'>$</span><span class='va'>NCBI.ENTREZID</span>, <span class='st'>"; "</span><span class='op'>)</span>, <span class='st'>"[["</span>, <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>list_of_camera</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/lapply.html'>lapply</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>ym_contrast</span><span class='op'>)</span>, <span class='kw'>function</span><span class='op'>(</span><span class='va'>j</span><span class='op'>)</span> <span class='op'>{</span>
  <span class='va'>cam</span> <span class='op'>&lt;-</span> <span class='fu'>camera</span><span class='op'>(</span>
    y <span class='op'>=</span> <span class='va'>ym</span>, 
    index <span class='op'>=</span> <span class='va'>ym_idx</span>,
    design <span class='op'>=</span> <span class='va'>ym_design</span>, 
    contrast <span class='op'>=</span> <span class='va'>ym_contrast</span><span class='op'>[</span>, <span class='va'>j</span><span class='op'>]</span><span class='op'>)</span>
  <span class='va'>gzout</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/connections.html'>gzfile</a></span><span class='op'>(</span>
    description <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span><span class='va'>outdir</span>, <span class='st'>"aggregated_tech_reps.males.camera.csv.gz"</span><span class='op'>)</span>,
    open <span class='op'>=</span> <span class='st'>"wb"</span><span class='op'>)</span>
  <span class='fu'><a href='https://rdrr.io/r/utils/write.table.html'>write.csv</a></span><span class='op'>(</span>
    <span class='va'>cam</span>,
    <span class='va'>gzout</span>,
    row.names <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
  <span class='fu'><a href='https://rdrr.io/r/base/connections.html'>close</a></span><span class='op'>(</span><span class='va'>gzout</span><span class='op'>)</span>
  <span class='va'>cam</span>
<span class='op'>}</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>list_of_camera</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>ym_contrast</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>CSV files of the results are available from <a href="../output/DEGs/reads/"><code>output/DEGs/reads</code></a>.</p>
<p>An example of the results are shown below.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span><span class='op'>(</span><span class='va'>list_of_camera</span><span class='op'>[[</span><span class='fl'>1</span><span class='op'>]</span><span class='op'>]</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span>
    caption <span class='op'>=</span> <span class='st'>'`camera()` produces a table with a row for each gene set (prepended by which MSigDB collection it comes from) and the following columns: **NGenes** (number of genes in set); **Direction** (direction of change); **PValue** (two-tailed p-value); **FDR** (Benjamini and Hochberg FDR adjusted p-value).'</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<table>
<caption><span id="tab:unnamed-chunk-45">Table 9: </span><code>camera()</code> produces a table with a row for each gene set (prepended by which MSigDB collection it comes from) and the following columns: <strong>NGenes</strong> (number of genes in set); <strong>Direction</strong> (direction of change); <strong>PValue</strong> (two-tailed p-value); <strong>FDR</strong> (Benjamini and Hochberg FDR adjusted p-value).</caption>
<colgroup>
<col style="width: 59%" />
<col style="width: 7%" />
<col style="width: 10%" />
<col style="width: 10%" />
<col style="width: 10%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">NGenes</th>
<th style="text-align: left;">Direction</th>
<th style="text-align: right;">PValue</th>
<th style="text-align: right;">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">H.HALLMARK_MYC_TARGETS_V1</td>
<td style="text-align: right;">229</td>
<td style="text-align: left;">Up</td>
<td style="text-align: right;">0.0000535</td>
<td style="text-align: right;">0.2203915</td>
</tr>
<tr class="even">
<td style="text-align: left;">C2.JOHANSSON_GLIOMAGENESIS_BY_PDGFB_DN</td>
<td style="text-align: right;">19</td>
<td style="text-align: left;">Down</td>
<td style="text-align: right;">0.0000794</td>
<td style="text-align: right;">0.2203915</td>
</tr>
<tr class="odd">
<td style="text-align: left;">C2.REACTOME_ELECTRIC_TRANSMISSION_ACROSS_GAP_JUNCTIONS</td>
<td style="text-align: right;">2</td>
<td style="text-align: left;">Up</td>
<td style="text-align: right;">0.0001400</td>
<td style="text-align: right;">0.2592181</td>
</tr>
<tr class="even">
<td style="text-align: left;">C2.RAHMAN_TP53_TARGETS_PHOSPHORYLATED</td>
<td style="text-align: right;">25</td>
<td style="text-align: left;">Up</td>
<td style="text-align: right;">0.0002956</td>
<td style="text-align: right;">0.3456734</td>
</tr>
<tr class="odd">
<td style="text-align: left;">C2.REACTOME_UNWINDING_OF_DNA</td>
<td style="text-align: right;">12</td>
<td style="text-align: left;">Up</td>
<td style="text-align: right;">0.0003112</td>
<td style="text-align: right;">0.3456734</td>
</tr>
<tr class="even">
<td style="text-align: left;">C2.PID_MYC_ACTIV_PATHWAY</td>
<td style="text-align: right;">95</td>
<td style="text-align: left;">Up</td>
<td style="text-align: right;">0.0003762</td>
<td style="text-align: right;">0.3482163</td>
</tr>
</tbody>
</table>
</div>
<h3 id="sarahs-gene-lists-1">Sarah’s gene lists</h3>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>ym_X</span> <span class='op'>&lt;-</span> <span class='va'>ym</span><span class='op'>$</span><span class='va'>genes</span><span class='op'>$</span><span class='va'>ENSEMBL.SEQNAME</span> <span class='op'>==</span> <span class='st'>"X"</span>
<span class='va'>ym_protocadherin</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/grep.html'>grepl</a></span><span class='op'>(</span><span class='st'>"Pcdh"</span>, <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>ym</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>ym_pw</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>ym</span><span class='op'>)</span> <span class='op'>%in%</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Ndn"</span>, <span class='st'>"Mkrn3"</span>, <span class='st'>"Peg12"</span><span class='op'>)</span>
<span class='va'>ym_male</span> <span class='op'>&lt;-</span> <span class='va'>male_het_vs_del</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Probe"</span>, <span class='st'>"Differential expression"</span><span class='op'>)</span><span class='op'>]</span>
<span class='va'>ym_female</span> <span class='op'>&lt;-</span> <span class='va'>female_het_vs_del</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Probe"</span>, <span class='st'>"Differential expression"</span><span class='op'>)</span><span class='op'>]</span>
<span class='va'>ym_sun</span> <span class='op'>&lt;-</span> <span class='va'>sun_2014</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"geneSymbol"</span>, <span class='st'>"log2FC"</span><span class='op'>)</span><span class='op'>]</span>
<span class='va'>ym_chambers</span> <span class='op'>&lt;-</span> <span class='va'>chambers_2007</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Gene Symbol"</span>, <span class='st'>"Fold Difference"</span><span class='op'>)</span><span class='op'>]</span>
<span class='va'>ym_chambers</span><span class='op'>$</span><span class='va'>logFC</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log2</a></span><span class='op'>(</span><span class='va'>ym_chambers</span><span class='op'>$</span><span class='va'>`Fold Difference`</span><span class='op'>)</span>
<span class='va'>ym_chambers</span><span class='op'>$</span><span class='va'>`Fold Difference`</span> <span class='op'>&lt;-</span> <span class='cn'>NULL</span>
<span class='va'>ym_chambers</span> <span class='op'>&lt;-</span> <span class='va'>ym_chambers</span><span class='op'>[</span><span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/NA.html'>is.na</a></span><span class='op'>(</span><span class='va'>ym_chambers</span><span class='op'>$</span><span class='va'>`Gene Symbol`</span><span class='op'>)</span>, <span class='op'>]</span>
<span class='va'>ym_chambers</span> <span class='op'>&lt;-</span> <span class='va'>ym_chambers</span><span class='op'>[</span><span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/duplicated.html'>duplicated</a></span><span class='op'>(</span><span class='va'>ym_chambers</span><span class='op'>$</span><span class='va'>`Gene Symbol`</span><span class='op'>)</span>, <span class='op'>]</span>
<span class='va'>ym_gazit</span> <span class='op'>&lt;-</span> <span class='va'>gazit_2013</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Symbol"</span><span class='op'>)</span>, drop <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>]</span>
<span class='va'>ym_gazit</span><span class='op'>$</span><span class='va'>Symbol</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/base/substr.html'>substring</a></span><span class='op'>(</span><span class='va'>ym_gazit</span><span class='op'>$</span><span class='va'>Symbol</span>, <span class='fl'>1</span>, <span class='fl'>1</span><span class='op'>)</span>, 
  <span class='fu'><a href='https://rdrr.io/r/base/chartr.html'>tolower</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/substr.html'>substring</a></span><span class='op'>(</span><span class='va'>ym_gazit</span><span class='op'>$</span><span class='va'>Symbol</span>, <span class='fl'>2</span><span class='op'>)</span><span class='op'>)</span>,
  sep <span class='op'>=</span> <span class='st'>""</span><span class='op'>)</span>
<span class='va'>ym_gazit</span><span class='op'>$</span><span class='va'>logFC</span> <span class='op'>&lt;-</span> <span class='fl'>1</span>
<span class='va'>ym_chambers_gazit</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>
  Symbol <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='va'>ym_chambers</span><span class='op'>$</span><span class='va'>`Gene Symbol`</span>, <span class='va'>ym_gazit</span><span class='op'>$</span><span class='va'>Symbol</span><span class='op'>)</span>,
  logFC <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='va'>ym_chambers</span><span class='op'>$</span><span class='va'>logFC</span>, <span class='va'>ym_gazit</span><span class='op'>$</span><span class='va'>logFC</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>ym_chambers_gazit</span> <span class='op'>&lt;-</span> <span class='va'>ym_chambers_gazit</span><span class='op'>[</span><span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/duplicated.html'>duplicated</a></span><span class='op'>(</span><span class='va'>ym_chambers_gazit</span><span class='op'>$</span><span class='va'>Symbol</span><span class='op'>)</span>, <span class='op'>]</span>
<span class='va'>ym_venezia_proliferation</span> <span class='op'>&lt;-</span> <span class='va'>venezia_proliferation</span><span class='op'>[</span>
  , <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Gene.Symbol"</span>, <span class='st'>"Log2.Fold.Change..FL.HSC.vs.Adult.HSC.."</span><span class='op'>)</span><span class='op'>]</span>
<span class='va'>ym_venezia_proliferation</span> <span class='op'>&lt;-</span> <span class='va'>ym_venezia_proliferation</span><span class='op'>[</span>
  <span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/duplicated.html'>duplicated</a></span><span class='op'>(</span><span class='va'>ym_venezia_proliferation</span><span class='op'>$</span><span class='va'>`Gene.Symbol`</span><span class='op'>)</span>, <span class='op'>]</span>
<span class='va'>ym_venezia_quiescence</span> <span class='op'>&lt;-</span> <span class='va'>venezia_quiescence</span><span class='op'>[</span>
  , <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Gene.Symbol"</span>, <span class='st'>"Log2.Fold.Change..FL.HSC.vs.Adult.HSC.."</span><span class='op'>)</span><span class='op'>]</span>
<span class='va'>ym_venezia_quiescence</span> <span class='op'>&lt;-</span> <span class='va'>ym_venezia_quiescence</span><span class='op'>[</span>
  <span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/duplicated.html'>duplicated</a></span><span class='op'>(</span><span class='va'>ym_venezia_quiescence</span><span class='op'>$</span><span class='va'>`Gene.Symbol`</span><span class='op'>)</span>, <span class='op'>]</span>
<span class='co'># TODO: Clarify how this gene set is to be used.</span>
<span class='co'># ym_smchd1_femBcells &lt;- smchd1_femBcells[</span>
<span class='co'>#   smchd1_femBcells$`FDR (EdgeR stats p&lt;1.0 after correction)` &lt; 0.05, </span>
<span class='co'>#   c("Feature", "Log2 Fold Change (EdgeR stats p&lt;1.0 after correction)")]</span>
<span class='co'># ym_smchd1_femBcells &lt;- ym_smchd1_femBcells[</span>
<span class='co'>#   !duplicated(ym_smchd1_femBcells$Feature), ]</span>
<span class='va'>ym_smchd1_femBcells</span> <span class='op'>&lt;-</span> <span class='va'>smchd1_femBcells</span><span class='op'>[</span>
    <span class='co'># smchd1_femBcells$Chromosome == "X", </span>
  <span class='va'>smchd1_femBcells</span><span class='op'>$</span><span class='va'>Chromosome</span> <span class='op'>==</span> <span class='st'>"X"</span> <span class='op'>&amp;</span> 
    <span class='va'>smchd1_femBcells</span><span class='op'>$</span><span class='va'>`FDR (EdgeR stats p&lt;1.0 after correction)`</span> <span class='op'>&lt;</span> <span class='fl'>0.05</span>,
  <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Feature"</span>, <span class='st'>"Log2 Fold Change (EdgeR stats p&lt;1.0 after correction)"</span><span class='op'>)</span><span class='op'>]</span>
<span class='va'>ym_smchd1_femBcells</span> <span class='op'>&lt;-</span> <span class='va'>ym_smchd1_femBcells</span><span class='op'>[</span>
  <span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/duplicated.html'>duplicated</a></span><span class='op'>(</span><span class='va'>ym_smchd1_femBcells</span><span class='op'>$</span><span class='va'>Feature</span><span class='op'>)</span>, <span class='op'>]</span>

<span class='va'>ym_index</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>
  `X-linked` <span class='op'>=</span> <span class='va'>ym_X</span>,
  Protocadherin <span class='op'>=</span> <span class='va'>ym_protocadherin</span>,
  `Prader-Willi` <span class='op'>=</span> <span class='va'>ym_pw</span>,
  `Male: Het vs. Del` <span class='op'>=</span> <span class='va'>ym_male</span>,
  `Female: Het vs. Del` <span class='op'>=</span> <span class='va'>ym_female</span>,
  `Sun (2014)` <span class='op'>=</span> <span class='va'>ym_sun</span>,
  `Chambers (2007)` <span class='op'>=</span> <span class='va'>ym_chambers</span>,
  `Gazit (2013)` <span class='op'>=</span> <span class='va'>ym_gazit</span>,
  `Chambers (2007) &amp; Gazit (2013)` <span class='op'>=</span> <span class='va'>ym_chambers_gazit</span>,
  `Proliferation Group Venezia (2004)` <span class='op'>=</span> <span class='va'>ym_venezia_proliferation</span>,
  `Quiescence Group Venezia (2004)` <span class='op'>=</span> <span class='va'>ym_venezia_quiescence</span>,
  `Smchd1 female B cells (chrX)` <span class='op'>=</span> <span class='va'>ym_smchd1_femBcells</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>Sarah requested testing a number of gene sets (see <a href="#gene-sets">Gene sets</a>).</p>
<p>We use the <code>fry()</code> function from the <em><a href="https://bioconductor.org/packages/3.12/edgeR">edgeR</a></em> R/Bioconductor package to perform a self-contained gene set test against the null hypothesis that none of the genes in the set are differentially expressed.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'>fry</span><span class='op'>(</span>
  <span class='va'>ym</span>, 
  index <span class='op'>=</span> <span class='va'>ym_index</span>, 
  design <span class='op'>=</span> <span class='va'>ym_design</span>,
  contrast <span class='op'>=</span> <span class='va'>ym_contrast</span>,
  sort <span class='op'>=</span> <span class='st'>"none"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
    <span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span>
      caption <span class='op'>=</span> <span class='st'>"Results of applying `fry()` to the RNA-seq differential expression results using the supplied gene sets. A significant 'Up' P-value means that the differential expression results found in the RNA-seq data are positively correlated with the expression signature from the corresponding gene set. Conversely, a significant 'Down' P-value means that the differential expression log-fold-changes are negatively correlated with the expression signature from the corresponding gene set. A significant 'Mixed' P-value means that the genes in the set tend to be differentially expressed without regard for direction."</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<table style="width:100%;">
<caption><span id="tab:unnamed-chunk-47">Table 10: </span>Results of applying <code>fry()</code> to the RNA-seq differential expression results using the supplied gene sets. A significant ‘Up’ P-value means that the differential expression results found in the RNA-seq data are positively correlated with the expression signature from the corresponding gene set. Conversely, a significant ‘Down’ P-value means that the differential expression log-fold-changes are negatively correlated with the expression signature from the corresponding gene set. A significant ‘Mixed’ P-value means that the genes in the set tend to be differentially expressed without regard for direction.</caption>
<colgroup>
<col style="width: 36%" />
<col style="width: 7%" />
<col style="width: 10%" />
<col style="width: 10%" />
<col style="width: 10%" />
<col style="width: 13%" />
<col style="width: 10%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">NGenes</th>
<th style="text-align: left;">Direction</th>
<th style="text-align: right;">PValue</th>
<th style="text-align: right;">FDR</th>
<th style="text-align: right;">PValue.Mixed</th>
<th style="text-align: right;">FDR.Mixed</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">X-linked</td>
<td style="text-align: right;">392</td>
<td style="text-align: left;">Down</td>
<td style="text-align: right;">0.0346948</td>
<td style="text-align: right;">0.1387794</td>
<td style="text-align: right;">0.0085836</td>
<td style="text-align: right;">0.0206006</td>
</tr>
<tr class="even">
<td style="text-align: left;">Protocadherin</td>
<td style="text-align: right;">6</td>
<td style="text-align: left;">Up</td>
<td style="text-align: right;">0.3718805</td>
<td style="text-align: right;">0.6375094</td>
<td style="text-align: right;">0.0278469</td>
<td style="text-align: right;">0.0477375</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Prader-Willi</td>
<td style="text-align: right;">3</td>
<td style="text-align: left;">Up</td>
<td style="text-align: right;">0.5392710</td>
<td style="text-align: right;">0.7190280</td>
<td style="text-align: right;">0.0766661</td>
<td style="text-align: right;">0.1047170</td>
</tr>
<tr class="even">
<td style="text-align: left;">Male: Het vs. Del</td>
<td style="text-align: right;">18</td>
<td style="text-align: left;">Up</td>
<td style="text-align: right;">0.6454682</td>
<td style="text-align: right;">0.7745619</td>
<td style="text-align: right;">0.1007001</td>
<td style="text-align: right;">0.1208401</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Female: Het vs. Del</td>
<td style="text-align: right;">51</td>
<td style="text-align: left;">Up</td>
<td style="text-align: right;">0.4434587</td>
<td style="text-align: right;">0.6651881</td>
<td style="text-align: right;">0.0785377</td>
<td style="text-align: right;">0.1047170</td>
</tr>
<tr class="even">
<td style="text-align: left;">Sun (2014)</td>
<td style="text-align: right;">2023</td>
<td style="text-align: left;">Up</td>
<td style="text-align: right;">0.7412489</td>
<td style="text-align: right;">0.8086352</td>
<td style="text-align: right;">0.0029447</td>
<td style="text-align: right;">0.0088761</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Chambers (2007)</td>
<td style="text-align: right;">206</td>
<td style="text-align: left;">Down</td>
<td style="text-align: right;">0.0047591</td>
<td style="text-align: right;">0.0285547</td>
<td style="text-align: right;">0.0029587</td>
<td style="text-align: right;">0.0088761</td>
</tr>
<tr class="even">
<td style="text-align: left;">Gazit (2013)</td>
<td style="text-align: right;">248</td>
<td style="text-align: left;">Down</td>
<td style="text-align: right;">0.1999566</td>
<td style="text-align: right;">0.3999131</td>
<td style="text-align: right;">0.0007036</td>
<td style="text-align: right;">0.0084427</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Chambers (2007) &amp; Gazit (2013)</td>
<td style="text-align: right;">402</td>
<td style="text-align: left;">Down</td>
<td style="text-align: right;">0.0030405</td>
<td style="text-align: right;">0.0285547</td>
<td style="text-align: right;">0.0028950</td>
<td style="text-align: right;">0.0088761</td>
</tr>
<tr class="even">
<td style="text-align: left;">Proliferation Group Venezia (2004)</td>
<td style="text-align: right;">364</td>
<td style="text-align: left;">Up</td>
<td style="text-align: right;">0.1363442</td>
<td style="text-align: right;">0.3469805</td>
<td style="text-align: right;">0.1138137</td>
<td style="text-align: right;">0.1241604</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Quiescence Group Venezia (2004)</td>
<td style="text-align: right;">383</td>
<td style="text-align: left;">Up</td>
<td style="text-align: right;">0.1445752</td>
<td style="text-align: right;">0.3469805</td>
<td style="text-align: right;">0.0246693</td>
<td style="text-align: right;">0.0477375</td>
</tr>
<tr class="even">
<td style="text-align: left;">Smchd1 female B cells (chrX)</td>
<td style="text-align: right;">6</td>
<td style="text-align: left;">Down</td>
<td style="text-align: right;">0.9696160</td>
<td style="text-align: right;">0.9696160</td>
<td style="text-align: right;">0.8636350</td>
<td style="text-align: right;">0.8636350</td>
</tr>
</tbody>
</table>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/r/graphics/par.html'>par</a></span><span class='op'>(</span>mfrow <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span><span class='op'>(</span><span class='va'>ym_index</span><span class='op'>)</span>, <span class='fl'>2</span><span class='op'>)</span><span class='op'>)</span>
<span class='kw'>for</span> <span class='op'>(</span><span class='va'>n</span> <span class='kw'>in</span> <span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>ym_index</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>{</span>
  <span class='kw'>if</span> <span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>is.data.frame</a></span><span class='op'>(</span><span class='va'>ym_index</span><span class='op'>[[</span><span class='va'>n</span><span class='op'>]</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>{</span>
    <span class='va'>ym_status</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/rep.html'>rep</a></span><span class='op'>(</span><span class='fl'>0</span>, <span class='fu'><a href='https://rdrr.io/r/base/nrow.html'>nrow</a></span><span class='op'>(</span><span class='va'>ym_dgelrt</span><span class='op'>)</span><span class='op'>)</span>
    <span class='va'>ym_status</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>ym</span><span class='op'>)</span> <span class='op'>%in%</span> <span class='va'>ym_index</span><span class='op'>[[</span><span class='va'>n</span><span class='op'>]</span><span class='op'>]</span><span class='op'>[</span>, <span class='fl'>1</span><span class='op'>]</span><span class='op'>]</span> <span class='op'>&lt;-</span>
      <span class='fu'><a href='https://rdrr.io/r/base/sign.html'>sign</a></span><span class='op'>(</span><span class='va'>ym_index</span><span class='op'>[[</span><span class='va'>n</span><span class='op'>]</span><span class='op'>]</span><span class='op'>[</span><span class='va'>ym_index</span><span class='op'>[[</span><span class='va'>n</span><span class='op'>]</span><span class='op'>]</span><span class='op'>[</span>, <span class='fl'>1</span><span class='op'>]</span> <span class='op'>%in%</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>ym</span><span class='op'>)</span>, <span class='fl'>2</span><span class='op'>]</span><span class='op'>)</span>
    <span class='fu'>plotMD</span><span class='op'>(</span>
      <span class='va'>ym_dgelrt</span>,
      status <span class='op'>=</span> <span class='va'>ym_status</span>,
      hl.col <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"red"</span>, <span class='st'>"blue"</span><span class='op'>)</span>,
      legend <span class='op'>=</span> <span class='st'>"bottomright"</span>,
      main <span class='op'>=</span> <span class='va'>n</span><span class='op'>)</span>
    <span class='fu'><a href='https://rdrr.io/r/graphics/abline.html'>abline</a></span><span class='op'>(</span>h <span class='op'>=</span> <span class='fl'>0</span>, col <span class='op'>=</span> <span class='st'>"darkgrey"</span><span class='op'>)</span>
  <span class='op'>}</span> <span class='kw'>else</span> <span class='op'>{</span>
    <span class='va'>ym_status</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/rep.html'>rep</a></span><span class='op'>(</span><span class='st'>"Other"</span>, <span class='fu'><a href='https://rdrr.io/r/base/nrow.html'>nrow</a></span><span class='op'>(</span><span class='va'>ym_dgelrt</span><span class='op'>)</span><span class='op'>)</span>
    <span class='va'>ym_status</span><span class='op'>[</span><span class='va'>ym_index</span><span class='op'>[[</span><span class='va'>n</span><span class='op'>]</span><span class='op'>]</span><span class='op'>]</span> <span class='op'>&lt;-</span> <span class='va'>n</span>
    <span class='fu'>plotMD</span><span class='op'>(</span>
      <span class='va'>ym_dgelrt</span>,
      status <span class='op'>=</span> <span class='va'>ym_status</span>,
      values <span class='op'>=</span> <span class='va'>n</span>,
      hl.col <span class='op'>=</span> <span class='st'>"red"</span>,
      legend <span class='op'>=</span> <span class='st'>"bottomright"</span>,
      main <span class='op'>=</span> <span class='va'>n</span><span class='op'>)</span>
    <span class='fu'><a href='https://rdrr.io/r/graphics/abline.html'>abline</a></span><span class='op'>(</span>h <span class='op'>=</span> <span class='fl'>0</span>, col <span class='op'>=</span> <span class='st'>"darkgrey"</span><span class='op'>)</span>
  <span class='op'>}</span>
  
  <span class='kw'>if</span> <span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>is.data.frame</a></span><span class='op'>(</span><span class='va'>ym_index</span><span class='op'>[[</span><span class='va'>n</span><span class='op'>]</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>{</span>
    <span class='fu'>barcodeplot</span><span class='op'>(</span>
      <span class='va'>ym_dgelrt</span><span class='op'>$</span><span class='va'>table</span><span class='op'>$</span><span class='va'>logFC</span>,
      <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>ym</span><span class='op'>)</span> <span class='op'>%in%</span> <span class='va'>ym_index</span><span class='op'>[[</span><span class='va'>n</span><span class='op'>]</span><span class='op'>]</span><span class='op'>[</span>, <span class='fl'>1</span><span class='op'>]</span>,
      gene.weights <span class='op'>=</span> <span class='va'>ym_index</span><span class='op'>[[</span><span class='va'>n</span><span class='op'>]</span><span class='op'>]</span><span class='op'>[</span>
        <span class='va'>ym_index</span><span class='op'>[[</span><span class='va'>n</span><span class='op'>]</span><span class='op'>]</span><span class='op'>[</span>, <span class='fl'>1</span><span class='op'>]</span> <span class='op'>%in%</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>ym</span><span class='op'>)</span>, <span class='fl'>2</span><span class='op'>]</span><span class='op'>)</span>
  <span class='op'>}</span> <span class='kw'>else</span> <span class='op'>{</span>
    <span class='fu'>barcodeplot</span><span class='op'>(</span>
      <span class='va'>ym_dgelrt</span><span class='op'>$</span><span class='va'>table</span><span class='op'>$</span><span class='va'>logFC</span>,
      <span class='va'>ym_index</span><span class='op'>[[</span><span class='va'>n</span><span class='op'>]</span><span class='op'>]</span><span class='op'>)</span>
  <span class='op'>}</span>
<span class='op'>}</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:unnamed-chunk-48"></span>
<img src="C086_Kinkel.multi-sample_comparisons_files/figure-html5/unnamed-chunk-48-1.png" alt="MD-plot and barcode plot of DEGs in supplied gene sets. Directional gene sets have (1) MD plot points coloured according to the statistical significance of the gene **in the gene set** (2) barcode plot 'weights' given by the logFC of the gene **in the gene set**. For the barcode plot, genes are represented by bars and are ranked from left to right by increasing log-fold change. This forms the barcode-like pattern. The line (or *worm*) above the barcode shows the relative local enrichment of the vertical bars in each part of the plot. The dotted horizontal line indicates neutral enrichment; the worm above the dotted line shows enrichment while the worm below the dotted line shows depletion." width="624" />
<p class="caption">
Figure 12: MD-plot and barcode plot of DEGs in supplied gene sets. Directional gene sets have (1) MD plot points coloured according to the statistical significance of the gene <strong>in the gene set</strong> (2) barcode plot ‘weights’ given by the logFC of the gene <strong>in the gene set</strong>. For the barcode plot, genes are represented by bars and are ranked from left to right by increasing log-fold change. This forms the barcode-like pattern. The line (or <em>worm</em>) above the barcode shows the relative local enrichment of the vertical bars in each part of the plot. The dotted horizontal line indicates neutral enrichment; the worm above the dotted line shows enrichment while the worm below the dotted line shows depletion.
</p>
</div>
</div>
<h1 id="female-only-analysis-of-aggregated-technical-replicates">Female-only: analysis of aggregated technical replicates</h1>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>yf_differences</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/RUVSeq/man/makeGroups.html'>makeGroups</a></span><span class='op'>(</span><span class='va'>yf</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>smchd1_genotype_updated</span><span class='op'>)</span>
<span class='va'>yf_ruv</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/RUVSeq/man/RUVs.html'>RUVs</a></span><span class='op'>(</span>
    x <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/EDASeq/man/betweenLaneNormalization-methods.html'>betweenLaneNormalization</a></span><span class='op'>(</span><span class='va'>yf</span><span class='op'>$</span><span class='va'>counts</span>, <span class='st'>"upper"</span><span class='op'>)</span>,
    cIdx <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>yf</span><span class='op'>)</span>,
    k <span class='op'>=</span> <span class='fl'>1</span>,
    scIdx <span class='op'>=</span> <span class='va'>yf_differences</span><span class='op'>)</span>

<span class='va'>yf_design</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/model.matrix.html'>model.matrix</a></span><span class='op'>(</span>
  <span class='op'>~</span><span class='fl'>0</span> <span class='op'>+</span> <span class='va'>smchd1_genotype_updated</span> <span class='op'>+</span> <span class='va'>yf_ruv</span><span class='op'>$</span><span class='va'>W</span>,
  <span class='va'>yf</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>yf_design</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/grep.html'>sub</a></span><span class='op'>(</span>
  <span class='st'>"ruv\\$|smchd1_genotype_updated"</span>,
  <span class='st'>""</span>,
  <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>yf_design</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<h2 id="mds-2">MDS</h2>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>yf_mds</span> <span class='op'>&lt;-</span> <span class='fu'>plotMDS</span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/SingleCellExperiment/man/assays.html'>cpm</a></span><span class='op'>(</span><span class='va'>yf_ruv</span><span class='op'>$</span><span class='va'>normalizedCounts</span>, log <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>, plot <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>

<span class='va'>yf_df</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cbind.html'>cbind</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>yf_mds</span><span class='op'>$</span><span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>yf_mds</span><span class='op'>$</span><span class='va'>y</span><span class='op'>)</span>, <span class='va'>yf</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>yf_df</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>yf</span><span class='op'>)</span>

<span class='va'>p1</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span>, colour <span class='op'>=</span> <span class='va'>smchd1_genotype_updated</span>, label <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>yf_df</span><span class='op'>)</span><span class='op'>)</span>,
  data <span class='op'>=</span> <span class='va'>yf_df</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html'>geom_text_repel</a></span><span class='op'>(</span>size <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 1"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 2"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>smchd1_genotype_updated_colours</span><span class='op'>)</span>
<span class='va'>p2</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span>, colour <span class='op'>=</span> <span class='va'>sex</span>, label <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>yf_df</span><span class='op'>)</span><span class='op'>)</span>,
  data <span class='op'>=</span> <span class='va'>yf_df</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html'>geom_text_repel</a></span><span class='op'>(</span>size <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 1"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 2"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>sex_colours</span><span class='op'>)</span>
<span class='va'>p3</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span>, colour <span class='op'>=</span> <span class='va'>mouse_number</span>, label <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>yf_df</span><span class='op'>)</span><span class='op'>)</span>,
  data <span class='op'>=</span> <span class='va'>yf_df</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html'>geom_text_repel</a></span><span class='op'>(</span>size <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 1"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 2"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>mouse_number_colours</span><span class='op'>)</span>
<span class='va'>p4</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span>, colour <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log10</a></span><span class='op'>(</span><span class='va'>lib.size</span><span class='op'>)</span>, label <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>yf_df</span><span class='op'>)</span><span class='op'>)</span>,
  data <span class='op'>=</span> <span class='va'>yf_df</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html'>geom_text_repel</a></span><span class='op'>(</span>size <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_viridis.html'>scale_colour_viridis_c</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 1"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Leading logFC dim 2"</span><span class='op'>)</span>

<span class='va'>p1</span> <span class='op'>+</span> <span class='va'>p2</span> <span class='op'>+</span> <span class='va'>p3</span> <span class='op'>+</span> <span class='va'>p4</span> <span class='op'>+</span> <span class='fu'><a href='https://patchwork.data-imaginist.com/reference/plot_layout.html'>plot_layout</a></span><span class='op'>(</span>ncol <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:unnamed-chunk-50"></span>
<img src="C086_Kinkel.multi-sample_comparisons_files/figure-html5/unnamed-chunk-50-1.png" alt="MDS plots of aggregated technical replicates following RUV coloured by various experimental factors." width="624" />
<p class="caption">
Figure 13: MDS plots of aggregated technical replicates following RUV coloured by various experimental factors.
</p>
</div>
</div>
<h2 id="de-analysis-2">DE analysis</h2>
<p>Fit a model using <em><a href="https://bioconductor.org/packages/3.12/edgeR">edgeR</a></em> with a term for <code>smchd1_genotype_updated</code>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>yf</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/estimateDisp.html'>estimateDisp</a></span><span class='op'>(</span><span class='va'>yf</span>, <span class='va'>yf_design</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/graphics/par.html'>par</a></span><span class='op'>(</span>mfrow <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/plotBCV.html'>plotBCV</a></span><span class='op'>(</span><span class='va'>yf</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>yf_dgeglm</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/glmfit.html'>glmFit</a></span><span class='op'>(</span><span class='va'>yf</span>, <span class='va'>yf_design</span><span class='op'>)</span>
<span class='va'>yf_contrast</span> <span class='op'>&lt;-</span> <span class='fu'>makeContrasts</span><span class='op'>(</span><span class='va'>Del</span> <span class='op'>-</span> <span class='va'>Het</span>, levels <span class='op'>=</span> <span class='va'>yf_design</span><span class='op'>)</span>
<span class='va'>yf_dgelrt</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/glmfit.html'>glmLRT</a></span><span class='op'>(</span><span class='va'>yf_dgeglm</span>, contrast <span class='op'>=</span> <span class='va'>yf_contrast</span><span class='op'>)</span>
<span class='va'>yf_dt</span> <span class='op'>&lt;-</span> <span class='fu'>decideTests</span><span class='op'>(</span><span class='va'>yf_dgelrt</span><span class='op'>)</span>

<span class='fu'><a href='https://rdrr.io/r/base/summary.html'>summary</a></span><span class='op'>(</span><span class='va'>yf_dt</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span>caption <span class='op'>=</span> <span class='st'>"Number of DEGs (FDR &lt; 0.05)"</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<table>
<caption><span id="tab:unnamed-chunk-52">Table 11: </span>Number of DEGs (FDR &lt; 0.05)</caption>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">1<em>Del -1</em>Het</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Down</td>
<td style="text-align: right;">40</td>
</tr>
<tr class="even">
<td style="text-align: left;">NotSig</td>
<td style="text-align: right;">11468</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Up</td>
<td style="text-align: right;">5</td>
</tr>
</tbody>
</table>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/r/graphics/par.html'>par</a></span><span class='op'>(</span>mfrow <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span>
<span class='fu'>plotMD</span><span class='op'>(</span><span class='va'>yf_dgelrt</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:unnamed-chunk-53"></span>
<img src="C086_Kinkel.multi-sample_comparisons_files/figure-html5/unnamed-chunk-53-1.png" alt="MD plot highlighting DEGs (red)." width="624" />
<p class="caption">
Figure 14: MD plot highlighting DEGs (red).
</p>
</div>
</div>
<p>An interactive <em><a href="https://bioconductor.org/packages/3.12/Glimma">Glimma</a></em> MD plot of the differential expression results is available from <a href="../output/DEGs/reads/glimma-plots/aggregated_tech_reps.females.md-plot.html"><code>output/DEGs/reads/glimma-plots/aggregated_tech_reps.females.md-plot.html</code></a>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>yf_tt</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/topTags.html'>topTags</a></span><span class='op'>(</span><span class='va'>yf_dgelrt</span>, n <span class='op'>=</span> <span class='cn'>Inf</span>, sort.by <span class='op'>=</span> <span class='st'>"none"</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/pkg/Glimma/man/glMDPlot.html'>glMDPlot</a></span><span class='op'>(</span>
  x <span class='op'>=</span> <span class='va'>yf_dgelrt</span>,
  counts <span class='op'>=</span> <span class='va'>yf</span>,
  anno <span class='op'>=</span> <span class='va'>yf</span><span class='op'>$</span><span class='va'>genes</span>,
  display.columns <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"ENSEMBL.SYMBOL"</span><span class='op'>)</span>,
  groups <span class='op'>=</span> <span class='va'>yf</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span>,
  samples <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>yf</span><span class='op'>)</span>,
  status <span class='op'>=</span> <span class='va'>yf_dt</span>,
  transform <span class='op'>=</span> <span class='cn'>TRUE</span>,
  sample.cols <span class='op'>=</span> <span class='va'>yf</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>smchd1_genotype_updated_colours</span>,
  path <span class='op'>=</span> <span class='va'>outdir</span>,
  html <span class='op'>=</span> <span class='st'>"aggregated_tech_reps.females.md-plot"</span>,
  launch <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>yf_tt</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/topTags.html'>topTags</a></span><span class='op'>(</span><span class='va'>yf_dgelrt</span>, p.value <span class='op'>=</span> <span class='fl'>0.05</span>, n <span class='op'>=</span> <span class='cn'>Inf</span><span class='op'>)</span>
<span class='kw'>if</span> <span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/nrow.html'>nrow</a></span><span class='op'>(</span><span class='va'>yf_tt</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>{</span>
  <span class='va'>yf_tt</span><span class='op'>$</span><span class='va'>table</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"logFC"</span>, <span class='st'>"logCPM"</span>, <span class='st'>"LR"</span>, <span class='st'>"PValue"</span>, <span class='st'>"FDR"</span><span class='op'>)</span><span class='op'>]</span> <span class='op'>%&gt;%</span>
    <span class='fu'>DT</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/DT/man/datatable.html'>datatable</a></span><span class='op'>(</span>caption <span class='op'>=</span> <span class='st'>"DEGs (FDR &lt; 0.05)"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
    <span class='fu'>DT</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/DT/man/formatCurrency.html'>formatSignif</a></span><span class='op'>(</span>
      <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"logFC"</span>, <span class='st'>"logCPM"</span>, <span class='st'>"LR"</span>, <span class='st'>"PValue"</span>, <span class='st'>"FDR"</span><span class='op'>)</span>,
      digits <span class='op'>=</span> <span class='fl'>3</span><span class='op'>)</span>
<span class='op'>}</span>
</code></pre>
</div>
</details>
<div id="htmlwidget-a574e3ac2363513b0633" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-a574e3ac2363513b0633">{"x":{"filter":"none","caption":"<caption>DEGs (FDR &lt; 0.05)<\/caption>","data":[["Cbx2","Ciart","Myof","Ppp1r12b","Tbc1d30","Frs3","Cap1","Abcd2","Gli3","Eya4","Scx","Il12rb2","Tmem43","Mmp16","1810055G02Rik","Zswim4","Homer2","Rab2b","Epha2","Boc","Obsl1","Pard3b","Chst7","Ehhadh","Dixdc1","Angptl3","Acvr1","Met","Smchd1","Evi5","Trpc2","Stard9","Zg16","Nov","Lpcat2","Pitx2","Tepp","Epas1","Arhgef26","Stat2","Cpne5","Cd48","Ndrg4","Zfp964","F830016B08Rik"],[-3.33340556668766,-7.36124862427518,-1.81001349014673,-2.61649453725001,-9.31477003185678,-7.72856454840923,-3.78320562877943,8.57402677129352,-7.62450423681063,-3.60071558196932,-8.72750710653342,2.41116575386001,-2.42992002359739,-5.57220269787109,-5.00982384326111,-1.84844312181358,-5.90107749410308,-1.81529370630774,-5.83097039708437,-7.14560688938933,-2.16202556528404,-1.42672975905515,7.72549854726001,-6.9544585940358,-8.10500889539785,5.53112673443314,-2.27330035619674,-6.12949917689081,-1.19985406501546,-2.54895686501839,-7.77072770555825,-7.56856917700944,-5.82109678512071,-6.108592539134,-5.52476774009755,-7.04918248604675,-7.96015680754604,-4.973684114471,-4.82020685435736,-1.57094460637574,-5.72725050362082,5.7042746828756,-5.29729829085284,-5.67472655017762,-2.57154557182144],[5.4001574825125,2.6794686682344,6.46529892025047,4.85156123458674,2.0090405570818,2.13722259502331,5.90422717276023,1.3351714496492,2.09514569100563,2.52736634824971,1.39718142385171,4.83498709391503,4.58850629548745,2.97048385883735,2.86496682345634,5.20073661918654,2.58643486774099,5.89322207895469,2.20033479649356,1.62392391057993,4.95891319736005,6.02787712934545,0.517883802907564,2.15140030538593,0.901767417043129,2.5478262567689,5.41263490728589,1.51164592826699,6.67689039739311,4.20545041601709,0.567591199526342,0.43434085093078,0.945352581241706,0.408166174976144,2.00916667356489,0.700824158924326,0.776360278866279,2.81402535830118,2.06241183727126,5.52489736226029,0.183741362389105,1.84486862262646,1.41434707083858,1.80918716414133,4.92562969662688],[46.2912493845141,28.673821146274,27.6535457174643,27.0936658716971,23.6343062270877,22.5393471765993,21.5415155448841,21.4005149100007,21.3571020540174,21.1221929531072,21.0060040304759,20.5693416332071,20.1278648760503,19.6112253229634,19.5270501013028,18.6627215410019,18.586219472734,18.3333554376926,18.2190018973839,17.7310028586791,17.7117093174675,17.6378055915987,17.3936333573772,17.099299651754,16.9706900201559,16.5786622263442,16.2674797434477,16.0389357766669,15.8761567587209,15.3829048105792,15.36786219293,15.0543732943527,14.9988371906595,14.911379686967,14.570105004886,14.4986668322858,14.4938785352811,14.4565908959145,14.3996302769674,14.2130084157672,14.0969966296566,14.0891195802164,14.0319560156648,13.9985012005063,13.9456264502067],[1.01918122399409e-11,8.5653840103287e-08,1.45105843534815e-07,1.93832009694549e-07,1.16492075319535e-06,2.05883049438857e-06,3.46251137205583e-06,3.72671089323356e-06,3.81205956717768e-06,4.30908678777553e-06,4.57846332332398e-06,5.75097800152221e-06,7.24336650727477e-06,9.4909937320055e-06,9.918517051223e-06,1.56002889769628e-05,1.62390206300236e-05,1.85431930077124e-05,1.96904517376871e-05,2.54445721716427e-05,2.57039494007433e-05,2.67223123589277e-05,3.03842024434647e-05,3.54753843418692e-05,3.79613482062078e-05,4.66732006168026e-05,5.49998380161622e-05,6.20531675657681e-05,6.76253405052443e-05,8.77788528568424e-05,8.84804689919593e-05,0.000104457942765126,0.000107577443988603,0.000112681218956403,0.000135040015512074,0.000140258769797777,0.000140615744527728,0.000143427076401639,0.000147831332377855,0.000163238142454253,0.000173620783945543,0.000174349406577065,0.000179730103401742,0.000182956414536748,0.000188174938199812],[1.1733833431844e-07,0.000493066330554572,0.000556867858872108,0.000557896981903335,0.00268234652630762,0.00395055258031593,0.00479198620376627,0.00479198620376627,0.00479198620376627,0.00479198620376627,0.00479198620376627,0.00551758414429376,0.00641483681525034,0.0076127924540487,0.0076127924540487,0.0109976379125566,0.0109976379125566,0.0118604322832107,0.0119313774134732,0.0139842719176516,0.0139842719176516,0.0139842719176516,0.0152092749013743,0.0170178374969975,0.0174819600759228,0.0206672522577403,0.0234523383362991,0.0255149327923103,0.026847260180582,0.0328605045001428,0.0328605045001428,0.0375314882618419,0.0375314882618419,0.0381558492307374,0.0434546297529491,0.0434546297529491,0.0434546297529491,0.0434546297529491,0.0436405674273395,0.0469840183518953,0.0477924932838511,0.0477924932838511,0.0478722091036723,0.0478722091036723,0.0481435125220986]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>logFC<\/th>\n      <th>logCPM<\/th>\n      <th>LR<\/th>\n      <th>PValue<\/th>\n      <th>FDR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"targets":1,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 3, 3, \",\", \".\");\n  }"},{"targets":2,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 3, 3, \",\", \".\");\n  }"},{"targets":3,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 3, 3, \",\", \".\");\n  }"},{"targets":4,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 3, 3, \",\", \".\");\n  }"},{"targets":5,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 3, 3, \",\", \".\");\n  }"},{"className":"dt-right","targets":[1,2,3,4,5]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":["options.columnDefs.0.render","options.columnDefs.1.render","options.columnDefs.2.render","options.columnDefs.3.render","options.columnDefs.4.render"],"jsHooks":[]}</script>
</div>
<p>CSV file of results available from <a href="../output/DEGs/reads/aggregated_tech_reps.females.DEGs.csv.gz"><code>output/DEGs/reads/aggregated_tech_reps.females.DEGs.csv.gz</code></a>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>gzout</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/connections.html'>gzfile</a></span><span class='op'>(</span>
  description <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span><span class='va'>outdir</span>, <span class='st'>"aggregated_tech_reps.females.DEGs.csv.gz"</span><span class='op'>)</span>,
  open <span class='op'>=</span> <span class='st'>"wb"</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/utils/write.table.html'>write.csv</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/topTags.html'>topTags</a></span><span class='op'>(</span><span class='va'>yf_dgelrt</span>, n <span class='op'>=</span> <span class='cn'>Inf</span><span class='op'>)</span>,
  <span class='va'>gzout</span>,
  <span class='co'># NOTE: quote = TRUE needed because some fields contain commas.</span>
  quote <span class='op'>=</span> <span class='cn'>TRUE</span>,
  row.names <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/connections.html'>close</a></span><span class='op'>(</span><span class='va'>gzout</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>heatmapForPaper</span> <span class='op'>&lt;-</span> <span class='kw'>function</span><span class='op'>(</span><span class='op'>)</span> <span class='op'>{</span>
  <span class='fu'><a href='https://rdrr.io/pkg/pheatmap/man/pheatmap.html'>pheatmap</a></span><span class='op'>(</span>
    <span class='fu'><a href='https://rdrr.io/pkg/SingleCellExperiment/man/assays.html'>cpm</a></span><span class='op'>(</span><span class='va'>yf</span>, log <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>yf_tt</span><span class='op'>)</span>, <span class='op'>]</span>,
    cluster_rows <span class='op'>=</span> <span class='cn'>TRUE</span>,
    cluster_cols <span class='op'>=</span> <span class='cn'>FALSE</span>,
    color <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/grDevices/palettes.html'>hcl.colors</a></span><span class='op'>(</span><span class='fl'>101</span>, <span class='st'>"Blue-Red 3"</span><span class='op'>)</span>,
    breaks <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span><span class='op'>(</span><span class='op'>-</span><span class='fl'>3</span>, <span class='fl'>3</span>, length.out <span class='op'>=</span> <span class='fl'>101</span><span class='op'>)</span>,
    scale <span class='op'>=</span> <span class='st'>"row"</span>,
    annotation_col <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>
      genotype <span class='op'>=</span> <span class='va'>yf</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>smchd1_genotype_updated</span>,
      sex <span class='op'>=</span> <span class='va'>yf</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>sex</span>,
      row.names <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>yf</span><span class='op'>)</span><span class='op'>)</span>,
    annotation_colors <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>
      genotype <span class='op'>=</span> <span class='va'>smchd1_genotype_updated_colours</span>,
      sex <span class='op'>=</span> <span class='va'>sex_colours</span><span class='op'>)</span>,
    main <span class='op'>=</span> <span class='st'>"Female samples"</span>,
    fontsize <span class='op'>=</span> <span class='fl'>9</span><span class='op'>)</span>
<span class='op'>}</span>
</code></pre>
</div>
</details>
</div>
<p>Figure <a href="#fig:heatmap-yf">15</a> is a heatmap showing the expression of the DEGs.</p>
<aside>
A PDF version of this figure is available in <a href="../output/figures/paper/heatmap.pdf"><code>output/figures/paper/heatmap.pdf</code></a>.
</aside>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'>heatmapForPaper</span><span class='op'>(</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:heatmap-yf"></span>
<img src="C086_Kinkel.multi-sample_comparisons_files/figure-html5/heatmap-yf-1.png" alt="Heatmap of upper-quartile-normalzied logCPM values for the DEGs." width="624" />
<p class="caption">
Figure 15: Heatmap of upper-quartile-normalzied logCPM values for the DEGs.
</p>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/r/base/files2.html'>dir.create</a></span><span class='op'>(</span><span class='fu'><a href='https://here.r-lib.org//reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"output/figures/paper"</span><span class='op'>)</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/grDevices/pdf.html'>pdf</a></span><span class='op'>(</span><span class='fu'><a href='https://here.r-lib.org//reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"output/figures/paper/heatmap.pdf"</span><span class='op'>)</span>, height <span class='op'>=</span> <span class='fl'>9</span>, width <span class='op'>=</span> <span class='fl'>6</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/graphics/par.html'>par</a></span><span class='op'>(</span>mfrow <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span>
<span class='fu'>heatmapForPaper</span><span class='op'>(</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/grDevices/dev.html'>dev.off</a></span><span class='op'>(</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<h2 id="gene-set-tests-2">Gene set tests</h2>
<p>Gene set testing is commonly used to interpret the differential expression results in a biological context. Here we apply various gene set tests to the DEG lists.</p>
<h3 id="goana-2">goana</h3>
<p>We use the <code>goana()</code> function from the <em><a href="https://bioconductor.org/packages/3.12/limma">limma</a></em> R/Bioconductor package to test for over-representation of gene ontology (GO) terms in each DEG list.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>list_of_go</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/lapply.html'>lapply</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>yf_contrast</span><span class='op'>)</span>, <span class='kw'>function</span><span class='op'>(</span><span class='va'>j</span><span class='op'>)</span> <span class='op'>{</span>
  <span class='va'>dgelrt</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/glmfit.html'>glmLRT</a></span><span class='op'>(</span><span class='va'>yf_dgeglm</span>, contrast <span class='op'>=</span> <span class='va'>yf_contrast</span><span class='op'>[</span>, <span class='va'>j</span><span class='op'>]</span><span class='op'>)</span>
  
  <span class='va'>go</span> <span class='op'>&lt;-</span> <span class='fu'>goana</span><span class='op'>(</span>
    de <span class='op'>=</span> <span class='va'>dgelrt</span>,
    geneid <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/lapply.html'>sapply</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/strsplit.html'>strsplit</a></span><span class='op'>(</span><span class='va'>dgelrt</span><span class='op'>$</span><span class='va'>genes</span><span class='op'>$</span><span class='va'>NCBI.ENTREZID</span>, <span class='st'>"; "</span><span class='op'>)</span>, <span class='st'>"[["</span>, <span class='fl'>1</span><span class='op'>)</span>,
    species <span class='op'>=</span> <span class='st'>"Mm"</span><span class='op'>)</span>
  <span class='va'>gzout</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/connections.html'>gzfile</a></span><span class='op'>(</span>
    description <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span>
      <span class='va'>outdir</span>, 
      <span class='st'>"aggregated_tech_reps.females.goana.csv.gz"</span><span class='op'>)</span>,
    open <span class='op'>=</span> <span class='st'>"wb"</span><span class='op'>)</span>
  <span class='fu'><a href='https://rdrr.io/r/utils/write.table.html'>write.csv</a></span><span class='op'>(</span>
    <span class='fu'>topGO</span><span class='op'>(</span><span class='va'>go</span>, number <span class='op'>=</span> <span class='cn'>Inf</span><span class='op'>)</span>,
    <span class='va'>gzout</span>,
    row.names <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
  <span class='fu'><a href='https://rdrr.io/r/base/connections.html'>close</a></span><span class='op'>(</span><span class='va'>gzout</span><span class='op'>)</span>
  <span class='va'>go</span>
<span class='op'>}</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>list_of_go</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>yf_contrast</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>CSV files of the results are available from <a href="../output/DEGs/reads/"><code>output/DEGs/reads/</code></a>.</p>
<p>An example of the results are shown below.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'>topGO</span><span class='op'>(</span><span class='va'>list_of_go</span><span class='op'>[[</span><span class='fl'>1</span><span class='op'>]</span><span class='op'>]</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span>
    caption <span class='op'>=</span> <span class='st'>'`goana()` produces a table with a row for each GO term and the following columns: **Term** (GO term); **Ont** (ontology that the GO term belongs to. Possible values are "BP", "CC" and "MF"); **N** (number of genes in the GO term); **Up** (number of up-regulated differentially expressed genes); **Down** (number of down-regulated differentially expressed genes); **P.Up** (p-value for over-representation of GO term in up-regulated genes); **P.Down** (p-value for over-representation of GO term in down-regulated genes)'</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<table>
<caption><span id="tab:unnamed-chunk-60">Table 12: </span><code>goana()</code> produces a table with a row for each GO term and the following columns: <strong>Term</strong> (GO term); <strong>Ont</strong> (ontology that the GO term belongs to. Possible values are “BP,” “CC” and “MF”); <strong>N</strong> (number of genes in the GO term); <strong>Up</strong> (number of up-regulated differentially expressed genes); <strong>Down</strong> (number of down-regulated differentially expressed genes); <strong>P.Up</strong> (p-value for over-representation of GO term in up-regulated genes); <strong>P.Down</strong> (p-value for over-representation of GO term in down-regulated genes)</caption>
<colgroup>
<col style="width: 11%" />
<col style="width: 49%" />
<col style="width: 4%" />
<col style="width: 5%" />
<col style="width: 3%" />
<col style="width: 5%" />
<col style="width: 10%" />
<col style="width: 10%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;">Term</th>
<th style="text-align: left;">Ont</th>
<th style="text-align: right;">N</th>
<th style="text-align: right;">Up</th>
<th style="text-align: right;">Down</th>
<th style="text-align: right;">P.Up</th>
<th style="text-align: right;">P.Down</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"><a href="GO:0009653" class="uri">GO:0009653</a></td>
<td style="text-align: left;">anatomical structure morphogenesis</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">1565</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">19</td>
<td style="text-align: right;">0.5223220</td>
<td style="text-align: right;">0.0000003</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="GO:0048562" class="uri">GO:0048562</a></td>
<td style="text-align: left;">embryonic organ morphogenesis</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">156</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0000156</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="GO:0050996" class="uri">GO:0050996</a></td>
<td style="text-align: left;">positive regulation of lipid catabolic process</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">15</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0000161</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="GO:0048754" class="uri">GO:0048754</a></td>
<td style="text-align: left;">branching morphogenesis of an epithelial tube</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">93</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0000171</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="GO:0048568" class="uri">GO:0048568</a></td>
<td style="text-align: left;">embryonic organ development</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">268</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">7</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0000353</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="GO:0035993" class="uri">GO:0035993</a></td>
<td style="text-align: left;">deltoid tuberosity development</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0000360</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="GO:0061138" class="uri">GO:0061138</a></td>
<td style="text-align: left;">morphogenesis of a branching epithelium</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">110</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0000385</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="GO:0007166" class="uri">GO:0007166</a></td>
<td style="text-align: left;">cell surface receptor signaling pathway</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">1413</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">15</td>
<td style="text-align: right;">0.4842227</td>
<td style="text-align: right;">0.0000454</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="GO:0001763" class="uri">GO:0001763</a></td>
<td style="text-align: left;">morphogenesis of a branching structure</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">118</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0000539</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="GO:0061061" class="uri">GO:0061061</a></td>
<td style="text-align: left;">muscle structure development</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">393</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0000544</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="GO:0009887" class="uri">GO:0009887</a></td>
<td style="text-align: left;">animal organ morphogenesis</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">533</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">9</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0000746</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="GO:0014902" class="uri">GO:0014902</a></td>
<td style="text-align: left;">myotube differentiation</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">68</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0000902</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="GO:0050994" class="uri">GO:0050994</a></td>
<td style="text-align: left;">regulation of lipid catabolic process</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">35</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0000911</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="GO:0090596" class="uri">GO:0090596</a></td>
<td style="text-align: left;">sensory organ morphogenesis</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">137</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0001095</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="GO:0048593" class="uri">GO:0048593</a></td>
<td style="text-align: left;">camera-type eye morphogenesis</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">72</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0001128</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="GO:0090630" class="uri">GO:0090630</a></td>
<td style="text-align: left;">activation of GTPase activity</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">75</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0001322</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="GO:0019199" class="uri">GO:0019199</a></td>
<td style="text-align: left;">transmembrane receptor protein kinase activity</td>
<td style="text-align: left;">MF</td>
<td style="text-align: right;">30</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0001523</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="GO:0048557" class="uri">GO:0048557</a></td>
<td style="text-align: left;">embryonic digestive tract morphogenesis</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0001786</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="GO:0003170" class="uri">GO:0003170</a></td>
<td style="text-align: left;">heart valve development</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">32</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0001852</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="GO:0001707" class="uri">GO:0001707</a></td>
<td style="text-align: left;">mesoderm formation</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">33</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0002032</td>
</tr>
</tbody>
</table>
</div>
<h3 id="kegga-2">kegga</h3>
<p>We use the <code>kegga()</code> function from the <em><a href="https://bioconductor.org/packages/3.12/limma">limma</a></em> R/Bioconductor package to test for over-representation of KEGG pathways in each DEG list.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>list_of_kegg</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/lapply.html'>lapply</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>yf_contrast</span><span class='op'>)</span>, <span class='kw'>function</span><span class='op'>(</span><span class='va'>j</span><span class='op'>)</span> <span class='op'>{</span>
  <span class='va'>dgelrt</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/glmfit.html'>glmLRT</a></span><span class='op'>(</span><span class='va'>yf_dgeglm</span>, contrast <span class='op'>=</span> <span class='va'>yf_contrast</span><span class='op'>[</span>, <span class='va'>j</span><span class='op'>]</span><span class='op'>)</span>
  
  <span class='va'>kegg</span> <span class='op'>&lt;-</span> <span class='fu'>kegga</span><span class='op'>(</span>
    de <span class='op'>=</span> <span class='va'>dgelrt</span>,
    geneid <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/lapply.html'>sapply</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/strsplit.html'>strsplit</a></span><span class='op'>(</span><span class='va'>dgelrt</span><span class='op'>$</span><span class='va'>genes</span><span class='op'>$</span><span class='va'>NCBI.ENTREZID</span>, <span class='st'>"; "</span><span class='op'>)</span>, <span class='st'>"[["</span>, <span class='fl'>1</span><span class='op'>)</span>,
    species <span class='op'>=</span> <span class='st'>"Mm"</span><span class='op'>)</span>
  <span class='va'>gzout</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/connections.html'>gzfile</a></span><span class='op'>(</span>
    description <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span>
      <span class='va'>outdir</span>,
      <span class='st'>"aggregated_tech_reps.females.kegga.csv.gz"</span><span class='op'>)</span>,
    open <span class='op'>=</span> <span class='st'>"wb"</span><span class='op'>)</span>
  <span class='fu'><a href='https://rdrr.io/r/utils/write.table.html'>write.csv</a></span><span class='op'>(</span>
    <span class='fu'>topKEGG</span><span class='op'>(</span><span class='va'>kegg</span>, number <span class='op'>=</span> <span class='cn'>Inf</span><span class='op'>)</span>,
    <span class='va'>gzout</span>,
    row.names <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
  <span class='fu'><a href='https://rdrr.io/r/base/connections.html'>close</a></span><span class='op'>(</span><span class='va'>gzout</span><span class='op'>)</span>
  <span class='va'>kegg</span>
<span class='op'>}</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>list_of_kegg</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>yf_contrast</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>CSV files of the results are available from <a href="../output/DEGs/reads/"><code>output/DEGs/reads/</code></a>.</p>
<p>An example of the results are shown below.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'>topKEGG</span><span class='op'>(</span><span class='va'>list_of_kegg</span><span class='op'>[[</span><span class='fl'>1</span><span class='op'>]</span><span class='op'>]</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span>
    caption <span class='op'>=</span> <span class='st'>'`kegga()` produces a table with a row for each KEGG pathway ID and the following columns: **Pathway** (KEGG pathway); **N** (number of genes in the GO term); **Up** (number of up-regulated differentially expressed genes); **Down** (number of down-regulated differentially expressed genes); **P.Up** (p-value for over-representation of KEGG pathway in up-regulated genes); **P.Down** (p-value for over-representation of KEGG pathway in down-regulated genes)'</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<table>
<caption><span id="tab:unnamed-chunk-62">Table 13: </span><code>kegga()</code> produces a table with a row for each KEGG pathway ID and the following columns: <strong>Pathway</strong> (KEGG pathway); <strong>N</strong> (number of genes in the GO term); <strong>Up</strong> (number of up-regulated differentially expressed genes); <strong>Down</strong> (number of down-regulated differentially expressed genes); <strong>P.Up</strong> (p-value for over-representation of KEGG pathway in up-regulated genes); <strong>P.Down</strong> (p-value for over-representation of KEGG pathway in down-regulated genes)</caption>
<colgroup>
<col style="width: 11%" />
<col style="width: 61%" />
<col style="width: 3%" />
<col style="width: 2%" />
<col style="width: 4%" />
<col style="width: 8%" />
<col style="width: 8%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;">Pathway</th>
<th style="text-align: right;">N</th>
<th style="text-align: right;">Up</th>
<th style="text-align: right;">Down</th>
<th style="text-align: right;">P.Up</th>
<th style="text-align: right;">P.Down</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">path:mmu04360</td>
<td style="text-align: left;">Axon guidance</td>
<td style="text-align: right;">109</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0065037</td>
</tr>
<tr class="even">
<td style="text-align: left;">path:mmu00532</td>
<td style="text-align: left;">Glycosaminoglycan biosynthesis - chondroitin sulfate / dermatan sulfate</td>
<td style="text-align: right;">15</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0065651</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="odd">
<td style="text-align: left;">path:mmu04340</td>
<td style="text-align: left;">Hedgehog signaling pathway</td>
<td style="text-align: right;">42</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0094658</td>
</tr>
<tr class="even">
<td style="text-align: left;">path:mmu02010</td>
<td style="text-align: left;">ABC transporters</td>
<td style="text-align: right;">25</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0109226</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="odd">
<td style="text-align: left;">path:mmu04979</td>
<td style="text-align: left;">Cholesterol metabolism</td>
<td style="text-align: right;">29</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0126614</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="even">
<td style="text-align: left;">path:mmu04350</td>
<td style="text-align: left;">TGF-beta signaling pathway</td>
<td style="text-align: right;">55</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0158649</td>
</tr>
<tr class="odd">
<td style="text-align: left;">path:mmu05321</td>
<td style="text-align: left;">Inflammatory bowel disease</td>
<td style="text-align: right;">38</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0165646</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="even">
<td style="text-align: left;">path:mmu05100</td>
<td style="text-align: left;">Bacterial invasion of epithelial cells</td>
<td style="text-align: right;">61</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0192945</td>
</tr>
<tr class="odd">
<td style="text-align: left;">path:mmu05211</td>
<td style="text-align: left;">Renal cell carcinoma</td>
<td style="text-align: right;">62</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0198939</td>
</tr>
<tr class="even">
<td style="text-align: left;">path:mmu04658</td>
<td style="text-align: left;">Th1 and Th2 cell differentiation</td>
<td style="text-align: right;">65</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0282002</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="odd">
<td style="text-align: left;">path:mmu04650</td>
<td style="text-align: left;">Natural killer cell mediated cytotoxicity</td>
<td style="text-align: right;">67</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0290577</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="even">
<td style="text-align: left;">path:mmu04146</td>
<td style="text-align: left;">Peroxisome</td>
<td style="text-align: right;">67</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0.0290577</td>
<td style="text-align: right;">0.2104325</td>
</tr>
<tr class="odd">
<td style="text-align: left;">path:mmu05200</td>
<td style="text-align: left;">Pathways in cancer</td>
<td style="text-align: right;">352</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">0.1452135</td>
<td style="text-align: right;">0.0341319</td>
</tr>
<tr class="even">
<td style="text-align: left;">path:mmu04630</td>
<td style="text-align: left;">JAK-STAT signaling pathway</td>
<td style="text-align: right;">94</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0.0405748</td>
<td style="text-align: right;">0.2824260</td>
</tr>
<tr class="odd">
<td style="text-align: left;">path:mmu04060</td>
<td style="text-align: left;">Cytokine-cytokine receptor interaction</td>
<td style="text-align: right;">96</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0.0414235</td>
<td style="text-align: right;">0.2874965</td>
</tr>
<tr class="even">
<td style="text-align: left;">path:mmu00650</td>
<td style="text-align: left;">Butanoate metabolism</td>
<td style="text-align: right;">16</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0547407</td>
</tr>
<tr class="odd">
<td style="text-align: left;">path:mmu00410</td>
<td style="text-align: left;">beta-Alanine metabolism</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0679626</td>
</tr>
<tr class="even">
<td style="text-align: left;">path:mmu00380</td>
<td style="text-align: left;">Tryptophan metabolism</td>
<td style="text-align: right;">21</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0712398</td>
</tr>
<tr class="odd">
<td style="text-align: left;">path:mmu04510</td>
<td style="text-align: left;">Focal adhesion</td>
<td style="text-align: right;">127</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0730508</td>
</tr>
<tr class="even">
<td style="text-align: left;">path:mmu05206</td>
<td style="text-align: left;">MicroRNAs in cancer</td>
<td style="text-align: right;">131</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0770741</td>
</tr>
</tbody>
</table>
</div>
<h3 id="camera-with-msigdb-gene-sets-2">camera with MSigDB gene sets</h3>
<p>We use the <code>camera()</code> function from the <em><a href="https://bioconductor.org/packages/3.12/limma">limma</a></em> R/Bioconductor package to test whether a set of genes is highly ranked relative to other genes in terms of differential expression, accounting for inter-gene correlation. Specifically, we test using gene sets from <a href="https://www.gsea-msigdb.org/gsea/msigdb">MSigDB</a>, namely:</p>
<aside>
We download these gene sets from <a href="http://bioinf.wehi.edu.au/MSigDB/index.html">http://bioinf.wehi.edu.au/MSigDB/index.html</a>
</aside>
<ul>
<li><strong>H</strong>: <strong>hallmark gene sets</strong> are coherently expressed signatures derived by aggregating many MSigDB gene sets to represent well-defined biological states or processes.</li>
<li><strong>C2</strong>: <strong>curated gene sets</strong> from online pathway databases, publications in PubMed, and knowledge of domain experts.</li>
</ul>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>yf_idx</span> <span class='op'>&lt;-</span> <span class='fu'>ids2indices</span><span class='op'>(</span>
  <span class='va'>msigdb</span>, 
  id <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/lapply.html'>sapply</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/strsplit.html'>strsplit</a></span><span class='op'>(</span><span class='va'>yf</span><span class='op'>$</span><span class='va'>genes</span><span class='op'>$</span><span class='va'>NCBI.ENTREZID</span>, <span class='st'>"; "</span><span class='op'>)</span>, <span class='st'>"[["</span>, <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>list_of_camera</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/lapply.html'>lapply</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>yf_contrast</span><span class='op'>)</span>, <span class='kw'>function</span><span class='op'>(</span><span class='va'>j</span><span class='op'>)</span> <span class='op'>{</span>
  <span class='va'>cam</span> <span class='op'>&lt;-</span> <span class='fu'>camera</span><span class='op'>(</span>
    y <span class='op'>=</span> <span class='va'>yf</span>, 
    index <span class='op'>=</span> <span class='va'>yf_idx</span>,
    design <span class='op'>=</span> <span class='va'>yf_design</span>, 
    contrast <span class='op'>=</span> <span class='va'>yf_contrast</span><span class='op'>[</span>, <span class='va'>j</span><span class='op'>]</span><span class='op'>)</span>
  <span class='va'>gzout</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/connections.html'>gzfile</a></span><span class='op'>(</span>
    description <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span>
      <span class='va'>outdir</span>, 
      <span class='st'>"aggregated_tech_reps.females.camera.csv.gz"</span><span class='op'>)</span>,
    open <span class='op'>=</span> <span class='st'>"wb"</span><span class='op'>)</span>
  <span class='fu'><a href='https://rdrr.io/r/utils/write.table.html'>write.csv</a></span><span class='op'>(</span>
    <span class='va'>cam</span>,
    <span class='va'>gzout</span>,
    row.names <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
  <span class='fu'><a href='https://rdrr.io/r/base/connections.html'>close</a></span><span class='op'>(</span><span class='va'>gzout</span><span class='op'>)</span>
  <span class='va'>cam</span>
<span class='op'>}</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>list_of_camera</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>yf_contrast</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>CSV files of the results are available from <a href="../output/DEGs/reads/"><code>output/DEGs/reads</code></a>.</p>
<p>An example of the results are shown below.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span><span class='op'>(</span><span class='va'>list_of_camera</span><span class='op'>[[</span><span class='fl'>1</span><span class='op'>]</span><span class='op'>]</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span>
    caption <span class='op'>=</span> <span class='st'>'`camera()` produces a table with a row for each gene set (prepended by which MSigDB collection it comes from) and the following columns: **NGenes** (number of genes in set); **Direction** (direction of change); **PValue** (two-tailed p-value); **FDR** (Benjamini and Hochberg FDR adjusted p-value).'</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<table>
<caption><span id="tab:unnamed-chunk-64">Table 14: </span><code>camera()</code> produces a table with a row for each gene set (prepended by which MSigDB collection it comes from) and the following columns: <strong>NGenes</strong> (number of genes in set); <strong>Direction</strong> (direction of change); <strong>PValue</strong> (two-tailed p-value); <strong>FDR</strong> (Benjamini and Hochberg FDR adjusted p-value).</caption>
<colgroup>
<col style="width: 79%" />
<col style="width: 4%" />
<col style="width: 6%" />
<col style="width: 4%" />
<col style="width: 5%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">NGenes</th>
<th style="text-align: left;">Direction</th>
<th style="text-align: right;">PValue</th>
<th style="text-align: right;">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">C2.KEGG_RIBOSOME</td>
<td style="text-align: right;">102</td>
<td style="text-align: left;">Up</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1.0e-07</td>
</tr>
<tr class="even">
<td style="text-align: left;">C2.REACTOME_EUKARYOTIC_TRANSLATION_ELONGATION</td>
<td style="text-align: right;">107</td>
<td style="text-align: left;">Up</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1.0e-07</td>
</tr>
<tr class="odd">
<td style="text-align: left;">C2.REACTOME_RESPONSE_OF_EIF2AK4_GCN2_TO_AMINO_ACID_DEFICIENCY</td>
<td style="text-align: right;">115</td>
<td style="text-align: left;">Up</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">9.0e-07</td>
</tr>
<tr class="even">
<td style="text-align: left;">C2.REACTOME_RESPIRATORY_ELECTRON_TRANSPORT_ATP_SYNTHESIS_BY_CHEMIOSMOTIC_COUPLING_AND_HEAT_PRODUCTION_BY_UNCOUPLING_PROTEINS</td>
<td style="text-align: right;">121</td>
<td style="text-align: left;">Up</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1.3e-06</td>
</tr>
<tr class="odd">
<td style="text-align: left;">C2.REACTOME_EUKARYOTIC_TRANSLATION_INITIATION</td>
<td style="text-align: right;">139</td>
<td style="text-align: left;">Up</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1.3e-06</td>
</tr>
<tr class="even">
<td style="text-align: left;">C2.REACTOME_SRP_DEPENDENT_COTRANSLATIONAL_PROTEIN_TARGETING_TO_MEMBRANE</td>
<td style="text-align: right;">130</td>
<td style="text-align: left;">Up</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1.9e-06</td>
</tr>
</tbody>
</table>
</div>
<h3 id="sarahs-gene-lists-2">Sarah’s gene lists</h3>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>yf_X</span> <span class='op'>&lt;-</span> <span class='va'>yf</span><span class='op'>$</span><span class='va'>genes</span><span class='op'>$</span><span class='va'>ENSEMBL.SEQNAME</span> <span class='op'>==</span> <span class='st'>"X"</span>
<span class='va'>yf_protocadherin</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/grep.html'>grepl</a></span><span class='op'>(</span><span class='st'>"Pcdh"</span>, <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>yf</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>yf_pw</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>yf</span><span class='op'>)</span> <span class='op'>%in%</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Ndn"</span>, <span class='st'>"Mkrn3"</span>, <span class='st'>"Peg12"</span><span class='op'>)</span>
<span class='va'>yf_female</span> <span class='op'>&lt;-</span> <span class='va'>female_het_vs_del</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Probe"</span>, <span class='st'>"Differential expression"</span><span class='op'>)</span><span class='op'>]</span>
<span class='va'>yf_male</span> <span class='op'>&lt;-</span> <span class='va'>male_het_vs_del</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Probe"</span>, <span class='st'>"Differential expression"</span><span class='op'>)</span><span class='op'>]</span>
<span class='va'>yf_sun</span> <span class='op'>&lt;-</span> <span class='va'>sun_2014</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"geneSymbol"</span>, <span class='st'>"log2FC"</span><span class='op'>)</span><span class='op'>]</span>
<span class='va'>yf_chambers</span> <span class='op'>&lt;-</span> <span class='va'>chambers_2007</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Gene Symbol"</span>, <span class='st'>"Fold Difference"</span><span class='op'>)</span><span class='op'>]</span>
<span class='va'>yf_chambers</span><span class='op'>$</span><span class='va'>logFC</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log2</a></span><span class='op'>(</span><span class='va'>yf_chambers</span><span class='op'>$</span><span class='va'>`Fold Difference`</span><span class='op'>)</span>
<span class='va'>yf_chambers</span><span class='op'>$</span><span class='va'>`Fold Difference`</span> <span class='op'>&lt;-</span> <span class='cn'>NULL</span>
<span class='va'>yf_chambers</span> <span class='op'>&lt;-</span> <span class='va'>yf_chambers</span><span class='op'>[</span><span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/NA.html'>is.na</a></span><span class='op'>(</span><span class='va'>yf_chambers</span><span class='op'>$</span><span class='va'>`Gene Symbol`</span><span class='op'>)</span>, <span class='op'>]</span>
<span class='va'>yf_chambers</span> <span class='op'>&lt;-</span> <span class='va'>yf_chambers</span><span class='op'>[</span><span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/duplicated.html'>duplicated</a></span><span class='op'>(</span><span class='va'>yf_chambers</span><span class='op'>$</span><span class='va'>`Gene Symbol`</span><span class='op'>)</span>, <span class='op'>]</span>
<span class='va'>yf_gazit</span> <span class='op'>&lt;-</span> <span class='va'>gazit_2013</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Symbol"</span><span class='op'>)</span>, drop <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>]</span>
<span class='va'>yf_gazit</span><span class='op'>$</span><span class='va'>Symbol</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/base/substr.html'>substring</a></span><span class='op'>(</span><span class='va'>yf_gazit</span><span class='op'>$</span><span class='va'>Symbol</span>, <span class='fl'>1</span>, <span class='fl'>1</span><span class='op'>)</span>, 
  <span class='fu'><a href='https://rdrr.io/r/base/chartr.html'>tolower</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/substr.html'>substring</a></span><span class='op'>(</span><span class='va'>yf_gazit</span><span class='op'>$</span><span class='va'>Symbol</span>, <span class='fl'>2</span><span class='op'>)</span><span class='op'>)</span>,
  sep <span class='op'>=</span> <span class='st'>""</span><span class='op'>)</span>
<span class='va'>yf_gazit</span><span class='op'>$</span><span class='va'>logFC</span> <span class='op'>&lt;-</span> <span class='fl'>1</span>
<span class='va'>yf_chambers_gazit</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>
  Symbol <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='va'>yf_chambers</span><span class='op'>$</span><span class='va'>`Gene Symbol`</span>, <span class='va'>yf_gazit</span><span class='op'>$</span><span class='va'>Symbol</span><span class='op'>)</span>,
  logFC <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='va'>yf_chambers</span><span class='op'>$</span><span class='va'>logFC</span>, <span class='va'>yf_gazit</span><span class='op'>$</span><span class='va'>logFC</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>yf_chambers_gazit</span> <span class='op'>&lt;-</span> <span class='va'>yf_chambers_gazit</span><span class='op'>[</span><span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/duplicated.html'>duplicated</a></span><span class='op'>(</span><span class='va'>yf_chambers_gazit</span><span class='op'>$</span><span class='va'>Symbol</span><span class='op'>)</span>, <span class='op'>]</span>
<span class='va'>yf_venezia_proliferation</span> <span class='op'>&lt;-</span> <span class='va'>venezia_proliferation</span><span class='op'>[</span>
  , <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Gene.Symbol"</span>, <span class='st'>"Log2.Fold.Change..FL.HSC.vs.Adult.HSC.."</span><span class='op'>)</span><span class='op'>]</span>
<span class='va'>yf_venezia_proliferation</span> <span class='op'>&lt;-</span> <span class='va'>yf_venezia_proliferation</span><span class='op'>[</span>
  <span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/duplicated.html'>duplicated</a></span><span class='op'>(</span><span class='va'>yf_venezia_proliferation</span><span class='op'>$</span><span class='va'>`Gene.Symbol`</span><span class='op'>)</span>, <span class='op'>]</span>
<span class='va'>yf_venezia_quiescence</span> <span class='op'>&lt;-</span> <span class='va'>venezia_quiescence</span><span class='op'>[</span>
  , <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Gene.Symbol"</span>, <span class='st'>"Log2.Fold.Change..FL.HSC.vs.Adult.HSC.."</span><span class='op'>)</span><span class='op'>]</span>
<span class='va'>yf_venezia_quiescence</span> <span class='op'>&lt;-</span> <span class='va'>yf_venezia_quiescence</span><span class='op'>[</span>
  <span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/duplicated.html'>duplicated</a></span><span class='op'>(</span><span class='va'>yf_venezia_quiescence</span><span class='op'>$</span><span class='va'>`Gene.Symbol`</span><span class='op'>)</span>, <span class='op'>]</span>
<span class='co'># TODO: Clarify how this gene set is to be used.</span>
<span class='co'># yf_smchd1_femBcells &lt;- smchd1_femBcells[</span>
<span class='co'>#   smchd1_femBcells$`FDR (EdgeR stats p&lt;1.0 after correction)` &lt; 0.05, </span>
<span class='co'>#   c("Feature", "Log2 Fold Change (EdgeR stats p&lt;1.0 after correction)")]</span>
<span class='co'># yf_smchd1_femBcells &lt;- yf_smchd1_femBcells[</span>
<span class='co'>#   !duplicated(yf_smchd1_femBcells$Feature), ]</span>
<span class='va'>yf_smchd1_femBcells</span> <span class='op'>&lt;-</span> <span class='va'>smchd1_femBcells</span><span class='op'>[</span>
    <span class='co'># smchd1_femBcells$Chromosome == "X", </span>
  <span class='va'>smchd1_femBcells</span><span class='op'>$</span><span class='va'>Chromosome</span> <span class='op'>==</span> <span class='st'>"X"</span> <span class='op'>&amp;</span> 
    <span class='va'>smchd1_femBcells</span><span class='op'>$</span><span class='va'>`FDR (EdgeR stats p&lt;1.0 after correction)`</span> <span class='op'>&lt;</span> <span class='fl'>0.05</span>,
  <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Feature"</span>, <span class='st'>"Log2 Fold Change (EdgeR stats p&lt;1.0 after correction)"</span><span class='op'>)</span><span class='op'>]</span>
<span class='va'>yf_smchd1_femBcells</span> <span class='op'>&lt;-</span> <span class='va'>yf_smchd1_femBcells</span><span class='op'>[</span>
  <span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/duplicated.html'>duplicated</a></span><span class='op'>(</span><span class='va'>yf_smchd1_femBcells</span><span class='op'>$</span><span class='va'>Feature</span><span class='op'>)</span>, <span class='op'>]</span>

<span class='va'>yf_index</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>
  `X-linked` <span class='op'>=</span> <span class='va'>yf_X</span>,
  Protocadherin <span class='op'>=</span> <span class='va'>yf_protocadherin</span>,
  `Prader-Willi` <span class='op'>=</span> <span class='va'>yf_pw</span>,
  `Male: Het vs. Del` <span class='op'>=</span> <span class='va'>yf_male</span>,
  `Female: Het vs. Del` <span class='op'>=</span> <span class='va'>yf_female</span>,
  `Sun (2014)` <span class='op'>=</span> <span class='va'>yf_sun</span>,
  `Chambers (2007)` <span class='op'>=</span> <span class='va'>yf_chambers</span>,
  `Gazit (2013)` <span class='op'>=</span> <span class='va'>yf_gazit</span>,
  `Chambers (2007) &amp; Gazit (2013)` <span class='op'>=</span> <span class='va'>yf_chambers_gazit</span>,
  `Proliferation Group Venezia (2004)` <span class='op'>=</span> <span class='va'>yf_venezia_proliferation</span>,
  `Quiescence Group Venezia (2004)` <span class='op'>=</span> <span class='va'>yf_venezia_quiescence</span>,
  `Smchd1 female B cell (chrX)` <span class='op'>=</span> <span class='va'>yf_smchd1_femBcells</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>Sarah requested testing a number of gene sets (see <a href="#gene-sets">Gene sets</a>).</p>
<p>We use the <code>fry()</code> function from the <em><a href="https://bioconductor.org/packages/3.12/edgeR">edgeR</a></em> R/Bioconductor package to perform a self-contained gene set test against the null hypothesis that none of the genes in the set are differentially expressed.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'>fry</span><span class='op'>(</span>
  <span class='va'>yf</span>, 
  index <span class='op'>=</span> <span class='va'>yf_index</span>, 
  design <span class='op'>=</span> <span class='va'>yf_design</span>,
  contrast <span class='op'>=</span> <span class='va'>yf_contrast</span>,
  sort <span class='op'>=</span> <span class='st'>"none"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
    <span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span>
      caption <span class='op'>=</span> <span class='st'>"Results of applying `fry()` to the RNA-seq differential expression results using the supplied gene sets. A significant 'Up' P-value means that the differential expression results found in the RNA-seq data are positively correlated with the expression signature from the corresponding gene set. Conversely, a significant 'Down' P-value means that the differential expression log-fold-changes are negatively correlated with the expression signature from the corresponding gene set. A significant 'Mixed' P-value means that the genes in the set tend to be differentially expressed without regard for direction."</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<table style="width:100%;">
<caption><span id="tab:unnamed-chunk-66">Table 15: </span>Results of applying <code>fry()</code> to the RNA-seq differential expression results using the supplied gene sets. A significant ‘Up’ P-value means that the differential expression results found in the RNA-seq data are positively correlated with the expression signature from the corresponding gene set. Conversely, a significant ‘Down’ P-value means that the differential expression log-fold-changes are negatively correlated with the expression signature from the corresponding gene set. A significant ‘Mixed’ P-value means that the genes in the set tend to be differentially expressed without regard for direction.</caption>
<colgroup>
<col style="width: 36%" />
<col style="width: 7%" />
<col style="width: 10%" />
<col style="width: 10%" />
<col style="width: 10%" />
<col style="width: 13%" />
<col style="width: 10%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">NGenes</th>
<th style="text-align: left;">Direction</th>
<th style="text-align: right;">PValue</th>
<th style="text-align: right;">FDR</th>
<th style="text-align: right;">PValue.Mixed</th>
<th style="text-align: right;">FDR.Mixed</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">X-linked</td>
<td style="text-align: right;">385</td>
<td style="text-align: left;">Down</td>
<td style="text-align: right;">0.7932572</td>
<td style="text-align: right;">0.7932572</td>
<td style="text-align: right;">0.0009255</td>
<td style="text-align: right;">0.0055529</td>
</tr>
<tr class="even">
<td style="text-align: left;">Protocadherin</td>
<td style="text-align: right;">6</td>
<td style="text-align: left;">Up</td>
<td style="text-align: right;">0.3450358</td>
<td style="text-align: right;">0.5175538</td>
<td style="text-align: right;">0.5544190</td>
<td style="text-align: right;">0.6653028</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Prader-Willi</td>
<td style="text-align: right;">3</td>
<td style="text-align: left;">Up</td>
<td style="text-align: right;">0.5275007</td>
<td style="text-align: right;">0.6330009</td>
<td style="text-align: right;">0.8358759</td>
<td style="text-align: right;">0.8358759</td>
</tr>
<tr class="even">
<td style="text-align: left;">Male: Het vs. Del</td>
<td style="text-align: right;">15</td>
<td style="text-align: left;">Up</td>
<td style="text-align: right;">0.6995272</td>
<td style="text-align: right;">0.7631205</td>
<td style="text-align: right;">0.6569840</td>
<td style="text-align: right;">0.7167098</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Female: Het vs. Del</td>
<td style="text-align: right;">42</td>
<td style="text-align: left;">Down</td>
<td style="text-align: right;">0.0677059</td>
<td style="text-align: right;">0.1624941</td>
<td style="text-align: right;">0.2192534</td>
<td style="text-align: right;">0.3758630</td>
</tr>
<tr class="even">
<td style="text-align: left;">Sun (2014)</td>
<td style="text-align: right;">1986</td>
<td style="text-align: left;">Down</td>
<td style="text-align: right;">0.1441139</td>
<td style="text-align: right;">0.2593703</td>
<td style="text-align: right;">0.0025603</td>
<td style="text-align: right;">0.0102413</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Chambers (2007)</td>
<td style="text-align: right;">204</td>
<td style="text-align: left;">Down</td>
<td style="text-align: right;">0.0110643</td>
<td style="text-align: right;">0.1048824</td>
<td style="text-align: right;">0.0219285</td>
<td style="text-align: right;">0.0438569</td>
</tr>
<tr class="even">
<td style="text-align: left;">Gazit (2013)</td>
<td style="text-align: right;">248</td>
<td style="text-align: left;">Down</td>
<td style="text-align: right;">0.0573814</td>
<td style="text-align: right;">0.1624941</td>
<td style="text-align: right;">0.0189400</td>
<td style="text-align: right;">0.0438569</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Chambers (2007) &amp; Gazit (2013)</td>
<td style="text-align: right;">399</td>
<td style="text-align: left;">Down</td>
<td style="text-align: right;">0.0174804</td>
<td style="text-align: right;">0.1048824</td>
<td style="text-align: right;">0.0210045</td>
<td style="text-align: right;">0.0438569</td>
</tr>
<tr class="even">
<td style="text-align: left;">Proliferation Group Venezia (2004)</td>
<td style="text-align: right;">365</td>
<td style="text-align: left;">Up</td>
<td style="text-align: right;">0.0282581</td>
<td style="text-align: right;">0.1130323</td>
<td style="text-align: right;">0.0003136</td>
<td style="text-align: right;">0.0037631</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Quiescence Group Venezia (2004)</td>
<td style="text-align: right;">381</td>
<td style="text-align: left;">Up</td>
<td style="text-align: right;">0.1512994</td>
<td style="text-align: right;">0.2593703</td>
<td style="text-align: right;">0.3284232</td>
<td style="text-align: right;">0.4926348</td>
</tr>
<tr class="even">
<td style="text-align: left;">Smchd1 female B cell (chrX)</td>
<td style="text-align: right;">6</td>
<td style="text-align: left;">Up</td>
<td style="text-align: right;">0.4888882</td>
<td style="text-align: right;">0.6330009</td>
<td style="text-align: right;">0.3862832</td>
<td style="text-align: right;">0.5150442</td>
</tr>
</tbody>
</table>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>geneSetTestFiguresForPaper</span> <span class='op'>&lt;-</span> <span class='kw'>function</span><span class='op'>(</span><span class='op'>)</span> <span class='op'>{</span>
  <span class='kw'>for</span> <span class='op'>(</span><span class='va'>n</span> <span class='kw'>in</span> <span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>yf_index</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>{</span>
    <span class='kw'>if</span> <span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>is.data.frame</a></span><span class='op'>(</span><span class='va'>yf_index</span><span class='op'>[[</span><span class='va'>n</span><span class='op'>]</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>{</span>
      <span class='va'>yf_status</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/rep.html'>rep</a></span><span class='op'>(</span><span class='fl'>0</span>, <span class='fu'><a href='https://rdrr.io/r/base/nrow.html'>nrow</a></span><span class='op'>(</span><span class='va'>yf_dgelrt</span><span class='op'>)</span><span class='op'>)</span>
      <span class='va'>yf_status</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>yf</span><span class='op'>)</span> <span class='op'>%in%</span> <span class='va'>yf_index</span><span class='op'>[[</span><span class='va'>n</span><span class='op'>]</span><span class='op'>]</span><span class='op'>[</span>, <span class='fl'>1</span><span class='op'>]</span><span class='op'>]</span> <span class='op'>&lt;-</span>
        <span class='fu'><a href='https://rdrr.io/r/base/sign.html'>sign</a></span><span class='op'>(</span><span class='va'>yf_index</span><span class='op'>[[</span><span class='va'>n</span><span class='op'>]</span><span class='op'>]</span><span class='op'>[</span><span class='va'>yf_index</span><span class='op'>[[</span><span class='va'>n</span><span class='op'>]</span><span class='op'>]</span><span class='op'>[</span>, <span class='fl'>1</span><span class='op'>]</span> <span class='op'>%in%</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>yf</span><span class='op'>)</span>, <span class='fl'>2</span><span class='op'>]</span><span class='op'>)</span>
      <span class='fu'>plotMD</span><span class='op'>(</span>
        <span class='va'>yf_dgelrt</span>,
        status <span class='op'>=</span> <span class='va'>yf_status</span>,
        hl.col <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"red"</span>, <span class='st'>"blue"</span><span class='op'>)</span>,
        legend <span class='op'>=</span> <span class='st'>"bottomright"</span>,
        main <span class='op'>=</span> <span class='va'>n</span><span class='op'>)</span>
      <span class='fu'><a href='https://rdrr.io/r/graphics/abline.html'>abline</a></span><span class='op'>(</span>h <span class='op'>=</span> <span class='fl'>0</span>, col <span class='op'>=</span> <span class='st'>"darkgrey"</span><span class='op'>)</span>
    <span class='op'>}</span> <span class='kw'>else</span> <span class='op'>{</span>
      <span class='va'>yf_status</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/rep.html'>rep</a></span><span class='op'>(</span><span class='st'>"Other"</span>, <span class='fu'><a href='https://rdrr.io/r/base/nrow.html'>nrow</a></span><span class='op'>(</span><span class='va'>yf_dgelrt</span><span class='op'>)</span><span class='op'>)</span>
      <span class='va'>yf_status</span><span class='op'>[</span><span class='va'>yf_index</span><span class='op'>[[</span><span class='va'>n</span><span class='op'>]</span><span class='op'>]</span><span class='op'>]</span> <span class='op'>&lt;-</span> <span class='va'>n</span>
      <span class='fu'>plotMD</span><span class='op'>(</span>
        <span class='va'>yf_dgelrt</span>,
        status <span class='op'>=</span> <span class='va'>yf_status</span>,
        values <span class='op'>=</span> <span class='va'>n</span>,
        hl.col <span class='op'>=</span> <span class='st'>"red"</span>,
        legend <span class='op'>=</span> <span class='st'>"bottomright"</span>,
        main <span class='op'>=</span> <span class='va'>n</span><span class='op'>)</span>
      <span class='fu'><a href='https://rdrr.io/r/graphics/abline.html'>abline</a></span><span class='op'>(</span>h <span class='op'>=</span> <span class='fl'>0</span>, col <span class='op'>=</span> <span class='st'>"darkgrey"</span><span class='op'>)</span>
    <span class='op'>}</span>
    
    <span class='kw'>if</span> <span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>is.data.frame</a></span><span class='op'>(</span><span class='va'>yf_index</span><span class='op'>[[</span><span class='va'>n</span><span class='op'>]</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>{</span>
      <span class='fu'>barcodeplot</span><span class='op'>(</span>
        <span class='va'>yf_dgelrt</span><span class='op'>$</span><span class='va'>table</span><span class='op'>$</span><span class='va'>logFC</span>,
        <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>yf</span><span class='op'>)</span> <span class='op'>%in%</span> <span class='va'>yf_index</span><span class='op'>[[</span><span class='va'>n</span><span class='op'>]</span><span class='op'>]</span><span class='op'>[</span>, <span class='fl'>1</span><span class='op'>]</span>,
        gene.weights <span class='op'>=</span> <span class='va'>yf_index</span><span class='op'>[[</span><span class='va'>n</span><span class='op'>]</span><span class='op'>]</span><span class='op'>[</span>
          <span class='va'>yf_index</span><span class='op'>[[</span><span class='va'>n</span><span class='op'>]</span><span class='op'>]</span><span class='op'>[</span>, <span class='fl'>1</span><span class='op'>]</span> <span class='op'>%in%</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>yf</span><span class='op'>)</span>, <span class='fl'>2</span><span class='op'>]</span>,
        main <span class='op'>=</span> <span class='va'>n</span><span class='op'>)</span>
    <span class='op'>}</span> <span class='kw'>else</span> <span class='op'>{</span>
      <span class='fu'>barcodeplot</span><span class='op'>(</span>
        <span class='va'>yf_dgelrt</span><span class='op'>$</span><span class='va'>table</span><span class='op'>$</span><span class='va'>logFC</span>,
        <span class='va'>yf_index</span><span class='op'>[[</span><span class='va'>n</span><span class='op'>]</span><span class='op'>]</span>,
        main <span class='op'>=</span> <span class='va'>n</span><span class='op'>)</span>
    <span class='op'>}</span>
  <span class='op'>}</span>
<span class='op'>}</span>
</code></pre>
</div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/r/graphics/par.html'>par</a></span><span class='op'>(</span>mfrow <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span><span class='op'>(</span><span class='va'>yf_index</span><span class='op'>)</span>, <span class='fl'>2</span><span class='op'>)</span><span class='op'>)</span>
<span class='fu'>geneSetTestFiguresForPaper</span><span class='op'>(</span><span class='op'>)</span>  
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:unnamed-chunk-68"></span>
<img src="C086_Kinkel.multi-sample_comparisons_files/figure-html5/unnamed-chunk-68-1.png" alt="MD-plot and barcode plot of DEGs in supplied gene sets. Directional gene sets have (1) MD plot points coloured according to the statistical significance of the gene **in the gene set** (2) barcode plot 'weights' given by the logFC of the gene **in the gene set**. For the barcode plot, genes are represented by bars and are ranked from left to right by increasing log-fold change. This forms the barcode-like pattern. The line (or *worm*) above the barcode shows the relative local enrichment of the vertical bars in each part of the plot. The dotted horizontal line indicates neutral enrichment; the worm above the dotted line shows enrichment while the worm below the dotted line shows depletion." width="624" />
<p class="caption">
Figure 16: MD-plot and barcode plot of DEGs in supplied gene sets. Directional gene sets have (1) MD plot points coloured according to the statistical significance of the gene <strong>in the gene set</strong> (2) barcode plot ‘weights’ given by the logFC of the gene <strong>in the gene set</strong>. For the barcode plot, genes are represented by bars and are ranked from left to right by increasing log-fold change. This forms the barcode-like pattern. The line (or <em>worm</em>) above the barcode shows the relative local enrichment of the vertical bars in each part of the plot. The dotted horizontal line indicates neutral enrichment; the worm above the dotted line shows enrichment while the worm below the dotted line shows depletion.
</p>
</div>
</div>
<aside>
A PDF version of these figures are available in <a href="../output/figures/paper/gene_set_tests.pdf"><code>output/figures/paper/gene_set_tests.pdf</code></a>.
</aside>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/r/base/files2.html'>dir.create</a></span><span class='op'>(</span><span class='fu'><a href='https://here.r-lib.org//reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"output/figures/paper"</span><span class='op'>)</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/grDevices/pdf.html'>pdf</a></span><span class='op'>(</span><span class='fu'><a href='https://here.r-lib.org//reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"output/figures/paper/gene_set_tests.pdf"</span><span class='op'>)</span>, height <span class='op'>=</span> <span class='fl'>6</span>, width <span class='op'>=</span> <span class='fl'>6</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/graphics/par.html'>par</a></span><span class='op'>(</span>mfrow <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span>
<span class='fu'>geneSetTestFiguresForPaper</span><span class='op'>(</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/grDevices/dev.html'>dev.off</a></span><span class='op'>(</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<h1 class="appendix" id="session-info">Session info</h1>
<summary>
The analysis and this document were prepared using the following software (click triangle to expand)
</summary>
<details>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'>sessioninfo</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/sessioninfo/man/session_info.html'>session_info</a></span><span class='op'>(</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<pre><code>─ Session info ─────────────────────────────────────────────────────
 setting  value                       
 version  R version 4.0.3 (2020-10-10)
 os       CentOS Linux 7 (Core)       
 system   x86_64, linux-gnu           
 ui       X11                         
 language (EN)                        
 collate  en_US.UTF-8                 
 ctype    en_US.UTF-8                 
 tz       Australia/Melbourne         
 date     2021-10-06                  

─ Packages ─────────────────────────────────────────────────────────
 ! package              * version  date       lib source        
 P annotate               1.68.0   2020-10-27 [?] Bioconductor  
 P AnnotationDbi          1.52.0   2020-10-27 [?] Bioconductor  
 P aroma.light            3.20.0   2020-10-27 [?] Bioconductor  
 P askpass                1.1      2019-01-13 [?] CRAN (R 4.0.0)
 P assertthat             0.2.1    2019-03-21 [?] CRAN (R 4.0.0)
 P Biobase              * 2.50.0   2020-10-27 [?] Bioconductor  
 P BiocFileCache        * 1.14.0   2020-10-27 [?] Bioconductor  
 P BiocGenerics         * 0.36.0   2020-10-27 [?] Bioconductor  
 P BiocManager            1.30.10  2019-11-16 [?] CRAN (R 4.0.0)
 P BiocParallel         * 1.24.1   2020-11-06 [?] Bioconductor  
 P BiocStyle              2.18.1   2020-11-24 [?] Bioconductor  
 P biomaRt                2.46.3   2021-02-09 [?] Bioconductor  
 P Biostrings           * 2.58.0   2020-10-27 [?] Bioconductor  
 P bit                    4.0.4    2020-08-04 [?] CRAN (R 4.0.0)
 P bit64                  4.0.5    2020-08-30 [?] CRAN (R 4.0.0)
 P bitops                 1.0-6    2013-08-17 [?] CRAN (R 4.0.0)
 P blob                   1.2.1    2020-01-20 [?] CRAN (R 4.0.0)
 P cachem                 1.0.4    2021-02-13 [?] CRAN (R 4.0.3)
 P cellranger             1.1.0    2016-07-27 [?] CRAN (R 4.0.0)
 P cli                    2.3.1    2021-02-23 [?] CRAN (R 4.0.3)
 P colorspace             2.0-0    2020-11-11 [?] CRAN (R 4.0.3)
 P cowplot              * 1.1.1    2020-12-30 [?] CRAN (R 4.0.3)
 P crayon                 1.4.1    2021-02-08 [?] CRAN (R 4.0.3)
 P crosstalk              1.1.1    2021-01-12 [?] CRAN (R 4.0.3)
 P curl                   4.3      2019-12-02 [?] CRAN (R 4.0.0)
 P data.table             1.14.0   2021-02-21 [?] CRAN (R 4.0.3)
 P DBI                    1.1.1    2021-01-15 [?] CRAN (R 4.0.3)
 P dbplyr               * 2.1.0    2021-02-03 [?] CRAN (R 4.0.3)
 P DelayedArray           0.16.1   2021-01-22 [?] Bioconductor  
 P DESeq2                 1.30.1   2021-02-19 [?] Bioconductor  
 P digest                 0.6.27   2020-10-24 [?] CRAN (R 4.0.2)
 P distill                1.2      2021-01-13 [?] CRAN (R 4.0.3)
 P downlit                0.2.1    2020-11-04 [?] CRAN (R 4.0.3)
 P dplyr                  1.0.4    2021-02-02 [?] CRAN (R 4.0.3)
 P DT                     0.17     2021-01-06 [?] CRAN (R 4.0.3)
 P EDASeq               * 2.24.0   2020-10-27 [?] Bioconductor  
 P edgeR                * 3.32.1   2021-01-14 [?] Bioconductor  
 P ellipsis               0.3.1    2020-05-15 [?] CRAN (R 4.0.0)
 P evaluate               0.14     2019-05-28 [?] CRAN (R 4.0.0)
 P fansi                  0.4.2    2021-01-15 [?] CRAN (R 4.0.3)
 P farver                 2.0.3    2020-01-16 [?] CRAN (R 4.0.0)
 P fastmap                1.1.0    2021-01-25 [?] CRAN (R 4.0.3)
 P genefilter             1.72.1   2021-01-21 [?] Bioconductor  
 P geneplotter            1.68.0   2020-10-27 [?] Bioconductor  
 P generics               0.1.0    2020-10-31 [?] CRAN (R 4.0.3)
 P GenomeInfoDb         * 1.26.2   2020-12-08 [?] Bioconductor  
 P GenomeInfoDbData       1.2.4    2020-10-20 [?] Bioconductor  
 P GenomicAlignments    * 1.26.0   2020-10-27 [?] Bioconductor  
 P GenomicFeatures        1.42.1   2020-11-12 [?] Bioconductor  
 P GenomicRanges        * 1.42.0   2020-10-27 [?] Bioconductor  
 P ggplot2              * 3.3.3    2020-12-30 [?] CRAN (R 4.0.3)
 P ggrepel              * 0.9.1    2021-01-15 [?] CRAN (R 4.0.3)
 P Glimma               * 2.0.0    2020-10-27 [?] Bioconductor  
 P glue                   1.4.2    2020-08-27 [?] CRAN (R 4.0.0)
 P GO.db                  3.12.1   2020-11-04 [?] Bioconductor  
 P gtable                 0.3.0    2019-03-25 [?] CRAN (R 4.0.0)
 P here                 * 1.0.1    2020-12-13 [?] CRAN (R 4.0.3)
 P highr                  0.8      2019-03-20 [?] CRAN (R 4.0.0)
 P hms                    1.0.0    2021-01-13 [?] CRAN (R 4.0.3)
 P htmltools              0.5.1.1  2021-01-22 [?] CRAN (R 4.0.3)
 P htmlwidgets            1.5.3    2020-12-10 [?] CRAN (R 4.0.3)
 P httr                   1.4.2    2020-07-20 [?] CRAN (R 4.0.0)
 P hwriter                1.3.2    2014-09-10 [?] CRAN (R 4.0.0)
 P IRanges              * 2.24.1   2020-12-12 [?] Bioconductor  
 P jpeg                   0.1-8.1  2019-10-24 [?] CRAN (R 4.0.0)
 P jsonlite               1.7.2    2020-12-09 [?] CRAN (R 4.0.3)
 P knitr                  1.31     2021-01-27 [?] CRAN (R 4.0.3)
 P labeling               0.4.2    2020-10-20 [?] CRAN (R 4.0.0)
 P lattice                0.20-41  2020-04-02 [3] CRAN (R 4.0.3)
 P latticeExtra           0.6-29   2019-12-19 [?] CRAN (R 4.0.0)
 P lifecycle              1.0.0    2021-02-15 [?] CRAN (R 4.0.3)
 P limma                * 3.46.0   2020-10-27 [?] Bioconductor  
 P locfit                 1.5-9.4  2020-03-25 [?] CRAN (R 4.0.0)
 P magrittr             * 2.0.1    2020-11-17 [?] CRAN (R 4.0.3)
 P MASS                   7.3-53   2020-09-09 [3] CRAN (R 4.0.3)
 P Matrix                 1.2-18   2019-11-27 [3] CRAN (R 4.0.3)
 P MatrixGenerics       * 1.2.1    2021-01-30 [?] Bioconductor  
 P matrixStats          * 0.58.0   2021-01-29 [?] CRAN (R 4.0.3)
 P memoise                2.0.0    2021-01-26 [?] CRAN (R 4.0.3)
 P munsell                0.5.0    2018-06-12 [?] CRAN (R 4.0.0)
 P openssl                1.4.3    2020-09-18 [?] CRAN (R 4.0.2)
 P org.Mm.eg.db           3.12.0   2020-10-20 [?] Bioconductor  
 P patchwork            * 1.1.1    2020-12-17 [?] CRAN (R 4.0.3)
 P pheatmap             * 1.0.12   2019-01-04 [?] CRAN (R 4.0.0)
 P pillar                 1.5.0    2021-02-22 [?] CRAN (R 4.0.3)
 P pkgconfig              2.0.3    2019-09-22 [?] CRAN (R 4.0.0)
 P png                    0.1-7    2013-12-03 [?] CRAN (R 4.0.0)
 P prettyunits            1.1.1    2020-01-24 [?] CRAN (R 4.0.0)
 P progress               1.2.2    2019-05-16 [?] CRAN (R 4.0.0)
 P purrr                  0.3.4    2020-04-17 [?] CRAN (R 4.0.0)
 P R.methodsS3            1.8.1    2020-08-26 [?] CRAN (R 4.0.0)
 P R.oo                   1.24.0   2020-08-26 [?] CRAN (R 4.0.0)
 P R.utils                2.10.1   2020-08-26 [?] CRAN (R 4.0.0)
 P R6                     2.5.0    2020-10-28 [?] CRAN (R 4.0.2)
 P rappdirs               0.3.3    2021-01-31 [?] CRAN (R 4.0.3)
 P RColorBrewer           1.1-2    2014-12-07 [?] CRAN (R 4.0.0)
 P Rcpp                   1.0.6    2021-01-15 [?] CRAN (R 4.0.3)
 P RCurl                  1.98-1.2 2020-04-18 [?] CRAN (R 4.0.0)
 P readxl               * 1.3.1    2019-03-13 [?] CRAN (R 4.0.0)
 P rematch                1.0.1    2016-04-21 [?] CRAN (R 4.0.0)
 P rlang                  0.4.10   2020-12-30 [?] CRAN (R 4.0.3)
 P rmarkdown              2.7      2021-02-19 [?] CRAN (R 4.0.3)
 P rprojroot              2.0.2    2020-11-15 [?] CRAN (R 4.0.3)
 P Rsamtools            * 2.6.0    2020-10-27 [?] Bioconductor  
 P RSQLite                2.2.3    2021-01-24 [?] CRAN (R 4.0.3)
 P rtracklayer            1.50.0   2020-10-27 [?] Bioconductor  
 P RUVSeq               * 1.24.0   2020-10-27 [?] Bioconductor  
 P S4Vectors            * 0.28.1   2020-12-09 [?] Bioconductor  
 P scales                 1.1.1    2020-05-11 [?] CRAN (R 4.0.0)
 P sessioninfo            1.1.1    2018-11-05 [?] CRAN (R 4.0.0)
 P ShortRead            * 1.48.0   2020-10-27 [?] Bioconductor  
 P SingleCellExperiment * 1.12.0   2020-10-27 [?] Bioconductor  
 P stringi                1.5.3    2020-09-09 [?] CRAN (R 4.0.0)
 P stringr                1.4.0    2019-02-10 [?] CRAN (R 4.0.0)
 P SummarizedExperiment * 1.20.0   2020-10-27 [?] Bioconductor  
 P survival               3.2-7    2020-09-28 [3] CRAN (R 4.0.3)
 P tibble                 3.0.6    2021-01-29 [?] CRAN (R 4.0.3)
 P tidyselect             1.1.0    2020-05-11 [?] CRAN (R 4.0.0)
 P utf8                   1.1.4    2018-05-24 [?] CRAN (R 4.0.0)
 P vctrs                  0.3.6    2020-12-17 [?] CRAN (R 4.0.3)
 P viridisLite            0.3.0    2018-02-01 [?] CRAN (R 4.0.0)
 P withr                  2.4.1    2021-01-26 [?] CRAN (R 4.0.3)
 P xfun                   0.21     2021-02-10 [?] CRAN (R 4.0.3)
 P XML                    3.99-0.5 2020-07-23 [?] CRAN (R 4.0.0)
 P xml2                   1.3.2    2020-04-23 [?] CRAN (R 4.0.0)
 P xtable                 1.8-4    2019-04-21 [?] CRAN (R 4.0.0)
 P XVector              * 0.30.0   2020-10-27 [?] Bioconductor  
 P yaml                   2.2.1    2020-02-01 [?] CRAN (R 4.0.0)
 P zlibbioc               1.36.0   2020-10-27 [?] Bioconductor  

[1] /stornext/Projects/score/Analyses/C086_Kinkel/renv/library/R-4.0/x86_64-pc-linux-gnu
[2] /tmp/RtmpBUin6g/renv-system-library
[3] /stornext/System/data/apps/R/R-4.0.3/lib64/R/library

 P ── Loaded and on-disk path mismatch.</code></pre>
</div>
</details>
<div class="sourceCode" id="cb5"><pre class="sourceCode r distill-force-highlighting-css"><code class="sourceCode r"></code></pre></div>
<div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography">
<div id="ref-chambers2007hematopoietic" class="csl-entry" role="doc-biblioentry">
Chambers, Stuart M, Nathan C Boles, Kuan-Yin K Lin, Megan P Tierney, Teresa V Bowman, Steven B Bradfute, Alice J Chen, et al. 2007. <span>“Hematopoietic Fingerprints: An Expression Database of Stem Cells and Their Progeny.”</span> <em>Cell Stem Cell</em> 1 (5): 578–91.
</div>
<div id="ref-gazit2013transcriptome" class="csl-entry" role="doc-biblioentry">
Gazit, Roi, Brian S Garrison, Tata Nageswara Rao, Tal Shay, James Costello, Jeff Ericson, Francis Kim, et al. 2013. <span>“Transcriptome Analysis Identifies Regulators of Hematopoietic Stem and Progenitor Cells.”</span> <em>Stem Cell Reports</em> 1 (3): 266–80.
</div>
<div id="ref-sun2014epigenomic" class="csl-entry" role="doc-biblioentry">
Sun, Deqiang, Min Luo, Mira Jeong, Benjamin Rodriguez, Zheng Xia, Rebecca Hannah, Hui Wang, et al. 2014. <span>“Epigenomic Profiling of Young and Aged HSCs Reveals Concerted Changes During Aging That Reinforce Self-Renewal.”</span> <em>Cell Stem Cell</em> 14 (5): 673–88.
</div>
<div id="ref-venezia2004molecular" class="csl-entry" role="doc-biblioentry">
Venezia, Teresa A, Akil A Merchant, Carlos A Ramos, Nathan L Whitehouse, Andrew S Young, Chad A Shaw, Margaret A Goodell, and Bing Lim. 2004. <span>“Molecular Signatures of Proliferation and Quiescence in Hematopoietic Stem Cells.”</span> <em>PLoS Biology</em> 2 (10): e301.
</div>
<div id="ref-vizan2020polycomb" class="csl-entry" role="doc-biblioentry">
Vizán, Pedro, Arantxa Gutiérrez, Isabel Espejo, Marc Garcı́a-Montolio, Martin Lange, Ana Carretero, Atefeh Lafzi, et al. 2020. <span>“The Polycomb-Associated Factor Phf19 Controls Hematopoietic Stem Cell State and Differentiation.”</span> <em>Science Advances</em> 6 (32): eabb2745.
</div>
</div>
<section class="footnotes" role="doc-endnotes">
<hr />
<ol>
<li id="fn1" role="doc-endnote"><p>Details of RUVs parameters are given separately for each analysis.<a href="#fnref1" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn2" role="doc-endnote"><p>Sarah “used SeqMonk/EdgeR to compare Smchd1-del to Smchd1-het, keeping males and females separate” (email from Sarah).<a href="#fnref2" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn3" role="doc-endnote"><p>Sarah “used SeqMonk/EdgeR to compare Smchd1-del to Smchd1-het, keeping males and females separate” (email from Sarah).<a href="#fnref3" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn4" role="doc-endnote"><p>My understanding of this terms translates as ‘upregulated DEGs.’<a href="#fnref4" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
</ol>
</section>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom">
<h3 id="references">References</h3>
<div id="references-listing"></div>
</div>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<!--/radix_placeholder_navigation_after_body-->

</body>

</html>
