<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ad0000; } /* Alert */
code span.an { color: #5e5e5e; } /* Annotation */
code span.at { color: #20794d; } /* Attribute */
code span.bn { color: #ad0000; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007ba5; } /* ControlFlow */
code span.ch { color: #20794d; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #5e5e5e; } /* Comment */
code span.cv { color: #5e5e5e; font-style: italic; } /* CommentVar */
code span.do { color: #5e5e5e; font-style: italic; } /* Documentation */
code span.dt { color: #ad0000; } /* DataType */
code span.dv { color: #ad0000; } /* DecVal */
code span.er { color: #ad0000; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #ad0000; } /* Float */
code span.fu { color: #4758ab; } /* Function */
code span.im { } /* Import */
code span.in { color: #5e5e5e; } /* Information */
code span.kw { color: #007ba5; } /* Keyword */
code span.op { color: #5e5e5e; } /* Operator */
code span.ot { color: #007ba5; } /* Other */
code span.pp { color: #ad0000; } /* Preprocessor */
code span.sc { color: #20794d; } /* SpecialChar */
code span.ss { color: #20794d; } /* SpecialString */
code span.st { color: #20794d; } /* String */
code span.va { color: #111111; } /* Variable */
code span.vs { color: #20794d; } /* VerbatimString */
code span.wa { color: #5e5e5e; font-style: italic; } /* Warning */
</style>

  <!--radix_placeholder_meta_tags-->
  <title>C086: Multi-sample comparisons of the Kinkel (C086) male B cells RNA-seq data set</title>

  <meta property="description" itemprop="description" content="Read-based analysis"/>


  <!--  https://schema.org/Article -->
  <meta property="article:published" itemprop="datePublished" content="2021-10-15"/>
  <meta property="article:created" itemprop="dateCreated" content="2021-10-15"/>
  <meta name="article:author" content="Peter Hickey"/>

  <!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
  <meta property="og:title" content="C086: Multi-sample comparisons of the Kinkel (C086) male B cells RNA-seq data set"/>
  <meta property="og:type" content="article"/>
  <meta property="og:description" content="Read-based analysis"/>
  <meta property="og:locale" content="en_US"/>
  <meta property="og:site_name" content="C086"/>

  <!--  https://dev.twitter.com/cards/types/summary -->
  <meta property="twitter:card" content="summary"/>
  <meta property="twitter:title" content="C086: Multi-sample comparisons of the Kinkel (C086) male B cells RNA-seq data set"/>
  <meta property="twitter:description" content="Read-based analysis"/>

  <!--/radix_placeholder_meta_tags-->
  
  <!--radix_placeholder_rmarkdown_metadata-->

  <script type="text/json" id="radix-rmarkdown-metadata">
  {"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","description","author","date","editor_options","bibliography"]}},"value":[{"type":"character","attributes":{},"value":["Multi-sample comparisons of the Kinkel (C086) male B cells RNA-seq data set"]},{"type":"character","attributes":{},"value":["Read-based analysis"]},{"type":"list","attributes":{},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","url","affiliation","affiliation_url"]}},"value":[{"type":"character","attributes":{},"value":["Peter Hickey"]},{"type":"character","attributes":{},"value":["https://peterhickey.org"]},{"type":"character","attributes":{},"value":["WEHI SCORE"]},{"type":"character","attributes":{},"value":["https://www.wehi.edu.au/people/shalin-naik/3310/score"]}]}]},{"type":"character","attributes":{},"value":["2021-10-15"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["chunk_output_type"]}},"value":[{"type":"character","attributes":{},"value":["console"]}]},{"type":"character","attributes":{},"value":["ref.bib"]}]}
  </script>
  <!--/radix_placeholder_rmarkdown_metadata-->
  <!--radix_placeholder_navigation_in_header-->
  <meta name="distill:offset" content=""/>

  <script type="application/javascript">

    window.headroom_prevent_pin = false;

    window.document.addEventListener("DOMContentLoaded", function (event) {

      // initialize headroom for banner
      var header = $('header').get(0);
      var headerHeight = header.offsetHeight;
      var headroom = new Headroom(header, {
        tolerance: 5,
        onPin : function() {
          if (window.headroom_prevent_pin) {
            window.headroom_prevent_pin = false;
            headroom.unpin();
          }
        }
      });
      headroom.init();
      if(window.location.hash)
        headroom.unpin();
      $(header).addClass('headroom--transition');

      // offset scroll location for banner on hash change
      // (see: https://github.com/WickyNilliams/headroom.js/issues/38)
      window.addEventListener("hashchange", function(event) {
        window.scrollTo(0, window.pageYOffset - (headerHeight + 25));
      });

      // responsive menu
      $('.distill-site-header').each(function(i, val) {
        var topnav = $(this);
        var toggle = topnav.find('.nav-toggle');
        toggle.on('click', function() {
          topnav.toggleClass('responsive');
        });
      });

      // nav dropdowns
      $('.nav-dropbtn').click(function(e) {
        $(this).next('.nav-dropdown-content').toggleClass('nav-dropdown-active');
        $(this).parent().siblings('.nav-dropdown')
           .children('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $("body").click(function(e){
        $('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $(".nav-dropdown").click(function(e){
        e.stopPropagation();
      });
    });
  </script>

  <style type="text/css">

  /* Theme (user-documented overrideables for nav appearance) */

  .distill-site-nav {
    color: rgba(255, 255, 255, 0.8);
    background-color: #0F2E3D;
    font-size: 15px;
    font-weight: 300;
  }

  .distill-site-nav a {
    color: inherit;
    text-decoration: none;
  }

  .distill-site-nav a:hover {
    color: white;
  }

  @media print {
    .distill-site-nav {
      display: none;
    }
  }

  .distill-site-header {

  }

  .distill-site-footer {

  }


  /* Site Header */

  .distill-site-header {
    width: 100%;
    box-sizing: border-box;
    z-index: 3;
  }

  .distill-site-header .nav-left {
    display: inline-block;
    margin-left: 8px;
  }

  @media screen and (max-width: 768px) {
    .distill-site-header .nav-left {
      margin-left: 0;
    }
  }


  .distill-site-header .nav-right {
    float: right;
    margin-right: 8px;
  }

  .distill-site-header a,
  .distill-site-header .title {
    display: inline-block;
    text-align: center;
    padding: 14px 10px 14px 10px;
  }

  .distill-site-header .title {
    font-size: 18px;
    min-width: 150px;
  }

  .distill-site-header .logo {
    padding: 0;
  }

  .distill-site-header .logo img {
    display: none;
    max-height: 20px;
    width: auto;
    margin-bottom: -4px;
  }

  .distill-site-header .nav-image img {
    max-height: 18px;
    width: auto;
    display: inline-block;
    margin-bottom: -3px;
  }



  @media screen and (min-width: 1000px) {
    .distill-site-header .logo img {
      display: inline-block;
    }
    .distill-site-header .nav-left {
      margin-left: 20px;
    }
    .distill-site-header .nav-right {
      margin-right: 20px;
    }
    .distill-site-header .title {
      padding-left: 12px;
    }
  }


  .distill-site-header .nav-toggle {
    display: none;
  }

  .nav-dropdown {
    display: inline-block;
    position: relative;
  }

  .nav-dropdown .nav-dropbtn {
    border: none;
    outline: none;
    color: rgba(255, 255, 255, 0.8);
    padding: 16px 10px;
    background-color: transparent;
    font-family: inherit;
    font-size: inherit;
    font-weight: inherit;
    margin: 0;
    margin-top: 1px;
    z-index: 2;
  }

  .nav-dropdown-content {
    display: none;
    position: absolute;
    background-color: white;
    min-width: 200px;
    border: 1px solid rgba(0,0,0,0.15);
    border-radius: 4px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.1);
    z-index: 1;
    margin-top: 2px;
    white-space: nowrap;
    padding-top: 4px;
    padding-bottom: 4px;
  }

  .nav-dropdown-content hr {
    margin-top: 4px;
    margin-bottom: 4px;
    border: none;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .nav-dropdown-active {
    display: block;
  }

  .nav-dropdown-content a, .nav-dropdown-content .nav-dropdown-header {
    color: black;
    padding: 6px 24px;
    text-decoration: none;
    display: block;
    text-align: left;
  }

  .nav-dropdown-content .nav-dropdown-header {
    display: block;
    padding: 5px 24px;
    padding-bottom: 0;
    text-transform: uppercase;
    font-size: 14px;
    color: #999999;
    white-space: nowrap;
  }

  .nav-dropdown:hover .nav-dropbtn {
    color: white;
  }

  .nav-dropdown-content a:hover {
    background-color: #ddd;
    color: black;
  }

  .nav-right .nav-dropdown-content {
    margin-left: -45%;
    right: 0;
  }

  @media screen and (max-width: 768px) {
    .distill-site-header a, .distill-site-header .nav-dropdown  {display: none;}
    .distill-site-header a.nav-toggle {
      float: right;
      display: block;
    }
    .distill-site-header .title {
      margin-left: 0;
    }
    .distill-site-header .nav-right {
      margin-right: 0;
    }
    .distill-site-header {
      overflow: hidden;
    }
    .nav-right .nav-dropdown-content {
      margin-left: 0;
    }
  }


  @media screen and (max-width: 768px) {
    .distill-site-header.responsive {position: relative; min-height: 500px; }
    .distill-site-header.responsive a.nav-toggle {
      position: absolute;
      right: 0;
      top: 0;
    }
    .distill-site-header.responsive a,
    .distill-site-header.responsive .nav-dropdown {
      display: block;
      text-align: left;
    }
    .distill-site-header.responsive .nav-left,
    .distill-site-header.responsive .nav-right {
      width: 100%;
    }
    .distill-site-header.responsive .nav-dropdown {float: none;}
    .distill-site-header.responsive .nav-dropdown-content {position: relative;}
    .distill-site-header.responsive .nav-dropdown .nav-dropbtn {
      display: block;
      width: 100%;
      text-align: left;
    }
  }

  /* Site Footer */

  .distill-site-footer {
    width: 100%;
    overflow: hidden;
    box-sizing: border-box;
    z-index: 3;
    margin-top: 30px;
    padding-top: 30px;
    padding-bottom: 30px;
    text-align: center;
  }

  /* Headroom */

  d-title {
    padding-top: 6rem;
  }

  @media print {
    d-title {
      padding-top: 4rem;
    }
  }

  .headroom {
    z-index: 1000;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
  }

  .headroom--transition {
    transition: all .4s ease-in-out;
  }

  .headroom--unpinned {
    top: -100px;
  }

  .headroom--pinned {
    top: 0;
  }

  /* adjust viewport for navbar height */
  /* helps vertically center bootstrap (non-distill) content */
  .min-vh-100 {
    min-height: calc(100vh - 100px) !important;
  }

  </style>

  <script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
  <link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet"/>
  <link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet"/>
  <script src="site_libs/headroom-0.9.4/headroom.min.js"></script>
  <script src="site_libs/autocomplete-0.37.1/autocomplete.min.js"></script>
  <script src="site_libs/fuse-6.4.1/fuse.min.js"></script>

  <script type="application/javascript">

  function getMeta(metaName) {
    var metas = document.getElementsByTagName('meta');
    for (let i = 0; i < metas.length; i++) {
      if (metas[i].getAttribute('name') === metaName) {
        return metas[i].getAttribute('content');
      }
    }
    return '';
  }

  function offsetURL(url) {
    var offset = getMeta('distill:offset');
    return offset ? offset + '/' + url : url;
  }

  function createFuseIndex() {

    // create fuse index
    var options = {
      keys: [
        { name: 'title', weight: 20 },
        { name: 'categories', weight: 15 },
        { name: 'description', weight: 10 },
        { name: 'contents', weight: 5 },
      ],
      ignoreLocation: true,
      threshold: 0
    };
    var fuse = new window.Fuse([], options);

    // fetch the main search.json
    return fetch(offsetURL('search.json'))
      .then(function(response) {
        if (response.status == 200) {
          return response.json().then(function(json) {
            // index main articles
            json.articles.forEach(function(article) {
              fuse.add(article);
            });
            // download collections and index their articles
            return Promise.all(json.collections.map(function(collection) {
              return fetch(offsetURL(collection)).then(function(response) {
                if (response.status === 200) {
                  return response.json().then(function(articles) {
                    articles.forEach(function(article) {
                      fuse.add(article);
                    });
                  })
                } else {
                  return Promise.reject(
                    new Error('Unexpected status from search index request: ' +
                              response.status)
                  );
                }
              });
            })).then(function() {
              return fuse;
            });
          });

        } else {
          return Promise.reject(
            new Error('Unexpected status from search index request: ' +
                        response.status)
          );
        }
      });
  }

  window.document.addEventListener("DOMContentLoaded", function (event) {

    // get search element (bail if we don't have one)
    var searchEl = window.document.getElementById('distill-search');
    if (!searchEl)
      return;

    createFuseIndex()
      .then(function(fuse) {

        // make search box visible
        searchEl.classList.remove('hidden');

        // initialize autocomplete
        var options = {
          autoselect: true,
          hint: false,
          minLength: 2,
        };
        window.autocomplete(searchEl, options, [{
          source: function(query, callback) {
            const searchOptions = {
              isCaseSensitive: false,
              shouldSort: true,
              minMatchCharLength: 2,
              limit: 10,
            };
            var results = fuse.search(query, searchOptions);
            callback(results
              .map(function(result) { return result.item; })
            );
          },
          templates: {
            suggestion: function(suggestion) {
              var img = suggestion.preview && Object.keys(suggestion.preview).length > 0
                ? `<img src="${offsetURL(suggestion.preview)}"</img>`
                : '';
              var html = `
                <div class="search-item">
                  <h3>${suggestion.title}</h3>
                  <div class="search-item-description">
                    ${suggestion.description || ''}
                  </div>
                  <div class="search-item-preview">
                    ${img}
                  </div>
                </div>
              `;
              return html;
            }
          }
        }]).on('autocomplete:selected', function(event, suggestion) {
          window.location.href = offsetURL(suggestion.path);
        });
        // remove inline display style on autocompleter (we want to
        // manage responsive display via css)
        $('.algolia-autocomplete').css("display", "");
      })
      .catch(function(error) {
        console.log(error);
      });

  });

  </script>

  <style type="text/css">

  .nav-search {
    font-size: x-small;
  }

  /* Algolioa Autocomplete */

  .algolia-autocomplete {
    display: inline-block;
    margin-left: 10px;
    vertical-align: sub;
    background-color: white;
    color: black;
    padding: 6px;
    padding-top: 8px;
    padding-bottom: 0;
    border-radius: 6px;
    border: 1px #0F2E3D solid;
    width: 180px;
  }


  @media screen and (max-width: 768px) {
    .distill-site-nav .algolia-autocomplete {
      display: none;
      visibility: hidden;
    }
    .distill-site-nav.responsive .algolia-autocomplete {
      display: inline-block;
      visibility: visible;
    }
    .distill-site-nav.responsive .algolia-autocomplete .aa-dropdown-menu {
      margin-left: 0;
      width: 400px;
      max-height: 400px;
    }
  }

  .algolia-autocomplete .aa-input, .algolia-autocomplete .aa-hint {
    width: 90%;
    outline: none;
    border: none;
  }

  .algolia-autocomplete .aa-hint {
    color: #999;
  }
  .algolia-autocomplete .aa-dropdown-menu {
    width: 550px;
    max-height: 70vh;
    overflow-x: visible;
    overflow-y: scroll;
    padding: 5px;
    margin-top: 3px;
    margin-left: -150px;
    background-color: #fff;
    border-radius: 5px;
    border: 1px solid #999;
    border-top: none;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion {
    cursor: pointer;
    padding: 5px 4px;
    border-bottom: 1px solid #eee;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion:last-of-type {
    border-bottom: none;
    margin-bottom: 2px;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item {
    overflow: hidden;
    font-size: 0.8em;
    line-height: 1.4em;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item h3 {
    font-size: 1rem;
    margin-block-start: 0;
    margin-block-end: 5px;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-description {
    display: inline-block;
    overflow: hidden;
    height: 2.8em;
    width: 80%;
    margin-right: 4%;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview {
    display: inline-block;
    width: 15%;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview img {
    height: 3em;
    width: auto;
    display: none;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview img[src] {
    display: initial;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion.aa-cursor {
    background-color: #eee;
  }
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion em {
    font-weight: bold;
    font-style: normal;
  }

  </style>


  <!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->

  <style type="text/css">

  body {
    background-color: white;
  }

  .pandoc-table {
    width: 100%;
  }

  .pandoc-table>caption {
    margin-bottom: 10px;
  }

  .pandoc-table th:not([align]) {
    text-align: left;
  }

  .pagedtable-footer {
    font-size: 15px;
  }

  d-byline .byline {
    grid-template-columns: 2fr 2fr;
  }

  d-byline .byline h3 {
    margin-block-start: 1.5em;
  }

  d-byline .byline .authors-affiliations h3 {
    margin-block-start: 0.5em;
  }

  .authors-affiliations .orcid-id {
    width: 16px;
    height:16px;
    margin-left: 4px;
    margin-right: 4px;
    vertical-align: middle;
    padding-bottom: 2px;
  }

  d-title .dt-tags {
    margin-top: 1em;
    grid-column: text;
  }

  .dt-tags .dt-tag {
    text-decoration: none;
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0em 0.4em;
    margin-right: 0.5em;
    margin-bottom: 0.4em;
    font-size: 70%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  d-article table.gt_table td,
  d-article table.gt_table th {
    border-bottom: none;
  }

  .html-widget {
    margin-bottom: 2.0em;
  }

  .l-screen-inset {
    padding-right: 16px;
  }

  .l-screen .caption {
    margin-left: 10px;
  }

  .shaded {
    background: rgb(247, 247, 247);
    padding-top: 20px;
    padding-bottom: 20px;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .html-widget {
    margin-bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .shaded-content {
    background: white;
  }

  .text-output {
    margin-top: 0;
    line-height: 1.5em;
  }

  .hidden {
    display: none !important;
  }

  d-article {
    padding-top: 2.5rem;
    padding-bottom: 30px;
  }

  d-appendix {
    padding-top: 30px;
  }

  d-article>p>img {
    width: 100%;
  }

  d-article h2 {
    margin: 1rem 0 1.5rem 0;
  }

  d-article h3 {
    margin-top: 1.5rem;
  }

  d-article iframe {
    border: 1px solid rgba(0, 0, 0, 0.1);
    margin-bottom: 2.0em;
    width: 100%;
  }

  /* Tweak code blocks */

  d-article div.sourceCode code,
  d-article pre code {
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
  }

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: auto;
  }

  d-article div.sourceCode {
    background-color: white;
  }

  d-article div.sourceCode pre {
    padding-left: 10px;
    font-size: 12px;
    border-left: 2px solid rgba(0,0,0,0.1);
  }

  d-article pre {
    font-size: 12px;
    color: black;
    background: none;
    margin-top: 0;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  d-article pre a {
    border-bottom: none;
  }

  d-article pre a:hover {
    border-bottom: none;
    text-decoration: underline;
  }

  d-article details {
    grid-column: text;
    margin-bottom: 0.8em;
  }

  @media(min-width: 768px) {

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: visible !important;
  }

  d-article div.sourceCode pre {
    padding-left: 18px;
    font-size: 14px;
  }

  d-article pre {
    font-size: 14px;
  }

  }

  figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  /* CSS for d-contents */

  .d-contents {
    grid-column: text;
    color: rgba(0,0,0,0.8);
    font-size: 0.9em;
    padding-bottom: 1em;
    margin-bottom: 1em;
    padding-bottom: 0.5em;
    margin-bottom: 1em;
    padding-left: 0.25em;
    justify-self: start;
  }

  @media(min-width: 1000px) {
    .d-contents.d-contents-float {
      height: 0;
      grid-column-start: 1;
      grid-column-end: 4;
      justify-self: center;
      padding-right: 3em;
      padding-left: 2em;
    }
  }

  .d-contents nav h3 {
    font-size: 18px;
    margin-top: 0;
    margin-bottom: 1em;
  }

  .d-contents li {
    list-style-type: none
  }

  .d-contents nav > ul {
    padding-left: 0;
  }

  .d-contents ul {
    padding-left: 1em
  }

  .d-contents nav ul li {
    margin-top: 0.6em;
    margin-bottom: 0.2em;
  }

  .d-contents nav a {
    font-size: 13px;
    border-bottom: none;
    text-decoration: none
    color: rgba(0, 0, 0, 0.8);
  }

  .d-contents nav a:hover {
    text-decoration: underline solid rgba(0, 0, 0, 0.6)
  }

  .d-contents nav > ul > li > a {
    font-weight: 600;
  }

  .d-contents nav > ul > li > ul {
    font-weight: inherit;
  }

  .d-contents nav > ul > li > ul > li {
    margin-top: 0.2em;
  }


  .d-contents nav ul {
    margin-top: 0;
    margin-bottom: 0.25em;
  }

  .d-article-with-toc h2:nth-child(2) {
    margin-top: 0;
  }


  /* Figure */

  .figure {
    position: relative;
    margin-bottom: 2.5em;
    margin-top: 1.5em;
  }

  .figure img {
    width: 100%;
  }

  .figure .caption {
    color: rgba(0, 0, 0, 0.6);
    font-size: 12px;
    line-height: 1.5em;
  }

  .figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  .figure .caption a {
    color: rgba(0, 0, 0, 0.6);
  }

  .figure .caption b,
  .figure .caption strong, {
    font-weight: 600;
    color: rgba(0, 0, 0, 1.0);
  }

  /* Citations */

  d-article .citation {
    color: inherit;
    cursor: inherit;
  }

  div.hanging-indent{
    margin-left: 1em; text-indent: -1em;
  }

  /* Citation hover box */

  .tippy-box[data-theme~=light-border] {
    background-color: rgba(250, 250, 250, 0.95);
  }

  .tippy-content > p {
    margin-bottom: 0;
    padding: 2px;
  }


  /* Tweak 1000px media break to show more text */

  @media(min-width: 1000px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 16px;
    }

    .grid {
      grid-column-gap: 16px;
    }

    d-article {
      font-size: 1.06rem;
      line-height: 1.7em;
    }
    figure .caption, .figure .caption, figure figcaption {
      font-size: 13px;
    }
  }

  @media(min-width: 1180px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 32px;
    }

    .grid {
      grid-column-gap: 32px;
    }
  }


  /* Get the citation styles for the appendix (not auto-injected on render since
     we do our own rendering of the citation appendix) */

  d-appendix .citation-appendix,
  .d-appendix .citation-appendix {
    font-size: 11px;
    line-height: 15px;
    border-left: 1px solid rgba(0, 0, 0, 0.1);
    padding-left: 18px;
    border: 1px solid rgba(0,0,0,0.1);
    background: rgba(0, 0, 0, 0.02);
    padding: 10px 18px;
    border-radius: 3px;
    color: rgba(150, 150, 150, 1);
    overflow: hidden;
    margin-top: -12px;
    white-space: pre-wrap;
    word-wrap: break-word;
  }

  /* Include appendix styles here so they can be overridden */

  d-appendix {
    contain: layout style;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-top: 60px;
    margin-bottom: 0;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    color: rgba(0,0,0,0.5);
    padding-top: 60px;
    padding-bottom: 48px;
  }

  d-appendix h3 {
    grid-column: page-start / text-start;
    font-size: 15px;
    font-weight: 500;
    margin-top: 1em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.65);
  }

  d-appendix h3 + * {
    margin-top: 1em;
  }

  d-appendix ol {
    padding: 0 0 0 15px;
  }

  @media (min-width: 768px) {
    d-appendix ol {
      padding: 0 0 0 30px;
      margin-left: -30px;
    }
  }

  d-appendix li {
    margin-bottom: 1em;
  }

  d-appendix a {
    color: rgba(0, 0, 0, 0.6);
  }

  d-appendix > * {
    grid-column: text;
  }

  d-appendix > d-footnote-list,
  d-appendix > d-citation-list,
  d-appendix > distill-appendix {
    grid-column: screen;
  }

  /* Include footnote styles here so they can be overridden */

  d-footnote-list {
    contain: layout style;
  }

  d-footnote-list > * {
    grid-column: text;
  }

  d-footnote-list a.footnote-backlink {
    color: rgba(0,0,0,0.3);
    padding-left: 0.5em;
  }



  /* Anchor.js */

  .anchorjs-link {
    /*transition: all .25s linear; */
    text-decoration: none;
    border-bottom: none;
  }
  *:hover > .anchorjs-link {
    margin-left: -1.125em !important;
    text-decoration: none;
    border-bottom: none;
  }

  /* Social footer */

  .social_footer {
    margin-top: 30px;
    margin-bottom: 0;
    color: rgba(0,0,0,0.67);
  }

  .disqus-comments {
    margin-right: 30px;
  }

  .disqus-comment-count {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    cursor: pointer;
  }

  #disqus_thread {
    margin-top: 30px;
  }

  .article-sharing a {
    border-bottom: none;
    margin-right: 8px;
  }

  .article-sharing a:hover {
    border-bottom: none;
  }

  .sidebar-section.subscribe {
    font-size: 12px;
    line-height: 1.6em;
  }

  .subscribe p {
    margin-bottom: 0.5em;
  }


  .article-footer .subscribe {
    font-size: 15px;
    margin-top: 45px;
  }


  .sidebar-section.custom {
    font-size: 12px;
    line-height: 1.6em;
  }

  .custom p {
    margin-bottom: 0.5em;
  }

  /* Styles for listing layout (hide title) */
  .layout-listing d-title, .layout-listing .d-title {
    display: none;
  }

  /* Styles for posts lists (not auto-injected) */


  .posts-with-sidebar {
    padding-left: 45px;
    padding-right: 45px;
  }

  .posts-list .description h2,
  .posts-list .description p {
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
  }

  .posts-list .description h2 {
    font-weight: 700;
    border-bottom: none;
    padding-bottom: 0;
  }

  .posts-list h2.post-tag {
    border-bottom: 1px solid rgba(0, 0, 0, 0.2);
    padding-bottom: 12px;
  }
  .posts-list {
    margin-top: 60px;
    margin-bottom: 24px;
  }

  .posts-list .post-preview {
    text-decoration: none;
    overflow: hidden;
    display: block;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
    padding: 24px 0;
  }

  .post-preview-last {
    border-bottom: none !important;
  }

  .posts-list .posts-list-caption {
    grid-column: screen;
    font-weight: 400;
  }

  .posts-list .post-preview h2 {
    margin: 0 0 6px 0;
    line-height: 1.2em;
    font-style: normal;
    font-size: 24px;
  }

  .posts-list .post-preview p {
    margin: 0 0 12px 0;
    line-height: 1.4em;
    font-size: 16px;
  }

  .posts-list .post-preview .thumbnail {
    box-sizing: border-box;
    margin-bottom: 24px;
    position: relative;
    max-width: 500px;
  }
  .posts-list .post-preview img {
    width: 100%;
    display: block;
  }

  .posts-list .metadata {
    font-size: 12px;
    line-height: 1.4em;
    margin-bottom: 18px;
  }

  .posts-list .metadata > * {
    display: inline-block;
  }

  .posts-list .metadata .publishedDate {
    margin-right: 2em;
  }

  .posts-list .metadata .dt-authors {
    display: block;
    margin-top: 0.3em;
    margin-right: 2em;
  }

  .posts-list .dt-tags {
    display: block;
    line-height: 1em;
  }

  .posts-list .dt-tags .dt-tag {
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0.3em 0.4em;
    margin-right: 0.2em;
    margin-bottom: 0.4em;
    font-size: 60%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  .posts-list img {
    opacity: 1;
  }

  .posts-list img[data-src] {
    opacity: 0;
  }

  .posts-more {
    clear: both;
  }


  .posts-sidebar {
    font-size: 16px;
  }

  .posts-sidebar h3 {
    font-size: 16px;
    margin-top: 0;
    margin-bottom: 0.5em;
    font-weight: 400;
    text-transform: uppercase;
  }

  .sidebar-section {
    margin-bottom: 30px;
  }

  .categories ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
  }

  .categories li {
    color: rgba(0, 0, 0, 0.8);
    margin-bottom: 0;
  }

  .categories li>a {
    border-bottom: none;
  }

  .categories li>a:hover {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
  }

  .categories .active {
    font-weight: 600;
  }

  .categories .category-count {
    color: rgba(0, 0, 0, 0.4);
  }


  @media(min-width: 768px) {
    .posts-list .post-preview h2 {
      font-size: 26px;
    }
    .posts-list .post-preview .thumbnail {
      float: right;
      width: 30%;
      margin-bottom: 0;
    }
    .posts-list .post-preview .description {
      float: left;
      width: 45%;
    }
    .posts-list .post-preview .metadata {
      float: left;
      width: 20%;
      margin-top: 8px;
    }
    .posts-list .post-preview p {
      margin: 0 0 12px 0;
      line-height: 1.5em;
      font-size: 16px;
    }
    .posts-with-sidebar .posts-list {
      float: left;
      width: 75%;
    }
    .posts-with-sidebar .posts-sidebar {
      float: right;
      width: 20%;
      margin-top: 60px;
      padding-top: 24px;
      padding-bottom: 24px;
    }
  }


  /* Improve display for browsers without grid (IE/Edge <= 15) */

  .downlevel {
    line-height: 1.6em;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
    margin: 0;
  }

  .downlevel .d-title {
    padding-top: 6rem;
    padding-bottom: 1.5rem;
  }

  .downlevel .d-title h1 {
    font-size: 50px;
    font-weight: 700;
    line-height: 1.1em;
    margin: 0 0 0.5rem;
  }

  .downlevel .d-title p {
    font-weight: 300;
    font-size: 1.2rem;
    line-height: 1.55em;
    margin-top: 0;
  }

  .downlevel .d-byline {
    padding-top: 0.8em;
    padding-bottom: 0.8em;
    font-size: 0.8rem;
    line-height: 1.8em;
  }

  .downlevel .section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
  }

  .downlevel .d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
    padding-top: 1rem;
    padding-bottom: 2rem;
  }


  .downlevel .d-appendix {
    padding-left: 0;
    padding-right: 0;
    max-width: none;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.5);
    padding-top: 40px;
    padding-bottom: 48px;
  }

  .downlevel .footnotes ol {
    padding-left: 13px;
  }

  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
    padding-left: 40px;
    padding-right: 40px;
  }

  @media(min-width: 768px) {
    .downlevel .base-grid,
    .downlevel .distill-header,
    .downlevel .d-title,
    .downlevel .d-abstract,
    .downlevel .d-article,
    .downlevel .d-appendix,
    .downlevel .distill-appendix,
    .downlevel .d-byline,
    .downlevel .d-footnote-list,
    .downlevel .d-citation-list,
    .downlevel .distill-footer,
    .downlevel .appendix-bottom,
    .downlevel .posts-container {
    padding-left: 150px;
    padding-right: 150px;
    max-width: 900px;
  }
  }

  .downlevel pre code {
    display: block;
    border-left: 2px solid rgba(0, 0, 0, .1);
    padding: 0 0 0 20px;
    font-size: 14px;
  }

  .downlevel code, .downlevel pre {
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  .downlevel .posts-list .post-preview {
    color: inherit;
  }



  </style>

  <script type="application/javascript">

  function is_downlevel_browser() {
    if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                   window.navigator.userAgent)) {
      return true;
    } else {
      return window.load_distill_framework === undefined;
    }
  }

  // show body when load is complete
  function on_load_complete() {

    // add anchors
    if (window.anchors) {
      window.anchors.options.placement = 'left';
      window.anchors.add('d-article > h2, d-article > h3, d-article > h4, d-article > h5');
    }


    // set body to visible
    document.body.style.visibility = 'visible';

    // force redraw for leaflet widgets
    if (window.HTMLWidgets) {
      var maps = window.HTMLWidgets.findAll(".leaflet");
      $.each(maps, function(i, el) {
        var map = this.getMap();
        map.invalidateSize();
        map.eachLayer(function(layer) {
          if (layer instanceof L.TileLayer)
            layer.redraw();
        });
      });
    }

    // trigger 'shown' so htmlwidgets resize
    $('d-article').trigger('shown');
  }

  function init_distill() {

    init_common();

    // create front matter
    var front_matter = $('<d-front-matter></d-front-matter>');
    $('#distill-front-matter').wrap(front_matter);

    // create d-title
    $('.d-title').changeElementType('d-title');

    // create d-byline
    var byline = $('<d-byline></d-byline>');
    $('.d-byline').replaceWith(byline);

    // create d-article
    var article = $('<d-article></d-article>');
    $('.d-article').wrap(article).children().unwrap();

    // move posts container into article
    $('.posts-container').appendTo($('d-article'));

    // create d-appendix
    $('.d-appendix').changeElementType('d-appendix');

    // flag indicating that we have appendix items
    var appendix = $('.appendix-bottom').children('h3').length > 0;

    // replace footnotes with <d-footnote>
    $('.footnote-ref').each(function(i, val) {
      appendix = true;
      var href = $(this).attr('href');
      var id = href.replace('#', '');
      var fn = $('#' + id);
      var fn_p = $('#' + id + '>p');
      fn_p.find('.footnote-back').remove();
      var text = fn_p.html();
      var dtfn = $('<d-footnote></d-footnote>');
      dtfn.html(text);
      $(this).replaceWith(dtfn);
    });
    // remove footnotes
    $('.footnotes').remove();

    // move refs into #references-listing
    $('#references-listing').replaceWith($('#refs'));

    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      var id = $(this).attr('id');
      $('.d-contents a[href="#' + id + '"]').parent().remove();
      appendix = true;
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
    });

    // show d-appendix if we have appendix content
    $("d-appendix").css('display', appendix ? 'grid' : 'none');

    // localize layout chunks to just output
    $('.layout-chunk').each(function(i, val) {

      // capture layout
      var layout = $(this).attr('data-layout');

      // apply layout to markdown level block elements
      var elements = $(this).children().not('details, div.sourceCode, pre, script');
      elements.each(function(i, el) {
        var layout_div = $('<div class="' + layout + '"></div>');
        if (layout_div.hasClass('shaded')) {
          var shaded_content = $('<div class="shaded-content"></div>');
          $(this).wrap(shaded_content);
          $(this).parent().wrap(layout_div);
        } else {
          $(this).wrap(layout_div);
        }
      });


      // unwrap the layout-chunk div
      $(this).children().unwrap();
    });

    // remove code block used to force  highlighting css
    $('.distill-force-highlighting-css').parent().remove();

    // remove empty line numbers inserted by pandoc when using a
    // custom syntax highlighting theme
    $('code.sourceCode a:empty').remove();

    // load distill framework
    load_distill_framework();

    // wait for window.distillRunlevel == 4 to do post processing
    function distill_post_process() {

      if (!window.distillRunlevel || window.distillRunlevel < 4)
        return;

      // hide author/affiliations entirely if we have no authors
      var front_matter = JSON.parse($("#distill-front-matter").html());
      var have_authors = front_matter.authors && front_matter.authors.length > 0;
      if (!have_authors)
        $('d-byline').addClass('hidden');

      // article with toc class
      $('.d-contents').parent().addClass('d-article-with-toc');

      // strip links that point to #
      $('.authors-affiliations').find('a[href="#"]').removeAttr('href');

      // add orcid ids
      $('.authors-affiliations').find('.author').each(function(i, el) {
        var orcid_id = front_matter.authors[i].orcidID;
        if (orcid_id) {
          var a = $('<a></a>');
          a.attr('href', 'https://orcid.org/' + orcid_id);
          var img = $('<img></img>');
          img.addClass('orcid-id');
          img.attr('alt', 'ORCID ID');
          img.attr('src','data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==');
          a.append(img);
          $(this).append(a);
        }
      });

      // hide elements of author/affiliations grid that have no value
      function hide_byline_column(caption) {
        $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
      }

      // affiliations
      var have_affiliations = false;
      for (var i = 0; i<front_matter.authors.length; ++i) {
        var author = front_matter.authors[i];
        if (author.affiliation !== "&nbsp;") {
          have_affiliations = true;
          break;
        }
      }
      if (!have_affiliations)
        $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');

      // published date
      if (!front_matter.publishedDate)
        hide_byline_column("Published");

      // document object identifier
      var doi = $('d-byline').find('h3:contains("DOI")');
      var doi_p = doi.next().empty();
      if (!front_matter.doi) {
        // if we have a citation and valid citationText then link to that
        if ($('#citation').length > 0 && front_matter.citationText) {
          doi.html('Citation');
          $('<a href="#citation"></a>')
            .text(front_matter.citationText)
            .appendTo(doi_p);
        } else {
          hide_byline_column("DOI");
        }
      } else {
        $('<a></a>')
           .attr('href', "https://doi.org/" + front_matter.doi)
           .html(front_matter.doi)
           .appendTo(doi_p);
      }

       // change plural form of authors/affiliations
      if (front_matter.authors.length === 1) {
        var grid = $('.authors-affiliations');
        grid.children('h3:contains("Authors")').text('Author');
        grid.children('h3:contains("Affiliations")').text('Affiliation');
      }

      // remove d-appendix and d-footnote-list local styles
      $('d-appendix > style:first-child').remove();
      $('d-footnote-list > style:first-child').remove();

      // move appendix-bottom entries to the bottom
      $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
      $('.appendix-bottom').remove();

      // hoverable references
      $('span.citation[data-cites]').each(function() {
        var refHtml = $('#ref-' + $(this).attr('data-cites')).html();
        window.tippy(this, {
          allowHTML: true,
          content: refHtml,
          maxWidth: 500,
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start'
        });
      });

      // clear polling timer
      clearInterval(tid);

      // show body now that everything is ready
      on_load_complete();
    }

    var tid = setInterval(distill_post_process, 50);
    distill_post_process();

  }

  function init_downlevel() {

    init_common();

     // insert hr after d-title
    $('.d-title').after($('<hr class="section-separator"/>'));

    // check if we have authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;

    // manage byline/border
    if (!have_authors)
      $('.d-byline').remove();
    $('.d-byline').after($('<hr class="section-separator"/>'));
    $('.d-byline a').remove();

    // remove toc
    $('.d-contents').remove();

    // move appendix elements
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
    });


    // inject headers into references and footnotes
    var refs_header = $('<h3></h3>');
    refs_header.text('References');
    $('#refs').prepend(refs_header);

    var footnotes_header = $('<h3></h3');
    footnotes_header.text('Footnotes');
    $('.footnotes').children('hr').first().replaceWith(footnotes_header);

    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
    $('.appendix-bottom').remove();

    // remove appendix if it's empty
    if ($('.d-appendix').children().length === 0)
      $('.d-appendix').remove();

    // prepend separator above appendix
    $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));

    // trim code
    $('pre>code').each(function(i, val) {
      $(this).html($.trim($(this).html()));
    });

    // move posts-container right before article
    $('.posts-container').insertBefore($('.d-article'));

    $('body').addClass('downlevel');

    on_load_complete();
  }


  function init_common() {

    // jquery plugin to change element types
    (function($) {
      $.fn.changeElementType = function(newType) {
        var attrs = {};

        $.each(this[0].attributes, function(idx, attr) {
          attrs[attr.nodeName] = attr.nodeValue;
        });

        this.replaceWith(function() {
          return $("<" + newType + "/>", attrs).append($(this).contents());
        });
      };
    })(jQuery);

    // prevent underline for linked images
    $('a > img').parent().css({'border-bottom' : 'none'});

    // mark non-body figures created by knitr chunks as 100% width
    $('.layout-chunk').each(function(i, val) {
      var figures = $(this).find('img, .html-widget');
      if ($(this).attr('data-layout') !== "l-body") {
        figures.css('width', '100%');
      } else {
        figures.css('max-width', '100%');
        figures.filter("[width]").each(function(i, val) {
          var fig = $(this);
          fig.css('width', fig.attr('width') + 'px');
        });

      }
    });

    // auto-append index.html to post-preview links in file: protocol
    // and in rstudio ide preview
    $('.post-preview').each(function(i, val) {
      if (window.location.protocol === "file:")
        $(this).attr('href', $(this).attr('href') + "index.html");
    });

    // get rid of index.html references in header
    if (window.location.protocol !== "file:") {
      $('.distill-site-header a[href]').each(function(i,val) {
        $(this).attr('href', $(this).attr('href').replace("index.html", "./"));
      });
    }

    // add class to pandoc style tables
    $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
    $('.kable-table').children('table').addClass('pandoc-table');

    // add figcaption style to table captions
    $('caption').parent('table').addClass("figcaption");

    // initialize posts list
    if (window.init_posts_list)
      window.init_posts_list();

    // implmement disqus comment link
    $('.disqus-comment-count').click(function() {
      window.headroom_prevent_pin = true;
      $('#disqus_thread').toggleClass('hidden');
      if (!$('#disqus_thread').hasClass('hidden')) {
        var offset = $(this).offset();
        $(window).resize();
        $('html, body').animate({
          scrollTop: offset.top - 35
        });
      }
    });
  }

  document.addEventListener('DOMContentLoaded', function() {
    if (is_downlevel_browser())
      init_downlevel();
    else
      window.addEventListener('WebComponentsReady', init_distill);
  });

  </script>

  <!--/radix_placeholder_distill-->
  <script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
  <script src="site_libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
  <script src="site_libs/jquery-3.5.1/jquery.min.js"></script>
  <link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
  <script src="site_libs/datatables-binding-0.17/datatables.js"></script>
  <link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
  <link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
  <script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
  <link href="site_libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
  <script src="site_libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
  <script src="site_libs/popper-2.6.0/popper.min.js"></script>
  <link href="site_libs/tippy-6.2.7/tippy.css" rel="stylesheet" />
  <link href="site_libs/tippy-6.2.7/tippy-light-border.css" rel="stylesheet" />
  <script src="site_libs/tippy-6.2.7/tippy.umd.min.js"></script>
  <script src="site_libs/anchor-4.2.2/anchor.min.js"></script>
  <script src="site_libs/bowser-1.9.3/bowser.min.js"></script>
  <script src="site_libs/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="site_libs/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
  <!--/radix_placeholder_site_in_header-->


</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"Multi-sample comparisons of the Kinkel (C086) male B cells RNA-seq data set","description":"Read-based analysis","authors":[{"author":"Peter Hickey","authorURL":"https://peterhickey.org","affiliation":"WEHI SCORE","affiliationURL":"https://www.wehi.edu.au/people/shalin-naik/3310/score","orcidID":""}],"publishedDate":"2021-10-15T00:00:00.000+11:00","citationText":"Hickey, 2021"}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<header class="header header--fixed" role="banner">
<nav class="distill-site-nav distill-site-header">
<div class="nav-left">
<a href="index.html" class="title">C086</a>
<a href="index.html">Introduction</a>
<a href="C086_Kinkel.preprocess.html">Preprocessing</a>
<div class="nav-dropdown">
<button class="nav-dropbtn">
Differential analyses
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<a href="C086_Kinkel.multi-sample_comparisons.html">HSCs</a>
<a href="C086_Kinkel.female_B_cells.multi-sample_comparisons.html">Female B cells</a>
<a href="C086_Kinkel.male_B_cells.multi-sample_comparisons.html">Male B cells</a>
</div>
</div>
<div class="nav-dropdown">
<button class="nav-dropbtn">
Supplementary files
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<a href="../output/scPipe/">scPipe reports</a>
<a href="../output/DEGs/">Differentially expressed genes</a>
</div>
</div>
</div>
<div class="nav-right">
<a href="https://github.com/WEHISCORE/C086_Kinkel">
<i class="fa fa-github" aria-hidden="true"></i>
</a>
<a href="javascript:void(0);" class="nav-toggle">&#9776;</a>
</div>
</nav>
</header>
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>Multi-sample comparisons of the Kinkel (C086) male B cells RNA-seq data set</h1>
<!--radix_placeholder_categories-->
<!--/radix_placeholder_categories-->
<p>Read-based analysis</p>
</div>

<div class="d-byline">
  Peter Hickey <a href="https://peterhickey.org" class="uri">https://peterhickey.org</a> (WEHI SCORE)<a href="https://www.wehi.edu.au/people/shalin-naik/3310/score" class="uri">https://www.wehi.edu.au/people/shalin-naik/3310/score</a>
  
<br/>2021-10-15
</div>

<div class="d-article">
<div class="d-contents d-contents-float">
<nav class="l-text toc figcaption" id="TOC">
<h3>Contents</h3>
<ul>
<li><a href="#summary">Summary</a></li>
<li><a href="#setup">Setup</a>
<ul>
<li><a href="#load-data">Load data</a></li>
<li><a href="#cleaning-up-duplicate-features">Cleaning up duplicate features</a></li>
<li><a href="#create-dgelist-objects">Create DGEList objects</a></li>
<li><a href="#incorporating-gene-based-annotation">Incorporating gene-based annotation</a></li>
<li><a href="#gene-filtering">Gene filtering</a></li>
<li><a href="#normalization">Normalization</a></li>
</ul></li>
<li><a href="#analysis-of-male-b-cell-samples">Analysis of male B cell samples</a>
<ul>
<li><a href="#mds">MDS</a></li>
<li><a href="#de-analysis">DE analysis</a></li>
<li><a href="#gene-set-tests">Gene set tests</a>
<ul>
<li><a href="#goana">goana</a></li>
<li><a href="#kegga">kegga</a></li>
<li><a href="#camera-with-msigdb-gene-sets">camera with MSigDB gene sets</a></li>
<li><a href="#sarahs-gene-lists">Sarahs gene lists</a></li>
</ul></li>
</ul></li>
<li><a href="#session-info">Session info</a></li>
</ul>
</nav>
</div>
<h1 id="summary">Summary</h1>
<p>This report analyses the <code>Del</code> and <code>Het</code> male samples only. The report begins with some general <a href="#setup">Setup</a> and is followed by various analyses denoted by a prefix (<code>y</code>)</p>
<ul>
<li><code>y</code>: <a href="#analysis-of-male-b-cell-samples">Analysis of male B cell samples</a></li>
</ul>
<h1 id="setup">Setup</h1>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://here.r-lib.org/'>here</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='http://bioinf.wehi.edu.au/edgeR'>edgeR</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://scales.r-lib.org'>scales</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/hasaru-k/GlimmaV2'>Glimma</a></span><span class='op'>)</span>

<span class='kw'><a href='https://rdrr.io/r/base/source.html'>source</a></span><span class='op'>(</span><span class='fu'><a href='https://here.r-lib.org//reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"code/helper_functions.R"</span><span class='op'>)</span><span class='op'>)</span>

<span class='va'>outdir</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://here.r-lib.org//reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"output"</span>, <span class='st'>"DEGs"</span>, <span class='st'>"male_B_cells"</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/files2.html'>dir.create</a></span><span class='op'>(</span><span class='va'>outdir</span>, recursive <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>

<span class='co'># Some useful colours (copied from HSC analysis)</span>
<span class='va'>genotype_colours</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Del"</span> <span class='op'>=</span> <span class='st'>"#1B9E77"</span>, <span class='st'>"Het"</span> <span class='op'>=</span> <span class='st'>"#D95F02"</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<h2 id="load-data">Load data</h2>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>x</span> <span class='op'>&lt;-</span> <span class='fu'>data.table</span><span class='fu'>::</span><span class='fu'><a href='https://Rdatatable.gitlab.io/data.table/reference/fread.html'>fread</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://here.r-lib.org//reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"data"</span>, <span class='st'>"raw_counts"</span>, <span class='st'>"Smchd1_maleBcell_rawcounts.txt"</span><span class='op'>)</span>,
  data.table <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>There are 25,965 features.</p>
<h2 id="cleaning-up-duplicate-features">Cleaning up duplicate features</h2>
<p>There are 56 features with a blank <code>ID</code><a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a>. These are shown in the table below.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>tmp</span> <span class='op'>&lt;-</span> <span class='va'>x</span><span class='op'>[</span>
  <span class='va'>x</span><span class='op'>$</span><span class='va'>ID</span> <span class='op'>==</span> <span class='st'>""</span>,
  <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Probe"</span>, <span class='st'>"Chromosome"</span>, <span class='st'>"Start"</span>, <span class='st'>"End"</span>, <span class='st'>"Probe Strand"</span>, <span class='st'>"Feature"</span>, <span class='st'>"ID"</span><span class='op'>)</span><span class='op'>]</span>

<span class='fu'>DT</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/DT/man/datatable.html'>datatable</a></span><span class='op'>(</span>
  <span class='va'>tmp</span>,
  caption <span class='op'>=</span> <span class='st'>"Features with duplicate 'Probe' values or blank 'ID' values."</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div id="htmlwidget-2a8d19eb606e8d464aa8" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-2a8d19eb606e8d464aa8">{"x":{"filter":"none","caption":"<caption>Features with duplicate 'Probe' values or blank 'ID' values.<\/caption>","data":[["54","1174","1175","2779","2780","2781","2782","3242","3243","3244","3382","4578","5741","6461","6784","8057","9539","10001","11210","11421","11514","11515","11668","13936","14496","14497","15749","15750","16263","16758","16759","16833","17160","17161","17596","17598","18119","18651","18652","18915","19514","19515","20820","20963","20964","21775","22541","22543","22563","22632","22846","23162","24809","25030","25237","25238"],["Kcnb2","Gm4955","Gm4955","Spnb5","Spnb5","Spnb5","Spnb5","Efcab8","Efcab8","Efcab8","Zhx3","Adora3","B230314M03Rik","Vps13d","Magi2","2810453I06Rik","Gm15494","Pou2f2","Tead1","Vkorc1","Jakmip3","Jakmip3","Tnfrsf22","Ccrl1","Akd1","Akd1","Gm12162","Gm12162","Rnasek","Gm11534","Gm11534","B230217C12Rik","Scn4a","Scn4a","Itgb1bp1","Itgb1bp1","Serpina11","Sycp2l","Sycp2l","Zfp708","Dnahc12","Dnahc12","Tob2","Muc19","Muc19","2810055G20Rik","C2","C2","Ly6g6c","A930015D03Rik","Prickle4","Gm10125","Gm362","Mtcp1","Tex16","Tex16"],["1","1","1","2","2","2","2","2","2","2","2","3","4","4","5","5","7","7","7","7","7","7","7","9","10","10","11","11","11","11","11","11","11","11","12","12","12","13","13","13","14","14","15","15","15","16","17","17","17","17","17","18","X","X","X","X"],[15287259,173468509,173480371,120046157,120059068,120061586,120076214,153780879,153798619,153842981,160743927,105870858,107806033,144972623,19227046,143548706,4445697,25091909,112679318,127886227,138940730,139064641,143634806,104084157,41316247,41351711,45409042,45452399,70235206,95869154,95875078,97840780,106318592,106349455,21240825,21269806,103980243,41114307,41148796,67069400,26693274,26831301,81848270,91838326,91908911,77329259,34856412,34862604,35065388,35994453,47686739,5491501,43590989,75404846,112011007,112093496],[15716367,173474052,173491041,120057497,120061436,120072523,120085772,153795182,153816482,153844752,160872998,105924036,107823010,145194995,20706049,143564531,4471285,25159922,112906807,127895617,139031950,139083976,143649661,104126643,41348269,41433288,45420784,45468504,70239842,95869724,95876380,97844099,106349389,106353288,21247323,21286237,103996579,41146658,41171684,67114028,26816334,26891703,81858796,91906298,91936388,77598531,34859310,34898265,35070050,36038441,47694736,5592439,43593532,75416588,112093089,112127326],["+","-","-","-","-","-","-","+","+","+","-","+","-","-","+","+","-","-","+","-","+","+","-","-","+","+","-","-","-","+","+","+","-","-","-","-","-","+","+","-","+","+","-","+","+","+","-","-","+","+","-","-","+","-","+","+"],["null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null"],["","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","",""]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Probe<\/th>\n      <th>Chromosome<\/th>\n      <th>Start<\/th>\n      <th>End<\/th>\n      <th>Probe Strand<\/th>\n      <th>Feature<\/th>\n      <th>ID<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<p>Its not clear why this has happened, but such features complicate downstream analysis. To remedy this, we use the value in the <code>Probe</code> column to look up the corresponding Ensembl ID.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'>EnsDb.Mmusculus.v79</span><span class='op'>)</span>

<span class='va'>id</span> <span class='op'>&lt;-</span> <span class='fu'>mapIds</span><span class='op'>(</span>
  x <span class='op'>=</span> <span class='va'>EnsDb.Mmusculus.v79</span>, 
  keys <span class='op'>=</span> <span class='va'>x</span><span class='op'>$</span><span class='va'>Probe</span><span class='op'>[</span><span class='va'>x</span><span class='op'>$</span><span class='va'>ID</span> <span class='op'>==</span> <span class='st'>""</span><span class='op'>]</span>,
  keytype <span class='op'>=</span> <span class='st'>"GENENAME"</span>,
  column <span class='op'>=</span> <span class='st'>"GENEID"</span>,
  multiVals <span class='op'>=</span> <span class='st'>"first"</span><span class='op'>)</span>

<span class='va'>x</span><span class='op'>$</span><span class='va'>ID</span><span class='op'>[</span><span class='va'>x</span><span class='op'>$</span><span class='va'>ID</span> <span class='op'>==</span> <span class='st'>""</span><span class='op'>]</span> <span class='op'>&lt;-</span> <span class='va'>id</span>
<span class='va'>tmp2</span> <span class='op'>&lt;-</span> <span class='va'>tmp</span><span class='op'>[</span><span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/NA.html'>is.na</a></span><span class='op'>(</span><span class='va'>id</span><span class='op'>)</span>, <span class='op'>]</span>
<span class='va'>tmp3</span> <span class='op'>&lt;-</span> <span class='va'>tmp</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/NA.html'>is.na</a></span><span class='op'>(</span><span class='va'>id</span><span class='op'>)</span>, <span class='op'>]</span>
</code></pre>
</div>
</details>
</div>
<p>This provides an <code>ID</code> for 44 / 56 features.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'>DT</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/DT/man/datatable.html'>datatable</a></span><span class='op'>(</span>
  <span class='va'>tmp2</span>,
  caption <span class='op'>=</span> <span class='st'>"Features successfully mapped to an 'ID' by the 'Probe' value."</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div id="htmlwidget-4e7aa81581795778de8c" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-4e7aa81581795778de8c">{"x":{"filter":"none","caption":"<caption>Features successfully mapped to an 'ID' by the 'Probe' value.<\/caption>","data":[["54","1174","1175","3242","3243","3244","3382","4578","6461","6784","9539","10001","11210","11421","11514","11515","11668","15749","15750","16263","16758","16759","16833","17160","17161","17596","17598","18119","18651","18652","18915","20820","20963","20964","22541","22543","22563","22632","22846","23162","24809","25030","25237","25238"],["Kcnb2","Gm4955","Gm4955","Efcab8","Efcab8","Efcab8","Zhx3","Adora3","Vps13d","Magi2","Gm15494","Pou2f2","Tead1","Vkorc1","Jakmip3","Jakmip3","Tnfrsf22","Gm12162","Gm12162","Rnasek","Gm11534","Gm11534","B230217C12Rik","Scn4a","Scn4a","Itgb1bp1","Itgb1bp1","Serpina11","Sycp2l","Sycp2l","Zfp708","Tob2","Muc19","Muc19","C2","C2","Ly6g6c","A930015D03Rik","Prickle4","Gm10125","Gm362","Mtcp1","Tex16","Tex16"],["1","1","1","2","2","2","2","3","4","5","7","7","7","7","7","7","7","11","11","11","11","11","11","11","11","12","12","12","13","13","13","15","15","15","17","17","17","17","17","18","X","X","X","X"],[15287259,173468509,173480371,153780879,153798619,153842981,160743927,105870858,144972623,19227046,4445697,25091909,112679318,127886227,138940730,139064641,143634806,45409042,45452399,70235206,95869154,95875078,97840780,106318592,106349455,21240825,21269806,103980243,41114307,41148796,67069400,81848270,91838326,91908911,34856412,34862604,35065388,35994453,47686739,5491501,43590989,75404846,112011007,112093496],[15716367,173474052,173491041,153795182,153816482,153844752,160872998,105924036,145194995,20706049,4471285,25159922,112906807,127895617,139031950,139083976,143649661,45420784,45468504,70239842,95869724,95876380,97844099,106349389,106353288,21247323,21286237,103996579,41146658,41171684,67114028,81858796,91906298,91936388,34859310,34898265,35070050,36038441,47694736,5592439,43593532,75416588,112093089,112127326],["+","-","-","+","+","+","-","+","-","+","-","-","+","-","+","+","-","-","-","-","+","+","+","-","-","-","-","-","+","+","-","-","+","+","-","-","+","+","-","-","+","-","+","+"],["null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null","null"],["","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","",""]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Probe<\/th>\n      <th>Chromosome<\/th>\n      <th>Start<\/th>\n      <th>End<\/th>\n      <th>Probe Strand<\/th>\n      <th>Feature<\/th>\n      <th>ID<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<p>This leaves 12 / 56 features for which we cant map the <code>Probe</code> to an <code>ID</code>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'>DT</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/DT/man/datatable.html'>datatable</a></span><span class='op'>(</span>
  <span class='va'>tmp3</span>,
  caption <span class='op'>=</span> <span class='st'>"Features that cannot be mapped to an 'ID' by the 'Probe' value."</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div id="htmlwidget-3ff612ab1db5743f56a8" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-3ff612ab1db5743f56a8">{"x":{"filter":"none","caption":"<caption>Features that cannot be mapped to an 'ID' by the 'Probe' value.<\/caption>","data":[["2779","2780","2781","2782","5741","8057","13936","14496","14497","19514","19515","21775"],["Spnb5","Spnb5","Spnb5","Spnb5","B230314M03Rik","2810453I06Rik","Ccrl1","Akd1","Akd1","Dnahc12","Dnahc12","2810055G20Rik"],["2","2","2","2","4","5","9","10","10","14","14","16"],[120046157,120059068,120061586,120076214,107806033,143548706,104084157,41316247,41351711,26693274,26831301,77329259],[120057497,120061436,120072523,120085772,107823010,143564531,104126643,41348269,41433288,26816334,26891703,77598531],["-","-","-","-","-","+","-","+","+","+","+","+"],["null","null","null","null","null","null","null","null","null","null","null","null"],["","","","","","","","","","","",""]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Probe<\/th>\n      <th>Chromosome<\/th>\n      <th>Start<\/th>\n      <th>End<\/th>\n      <th>Probe Strand<\/th>\n      <th>Feature<\/th>\n      <th>ID<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<p>We opt to remove these features from the dataset.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>x</span> <span class='op'>&lt;-</span> <span class='va'>x</span><span class='op'>[</span><span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/NA.html'>is.na</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>$</span><span class='va'>ID</span><span class='op'>)</span>, <span class='op'>]</span>
</code></pre>
</div>
</details>
</div>
<p>Finally, there are 12 <code>ID</code> values that appear more than once. We pool (sum) together counts from features with the same <code>ID</code>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>counts</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/rowsum.html'>rowsum</a></span><span class='op'>(</span>
  x <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/matrix.html'>as.matrix</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/grep.html'>grep</a></span><span class='op'>(</span><span class='st'>"bam"</span>, <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span><span class='op'>)</span><span class='op'>]</span><span class='op'>)</span>,
  group <span class='op'>=</span> <span class='va'>x</span><span class='op'>$</span><span class='va'>ID</span>,
  reorder <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<h2 id="create-dgelist-objects">Create DGEList objects</h2>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='co'># NOTE: Only retain a small subset of columns since we re-annotate the </span>
<span class='co'>#       gene-based metadata, below.</span>
<span class='va'>genes</span> <span class='op'>&lt;-</span> <span class='va'>x</span><span class='op'>[</span><span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/duplicated.html'>duplicated</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>$</span><span class='va'>ID</span><span class='op'>)</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"ID"</span>, <span class='st'>"Description"</span><span class='op'>)</span><span class='op'>]</span>
<span class='fu'><a href='https://rdrr.io/r/base/stopifnot.html'>stopifnot</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/identical.html'>identical</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>counts</span><span class='op'>)</span>, <span class='va'>genes</span><span class='op'>$</span><span class='va'>ID</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>y</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/DGEList.html'>DGEList</a></span><span class='op'>(</span>counts <span class='op'>=</span> <span class='va'>counts</span>, genes <span class='op'>=</span> <span class='va'>genes</span><span class='op'>)</span>
<span class='va'>y</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>genotype</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/ifelse.html'>ifelse</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/grep.html'>grepl</a></span><span class='op'>(</span><span class='st'>"het"</span>, <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>)</span><span class='op'>)</span>, <span class='st'>"Het"</span>, <span class='st'>"Del"</span><span class='op'>)</span>
<span class='va'>y</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>mouse</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/substr.html'>substr</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>)</span>, <span class='fl'>7</span>, <span class='fl'>10</span><span class='op'>)</span>
<span class='va'>y</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span> <span class='op'>&lt;-</span> <span class='va'>y</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>genotype</span>
<span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/grep.html'>sub</a></span><span class='op'>(</span><span class='st'>"\\_trim\\.bam"</span>, <span class='st'>""</span>, <span class='fu'><a href='https://rdrr.io/r/base/grep.html'>sub</a></span><span class='op'>(</span><span class='st'>"Bcell\\_"</span>, <span class='st'>""</span>, <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>zero_counts</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colSums.html'>rowSums</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>$</span><span class='va'>counts</span><span class='op'>)</span> <span class='op'>==</span> <span class='fl'>0</span>
</code></pre>
</div>
</details>
</div>
<p>We exclude 7,932 genes with zero counts in all 8 samples.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>y</span> <span class='op'>&lt;-</span> <span class='va'>y</span><span class='op'>[</span><span class='op'>!</span><span class='va'>zero_counts</span>, <span class='op'>]</span>
</code></pre>
</div>
</details>
</div>
<h2 id="incorporating-gene-based-annotation">Incorporating gene-based annotation</h2>
<p>We obtain gene-based annotations from the NCBI/RefSeq and Ensembl databases, such as the chromosome and gene symbol, using the <em><a href="https://bioconductor.org/packages/3.12/Mus.musculus">Mus.musculus</a></em> and <em><a href="https://bioconductor.org/packages/3.12/EnsDb.Mmusculus.v79">EnsDb.Mmusculus.v79</a></em> packages.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='co'># Extract rownames (Ensembl IDs) to use as key in database lookups.</span>
<span class='va'>ensembl</span> <span class='op'>&lt;-</span> <span class='va'>y</span><span class='op'>$</span><span class='va'>genes</span><span class='op'>$</span><span class='va'>ID</span>

<span class='co'># Pull out useful gene-based annotations from the Ensembl-based database.</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'>EnsDb.Mmusculus.v79</span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/jorainer/ensembldb'>ensembldb</a></span><span class='op'>)</span>
<span class='co'># NOTE: These columns were customised for this project.</span>
<span class='va'>ensdb_columns</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span>
  <span class='st'>"GENEBIOTYPE"</span>, <span class='st'>"GENENAME"</span>, <span class='st'>"GENESEQSTART"</span>, <span class='st'>"GENESEQEND"</span>, <span class='st'>"SEQNAME"</span>, <span class='st'>"SYMBOL"</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>ensdb_columns</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='st'>"ENSEMBL."</span>, <span class='va'>ensdb_columns</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/stopifnot.html'>stopifnot</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/all.html'>all</a></span><span class='op'>(</span><span class='va'>ensdb_columns</span> <span class='op'>%in%</span> <span class='fu'>columns</span><span class='op'>(</span><span class='va'>EnsDb.Mmusculus.v79</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>ensdb_df</span> <span class='op'>&lt;-</span> <span class='fu'>DataFrame</span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/base/lapply.html'>lapply</a></span><span class='op'>(</span><span class='va'>ensdb_columns</span>, <span class='kw'>function</span><span class='op'>(</span><span class='va'>column</span><span class='op'>)</span> <span class='op'>{</span>
    <span class='fu'>mapIds</span><span class='op'>(</span>
      x <span class='op'>=</span> <span class='va'>EnsDb.Mmusculus.v79</span>, 
      <span class='co'># NOTE: Need to remove gene version number prior to lookup.</span>
      keys <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/grep.html'>gsub</a></span><span class='op'>(</span><span class='st'>"\\.[0-9]+$"</span>, <span class='st'>""</span>, <span class='va'>ensembl</span><span class='op'>)</span>,
      keytype <span class='op'>=</span> <span class='st'>"GENEID"</span>,
      column <span class='op'>=</span> <span class='va'>column</span>,
      multiVals <span class='op'>=</span> <span class='st'>"CharacterList"</span><span class='op'>)</span>
  <span class='op'>}</span><span class='op'>)</span>,
  row.names <span class='op'>=</span> <span class='va'>ensembl</span><span class='op'>)</span>
<span class='co'># NOTE: Can't look up GENEID column with GENEID key, so have to add manually.</span>
<span class='va'>ensdb_df</span><span class='op'>$</span><span class='va'>ENSEMBL.GENEID</span> <span class='op'>&lt;-</span> <span class='va'>ensembl</span>

<span class='co'># NOTE: Mus.musculus combines org.Mm.eg.db and</span>
<span class='co'>#       TxDb.Mmusculus.UCSC.mm10.knownGene (as well as others) and therefore </span>
<span class='co'>#       uses entrez gene and RefSeq based data.</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'>Mus.musculus</span><span class='op'>)</span>
<span class='co'># NOTE: These columns were customised for this project.</span>
<span class='va'>ncbi_columns</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span>
  <span class='co'># From TxDB: None required</span>
  <span class='co'># From OrgDB</span>
  <span class='st'>"ALIAS"</span>, <span class='st'>"ENTREZID"</span>, <span class='st'>"GENENAME"</span>, <span class='st'>"REFSEQ"</span>, <span class='st'>"SYMBOL"</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>ncbi_columns</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='st'>"NCBI."</span>, <span class='va'>ncbi_columns</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/stopifnot.html'>stopifnot</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/all.html'>all</a></span><span class='op'>(</span><span class='va'>ncbi_columns</span> <span class='op'>%in%</span> <span class='fu'>columns</span><span class='op'>(</span><span class='va'>Mus.musculus</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>ncbi_df</span> <span class='op'>&lt;-</span> <span class='fu'>DataFrame</span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/base/lapply.html'>lapply</a></span><span class='op'>(</span><span class='va'>ncbi_columns</span>, <span class='kw'>function</span><span class='op'>(</span><span class='va'>column</span><span class='op'>)</span> <span class='op'>{</span>
    <span class='fu'>mapIds</span><span class='op'>(</span>
      x <span class='op'>=</span> <span class='va'>Mus.musculus</span>, 
      <span class='co'># NOTE: Need to remove gene version number prior to lookup.</span>
      keys <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/grep.html'>gsub</a></span><span class='op'>(</span><span class='st'>"\\.[0-9]+$"</span>, <span class='st'>""</span>, <span class='va'>ensembl</span><span class='op'>)</span>,
      keytype <span class='op'>=</span> <span class='st'>"ENSEMBL"</span>,
      column <span class='op'>=</span> <span class='va'>column</span>,
      multiVals <span class='op'>=</span> <span class='st'>"CharacterList"</span><span class='op'>)</span>
  <span class='op'>}</span><span class='op'>)</span>,
  row.names <span class='op'>=</span> <span class='va'>ensembl</span><span class='op'>)</span>

<span class='va'>y</span><span class='op'>$</span><span class='va'>genes</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cbind.html'>cbind</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>$</span><span class='va'>genes</span>, <span class='fu'>flattenDF</span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cbind.html'>cbind</a></span><span class='op'>(</span><span class='va'>ensdb_df</span>, <span class='va'>ncbi_df</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>Having quantified gene expression against the Ensembl gene annotation, we have Ensembl-style identifiers for the genes. These identifiers are used as they are unambiguous and highly stable. However, they are difficult to interpret compared to the gene symbols which are more commonly used in the literature. Henceforth, we will use gene symbols (where available) to refer to genes in our analysis and otherwise use the Ensembl-style gene identifiers<a href="#fn2" class="footnote-ref" id="fnref2" role="doc-noteref"><sup>2</sup></a>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='co'># Replace the row names of the SCE by the gene symbols (where available).</span>
<span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'>scuttle</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/scuttle/man/uniquifyFeatureNames.html'>uniquifyFeatureNames</a></span><span class='op'>(</span>
  ID <span class='op'>=</span> <span class='va'>ensembl</span>, 
  <span class='co'># NOTE: An Ensembl ID may map to 0, 1, 2, 3, ... gene symbols.</span>
  <span class='co'>#       When there are multiple matches only the 1st match is used.</span>
  names <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/lapply.html'>sapply</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/strsplit.html'>strsplit</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>$</span><span class='va'>genes</span><span class='op'>$</span><span class='va'>ENSEMBL.SYMBOL</span>, <span class='st'>"; "</span><span class='op'>)</span>, <span class='kw'>function</span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span> <span class='op'>{</span>
    <span class='kw'>if</span> <span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>{</span>
      <span class='va'>x</span><span class='op'>[[</span><span class='fl'>1</span><span class='op'>]</span><span class='op'>]</span>
    <span class='op'>}</span> <span class='kw'>else</span> <span class='op'>{</span>
      <span class='cn'>NA_character_</span>
    <span class='op'>}</span>
  <span class='op'>}</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='co'># Some useful gene sets</span>
<span class='va'>mito_set</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>)</span><span class='op'>[</span><span class='va'>y</span><span class='op'>$</span><span class='va'>genes</span><span class='op'>$</span><span class='va'>ENSEMBL.SEQNAME</span> <span class='op'>==</span> <span class='st'>"MT"</span><span class='op'>]</span>
<span class='va'>ribo_set</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/grep.html'>grep</a></span><span class='op'>(</span><span class='st'>"^Rp(s|l)"</span>, <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>)</span>, value <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
<span class='co'># NOTE: A more curated approach for identifying ribosomal protein genes </span>
<span class='co'>#       (https://github.com/Bioconductor/OrchestratingSingleCellAnalysis-base/blob/ae201bf26e3e4fa82d9165d8abf4f4dc4b8e5a68/feature-selection.Rmd#L376-L380)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://igordot.github.io/msigdbr'>msigdbr</a></span><span class='op'>)</span>
<span class='va'>c2_sets</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://igordot.github.io/msigdbr/reference/msigdbr.html'>msigdbr</a></span><span class='op'>(</span>species <span class='op'>=</span> <span class='st'>"Mus musculus"</span>, category <span class='op'>=</span> <span class='st'>"C2"</span><span class='op'>)</span>
<span class='va'>ribo_set</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/sets.html'>union</a></span><span class='op'>(</span>
  <span class='va'>ribo_set</span>,
  <span class='va'>c2_sets</span><span class='op'>[</span><span class='va'>c2_sets</span><span class='op'>$</span><span class='va'>gs_name</span> <span class='op'>==</span> <span class='st'>"KEGG_RIBOSOME"</span>, <span class='op'>]</span><span class='op'>$</span><span class='va'>gene_symbol</span><span class='op'>)</span>
<span class='va'>sex_set</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>)</span><span class='op'>[</span><span class='va'>y</span><span class='op'>$</span><span class='va'>genes</span><span class='op'>$</span><span class='va'>ENSEMBL.SEQNAME</span> <span class='op'>%in%</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"X"</span>, <span class='st'>"Y"</span><span class='op'>)</span><span class='op'>]</span>
<span class='va'>pseudogene_set</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>)</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/grep.html'>grepl</a></span><span class='op'>(</span><span class='st'>"pseudogene"</span>, <span class='va'>y</span><span class='op'>$</span><span class='va'>genes</span><span class='op'>$</span><span class='va'>ENSEMBL.GENEBIOTYPE</span><span class='op'>)</span><span class='op'>]</span>
</code></pre>
</div>
</details>
</div>
<h2 id="gene-filtering">Gene filtering</h2>
<p>We exclude genes with insufficient counts for DE testing. Shown below are the number of genes that have sufficient counts to test for DE (<code>TRUE</code>) or not (<code>FALSE</code>).</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>y_keep_exprs</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/filterByExpr.html'>filterByExpr</a></span><span class='op'>(</span><span class='va'>y</span>, group <span class='op'>=</span> <span class='va'>y</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span><span class='op'>)</span>
<span class='va'>y</span> <span class='op'>&lt;-</span> <span class='va'>y</span><span class='op'>[</span><span class='va'>y_keep_exprs</span>, , keep.lib.sizes <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>]</span>
<span class='fu'><a href='https://rdrr.io/r/base/table.html'>table</a></span><span class='op'>(</span><span class='va'>y_keep_exprs</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<pre><code>y_keep_exprs
FALSE  TRUE 
 4600 13409 </code></pre>
</div>
<h2 id="normalization">Normalization</h2>
<p>Normalization with upper-quartile (UQ) normalization.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>y</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/calcNormFactors.html'>calcNormFactors</a></span><span class='op'>(</span><span class='va'>y</span>, method <span class='op'>=</span> <span class='st'>"upperquartile"</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<h1 id="analysis-of-male-b-cell-samples">Analysis of male B cell samples</h1>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>y_design</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/model.matrix.html'>model.matrix</a></span><span class='op'>(</span><span class='op'>~</span><span class='fl'>0</span> <span class='op'>+</span> <span class='va'>genotype</span>, <span class='va'>y</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>y_design</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/grep.html'>sub</a></span><span class='op'>(</span><span class='st'>"genotype"</span>, <span class='st'>""</span>, <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>y_design</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<h2 id="mds">MDS</h2>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'>plotMDS</span><span class='op'>(</span><span class='va'>y</span>, col <span class='op'>=</span> <span class='va'>genotype_colours</span><span class='op'>[</span><span class='va'>y</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>genotype</span><span class='op'>]</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:unnamed-chunk-17"></span>
<img src="C086_Kinkel.male_B_cells.multi-sample_comparisons_files/figure-html5/unnamed-chunk-17-1.png" alt="MDS plot coloured by `genotype`." width="624" />
<p class="caption">
Figure 1: MDS plot coloured by <code>genotype</code>.
</p>
</div>
</div>
<h2 id="de-analysis">DE analysis</h2>
<p>Fit a model with a term for <code>genotype</code> using the quasi-likelihood pipeline from <em><a href="https://bioconductor.org/packages/3.12/edgeR">edgeR</a></em>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>y</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/estimateDisp.html'>estimateDisp</a></span><span class='op'>(</span><span class='va'>y</span>, <span class='va'>y_design</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/graphics/par.html'>par</a></span><span class='op'>(</span>mfrow <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>2</span><span class='op'>)</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/plotBCV.html'>plotBCV</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>)</span>
<span class='va'>y_dgeglm</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/glmQLFTest.html'>glmQLFit</a></span><span class='op'>(</span><span class='va'>y</span>, <span class='va'>y_design</span>, robust <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/plotQLDisp.html'>plotQLDisp</a></span><span class='op'>(</span><span class='va'>y_dgeglm</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>y_contrast</span> <span class='op'>&lt;-</span> <span class='fu'>makeContrasts</span><span class='op'>(</span><span class='va'>Del</span> <span class='op'>-</span> <span class='va'>Het</span>, levels <span class='op'>=</span> <span class='va'>y_design</span><span class='op'>)</span>
<span class='va'>y_dgelrt</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/glmQLFTest.html'>glmQLFTest</a></span><span class='op'>(</span><span class='va'>y_dgeglm</span>, contrast <span class='op'>=</span> <span class='va'>y_contrast</span><span class='op'>)</span>
<span class='va'>y_dt</span> <span class='op'>&lt;-</span> <span class='fu'>decideTests</span><span class='op'>(</span><span class='va'>y_dgelrt</span><span class='op'>)</span>

<span class='fu'><a href='https://rdrr.io/r/base/summary.html'>summary</a></span><span class='op'>(</span><span class='va'>y_dt</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span>caption <span class='op'>=</span> <span class='st'>"Number of DEGs (FDR &lt; 0.05)"</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<table>
<caption><span id="tab:unnamed-chunk-19">Table 1: </span>Number of DEGs (FDR &lt; 0.05)</caption>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">1<em>Del -1</em>Het</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Down</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">NotSig</td>
<td style="text-align: right;">13393</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Up</td>
<td style="text-align: right;">15</td>
</tr>
</tbody>
</table>
</div>
<p>CSV file of results available from <a href="../output/DEGs/male_B_cells/"><code>output/DEGs/male_B_cells/</code></a>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>gzout</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/connections.html'>gzfile</a></span><span class='op'>(</span>
  description <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span><span class='va'>outdir</span>, <span class='st'>"male_B_cells.DEGs.csv.gz"</span><span class='op'>)</span>,
  open <span class='op'>=</span> <span class='st'>"wb"</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/utils/write.table.html'>write.csv</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/topTags.html'>topTags</a></span><span class='op'>(</span><span class='va'>y_dgelrt</span>, n <span class='op'>=</span> <span class='cn'>Inf</span><span class='op'>)</span>,
  <span class='va'>gzout</span>,
  <span class='co'># NOTE: quote = TRUE needed because some fields contain commas.</span>
  quote <span class='op'>=</span> <span class='cn'>TRUE</span>,
  row.names <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/connections.html'>close</a></span><span class='op'>(</span><span class='va'>gzout</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/r/graphics/par.html'>par</a></span><span class='op'>(</span>mfrow <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span>
<span class='fu'>plotMD</span><span class='op'>(</span><span class='va'>y_dgelrt</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:unnamed-chunk-21"></span>
<img src="C086_Kinkel.male_B_cells.multi-sample_comparisons_files/figure-html5/unnamed-chunk-21-1.png" alt="MD plot highlighting DEGs (red)." width="624" />
<p class="caption">
Figure 2: MD plot highlighting DEGs (red).
</p>
</div>
</div>
<p>An interactive <em><a href="https://bioconductor.org/packages/3.12/Glimma">Glimma</a></em> MD plot of the differential expression results is available from <a href="../output/DEGs/male_B_cells/glimma-plots/male_B_cells.md-plot.html"><code>output/DEGs/male_B_cells/glimma-plots/male_B_cells.md-plot.html</code></a>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/pkg/Glimma/man/glMDPlot.html'>glMDPlot</a></span><span class='op'>(</span>
  x <span class='op'>=</span> <span class='va'>y_dgelrt</span>,
  counts <span class='op'>=</span> <span class='va'>y</span>,
  anno <span class='op'>=</span> <span class='va'>y</span><span class='op'>$</span><span class='va'>genes</span>,
  display.columns <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"ENSEMBL.SYMBOL"</span><span class='op'>)</span>,
  groups <span class='op'>=</span> <span class='va'>y</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>group</span>,
  samples <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>)</span>,
  status <span class='op'>=</span> <span class='va'>y_dt</span>,
  transform <span class='op'>=</span> <span class='cn'>TRUE</span>,
  sample.cols <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/integer.html'>as.integer</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>$</span><span class='va'>samples</span><span class='op'>$</span><span class='va'>genotype</span><span class='op'>)</span><span class='op'>)</span>,
  path <span class='op'>=</span> <span class='va'>outdir</span>,
  html <span class='op'>=</span> <span class='st'>"male_B_cells.md-plot"</span>,
  launch <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<h2 id="gene-set-tests">Gene set tests</h2>
<p>Gene set testing is commonly used to interpret the differential expression results in a biological context. Here we apply various gene set tests to the DEG lists.</p>
<h3 id="goana">goana</h3>
<p>We use the <code>goana()</code> function from the <em><a href="https://bioconductor.org/packages/3.12/limma">limma</a></em> R/Bioconductor package to test for over-representation of gene ontology (GO) terms in each DEG list.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>list_of_go</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/lapply.html'>lapply</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>y_contrast</span><span class='op'>)</span>, <span class='kw'>function</span><span class='op'>(</span><span class='va'>j</span><span class='op'>)</span> <span class='op'>{</span>
  <span class='va'>dgelrt</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/glmfit.html'>glmLRT</a></span><span class='op'>(</span><span class='va'>y_dgeglm</span>, contrast <span class='op'>=</span> <span class='va'>y_contrast</span><span class='op'>[</span>, <span class='va'>j</span><span class='op'>]</span><span class='op'>)</span>
  
  <span class='va'>go</span> <span class='op'>&lt;-</span> <span class='fu'>goana</span><span class='op'>(</span>
    de <span class='op'>=</span> <span class='va'>dgelrt</span>,
    geneid <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/lapply.html'>sapply</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/strsplit.html'>strsplit</a></span><span class='op'>(</span><span class='va'>dgelrt</span><span class='op'>$</span><span class='va'>genes</span><span class='op'>$</span><span class='va'>NCBI.ENTREZID</span>, <span class='st'>"; "</span><span class='op'>)</span>, <span class='st'>"[["</span>, <span class='fl'>1</span><span class='op'>)</span>,
    species <span class='op'>=</span> <span class='st'>"Mm"</span><span class='op'>)</span>
  <span class='va'>gzout</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/connections.html'>gzfile</a></span><span class='op'>(</span>
    description <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span><span class='va'>outdir</span>, <span class='st'>"male_B_cells.goana.csv.gz"</span><span class='op'>)</span>,
    open <span class='op'>=</span> <span class='st'>"wb"</span><span class='op'>)</span>
  <span class='fu'><a href='https://rdrr.io/r/utils/write.table.html'>write.csv</a></span><span class='op'>(</span>
    <span class='fu'>topGO</span><span class='op'>(</span><span class='va'>go</span>, number <span class='op'>=</span> <span class='cn'>Inf</span><span class='op'>)</span>,
    <span class='va'>gzout</span>,
    row.names <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
  <span class='fu'><a href='https://rdrr.io/r/base/connections.html'>close</a></span><span class='op'>(</span><span class='va'>gzout</span><span class='op'>)</span>
  <span class='va'>go</span>
<span class='op'>}</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>list_of_go</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>y_contrast</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>CSV files of the results are available from <a href="../output/DEGs/male_B_cells/"><code>output/DEGs/male_B_cells/</code></a>.</p>
<p>An example of the results are shown below.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'>topGO</span><span class='op'>(</span><span class='va'>list_of_go</span><span class='op'>[[</span><span class='fl'>1</span><span class='op'>]</span><span class='op'>]</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span>
    caption <span class='op'>=</span> <span class='st'>'`goana()` produces a table with a row for each GO term and the following columns: **Term** (GO term); **Ont** (ontology that the GO term belongs to. Possible values are "BP", "CC" and "MF"); **N** (number of genes in the GO term); **Up** (number of up-regulated differentially expressed genes); **Down** (number of down-regulated differentially expressed genes); **P.Up** (p-value for over-representation of GO term in up-regulated genes); **P.Down** (p-value for over-representation of GO term in down-regulated genes)'</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<table>
<caption><span id="tab:unnamed-chunk-24">Table 2: </span><code>goana()</code> produces a table with a row for each GO term and the following columns: <strong>Term</strong> (GO term); <strong>Ont</strong> (ontology that the GO term belongs to. Possible values are BP, CC and MF); <strong>N</strong> (number of genes in the GO term); <strong>Up</strong> (number of up-regulated differentially expressed genes); <strong>Down</strong> (number of down-regulated differentially expressed genes); <strong>P.Up</strong> (p-value for over-representation of GO term in up-regulated genes); <strong>P.Down</strong> (p-value for over-representation of GO term in down-regulated genes)</caption>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;">Term</th>
<th style="text-align: left;">Ont</th>
<th style="text-align: right;">N</th>
<th style="text-align: right;">Up</th>
<th style="text-align: right;">Down</th>
<th style="text-align: right;">P.Up</th>
<th style="text-align: right;">P.Down</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"><a href="GO:0031720" class="uri">GO:0031720</a></td>
<td style="text-align: left;">haptoglobin binding</td>
<td style="text-align: left;">MF</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0000000</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="GO:0005833" class="uri">GO:0005833</a></td>
<td style="text-align: left;">hemoglobin complex</td>
<td style="text-align: left;">CC</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0000000</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="GO:0048731" class="uri">GO:0048731</a></td>
<td style="text-align: left;">system development</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">2947</td>
<td style="text-align: right;">31</td>
<td style="text-align: right;">31</td>
<td style="text-align: right;">0.0131091</td>
<td style="text-align: right;">0.0000000</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="GO:0030426" class="uri">GO:0030426</a></td>
<td style="text-align: left;">growth cone</td>
<td style="text-align: left;">CC</td>
<td style="text-align: right;">164</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">0.6976242</td>
<td style="text-align: right;">0.0000000</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="GO:0030427" class="uri">GO:0030427</a></td>
<td style="text-align: left;">site of polarized growth</td>
<td style="text-align: left;">CC</td>
<td style="text-align: right;">170</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">0.7106582</td>
<td style="text-align: right;">0.0000000</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="GO:0031838" class="uri">GO:0031838</a></td>
<td style="text-align: left;">haptoglobin-hemoglobin complex</td>
<td style="text-align: left;">CC</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">0.0355835</td>
<td style="text-align: right;">0.0000000</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="GO:0007275" class="uri">GO:0007275</a></td>
<td style="text-align: left;">multicellular organism development</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">3301</td>
<td style="text-align: right;">34</td>
<td style="text-align: right;">31</td>
<td style="text-align: right;">0.0121146</td>
<td style="text-align: right;">0.0000000</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="GO:0048821" class="uri">GO:0048821</a></td>
<td style="text-align: left;">erythrocyte development</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">38</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0000000</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="GO:0048468" class="uri">GO:0048468</a></td>
<td style="text-align: left;">cell development</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">1486</td>
<td style="text-align: right;">16</td>
<td style="text-align: right;">21</td>
<td style="text-align: right;">0.0652232</td>
<td style="text-align: right;">0.0000000</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="GO:0007399" class="uri">GO:0007399</a></td>
<td style="text-align: left;">nervous system development</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">1490</td>
<td style="text-align: right;">17</td>
<td style="text-align: right;">21</td>
<td style="text-align: right;">0.0364729</td>
<td style="text-align: right;">0.0000000</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="GO:0019825" class="uri">GO:0019825</a></td>
<td style="text-align: left;">oxygen binding</td>
<td style="text-align: left;">MF</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0000000</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="GO:0150034" class="uri">GO:0150034</a></td>
<td style="text-align: left;">distal axon</td>
<td style="text-align: left;">CC</td>
<td style="text-align: right;">247</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">0.5353054</td>
<td style="text-align: right;">0.0000000</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="GO:0044297" class="uri">GO:0044297</a></td>
<td style="text-align: left;">cell body</td>
<td style="text-align: left;">CC</td>
<td style="text-align: right;">522</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">13</td>
<td style="text-align: right;">0.7331790</td>
<td style="text-align: right;">0.0000000</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="GO:0048856" class="uri">GO:0048856</a></td>
<td style="text-align: left;">anatomical structure development</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">3597</td>
<td style="text-align: right;">36</td>
<td style="text-align: right;">31</td>
<td style="text-align: right;">0.0148414</td>
<td style="text-align: right;">0.0000000</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="GO:0032502" class="uri">GO:0032502</a></td>
<td style="text-align: left;">developmental process</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">3893</td>
<td style="text-align: right;">36</td>
<td style="text-align: right;">32</td>
<td style="text-align: right;">0.0477809</td>
<td style="text-align: right;">0.0000000</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="GO:0005344" class="uri">GO:0005344</a></td>
<td style="text-align: left;">oxygen carrier activity</td>
<td style="text-align: left;">MF</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0000000</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="GO:0048858" class="uri">GO:0048858</a></td>
<td style="text-align: left;">cell projection morphogenesis</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">453</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">12</td>
<td style="text-align: right;">0.0164555</td>
<td style="text-align: right;">0.0000000</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="GO:0032501" class="uri">GO:0032501</a></td>
<td style="text-align: left;">multicellular organismal process</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">4220</td>
<td style="text-align: right;">42</td>
<td style="text-align: right;">33</td>
<td style="text-align: right;">0.0077364</td>
<td style="text-align: right;">0.0000001</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="GO:0001944" class="uri">GO:0001944</a></td>
<td style="text-align: left;">vasculature development</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">494</td>
<td style="text-align: right;">17</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">0.0000001</td>
<td style="text-align: right;">0.2635084</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="GO:0032990" class="uri">GO:0032990</a></td>
<td style="text-align: left;">cell part morphogenesis</td>
<td style="text-align: left;">BP</td>
<td style="text-align: right;">474</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">12</td>
<td style="text-align: right;">0.0209935</td>
<td style="text-align: right;">0.0000001</td>
</tr>
</tbody>
</table>
</div>
<h3 id="kegga">kegga</h3>
<p>We use the <code>kegga()</code> function from the <em><a href="https://bioconductor.org/packages/3.12/limma">limma</a></em> R/Bioconductor package to test for over-representation of KEGG pathways in each DEG list.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>list_of_kegg</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/lapply.html'>lapply</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>y_contrast</span><span class='op'>)</span>, <span class='kw'>function</span><span class='op'>(</span><span class='va'>j</span><span class='op'>)</span> <span class='op'>{</span>
  <span class='va'>dgelrt</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/glmfit.html'>glmLRT</a></span><span class='op'>(</span><span class='va'>y_dgeglm</span>, contrast <span class='op'>=</span> <span class='va'>y_contrast</span><span class='op'>[</span>, <span class='va'>j</span><span class='op'>]</span><span class='op'>)</span>
  
  <span class='va'>kegg</span> <span class='op'>&lt;-</span> <span class='fu'>kegga</span><span class='op'>(</span>
    de <span class='op'>=</span> <span class='va'>dgelrt</span>,
    geneid <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/lapply.html'>sapply</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/strsplit.html'>strsplit</a></span><span class='op'>(</span><span class='va'>dgelrt</span><span class='op'>$</span><span class='va'>genes</span><span class='op'>$</span><span class='va'>NCBI.ENTREZID</span>, <span class='st'>"; "</span><span class='op'>)</span>, <span class='st'>"[["</span>, <span class='fl'>1</span><span class='op'>)</span>,
    species <span class='op'>=</span> <span class='st'>"Mm"</span><span class='op'>)</span>
  <span class='va'>gzout</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/connections.html'>gzfile</a></span><span class='op'>(</span>
    description <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span><span class='va'>outdir</span>, <span class='st'>"male_B_cells.kegga.csv.gz"</span><span class='op'>)</span>,
    open <span class='op'>=</span> <span class='st'>"wb"</span><span class='op'>)</span>
  <span class='fu'><a href='https://rdrr.io/r/utils/write.table.html'>write.csv</a></span><span class='op'>(</span>
    <span class='fu'>topKEGG</span><span class='op'>(</span><span class='va'>kegg</span>, number <span class='op'>=</span> <span class='cn'>Inf</span><span class='op'>)</span>,
    <span class='va'>gzout</span>,
    row.names <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
  <span class='fu'><a href='https://rdrr.io/r/base/connections.html'>close</a></span><span class='op'>(</span><span class='va'>gzout</span><span class='op'>)</span>
  <span class='va'>kegg</span>
<span class='op'>}</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>list_of_kegg</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>y_contrast</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>CSV files of the results are available from <a href="../output/DEGs/male_B_cells/"><code>output/DEGs/male_B_cells/</code></a>.</p>
<p>An example of the results are shown below.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'>topKEGG</span><span class='op'>(</span><span class='va'>list_of_kegg</span><span class='op'>[[</span><span class='fl'>1</span><span class='op'>]</span><span class='op'>]</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span>
    caption <span class='op'>=</span> <span class='st'>'`kegga()` produces a table with a row for each KEGG pathway ID and the following columns: **Pathway** (KEGG pathway); **N** (number of genes in the GO term); **Up** (number of up-regulated differentially expressed genes); **Down** (number of down-regulated differentially expressed genes); **P.Up** (p-value for over-representation of KEGG pathway in up-regulated genes); **P.Down** (p-value for over-representation of KEGG pathway in down-regulated genes)'</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<table style="width:100%;">
<caption><span id="tab:unnamed-chunk-26">Table 3: </span><code>kegga()</code> produces a table with a row for each KEGG pathway ID and the following columns: <strong>Pathway</strong> (KEGG pathway); <strong>N</strong> (number of genes in the GO term); <strong>Up</strong> (number of up-regulated differentially expressed genes); <strong>Down</strong> (number of down-regulated differentially expressed genes); <strong>P.Up</strong> (p-value for over-representation of KEGG pathway in up-regulated genes); <strong>P.Down</strong> (p-value for over-representation of KEGG pathway in down-regulated genes)</caption>
<colgroup>
<col style="width: 15%" />
<col style="width: 50%" />
<col style="width: 4%" />
<col style="width: 3%" />
<col style="width: 5%" />
<col style="width: 10%" />
<col style="width: 10%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;">Pathway</th>
<th style="text-align: right;">N</th>
<th style="text-align: right;">Up</th>
<th style="text-align: right;">Down</th>
<th style="text-align: right;">P.Up</th>
<th style="text-align: right;">P.Down</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">path:mmu05143</td>
<td style="text-align: left;">African trypanosomiasis</td>
<td style="text-align: right;">26</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">0.1735812</td>
<td style="text-align: right;">0.0000021</td>
</tr>
<tr class="even">
<td style="text-align: left;">path:mmu04668</td>
<td style="text-align: left;">TNF signaling pathway</td>
<td style="text-align: right;">92</td>
<td style="text-align: right;">7</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0000046</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="odd">
<td style="text-align: left;">path:mmu04380</td>
<td style="text-align: left;">Osteoclast differentiation</td>
<td style="text-align: right;">111</td>
<td style="text-align: right;">7</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0000160</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="even">
<td style="text-align: left;">path:mmu05144</td>
<td style="text-align: left;">Malaria</td>
<td style="text-align: right;">46</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">0.2865069</td>
<td style="text-align: right;">0.0000216</td>
</tr>
<tr class="odd">
<td style="text-align: left;">path:mmu04657</td>
<td style="text-align: left;">IL-17 signaling pathway</td>
<td style="text-align: right;">57</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0000576</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="even">
<td style="text-align: left;">path:mmu05418</td>
<td style="text-align: left;">Fluid shear stress and atherosclerosis</td>
<td style="text-align: right;">112</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0001647</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="odd">
<td style="text-align: left;">path:mmu01522</td>
<td style="text-align: left;">Endocrine resistance</td>
<td style="text-align: right;">78</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0002573</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="even">
<td style="text-align: left;">path:mmu04915</td>
<td style="text-align: left;">Estrogen signaling pathway</td>
<td style="text-align: right;">79</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0.0002731</td>
<td style="text-align: right;">0.2491464</td>
</tr>
<tr class="odd">
<td style="text-align: left;">path:mmu04935</td>
<td style="text-align: left;">Growth hormone synthesis, secretion and action</td>
<td style="text-align: right;">88</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0004509</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="even">
<td style="text-align: left;">path:mmu05133</td>
<td style="text-align: left;">Pertussis</td>
<td style="text-align: right;">54</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0006355</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="odd">
<td style="text-align: left;">path:mmu04713</td>
<td style="text-align: left;">Circadian entrainment</td>
<td style="text-align: right;">57</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0007807</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="even">
<td style="text-align: left;">path:mmu05224</td>
<td style="text-align: left;">Breast cancer</td>
<td style="text-align: right;">101</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0008464</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="odd">
<td style="text-align: left;">path:mmu05030</td>
<td style="text-align: left;">Cocaine addiction</td>
<td style="text-align: right;">30</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0013240</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="even">
<td style="text-align: left;">path:mmu05200</td>
<td style="text-align: left;">Pathways in cancer</td>
<td style="text-align: right;">376</td>
<td style="text-align: right;">9</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">0.0016617</td>
<td style="text-align: right;">0.3953707</td>
</tr>
<tr class="odd">
<td style="text-align: left;">path:mmu04024</td>
<td style="text-align: left;">cAMP signaling pathway</td>
<td style="text-align: right;">118</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0016969</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="even">
<td style="text-align: left;">path:mmu05135</td>
<td style="text-align: left;">Yersinia infection</td>
<td style="text-align: right;">123</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0020368</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="odd">
<td style="text-align: left;">path:mmu04620</td>
<td style="text-align: left;">Toll-like receptor signaling pathway</td>
<td style="text-align: right;">78</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0025097</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="even">
<td style="text-align: left;">path:mmu05142</td>
<td style="text-align: left;">Chagas disease</td>
<td style="text-align: right;">83</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0031451</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="odd">
<td style="text-align: left;">path:mmu04670</td>
<td style="text-align: left;">Leukocyte transendothelial migration</td>
<td style="text-align: right;">83</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0031451</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="even">
<td style="text-align: left;">path:mmu04926</td>
<td style="text-align: left;">Relaxin signaling pathway</td>
<td style="text-align: right;">86</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0035748</td>
<td style="text-align: right;">1.0000000</td>
</tr>
</tbody>
</table>
</div>
<h3 id="camera-with-msigdb-gene-sets">camera with MSigDB gene sets</h3>
<p>We use the <code>camera()</code> function from the <em><a href="https://bioconductor.org/packages/3.12/limma">limma</a></em> R/Bioconductor package to test whether a set of genes is highly ranked relative to other genes in terms of differential expression, accounting for inter-gene correlation. Specifically, we test using gene sets from <a href="https://www.gsea-msigdb.org/gsea/msigdb">MSigDB</a>, namely:</p>
<aside>
We download these gene sets from <a href="http://bioinf.wehi.edu.au/MSigDB/index.html">http://bioinf.wehi.edu.au/MSigDB/index.html</a>
</aside>
<ul>
<li><strong>H</strong>: <strong>hallmark gene sets</strong> are coherently expressed signatures derived by aggregating many MSigDB gene sets to represent well-defined biological states or processes.</li>
<li><strong>C2</strong>: <strong>curated gene sets</strong> from online pathway databases, publications in PubMed, and knowledge of domain experts.</li>
</ul>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='co'># NOTE: Using BiocFileCache to avoid re-downloading these gene sets everytime </span>
<span class='co'>#       the report is rendered.</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'>BiocFileCache</span><span class='op'>)</span>
<span class='va'>bfc</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html'>BiocFileCache</a></span><span class='op'>(</span><span class='op'>)</span>
<span class='co'># NOTE: Creating list of gene sets in this slightly convoluted way so that each </span>
<span class='co'>#       gene set name is prepended by its origin (e.g. H, C2, or C7).</span>
<span class='va'>msigdb</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/do.call.html'>do.call</a></span><span class='op'>(</span>
  <span class='va'>c</span>, 
  <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>
    H <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/readRDS.html'>readRDS</a></span><span class='op'>(</span>
      <span class='fu'><a href='https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html'>bfcrpath</a></span><span class='op'>(</span>
        <span class='va'>bfc</span>,
        <span class='st'>"http://bioinf.wehi.edu.au/MSigDB/v7.1/Mm.h.all.v7.1.entrez.rds"</span><span class='op'>)</span><span class='op'>)</span>,
    C2 <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/readRDS.html'>readRDS</a></span><span class='op'>(</span>
      <span class='fu'><a href='https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html'>bfcrpath</a></span><span class='op'>(</span>
        <span class='va'>bfc</span>, 
        <span class='st'>"http://bioinf.wehi.edu.au/MSigDB/v7.1/Mm.c2.all.v7.1.entrez.rds"</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>

<span class='va'>y_idx</span> <span class='op'>&lt;-</span> <span class='fu'>ids2indices</span><span class='op'>(</span>
  <span class='va'>msigdb</span>, 
  id <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/lapply.html'>sapply</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/strsplit.html'>strsplit</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>$</span><span class='va'>genes</span><span class='op'>$</span><span class='va'>NCBI.ENTREZID</span>, <span class='st'>"; "</span><span class='op'>)</span>, <span class='st'>"[["</span>, <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>list_of_camera</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/lapply.html'>lapply</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>y_contrast</span><span class='op'>)</span>, <span class='kw'>function</span><span class='op'>(</span><span class='va'>j</span><span class='op'>)</span> <span class='op'>{</span>
  <span class='va'>cam</span> <span class='op'>&lt;-</span> <span class='fu'>camera</span><span class='op'>(</span>
    y <span class='op'>=</span> <span class='va'>y</span>, 
    index <span class='op'>=</span> <span class='va'>y_idx</span>,
    design <span class='op'>=</span> <span class='va'>y_design</span>, 
    contrast <span class='op'>=</span> <span class='va'>y_contrast</span><span class='op'>[</span>, <span class='va'>j</span><span class='op'>]</span><span class='op'>)</span>
  <span class='va'>gzout</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/connections.html'>gzfile</a></span><span class='op'>(</span>
    description <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span><span class='va'>outdir</span>, <span class='st'>"male_B_cells.camera.csv.gz"</span><span class='op'>)</span>,
    open <span class='op'>=</span> <span class='st'>"wb"</span><span class='op'>)</span>
  <span class='fu'><a href='https://rdrr.io/r/utils/write.table.html'>write.csv</a></span><span class='op'>(</span>
    <span class='va'>cam</span>,
    <span class='va'>gzout</span>,
    row.names <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
  <span class='fu'><a href='https://rdrr.io/r/base/connections.html'>close</a></span><span class='op'>(</span><span class='va'>gzout</span><span class='op'>)</span>
  <span class='va'>cam</span>
<span class='op'>}</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>list_of_camera</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>y_contrast</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>CSV files of the results are available from <a href="../output/DEGs/male_B_cells/"><code>output/DEGs/male_B_cells</code></a>.</p>
<p>An example of the results are shown below.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span><span class='op'>(</span><span class='va'>list_of_camera</span><span class='op'>[[</span><span class='fl'>1</span><span class='op'>]</span><span class='op'>]</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span>
    caption <span class='op'>=</span> <span class='st'>'`camera()` produces a table with a row for each gene set (prepended by which MSigDB collection it comes from) and the following columns: **NGenes** (number of genes in set); **Direction** (direction of change); **PValue** (two-tailed p-value); **FDR** (Benjamini and Hochberg FDR adjusted p-value).'</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<table>
<caption><span id="tab:unnamed-chunk-28">Table 4: </span><code>camera()</code> produces a table with a row for each gene set (prepended by which MSigDB collection it comes from) and the following columns: <strong>NGenes</strong> (number of genes in set); <strong>Direction</strong> (direction of change); <strong>PValue</strong> (two-tailed p-value); <strong>FDR</strong> (Benjamini and Hochberg FDR adjusted p-value).</caption>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">NGenes</th>
<th style="text-align: left;">Direction</th>
<th style="text-align: right;">PValue</th>
<th style="text-align: right;">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">C2.ZHOU_PANCREATIC_EXOCRINE_PROGENITOR</td>
<td style="text-align: right;">3</td>
<td style="text-align: left;">Up</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0e+00</td>
</tr>
<tr class="even">
<td style="text-align: left;">C2.MORI_MATURE_B_LYMPHOCYTE_DN</td>
<td style="text-align: right;">92</td>
<td style="text-align: left;">Down</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0e+00</td>
</tr>
<tr class="odd">
<td style="text-align: left;">C2.REACTOME_EUKARYOTIC_TRANSLATION_ELONGATION</td>
<td style="text-align: right;">108</td>
<td style="text-align: left;">Up</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0e+00</td>
</tr>
<tr class="even">
<td style="text-align: left;">C2.KEGG_RIBOSOME</td>
<td style="text-align: right;">103</td>
<td style="text-align: left;">Up</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0e+00</td>
</tr>
<tr class="odd">
<td style="text-align: left;">C2.LEE_EARLY_T_LYMPHOCYTE_UP</td>
<td style="text-align: right;">116</td>
<td style="text-align: left;">Down</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1e-07</td>
</tr>
<tr class="even">
<td style="text-align: left;">C2.REACTOME_RESPONSE_OF_EIF2AK4_GCN2_TO_AMINO_ACID_DEFICIENCY</td>
<td style="text-align: right;">117</td>
<td style="text-align: left;">Up</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2e-07</td>
</tr>
</tbody>
</table>
</div>
<h3 id="sarahs-gene-lists">Sarahs gene lists</h3>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>y_X</span> <span class='op'>&lt;-</span> <span class='va'>y</span><span class='op'>$</span><span class='va'>genes</span><span class='op'>$</span><span class='va'>ENSEMBL.SEQNAME</span> <span class='op'>==</span> <span class='st'>"X"</span>

<span class='va'>y_index</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>`X-linked` <span class='op'>=</span> <span class='va'>y_X</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>Sarah requested testing all X-linked genes as a gene set.</p>
<p>We use the <code>fry()</code> function from the <em><a href="https://bioconductor.org/packages/3.12/edgeR">edgeR</a></em> R/Bioconductor package to perform a self-contained gene set test against the null hypothesis that none of the genes in the set are differentially expressed.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'>fry</span><span class='op'>(</span>
  <span class='va'>y</span>,
  index <span class='op'>=</span> <span class='va'>y_index</span>,
  design <span class='op'>=</span> <span class='va'>y_design</span>,
  contrast <span class='op'>=</span> <span class='va'>y_contrast</span>, 
  sort <span class='op'>=</span> <span class='st'>"none"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
    <span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span>
      caption <span class='op'>=</span> <span class='st'>"Results of applying `fry()` to the RNA-seq differential expression results using the supplied gene sets. A significant 'Up' P-value means that the differential expression results found in the RNA-seq data are positively correlated with the expression signature from the corresponding gene set. Conversely, a significant 'Down' P-value means that the differential expression log-fold-changes are negatively correlated with the expression signature from the corresponding gene set. A significant 'Mixed' P-value means that the genes in the set tend to be differentially expressed without regard for direction."</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<table>
<caption><span id="tab:unnamed-chunk-30">Table 5: </span>Results of applying <code>fry()</code> to the RNA-seq differential expression results using the supplied gene sets. A significant Up P-value means that the differential expression results found in the RNA-seq data are positively correlated with the expression signature from the corresponding gene set. Conversely, a significant Down P-value means that the differential expression log-fold-changes are negatively correlated with the expression signature from the corresponding gene set. A significant Mixed P-value means that the genes in the set tend to be differentially expressed without regard for direction.</caption>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">NGenes</th>
<th style="text-align: left;">Direction</th>
<th style="text-align: right;">PValue</th>
<th style="text-align: right;">PValue.Mixed</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">X-linked</td>
<td style="text-align: right;">416</td>
<td style="text-align: left;">Down</td>
<td style="text-align: right;">0.9861314</td>
<td style="text-align: right;">0.6496793</td>
</tr>
</tbody>
</table>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/r/graphics/par.html'>par</a></span><span class='op'>(</span>mfrow <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span><span class='op'>(</span><span class='va'>y_index</span><span class='op'>)</span>, <span class='fl'>2</span><span class='op'>)</span><span class='op'>)</span>
<span class='kw'>for</span> <span class='op'>(</span><span class='va'>n</span> <span class='kw'>in</span> <span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>y_index</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>{</span>
  <span class='kw'>if</span> <span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>is.data.frame</a></span><span class='op'>(</span><span class='va'>y_index</span><span class='op'>[[</span><span class='va'>n</span><span class='op'>]</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>{</span>
    <span class='va'>y_status</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/rep.html'>rep</a></span><span class='op'>(</span><span class='fl'>0</span>, <span class='fu'><a href='https://rdrr.io/r/base/nrow.html'>nrow</a></span><span class='op'>(</span><span class='va'>y_dgelrt</span><span class='op'>)</span><span class='op'>)</span>
    <span class='va'>y_status</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>)</span> <span class='op'>%in%</span> <span class='va'>y_index</span><span class='op'>[[</span><span class='va'>n</span><span class='op'>]</span><span class='op'>]</span><span class='op'>[</span>, <span class='fl'>1</span><span class='op'>]</span><span class='op'>]</span> <span class='op'>&lt;-</span>
      <span class='fu'><a href='https://rdrr.io/r/base/sign.html'>sign</a></span><span class='op'>(</span><span class='va'>y_index</span><span class='op'>[[</span><span class='va'>n</span><span class='op'>]</span><span class='op'>]</span><span class='op'>[</span><span class='va'>y_index</span><span class='op'>[[</span><span class='va'>n</span><span class='op'>]</span><span class='op'>]</span><span class='op'>[</span>, <span class='fl'>1</span><span class='op'>]</span> <span class='op'>%in%</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>)</span>, <span class='fl'>2</span><span class='op'>]</span><span class='op'>)</span>
    <span class='fu'>plotMD</span><span class='op'>(</span>
      <span class='va'>y_dgelrt</span>,
      status <span class='op'>=</span> <span class='va'>y_status</span>,
      hl.col <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"red"</span>, <span class='st'>"blue"</span><span class='op'>)</span>,
      legend <span class='op'>=</span> <span class='st'>"bottomright"</span>,
      main <span class='op'>=</span> <span class='va'>n</span><span class='op'>)</span>
    <span class='fu'><a href='https://rdrr.io/r/graphics/abline.html'>abline</a></span><span class='op'>(</span>h <span class='op'>=</span> <span class='fl'>0</span>, col <span class='op'>=</span> <span class='st'>"darkgrey"</span><span class='op'>)</span>
  <span class='op'>}</span> <span class='kw'>else</span> <span class='op'>{</span>
    <span class='va'>y_status</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/rep.html'>rep</a></span><span class='op'>(</span><span class='st'>"Other"</span>, <span class='fu'><a href='https://rdrr.io/r/base/nrow.html'>nrow</a></span><span class='op'>(</span><span class='va'>y_dgelrt</span><span class='op'>)</span><span class='op'>)</span>
    <span class='va'>y_status</span><span class='op'>[</span><span class='va'>y_index</span><span class='op'>[[</span><span class='va'>n</span><span class='op'>]</span><span class='op'>]</span><span class='op'>]</span> <span class='op'>&lt;-</span> <span class='va'>n</span>
    <span class='fu'>plotMD</span><span class='op'>(</span>
      <span class='va'>y_dgelrt</span>,
      status <span class='op'>=</span> <span class='va'>y_status</span>,
      values <span class='op'>=</span> <span class='va'>n</span>,
      hl.col <span class='op'>=</span> <span class='st'>"red"</span>,
      legend <span class='op'>=</span> <span class='st'>"topright"</span>,
      main <span class='op'>=</span> <span class='va'>n</span><span class='op'>)</span>
    <span class='fu'><a href='https://rdrr.io/r/graphics/abline.html'>abline</a></span><span class='op'>(</span>h <span class='op'>=</span> <span class='fl'>0</span>, col <span class='op'>=</span> <span class='st'>"darkgrey"</span><span class='op'>)</span>
  <span class='op'>}</span>
  
  <span class='kw'>if</span> <span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>is.data.frame</a></span><span class='op'>(</span><span class='va'>y_index</span><span class='op'>[[</span><span class='va'>n</span><span class='op'>]</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>{</span>
    <span class='fu'>barcodeplot</span><span class='op'>(</span>
      <span class='va'>y_dgelrt</span><span class='op'>$</span><span class='va'>table</span><span class='op'>$</span><span class='va'>logFC</span>,
      <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>)</span> <span class='op'>%in%</span> <span class='va'>y_index</span><span class='op'>[[</span><span class='va'>n</span><span class='op'>]</span><span class='op'>]</span><span class='op'>[</span>, <span class='fl'>1</span><span class='op'>]</span>,
      gene.weights <span class='op'>=</span> <span class='va'>y_index</span><span class='op'>[[</span><span class='va'>n</span><span class='op'>]</span><span class='op'>]</span><span class='op'>[</span>
        <span class='va'>y_index</span><span class='op'>[[</span><span class='va'>n</span><span class='op'>]</span><span class='op'>]</span><span class='op'>[</span>, <span class='fl'>1</span><span class='op'>]</span> <span class='op'>%in%</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>)</span>, <span class='fl'>2</span><span class='op'>]</span><span class='op'>)</span>
  <span class='op'>}</span> <span class='kw'>else</span> <span class='op'>{</span>
    <span class='fu'>barcodeplot</span><span class='op'>(</span>
      <span class='va'>y_dgelrt</span><span class='op'>$</span><span class='va'>table</span><span class='op'>$</span><span class='va'>logFC</span>,
      <span class='va'>y_index</span><span class='op'>[[</span><span class='va'>n</span><span class='op'>]</span><span class='op'>]</span><span class='op'>)</span>
  <span class='op'>}</span>
<span class='op'>}</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:unnamed-chunk-31"></span>
<img src="C086_Kinkel.male_B_cells.multi-sample_comparisons_files/figure-html5/unnamed-chunk-31-1.png" alt="MD-plot and barcode plot of DEGs in supplied gene sets. Directional gene sets have (1) MD plot points coloured according to the statistical significance of the gene **in the gene set** (2) barcode plot 'weights' given by the logFC of the gene **in the gene set**. For the barcode plot, genes are represented by bars and are ranked from left to right by increasing log-fold change. This forms the barcode-like pattern. The line (or *worm*) above the barcode shows the relative local enrichment of the vertical bars in each part of the plot. The dotted horizontal line indicates neutral enrichment; the worm above the dotted line shows enrichment while the worm below the dotted line shows depletion." width="624" />
<p class="caption">
Figure 3: MD-plot and barcode plot of DEGs in supplied gene sets. Directional gene sets have (1) MD plot points coloured according to the statistical significance of the gene <strong>in the gene set</strong> (2) barcode plot weights given by the logFC of the gene <strong>in the gene set</strong>. For the barcode plot, genes are represented by bars and are ranked from left to right by increasing log-fold change. This forms the barcode-like pattern. The line (or <em>worm</em>) above the barcode shows the relative local enrichment of the vertical bars in each part of the plot. The dotted horizontal line indicates neutral enrichment; the worm above the dotted line shows enrichment while the worm below the dotted line shows depletion.
</p>
</div>
</div>
<h1 class="appendix" id="session-info">Session info</h1>
<summary>
The analysis and this document were prepared using the following software (click triangle to expand)
</summary>
<details>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'>sessioninfo</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/sessioninfo/man/session_info.html'>session_info</a></span><span class='op'>(</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<pre><code> Session info 
 setting  value                       
 version  R version 4.0.3 (2020-10-10)
 os       CentOS Linux 7 (Core)       
 system   x86_64, linux-gnu           
 ui       X11                         
 language (EN)                        
 collate  en_US.UTF-8                 
 ctype    en_US.UTF-8                 
 tz       Australia/Melbourne         
 date     2021-10-15                  

 Packages 
 ! package                            * version  date       lib
 P annotate                             1.68.0   2020-10-27 [?]
 P AnnotationDbi                      * 1.52.0   2020-10-27 [?]
 P AnnotationFilter                   * 1.14.0   2020-10-27 [?]
 P askpass                              1.1      2019-01-13 [?]
 P assertthat                           0.2.1    2019-03-21 [?]
 P beachmat                             2.6.4    2020-12-20 [?]
 P Biobase                            * 2.50.0   2020-10-27 [?]
 P BiocFileCache                      * 1.14.0   2020-10-27 [?]
 P BiocGenerics                       * 0.36.0   2020-10-27 [?]
 P BiocManager                          1.30.10  2019-11-16 [?]
 P BiocParallel                         1.24.1   2020-11-06 [?]
 P BiocStyle                            2.18.1   2020-11-24 [?]
 P biomaRt                              2.46.3   2021-02-09 [?]
 P Biostrings                           2.58.0   2020-10-27 [?]
 P bit                                  4.0.4    2020-08-04 [?]
 P bit64                                4.0.5    2020-08-30 [?]
 P bitops                               1.0-6    2013-08-17 [?]
 P blob                                 1.2.1    2020-01-20 [?]
 P cachem                               1.0.4    2021-02-13 [?]
 P cli                                  2.3.1    2021-02-23 [?]
 P colorspace                           2.0-0    2020-11-11 [?]
 P crayon                               1.4.1    2021-02-08 [?]
 P crosstalk                            1.1.1    2021-01-12 [?]
 P curl                                 4.3      2019-12-02 [?]
 P data.table                           1.14.0   2021-02-21 [?]
 P DBI                                  1.1.1    2021-01-15 [?]
 P dbplyr                             * 2.1.0    2021-02-03 [?]
 P DelayedArray                         0.16.1   2021-01-22 [?]
 P DelayedMatrixStats                   1.12.3   2021-02-03 [?]
 P DESeq2                               1.30.1   2021-02-19 [?]
 P digest                               0.6.27   2020-10-24 [?]
 P distill                              1.2      2021-01-13 [?]
 P downlit                              0.2.1    2020-11-04 [?]
 P dplyr                                1.0.4    2021-02-02 [?]
 P DT                                   0.17     2021-01-06 [?]
 P edgeR                              * 3.32.1   2021-01-14 [?]
 P ellipsis                             0.3.1    2020-05-15 [?]
 P EnsDb.Mmusculus.v79                * 2.99.0   2020-07-15 [?]
 P ensembldb                          * 2.14.0   2020-10-27 [?]
 P evaluate                             0.14     2019-05-28 [?]
 P fansi                                0.4.2    2021-01-15 [?]
 P fastmap                              1.1.0    2021-01-25 [?]
 P genefilter                           1.72.1   2021-01-21 [?]
 P geneplotter                          1.68.0   2020-10-27 [?]
 P generics                             0.1.0    2020-10-31 [?]
 P GenomeInfoDb                       * 1.26.2   2020-12-08 [?]
 P GenomeInfoDbData                     1.2.4    2020-10-20 [?]
 P GenomicAlignments                    1.26.0   2020-10-27 [?]
 P GenomicFeatures                    * 1.42.1   2020-11-12 [?]
 P GenomicRanges                      * 1.42.0   2020-10-27 [?]
 P ggplot2                              3.3.3    2020-12-30 [?]
 P Glimma                             * 2.0.0    2020-10-27 [?]
 P glue                                 1.4.2    2020-08-27 [?]
 P GO.db                              * 3.12.1   2020-11-04 [?]
 P graph                                1.68.0   2020-10-27 [?]
 P gtable                               0.3.0    2019-03-25 [?]
 P here                               * 1.0.1    2020-12-13 [?]
 P highr                                0.8      2019-03-20 [?]
 P hms                                  1.0.0    2021-01-13 [?]
 P htmltools                            0.5.1.1  2021-01-22 [?]
 P htmlwidgets                          1.5.3    2020-12-10 [?]
 P httr                                 1.4.2    2020-07-20 [?]
 P IRanges                            * 2.24.1   2020-12-12 [?]
 P jsonlite                             1.7.2    2020-12-09 [?]
 P knitr                                1.31     2021-01-27 [?]
 P lattice                              0.20-41  2020-04-02 [3]
 P lazyeval                             0.2.2    2019-03-15 [?]
 P lifecycle                            1.0.0    2021-02-15 [?]
 P limma                              * 3.46.0   2020-10-27 [?]
 P locfit                               1.5-9.4  2020-03-25 [?]
 P magrittr                             2.0.1    2020-11-17 [?]
 P Matrix                               1.2-18   2019-11-27 [3]
 P MatrixGenerics                       1.2.1    2021-01-30 [?]
 P matrixStats                          0.58.0   2021-01-29 [?]
 P memoise                              2.0.0    2021-01-26 [?]
 P msigdbr                            * 7.2.1    2020-10-02 [?]
 P munsell                              0.5.0    2018-06-12 [?]
 P Mus.musculus                       * 1.3.1    2020-07-15 [?]
 P openssl                              1.4.3    2020-09-18 [?]
 P org.Mm.eg.db                       * 3.12.0   2020-10-20 [?]
 P OrganismDbi                        * 1.32.0   2020-10-27 [?]
 P pillar                               1.5.0    2021-02-22 [?]
 P pkgconfig                            2.0.3    2019-09-22 [?]
 P prettyunits                          1.1.1    2020-01-24 [?]
 P progress                             1.2.2    2019-05-16 [?]
 P ProtGenerics                         1.22.0   2020-10-27 [?]
 P purrr                                0.3.4    2020-04-17 [?]
 P R6                                   2.5.0    2020-10-28 [?]
 P rappdirs                             0.3.3    2021-01-31 [?]
 P RBGL                                 1.66.0   2020-10-27 [?]
 P RColorBrewer                         1.1-2    2014-12-07 [?]
 P Rcpp                                 1.0.6    2021-01-15 [?]
 P RCurl                                1.98-1.2 2020-04-18 [?]
 P rlang                                0.4.10   2020-12-30 [?]
 P rmarkdown                            2.11     2021-09-14 [?]
 P rprojroot                            2.0.2    2020-11-15 [?]
 P Rsamtools                            2.6.0    2020-10-27 [?]
 P RSQLite                              2.2.3    2021-01-24 [?]
 P rtracklayer                          1.50.0   2020-10-27 [?]
 P S4Vectors                          * 0.28.1   2020-12-09 [?]
 P scales                             * 1.1.1    2020-05-11 [?]
 P scuttle                              1.0.4    2020-12-17 [?]
 P sessioninfo                          1.1.1    2018-11-05 [?]
 P SingleCellExperiment                 1.12.0   2020-10-27 [?]
 P sparseMatrixStats                    1.2.1    2021-02-02 [?]
 P statmod                              1.4.35   2020-10-19 [?]
 P stringi                              1.5.3    2020-09-09 [?]
 P stringr                              1.4.0    2019-02-10 [?]
 P SummarizedExperiment                 1.20.0   2020-10-27 [?]
 P survival                             3.2-7    2020-09-28 [3]
 P tibble                               3.0.6    2021-01-29 [?]
 P tidyselect                           1.1.0    2020-05-11 [?]
 P TxDb.Mmusculus.UCSC.mm10.knownGene * 3.10.0   2020-07-15 [?]
 P utf8                                 1.1.4    2018-05-24 [?]
 P vctrs                                0.3.6    2020-12-17 [?]
 P withr                                2.4.1    2021-01-26 [?]
 P xfun                                 0.26     2021-09-14 [?]
 P XML                                  3.99-0.5 2020-07-23 [?]
 P xml2                                 1.3.2    2020-04-23 [?]
 P xtable                               1.8-4    2019-04-21 [?]
 P XVector                              0.30.0   2020-10-27 [?]
 P yaml                                 2.2.1    2020-02-01 [?]
 P zlibbioc                             1.36.0   2020-10-27 [?]
 source        
 Bioconductor  
 Bioconductor  
 Bioconductor  
 CRAN (R 4.0.0)
 CRAN (R 4.0.0)
 Bioconductor  
 Bioconductor  
 Bioconductor  
 Bioconductor  
 CRAN (R 4.0.0)
 Bioconductor  
 Bioconductor  
 Bioconductor  
 Bioconductor  
 CRAN (R 4.0.0)
 CRAN (R 4.0.0)
 CRAN (R 4.0.0)
 CRAN (R 4.0.0)
 CRAN (R 4.0.3)
 CRAN (R 4.0.3)
 CRAN (R 4.0.3)
 CRAN (R 4.0.3)
 CRAN (R 4.0.3)
 CRAN (R 4.0.0)
 CRAN (R 4.0.3)
 CRAN (R 4.0.3)
 CRAN (R 4.0.3)
 Bioconductor  
 Bioconductor  
 Bioconductor  
 CRAN (R 4.0.2)
 CRAN (R 4.0.3)
 CRAN (R 4.0.3)
 CRAN (R 4.0.3)
 CRAN (R 4.0.3)
 Bioconductor  
 CRAN (R 4.0.0)
 Bioconductor  
 Bioconductor  
 CRAN (R 4.0.0)
 CRAN (R 4.0.3)
 CRAN (R 4.0.3)
 Bioconductor  
 Bioconductor  
 CRAN (R 4.0.3)
 Bioconductor  
 Bioconductor  
 Bioconductor  
 Bioconductor  
 Bioconductor  
 CRAN (R 4.0.3)
 Bioconductor  
 CRAN (R 4.0.0)
 Bioconductor  
 Bioconductor  
 CRAN (R 4.0.0)
 CRAN (R 4.0.3)
 CRAN (R 4.0.0)
 CRAN (R 4.0.3)
 CRAN (R 4.0.3)
 CRAN (R 4.0.3)
 CRAN (R 4.0.0)
 Bioconductor  
 CRAN (R 4.0.3)
 CRAN (R 4.0.3)
 CRAN (R 4.0.3)
 CRAN (R 4.0.0)
 CRAN (R 4.0.3)
 Bioconductor  
 CRAN (R 4.0.0)
 CRAN (R 4.0.3)
 CRAN (R 4.0.3)
 Bioconductor  
 CRAN (R 4.0.3)
 CRAN (R 4.0.3)
 CRAN (R 4.0.2)
 CRAN (R 4.0.0)
 Bioconductor  
 CRAN (R 4.0.2)
 Bioconductor  
 Bioconductor  
 CRAN (R 4.0.3)
 CRAN (R 4.0.0)
 CRAN (R 4.0.0)
 CRAN (R 4.0.0)
 Bioconductor  
 CRAN (R 4.0.0)
 CRAN (R 4.0.2)
 CRAN (R 4.0.3)
 Bioconductor  
 CRAN (R 4.0.0)
 CRAN (R 4.0.3)
 CRAN (R 4.0.0)
 CRAN (R 4.0.3)
 CRAN (R 4.0.0)
 CRAN (R 4.0.3)
 Bioconductor  
 CRAN (R 4.0.3)
 Bioconductor  
 Bioconductor  
 CRAN (R 4.0.0)
 Bioconductor  
 CRAN (R 4.0.0)
 Bioconductor  
 Bioconductor  
 CRAN (R 4.0.2)
 CRAN (R 4.0.0)
 CRAN (R 4.0.0)
 Bioconductor  
 CRAN (R 4.0.3)
 CRAN (R 4.0.3)
 CRAN (R 4.0.0)
 Bioconductor  
 CRAN (R 4.0.0)
 CRAN (R 4.0.3)
 CRAN (R 4.0.3)
 CRAN (R 4.0.0)
 CRAN (R 4.0.0)
 CRAN (R 4.0.0)
 CRAN (R 4.0.0)
 Bioconductor  
 CRAN (R 4.0.0)
 Bioconductor  

[1] /stornext/Projects/score/Analyses/C086_Kinkel/renv/library/R-4.0/x86_64-pc-linux-gnu
[2] /tmp/Rtmp6h4rt0/renv-system-library
[3] /stornext/System/data/apps/R/R-4.0.3/lib64/R/library

 P  Loaded and on-disk path mismatch.</code></pre>
</div>
</details>
<div class="sourceCode" id="cb3"><pre class="sourceCode r distill-force-highlighting-css"><code class="sourceCode r"></code></pre></div>
<section class="footnotes" role="doc-endnotes">
<hr />
<ol>
<li id="fn1" role="doc-endnote"><p>Ensembl gene ID<a href="#fnref1" class="footnote-back" role="doc-backlink"></a></p></li>
<li id="fn2" role="doc-endnote"><p>Some care is taken to account for missing and duplicate gene symbols; missing symbols are replaced with the Ensembl identifier and duplicated symbols are concatenated with the (unique) Ensembl identifiers.<a href="#fnref2" class="footnote-back" role="doc-backlink"></a></p></li>
</ol>
</section>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>


<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom">
<h3 id="references">References</h3>
<div id="references-listing"></div>
</div>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<!--/radix_placeholder_navigation_after_body-->

</body>

</html>
