<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ad0000; } /* Alert */
code span.an { color: #5e5e5e; } /* Annotation */
code span.at { color: #20794d; } /* Attribute */
code span.bn { color: #ad0000; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007ba5; } /* ControlFlow */
code span.ch { color: #20794d; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #5e5e5e; } /* Comment */
code span.cv { color: #5e5e5e; font-style: italic; } /* CommentVar */
code span.do { color: #5e5e5e; font-style: italic; } /* Documentation */
code span.dt { color: #ad0000; } /* DataType */
code span.dv { color: #ad0000; } /* DecVal */
code span.er { color: #ad0000; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #ad0000; } /* Float */
code span.fu { color: #4758ab; } /* Function */
code span.im { } /* Import */
code span.in { color: #5e5e5e; } /* Information */
code span.kw { color: #007ba5; } /* Keyword */
code span.op { color: #5e5e5e; } /* Operator */
code span.ot { color: #007ba5; } /* Other */
code span.pp { color: #ad0000; } /* Preprocessor */
code span.sc { color: #20794d; } /* SpecialChar */
code span.ss { color: #20794d; } /* SpecialString */
code span.st { color: #20794d; } /* String */
code span.va { color: #111111; } /* Variable */
code span.vs { color: #20794d; } /* VerbatimString */
code span.wa { color: #5e5e5e; font-style: italic; } /* Warning */
</style>

  <!--radix_placeholder_meta_tags-->
  <title>C086: Preprocessing the Kinkel (C086) Smchd1 LSK mini-bulk data set</title>
  
  
  
  <!--  https://schema.org/Article -->
  <meta property="article:published" itemprop="datePublished" content="2021-03-31"/>
  <meta property="article:created" itemprop="dateCreated" content="2021-03-31"/>
  <meta name="article:author" content="Peter Hickey"/>
  
  <!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
  <meta property="og:title" content="C086: Preprocessing the Kinkel (C086) Smchd1 LSK mini-bulk data set"/>
  <meta property="og:type" content="article"/>
  <meta property="og:locale" content="en_US"/>
  <meta property="og:site_name" content="C086"/>
  
  <!--  https://dev.twitter.com/cards/types/summary -->
  <meta property="twitter:card" content="summary"/>
  <meta property="twitter:title" content="C086: Preprocessing the Kinkel (C086) Smchd1 LSK mini-bulk data set"/>
  
  <!--/radix_placeholder_meta_tags-->
  
  <!--radix_placeholder_rmarkdown_metadata-->
  
  <script type="text/json" id="radix-rmarkdown-metadata">
  {"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","description","author","date","editor_options","bibliography"]}},"value":[{"type":"character","attributes":{},"value":["Preprocessing the Kinkel (C086) Smchd1 LSK mini-bulk data set"]},{"type":"NULL"},{"type":"list","attributes":{},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","url","affiliation","affiliation_url"]}},"value":[{"type":"character","attributes":{},"value":["Peter Hickey"]},{"type":"character","attributes":{},"value":["https://peterhickey.org"]},{"type":"character","attributes":{},"value":["WEHI SCORE"]},{"type":"character","attributes":{},"value":["https://www.wehi.edu.au/people/shalin-naik/3310/score"]}]}]},{"type":"character","attributes":{},"value":["2021-03-31"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["chunk_output_type"]}},"value":[{"type":"character","attributes":{},"value":["console"]}]},{"type":"character","attributes":{},"value":["ref.bib"]}]}
  </script>
  <!--/radix_placeholder_rmarkdown_metadata-->
  <!--radix_placeholder_navigation_in_header-->
  <meta name="distill:offset" content=""/>
  
  <script type="application/javascript">
  
    window.headroom_prevent_pin = false;
  
    window.document.addEventListener("DOMContentLoaded", function (event) {
  
      // initialize headroom for banner
      var header = $('header').get(0);
      var headerHeight = header.offsetHeight;
      var headroom = new Headroom(header, {
        tolerance: 5,
        onPin : function() {
          if (window.headroom_prevent_pin) {
            window.headroom_prevent_pin = false;
            headroom.unpin();
          }
        }
      });
      headroom.init();
      if(window.location.hash)
        headroom.unpin();
      $(header).addClass('headroom--transition');
  
      // offset scroll location for banner on hash change
      // (see: https://github.com/WickyNilliams/headroom.js/issues/38)
      window.addEventListener("hashchange", function(event) {
        window.scrollTo(0, window.pageYOffset - (headerHeight + 25));
      });
  
      // responsive menu
      $('.distill-site-header').each(function(i, val) {
        var topnav = $(this);
        var toggle = topnav.find('.nav-toggle');
        toggle.on('click', function() {
          topnav.toggleClass('responsive');
        });
      });
  
      // nav dropdowns
      $('.nav-dropbtn').click(function(e) {
        $(this).next('.nav-dropdown-content').toggleClass('nav-dropdown-active');
        $(this).parent().siblings('.nav-dropdown')
           .children('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $("body").click(function(e){
        $('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $(".nav-dropdown").click(function(e){
        e.stopPropagation();
      });
    });
  </script>
  
  <style type="text/css">
  
  /* Theme (user-documented overrideables for nav appearance) */
  
  .distill-site-nav {
    color: rgba(255, 255, 255, 0.8);
    background-color: #0F2E3D;
    font-size: 15px;
    font-weight: 300;
  }
  
  .distill-site-nav a {
    color: inherit;
    text-decoration: none;
  }
  
  .distill-site-nav a:hover {
    color: white;
  }
  
  @media print {
    .distill-site-nav {
      display: none;
    }
  }
  
  .distill-site-header {
  
  }
  
  .distill-site-footer {
  
  }
  
  
  /* Site Header */
  
  .distill-site-header {
    width: 100%;
    box-sizing: border-box;
    z-index: 3;
  }
  
  .distill-site-header .nav-left {
    display: inline-block;
    margin-left: 8px;
  }
  
  @media screen and (max-width: 768px) {
    .distill-site-header .nav-left {
      margin-left: 0;
    }
  }
  
  
  .distill-site-header .nav-right {
    float: right;
    margin-right: 8px;
  }
  
  .distill-site-header a,
  .distill-site-header .title {
    display: inline-block;
    text-align: center;
    padding: 14px 10px 14px 10px;
  }
  
  .distill-site-header .title {
    font-size: 18px;
    min-width: 150px;
  }
  
  .distill-site-header .logo {
    padding: 0;
  }
  
  .distill-site-header .logo img {
    display: none;
    max-height: 20px;
    width: auto;
    margin-bottom: -4px;
  }
  
  .distill-site-header .nav-image img {
    max-height: 18px;
    width: auto;
    display: inline-block;
    margin-bottom: -3px;
  }
  
  
  
  @media screen and (min-width: 1000px) {
    .distill-site-header .logo img {
      display: inline-block;
    }
    .distill-site-header .nav-left {
      margin-left: 20px;
    }
    .distill-site-header .nav-right {
      margin-right: 20px;
    }
    .distill-site-header .title {
      padding-left: 12px;
    }
  }
  
  
  .distill-site-header .nav-toggle {
    display: none;
  }
  
  .nav-dropdown {
    display: inline-block;
    position: relative;
  }
  
  .nav-dropdown .nav-dropbtn {
    border: none;
    outline: none;
    color: rgba(255, 255, 255, 0.8);
    padding: 16px 10px;
    background-color: transparent;
    font-family: inherit;
    font-size: inherit;
    font-weight: inherit;
    margin: 0;
    margin-top: 1px;
    z-index: 2;
  }
  
  .nav-dropdown-content {
    display: none;
    position: absolute;
    background-color: white;
    min-width: 200px;
    border: 1px solid rgba(0,0,0,0.15);
    border-radius: 4px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.1);
    z-index: 1;
    margin-top: 2px;
    white-space: nowrap;
    padding-top: 4px;
    padding-bottom: 4px;
  }
  
  .nav-dropdown-content hr {
    margin-top: 4px;
    margin-bottom: 4px;
    border: none;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .nav-dropdown-active {
    display: block;
  }
  
  .nav-dropdown-content a, .nav-dropdown-content .nav-dropdown-header {
    color: black;
    padding: 6px 24px;
    text-decoration: none;
    display: block;
    text-align: left;
  }
  
  .nav-dropdown-content .nav-dropdown-header {
    display: block;
    padding: 5px 24px;
    padding-bottom: 0;
    text-transform: uppercase;
    font-size: 14px;
    color: #999999;
    white-space: nowrap;
  }
  
  .nav-dropdown:hover .nav-dropbtn {
    color: white;
  }
  
  .nav-dropdown-content a:hover {
    background-color: #ddd;
    color: black;
  }
  
  .nav-right .nav-dropdown-content {
    margin-left: -45%;
    right: 0;
  }
  
  @media screen and (max-width: 768px) {
    .distill-site-header a, .distill-site-header .nav-dropdown  {display: none;}
    .distill-site-header a.nav-toggle {
      float: right;
      display: block;
    }
    .distill-site-header .title {
      margin-left: 0;
    }
    .distill-site-header .nav-right {
      margin-right: 0;
    }
    .distill-site-header {
      overflow: hidden;
    }
    .nav-right .nav-dropdown-content {
      margin-left: 0;
    }
  }
  
  
  @media screen and (max-width: 768px) {
    .distill-site-header.responsive {position: relative; min-height: 500px; }
    .distill-site-header.responsive a.nav-toggle {
      position: absolute;
      right: 0;
      top: 0;
    }
    .distill-site-header.responsive a,
    .distill-site-header.responsive .nav-dropdown {
      display: block;
      text-align: left;
    }
    .distill-site-header.responsive .nav-left,
    .distill-site-header.responsive .nav-right {
      width: 100%;
    }
    .distill-site-header.responsive .nav-dropdown {float: none;}
    .distill-site-header.responsive .nav-dropdown-content {position: relative;}
    .distill-site-header.responsive .nav-dropdown .nav-dropbtn {
      display: block;
      width: 100%;
      text-align: left;
    }
  }
  
  /* Site Footer */
  
  .distill-site-footer {
    width: 100%;
    overflow: hidden;
    box-sizing: border-box;
    z-index: 3;
    margin-top: 30px;
    padding-top: 30px;
    padding-bottom: 30px;
    text-align: center;
  }
  
  /* Headroom */
  
  d-title {
    padding-top: 6rem;
  }
  
  @media print {
    d-title {
      padding-top: 4rem;
    }
  }
  
  .headroom {
    z-index: 1000;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
  }
  
  .headroom--transition {
    transition: all .4s ease-in-out;
  }
  
  .headroom--unpinned {
    top: -100px;
  }
  
  .headroom--pinned {
    top: 0;
  }
  
  /* adjust viewport for navbar height */
  /* helps vertically center bootstrap (non-distill) content */
  .min-vh-100 {
    min-height: calc(100vh - 100px) !important;
  }
  
  </style>
  
  <script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
  <link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet"/>
  <link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet"/>
  <script src="site_libs/headroom-0.9.4/headroom.min.js"></script>
  <script src="site_libs/autocomplete-0.37.1/autocomplete.min.js"></script>
  <script src="site_libs/fuse-6.4.1/fuse.min.js"></script>
  
  <script type="application/javascript">
  
  function getMeta(metaName) {
    var metas = document.getElementsByTagName('meta');
    for (let i = 0; i < metas.length; i++) {
      if (metas[i].getAttribute('name') === metaName) {
        return metas[i].getAttribute('content');
      }
    }
    return '';
  }
  
  function offsetURL(url) {
    var offset = getMeta('distill:offset');
    return offset ? offset + '/' + url : url;
  }
  
  function createFuseIndex() {
  
    // create fuse index
    var options = {
      keys: [
        { name: 'title', weight: 20 },
        { name: 'categories', weight: 15 },
        { name: 'description', weight: 10 },
        { name: 'contents', weight: 5 },
      ],
      ignoreLocation: true,
      threshold: 0
    };
    var fuse = new window.Fuse([], options);
  
    // fetch the main search.json
    return fetch(offsetURL('search.json'))
      .then(function(response) {
        if (response.status == 200) {
          return response.json().then(function(json) {
            // index main articles
            json.articles.forEach(function(article) {
              fuse.add(article);
            });
            // download collections and index their articles
            return Promise.all(json.collections.map(function(collection) {
              return fetch(offsetURL(collection)).then(function(response) {
                if (response.status === 200) {
                  return response.json().then(function(articles) {
                    articles.forEach(function(article) {
                      fuse.add(article);
                    });
                  })
                } else {
                  return Promise.reject(
                    new Error('Unexpected status from search index request: ' +
                              response.status)
                  );
                }
              });
            })).then(function() {
              return fuse;
            });
          });
  
        } else {
          return Promise.reject(
            new Error('Unexpected status from search index request: ' +
                        response.status)
          );
        }
      });
  }
  
  window.document.addEventListener("DOMContentLoaded", function (event) {
  
    // get search element (bail if we don't have one)
    var searchEl = window.document.getElementById('distill-search');
    if (!searchEl)
      return;
  
    createFuseIndex()
      .then(function(fuse) {
  
        // make search box visible
        searchEl.classList.remove('hidden');
  
        // initialize autocomplete
        var options = {
          autoselect: true,
          hint: false,
          minLength: 2,
        };
        window.autocomplete(searchEl, options, [{
          source: function(query, callback) {
            const searchOptions = {
              isCaseSensitive: false,
              shouldSort: true,
              minMatchCharLength: 2,
              limit: 10,
            };
            var results = fuse.search(query, searchOptions);
            callback(results
              .map(function(result) { return result.item; })
            );
          },
          templates: {
            suggestion: function(suggestion) {
              var img = suggestion.preview && Object.keys(suggestion.preview).length > 0
                ? `<img src="${offsetURL(suggestion.preview)}"</img>`
                : '';
              var html = `
                <div class="search-item">
                  <h3>${suggestion.title}</h3>
                  <div class="search-item-description">
                    ${suggestion.description || ''}
                  </div>
                  <div class="search-item-preview">
                    ${img}
                  </div>
                </div>
              `;
              return html;
            }
          }
        }]).on('autocomplete:selected', function(event, suggestion) {
          window.location.href = offsetURL(suggestion.path);
        });
        // remove inline display style on autocompleter (we want to
        // manage responsive display via css)
        $('.algolia-autocomplete').css("display", "");
      })
      .catch(function(error) {
        console.log(error);
      });
  
  });
  
  </script>
  
  <style type="text/css">
  
  .nav-search {
    font-size: x-small;
  }
  
  /* Algolioa Autocomplete */
  
  .algolia-autocomplete {
    display: inline-block;
    margin-left: 10px;
    vertical-align: sub;
    background-color: white;
    color: black;
    padding: 6px;
    padding-top: 8px;
    padding-bottom: 0;
    border-radius: 6px;
    border: 1px #0F2E3D solid;
    width: 180px;
  }
  
  
  @media screen and (max-width: 768px) {
    .distill-site-nav .algolia-autocomplete {
      display: none;
      visibility: hidden;
    }
    .distill-site-nav.responsive .algolia-autocomplete {
      display: inline-block;
      visibility: visible;
    }
    .distill-site-nav.responsive .algolia-autocomplete .aa-dropdown-menu {
      margin-left: 0;
      width: 400px;
      max-height: 400px;
    }
  }
  
  .algolia-autocomplete .aa-input, .algolia-autocomplete .aa-hint {
    width: 90%;
    outline: none;
    border: none;
  }
  
  .algolia-autocomplete .aa-hint {
    color: #999;
  }
  .algolia-autocomplete .aa-dropdown-menu {
    width: 550px;
    max-height: 70vh;
    overflow-x: visible;
    overflow-y: scroll;
    padding: 5px;
    margin-top: 3px;
    margin-left: -150px;
    background-color: #fff;
    border-radius: 5px;
    border: 1px solid #999;
    border-top: none;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion {
    cursor: pointer;
    padding: 5px 4px;
    border-bottom: 1px solid #eee;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion:last-of-type {
    border-bottom: none;
    margin-bottom: 2px;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item {
    overflow: hidden;
    font-size: 0.8em;
    line-height: 1.4em;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item h3 {
    font-size: 1rem;
    margin-block-start: 0;
    margin-block-end: 5px;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-description {
    display: inline-block;
    overflow: hidden;
    height: 2.8em;
    width: 80%;
    margin-right: 4%;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview {
    display: inline-block;
    width: 15%;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview img {
    height: 3em;
    width: auto;
    display: none;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview img[src] {
    display: initial;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion.aa-cursor {
    background-color: #eee;
  }
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion em {
    font-weight: bold;
    font-style: normal;
  }
  
  </style>
  
  
  <!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->
  
  <style type="text/css">
  
  body {
    background-color: white;
  }
  
  .pandoc-table {
    width: 100%;
  }
  
  .pandoc-table>caption {
    margin-bottom: 10px;
  }
  
  .pandoc-table th:not([align]) {
    text-align: left;
  }
  
  .pagedtable-footer {
    font-size: 15px;
  }
  
  d-byline .byline {
    grid-template-columns: 2fr 2fr;
  }
  
  d-byline .byline h3 {
    margin-block-start: 1.5em;
  }
  
  d-byline .byline .authors-affiliations h3 {
    margin-block-start: 0.5em;
  }
  
  .authors-affiliations .orcid-id {
    width: 16px;
    height:16px;
    margin-left: 4px;
    margin-right: 4px;
    vertical-align: middle;
    padding-bottom: 2px;
  }
  
  d-title .dt-tags {
    margin-top: 1em;
    grid-column: text;
  }
  
  .dt-tags .dt-tag {
    text-decoration: none;
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0em 0.4em;
    margin-right: 0.5em;
    margin-bottom: 0.4em;
    font-size: 70%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }
  
  d-article table.gt_table td,
  d-article table.gt_table th {
    border-bottom: none;
  }
  
  .html-widget {
    margin-bottom: 2.0em;
  }
  
  .l-screen-inset {
    padding-right: 16px;
  }
  
  .l-screen .caption {
    margin-left: 10px;
  }
  
  .shaded {
    background: rgb(247, 247, 247);
    padding-top: 20px;
    padding-bottom: 20px;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .shaded .html-widget {
    margin-bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .shaded .shaded-content {
    background: white;
  }
  
  .text-output {
    margin-top: 0;
    line-height: 1.5em;
  }
  
  .hidden {
    display: none !important;
  }
  
  d-article {
    padding-top: 2.5rem;
    padding-bottom: 30px;
  }
  
  d-appendix {
    padding-top: 30px;
  }
  
  d-article>p>img {
    width: 100%;
  }
  
  d-article h2 {
    margin: 1rem 0 1.5rem 0;
  }
  
  d-article h3 {
    margin-top: 1.5rem;
  }
  
  d-article iframe {
    border: 1px solid rgba(0, 0, 0, 0.1);
    margin-bottom: 2.0em;
    width: 100%;
  }
  
  /* Tweak code blocks */
  
  d-article div.sourceCode code,
  d-article pre code {
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
  }
  
  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: auto;
  }
  
  d-article div.sourceCode {
    background-color: white;
  }
  
  d-article div.sourceCode pre {
    padding-left: 10px;
    font-size: 12px;
    border-left: 2px solid rgba(0,0,0,0.1);
  }
  
  d-article pre {
    font-size: 12px;
    color: black;
    background: none;
    margin-top: 0;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;
  
    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;
  
    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }
  
  d-article pre a {
    border-bottom: none;
  }
  
  d-article pre a:hover {
    border-bottom: none;
    text-decoration: underline;
  }
  
  d-article details {
    grid-column: text;
    margin-bottom: 0.8em;
  }
  
  @media(min-width: 768px) {
  
  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: visible !important;
  }
  
  d-article div.sourceCode pre {
    padding-left: 18px;
    font-size: 14px;
  }
  
  d-article pre {
    font-size: 14px;
  }
  
  }
  
  figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }
  
  /* CSS for d-contents */
  
  .d-contents {
    grid-column: text;
    color: rgba(0,0,0,0.8);
    font-size: 0.9em;
    padding-bottom: 1em;
    margin-bottom: 1em;
    padding-bottom: 0.5em;
    margin-bottom: 1em;
    padding-left: 0.25em;
    justify-self: start;
  }
  
  @media(min-width: 1000px) {
    .d-contents.d-contents-float {
      height: 0;
      grid-column-start: 1;
      grid-column-end: 4;
      justify-self: center;
      padding-right: 3em;
      padding-left: 2em;
    }
  }
  
  .d-contents nav h3 {
    font-size: 18px;
    margin-top: 0;
    margin-bottom: 1em;
  }
  
  .d-contents li {
    list-style-type: none
  }
  
  .d-contents nav > ul {
    padding-left: 0;
  }
  
  .d-contents ul {
    padding-left: 1em
  }
  
  .d-contents nav ul li {
    margin-top: 0.6em;
    margin-bottom: 0.2em;
  }
  
  .d-contents nav a {
    font-size: 13px;
    border-bottom: none;
    text-decoration: none
    color: rgba(0, 0, 0, 0.8);
  }
  
  .d-contents nav a:hover {
    text-decoration: underline solid rgba(0, 0, 0, 0.6)
  }
  
  .d-contents nav > ul > li > a {
    font-weight: 600;
  }
  
  .d-contents nav > ul > li > ul {
    font-weight: inherit;
  }
  
  .d-contents nav > ul > li > ul > li {
    margin-top: 0.2em;
  }
  
  
  .d-contents nav ul {
    margin-top: 0;
    margin-bottom: 0.25em;
  }
  
  .d-article-with-toc h2:nth-child(2) {
    margin-top: 0;
  }
  
  
  /* Figure */
  
  .figure {
    position: relative;
    margin-bottom: 2.5em;
    margin-top: 1.5em;
  }
  
  .figure img {
    width: 100%;
  }
  
  .figure .caption {
    color: rgba(0, 0, 0, 0.6);
    font-size: 12px;
    line-height: 1.5em;
  }
  
  .figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }
  
  .figure .caption a {
    color: rgba(0, 0, 0, 0.6);
  }
  
  .figure .caption b,
  .figure .caption strong, {
    font-weight: 600;
    color: rgba(0, 0, 0, 1.0);
  }
  
  /* Citations */
  
  d-article .citation {
    color: inherit;
    cursor: inherit;
  }
  
  div.hanging-indent{
    margin-left: 1em; text-indent: -1em;
  }
  
  /* Citation hover box */
  
  .tippy-box[data-theme~=light-border] {
    background-color: rgba(250, 250, 250, 0.95);
  }
  
  .tippy-content > p {
    margin-bottom: 0;
    padding: 2px;
  }
  
  
  /* Tweak 1000px media break to show more text */
  
  @media(min-width: 1000px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 16px;
    }
  
    .grid {
      grid-column-gap: 16px;
    }
  
    d-article {
      font-size: 1.06rem;
      line-height: 1.7em;
    }
    figure .caption, .figure .caption, figure figcaption {
      font-size: 13px;
    }
  }
  
  @media(min-width: 1180px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 32px;
    }
  
    .grid {
      grid-column-gap: 32px;
    }
  }
  
  
  /* Get the citation styles for the appendix (not auto-injected on render since
     we do our own rendering of the citation appendix) */
  
  d-appendix .citation-appendix,
  .d-appendix .citation-appendix {
    font-size: 11px;
    line-height: 15px;
    border-left: 1px solid rgba(0, 0, 0, 0.1);
    padding-left: 18px;
    border: 1px solid rgba(0,0,0,0.1);
    background: rgba(0, 0, 0, 0.02);
    padding: 10px 18px;
    border-radius: 3px;
    color: rgba(150, 150, 150, 1);
    overflow: hidden;
    margin-top: -12px;
    white-space: pre-wrap;
    word-wrap: break-word;
  }
  
  /* Include appendix styles here so they can be overridden */
  
  d-appendix {
    contain: layout style;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-top: 60px;
    margin-bottom: 0;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    color: rgba(0,0,0,0.5);
    padding-top: 60px;
    padding-bottom: 48px;
  }
  
  d-appendix h3 {
    grid-column: page-start / text-start;
    font-size: 15px;
    font-weight: 500;
    margin-top: 1em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.65);
  }
  
  d-appendix h3 + * {
    margin-top: 1em;
  }
  
  d-appendix ol {
    padding: 0 0 0 15px;
  }
  
  @media (min-width: 768px) {
    d-appendix ol {
      padding: 0 0 0 30px;
      margin-left: -30px;
    }
  }
  
  d-appendix li {
    margin-bottom: 1em;
  }
  
  d-appendix a {
    color: rgba(0, 0, 0, 0.6);
  }
  
  d-appendix > * {
    grid-column: text;
  }
  
  d-appendix > d-footnote-list,
  d-appendix > d-citation-list,
  d-appendix > distill-appendix {
    grid-column: screen;
  }
  
  /* Include footnote styles here so they can be overridden */
  
  d-footnote-list {
    contain: layout style;
  }
  
  d-footnote-list > * {
    grid-column: text;
  }
  
  d-footnote-list a.footnote-backlink {
    color: rgba(0,0,0,0.3);
    padding-left: 0.5em;
  }
  
  
  
  /* Anchor.js */
  
  .anchorjs-link {
    /*transition: all .25s linear; */
    text-decoration: none;
    border-bottom: none;
  }
  *:hover > .anchorjs-link {
    margin-left: -1.125em !important;
    text-decoration: none;
    border-bottom: none;
  }
  
  /* Social footer */
  
  .social_footer {
    margin-top: 30px;
    margin-bottom: 0;
    color: rgba(0,0,0,0.67);
  }
  
  .disqus-comments {
    margin-right: 30px;
  }
  
  .disqus-comment-count {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    cursor: pointer;
  }
  
  #disqus_thread {
    margin-top: 30px;
  }
  
  .article-sharing a {
    border-bottom: none;
    margin-right: 8px;
  }
  
  .article-sharing a:hover {
    border-bottom: none;
  }
  
  .sidebar-section.subscribe {
    font-size: 12px;
    line-height: 1.6em;
  }
  
  .subscribe p {
    margin-bottom: 0.5em;
  }
  
  
  .article-footer .subscribe {
    font-size: 15px;
    margin-top: 45px;
  }
  
  
  .sidebar-section.custom {
    font-size: 12px;
    line-height: 1.6em;
  }
  
  .custom p {
    margin-bottom: 0.5em;
  }
  
  /* Styles for listing layout (hide title) */
  .layout-listing d-title, .layout-listing .d-title {
    display: none;
  }
  
  /* Styles for posts lists (not auto-injected) */
  
  
  .posts-with-sidebar {
    padding-left: 45px;
    padding-right: 45px;
  }
  
  .posts-list .description h2,
  .posts-list .description p {
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
  }
  
  .posts-list .description h2 {
    font-weight: 700;
    border-bottom: none;
    padding-bottom: 0;
  }
  
  .posts-list h2.post-tag {
    border-bottom: 1px solid rgba(0, 0, 0, 0.2);
    padding-bottom: 12px;
  }
  .posts-list {
    margin-top: 60px;
    margin-bottom: 24px;
  }
  
  .posts-list .post-preview {
    text-decoration: none;
    overflow: hidden;
    display: block;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
    padding: 24px 0;
  }
  
  .post-preview-last {
    border-bottom: none !important;
  }
  
  .posts-list .posts-list-caption {
    grid-column: screen;
    font-weight: 400;
  }
  
  .posts-list .post-preview h2 {
    margin: 0 0 6px 0;
    line-height: 1.2em;
    font-style: normal;
    font-size: 24px;
  }
  
  .posts-list .post-preview p {
    margin: 0 0 12px 0;
    line-height: 1.4em;
    font-size: 16px;
  }
  
  .posts-list .post-preview .thumbnail {
    box-sizing: border-box;
    margin-bottom: 24px;
    position: relative;
    max-width: 500px;
  }
  .posts-list .post-preview img {
    width: 100%;
    display: block;
  }
  
  .posts-list .metadata {
    font-size: 12px;
    line-height: 1.4em;
    margin-bottom: 18px;
  }
  
  .posts-list .metadata > * {
    display: inline-block;
  }
  
  .posts-list .metadata .publishedDate {
    margin-right: 2em;
  }
  
  .posts-list .metadata .dt-authors {
    display: block;
    margin-top: 0.3em;
    margin-right: 2em;
  }
  
  .posts-list .dt-tags {
    display: block;
    line-height: 1em;
  }
  
  .posts-list .dt-tags .dt-tag {
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0.3em 0.4em;
    margin-right: 0.2em;
    margin-bottom: 0.4em;
    font-size: 60%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }
  
  .posts-list img {
    opacity: 1;
  }
  
  .posts-list img[data-src] {
    opacity: 0;
  }
  
  .posts-more {
    clear: both;
  }
  
  
  .posts-sidebar {
    font-size: 16px;
  }
  
  .posts-sidebar h3 {
    font-size: 16px;
    margin-top: 0;
    margin-bottom: 0.5em;
    font-weight: 400;
    text-transform: uppercase;
  }
  
  .sidebar-section {
    margin-bottom: 30px;
  }
  
  .categories ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
  }
  
  .categories li {
    color: rgba(0, 0, 0, 0.8);
    margin-bottom: 0;
  }
  
  .categories li>a {
    border-bottom: none;
  }
  
  .categories li>a:hover {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
  }
  
  .categories .active {
    font-weight: 600;
  }
  
  .categories .category-count {
    color: rgba(0, 0, 0, 0.4);
  }
  
  
  @media(min-width: 768px) {
    .posts-list .post-preview h2 {
      font-size: 26px;
    }
    .posts-list .post-preview .thumbnail {
      float: right;
      width: 30%;
      margin-bottom: 0;
    }
    .posts-list .post-preview .description {
      float: left;
      width: 45%;
    }
    .posts-list .post-preview .metadata {
      float: left;
      width: 20%;
      margin-top: 8px;
    }
    .posts-list .post-preview p {
      margin: 0 0 12px 0;
      line-height: 1.5em;
      font-size: 16px;
    }
    .posts-with-sidebar .posts-list {
      float: left;
      width: 75%;
    }
    .posts-with-sidebar .posts-sidebar {
      float: right;
      width: 20%;
      margin-top: 60px;
      padding-top: 24px;
      padding-bottom: 24px;
    }
  }
  
  
  /* Improve display for browsers without grid (IE/Edge <= 15) */
  
  .downlevel {
    line-height: 1.6em;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
    margin: 0;
  }
  
  .downlevel .d-title {
    padding-top: 6rem;
    padding-bottom: 1.5rem;
  }
  
  .downlevel .d-title h1 {
    font-size: 50px;
    font-weight: 700;
    line-height: 1.1em;
    margin: 0 0 0.5rem;
  }
  
  .downlevel .d-title p {
    font-weight: 300;
    font-size: 1.2rem;
    line-height: 1.55em;
    margin-top: 0;
  }
  
  .downlevel .d-byline {
    padding-top: 0.8em;
    padding-bottom: 0.8em;
    font-size: 0.8rem;
    line-height: 1.8em;
  }
  
  .downlevel .section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .downlevel .d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
    padding-top: 1rem;
    padding-bottom: 2rem;
  }
  
  
  .downlevel .d-appendix {
    padding-left: 0;
    padding-right: 0;
    max-width: none;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.5);
    padding-top: 40px;
    padding-bottom: 48px;
  }
  
  .downlevel .footnotes ol {
    padding-left: 13px;
  }
  
  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
    padding-left: 40px;
    padding-right: 40px;
  }
  
  @media(min-width: 768px) {
    .downlevel .base-grid,
    .downlevel .distill-header,
    .downlevel .d-title,
    .downlevel .d-abstract,
    .downlevel .d-article,
    .downlevel .d-appendix,
    .downlevel .distill-appendix,
    .downlevel .d-byline,
    .downlevel .d-footnote-list,
    .downlevel .d-citation-list,
    .downlevel .distill-footer,
    .downlevel .appendix-bottom,
    .downlevel .posts-container {
    padding-left: 150px;
    padding-right: 150px;
    max-width: 900px;
  }
  }
  
  .downlevel pre code {
    display: block;
    border-left: 2px solid rgba(0, 0, 0, .1);
    padding: 0 0 0 20px;
    font-size: 14px;
  }
  
  .downlevel code, .downlevel pre {
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;
  
    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;
  
    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }
  
  .downlevel .posts-list .post-preview {
    color: inherit;
  }
  
  
  
  </style>
  
  <script type="application/javascript">
  
  function is_downlevel_browser() {
    if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                   window.navigator.userAgent)) {
      return true;
    } else {
      return window.load_distill_framework === undefined;
    }
  }
  
  // show body when load is complete
  function on_load_complete() {
  
    // add anchors
    if (window.anchors) {
      window.anchors.options.placement = 'left';
      window.anchors.add('d-article > h2, d-article > h3, d-article > h4, d-article > h5');
    }
  
  
    // set body to visible
    document.body.style.visibility = 'visible';
  
    // force redraw for leaflet widgets
    if (window.HTMLWidgets) {
      var maps = window.HTMLWidgets.findAll(".leaflet");
      $.each(maps, function(i, el) {
        var map = this.getMap();
        map.invalidateSize();
        map.eachLayer(function(layer) {
          if (layer instanceof L.TileLayer)
            layer.redraw();
        });
      });
    }
  
    // trigger 'shown' so htmlwidgets resize
    $('d-article').trigger('shown');
  }
  
  function init_distill() {
  
    init_common();
  
    // create front matter
    var front_matter = $('<d-front-matter></d-front-matter>');
    $('#distill-front-matter').wrap(front_matter);
  
    // create d-title
    $('.d-title').changeElementType('d-title');
  
    // create d-byline
    var byline = $('<d-byline></d-byline>');
    $('.d-byline').replaceWith(byline);
  
    // create d-article
    var article = $('<d-article></d-article>');
    $('.d-article').wrap(article).children().unwrap();
  
    // move posts container into article
    $('.posts-container').appendTo($('d-article'));
  
    // create d-appendix
    $('.d-appendix').changeElementType('d-appendix');
  
    // flag indicating that we have appendix items
    var appendix = $('.appendix-bottom').children('h3').length > 0;
  
    // replace footnotes with <d-footnote>
    $('.footnote-ref').each(function(i, val) {
      appendix = true;
      var href = $(this).attr('href');
      var id = href.replace('#', '');
      var fn = $('#' + id);
      var fn_p = $('#' + id + '>p');
      fn_p.find('.footnote-back').remove();
      var text = fn_p.html();
      var dtfn = $('<d-footnote></d-footnote>');
      dtfn.html(text);
      $(this).replaceWith(dtfn);
    });
    // remove footnotes
    $('.footnotes').remove();
  
    // move refs into #references-listing
    $('#references-listing').replaceWith($('#refs'));
  
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      var id = $(this).attr('id');
      $('.d-contents a[href="#' + id + '"]').parent().remove();
      appendix = true;
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
    });
  
    // show d-appendix if we have appendix content
    $("d-appendix").css('display', appendix ? 'grid' : 'none');
  
    // localize layout chunks to just output
    $('.layout-chunk').each(function(i, val) {
  
      // capture layout
      var layout = $(this).attr('data-layout');
  
      // apply layout to markdown level block elements
      var elements = $(this).children().not('details, div.sourceCode, pre, script');
      elements.each(function(i, el) {
        var layout_div = $('<div class="' + layout + '"></div>');
        if (layout_div.hasClass('shaded')) {
          var shaded_content = $('<div class="shaded-content"></div>');
          $(this).wrap(shaded_content);
          $(this).parent().wrap(layout_div);
        } else {
          $(this).wrap(layout_div);
        }
      });
  
  
      // unwrap the layout-chunk div
      $(this).children().unwrap();
    });
  
    // remove code block used to force  highlighting css
    $('.distill-force-highlighting-css').parent().remove();
  
    // remove empty line numbers inserted by pandoc when using a
    // custom syntax highlighting theme
    $('code.sourceCode a:empty').remove();
  
    // load distill framework
    load_distill_framework();
  
    // wait for window.distillRunlevel == 4 to do post processing
    function distill_post_process() {
  
      if (!window.distillRunlevel || window.distillRunlevel < 4)
        return;
  
      // hide author/affiliations entirely if we have no authors
      var front_matter = JSON.parse($("#distill-front-matter").html());
      var have_authors = front_matter.authors && front_matter.authors.length > 0;
      if (!have_authors)
        $('d-byline').addClass('hidden');
  
      // article with toc class
      $('.d-contents').parent().addClass('d-article-with-toc');
  
      // strip links that point to #
      $('.authors-affiliations').find('a[href="#"]').removeAttr('href');
  
      // add orcid ids
      $('.authors-affiliations').find('.author').each(function(i, el) {
        var orcid_id = front_matter.authors[i].orcidID;
        if (orcid_id) {
          var a = $('<a></a>');
          a.attr('href', 'https://orcid.org/' + orcid_id);
          var img = $('<img></img>');
          img.addClass('orcid-id');
          img.attr('alt', 'ORCID ID');
          img.attr('src','data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==');
          a.append(img);
          $(this).append(a);
        }
      });
  
      // hide elements of author/affiliations grid that have no value
      function hide_byline_column(caption) {
        $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
      }
  
      // affiliations
      var have_affiliations = false;
      for (var i = 0; i<front_matter.authors.length; ++i) {
        var author = front_matter.authors[i];
        if (author.affiliation !== "&nbsp;") {
          have_affiliations = true;
          break;
        }
      }
      if (!have_affiliations)
        $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');
  
      // published date
      if (!front_matter.publishedDate)
        hide_byline_column("Published");
  
      // document object identifier
      var doi = $('d-byline').find('h3:contains("DOI")');
      var doi_p = doi.next().empty();
      if (!front_matter.doi) {
        // if we have a citation and valid citationText then link to that
        if ($('#citation').length > 0 && front_matter.citationText) {
          doi.html('Citation');
          $('<a href="#citation"></a>')
            .text(front_matter.citationText)
            .appendTo(doi_p);
        } else {
          hide_byline_column("DOI");
        }
      } else {
        $('<a></a>')
           .attr('href', "https://doi.org/" + front_matter.doi)
           .html(front_matter.doi)
           .appendTo(doi_p);
      }
  
       // change plural form of authors/affiliations
      if (front_matter.authors.length === 1) {
        var grid = $('.authors-affiliations');
        grid.children('h3:contains("Authors")').text('Author');
        grid.children('h3:contains("Affiliations")').text('Affiliation');
      }
  
      // remove d-appendix and d-footnote-list local styles
      $('d-appendix > style:first-child').remove();
      $('d-footnote-list > style:first-child').remove();
  
      // move appendix-bottom entries to the bottom
      $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
      $('.appendix-bottom').remove();
  
      // hoverable references
      $('span.citation[data-cites]').each(function() {
        var refHtml = $('#ref-' + $(this).attr('data-cites')).html();
        window.tippy(this, {
          allowHTML: true,
          content: refHtml,
          maxWidth: 500,
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start'
        });
      });
  
      // clear polling timer
      clearInterval(tid);
  
      // show body now that everything is ready
      on_load_complete();
    }
  
    var tid = setInterval(distill_post_process, 50);
    distill_post_process();
  
  }
  
  function init_downlevel() {
  
    init_common();
  
     // insert hr after d-title
    $('.d-title').after($('<hr class="section-separator"/>'));
  
    // check if we have authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;
  
    // manage byline/border
    if (!have_authors)
      $('.d-byline').remove();
    $('.d-byline').after($('<hr class="section-separator"/>'));
    $('.d-byline a').remove();
  
    // remove toc
    $('.d-contents').remove();
  
    // move appendix elements
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
    });
  
  
    // inject headers into references and footnotes
    var refs_header = $('<h3></h3>');
    refs_header.text('References');
    $('#refs').prepend(refs_header);
  
    var footnotes_header = $('<h3></h3');
    footnotes_header.text('Footnotes');
    $('.footnotes').children('hr').first().replaceWith(footnotes_header);
  
    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
    $('.appendix-bottom').remove();
  
    // remove appendix if it's empty
    if ($('.d-appendix').children().length === 0)
      $('.d-appendix').remove();
  
    // prepend separator above appendix
    $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));
  
    // trim code
    $('pre>code').each(function(i, val) {
      $(this).html($.trim($(this).html()));
    });
  
    // move posts-container right before article
    $('.posts-container').insertBefore($('.d-article'));
  
    $('body').addClass('downlevel');
  
    on_load_complete();
  }
  
  
  function init_common() {
  
    // jquery plugin to change element types
    (function($) {
      $.fn.changeElementType = function(newType) {
        var attrs = {};
  
        $.each(this[0].attributes, function(idx, attr) {
          attrs[attr.nodeName] = attr.nodeValue;
        });
  
        this.replaceWith(function() {
          return $("<" + newType + "/>", attrs).append($(this).contents());
        });
      };
    })(jQuery);
  
    // prevent underline for linked images
    $('a > img').parent().css({'border-bottom' : 'none'});
  
    // mark non-body figures created by knitr chunks as 100% width
    $('.layout-chunk').each(function(i, val) {
      var figures = $(this).find('img, .html-widget');
      if ($(this).attr('data-layout') !== "l-body") {
        figures.css('width', '100%');
      } else {
        figures.css('max-width', '100%');
        figures.filter("[width]").each(function(i, val) {
          var fig = $(this);
          fig.css('width', fig.attr('width') + 'px');
        });
  
      }
    });
  
    // auto-append index.html to post-preview links in file: protocol
    // and in rstudio ide preview
    $('.post-preview').each(function(i, val) {
      if (window.location.protocol === "file:")
        $(this).attr('href', $(this).attr('href') + "index.html");
    });
  
    // get rid of index.html references in header
    if (window.location.protocol !== "file:") {
      $('.distill-site-header a[href]').each(function(i,val) {
        $(this).attr('href', $(this).attr('href').replace("index.html", "./"));
      });
    }
  
    // add class to pandoc style tables
    $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
    $('.kable-table').children('table').addClass('pandoc-table');
  
    // add figcaption style to table captions
    $('caption').parent('table').addClass("figcaption");
  
    // initialize posts list
    if (window.init_posts_list)
      window.init_posts_list();
  
    // implmement disqus comment link
    $('.disqus-comment-count').click(function() {
      window.headroom_prevent_pin = true;
      $('#disqus_thread').toggleClass('hidden');
      if (!$('#disqus_thread').hasClass('hidden')) {
        var offset = $(this).offset();
        $(window).resize();
        $('html, body').animate({
          scrollTop: offset.top - 35
        });
      }
    });
  }
  
  document.addEventListener('DOMContentLoaded', function() {
    if (is_downlevel_browser())
      init_downlevel();
    else
      window.addEventListener('WebComponentsReady', init_distill);
  });
  
  </script>
  
  <!--/radix_placeholder_distill-->
  <link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
  <script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
  <script src="site_libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
  <script src="site_libs/jquery-3.5.1/jquery.min.js"></script>
  <link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
  <script src="site_libs/datatables-binding-0.17/datatables.js"></script>
  <link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
  <link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
  <script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
  <link href="site_libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
  <script src="site_libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
  <script src="site_libs/popper-2.6.0/popper.min.js"></script>
  <link href="site_libs/tippy-6.2.7/tippy.css" rel="stylesheet" />
  <link href="site_libs/tippy-6.2.7/tippy-light-border.css" rel="stylesheet" />
  <script src="site_libs/tippy-6.2.7/tippy.umd.min.js"></script>
  <script src="site_libs/anchor-4.2.2/anchor.min.js"></script>
  <script src="site_libs/bowser-1.9.3/bowser.min.js"></script>
  <script src="site_libs/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="site_libs/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
  <!--/radix_placeholder_site_in_header-->


</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"Preprocessing the Kinkel (C086) Smchd1 LSK mini-bulk data set","authors":[{"author":"Peter Hickey","authorURL":"https://peterhickey.org","affiliation":"WEHI SCORE","affiliationURL":"https://www.wehi.edu.au/people/shalin-naik/3310/score","orcidID":""}],"publishedDate":"2021-03-31T00:00:00.000+11:00","citationText":"Hickey, 2021"}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<header class="header header--fixed" role="banner">
<nav class="distill-site-nav distill-site-header">
<div class="nav-left">
<a href="index.html" class="title">C086</a>
<a href="index.html">Introduction</a>
<a href="C086_Kinkel.preprocess.html">Preprocessing</a>
<div class="nav-dropdown">
<button class="nav-dropbtn">
Differential analyses
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<a href="C086_Kinkel.multi-sample_comparisons.reads.html">Read-based</a>
<a href="C086_Kinkel.multi-sample_comparisons.UMIs.html">UMI-based</a>
</div>
</div>
<div class="nav-dropdown">
<button class="nav-dropbtn">
Supplementary files
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<a href="../output/scPipe/">scPipe reports</a>
<a href="../output/DEGs/">Differentially expressed genes</a>
</div>
</div>
</div>
<div class="nav-right">
<a href="https://github.com/WEHISCORE/C086_Kinkel">
<i class="fa fa-github" aria-hidden="true"></i>
</a>
<a href="javascript:void(0);" class="nav-toggle">&#9776;</a>
</div>
</nav>
</header>
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>Preprocessing the Kinkel (C086) Smchd1 LSK mini-bulk data set</h1>
<!--radix_placeholder_categories-->
<!--/radix_placeholder_categories-->

</div>

<div class="d-byline">
  Peter Hickey <a href="https://peterhickey.org" class="uri">https://peterhickey.org</a> (WEHI SCORE)<a href="https://www.wehi.edu.au/people/shalin-naik/3310/score" class="uri">https://www.wehi.edu.au/people/shalin-naik/3310/score</a>
  
<br/>2021-03-31
</div>

<div class="d-article">
<div class="d-contents d-contents-float">
<nav class="l-text toc figcaption" id="TOC">
<h3>Contents</h3>
<ul>
<li><a href="#setting-up-the-data">Setting up the data</a></li>
<li><a href="#read-mapping-metrics">Read mapping metrics</a><ul>
<li><a href="#motivation">Motivation</a></li>
<li><a href="#analysis">Analysis</a></li>
<li><a href="#summary">Summary</a></li>
</ul></li>
<li><a href="#incorporating-metadata">Incorporating metadata</a><ul>
<li><a href="#incorporating-gene-based-annotation">Incorporating gene-based annotation</a></li>
<li><a href="#incorporating-sample-based-annotation">Incorporating sample-based annotation</a></li>
</ul></li>
<li><a href="#have-we-sequenced-enough">Have we sequenced enough?</a><ul>
<li><a href="#motivation-1">Motivation</a></li>
<li><a href="#analysis-1">Analysis</a></li>
<li><a href="#summary-1">Summary</a></li>
</ul></li>
<li><a href="#umi-counts-vs.-read-counts">UMI counts vs. read counts</a><ul>
<li><a href="#motivation-2">Motivation</a></li>
<li><a href="#analysis-2">Analysis</a></li>
<li><a href="#summary-2">Summary</a></li>
</ul></li>
<li><a href="#quality-control-of-samples">Quality control of samples</a><ul>
<li><a href="#defining-the-quality-control-metrics">Defining the quality control metrics</a></li>
<li><a href="#visualizing-the-qc-metrics">Visualizing the QC metrics</a></li>
<li><a href="#identifying-outliers-by-each-metric">Identifying outliers by each metric</a></li>
<li><a href="#summary-3">Summary</a></li>
</ul></li>
<li><a href="#examining-gene-level-metrics">Examining gene-level metrics</a><ul>
<li><a href="#inspecting-the-most-highly-expressed-genes">Inspecting the most highly expressed genes</a></li>
<li><a href="#sex-check-using-xist">Sex check using <em>Xist</em></a></li>
<li><a href="#genotype-check-using-smchd1">Genotype check using <em>Smchd1</em></a></li>
</ul></li>
<li><a href="#concluding-remarks">Concluding remarks</a></li>
<li><a href="#additional-information">Additional information</a></li>
<li><a href="#session-info">Session info</a></li>
</ul>
</nav>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'>SingleCellExperiment</span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://here.r-lib.org/'>here</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/rstudio/rmarkdown'>rmarkdown</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://ggplot2.tidyverse.org'>ggplot2</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://patchwork.data-imaginist.com'>patchwork</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/Bioconductor/BiocParallel'>BiocParallel</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='http://bioconductor.org/packages/scater/'>scater</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://wilkelab.org/cowplot/'>cowplot</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='http://bioinf.wehi.edu.au/limma'>limma</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://scales.r-lib.org'>scales</a></span><span class='op'>)</span>

<span class='co'># NOTE: Using &gt;= 4 cores seizes up my laptop. Can use more on RStudio server.</span>
<span class='fu'><a href='https://rdrr.io/r/base/options.html'>options</a></span><span class='op'>(</span>
  <span class='st'>"mc.cores"</span> <span class='op'>=</span> 
    <span class='fu'><a href='https://rdrr.io/r/base/ifelse.html'>ifelse</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/Sys.info.html'>Sys.info</a></span><span class='op'>(</span><span class='op'>)</span><span class='op'>[[</span><span class='st'>"nodename"</span><span class='op'>]</span><span class='op'>]</span> <span class='op'>==</span> <span class='st'>"rstudio-1.hpc.wehi.edu.au"</span>, <span class='fl'>8L</span>, <span class='fl'>2L</span><span class='op'>)</span><span class='op'>)</span>

<span class='fu'><a href='https://rdrr.io/pkg/BiocParallel/man/register.html'>register</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/BiocParallel/man/MulticoreParam-class.html'>MulticoreParam</a></span><span class='op'>(</span>workers <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/options.html'>getOption</a></span><span class='op'>(</span><span class='st'>"mc.cores"</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>

<span class='fu'>knitr</span><span class='fu'>::</span><span class='va'><a href='https://rdrr.io/pkg/knitr/man/opts_chunk.html'>opts_chunk</a></span><span class='op'>$</span><span class='fu'>set</span><span class='op'>(</span>fig.path <span class='op'>=</span> <span class='st'>"C086_Kinkel.preprocess_files/"</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<h1 id="setting-up-the-data">Setting up the data</h1>
<p>We start from the count matrices created in <a href="../code/scPipe.R%60"><code>code/scPipe.R</code></a>, specifically the <a href="../data/SCEs/C086_Kinkel.UMI_deduped.scPipe.SCE.rds">UMI counts (a.k.a. UMI-deduplicated) data</a> and <a href="../data/SCEs/C086_Kinkel.not_UMI_deduped.scPipe.SCE.rds">read counts (a.k.a. not UMI-deduplicated) data</a>.</p>
<aside>
UMI stands for unique molecular identifier.
</aside>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>sce_deduped</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/readRDS.html'>readRDS</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://here.r-lib.org//reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"data"</span>, <span class='st'>"SCEs"</span>, <span class='st'>"C086_Kinkel.UMI_deduped.scPipe.SCE.rds"</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>sce_not_deduped</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/readRDS.html'>readRDS</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://here.r-lib.org//reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"data"</span>, <span class='st'>"SCEs"</span>, <span class='st'>"C086_Kinkel.not_UMI_deduped.scPipe.SCE.rds"</span><span class='op'>)</span><span class='op'>)</span>

<span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>sce_not_deduped</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/grep.html'>sub</a></span><span class='op'>(</span>
  <span class='st'>"\\.not_UMI_deduped"</span>,
  <span class='st'>""</span>,
  <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>sce_not_deduped</span><span class='op'>)</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/stopifnot.html'>stopifnot</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/base/identical.html'>identical</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>sce_deduped</span><span class='op'>)</span>, <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>sce_not_deduped</span><span class='op'>)</span><span class='op'>)</span>,
  <span class='fu'><a href='https://rdrr.io/r/base/identical.html'>identical</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>sce_deduped</span><span class='op'>)</span>, <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>sce_not_deduped</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span> 

<span class='co'># Combine UMI and read counts a single SCE.</span>
<span class='va'>sce</span> <span class='op'>&lt;-</span> <span class='va'>sce_deduped</span>
<span class='fu'>assay</span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"UMI_counts"</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'>assay</span><span class='op'>(</span><span class='va'>sce_deduped</span>, <span class='st'>"counts"</span><span class='op'>)</span>
<span class='fu'>assay</span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"read_counts"</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'>assay</span><span class='op'>(</span><span class='va'>sce_not_deduped</span>, <span class='st'>"counts"</span><span class='op'>)</span>
<span class='co'># NOTE: Nullify some now-unrequired data.</span>
<span class='fu'>assay</span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"counts"</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='cn'>NULL</span>
<span class='va'>sce</span><span class='op'>$</span><span class='va'>UMI_deduped</span> <span class='op'>&lt;-</span> <span class='cn'>NULL</span>
</code></pre>
</div>
</details>
</div>
<p>We combine these into a single object<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a> containing both the UMI counts and the read counts for 25587 genes and 36 samples.</p>
<h1 id="read-mapping-metrics">Read mapping metrics</h1>
<h2 id="motivation">Motivation</h2>
<p>We aimed to sequence 5 million reads/sample and hoped that the majority of these reads would be exonic.</p>
<h2 id="analysis">Analysis</h2>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/LuyiTian/scPipe'>scPipe</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://tidyr.tidyverse.org'>tidyr</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://dplyr.tidyverse.org'>dplyr</a></span><span class='op'>)</span>

<span class='va'>qc_df</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/scPipe/man/QC_metrics.html'>QC_metrics</a></span><span class='op'>(</span><span class='va'>sce_deduped</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>tibble</span><span class='fu'>::</span><span class='fu'><a href='https://tibble.tidyverse.org/reference/rownames.html'>rownames_to_column</a></span><span class='op'>(</span>var <span class='op'>=</span> <span class='st'>"rowname"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://tidyr.tidyverse.org/reference/pivot_longer.html'>pivot_longer</a></span><span class='op'>(</span><span class='op'>!</span><span class='va'>rowname</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate.html'>mutate</a></span><span class='op'>(</span>
    well_position <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/grep.html'>sub</a></span><span class='op'>(</span><span class='st'>"LCE[0-9]+_"</span>, <span class='st'>""</span>, <span class='va'>rowname</span><span class='op'>)</span>,
    plate_number <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/substr.html'>substr</a></span><span class='op'>(</span><span class='va'>rowname</span>, <span class='fl'>1</span>, <span class='fl'>6</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate-joins.html'>inner_join</a></span><span class='op'>(</span>
    <span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce_deduped</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
      <span class='fu'>tibble</span><span class='fu'>::</span><span class='fu'><a href='https://tibble.tidyverse.org/reference/rownames.html'>rownames_to_column</a></span><span class='op'>(</span>var <span class='op'>=</span> <span class='st'>"rowname"</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate.html'>mutate</a></span><span class='op'>(</span>well_position <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span>
    <span class='va'>well_position</span>,
    <span class='fu'><a href='https://rdrr.io/r/base/unlist.html'>unlist</a></span><span class='op'>(</span>
      <span class='fu'><a href='https://rdrr.io/r/base/lapply.html'>lapply</a></span><span class='op'>(</span><span class='va'>LETTERS</span><span class='op'>[</span><span class='fl'>1</span><span class='op'>:</span><span class='fl'>16</span><span class='op'>]</span>, <span class='kw'>function</span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='va'>x</span>, <span class='fl'>1</span><span class='op'>:</span><span class='fl'>24</span><span class='op'>)</span><span class='op'>)</span>,
      use.names <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>Figure <a href="#fig:nreads">1</a> shows that we generated more than 5 million reads for the majority of samples and that most of these reads are exonic. As expected, the negative control sample generated very few reads.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>mapped_category_colours</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/setNames.html'>setNames</a></span><span class='op'>(</span>
  <span class='fu'>RColorBrewer</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html'>brewer.pal</a></span><span class='op'>(</span>
    <span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/unique.html'>unique</a></span><span class='op'>(</span><span class='va'>qc_df</span><span class='op'>$</span><span class='va'>name</span><span class='op'>)</span><span class='op'>)</span>, 
    <span class='st'>"Dark2"</span><span class='op'>)</span>,
  <span class='fu'><a href='https://rdrr.io/r/base/sort.html'>sort</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/unique.html'>unique</a></span><span class='op'>(</span><span class='va'>qc_df</span><span class='op'>$</span><span class='va'>name</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>
<span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='va'>qc_df</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_bar.html'>geom_col</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>well_position</span>, fill <span class='op'>=</span> <span class='va'>name</span>, y <span class='op'>=</span> <span class='va'>value</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>cowplot</span><span class='fu'>::</span><span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Number of reads"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Well position"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_abline.html'>geom_hline</a></span><span class='op'>(</span>yintercept <span class='op'>=</span> <span class='fl'>5e6</span>, lty <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_fill_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>mapped_category_colours</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/facet_grid.html'>facet_grid</a></span><span class='op'>(</span><span class='va'>sample_name</span> <span class='op'>~</span> <span class='va'>.</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span><span class='op'>(</span>
    <span class='st'>"C086_Kinkel: Read mapping metrics"</span>,
    subtitle <span class='op'>=</span> <span class='st'>"Number of reads per category"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_discrete.html'>scale_x_discrete</a></span><span class='op'>(</span>guide <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guide_axis.html'>guide_axis</a></span><span class='op'>(</span>n.dodge <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/theme.html'>theme</a></span><span class='op'>(</span>axis.text.x <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/element.html'>element_text</a></span><span class='op'>(</span>size <span class='op'>=</span> <span class='fl'>6</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>fill <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guide_legend.html'>guide_legend</a></span><span class='op'>(</span>title <span class='op'>=</span> <span class='st'>"Mapping category"</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:nreads"></span>
<img src="C086_Kinkel.preprocess_files/nreads-1.png" alt="Number of reads generated per sample. The dashed horizional line denotes 5 million reads." width="624" />
<p class="caption">
Figure 1: Number of reads generated per sample. The dashed horizional line denotes 5 million reads.
</p>
</div>
</div>
<p>Figure <a href="#fig:prop-reads">2</a> shows for each sample that around 60% of reads are exonic and a further 20% are intronic.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='va'>qc_df</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_bar.html'>geom_col</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>well_position</span>, fill <span class='op'>=</span> <span class='va'>name</span>, y <span class='op'>=</span> <span class='va'>value</span><span class='op'>)</span>, position <span class='op'>=</span> <span class='st'>"fill"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>cowplot</span><span class='fu'>::</span><span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Proportion of reads"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Well position"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_fill_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>mapped_category_colours</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/facet_grid.html'>facet_grid</a></span><span class='op'>(</span><span class='va'>plate_number</span> <span class='op'>+</span> <span class='va'>sample_name</span> <span class='op'>~</span> <span class='va'>.</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span><span class='op'>(</span>
    <span class='st'>"C086_Kinkel: Read mapping metrics"</span>,
    subtitle <span class='op'>=</span> <span class='st'>"Proportion of reads per category"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_discrete.html'>scale_x_discrete</a></span><span class='op'>(</span>guide <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guide_axis.html'>guide_axis</a></span><span class='op'>(</span>n.dodge <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/theme.html'>theme</a></span><span class='op'>(</span>axis.text.x <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/element.html'>element_text</a></span><span class='op'>(</span>size <span class='op'>=</span> <span class='fl'>6</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>fill <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guide_legend.html'>guide_legend</a></span><span class='op'>(</span>title <span class='op'>=</span> <span class='st'>"Mapping category"</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:prop-reads"></span>
<img src="C086_Kinkel.preprocess_files/prop-reads-1.png" alt="Proportion of reads generated in each mapping category per sample." width="624" />
<p class="caption">
Figure 2: Proportion of reads generated in each mapping category per sample.
</p>
</div>
</div>
<p>Figure <a href="#fig:nreads-exonic-intronic">3</a> shows for each sample that on average 4 million reads are exonic and a further 1 million are intronic</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>qc_df</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_boxplot.html'>geom_boxplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>name</span>, y <span class='op'>=</span> <span class='va'>value</span>, fill <span class='op'>=</span> <span class='va'>name</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>cowplot</span><span class='fu'>::</span><span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>12</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Number of reads"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Mapping category"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_fill_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>mapped_category_colours</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span><span class='op'>(</span>
    <span class='st'>"C086_Kinkel: Number of reads per category"</span>,
    subtitle <span class='op'>=</span> <span class='st'>"Includes negative control sample"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>fill <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_discrete.html'>scale_x_discrete</a></span><span class='op'>(</span>guide <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guide_axis.html'>guide_axis</a></span><span class='op'>(</span>angle <span class='op'>=</span> <span class='fl'>45</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:nreads-exonic-intronic"></span>
<img src="C086_Kinkel.preprocess_files/nreads-exonic-intronic-1.png" alt="Number of exonic and intronic reads generated per sample." width="624" />
<p class="caption">
Figure 3: Number of exonic and intronic reads generated per sample.
</p>
</div>
</div>
<h2 id="summary">Summary</h2>
<p>We largely achieved our aim of sequencing ~5 million reads/sample and with the majority of reads being exonic. <strong>Henceforth, we only use the exonic reads.</strong> Although the intronic reads are potentially useful, to do so would require substantial changes to the data analysis workflow and additional analysis.</p>
<p>We now remove the negative control sample.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>sce</span> <span class='op'>&lt;-</span> <span class='va'>sce</span><span class='op'>[</span>, <span class='va'>sce</span><span class='op'>$</span><span class='va'>sample_name</span> <span class='op'>!=</span> <span class='st'>"negative control"</span><span class='op'>]</span>
<span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/droplevels.html'>droplevels</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>We look more closely at which samples have smaller numbers of exonic reads in <a href="#have-we-sequenced-enough">Have we sequenced enough?</a> and <a href="#quality-control-of-samples">Quality control of samples</a>.</p>
<h1 id="incorporating-metadata">Incorporating metadata</h1>
<h2 id="incorporating-gene-based-annotation">Incorporating gene-based annotation</h2>
<p>We obtain gene-based annotations from the NCBI/RefSeq and Ensembl databases, such as the chromosome and gene symbol, using the <em><a href="https://bioconductor.org/packages/3.12/Mus.musculus">Mus.musculus</a></em> and <em><a href="https://bioconductor.org/packages/3.12/EnsDb.Mmusculus.v79">EnsDb.Mmusculus.v79</a></em> packages.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='co'># Extract rownames (Ensembl IDs) to use as key in database lookups.</span>
<span class='va'>ensembl</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span>

<span class='co'># Pull out useful gene-based annotations from the Ensembl-based database.</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'>EnsDb.Mmusculus.v79</span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/jorainer/ensembldb'>ensembldb</a></span><span class='op'>)</span>
<span class='co'># NOTE: These columns were customised for this project.</span>
<span class='va'>ensdb_columns</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span>
  <span class='st'>"GENEBIOTYPE"</span>, <span class='st'>"GENENAME"</span>, <span class='st'>"GENESEQSTART"</span>, <span class='st'>"GENESEQEND"</span>, <span class='st'>"SEQNAME"</span>, <span class='st'>"SYMBOL"</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>ensdb_columns</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='st'>"ENSEMBL."</span>, <span class='va'>ensdb_columns</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/stopifnot.html'>stopifnot</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/all.html'>all</a></span><span class='op'>(</span><span class='va'>ensdb_columns</span> <span class='op'>%in%</span> <span class='fu'>columns</span><span class='op'>(</span><span class='va'>EnsDb.Mmusculus.v79</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>ensdb_df</span> <span class='op'>&lt;-</span> <span class='fu'>DataFrame</span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/base/lapply.html'>lapply</a></span><span class='op'>(</span><span class='va'>ensdb_columns</span>, <span class='kw'>function</span><span class='op'>(</span><span class='va'>column</span><span class='op'>)</span> <span class='op'>{</span>
    <span class='fu'>mapIds</span><span class='op'>(</span>
      x <span class='op'>=</span> <span class='va'>EnsDb.Mmusculus.v79</span>, 
      <span class='co'># NOTE: Need to remove gene version number prior to lookup.</span>
      keys <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/grep.html'>gsub</a></span><span class='op'>(</span><span class='st'>"\\.[0-9]+$"</span>, <span class='st'>""</span>, <span class='va'>ensembl</span><span class='op'>)</span>,
      keytype <span class='op'>=</span> <span class='st'>"GENEID"</span>,
      column <span class='op'>=</span> <span class='va'>column</span>,
      multiVals <span class='op'>=</span> <span class='st'>"CharacterList"</span><span class='op'>)</span>
  <span class='op'>}</span><span class='op'>)</span>,
  row.names <span class='op'>=</span> <span class='va'>ensembl</span><span class='op'>)</span>
<span class='co'># NOTE: Can't look up GENEID column with GENEID key, so have to add manually.</span>
<span class='va'>ensdb_df</span><span class='op'>$</span><span class='va'>ENSEMBL.GENEID</span> <span class='op'>&lt;-</span> <span class='va'>ensembl</span>

<span class='co'># NOTE: Mus.musculus combines org.Mm.eg.db and</span>
<span class='co'>#       TxDb.Mmusculus.UCSC.mm10.knownGene (as well as others) and therefore </span>
<span class='co'>#       uses entrez gene and RefSeq based data.</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'>Mus.musculus</span><span class='op'>)</span>
<span class='co'># NOTE: These columns were customised for this project.</span>
<span class='va'>ncbi_columns</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span>
  <span class='co'># From TxDB: None required</span>
  <span class='co'># From OrgDB</span>
  <span class='st'>"ALIAS"</span>, <span class='st'>"ENTREZID"</span>, <span class='st'>"GENENAME"</span>, <span class='st'>"REFSEQ"</span>, <span class='st'>"SYMBOL"</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>ncbi_columns</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='st'>"NCBI."</span>, <span class='va'>ncbi_columns</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/stopifnot.html'>stopifnot</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/all.html'>all</a></span><span class='op'>(</span><span class='va'>ncbi_columns</span> <span class='op'>%in%</span> <span class='fu'>columns</span><span class='op'>(</span><span class='va'>Mus.musculus</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>ncbi_df</span> <span class='op'>&lt;-</span> <span class='fu'>DataFrame</span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/base/lapply.html'>lapply</a></span><span class='op'>(</span><span class='va'>ncbi_columns</span>, <span class='kw'>function</span><span class='op'>(</span><span class='va'>column</span><span class='op'>)</span> <span class='op'>{</span>
    <span class='fu'>mapIds</span><span class='op'>(</span>
      x <span class='op'>=</span> <span class='va'>Mus.musculus</span>, 
      <span class='co'># NOTE: Need to remove gene version number prior to lookup.</span>
      keys <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/grep.html'>gsub</a></span><span class='op'>(</span><span class='st'>"\\.[0-9]+$"</span>, <span class='st'>""</span>, <span class='va'>ensembl</span><span class='op'>)</span>,
      keytype <span class='op'>=</span> <span class='st'>"ENSEMBL"</span>,
      column <span class='op'>=</span> <span class='va'>column</span>,
      multiVals <span class='op'>=</span> <span class='st'>"CharacterList"</span><span class='op'>)</span>
  <span class='op'>}</span><span class='op'>)</span>,
  row.names <span class='op'>=</span> <span class='va'>ensembl</span><span class='op'>)</span>

<span class='fu'>rowData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/limma/man/cbind.html'>cbind</a></span><span class='op'>(</span><span class='va'>ensdb_df</span>, <span class='va'>ncbi_df</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>Having quantified gene expression against the GENCODE gene annotation, we have Ensembl-style identifiers for the genes. These identifiers are used as they are unambiguous and highly stable. However, they are difficult to interpret compared to the gene symbols which are more commonly used in the literature. Henceforth, we will use gene symbols (where available) to refer to genes in our analysis and otherwise use the Ensembl-style gene identifiers<a href="#fn2" class="footnote-ref" id="fnref2" role="doc-noteref"><sup>2</sup></a>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='co'># Replace the row names of the SCE by the gene symbols (where available).</span>
<span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scuttle/man/uniquifyFeatureNames.html'>uniquifyFeatureNames</a></span><span class='op'>(</span>
  ID <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span>, 
  <span class='co'># NOTE: An Ensembl ID may map to 0, 1, 2, 3, ... gene symbols.</span>
  <span class='co'>#       When there are multiple matches only the 1st match is used.</span>
  names <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/lapply.html'>sapply</a></span><span class='op'>(</span><span class='fu'>rowData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>$</span><span class='va'>ENSEMBL.SYMBOL</span>, <span class='kw'>function</span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span> <span class='op'>{</span>
    <span class='kw'>if</span> <span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>{</span>
      <span class='va'>x</span><span class='op'>[[</span><span class='fl'>1</span><span class='op'>]</span><span class='op'>]</span>
    <span class='op'>}</span> <span class='kw'>else</span> <span class='op'>{</span>
      <span class='cn'>NA_character_</span>
    <span class='op'>}</span>
  <span class='op'>}</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='co'># Some useful gene sets</span>
<span class='va'>mito_set</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/any.html'>any</a></span><span class='op'>(</span><span class='fu'>rowData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>$</span><span class='va'>ENSEMBL.SEQNAME</span> <span class='op'>==</span> <span class='st'>"MT"</span><span class='op'>)</span><span class='op'>]</span>
<span class='va'>ribo_set</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/grep.html'>grep</a></span><span class='op'>(</span><span class='st'>"^Rp(s|l)"</span>, <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span>, value <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
<span class='co'># NOTE: A more curated approach for identifying ribosomal protein genes </span>
<span class='co'>#       (https://github.com/Bioconductor/OrchestratingSingleCellAnalysis-base/blob/ae201bf26e3e4fa82d9165d8abf4f4dc4b8e5a68/feature-selection.Rmd#L376-L380)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://igordot.github.io/msigdbr'>msigdbr</a></span><span class='op'>)</span>
<span class='va'>c2_sets</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://igordot.github.io/msigdbr/reference/msigdbr.html'>msigdbr</a></span><span class='op'>(</span>species <span class='op'>=</span> <span class='st'>"Mus musculus"</span>, category <span class='op'>=</span> <span class='st'>"C2"</span><span class='op'>)</span>
<span class='va'>ribo_set</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://generics.r-lib.org/reference/setops.html'>union</a></span><span class='op'>(</span>
  <span class='va'>ribo_set</span>,
  <span class='va'>c2_sets</span><span class='op'>[</span><span class='va'>c2_sets</span><span class='op'>$</span><span class='va'>gs_name</span> <span class='op'>==</span> <span class='st'>"KEGG_RIBOSOME"</span>, <span class='op'>]</span><span class='op'>$</span><span class='va'>gene_symbol</span><span class='op'>)</span>
<span class='va'>sex_set</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/any.html'>any</a></span><span class='op'>(</span><span class='fu'>rowData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>$</span><span class='va'>ENSEMBL.SEQNAME</span> <span class='op'>%in%</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"X"</span>, <span class='st'>"Y"</span><span class='op'>)</span><span class='op'>)</span><span class='op'>]</span>
<span class='va'>pseudogene_set</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span>
  <span class='fu'><a href='https://rdrr.io/r/base/any.html'>any</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/grep.html'>grepl</a></span><span class='op'>(</span><span class='st'>"pseudogene"</span>, <span class='fu'>rowData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>$</span><span class='va'>ENSEMBL.GENEBIOTYPE</span><span class='op'>)</span><span class='op'>)</span><span class='op'>]</span>
</code></pre>
</div>
</details>
</div>
<h2 id="incorporating-sample-based-annotation">Incorporating sample-based annotation</h2>
<p>Sarah provided various sample-based metadata about the samples such as the sex, genotype, date of birth, and date of death of each mouse.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='co'># Add sample metadata</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://readxl.tidyverse.org'>readxl</a></span><span class='op'>)</span>

<span class='va'>sample_sheet</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://readxl.tidyverse.org/reference/read_excel.html'>read_excel</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://here.r-lib.org//reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"data"</span>, <span class='st'>"sample_sheets"</span>, <span class='st'>"Minibulk_RNAseq_samples.xlsx"</span><span class='op'>)</span>,
  sheet <span class='op'>=</span> <span class='st'>"Sheet1"</span>,
  range <span class='op'>=</span> <span class='st'>"A3:I15"</span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/sfirke/janitor'>janitor</a></span><span class='op'>)</span>
<span class='va'>sample_sheet</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/janitor/man/clean_names.html'>clean_names</a></span><span class='op'>(</span><span class='va'>sample_sheet</span><span class='op'>)</span>
<span class='co'># Some manual tweaks</span>
<span class='va'>sample_sheet</span><span class='op'>$</span><span class='va'>mouse_number</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/grep.html'>sub</a></span><span class='op'>(</span><span class='st'>"#"</span>, <span class='st'>""</span>, <span class='va'>sample_sheet</span><span class='op'>$</span><span class='va'>mouse_number</span><span class='op'>)</span>
<span class='va'>sample_sheet</span><span class='op'>$</span><span class='va'>genotype.mouse</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='va'>sample_sheet</span><span class='op'>$</span><span class='va'>smchd1_genotype_updated</span>, <span class='st'>"."</span>, <span class='va'>sample_sheet</span><span class='op'>$</span><span class='va'>mouse_number</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>Sarah sent me a subsequent email (2021-02-22) in which she reported some updated genotyping results, which showed that:</p>
<blockquote>
<p>Given the latest genotyping results we have only two male Smchd1-del biological replicates. Both #1269 and #1271 were initially incorrectly genotyped as having the Vav-cre transgene but on repeat genotyping of BM cells we found that they do not, which is why there has been a failure to delete the Smchd1 floxed allele in these samples. This means that #1269 is effectively Smchd1-het and #1271 is WT. After consultation with Marnie, we think it could be useful to include #1269 as a Smchd1-het, although it is lacking the Vav-cre transgene so it may/may not look the same as the other two Smchd1-het males due to this. Also, I’d be keen to know how similar the WT is to the Smchd1-hets. I know there is only one replicate, but we are using the Smchd1-hets as our controls in these experiments, so it would be nice to see how they compare to an actual WT in this context.</p>
</blockquote>
<p>These sample-based annotations are included in the <em>colData</em> of the <em>SingleCellExperiment</em>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='co'># Join with SCE</span>
<span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate-joins.html'>left_join</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>)</span>,
  <span class='va'>sample_sheet</span>,
  by <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"mouse_number"</span> <span class='op'>=</span> <span class='st'>"mouse_number"</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>DataFrame</span><span class='op'>(</span><span class='va'>.</span>, row.names <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>)</span>

<span class='co'># Make some key columns factors</span>
<span class='va'>sce</span><span class='op'>$</span><span class='va'>smchd1_genotype_updated</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>smchd1_genotype_updated</span><span class='op'>)</span>
<span class='va'>sce</span><span class='op'>$</span><span class='va'>sex</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>sex</span><span class='op'>)</span>
<span class='va'>sce</span><span class='op'>$</span><span class='va'>mouse_number</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>mouse_number</span><span class='op'>)</span>

<span class='va'>sce</span><span class='op'>$</span><span class='va'>genotype.mouse</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>smchd1_genotype_updated</span>, <span class='st'>"."</span>, <span class='va'>sce</span><span class='op'>$</span><span class='va'>mouse_number</span><span class='op'>)</span><span class='op'>)</span>
<span class='co'># Convoluted way to add tech rep ID</span>
<span class='va'>sce</span><span class='op'>$</span><span class='va'>rep</span> <span class='op'>&lt;-</span> <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/group_by.html'>group_by</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>)</span>, <span class='va'>genotype.mouse</span><span class='op'>)</span> <span class='op'>%&gt;%</span> 
    <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate.html'>mutate</a></span><span class='op'>(</span>rep <span class='op'>=</span> <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/ranking.html'>row_number</a></span><span class='op'>(</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/pull.html'>pull</a></span><span class='op'>(</span><span class='va'>rep</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>genotype.mouse</span>, <span class='st'>"."</span>, <span class='va'>sce</span><span class='op'>$</span><span class='va'>rep</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>Sarah aimed for technical triplicates of each sample as a 100-cell mini-bulk, but this was not always achievable, as shown in the table below.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>x</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/count.html'>count</a></span><span class='op'>(</span>
    <span class='va'>smchd1_genotype_updated</span>,
    <span class='va'>genotyping_bm_cells</span>,
    <span class='va'>sex</span>,
    <span class='va'>mouse_number</span>,
    <span class='va'>dob</span>,
    <span class='va'>dod</span>,
    <span class='va'>age_days</span>,
    <span class='va'>sample_type</span><span class='op'>)</span>

<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/rstudio/gt'>gt</a></span><span class='op'>)</span>
<span class='va'>x</span> <span class='op'>%&gt;%</span>
  <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/group_by.html'>group_by</a></span><span class='op'>(</span><span class='va'>smchd1_genotype_updated</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/pkg/gt/man/gt.html'>gt</a></span><span class='op'>(</span>rowname_col <span class='op'>=</span> <span class='st'>"mouse_number"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/pkg/gt/man/tab_spanner.html'>tab_spanner</a></span><span class='op'>(</span>
    label <span class='op'>=</span> <span class='st'>"Sample metadata"</span>,
    columns <span class='op'>=</span> <span class='fu'><a href='https://dplyr.tidyverse.org/reference/vars.html'>vars</a></span><span class='op'>(</span><span class='va'>genotyping_bm_cells</span>, <span class='va'>sex</span>, <span class='va'>dob</span>, <span class='va'>dod</span>, <span class='va'>age_days</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/pkg/gt/man/tab_header.html'>tab_header</a></span><span class='op'>(</span>
    title <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/gt/man/md.html'>md</a></span><span class='op'>(</span><span class='st'>"**Samples in C086_Kinkel dataset**"</span><span class='op'>)</span>,
    subtitle <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/gt/man/md.html'>md</a></span><span class='op'>(</span><span class='st'>"*Rownames are mouse number IDs*"</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/pkg/gt/man/tab_source_note.html'>tab_source_note</a></span><span class='op'>(</span>source_note <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/gt/man/md.html'>md</a></span><span class='op'>(</span><span class='st'>"Source: [`data/sample_sheets/Minibulk_RNAseq_samples.xlsx`](../data/sample_sheets/Minibulk_RNAseq_samples.xlsx)"</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span> 
  <span class='fu'><a href='https://rdrr.io/pkg/gt/man/summary_rows.html'>summary_rows</a></span><span class='op'>(</span>
    groups <span class='op'>=</span> <span class='cn'>TRUE</span>,
    columns <span class='op'>=</span> <span class='fu'><a href='https://dplyr.tidyverse.org/reference/vars.html'>vars</a></span><span class='op'>(</span><span class='va'>n</span><span class='op'>)</span>,
    fns <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>total <span class='op'>=</span> <span class='st'>"sum"</span><span class='op'>)</span>,
    formatter <span class='op'>=</span> <span class='va'>fmt_number</span>,
    decimals <span class='op'>=</span> <span class='fl'>0</span><span class='op'>)</span> 
</code></pre>
</div>
</details>
<style>html {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Helvetica Neue', 'Fira Sans', 'Droid Sans', Arial, sans-serif;
}

#szcvlsmmen .gt_table {
  display: table;
  border-collapse: collapse;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#szcvlsmmen .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#szcvlsmmen .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#szcvlsmmen .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 0;
  padding-bottom: 4px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#szcvlsmmen .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#szcvlsmmen .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#szcvlsmmen .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#szcvlsmmen .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#szcvlsmmen .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#szcvlsmmen .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#szcvlsmmen .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#szcvlsmmen .gt_group_heading {
  padding: 8px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
}

#szcvlsmmen .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#szcvlsmmen .gt_from_md > :first-child {
  margin-top: 0;
}

#szcvlsmmen .gt_from_md > :last-child {
  margin-bottom: 0;
}

#szcvlsmmen .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#szcvlsmmen .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 12px;
}

#szcvlsmmen .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#szcvlsmmen .gt_first_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
}

#szcvlsmmen .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#szcvlsmmen .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#szcvlsmmen .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#szcvlsmmen .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#szcvlsmmen .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#szcvlsmmen .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding: 4px;
}

#szcvlsmmen .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#szcvlsmmen .gt_sourcenote {
  font-size: 90%;
  padding: 4px;
}

#szcvlsmmen .gt_left {
  text-align: left;
}

#szcvlsmmen .gt_center {
  text-align: center;
}

#szcvlsmmen .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#szcvlsmmen .gt_font_normal {
  font-weight: normal;
}

#szcvlsmmen .gt_font_bold {
  font-weight: bold;
}

#szcvlsmmen .gt_font_italic {
  font-style: italic;
}

#szcvlsmmen .gt_super {
  font-size: 65%;
}

#szcvlsmmen .gt_footnote_marks {
  font-style: italic;
  font-size: 65%;
}
</style>
<div id="szcvlsmmen" style="overflow-x:auto;overflow-y:auto;width:auto;height:auto;"><table class="gt_table">
  <thead class="gt_header">
    <tr>
      <th colspan="8" class="gt_heading gt_title gt_font_normal" style><strong>Samples in C086_Kinkel dataset</strong></th>
    </tr>
    <tr>
      <th colspan="8" class="gt_heading gt_subtitle gt_font_normal gt_bottom_border" style><em>Rownames are mouse number IDs</em></th>
    </tr>
  </thead>
  <thead class="gt_col_headings">
    <tr>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="2" colspan="1"></th>
      <th class="gt_center gt_columns_top_border gt_column_spanner_outer" rowspan="1" colspan="5">
        <span class="gt_column_spanner">Sample metadata</span>
      </th>
      <th class="gt_col_heading gt_center gt_columns_bottom_border" rowspan="2" colspan="1">sample_type</th>
      <th class="gt_col_heading gt_center gt_columns_bottom_border" rowspan="2" colspan="1">n</th>
    </tr>
    <tr>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1">genotyping_bm_cells</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1">sex</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1">dob</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1">dod</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1">age_days</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr class="gt_group_heading_row">
      <td colspan="8" class="gt_group_heading">Del</td>
    </tr>
    <tr>
      <td class="gt_row gt_left gt_stub">1253</td>
      <td class="gt_row gt_left">del/del</td>
      <td class="gt_row gt_center">F</td>
      <td class="gt_row gt_left">184/20</td>
      <td class="gt_row gt_left">344/20</td>
      <td class="gt_row gt_right">160</td>
      <td class="gt_row gt_left">MiniBulk 100 cells</td>
      <td class="gt_row gt_center">3</td>
    </tr>
    <tr>
      <td class="gt_row gt_left gt_stub">1266</td>
      <td class="gt_row gt_left">del/del</td>
      <td class="gt_row gt_center">F</td>
      <td class="gt_row gt_left">207/20</td>
      <td class="gt_row gt_left">344/20</td>
      <td class="gt_row gt_right">137</td>
      <td class="gt_row gt_left">MiniBulk 100 cells</td>
      <td class="gt_row gt_center">2</td>
    </tr>
    <tr>
      <td class="gt_row gt_left gt_stub">1266</td>
      <td class="gt_row gt_left">del/del</td>
      <td class="gt_row gt_center">F</td>
      <td class="gt_row gt_left">207/20</td>
      <td class="gt_row gt_left">344/20</td>
      <td class="gt_row gt_right">137</td>
      <td class="gt_row gt_left">MiniBulk 58 cells</td>
      <td class="gt_row gt_center">1</td>
    </tr>
    <tr>
      <td class="gt_row gt_left gt_stub">1283</td>
      <td class="gt_row gt_left">del/del</td>
      <td class="gt_row gt_center">F</td>
      <td class="gt_row gt_left">258/20</td>
      <td class="gt_row gt_left">344/20</td>
      <td class="gt_row gt_right">86</td>
      <td class="gt_row gt_left">MiniBulk 100 cells</td>
      <td class="gt_row gt_center">1</td>
    </tr>
    <tr>
      <td class="gt_row gt_left gt_stub">1283</td>
      <td class="gt_row gt_left">del/del</td>
      <td class="gt_row gt_center">F</td>
      <td class="gt_row gt_left">258/20</td>
      <td class="gt_row gt_left">344/20</td>
      <td class="gt_row gt_right">86</td>
      <td class="gt_row gt_left">MiniBulk 50 cells</td>
      <td class="gt_row gt_center">1</td>
    </tr>
    <tr>
      <td class="gt_row gt_left gt_stub">1250</td>
      <td class="gt_row gt_left">del/del</td>
      <td class="gt_row gt_center">M</td>
      <td class="gt_row gt_left">165/20</td>
      <td class="gt_row gt_left">344/20</td>
      <td class="gt_row gt_right">178</td>
      <td class="gt_row gt_left">MiniBulk 100 cells</td>
      <td class="gt_row gt_center">3</td>
    </tr>
    <tr>
      <td class="gt_row gt_left gt_stub">1282</td>
      <td class="gt_row gt_left">del/del</td>
      <td class="gt_row gt_center">M</td>
      <td class="gt_row gt_left">258/20</td>
      <td class="gt_row gt_left">344/20</td>
      <td class="gt_row gt_right">86</td>
      <td class="gt_row gt_left">MiniBulk 100 cells</td>
      <td class="gt_row gt_center">3</td>
    </tr>
    <tr>
      <td class="gt_row gt_stub gt_right gt_summary_row gt_first_summary_row">total</td>
      <td class="gt_row gt_left gt_summary_row gt_first_summary_row">&mdash;</td>
      <td class="gt_row gt_center gt_summary_row gt_first_summary_row">&mdash;</td>
      <td class="gt_row gt_left gt_summary_row gt_first_summary_row">&mdash;</td>
      <td class="gt_row gt_left gt_summary_row gt_first_summary_row">&mdash;</td>
      <td class="gt_row gt_right gt_summary_row gt_first_summary_row">&mdash;</td>
      <td class="gt_row gt_left gt_summary_row gt_first_summary_row">&mdash;</td>
      <td class="gt_row gt_center gt_summary_row gt_first_summary_row">14</td>
    </tr>
    <tr class="gt_group_heading_row">
      <td colspan="8" class="gt_group_heading">Het</td>
    </tr>
    <tr>
      <td class="gt_row gt_left gt_stub">1264</td>
      <td class="gt_row gt_left">del/+</td>
      <td class="gt_row gt_center">F</td>
      <td class="gt_row gt_left">207/20</td>
      <td class="gt_row gt_left">344/20</td>
      <td class="gt_row gt_right">137</td>
      <td class="gt_row gt_left">MiniBulk 100 cells</td>
      <td class="gt_row gt_center">3</td>
    </tr>
    <tr>
      <td class="gt_row gt_left gt_stub">1272</td>
      <td class="gt_row gt_left">del/+</td>
      <td class="gt_row gt_center">F</td>
      <td class="gt_row gt_left">221/20</td>
      <td class="gt_row gt_left">344/20</td>
      <td class="gt_row gt_right">123</td>
      <td class="gt_row gt_left">MiniBulk 100 cells</td>
      <td class="gt_row gt_center">3</td>
    </tr>
    <tr>
      <td class="gt_row gt_left gt_stub">1291</td>
      <td class="gt_row gt_left">del/+</td>
      <td class="gt_row gt_center">F</td>
      <td class="gt_row gt_left">266/20</td>
      <td class="gt_row gt_left">344/20</td>
      <td class="gt_row gt_right">78</td>
      <td class="gt_row gt_left">MiniBulk 100 cells</td>
      <td class="gt_row gt_center">3</td>
    </tr>
    <tr>
      <td class="gt_row gt_left gt_stub">1248</td>
      <td class="gt_row gt_left">del/+</td>
      <td class="gt_row gt_center">M</td>
      <td class="gt_row gt_left">165/20</td>
      <td class="gt_row gt_left">344/20</td>
      <td class="gt_row gt_right">178</td>
      <td class="gt_row gt_left">MiniBulk 100 cells</td>
      <td class="gt_row gt_center">3</td>
    </tr>
    <tr>
      <td class="gt_row gt_left gt_stub">1286</td>
      <td class="gt_row gt_left">del/+</td>
      <td class="gt_row gt_center">M</td>
      <td class="gt_row gt_left">258/20</td>
      <td class="gt_row gt_left">344/20</td>
      <td class="gt_row gt_right">86</td>
      <td class="gt_row gt_left">MiniBulk 100 cells</td>
      <td class="gt_row gt_center">3</td>
    </tr>
    <tr>
      <td class="gt_row gt_left gt_stub">1269</td>
      <td class="gt_row gt_left">del/fl</td>
      <td class="gt_row gt_center">M</td>
      <td class="gt_row gt_left">207/20</td>
      <td class="gt_row gt_left">344/20</td>
      <td class="gt_row gt_right">137</td>
      <td class="gt_row gt_left">MiniBulk 100 cells</td>
      <td class="gt_row gt_center">3</td>
    </tr>
    <tr>
      <td class="gt_row gt_stub gt_right gt_summary_row gt_first_summary_row">total</td>
      <td class="gt_row gt_left gt_summary_row gt_first_summary_row">&mdash;</td>
      <td class="gt_row gt_center gt_summary_row gt_first_summary_row">&mdash;</td>
      <td class="gt_row gt_left gt_summary_row gt_first_summary_row">&mdash;</td>
      <td class="gt_row gt_left gt_summary_row gt_first_summary_row">&mdash;</td>
      <td class="gt_row gt_right gt_summary_row gt_first_summary_row">&mdash;</td>
      <td class="gt_row gt_left gt_summary_row gt_first_summary_row">&mdash;</td>
      <td class="gt_row gt_center gt_summary_row gt_first_summary_row">18</td>
    </tr>
    <tr class="gt_group_heading_row">
      <td colspan="8" class="gt_group_heading">WT</td>
    </tr>
    <tr>
      <td class="gt_row gt_left gt_stub">1271</td>
      <td class="gt_row gt_left">fl/+</td>
      <td class="gt_row gt_center">M</td>
      <td class="gt_row gt_left">207/20</td>
      <td class="gt_row gt_left">344/20</td>
      <td class="gt_row gt_right">137</td>
      <td class="gt_row gt_left">MiniBulk 100 cells</td>
      <td class="gt_row gt_center">3</td>
    </tr>
    <tr>
      <td class="gt_row gt_stub gt_right gt_summary_row gt_first_summary_row">total</td>
      <td class="gt_row gt_left gt_summary_row gt_first_summary_row">&mdash;</td>
      <td class="gt_row gt_center gt_summary_row gt_first_summary_row">&mdash;</td>
      <td class="gt_row gt_left gt_summary_row gt_first_summary_row">&mdash;</td>
      <td class="gt_row gt_left gt_summary_row gt_first_summary_row">&mdash;</td>
      <td class="gt_row gt_right gt_summary_row gt_first_summary_row">&mdash;</td>
      <td class="gt_row gt_left gt_summary_row gt_first_summary_row">&mdash;</td>
      <td class="gt_row gt_center gt_summary_row gt_first_summary_row">3</td>
    </tr>
  </tbody>
  <tfoot class="gt_sourcenotes">
    <tr>
      <td class="gt_sourcenote" colspan="8">Source: <a href="../data/sample_sheets/Minibulk_RNAseq_samples.xlsx"><code>data/sample_sheets/Minibulk_RNAseq_samples.xlsx</code></a></td>
    </tr>
  </tfoot>
  
</table></div>
</div>
<p>The complete sample metadata is shown below.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://pkgs.rstudio.com/rmarkdown/reference/paged_table.html'>paged_table</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["unaligned"],"name":[1],"type":["int"],"align":["right"]},{"label":["aligned_unmapped"],"name":[2],"type":["int"],"align":["right"]},{"label":["mapped_to_exon"],"name":[3],"type":["int"],"align":["right"]},{"label":["mapped_to_intron"],"name":[4],"type":["int"],"align":["right"]},{"label":["ambiguous_mapping"],"name":[5],"type":["int"],"align":["right"]},{"label":["mapped_to_ERCC"],"name":[6],"type":["int"],"align":["right"]},{"label":["mapped_to_MT"],"name":[7],"type":["int"],"align":["right"]},{"label":["plate_number"],"name":[8],"type":["chr"],"align":["left"]},{"label":["well_position"],"name":[9],"type":["fct"],"align":["left"]},{"label":["sample_type"],"name":[10],"type":["chr"],"align":["left"]},{"label":["mouse_number"],"name":[11],"type":["fct"],"align":["left"]},{"label":["sample_name"],"name":[12],"type":["chr"],"align":["left"]},{"label":["c_rt1_primer_name"],"name":[13],"type":["chr"],"align":["left"]},{"label":["rd1_index_cell_index_index_sequence_as_in_c_rt1_primer"],"name":[14],"type":["chr"],"align":["left"]},{"label":["illumina_index_index_number_separate_index_read"],"name":[15],"type":["chr"],"align":["left"]},{"label":["illumina_index_index_sequence_separate_index_read"],"name":[16],"type":["chr"],"align":["left"]},{"label":["rt1_index_primer_sequences"],"name":[17],"type":["chr"],"align":["left"]},{"label":["sequencing_run"],"name":[18],"type":["chr"],"align":["left"]},{"label":["sex"],"name":[19],"type":["fct"],"align":["left"]},{"label":["smchd1_genotype_3"],"name":[20],"type":["chr"],"align":["left"]},{"label":["smchd1_genotype_4"],"name":[21],"type":["chr"],"align":["left"]},{"label":["dob"],"name":[22],"type":["chr"],"align":["left"]},{"label":["dod"],"name":[23],"type":["chr"],"align":["left"]},{"label":["age_days"],"name":[24],"type":["dbl"],"align":["right"]},{"label":["genotyping_bm_cells"],"name":[25],"type":["chr"],"align":["left"]},{"label":["smchd1_genotype_updated"],"name":[26],"type":["fct"],"align":["left"]},{"label":["genotype.mouse"],"name":[27],"type":["fct"],"align":["left"]},{"label":["rep"],"name":[28],"type":["int"],"align":["right"]}],"data":[{"1":"419784","2":"264425","3":"3057246","4":"1029825","5":"41796","6":"1190","7":"167349","8":"LCE504","9":"D10","10":"MiniBulk 100 cells","11":"1250","12":"LSK slam","13":"178","14":"CGTGAATC","15":"RPI 8","16":"ACTTGA","17":"CGATTGAGGCCGGTAATACGACTCACTATAGGGGTTCAGAGTTCTACAGTCCGACGATCNNNNNNCGTGAATCTTTTTTTTTTTTTTTTTTTTTTTTV","18":"NN206","19":"M","20":"del/fl","21":"Del","22":"165/20","23":"344/20","24":"178","25":"del/del","26":"Del","27":"Del.1250","28":"1","_rn_":"Del.1250.1"},{"1":"335787","2":"206480","3":"2361371","4":"789133","5":"31841","6":"1181","7":"131361","8":"LCE504","9":"D11","10":"MiniBulk 100 cells","11":"1250","12":"LSK slam","13":"179","14":"GGTAACTC","15":"RPI 8","16":"ACTTGA","17":"CGATTGAGGCCGGTAATACGACTCACTATAGGGGTTCAGAGTTCTACAGTCCGACGATCNNNNNNGGTAACTCTTTTTTTTTTTTTTTTTTTTTTTTV","18":"NN206","19":"M","20":"del/fl","21":"Del","22":"165/20","23":"344/20","24":"178","25":"del/del","26":"Del","27":"Del.1250","28":"2","_rn_":"Del.1250.2"},{"1":"552804","2":"294638","3":"3269790","4":"1232199","5":"44704","6":"1846","7":"166589","8":"LCE504","9":"D12","10":"MiniBulk 100 cells","11":"1253","12":"LSK slam","13":"180","14":"CACTAGCA","15":"RPI 8","16":"ACTTGA","17":"CGATTGAGGCCGGTAATACGACTCACTATAGGGGTTCAGAGTTCTACAGTCCGACGATCNNNNNNCACTAGCATTTTTTTTTTTTTTTTTTTTTTTTV","18":"NN206","19":"F","20":"del/fl","21":"Del","22":"184/20","23":"344/20","24":"160","25":"del/del","26":"Del","27":"Del.1253","28":"1","_rn_":"Del.1253.1"},{"1":"762686","2":"489731","3":"5050320","4":"1644220","5":"66576","6":"2717","7":"266753","8":"LCE504","9":"D13","10":"MiniBulk 100 cells","11":"1253","12":"LSK slam","13":"181","14":"GATACTGC","15":"RPI 8","16":"ACTTGA","17":"CGATTGAGGCCGGTAATACGACTCACTATAGGGGTTCAGAGTTCTACAGTCCGACGATCNNNNNNGATACTGCTTTTTTTTTTTTTTTTTTTTTTTTV","18":"NN206","19":"F","20":"del/fl","21":"Del","22":"184/20","23":"344/20","24":"160","25":"del/del","26":"Del","27":"Del.1253","28":"2","_rn_":"Del.1253.2"},{"1":"662658","2":"362116","3":"3795497","4":"1346683","5":"50739","6":"1788","7":"318560","8":"LCE504","9":"D14","10":"MiniBulk 100 cells","11":"1253","12":"LSK slam","13":"182","14":"GTGCGTTA","15":"RPI 8","16":"ACTTGA","17":"CGATTGAGGCCGGTAATACGACTCACTATAGGGGTTCAGAGTTCTACAGTCCGACGATCNNNNNNGTGCGTTATTTTTTTTTTTTTTTTTTTTTTTTV","18":"NN206","19":"F","20":"del/fl","21":"Del","22":"184/20","23":"344/20","24":"160","25":"del/del","26":"Del","27":"Del.1253","28":"3","_rn_":"Del.1253.3"},{"1":"551175","2":"320289","3":"3338054","4":"1407487","5":"42276","6":"1497","7":"167953","8":"LCE504","9":"D15","10":"MiniBulk 100 cells","11":"1264","12":"LSK slam","13":"183","14":"GGATGCTT","15":"RPI 8","16":"ACTTGA","17":"CGATTGAGGCCGGTAATACGACTCACTATAGGGGTTCAGAGTTCTACAGTCCGACGATCNNNNNNGGATGCTTTTTTTTTTTTTTTTTTTTTTTTTTV","18":"NN206","19":"F","20":"del/+","21":"Het","22":"207/20","23":"344/20","24":"137","25":"del/+","26":"Het","27":"Het.1264","28":"1","_rn_":"Het.1264.1"},{"1":"806239","2":"348573","3":"3809314","4":"1506786","5":"48706","6":"1539","7":"212785","8":"LCE504","9":"D16","10":"MiniBulk 100 cells","11":"1264","12":"LSK slam","13":"184","14":"TCAGGTGT","15":"RPI 8","16":"ACTTGA","17":"CGATTGAGGCCGGTAATACGACTCACTATAGGGGTTCAGAGTTCTACAGTCCGACGATCNNNNNNTCAGGTGTTTTTTTTTTTTTTTTTTTTTTTTTV","18":"NN206","19":"F","20":"del/+","21":"Het","22":"207/20","23":"344/20","24":"137","25":"del/+","26":"Het","27":"Het.1264","28":"2","_rn_":"Het.1264.2"},{"1":"890481","2":"504815","3":"5501653","4":"2111838","5":"71100","6":"2415","7":"320706","8":"LCE504","9":"D17","10":"MiniBulk 100 cells","11":"1264","12":"LSK slam","13":"185","14":"TAGCTCAG","15":"RPI 8","16":"ACTTGA","17":"CGATTGAGGCCGGTAATACGACTCACTATAGGGGTTCAGAGTTCTACAGTCCGACGATCNNNNNNTAGCTCAGTTTTTTTTTTTTTTTTTTTTTTTTV","18":"NN206","19":"F","20":"del/+","21":"Het","22":"207/20","23":"344/20","24":"137","25":"del/+","26":"Het","27":"Het.1264","28":"3","_rn_":"Het.1264.3"},{"1":"450016","2":"217171","3":"2569723","4":"846824","5":"35859","6":"1394","7":"175967","8":"LCE504","9":"D6","10":"MiniBulk 100 cells","11":"1248","12":"LSK slam","13":"174","14":"ATGCCAGT","15":"RPI 8","16":"ACTTGA","17":"CGATTGAGGCCGGTAATACGACTCACTATAGGGGTTCAGAGTTCTACAGTCCGACGATCNNNNNNATGCCAGTTTTTTTTTTTTTTTTTTTTTTTTTV","18":"NN206","19":"M","20":"del/+","21":"Het","22":"165/20","23":"344/20","24":"178","25":"del/+","26":"Het","27":"Het.1248","28":"1","_rn_":"Het.1248.1"},{"1":"252129","2":"146749","3":"1667259","4":"585668","5":"22722","6":"849","7":"85390","8":"LCE504","9":"D7","10":"MiniBulk 100 cells","11":"1248","12":"LSK slam","13":"175","14":"CATCCAGA","15":"RPI 8","16":"ACTTGA","17":"CGATTGAGGCCGGTAATACGACTCACTATAGGGGTTCAGAGTTCTACAGTCCGACGATCNNNNNNCATCCAGATTTTTTTTTTTTTTTTTTTTTTTTV","18":"NN206","19":"M","20":"del/+","21":"Het","22":"165/20","23":"344/20","24":"178","25":"del/+","26":"Het","27":"Het.1248","28":"2","_rn_":"Het.1248.2"},{"1":"420762","2":"236748","3":"2652487","4":"954461","5":"37079","6":"1350","7":"127034","8":"LCE504","9":"D8","10":"MiniBulk 100 cells","11":"1248","12":"LSK slam","13":"176","14":"GACTACCA","15":"RPI 8","16":"ACTTGA","17":"CGATTGAGGCCGGTAATACGACTCACTATAGGGGTTCAGAGTTCTACAGTCCGACGATCNNNNNNGACTACCATTTTTTTTTTTTTTTTTTTTTTTTV","18":"NN206","19":"M","20":"del/+","21":"Het","22":"165/20","23":"344/20","24":"178","25":"del/+","26":"Het","27":"Het.1248","28":"3","_rn_":"Het.1248.3"},{"1":"406472","2":"216090","3":"2462512","4":"895034","5":"35100","6":"995","7":"130040","8":"LCE504","9":"D9","10":"MiniBulk 100 cells","11":"1250","12":"LSK slam","13":"177","14":"CAGTGCTA","15":"RPI 8","16":"ACTTGA","17":"CGATTGAGGCCGGTAATACGACTCACTATAGGGGTTCAGAGTTCTACAGTCCGACGATCNNNNNNCAGTGCTATTTTTTTTTTTTTTTTTTTTTTTTV","18":"NN206","19":"M","20":"del/fl","21":"Del","22":"165/20","23":"344/20","24":"178","25":"del/del","26":"Del","27":"Del.1250","28":"3","_rn_":"Del.1250.3"},{"1":"845484","2":"519140","3":"6069536","4":"1904738","5":"77995","6":"3763","7":"331240","8":"LCE504","9":"E10","10":"MiniBulk 100 cells","11":"1269","12":"LSK slam","13":"202","14":"TACGAGTC","15":"RPI 8","16":"ACTTGA","17":"CGATTGAGGCCGGTAATACGACTCACTATAGGGGTTCAGAGTTCTACAGTCCGACGATCNNNNNNTACGAGTCTTTTTTTTTTTTTTTTTTTTTTTTV","18":"NN206","19":"M","20":"del/fl","21":"Del","22":"207/20","23":"344/20","24":"137","25":"del/fl","26":"Het","27":"Het.1269","28":"1","_rn_":"Het.1269.1"},{"1":"649517","2":"382973","3":"4677176","4":"1538531","5":"62485","6":"2306","7":"401279","8":"LCE504","9":"E11","10":"MiniBulk 100 cells","11":"1269","12":"LSK slam","13":"203","14":"GTGCGTAT","15":"RPI 8","16":"ACTTGA","17":"CGATTGAGGCCGGTAATACGACTCACTATAGGGGTTCAGAGTTCTACAGTCCGACGATCNNNNNNGTGCGTATTTTTTTTTTTTTTTTTTTTTTTTTV","18":"NN206","19":"M","20":"del/fl","21":"Del","22":"207/20","23":"344/20","24":"137","25":"del/fl","26":"Het","27":"Het.1269","28":"2","_rn_":"Het.1269.2"},{"1":"832971","2":"422436","3":"5112634","4":"1670034","5":"74275","6":"3916","7":"238011","8":"LCE504","9":"E12","10":"MiniBulk 100 cells","11":"1271","12":"LSK slam","13":"204","14":"CGACACAT","15":"RPI 8","16":"ACTTGA","17":"CGATTGAGGCCGGTAATACGACTCACTATAGGGGTTCAGAGTTCTACAGTCCGACGATCNNNNNNCGACACATTTTTTTTTTTTTTTTTTTTTTTTTV","18":"NN206","19":"M","20":"fl/+","21":"Het","22":"207/20","23":"344/20","24":"137","25":"fl/+","26":"WT","27":"WT.1271","28":"1","_rn_":"WT.1271.1"},{"1":"1069410","2":"620193","3":"7236478","4":"2474387","5":"101307","6":"4796","7":"397183","8":"LCE504","9":"E13","10":"MiniBulk 100 cells","11":"1271","12":"LSK slam","13":"205","14":"TCGACTAG","15":"RPI 8","16":"ACTTGA","17":"CGATTGAGGCCGGTAATACGACTCACTATAGGGGTTCAGAGTTCTACAGTCCGACGATCNNNNNNTCGACTAGTTTTTTTTTTTTTTTTTTTTTTTTV","18":"NN206","19":"M","20":"fl/+","21":"Het","22":"207/20","23":"344/20","24":"137","25":"fl/+","26":"WT","27":"WT.1271","28":"2","_rn_":"WT.1271.2"},{"1":"1063134","2":"540189","3":"6375836","4":"2197707","5":"90456","6":"4577","7":"422900","8":"LCE504","9":"E14","10":"MiniBulk 100 cells","11":"1271","12":"LSK slam","13":"206","14":"TCACGGTA","15":"RPI 8","16":"ACTTGA","17":"CGATTGAGGCCGGTAATACGACTCACTATAGGGGTTCAGAGTTCTACAGTCCGACGATCNNNNNNTCACGGTATTTTTTTTTTTTTTTTTTTTTTTTV","18":"NN206","19":"M","20":"fl/+","21":"Het","22":"207/20","23":"344/20","24":"137","25":"fl/+","26":"WT","27":"WT.1271","28":"3","_rn_":"WT.1271.3"},{"1":"1354016","2":"562035","3":"6746558","4":"2282616","5":"92200","6":"4517","7":"366975","8":"LCE504","9":"E15","10":"MiniBulk 100 cells","11":"1272","12":"LSK slam","13":"207","14":"ATGGTGCT","15":"RPI 8","16":"ACTTGA","17":"CGATTGAGGCCGGTAATACGACTCACTATAGGGGTTCAGAGTTCTACAGTCCGACGATCNNNNNNATGGTGCTTTTTTTTTTTTTTTTTTTTTTTTTV","18":"NN206","19":"F","20":"del/+","21":"Het","22":"221/20","23":"344/20","24":"123","25":"del/+","26":"Het","27":"Het.1272","28":"1","_rn_":"Het.1272.1"},{"1":"438214","2":"209659","3":"2131620","4":"983426","5":"26999","6":"799","7":"93364","8":"LCE504","9":"E16","10":"MiniBulk 100 cells","11":"1272","12":"LSK slam","13":"208","14":"GTACTGTG","15":"RPI 8","16":"ACTTGA","17":"CGATTGAGGCCGGTAATACGACTCACTATAGGGGTTCAGAGTTCTACAGTCCGACGATCNNNNNNGTACTGTGTTTTTTTTTTTTTTTTTTTTTTTTV","18":"NN206","19":"F","20":"del/+","21":"Het","22":"221/20","23":"344/20","24":"123","25":"del/+","26":"Het","27":"Het.1272","28":"2","_rn_":"Het.1272.2"},{"1":"791216","2":"475692","3":"5750758","4":"1891528","5":"79246","6":"3381","7":"314017","8":"LCE504","9":"E17","10":"MiniBulk 100 cells","11":"1272","12":"LSK slam","13":"209","14":"CGATTGTG","15":"RPI 8","16":"ACTTGA","17":"CGATTGAGGCCGGTAATACGACTCACTATAGGGGTTCAGAGTTCTACAGTCCGACGATCNNNNNNCGATTGTGTTTTTTTTTTTTTTTTTTTTTTTTV","18":"NN206","19":"F","20":"del/+","21":"Het","22":"221/20","23":"344/20","24":"123","25":"del/+","26":"Het","27":"Het.1272","28":"3","_rn_":"Het.1272.3"},{"1":"1237391","2":"583513","3":"7126330","4":"2197647","5":"103329","6":"4587","7":"501125","8":"LCE504","9":"E6","10":"MiniBulk 100 cells","11":"1266","12":"LSK slam","13":"198","14":"GTACTTGG","15":"RPI 8","16":"ACTTGA","17":"CGATTGAGGCCGGTAATACGACTCACTATAGGGGTTCAGAGTTCTACAGTCCGACGATCNNNNNNGTACTTGGTTTTTTTTTTTTTTTTTTTTTTTTV","18":"NN206","19":"F","20":"del/fl","21":"Del","22":"207/20","23":"344/20","24":"137","25":"del/del","26":"Del","27":"Del.1266","28":"1","_rn_":"Del.1266.1"},{"1":"843446","2":"350347","3":"4268451","4":"1286286","5":"61797","6":"2945","7":"255906","8":"LCE504","9":"E7","10":"MiniBulk 100 cells","11":"1266","12":"LSK slam","13":"199","14":"TAGGTGCT","15":"RPI 8","16":"ACTTGA","17":"CGATTGAGGCCGGTAATACGACTCACTATAGGGGTTCAGAGTTCTACAGTCCGACGATCNNNNNNTAGGTGCTTTTTTTTTTTTTTTTTTTTTTTTTV","18":"NN206","19":"F","20":"del/fl","21":"Del","22":"207/20","23":"344/20","24":"137","25":"del/del","26":"Del","27":"Del.1266","28":"2","_rn_":"Del.1266.2"},{"1":"509139","2":"274447","3":"3517525","4":"907175","5":"52724","6":"3190","7":"187734","8":"LCE504","9":"E8","10":"MiniBulk 58 cells","11":"1266","12":"LSK slam","13":"200","14":"AGCTTGCA","15":"RPI 8","16":"ACTTGA","17":"CGATTGAGGCCGGTAATACGACTCACTATAGGGGTTCAGAGTTCTACAGTCCGACGATCNNNNNNAGCTTGCATTTTTTTTTTTTTTTTTTTTTTTTV","18":"NN206","19":"F","20":"del/fl","21":"Del","22":"207/20","23":"344/20","24":"137","25":"del/del","26":"Del","27":"Del.1266","28":"3","_rn_":"Del.1266.3"},{"1":"527433","2":"287372","3":"3340891","4":"1229308","5":"46709","6":"1978","7":"151062","8":"LCE504","9":"E9","10":"MiniBulk 100 cells","11":"1269","12":"LSK slam","13":"201","14":"AGTGACTC","15":"RPI 8","16":"ACTTGA","17":"CGATTGAGGCCGGTAATACGACTCACTATAGGGGTTCAGAGTTCTACAGTCCGACGATCNNNNNNAGTGACTCTTTTTTTTTTTTTTTTTTTTTTTTV","18":"NN206","19":"M","20":"del/fl","21":"Del","22":"207/20","23":"344/20","24":"137","25":"del/fl","26":"Het","27":"Het.1269","28":"3","_rn_":"Het.1269.3"},{"1":"610136","2":"375723","3":"4933566","4":"1172086","5":"72759","6":"4772","7":"356231","8":"LCE504","9":"F10","10":"MiniBulk 50 cells","11":"1283","12":"LSK slam","13":"226","14":"GTTCCAAG","15":"RPI 8","16":"ACTTGA","17":"CGATTGAGGCCGGTAATACGACTCACTATAGGGGTTCAGAGTTCTACAGTCCGACGATCNNNNNNGTTCCAAGTTTTTTTTTTTTTTTTTTTTTTTTV","18":"NN206","19":"F","20":"del/fl","21":"Del","22":"258/20","23":"344/20","24":"86","25":"del/del","26":"Del","27":"Del.1283","28":"1","_rn_":"Del.1283.1"},{"1":"798900","2":"407601","3":"4861771","4":"1755389","5":"65710","6":"3689","7":"357490","8":"LCE504","9":"F12","10":"MiniBulk 100 cells","11":"1286","12":"LSK slam","13":"228","14":"TGATGCTG","15":"RPI 8","16":"ACTTGA","17":"CGATTGAGGCCGGTAATACGACTCACTATAGGGGTTCAGAGTTCTACAGTCCGACGATCNNNNNNTGATGCTGTTTTTTTTTTTTTTTTTTTTTTTTV","18":"NN206","19":"M","20":"del/+","21":"Het","22":"258/20","23":"344/20","24":"86","25":"del/+","26":"Het","27":"Het.1286","28":"1","_rn_":"Het.1286.1"},{"1":"968391","2":"505620","3":"6304428","4":"2060114","5":"88199","6":"3792","7":"350840","8":"LCE504","9":"F13","10":"MiniBulk 100 cells","11":"1286","12":"LSK slam","13":"229","14":"TACCGATG","15":"RPI 8","16":"ACTTGA","17":"CGATTGAGGCCGGTAATACGACTCACTATAGGGGTTCAGAGTTCTACAGTCCGACGATCNNNNNNTACCGATGTTTTTTTTTTTTTTTTTTTTTTTTV","18":"NN206","19":"M","20":"del/+","21":"Het","22":"258/20","23":"344/20","24":"86","25":"del/+","26":"Het","27":"Het.1286","28":"2","_rn_":"Het.1286.2"},{"1":"767410","2":"309472","3":"3815812","4":"1319976","5":"52551","6":"3236","7":"218044","8":"LCE504","9":"F14","10":"MiniBulk 100 cells","11":"1286","12":"LSK slam","13":"230","14":"TACAGGCT","15":"RPI 8","16":"ACTTGA","17":"CGATTGAGGCCGGTAATACGACTCACTATAGGGGTTCAGAGTTCTACAGTCCGACGATCNNNNNNTACAGGCTTTTTTTTTTTTTTTTTTTTTTTTTV","18":"NN206","19":"M","20":"del/+","21":"Het","22":"258/20","23":"344/20","24":"86","25":"del/+","26":"Het","27":"Het.1286","28":"3","_rn_":"Het.1286.3"},{"1":"1043391","2":"596408","3":"8058749","4":"2084566","5":"115621","6":"4218","7":"485616","8":"LCE504","9":"F15","10":"MiniBulk 100 cells","11":"1291","12":"LSK slam","13":"231","14":"AGCTTACG","15":"RPI 8","16":"ACTTGA","17":"CGATTGAGGCCGGTAATACGACTCACTATAGGGGTTCAGAGTTCTACAGTCCGACGATCNNNNNNAGCTTACGTTTTTTTTTTTTTTTTTTTTTTTTV","18":"NN206","19":"F","20":"del/+","21":"Het","22":"266/20","23":"344/20","24":"78","25":"del/+","26":"Het","27":"Het.1291","28":"1","_rn_":"Het.1291.1"},{"1":"536391","2":"300483","3":"3731110","4":"1333296","5":"49264","6":"1661","7":"236815","8":"LCE504","9":"F16","10":"MiniBulk 100 cells","11":"1291","12":"LSK slam","13":"232","14":"GAGTTCCA","15":"RPI 8","16":"ACTTGA","17":"CGATTGAGGCCGGTAATACGACTCACTATAGGGGTTCAGAGTTCTACAGTCCGACGATCNNNNNNGAGTTCCATTTTTTTTTTTTTTTTTTTTTTTTV","18":"NN206","19":"F","20":"del/+","21":"Het","22":"266/20","23":"344/20","24":"78","25":"del/+","26":"Het","27":"Het.1291","28":"2","_rn_":"Het.1291.2"},{"1":"891989","2":"481600","3":"6249593","4":"1793175","5":"91413","6":"4725","7":"287882","8":"LCE504","9":"F17","10":"MiniBulk 100 cells","11":"1291","12":"LSK slam","13":"233","14":"GCTGCATA","15":"RPI 8","16":"ACTTGA","17":"CGATTGAGGCCGGTAATACGACTCACTATAGGGGTTCAGAGTTCTACAGTCCGACGATCNNNNNNGCTGCATATTTTTTTTTTTTTTTTTTTTTTTTV","18":"NN206","19":"F","20":"del/+","21":"Het","22":"266/20","23":"344/20","24":"78","25":"del/+","26":"Het","27":"Het.1291","28":"3","_rn_":"Het.1291.3"},{"1":"667653","2":"248269","3":"3004013","4":"915182","5":"46258","6":"2372","7":"166687","8":"LCE504","9":"F6","10":"MiniBulk 100 cells","11":"1282","12":"LSK slam","13":"222","14":"GAGAGGAA","15":"RPI 8","16":"ACTTGA","17":"CGATTGAGGCCGGTAATACGACTCACTATAGGGGTTCAGAGTTCTACAGTCCGACGATCNNNNNNGAGAGGAATTTTTTTTTTTTTTTTTTTTTTTTV","18":"NN206","19":"M","20":"del/fl","21":"Del","22":"258/20","23":"344/20","24":"86","25":"del/del","26":"Del","27":"Del.1282","28":"1","_rn_":"Del.1282.1"},{"1":"95921","2":"32481","3":"325500","4":"124764","5":"4884","6":"101","7":"9789","8":"LCE504","9":"F7","10":"MiniBulk 100 cells","11":"1282","12":"LSK slam","13":"223","14":"ACACAGCT","15":"RPI 8","16":"ACTTGA","17":"CGATTGAGGCCGGTAATACGACTCACTATAGGGGTTCAGAGTTCTACAGTCCGACGATCNNNNNNACACAGCTTTTTTTTTTTTTTTTTTTTTTTTTV","18":"NN206","19":"M","20":"del/fl","21":"Del","22":"258/20","23":"344/20","24":"86","25":"del/del","26":"Del","27":"Del.1282","28":"2","_rn_":"Del.1282.2"},{"1":"523226","2":"302638","3":"3750452","4":"1055844","5":"52586","6":"2296","7":"257461","8":"LCE504","9":"F8","10":"MiniBulk 100 cells","11":"1282","12":"LSK slam","13":"224","14":"CGACGTAT","15":"RPI 8","16":"ACTTGA","17":"CGATTGAGGCCGGTAATACGACTCACTATAGGGGTTCAGAGTTCTACAGTCCGACGATCNNNNNNCGACGTATTTTTTTTTTTTTTTTTTTTTTTTTV","18":"NN206","19":"M","20":"del/fl","21":"Del","22":"258/20","23":"344/20","24":"86","25":"del/del","26":"Del","27":"Del.1282","28":"3","_rn_":"Del.1282.3"},{"1":"995149","2":"481269","3":"6734852","4":"1403390","5":"104849","6":"2868","7":"270172","8":"LCE504","9":"F9","10":"MiniBulk 100 cells","11":"1283","12":"LSK slam","13":"225","14":"ACATGCCA","15":"RPI 8","16":"ACTTGA","17":"CGATTGAGGCCGGTAATACGACTCACTATAGGGGTTCAGAGTTCTACAGTCCGACGATCNNNNNNACATGCCATTTTTTTTTTTTTTTTTTTTTTTTV","18":"NN206","19":"F","20":"del/fl","21":"Del","22":"258/20","23":"344/20","24":"86","25":"del/del","26":"Del","27":"Del.1283","28":"2","_rn_":"Del.1283.2"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='co'># Some useful colours</span>
<span class='va'>smchd1_genotype_updated_colours</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/setNames.html'>setNames</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/grDevices/palette.html'>palette.colors</a></span><span class='op'>(</span>n <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/nlevels.html'>nlevels</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>smchd1_genotype_updated</span><span class='op'>)</span>, <span class='st'>"Dark 2"</span><span class='op'>)</span>,
  <span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>smchd1_genotype_updated</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>sce</span><span class='op'>$</span><span class='va'>smchd1_genotype_updated_colours</span> <span class='op'>&lt;-</span> <span class='va'>smchd1_genotype_updated_colours</span><span class='op'>[</span>
  <span class='va'>sce</span><span class='op'>$</span><span class='va'>smchd1_genotype_updated</span><span class='op'>]</span>
<span class='va'>sex_colours</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/setNames.html'>setNames</a></span><span class='op'>(</span>
  <span class='fu'>RColorBrewer</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html'>brewer.pal</a></span><span class='op'>(</span><span class='fl'>4</span>, <span class='st'>"Set1"</span><span class='op'>)</span><span class='op'>[</span><span class='fl'>1</span><span class='op'>:</span><span class='fl'>2</span><span class='op'>]</span>,
  <span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>sex</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>sce</span><span class='op'>$</span><span class='va'>sex_colours</span> <span class='op'>&lt;-</span> <span class='va'>sex_colours</span><span class='op'>[</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>sex</span><span class='op'>]</span>
<span class='va'>mouse_number_colours</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/setNames.html'>setNames</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/grDevices/palette.html'>palette.colors</a></span><span class='op'>(</span>palette <span class='op'>=</span> <span class='st'>"Alphabet"</span>, n <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/nlevels.html'>nlevels</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>mouse_number</span><span class='op'>)</span><span class='op'>)</span>,
  <span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>mouse_number</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>sce</span><span class='op'>$</span><span class='va'>mouse_number_colours</span> <span class='op'>&lt;-</span> <span class='va'>mouse_number_colours</span><span class='op'>[</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>mouse_number</span><span class='op'>]</span>
</code></pre>
</div>
</details>
</div>
<h1 id="have-we-sequenced-enough">Have we sequenced enough?</h1>
<h2 id="motivation-1">Motivation</h2>
<p>SCORE’s mini-bulk protocol is still being finalised and some technical questions remain. In particular, we want to know if we have sequenced enough to answer the biological questions of interest.</p>
<p>One way of estimating if we have sequenced enough uses a metric called ‘sequencing saturation’. 10x Genomics have some useful information about sequencing saturation available <a href="https://kb.10xgenomics.com/hc/en-us/articles/115005062366-What-is-sequencing-saturation-">here</a>. In particular,</p>
<blockquote>
<p>Sequencing saturation is a measure of the fraction of library complexity that was sequenced in a given experiment. The inverse of the sequencing saturation can be interpreted as the number of additional reads it would take to detect a new transcript. … Sequencing saturation is dependent on the library complexity and sequencing depth. Different cell types will have different amounts of RNA and thus will differ in the total number of different transcripts in the final library (also known as library complexity). … As sequencing depth increases, more genes are detected, but this reaches saturation at different sequencing depths depending on cell type. … Sequencing depth also affects sequencing saturation; generally, the more sequencing reads, the more additional unique transcripts you can detect. However, this is limited by the library complexity.</p>
</blockquote>
<p>This naturally leads to the question of <a href="https://kb.10xgenomics.com/hc/en-us/articles/115002474263-How-much-sequencing-saturation-should-I-aim-for-">‘how much sequencing saturation should I aim for?’</a>, which very much depends on the goals of the experiment. For this experiment, where we wish to assess differential expression between treatment arms, we will likely want quite high sequencing saturation.</p>
<h2 id="analysis-1">Analysis</h2>
<p>To begin, we simply examine the number of reads sequenced/sample.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>reads_ls</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log10</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colSums.html'>colSums</a></span><span class='op'>(</span><span class='fu'>assay</span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"read_counts"</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>Figure <a href="#fig:read-library-size">4</a> plots the distribution of the number of reads mapped to exons in each sample<a href="#fn3" class="footnote-ref" id="fnref3" role="doc-noteref"><sup>3</sup></a>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/r/graphics/plot.default.html'>plot</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/stats/density.html'>density</a></span><span class='op'>(</span><span class='va'>reads_ls</span><span class='op'>)</span>,
  xlab <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/expression.html'>expression</a></span><span class='op'>(</span><span class='va'>log</span><span class='op'>[</span><span class='fl'>10</span><span class='op'>]</span><span class='op'>(</span><span class='va'>n</span><span class='op'>[</span><span class='va'>reads</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span>,
  ylab <span class='op'>=</span> <span class='st'>"density"</span>,
  xlim <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/range.html'>range</a></span><span class='op'>(</span><span class='va'>reads_ls</span><span class='op'>)</span>,
  main <span class='op'>=</span> <span class='st'>""</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:read-library-size"></span>
<img src="C086_Kinkel.preprocess_files/read-library-size-1.png" alt="Distribution of the number of reads mapped to exons in each sample." width="624" />
<p class="caption">
Figure 4: Distribution of the number of reads mapped to exons in each sample.
</p>
</div>
</div>
<p>We estimate the sequencing saturation using <a href="https://kb.10xgenomics.com/hc/en-us/articles/115003646912-How-is-sequencing-saturation-calculated-">10x Genomics’ formula</a>,</p>
<p><span class="math display">\[
1 - (n_{deduped reads} / n_{reads})
\]</span> where <span class="math inline">\(n_{deduped reads}\)</span> = Number of unique (valid cell-barcode, valid UMI, gene)-combinations among confidently mapped reads and <span class="math inline">\(n_{reads}\)</span> = Total number of confidently mapped, valid cell-barcode, valid UMI reads.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>saturation</span> <span class='op'>&lt;-</span> <span class='fl'>1</span> <span class='op'>-</span> 
  <span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colSums.html'>colSums</a></span><span class='op'>(</span><span class='fu'>assay</span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"UMI_counts"</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>/</span> <span class='fu'><a href='https://rdrr.io/r/base/colSums.html'>colSums</a></span><span class='op'>(</span><span class='fu'>assay</span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"read_counts"</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>Figure <a href="#fig:saturation">5</a> shows that almost all samples are sequenced to near-total saturation.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/r/graphics/par.html'>par</a></span><span class='op'>(</span>mfrow <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/graphics/plot.default.html'>plot</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/stats/density.html'>density</a></span><span class='op'>(</span>
    <span class='va'>saturation</span>,
    from <span class='op'>=</span> <span class='fl'>0</span>,
    to <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span>,
  xlab <span class='op'>=</span> <span class='st'>"saturation"</span>,
  ylab <span class='op'>=</span> <span class='st'>"density"</span>,
  xlim <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>0</span>, <span class='fl'>1</span><span class='op'>)</span>,
  main <span class='op'>=</span> <span class='st'>"Sequencing saturation"</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:saturation"></span>
<img src="C086_Kinkel.preprocess_files/saturation-1.png" alt="Distribution of the sequencing saturation." width="624" />
<p class="caption">
Figure 5: Distribution of the sequencing saturation.
</p>
</div>
</div>
<h2 id="summary-1">Summary</h2>
<p>These results show that the libraries are sequenced to near-saturation.</p>
<p>One way to think of sequencing saturation is <a href="https://kb.10xgenomics.com/hc/en-us/articles/115002474263-How-much-sequencing-saturation-should-I-aim-for-">provided by 10x Genomics</a>:</p>
<blockquote>
<p>The inverse of the sequencing saturation can be interpreted as roughly the number of new transcripts you expect to find with one new read. If sequencing saturation is at 50%, it means that every 2 new reads will result in 1 new UMI count (unique transcript) detected. In contrast, 90% sequencing saturation means that 10 new reads are necessary to obtain one new UMI count. If the sequencing saturation is high, additional sequencing would not recover much new information for the library.</p>
</blockquote>
<p><strong>At this stage, it is not necessary to perform additional sequencing of the library.</strong></p>
<h1 id="umi-counts-vs.-read-counts">UMI counts vs. read counts</h1>
<h2 id="motivation-2">Motivation</h2>
<p>A unique feature of SCORE’s ‘mini-bulk’ RNA-sequencing protocol is the incorporation of unique molecular identifiers (UMIs) into the library preparation; an ordinary bulk RNA-seq protocol does not incorporate UMIs.</p>
<p>UMIs are commonly used in single-cell RNA-sequencing protocols, such as the CEL-Seq2 protocol from which the mini-bulk protocol is adapted. UMIs are molecular tags that are used to detect and quantify unique mRNA transcripts. In this method, mRNA libraries are generated by fragmentation and reverse-transcribed to cDNA. Oligo(dT) primers with specific sequencing linkers are added to the cDNA. Another sequencing linker with a 6 bp random label and an index sequence is added to the 5’ end of the template, which is amplified and sequenced.</p>
<aside>
This explanation is adapted from <a href="https://sapac.illumina.com/science/sequencing-method-explorer/kits-and-arrays/umi.html">https://sapac.illumina.com/science/sequencing-method-explorer/kits-and-arrays/umi.html</a>.
</aside>
<p>The UMI count is always less than or equal to the read count and a UMI count of zero also implies a read count of zero.</p>
<p>Pros of using UMIs:</p>
<ul>
<li>Can sequence unique mRNA transcripts</li>
<li>Can detect transcripts occurring at low frequencies</li>
<li>Transcripts can be quantified based on sequencing reads specific to each barcode</li>
</ul>
<p>Cons of using UMIs:</p>
<ul>
<li>SCORE’s mini-bulk protocol uses a <span class="math inline">\(k=6\)</span> bp UMI<a href="#fn4" class="footnote-ref" id="fnref4" role="doc-noteref"><sup>4</sup></a> and so (theoretically) the maximum UMI count for each gene is <span class="math inline">\(4^6 = 4096\)</span>.</li>
<li>There is therefore a risk that highly expressed genes may have their expression measurements artificially truncated, particularly if the sample is deeply sequenced.</li>
</ul>
<p>To investigate whether expression measurements are being ‘truncated’ in this dataset, and decide whether to use the UMI counts or the read counts, we generated and now analyse two count matrices:</p>
<ol type="1">
<li>UMI counts (i.e. deduplicating UMI counts)</li>
<li>Read counts (i.e. not deduplicating UMI counts, equivalent to ignoring the UMIs)</li>
</ol>
<h2 id="analysis-2">Analysis</h2>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>n</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/apply.html'>apply</a></span><span class='op'>(</span><span class='fu'>assay</span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"read_counts"</span><span class='op'>)</span>, <span class='fl'>1</span>, <span class='kw'>function</span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span> <span class='fu'><a href='https://rdrr.io/r/base/sum.html'>sum</a></span><span class='op'>(</span><span class='va'>x</span> <span class='op'>&gt;</span> <span class='fl'>4</span> <span class='op'>^</span> <span class='fl'>6</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>g</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>n</span><span class='op'>[</span><span class='va'>n</span> <span class='op'>&gt;</span> <span class='fl'>0</span><span class='op'>]</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>Figure <a href="#fig:read-distribution">6</a> plots the distribution of the logarithm of the read counts. From this we see that the vast majority of read counts are small. In fact:</p>
<ul>
<li>41.4% of read counts are 0</li>
<li>95% of the read counts are no greater than 650</li>
<li>99% of the read counts are no greater than 2,025</li>
<li>99.9% of the read counts are no greater than 11,234</li>
</ul>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>x</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/vector.html'>as.vector</a></span><span class='op'>(</span><span class='fu'>assay</span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"read_counts"</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> <span class='fl'>1L</span>

<span class='fu'><a href='https://rdrr.io/r/graphics/par.html'>par</a></span><span class='op'>(</span>mfrow <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>2</span><span class='op'>)</span>, cex <span class='op'>=</span> <span class='fl'>0.5</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/graphics/plot.default.html'>plot</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/stats/density.html'>density</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log2</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span>, from <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/Extremes.html'>min</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log2</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span><span class='op'>)</span>, to <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/Extremes.html'>max</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log2</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>,
  xlab <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/expression.html'>expression</a></span><span class='op'>(</span><span class='va'>log</span><span class='op'>[</span><span class='fl'>2</span><span class='op'>]</span><span class='op'>(</span><span class='va'>reads</span> <span class='op'>+</span> <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span>,
  main <span class='op'>=</span> <span class='st'>""</span>,
  ylab <span class='op'>=</span> <span class='st'>"density"</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/graphics/abline.html'>abline</a></span><span class='op'>(</span>v <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log2</a></span><span class='op'>(</span><span class='fl'>4</span> <span class='op'>^</span> <span class='fl'>6</span><span class='op'>)</span>, col <span class='op'>=</span> <span class='st'>"orange"</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/graphics/plot.default.html'>plot</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/stats/ecdf.html'>ecdf</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log2</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span><span class='op'>)</span>, 
  xlim <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/range.html'>range</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log2</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span><span class='op'>)</span>,
  main <span class='op'>=</span> <span class='st'>""</span>,
  xlab <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/expression.html'>expression</a></span><span class='op'>(</span><span class='va'>log</span><span class='op'>[</span><span class='fl'>2</span><span class='op'>]</span><span class='op'>(</span><span class='va'>reads</span> <span class='op'>+</span> <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span>,
  ylab <span class='op'>=</span> <span class='st'>"cumulative density"</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/graphics/abline.html'>abline</a></span><span class='op'>(</span>v <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log2</a></span><span class='op'>(</span><span class='fl'>4</span> <span class='op'>^</span> <span class='fl'>6</span><span class='op'>)</span>, col <span class='op'>=</span> <span class='st'>"orange"</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:read-distribution"></span>
<img src="C086_Kinkel.preprocess_files/read-distribution-1.png" alt="Distribution of the logarithm of the read counts as density plot (left) and empirical cumulative distribution function (right). A count of 1 is added to each UMI count to avoid taking the logarithm of zero. The orange vertical line denotes $4^6 = 4096$." width="624" />
<p class="caption">
Figure 6: Distribution of the logarithm of the read counts as density plot (left) and empirical cumulative distribution function (right). A count of 1 is added to each UMI count to avoid taking the logarithm of zero. The orange vertical line denotes <span class="math inline">\(4^6 = 4096\)</span>.
</p>
</div>
</div>
<p>Only 0.372% of read counts exceed the <span class="math inline">\(4^6 = 4096\)</span> threshold. However there are 245 genes for which at least one sample exceeds the <span class="math inline">\(4^6=4096\)</span> threshold; these genes are shown in the table below. If we were to use the UMI counts, the genes in this list may have their expression measurements artificially truncated in those samples where the number of copies of the transcript exceed the <span class="math inline">\(4^6=4096\)</span> threshold.</p>
<p>This table shows that many of these genes are mitochondrial and ribosomal protein genes.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>read_e</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/matrix.html'>as.matrix</a></span><span class='op'>(</span><span class='fu'>assay</span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"read_counts"</span><span class='op'>)</span><span class='op'>[</span><span class='va'>g</span>, <span class='op'>]</span><span class='op'>)</span>
<span class='va'>umi_e</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/matrix.html'>as.matrix</a></span><span class='op'>(</span><span class='fu'>assay</span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"UMI_counts"</span><span class='op'>)</span><span class='op'>[</span><span class='va'>g</span>, <span class='op'>]</span><span class='op'>)</span>
<span class='va'>high_count_tbl</span> <span class='op'>&lt;-</span> <span class='fu'>tibble</span><span class='fu'>::</span><span class='fu'><a href='https://tibble.tidyverse.org/reference/tibble.html'>tibble</a></span><span class='op'>(</span>
  gene <span class='op'>=</span> <span class='va'>g</span>,
  read_median_count <span class='op'>=</span> <span class='fu'>matrixStats</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/matrixStats/man/rowMedians.html'>rowMedians</a></span><span class='op'>(</span><span class='va'>read_e</span><span class='op'>)</span>,
  read_exceed_sum <span class='op'>=</span> <span class='fu'>matrixStats</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/matrixStats/man/rowSums2.html'>rowSums2</a></span><span class='op'>(</span><span class='va'>read_e</span> <span class='op'>&gt;</span> <span class='fl'>4</span> <span class='op'>^</span> <span class='fl'>6</span><span class='op'>)</span>,
  read_exceed_percent <span class='op'>=</span> <span class='va'>read_exceed_sum</span> <span class='op'>/</span> <span class='fu'><a href='https://rdrr.io/r/base/nrow.html'>ncol</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span>,
  umi_median_count <span class='op'>=</span> <span class='fu'>matrixStats</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/matrixStats/man/rowMedians.html'>rowMedians</a></span><span class='op'>(</span><span class='va'>umi_e</span><span class='op'>)</span>,
  umi_exceed_sum <span class='op'>=</span> <span class='fu'>matrixStats</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/matrixStats/man/rowSums2.html'>rowSums2</a></span><span class='op'>(</span><span class='va'>umi_e</span> <span class='op'>&gt;</span> <span class='fl'>4</span> <span class='op'>^</span> <span class='fl'>6</span><span class='op'>)</span>,
  umi_exceed_percent <span class='op'>=</span> <span class='va'>umi_exceed_sum</span> <span class='op'>/</span> <span class='fu'><a href='https://rdrr.io/r/base/nrow.html'>ncol</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/arrange.html'>arrange</a></span><span class='op'>(</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/desc.html'>desc</a></span><span class='op'>(</span><span class='va'>read_exceed_sum</span><span class='op'>)</span><span class='op'>)</span>

<span class='va'>sketch</span> <span class='op'>&lt;-</span> <span class='fu'>htmltools</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/htmltools/man/withTags.html'>withTags</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/table.html'>table</a></span><span class='op'>(</span>
  class <span class='op'>=</span> <span class='st'>"display"</span>,
  <span class='fu'>thead</span><span class='op'>(</span>
    <span class='fu'>tr</span><span class='op'>(</span>
      <span class='fu'>th</span><span class='op'>(</span>rowspan <span class='op'>=</span> <span class='fl'>2</span>, <span class='st'>"gene"</span><span class='op'>)</span>,
      <span class='fu'>th</span><span class='op'>(</span>colspan <span class='op'>=</span> <span class='fl'>3</span>, <span class='st'>"reads"</span><span class='op'>)</span>,
      <span class='fu'>th</span><span class='op'>(</span>colspan <span class='op'>=</span> <span class='fl'>3</span>, <span class='st'>"UMIs"</span><span class='op'>)</span>
    <span class='op'>)</span>,
    <span class='fu'>tr</span><span class='op'>(</span>
      <span class='fu'><a href='https://rdrr.io/r/base/lapply.html'>lapply</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/rep.html'>rep</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"median count"</span>, <span class='st'>"N &gt; 4096"</span>, <span class='st'>"% &gt; 4096"</span><span class='op'>)</span>, <span class='fl'>2</span><span class='op'>)</span>, <span class='va'>th</span><span class='op'>)</span>
    <span class='op'>)</span>
  <span class='op'>)</span>
<span class='op'>)</span><span class='op'>)</span>

<span class='fu'>DT</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/DT/man/datatable.html'>datatable</a></span><span class='op'>(</span>
  <span class='va'>high_count_tbl</span>, 
  caption <span class='op'>=</span> <span class='st'>"Genes for which any sample has a read count exceeding the 4096 threshold and summaries of the corresponding read and UMI counts, namely: the median count ('median count'); the number of samples for which the counts exceed the threshold ('N &gt; 4096'); and the percentage of samples for which the counts exceed the threshold ('% &gt; 4096')."</span>,
  container <span class='op'>=</span> <span class='va'>sketch</span>,
  rownames <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>DT</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/DT/man/formatCurrency.html'>formatPercentage</a></span><span class='op'>(</span>
    columns <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"read_exceed_percent"</span>, <span class='st'>"umi_exceed_percent"</span><span class='op'>)</span>,
    digits <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>DT</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/DT/man/formatCurrency.html'>formatRound</a></span><span class='op'>(</span>
    columns <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"read_median_count"</span>, <span class='st'>"umi_median_count"</span><span class='op'>)</span>,
    digits <span class='op'>=</span> <span class='fl'>0</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div id="htmlwidget-7df0a3796d013070ca78" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-7df0a3796d013070ca78">{"x":{"filter":"none","class":"display","caption":"<caption>Genes for which any sample has a read count exceeding the 4096 threshold and summaries of the corresponding read and UMI counts, namely: the median count ('median count'); the number of samples for which the counts exceed the threshold ('N &gt; 4096'); and the percentage of samples for which the counts exceed the threshold ('% &gt; 4096').<\/caption>","data":[["ENSMUSG00000106106.3","mt-Rnr1","Gm6472","Rpl18a","Rpl23","mt-Rnr2","Malat1","mt-Nd4","ENSMUSG00000117438.1","mt-Cytb","Gm29216","Srgn","Actb","Rpl4","ENSMUSG00000107176.1","Gm28661","Gm15500","mt-Nd5","Gm10925","Rps3a2","mt-Nd1","Gm15793","Rps4x","Eef1b2","Rplp1","Gm28437","Rps6","Rps5","Rps23-ps2","H2-K1","Car2","Rps14","B2m","Rpl31-ps8","mt-Nd2","Nedd4","Eif5a","Hsp90ab1","mt-Co1","Serinc3","ENSMUSG00000108437.1","Cfl1","Pcbp1","Tpt1-ps3","Rpl17-ps10","Tmsb4x","Pabpc1","Gm15427","Rps24","Rpl15-ps3","Rps6-ps3","Rps3a1","ENSMUSG00000110275.1","Pbx1","Rpl10a-ps1","Gm28439","ENSMUSG00000104649.1","Rps20","Ybx1","Npm1","Rps10-ps2","Gm14303","Ubb","Txnip","Gm10073","Gm14586","Rps9","Cd47","Gm8730","Arpc2","Ncl","Rps3","Gm13192","H2afy","Gm10275","Rpl32","Rps2","Kcnq1ot1","Rplp2","mt-Atp6","Lmo2","ENSMUSG00000109933.1","Ccnd2","Xist","Srsf2","Gm10222","Rpl36a-ps1","Ddx5","Rpl13","Eef1a1","Gm6136","Pnisr","Gnb2l1","ENSMUSG00000105944.1","Itm2b","ENSMUSG00000105388.1","Rpl38-ps2","Gnas","Hsp90aa1","ENSMUSG00000106926.1","Rpl7","Rpl14","Hnrnpa2b1","H2-D1","Serbp1","H3f3b","ENSMUSG00000108366.1","Pcbp2","ENSMUSG00000113059.1","Cdk6","Rps16-ps2","Rsrp1","Hlf","Rps18","Gnai2","Gm10260","Hnrnpab","Rps26","Sep15","Son","Gm12254","Gpx1","Atp5g3","Pcna","Kit","ENSMUSG00000117468.1","Gm12191","Cdc42","Eef2","Gm10132","Rplp0","Rps15","Ybx3","Fth1","Hnrnpu","mt-Co2","Actg1","Gm9385","Rps28","Sub1","Pfn1","Rpl22","Rps13-ps2","Sf3b1","Gas5","Chchd2","Gpr56","Rpl15","Rbm25","Gm12338","ENSMUSG00000109509.1","ENSMUSG00000107713.1","Rpl8","Gm8355","Gm10136","Srrm2","Mat2a","Rpl35a-ps6","Slc25a5","Celf2","Gata2","Tubb5","Rpl10-ps1","Rps8-ps2","Atp5a1","Myl6","Paics","Rpl41","Rpl14-ps1","Pim1","Rpl9-ps6","Ifitm1","Rps25","Gm9843","Ly6e","Cct6a","Akap13","Caprin1","Hnrnpk","Atrx","Atp5b","Rps11","Tagln2","Srsf11","Myb","Gm15710","Rps13-ps4","Rpl34","Capns1","Pf4","Rpl31-ps9","Ywhab","Gm10180","Tram1","Msi2","Sfpq","Hdgf","Banf1","Gm8451","Ldha","Ptp4a2","Prkacb","Ppif","Gm26384","Gm5963","Nucb1","Sod1","H2-Q7","Angpt1","Dpf2","Hist1h2ap","Jun","Eif4a1","Ssb","Cd164","Calm1","Eif4ebp2","Gm5428","Rbm3","Hnrnph1","Rps18-ps3","Rpl30","Mycn","Calr","Rps19","Btf3","Txn1","Gm21887","Krtcap2","Tln1","Sumo2","Erdr1","Atxn7l3b","Cux1","Zbtb20","Tspan13","Gm13340","Gm13436","Kmt2e","Ddx21","Vdac2","Srsf7","Pdcd4","Gm7866","Rbm39"],[96412,16128,11841,16027,13431,71652,32803,24543,16550,58907,54945,14551,17322,12629,9767,44419,16383,20825,54177,11496,17007,13207,13546,12979,8573,16622,10937,10372,7866,10804,9283,7563,11993,8234,9174,7187,7070,6769,7354,5648,6589,5763,6252,6459,6099,6213,7202,6147,5521,6010,6008,5504,5912,6477,5480,5749,4910,5309,5039,4879,5285,5644,5024,5250,5870,4939,4699,4294,5215,4581,4369,4537,4739,4262,4161,4134,4202,4123,4319,3983,3988,3941,4058,518,4018,3721,3672,4051,3770,3761,4009,3738,3887,4051,3868,3831,3962,3930,3973,3764,3896,3222,3468,3597,3598,3670,3234,3559,3509,3592,3732,3471,3174,3315,3035,3338,3418,3480,3215,2849,3019,3349,3422,2550,3182,3300,3143,3460,2789,2641,3299,2739,3099,3027,2942,2901,2653,2927,2909,2834,2813,2827,2602,3226,2749,2772,2775,2384,2811,2134,2551,2727,2668,2481,2639,2600,2610,2514,2512,2836,2871,1950,2305,2721,2155,2132,2712,2333,2428,1789,2397,1680,2371,2220,2119,1938,2557,1942,2435,2236,2126,2188,2397,2305,2444,2393,2448,1883,1871,1455,2217,2127,1678,1985,2001,2560,2199,2031,2047,2032,2307,2042,1858,2100,1943,2318,2075,2018,2301,992,1490,776,1707,1891,1475,1741,1756,1684,1542,1922,2229,2299,2260,1687,2045,1515,1781,837,1781,2000,1990,1088,1944,1435,1544,1088,2221,2187,2126,1583,1548,2030,2036,1798,2180],[35,34,34,34,34,34,34,34,34,34,34,34,34,34,34,34,34,34,34,34,34,34,34,34,33,33,33,33,33,33,33,33,33,32,32,32,32,31,31,29,29,29,29,29,29,29,29,28,27,27,26,26,26,26,26,25,24,24,23,23,23,23,23,22,22,22,21,21,21,21,21,20,19,19,18,18,18,18,18,17,17,17,17,17,17,16,16,16,16,16,16,16,16,16,16,16,15,15,15,15,15,14,14,14,14,14,14,14,13,13,13,13,13,13,12,12,12,12,11,11,11,11,10,10,10,10,10,10,9,9,9,9,9,9,9,8,8,8,8,8,8,8,8,8,7,7,7,7,6,6,6,6,5,5,5,5,5,5,5,5,5,4,4,4,4,4,4,4,4,4,4,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[1,0.971428571428571,0.971428571428571,0.971428571428571,0.971428571428571,0.971428571428571,0.971428571428571,0.971428571428571,0.971428571428571,0.971428571428571,0.971428571428571,0.971428571428571,0.971428571428571,0.971428571428571,0.971428571428571,0.971428571428571,0.971428571428571,0.971428571428571,0.971428571428571,0.971428571428571,0.971428571428571,0.971428571428571,0.971428571428571,0.971428571428571,0.942857142857143,0.942857142857143,0.942857142857143,0.942857142857143,0.942857142857143,0.942857142857143,0.942857142857143,0.942857142857143,0.942857142857143,0.914285714285714,0.914285714285714,0.914285714285714,0.914285714285714,0.885714285714286,0.885714285714286,0.828571428571429,0.828571428571429,0.828571428571429,0.828571428571429,0.828571428571429,0.828571428571429,0.828571428571429,0.828571428571429,0.8,0.771428571428571,0.771428571428571,0.742857142857143,0.742857142857143,0.742857142857143,0.742857142857143,0.742857142857143,0.714285714285714,0.685714285714286,0.685714285714286,0.657142857142857,0.657142857142857,0.657142857142857,0.657142857142857,0.657142857142857,0.628571428571429,0.628571428571429,0.628571428571429,0.6,0.6,0.6,0.6,0.6,0.571428571428571,0.542857142857143,0.542857142857143,0.514285714285714,0.514285714285714,0.514285714285714,0.514285714285714,0.514285714285714,0.485714285714286,0.485714285714286,0.485714285714286,0.485714285714286,0.485714285714286,0.485714285714286,0.457142857142857,0.457142857142857,0.457142857142857,0.457142857142857,0.457142857142857,0.457142857142857,0.457142857142857,0.457142857142857,0.457142857142857,0.457142857142857,0.457142857142857,0.428571428571429,0.428571428571429,0.428571428571429,0.428571428571429,0.428571428571429,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.371428571428571,0.371428571428571,0.371428571428571,0.371428571428571,0.371428571428571,0.371428571428571,0.342857142857143,0.342857142857143,0.342857142857143,0.342857142857143,0.314285714285714,0.314285714285714,0.314285714285714,0.314285714285714,0.285714285714286,0.285714285714286,0.285714285714286,0.285714285714286,0.285714285714286,0.285714285714286,0.257142857142857,0.257142857142857,0.257142857142857,0.257142857142857,0.257142857142857,0.257142857142857,0.257142857142857,0.228571428571429,0.228571428571429,0.228571428571429,0.228571428571429,0.228571428571429,0.228571428571429,0.228571428571429,0.228571428571429,0.228571428571429,0.2,0.2,0.2,0.2,0.171428571428571,0.171428571428571,0.171428571428571,0.171428571428571,0.142857142857143,0.142857142857143,0.142857142857143,0.142857142857143,0.142857142857143,0.142857142857143,0.142857142857143,0.142857142857143,0.142857142857143,0.114285714285714,0.114285714285714,0.114285714285714,0.114285714285714,0.114285714285714,0.114285714285714,0.114285714285714,0.114285714285714,0.114285714285714,0.114285714285714,0.0857142857142857,0.0857142857142857,0.0857142857142857,0.0857142857142857,0.0857142857142857,0.0857142857142857,0.0857142857142857,0.0857142857142857,0.0857142857142857,0.0857142857142857,0.0857142857142857,0.0857142857142857,0.0857142857142857,0.0857142857142857,0.0857142857142857,0.0857142857142857,0.0571428571428571,0.0571428571428571,0.0571428571428571,0.0571428571428571,0.0571428571428571,0.0571428571428571,0.0571428571428571,0.0571428571428571,0.0571428571428571,0.0571428571428571,0.0571428571428571,0.0571428571428571,0.0571428571428571,0.0571428571428571,0.0571428571428571,0.0571428571428571,0.0571428571428571,0.0571428571428571,0.0571428571428571,0.0285714285714286,0.0285714285714286,0.0285714285714286,0.0285714285714286,0.0285714285714286,0.0285714285714286,0.0285714285714286,0.0285714285714286,0.0285714285714286,0.0285714285714286,0.0285714285714286,0.0285714285714286,0.0285714285714286,0.0285714285714286,0.0285714285714286,0.0285714285714286,0.0285714285714286,0.0285714285714286,0.0285714285714286,0.0285714285714286,0.0285714285714286,0.0285714285714286,0.0285714285714286,0.0285714285714286,0.0285714285714286,0.0285714285714286,0.0285714285714286,0.0285714285714286,0.0285714285714286,0.0285714285714286,0.0285714285714286,0.0285714285714286,0.0285714285714286,0.0285714285714286,0.0285714285714286,0.0285714285714286,0.0285714285714286,0.0285714285714286,0.0285714285714286],[365,279,231,286,245,373,323,295,247,298,316,180,277,257,240,311,228,240,335,235,247,214,248,195,298,275,235,242,254,201,163,245,251,223,212,134,159,235,319,167,166,168,96,228,222,180,228,264,238,197,228,247,248,147,198,207,163,216,126,157,188,271,169,122,252,200,239,119,228,112,156,227,216,107,204,253,227,134,266,289,133,225,154,93,91,120,199,121,263,272,230,92,206,245,107,176,199,124,169,169,176,206,137,150,153,119,228,101,216,118,236,94,97,251,122,200,107,242,53,91,157,125,87,78,86,161,171,80,172,228,226,210,75,111,105,285,134,148,237,78,108,164,183,76,131,105,149,171,104,92,239,236,180,141,194,91,78,184,75,86,85,105,196,246,73,116,83,266,172,95,239,78,184,184,113,59,94,54,95,77,83,175,92,69,69,262,189,181,57,58,152,83,144,51,97,87,93,77,187,87,89,48,43,289,224,74,85,78,79,34,184,49,81,66,26,91,46,148,63,73,226,152,50,81,195,77,83,130,49,90,73,122,60,63,71,39,257,200,63,48,46,72,94,149,98],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th rowspan=\"2\">gene<\/th>\n      <th colspan=\"3\">reads<\/th>\n      <th colspan=\"3\">UMIs<\/th>\n    <\/tr>\n    <tr>\n      <th>median count<\/th>\n      <th>N &gt; 4096<\/th>\n      <th>% &gt; 4096<\/th>\n      <th>median count<\/th>\n      <th>N &gt; 4096<\/th>\n      <th>% &gt; 4096<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"targets":1,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 0, 3, \",\", \".\");\n  }"},{"targets":4,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 0, 3, \",\", \".\");\n  }"},{"targets":3,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatPercentage(data, 1, 3, \",\", \".\");\n  }"},{"targets":6,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatPercentage(data, 1, 3, \",\", \".\");\n  }"},{"className":"dt-right","targets":[1,2,3,4,5,6]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":["options.columnDefs.0.render","options.columnDefs.1.render","options.columnDefs.2.render","options.columnDefs.3.render"],"jsHooks":[]}</script>
</div>
<p>We can investigate this list more formally using a statistical test for over-representation of gene ontology (GO) terms in this gene set. This is done using the <code>goana()</code> function from the <em><a href="https://bioconductor.org/packages/3.12/limma">limma</a></em> package. These results are shown in the table below and confirm that these genes are enriched for GO terms related to ribosomal and mitochondrial processes. This is something we frequently see with the mini-bulk protocol and the CEL-Seq protocol from which it is derived.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='co'># NOTE: kegga() was uninformative.</span>
<span class='va'>go</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/limma/man/goana.html'>goana</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/unlist.html'>unlist</a></span><span class='op'>(</span><span class='fu'>rowData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>[</span><span class='va'>g</span>, <span class='op'>]</span><span class='op'>)</span><span class='op'>$</span><span class='va'>NCBI.ENTREZID</span><span class='op'>)</span>, species <span class='op'>=</span> <span class='st'>"Mm"</span><span class='op'>)</span>
<span class='va'>tg</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/limma/man/topGO.html'>topGO</a></span><span class='op'>(</span><span class='va'>go</span>, number <span class='op'>=</span> <span class='cn'>Inf</span><span class='op'>)</span>
<span class='fu'>DT</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/DT/man/datatable.html'>datatable</a></span><span class='op'>(</span>
  <span class='va'>tg</span><span class='op'>[</span><span class='va'>tg</span><span class='op'>$</span><span class='va'>P.DE</span> <span class='op'>&lt;</span> <span class='fl'>0.05</span>, <span class='op'>]</span>,
  caption <span class='op'>=</span> <span class='st'>"Most significant GO terms for genes with read counts that exceed the 4096 threshold. Report columns are the ontology that the GO term belongs to ('Ont'); the number of genes in the GO term ('N'); the number of genes in the DE set ('DE'); and the p-value for over-representation of the GO term in the set ('P.DE'). The p-values returned are unadjusted for multiple testing because GO terms are often overlapping, so standard methods of p-value adjustment may be very conservative. This means that only very small p-values should be used for published results."</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>DT</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/DT/man/formatCurrency.html'>formatRound</a></span><span class='op'>(</span>columns <span class='op'>=</span> <span class='st'>"P.DE"</span>, digits <span class='op'>=</span> <span class='fl'>9</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div id="htmlwidget-7651aa4b06615acea5e8" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-7651aa4b06615acea5e8">{"x":{"filter":"none","caption":"<caption>Most significant GO terms for genes with read counts that exceed the 4096 threshold. Report columns are the ontology that the GO term belongs to ('Ont'); the number of genes in the GO term ('N'); the number of genes in the DE set ('DE'); and the p-value for over-representation of the GO term in the set ('P.DE'). The p-values returned are unadjusted for multiple testing because GO terms are often overlapping, so standard methods of p-value adjustment may be very conservative. This means that only very small p-values should be used for published results.<\/caption>","data":[["GO:0022626","GO:0005840","GO:0044391","GO:1990904","GO:0003735","GO:0003723","GO:0005829","GO:0032991","GO:0010467","GO:0034641","GO:0006412","GO:0043043","GO:0044271","GO:0006518","GO:0022627","GO:0005622","GO:0043229","GO:0005198","GO:0043604","GO:0044249","GO:0043226","GO:0015935","GO:0003676","GO:0005737","GO:0009058","GO:1901576","GO:0009059","GO:0044237","GO:1901363","GO:0034645","GO:0008152","GO:0097159","GO:0006807","GO:1901566","GO:0043603","GO:0022625","GO:0043228","GO:0043232","GO:0003729","GO:0043170","GO:0071704","GO:0015934","GO:0044238","GO:0019843","GO:0044260","GO:0010468","GO:0043227","GO:0043231","GO:0005844","GO:0005488","GO:0060255","GO:0019222","GO:0045202","GO:0051171","GO:0009987","GO:0006396","GO:0006139","GO:0005634","GO:0005515","GO:0043484","GO:0048519","GO:0006725","GO:0002181","GO:0046483","GO:0019899","GO:0048024","GO:0031323","GO:0080090","GO:1901564","GO:1901360","GO:0044267","GO:0008380","GO:0090304","GO:0019538","GO:0016070","GO:0048523","GO:0009889","GO:0110165","GO:0098794","GO:0010629","GO:0019219","GO:0050684","GO:2000112","GO:0010604","GO:0031326","GO:0070013","GO:0031974","GO:0043233","GO:0030054","GO:0010556","GO:0048522","GO:0009890","GO:0006397","GO:0009893","GO:0009892","GO:0070887","GO:0010558","GO:0010033","GO:0048518","GO:0000028","GO:0051252","GO:1903311","GO:0042274","GO:0000398","GO:0000375","GO:0000377","GO:2000113","GO:0045934","GO:0022613","GO:0010628","GO:0031324","GO:0010605","GO:0051173","GO:0051253","GO:0051172","GO:0044877","GO:0031327","GO:0071310","GO:0043021","GO:0009891","GO:0010557","GO:0042788","GO:0016071","GO:0031328","GO:0034654","GO:0099572","GO:0042254","GO:0018130","GO:0042221","GO:0022618","GO:0019438","GO:0043209","GO:0014069","GO:0031981","GO:0006119","GO:1901362","GO:0032279","GO:0071826","GO:0045935","GO:0031325","GO:0015990","GO:0015988","GO:0045892","GO:1903507","GO:1902679","GO:0071345","GO:0098984","GO:0005681","GO:0042255","GO:0008219","GO:1902600","GO:0045182","GO:2001242","GO:0098800","GO:0046034","GO:0042824","GO:0010941","GO:0071840","GO:0051169","GO:0006913","GO:0045595","GO:0036464","GO:0000122","GO:0010608","GO:0065003","GO:0051246","GO:0003697","GO:0051130","GO:0034622","GO:0043933","GO:0005740","GO:0034097","GO:0048856","GO:0032502","GO:0035770","GO:0002262","GO:0034101","GO:0051128","GO:0009888","GO:0036002","GO:0000380","GO:0042127","GO:0030425","GO:0060548","GO:0097447","GO:0008284","GO:0098576","GO:0031975","GO:0031967","GO:0005654","GO:0012501","GO:0048869","GO:0030154","GO:0006364","GO:0006417","GO:0031625","GO:0019904","GO:0043069","GO:0007275","GO:0006351","GO:0006355","GO:0097659","GO:0050789","GO:1903506","GO:0016072","GO:0032774","GO:0050794","GO:2001141","GO:0006915","GO:0034613","GO:0042611","GO:0006366","GO:0070727","GO:0043067","GO:0016604","GO:0051641","GO:0044389","GO:0030218","GO:0065008","GO:0005743","GO:0006357","GO:0000381","GO:0030670","GO:0006414","GO:0006996","GO:0043066","GO:0008283","GO:0042612","GO:0044085","GO:0050793","GO:0006091","GO:0045335","GO:0015931","GO:0042981","GO:0097193","GO:0002474","GO:0034248","GO:0043005","GO:0033119","GO:0006810","GO:0016043","GO:2001233","GO:0051239","GO:0034470","GO:0030099","GO:1902533","GO:0098803","GO:0050657","GO:0050658","GO:0048513","GO:0051234","GO:0071013","GO:0032268","GO:0065007","GO:0097190","GO:0051236","GO:0019900","GO:0019866","GO:0046907","GO:0031966","GO:0008134","GO:1903312","GO:0044403","GO:0032200","GO:0061061","GO:0098798","GO:0006886","GO:0048731","GO:0070469","GO:0071556","GO:0098553","GO:0046933","GO:0002428","GO:0042773","GO:0048872","GO:1901701","GO:0098761","GO:0098760","GO:0003746","GO:0030097","GO:0098589","GO:0036477","GO:0003730","GO:0006403","GO:0005730","GO:1901700","GO:0019901","GO:0002475","GO:0051254","GO:0005739","GO:0048584","GO:0120025","GO:0007049","GO:0070182","GO:0042288","GO:0006979","GO:2001244","GO:0034504","GO:0051179","GO:0098796","GO:0002483","GO:0019885","GO:0042592","GO:0051168","GO:0005753","GO:0045787","GO:0048534","GO:0051240","GO:0033365","GO:0048002","GO:0080135","GO:0045259","GO:0031090","GO:0001906","GO:0030139","GO:0019883","GO:0030666","GO:0016032","GO:0015252","GO:1903827","GO:0045740","GO:2000573","GO:0033043","GO:0016310","GO:0045893","GO:0051247","GO:0009653","GO:1903508","GO:1902680","GO:0045121","GO:0051052","GO:0051338","GO:0098857","GO:0034660","GO:0016607","GO:0005746","GO:0002376","GO:0015986","GO:0015985","GO:0048025","GO:0051726","GO:0002682","GO:2000278","GO:0045597","GO:0022904","GO:0042542","GO:0042995","GO:2000026","GO:0032070","GO:0062061","GO:0051054","GO:0009790","GO:0002520","GO:0022607","GO:0048468","GO:0000723","GO:0031252","GO:0001916","GO:0022900","GO:0042802","GO:0048583","GO:1902531","GO:0006611","GO:0010638","GO:0051093","GO:0051649","GO:0048608","GO:0050821","GO:0006754","GO:0002135","GO:2000814","GO:0061458","GO:0001914","GO:0072593","GO:0008406","GO:0042610","GO:0010639","GO:0060249","GO:0035162","GO:1903706","GO:0045137","GO:0090079","GO:0032270","GO:0010942","GO:0035556","GO:0003682","GO:0030881","GO:0046977","GO:0050686","GO:0008187","GO:2001252","GO:0045185","GO:0022402","GO:0015980","GO:0032870","GO:1902882","GO:0002486","GO:0002697","GO:0080134","GO:0044297","GO:0008097","GO:0005200","GO:2000147","GO:0090068","GO:0002484","GO:0002476","GO:0007548","GO:0051347","GO:0009206","GO:0001726","GO:0035295","GO:0045944","GO:0034614","GO:0010634","GO:0044419","GO:0000302","GO:0009145","GO:0003677","GO:0035254","GO:0006796","GO:0005102","GO:0030176","GO:0051129","GO:0042287","GO:0060429","GO:0040017","GO:1902494","GO:0032507","GO:0009201","GO:0051241","GO:0071495","GO:2000377","GO:0006457","GO:0043085","GO:0001913","GO:0031369","GO:0009719","GO:0051272","GO:0006793","GO:0045333","GO:1990204","GO:0120034","GO:0002485","GO:1990829","GO:1990428","GO:0017098","GO:0071426","GO:0062197","GO:0032879","GO:0007569","GO:0009725","GO:1901796","GO:0071166","GO:0015078","GO:0034399","GO:0031227","GO:0002699","GO:0002711","GO:0044093","GO:0034599","GO:0019079","GO:0010833","GO:0019882","GO:0051094","GO:0008104","GO:0071364","GO:0008135","GO:0042776","GO:0042608","GO:0043009","GO:0045637","GO:0071897","GO:1903201","GO:0019887","GO:0017076","GO:0030335","GO:0009142","GO:0071705","GO:0009205","GO:0035639","GO:0070849","GO:0009792","GO:0043532","GO:0004691","GO:0032564","GO:0002134","GO:0097367","GO:0010631","GO:0031341","GO:0098805","GO:0006950","GO:0001912","GO:1901265","GO:0000166","GO:0090132","GO:0042605","GO:0032272","GO:0009199","GO:0045727","GO:0090130","GO:0002252","GO:0051082","GO:0031301","GO:0033044","GO:1903320","GO:0071157","GO:0072520","GO:0002761","GO:0065009","GO:2001235","GO:0006259","GO:0006405","GO:0009967","GO:0002705","GO:0016469","GO:0014706","GO:0016363","GO:0009144","GO:0072331","GO:0045646","GO:0070301","GO:0034341","GO:0043549","GO:0007568","GO:0005747","GO:0030964","GO:0045271","GO:0032558","GO:0070934","GO:0000275","GO:1990715","GO:0032077","GO:0032557","GO:0051022","GO:0098973","GO:0071987","GO:0007063","GO:0032555","GO:0060284","GO:1900407","GO:1903829","GO:0042277","GO:0009152","GO:0051098","GO:0005797","GO:0048585","GO:0032553","GO:0002684","GO:0001816","GO:0031503","GO:0043565","GO:0031401","GO:0031399","GO:0031343","GO:0045184","GO:0060537","GO:0017148","GO:0050792","GO:0042692","GO:0019207","GO:0009966","GO:0002709","GO:1902883","GO:0120035","GO:0036473","GO:0031300","GO:0031346","GO:0071880","GO:0007549","GO:1901798","GO:0070937","GO:0004690","GO:0098963","GO:0098961","GO:0097064","GO:0032075","GO:0045261","GO:0032554","GO:2000977","GO:0050733","GO:0035368","GO:1905323","GO:0051031","GO:0042775","GO:0043536","GO:0031396","GO:0009260","GO:0045596","GO:0031344","GO:0044087","GO:0099010","GO:0050136","GO:0008137","GO:0010592","GO:1902175","GO:0043065","GO:0015672","GO:0070925","GO:0010595","GO:0006278","GO:0007004","GO:0043068","GO:1902884","GO:0032069","GO:0050807","GO:0015833","GO:0001910","GO:0046390","GO:0034250","GO:0050790","GO:0048646","GO:0086023","GO:1902177","GO:1903608","GO:2000812","GO:0032071","GO:0031985","GO:0019080","GO:0071875","GO:0003954","GO:0008266","GO:0043903","GO:0017111","GO:0006164","GO:0061629","GO:0072594","GO:0002573","GO:0045639","GO:0010035","GO:0001667","GO:0019058","GO:0071353","GO:0070198","GO:0009141","GO:0006928","GO:0031647","GO:0002521","GO:0042886","GO:0033036","GO:0040029","GO:0006275","GO:0010564","GO:0050803","GO:0034249","GO:1903708","GO:0090398","GO:0050776","GO:0016477","GO:0033674","GO:0035198","GO:0009303","GO:0006606","GO:0072522","GO:0051147","GO:0045785","GO:0035239","GO:0032271","GO:0050673","GO:0050764","GO:0098685","GO:0099080","GO:0032552","GO:0051021","GO:0070180","GO:0030235","GO:0050815","GO:0034182","GO:0000056","GO:1903047","GO:0051961","GO:0015031","GO:0070069","GO:1904031","GO:0070670","GO:0045860","GO:1901698","GO:0071346","GO:2000379","GO:0051170","GO:0007566","GO:0030032","GO:0030837","GO:0002763","GO:2001234","GO:0043393","GO:0008585","GO:0033157","GO:0070971","GO:0051569","GO:0010939","GO:0051150","GO:0032587","GO:0050678","GO:1901699","GO:0051276","GO:0010632","GO:0010494","GO:0001541","GO:0046545","GO:0010646","GO:0045786","GO:0051984","GO:1902745","GO:0045937","GO:0010562","GO:0043534","GO:0071363","GO:0019220","GO:0051174","GO:0034091","GO:0030957","GO:0010647","GO:0032680","GO:0023051","GO:0099563","GO:0023056","GO:1902105","GO:0048589","GO:0006955","GO:0045807","GO:1903555","GO:0045664","GO:0071103","GO:0060341","GO:0001817","GO:1902107","GO:0050767","GO:0051973","GO:0010591","GO:0042273","GO:0002683","GO:0070848","GO:0016462","GO:0048870","GO:0051674","GO:0048666","GO:0016817","GO:0016818","GO:1903409","GO:0043168","GO:0007010","GO:0098772","GO:0042327","GO:0030554","GO:2000145","GO:0097067","GO:0075522","GO:0002318","GO:1900264","GO:1900262","GO:0006450","GO:0070181","GO:0000805","GO:0032640","GO:0031060","GO:0033218","GO:0043025","GO:0051248","GO:0071514","GO:0071706","GO:0032880","GO:0042325","GO:0030234","GO:1903362","GO:0031672","GO:0016655","GO:0001934","GO:0120032","GO:0050808","GO:0030141","GO:0033554","GO:0005524","GO:1904029","GO:0086103","GO:0005750","GO:0045876","GO:0019103","GO:0045275","GO:0002456","GO:0051049","GO:0051960","GO:0060491","GO:0035255","GO:0000803","GO:0032204","GO:0033045","GO:0002703","GO:0099522","GO:0019933","GO:0000278","GO:0016651","GO:0032501","GO:0031175","GO:0017069","GO:0050766","GO:0044294","GO:0098935","GO:0034088","GO:0031386","GO:0043023","GO:0042989","GO:0006323","GO:0048699","GO:0003725","GO:0048524","GO:0070997","GO:0006839","GO:0099173","GO:0001067","GO:0030334","GO:0008584","GO:2000772","GO:0022008","GO:0040012","GO:0097581","GO:0007189","GO:0046546","GO:0006812","GO:0032269","GO:0050778","GO:0007517","GO:0051716","GO:0001909","GO:0005795","GO:0000307","GO:0061980","GO:0051050","GO:0045859","GO:0009165","GO:0051301","GO:0034086","GO:0030182","GO:0008631","GO:0051702","GO:0002708","GO:0098978","GO:0071702","GO:1902743","GO:1901293","GO:0014070","GO:0051051","GO:0016458","GO:1902275","GO:0031333","GO:0002824","GO:0042110","GO:0002371","GO:1990124","GO:0051152","GO:0098871","GO:1905063","GO:0097066","GO:0098847","GO:0032559","GO:0005525","GO:0070062","GO:0031982","GO:0043542","GO:0030217","GO:0001944","GO:0046660","GO:0051651","GO:0003727","GO:0042645","GO:0009295","GO:0031256","GO:0043254","GO:0043535","GO:0032550","GO:0036094","GO:0021954","GO:2001243","GO:0090329","GO:0051972","GO:0071900","GO:0006402","GO:0017038","GO:0060627","GO:0150105","GO:0033178","GO:1904814","GO:1901836","GO:0019081","GO:0072359","GO:0003690","GO:0045638","GO:0031970","GO:0051270","GO:0140297","GO:0032549","GO:0030029","GO:0007188","GO:0002821","GO:0001883","GO:0030490","GO:0098781","GO:0033120","GO:0035196","GO:0090342","GO:0071902","GO:0030036","GO:0045069","GO:0010948","GO:1903050","GO:0051258","GO:0071407","GO:0030100","GO:0001046","GO:1902903","GO:0071396","GO:1903707","GO:0019935","GO:0000463","GO:0098885","GO:0045655","GO:0043024","GO:0000054","GO:0033750","GO:0001525","GO:0031050","GO:0032206","GO:0070918","GO:0051495","GO:1903561","GO:0090575","GO:1990837","GO:0051402","GO:0001882","GO:1901653","GO:0042491","GO:1903202","GO:1901214","GO:1901989","GO:0040007","GO:0005667","GO:0019001","GO:0032561","GO:0099640","GO:0050665","GO:0090336","GO:0045793","GO:0051571","GO:0033194","GO:0071428","GO:0030155","GO:1900408","GO:0051983","GO:0016874","GO:0031056","GO:0033135","GO:0010970","GO:0006413","GO:0055085","GO:0042743","GO:0010823","GO:0004860","GO:1904705","GO:1990874","GO:0150034","GO:0008022","GO:0000976","GO:0002444","GO:1990782","GO:0005938","GO:0061834","GO:0031002","GO:0072749","GO:0071228","GO:0090732","GO:0097132","GO:0014828","GO:0044796","GO:0060940","GO:1990631","GO:0071964","GO:0075713","GO:0033380","GO:0062072","GO:0098577","GO:1905877","GO:0097726","GO:0098547","GO:0033382","GO:0033373","GO:0033379","GO:0033370","GO:0033377","GO:0097326","GO:0035500","GO:1905454","GO:1905875","GO:1901581","GO:1901389","GO:0051511","GO:0071045","GO:1990707","GO:1990917","GO:0032358","GO:0043626","GO:0004638","GO:0004639","GO:0061826","GO:0017130","GO:1905062","GO:1902546","GO:1903226","GO:2000784","GO:1905926","GO:1905873","GO:0120072","GO:1901582","GO:0040032","GO:0035128","GO:0099143","GO:0033368","GO:0033375","GO:0033367","GO:1902889","GO:0033374","GO:0042543","GO:0120065","GO:1900022","GO:0060735","GO:2000782","GO:1905924","GO:1902629","GO:0010849","GO:0120071","GO:1901580","GO:0051510","GO:0036275","GO:1901328","GO:0061481","GO:1990280","GO:0005020","GO:0120064","GO:0033371","GO:0097393","GO:0097394","GO:0034693","GO:0061835","GO:0010594","GO:0051896","GO:0051131","GO:0038093","GO:0042790","GO:0031492","GO:0009886","GO:0042026","GO:0051018","GO:2000351","GO:0051100","GO:0051493","GO:0002367","GO:0110020","GO:0048145","GO:1903426","GO:0098655","GO:0051568","GO:0030219","GO:0070265","GO:0007062","GO:0048514","GO:0034330","GO:0031329","GO:0032091","GO:0008544","GO:0032989","GO:0048598","GO:1902373","GO:0045309","GO:0048144","GO:1900006","GO:0051897","GO:0090201","GO:0099571","GO:0046661","GO:0051494","GO:0045931","GO:0042176","GO:0040011","GO:0045936","GO:0010563","GO:0043230","GO:0032231","GO:0044092","GO:0000785","GO:0072577","GO:0035315","GO:0061614","GO:0035019","GO:0098662","GO:0008637","GO:0007399","GO:1901652","GO:0051235","GO:0019210","GO:0043462","GO:0071156","GO:1904356","GO:0001836","GO:0001221","GO:0009628","GO:0044292","GO:2000773","GO:0070064","GO:0006349","GO:0045649","GO:0045277","GO:0099186","GO:0070201","GO:0007187","GO:0048260","GO:0055114","GO:0007059","GO:0009150","GO:0016922","GO:0050896","GO:0048259","GO:0060055","GO:0036336","GO:0062033","GO:1903209","GO:0050434","GO:0090066","GO:1901215","GO:0034968","GO:0022414","GO:0000003","GO:0030330","GO:0031640","GO:0043022","GO:0032386","GO:0044265","GO:0051427","GO:0048814","GO:0071375","GO:0090087","GO:0048813","GO:0043524","GO:0006401","GO:0032535","GO:0098793","GO:0060999","GO:0032760","GO:0120036","GO:0004129","GO:0015002","GO:0006120","GO:0050998","GO:0016676","GO:2000010","GO:0048156","GO:0006325","GO:0044770","GO:0009259","GO:0001701","GO:1903557","GO:0031400","GO:0032446","GO:0048738","GO:0043434","GO:0050768","GO:0009048","GO:0051450","GO:0016675","GO:0046930","GO:0032930","GO:0097524","GO:0010770","GO:0008092","GO:0061136","GO:0030863","GO:0042826","GO:0009894","GO:0022610","GO:0003006","GO:0048568","GO:0006997","GO:0051783","GO:0101031","GO:0070199","GO:0031683","GO:0051457","GO:0046823","GO:0050765","GO:0005721","GO:1900409","GO:0070412","GO:0090335","GO:0000979","GO:0048812","GO:0019693","GO:0000987","GO:0031410","GO:0009055","GO:1902369","GO:0009913","GO:0046822","GO:0030705","GO:0097708","GO:0048146","GO:0034394","GO:0033047","GO:0060996","GO:0099503","GO:0005856","GO:0043197","GO:0099111","GO:0002183","GO:0048027","GO:1903206","GO:1901032","GO:0035813","GO:0043523","GO:0003779","GO:0010811","GO:0061013","GO:0043491","GO:0002706","GO:0007519","GO:1901137","GO:0002443","GO:0010769","GO:0006605","GO:0006305","GO:0006306","GO:1990841","GO:0010856","GO:0060936","GO:0060935","GO:0036018","GO:0071283","GO:0099115","GO:1990765","GO:0030142","GO:0071682","GO:0035854","GO:0060939","GO:0060938","GO:0038162","GO:0019043","GO:0098770","GO:1990565","GO:1990630","GO:0035026","GO:0099403","GO:0033364","GO:0097324","GO:0004478","GO:0048269","GO:0002397","GO:0099404","GO:1902340","GO:1905450","GO:1904908","GO:2000276","GO:1904815","GO:1903614","GO:0071159","GO:0070557","GO:0002502","GO:0047066","GO:1905053","GO:1904343","GO:0051343","GO:2001272","GO:2001150","GO:2000880","GO:2000771","GO:1904434","GO:1905437","GO:0045716","GO:2000878","GO:0060265","GO:1905663","GO:1904437","GO:0035120","GO:0035127","GO:0000701","GO:1905051","GO:1904341","GO:2001148","GO:0090089","GO:1902544","GO:2000769","GO:1904432","GO:1905435","GO:1904907","GO:1905453","GO:0051621","GO:0090088","GO:1905871","GO:1904435","GO:0140244","GO:0099577","GO:0036017","GO:0010041","GO:0006407","GO:0097226","GO:0120063","GO:0001097","GO:0030030","GO:0061844","GO:0033138","GO:0032273","GO:0047485","GO:0002822","GO:0050773","GO:0016491","GO:0110053","GO:0002039","GO:0006163","GO:0000902","GO:0120039","GO:0030855","GO:0030325","GO:0017134","GO:0007064","GO:0090344","GO:0099524","GO:0044183","GO:0051570","GO:0032928","GO:0005689","GO:0098660","GO:0018022","GO:0003712","GO:0044309","GO:1902905","GO:0022603","GO:0099175","GO:0048858","GO:0010975","GO:0000904","GO:0003007","GO:0071417","GO:0060997","GO:0021879","GO:0030971","GO:0009607","GO:0009986","GO:0000977","GO:0003785","GO:0002478","GO:0002102","GO:0035812","GO:0032956","GO:0001669","GO:0044839","GO:0002698","GO:0009060","GO:0022604","GO:1901655","GO:0001819","GO:0001932","GO:0042306","GO:0051145","GO:0060538","GO:0005884","GO:0051223","GO:0051016","GO:0030224","GO:1903131","GO:0008143","GO:0045737","GO:1901532","GO:0097435","GO:0007050","GO:0016234","GO:0000079","GO:0017053","GO:0016567","GO:0030426","GO:1901224","GO:1903322","GO:1900180","GO:0044305","GO:0045292","GO:0039694","GO:0072332","GO:0060563","GO:0044089","GO:0033993","GO:0034220","GO:0097305","GO:0032990","GO:0001775","GO:0002819","GO:0008361","GO:0000470","GO:0061082","GO:1902751","GO:0030947","GO:0061827","GO:0098918","GO:0048667","GO:0006897","GO:0071824","GO:0010810","GO:0098542","GO:0048839","GO:0008088","GO:0001568","GO:0006464","GO:0036211","GO:0030427","GO:0043167","GO:0016571","GO:0090316","GO:0051881","GO:1904589","GO:0095500","GO:0050681","GO:0030332","GO:0044062","GO:1902253","GO:0060074","GO:0003924","GO:0005575","GO:0043414","GO:1990932","GO:0002481","GO:0034751","GO:0072717","GO:0072752","GO:0000448","GO:0061830","GO:0048248","GO:0097129","GO:0032139","GO:0030337","GO:0005853","GO:0097451","GO:0060819","GO:1905863","GO:0035691","GO:0032398","GO:1990498","GO:1902990","GO:0036194","GO:0036195","GO:0005854","GO:0140199","GO:0033128","GO:0010593","GO:0034183","GO:0034092","GO:1904761","GO:0002501","GO:0090725","GO:0050816","GO:0061825","GO:1902425","GO:0045585","GO:2000373","GO:2000170","GO:1904349","GO:0045905","GO:1902423","GO:0045583","GO:2000371","GO:0060699","GO:1900169","GO:2000446","GO:1902445","GO:2000828","GO:0010725","GO:1905661","GO:0009608","GO:0009609","GO:0042134","GO:1990935","GO:0031509","GO:0140461","GO:0097100","GO:0006452","GO:0006409","GO:0071431","GO:0045298","GO:0051656","GO:0060113","GO:0042326","GO:0035270","GO:2001251","GO:0071354","GO:2000352","GO:0051491","GO:0048026","GO:0031297","GO:0021872","GO:0051899","GO:0005758","GO:1990542","GO:1902554","GO:0048511","GO:0007155","GO:0002009","GO:0050680","GO:2001020","GO:0010721","GO:0016209","GO:0000978","GO:0006952","GO:0007163","GO:0048471","GO:1905145","GO:0008200","GO:0070266","GO:0045672","GO:1903205","GO:0045652","GO:1905144","GO:0002065","GO:0004386","GO:0033613","GO:0002718","GO:0031497","GO:0016358","GO:0048729","GO:0006811","GO:0051707","GO:0035257","GO:0043207","GO:0001503","GO:0019884","GO:0008180","GO:0097718","GO:1902686","GO:0006356","GO:0070741","GO:0042162","GO:0044728","GO:0035304","GO:0031348","GO:0072521","GO:0010720","GO:1990830","GO:1901992","GO:1900182","GO:0034502","GO:0035258","GO:0043412","GO:0005813","GO:1990823","GO:0016248","GO:1905360","GO:0005834","GO:0006972","GO:0000462","GO:0035308","GO:0043243","GO:0035025","GO:0043516","GO:0046686","GO:0003724","GO:0003015","GO:0045087","GO:0030864","GO:0042490","GO:0051149","GO:0042770","GO:0008201","GO:0050728","GO:0009314","GO:0007044","GO:0051567","GO:0032232","GO:0033260","GO:0070717","GO:1904358","GO:1904707","GO:0035886","GO:0055067","GO:0016887","GO:0009968","GO:0021953","GO:0031334","GO:0006189","GO:0090135","GO:0030043","GO:0002477","GO:0071349","GO:0051365","GO:0036216","GO:0051430","GO:0045065","GO:0044547","GO:0071963","GO:0090498","GO:0043402","GO:0019659","GO:0019661","GO:0019660","GO:0021888","GO:0021886","GO:0035061","GO:0038109","GO:0004459","GO:0019244","GO:0006272","GO:0002396","GO:0000700","GO:0005757","GO:0031852","GO:0036446","GO:0071878","GO:2000173","GO:0006933","GO:1903660","GO:2000978","GO:1902219","GO:2000623","GO:1901837","GO:0032079","GO:0042998","GO:0051835","GO:0045901","GO:0045903","GO:0033366","GO:0047134","GO:0140192","GO:0090230","GO:1904760","GO:2000622","GO:1903613","GO:0072716","GO:1901355","GO:0036215","GO:0006556","GO:0000322","GO:0002309","GO:0032201","GO:0070053","GO:0035976","GO:0031800","GO:0009826","GO:0019042","GO:0006468","GO:0071383","GO:2000058","GO:1903900","GO:0009266","GO:0033002","GO:0007088","GO:0055072","GO:0036474","GO:1990748","GO:0030866","GO:0035794","GO:1902229","GO:0048863","GO:0030098","GO:0048638","GO:0150115","GO:0033077","GO:0032970","GO:0007346","GO:0051607","GO:0006898","GO:0006284","GO:0010661","GO:0070528","GO:0090322","GO:0042175","GO:0031490","GO:1903313","GO:1904035","GO:0009408","GO:0002244","GO:1902904","GO:0030278","GO:0008285","GO:0051492","GO:0019730","GO:0120111","GO:0032781","GO:1900087","GO:1905710","GO:1900026","GO:0042554","GO:0042311","GO:0070647","GO:1903578","GO:0060998","GO:0006260","GO:0007565","GO:0006338","GO:1902749","GO:0010660","GO:0043583","GO:0043531","GO:0045005","GO:0031941","GO:0006730","GO:0045981","GO:1900544","GO:0051205","GO:0098588","GO:0006334","GO:0051017","GO:0031047","GO:0097061","GO:1902911","GO:0012505","GO:0051693","GO:0044786","GO:0001965","GO:0030225","GO:0072595","GO:0097300","GO:0098801","GO:0007171","GO:0001740","GO:0006287","GO:0044838","GO:0070087","GO:0009597","GO:0035442","GO:0042938","GO:0032135","GO:0030222","GO:0006113","GO:0044027","GO:0004457","GO:0071817","GO:0045541","GO:0090206","GO:1902951","GO:0061086","GO:1902744","GO:0060266","GO:0045875","GO:0106119","GO:1990441","GO:0016681","GO:0016721","GO:0035898","GO:0031508","GO:0140462","GO:0015288","GO:1905605","GO:0032849","GO:0010729","GO:0010912","GO:0034184","GO:0033864","GO:0045040","GO:1902463","GO:0030292","GO:0060623","GO:2001270","GO:1902218","GO:0010911","GO:2000275","GO:1904347","GO:0001069","GO:0072344","GO:0010269","GO:0051029","GO:1990770","GO:0004784","GO:0048014","GO:0008121","GO:0031330","GO:0065004","GO:0030833","GO:0050890","GO:0006304","GO:0009117","GO:0010648","GO:0030659","GO:0061572","GO:0001709","GO:0021955","GO:0031062","GO:2000008","GO:0055006","GO:0005902","GO:0023057","GO:0016049","GO:0043410","GO:0005694","GO:0006333","GO:0001890","GO:1901654","GO:0007006","GO:0010657","GO:0007167","GO:0030424","GO:0035107","GO:0035108","GO:0050685","GO:0046782","GO:0045321","GO:0010243","GO:0019903","GO:0035064","GO:0140034","GO:0048525","GO:0051219","GO:0043488","GO:1901222","GO:0005576","GO:0006753","GO:0071385","GO:0097237","GO:2000144","GO:1902895","GO:0042169","GO:0007015","GO:0045665","GO:0009057","GO:0007507","GO:0051348","GO:0007051","GO:0000139","GO:0001894","GO:0106027","GO:0090199","GO:1901031","GO:0000245","GO:0032259","GO:0007219","GO:0047391","GO:0005642","GO:0051434","GO:0071316","GO:0051429","GO:0001651","GO:0015074","GO:0015616","GO:0044806","GO:0060789","GO:0030210","GO:1990712","GO:0071204","GO:0019249","GO:0097413","GO:0032762","GO:0032405","GO:0002372","GO:0045650","GO:0045347","GO:0046826","GO:1901164","GO:0071598","GO:0051620","GO:0016679","GO:0010726","GO:0034093","GO:0002735","GO:0010571","GO:1904751","GO:0060267","GO:1902949","GO:1905065","GO:0060319","GO:0071896","GO:0000182","GO:0051342","GO:1903224","GO:0002733","GO:0099576","GO:0060264","GO:0070671","GO:0071107","GO:0060708","GO:0048039","GO:1990948","GO:0000819","GO:0031589","GO:0034655","GO:0044295","GO:0071384","GO:0030261","GO:0061647","GO:0030835","GO:0090317","GO:0098926","GO:0003254","GO:0048536","GO:0019083","GO:0000956","GO:0019932","GO:0042393","GO:0000082","GO:0008213","GO:0006479","GO:0000781","GO:0051028","GO:0032388","GO:0048588","GO:0045214","GO:0099106","GO:0038061","GO:2001022","GO:0019725","GO:0032993","GO:0022898","GO:0007339","GO:0031491","GO:1903580","GO:0048010","GO:0030041","GO:0050900","GO:0030038","GO:0051225","GO:0043149","GO:0008064","GO:0015077","GO:0005759","GO:0048255","GO:1902808","GO:0001102","GO:0098797","GO:0012506","GO:0018193","GO:0021902","GO:0044026","GO:0010385","GO:0003160","GO:0044322","GO:0071169","GO:0060613","GO:0008627","GO:0030274","GO:0022605","GO:0060374","GO:0042629","GO:1990247","GO:0042985","GO:0070345","GO:0051572","GO:1903208","GO:0045656","GO:0060268","GO:0048387","GO:1901621","GO:0036476","GO:0035672","GO:0008228","GO:0042382","GO:0030578","GO:1905538","GO:0051987","GO:2000138","GO:2000510","GO:0060501","GO:1904306","GO:0048170","GO:1905461","GO:0030628","GO:0034214","GO:1902396","GO:0032847","GO:1905449","GO:1903207","GO:0045714","GO:1901388","GO:0140243","GO:0099547","GO:0002536","GO:0033762","GO:0000055","GO:0032202","GO:0008379","GO:0030911","GO:0055105","GO:0016461","GO:0019062","GO:0010038","GO:0055002","GO:0018205","GO:1904951"],["cytosolic ribosome","ribosome","ribosomal subunit","ribonucleoprotein complex","structural constituent of ribosome","RNA binding","cytosol","protein-containing complex","gene expression","cellular nitrogen compound metabolic process","translation","peptide biosynthetic process","cellular nitrogen compound biosynthetic process","peptide metabolic process","cytosolic small ribosomal subunit","intracellular","intracellular organelle","structural molecule activity","amide biosynthetic process","cellular biosynthetic process","organelle","small ribosomal subunit","nucleic acid binding","cytoplasm","biosynthetic process","organic substance biosynthetic process","macromolecule biosynthetic process","cellular metabolic process","heterocyclic compound binding","cellular macromolecule biosynthetic process","metabolic process","organic cyclic compound binding","nitrogen compound metabolic process","organonitrogen compound biosynthetic process","cellular amide metabolic process","cytosolic large ribosomal subunit","non-membrane-bounded organelle","intracellular non-membrane-bounded organelle","mRNA binding","macromolecule metabolic process","organic substance metabolic process","large ribosomal subunit","primary metabolic process","rRNA binding","cellular macromolecule metabolic process","regulation of gene expression","membrane-bounded organelle","intracellular membrane-bounded organelle","polysome","binding","regulation of macromolecule metabolic process","regulation of metabolic process","synapse","regulation of nitrogen compound metabolic process","cellular process","RNA processing","nucleobase-containing compound metabolic process","nucleus","protein binding","regulation of RNA splicing","negative regulation of biological process","cellular aromatic compound metabolic process","cytoplasmic translation","heterocycle metabolic process","enzyme binding","regulation of mRNA splicing, via spliceosome","regulation of cellular metabolic process","regulation of primary metabolic process","organonitrogen compound metabolic process","organic cyclic compound metabolic process","cellular protein metabolic process","RNA splicing","nucleic acid metabolic process","protein metabolic process","RNA metabolic process","negative regulation of cellular process","regulation of biosynthetic process","cellular anatomical entity","postsynapse","negative regulation of gene expression","regulation of nucleobase-containing compound metabolic process","regulation of mRNA processing","regulation of cellular macromolecule biosynthetic process","positive regulation of macromolecule metabolic process","regulation of cellular biosynthetic process","intracellular organelle lumen","membrane-enclosed lumen","organelle lumen","cell junction","regulation of macromolecule biosynthetic process","positive regulation of cellular process","negative regulation of biosynthetic process","mRNA processing","positive regulation of metabolic process","negative regulation of metabolic process","cellular response to chemical stimulus","negative regulation of macromolecule biosynthetic process","response to organic substance","positive regulation of biological process","ribosomal small subunit assembly","regulation of RNA metabolic process","regulation of mRNA metabolic process","ribosomal small subunit biogenesis","mRNA splicing, via spliceosome","RNA splicing, via transesterification reactions","RNA splicing, via transesterification reactions with bulged adenosine as nucleophile","negative regulation of cellular macromolecule biosynthetic process","negative regulation of nucleobase-containing compound metabolic process","ribonucleoprotein complex biogenesis","positive regulation of gene expression","negative regulation of cellular metabolic process","negative regulation of macromolecule metabolic process","positive regulation of nitrogen compound metabolic process","negative regulation of RNA metabolic process","negative regulation of nitrogen compound metabolic process","protein-containing complex binding","negative regulation of cellular biosynthetic process","cellular response to organic substance","ribonucleoprotein complex binding","positive regulation of biosynthetic process","positive regulation of macromolecule biosynthetic process","polysomal ribosome","mRNA metabolic process","positive regulation of cellular biosynthetic process","nucleobase-containing compound biosynthetic process","postsynaptic specialization","ribosome biogenesis","heterocycle biosynthetic process","response to chemical","ribonucleoprotein complex assembly","aromatic compound biosynthetic process","myelin sheath","postsynaptic density","nuclear lumen","oxidative phosphorylation","organic cyclic compound biosynthetic process","asymmetric synapse","ribonucleoprotein complex subunit organization","positive regulation of nucleobase-containing compound metabolic process","positive regulation of cellular metabolic process","electron transport coupled proton transport","energy coupled proton transmembrane transport, against electrochemical gradient","negative regulation of transcription, DNA-templated","negative regulation of nucleic acid-templated transcription","negative regulation of RNA biosynthetic process","cellular response to cytokine stimulus","neuron to neuron synapse","spliceosomal complex","ribosome assembly","cell death","proton transmembrane transport","translation regulator activity","regulation of intrinsic apoptotic signaling pathway","inner mitochondrial membrane protein complex","ATP metabolic process","MHC class I peptide loading complex","regulation of cell death","cellular component organization or biogenesis","nuclear transport","nucleocytoplasmic transport","regulation of cell differentiation","cytoplasmic ribonucleoprotein granule","negative regulation of transcription by RNA polymerase II","posttranscriptional regulation of gene expression","protein-containing complex assembly","regulation of protein metabolic process","single-stranded DNA binding","positive regulation of cellular component organization","cellular protein-containing complex assembly","protein-containing complex subunit organization","mitochondrial envelope","response to cytokine","anatomical structure development","developmental process","ribonucleoprotein granule","myeloid cell homeostasis","erythrocyte homeostasis","regulation of cellular component organization","tissue development","pre-mRNA binding","alternative mRNA splicing, via spliceosome","regulation of cell population proliferation","dendrite","negative regulation of cell death","dendritic tree","positive regulation of cell population proliferation","lumenal side of membrane","envelope","organelle envelope","nucleoplasm","programmed cell death","cellular developmental process","cell differentiation","rRNA processing","regulation of translation","ubiquitin protein ligase binding","protein domain specific binding","negative regulation of programmed cell death","multicellular organism development","transcription, DNA-templated","regulation of transcription, DNA-templated","nucleic acid-templated transcription","regulation of biological process","regulation of nucleic acid-templated transcription","rRNA metabolic process","RNA biosynthetic process","regulation of cellular process","regulation of RNA biosynthetic process","apoptotic process","cellular protein localization","MHC protein complex","transcription by RNA polymerase II","cellular macromolecule localization","regulation of programmed cell death","nuclear body","cellular localization","ubiquitin-like protein ligase binding","erythrocyte differentiation","regulation of biological quality","mitochondrial inner membrane","regulation of transcription by RNA polymerase II","regulation of alternative mRNA splicing, via spliceosome","phagocytic vesicle membrane","translational elongation","organelle organization","negative regulation of apoptotic process","cell population proliferation","MHC class I protein complex","cellular component biogenesis","regulation of developmental process","generation of precursor metabolites and energy","phagocytic vesicle","nucleobase-containing compound transport","regulation of apoptotic process","intrinsic apoptotic signaling pathway","antigen processing and presentation of peptide antigen via MHC class I","regulation of cellular amide metabolic process","neuron projection","negative regulation of RNA splicing","transport","cellular component organization","regulation of apoptotic signaling pathway","regulation of multicellular organismal process","ncRNA processing","myeloid cell differentiation","positive regulation of intracellular signal transduction","respiratory chain complex","nucleic acid transport","RNA transport","animal organ development","establishment of localization","catalytic step 2 spliceosome","regulation of cellular protein metabolic process","biological regulation","apoptotic signaling pathway","establishment of RNA localization","kinase binding","organelle inner membrane","intracellular transport","mitochondrial membrane","transcription factor binding","negative regulation of mRNA metabolic process","symbiotic process","telomere organization","muscle structure development","mitochondrial protein complex","intracellular protein transport","system development","respirasome","integral component of lumenal side of endoplasmic reticulum membrane","lumenal side of endoplasmic reticulum membrane","proton-transporting ATP synthase activity, rotational mechanism","antigen processing and presentation of peptide antigen via MHC class Ib","ATP synthesis coupled electron transport","homeostasis of number of cells","cellular response to oxygen-containing compound","cellular response to interleukin-7","response to interleukin-7","translation elongation factor activity","hemopoiesis","membrane region","somatodendritic compartment","mRNA 3'-UTR binding","RNA localization","nucleolus","response to oxygen-containing compound","protein kinase binding","antigen processing and presentation via MHC class Ib","positive regulation of RNA metabolic process","mitochondrion","positive regulation of response to stimulus","plasma membrane bounded cell projection","cell cycle","DNA polymerase binding","MHC class I protein binding","response to oxidative stress","positive regulation of intrinsic apoptotic signaling pathway","protein localization to nucleus","localization","membrane protein complex","antigen processing and presentation of endogenous peptide antigen","antigen processing and presentation of endogenous peptide antigen via MHC class I","homeostatic process","nuclear export","mitochondrial proton-transporting ATP synthase complex","positive regulation of cell cycle","hematopoietic or lymphoid organ development","positive regulation of multicellular organismal process","protein localization to organelle","antigen processing and presentation of peptide antigen","regulation of cellular response to stress","proton-transporting ATP synthase complex","organelle membrane","cell killing","endocytic vesicle","antigen processing and presentation of endogenous antigen","endocytic vesicle membrane","viral process","proton channel activity","regulation of cellular protein localization","positive regulation of DNA replication","positive regulation of DNA biosynthetic process","regulation of organelle organization","phosphorylation","positive regulation of transcription, DNA-templated","positive regulation of protein metabolic process","anatomical structure morphogenesis","positive regulation of nucleic acid-templated transcription","positive regulation of RNA biosynthetic process","membrane raft","regulation of DNA metabolic process","regulation of transferase activity","membrane microdomain","ncRNA metabolic process","nuclear speck","mitochondrial respirasome","immune system process","ATP synthesis coupled proton transport","energy coupled proton transport, down electrochemical gradient","negative regulation of mRNA splicing, via spliceosome","regulation of cell cycle","regulation of immune system process","regulation of DNA biosynthetic process","positive regulation of cell differentiation","respiratory electron transport chain","response to hydrogen peroxide","cell projection","regulation of multicellular organismal development","regulation of deoxyribonuclease activity","TAP complex binding","positive regulation of DNA metabolic process","embryo development","immune system development","cellular component assembly","cell development","telomere maintenance","cell leading edge","positive regulation of T cell mediated cytotoxicity","electron transport chain","identical protein binding","regulation of response to stimulus","regulation of intracellular signal transduction","protein export from nucleus","positive regulation of organelle organization","negative regulation of developmental process","establishment of localization in cell","reproductive structure development","protein stabilization","ATP biosynthetic process","CTP binding","positive regulation of barbed-end actin filament capping","reproductive system development","regulation of T cell mediated cytotoxicity","reactive oxygen species metabolic process","gonad development","CD8 receptor binding","negative regulation of organelle organization","anatomical structure homeostasis","embryonic hemopoiesis","regulation of hemopoiesis","development of primary sexual characteristics","translation regulator activity, nucleic acid binding","positive regulation of cellular protein metabolic process","positive regulation of cell death","intracellular signal transduction","chromatin binding","beta-2-microglobulin binding","TAP binding","negative regulation of mRNA processing","poly-pyrimidine tract binding","positive regulation of chromosome organization","maintenance of protein location","cell cycle process","energy derivation by oxidation of organic compounds","cellular response to hormone stimulus","regulation of response to oxidative stress","antigen processing and presentation of endogenous peptide antigen via MHC class I via ER pathway, TAP-independent","regulation of immune effector process","regulation of response to stress","cell body","5S rRNA binding","structural constituent of cytoskeleton","positive regulation of cell motility","positive regulation of cell cycle process","antigen processing and presentation of endogenous peptide antigen via MHC class I via ER pathway","antigen processing and presentation of endogenous peptide antigen via MHC class Ib","sex differentiation","positive regulation of transferase activity","purine ribonucleoside triphosphate biosynthetic process","ruffle","tube development","positive regulation of transcription by RNA polymerase II","cellular response to reactive oxygen species","positive regulation of epithelial cell migration","interspecies interaction between organisms","response to reactive oxygen species","purine nucleoside triphosphate biosynthetic process","DNA binding","glutamate receptor binding","phosphate-containing compound metabolic process","signaling receptor binding","integral component of endoplasmic reticulum membrane","negative regulation of cellular component organization","MHC protein binding","epithelium development","positive regulation of locomotion","catalytic complex","maintenance of protein location in cell","ribonucleoside triphosphate biosynthetic process","negative regulation of multicellular organismal process","cellular response to endogenous stimulus","regulation of reactive oxygen species metabolic process","protein folding","positive regulation of catalytic activity","T cell mediated cytotoxicity","translation initiation factor binding","response to endogenous stimulus","positive regulation of cellular component movement","phosphorus metabolic process","cellular respiration","oxidoreductase complex","positive regulation of plasma membrane bounded cell projection assembly","antigen processing and presentation of endogenous peptide antigen via MHC class I via ER pathway, TAP-dependent","C-rich single-stranded DNA binding","miRNA transport","sulfonylurea receptor binding","ribonucleoprotein complex export from nucleus","cellular response to chemical stress","regulation of localization","cell aging","response to hormone","regulation of signal transduction by p53 class mediator","ribonucleoprotein complex localization","proton transmembrane transporter activity","nuclear periphery","intrinsic component of endoplasmic reticulum membrane","positive regulation of immune effector process","positive regulation of T cell mediated immunity","positive regulation of molecular function","cellular response to oxidative stress","viral genome replication","telomere maintenance via telomere lengthening","antigen processing and presentation","positive regulation of developmental process","protein localization","cellular response to epidermal growth factor stimulus","translation factor activity, RNA binding","mitochondrial ATP synthesis coupled proton transport","T cell receptor binding","chordate embryonic development","regulation of myeloid cell differentiation","DNA biosynthetic process","regulation of oxidative stress-induced cell death","protein kinase regulator activity","purine nucleotide binding","positive regulation of cell migration","nucleoside triphosphate biosynthetic process","nitrogen compound transport","purine ribonucleoside triphosphate metabolic process","purine ribonucleoside triphosphate binding","response to epidermal growth factor","embryo development ending in birth or egg hatching","angiostatin binding","cAMP-dependent protein kinase activity","dATP binding","UTP binding","carbohydrate derivative binding","epithelial cell migration","regulation of cell killing","whole membrane","response to stress","positive regulation of leukocyte mediated cytotoxicity","nucleoside phosphate binding","nucleotide binding","epithelium migration","peptide antigen binding","negative regulation of protein polymerization","ribonucleoside triphosphate metabolic process","positive regulation of translation","tissue migration","immune effector process","unfolded protein binding","integral component of organelle membrane","regulation of chromosome organization","regulation of protein modification by small protein conjugation or removal","negative regulation of cell cycle arrest","seminiferous tubule development","regulation of myeloid leukocyte differentiation","regulation of molecular function","positive regulation of apoptotic signaling pathway","DNA metabolic process","RNA export from nucleus","positive regulation of signal transduction","positive regulation of leukocyte mediated immunity","proton-transporting two-sector ATPase complex","striated muscle tissue development","nuclear matrix","purine nucleoside triphosphate metabolic process","signal transduction by p53 class mediator","regulation of erythrocyte differentiation","cellular response to hydrogen peroxide","response to interferon-gamma","regulation of kinase activity","aging","mitochondrial respiratory chain complex I","NADH dehydrogenase complex","respiratory chain complex I","adenyl deoxyribonucleotide binding","CRD-mediated mRNA stabilization","mitochondrial proton-transporting ATP synthase complex, catalytic sector F(1)","mRNA CDS binding","positive regulation of deoxyribonuclease activity","pyrimidine ribonucleotide binding","Rho GDP-dissociation inhibitor binding","structural constituent of postsynaptic actin cytoskeleton","WD40-repeat domain binding","regulation of sister chromatid cohesion","purine ribonucleotide binding","regulation of cell development","regulation of cellular response to oxidative stress","positive regulation of cellular protein localization","peptide binding","purine ribonucleotide biosynthetic process","regulation of binding","Golgi medial cisterna","negative regulation of response to stimulus","ribonucleotide binding","positive regulation of immune system process","cytokine production","protein-containing complex localization","sequence-specific DNA binding","positive regulation of protein modification process","regulation of protein modification process","positive regulation of cell killing","establishment of protein localization","muscle tissue development","negative regulation of translation","regulation of viral process","muscle cell differentiation","kinase regulator activity","regulation of signal transduction","regulation of T cell mediated immunity","negative regulation of response to oxidative stress","regulation of plasma membrane bounded cell projection organization","cell death in response to oxidative stress","intrinsic component of organelle membrane","positive regulation of cell projection organization","adenylate cyclase-activating adrenergic receptor signaling pathway","dosage compensation","positive regulation of signal transduction by p53 class mediator","CRD-mediated mRNA stability complex","cyclic nucleotide-dependent protein kinase activity","dendritic transport of messenger ribonucleoprotein complex","dendritic transport of ribonucleoprotein complex","ncRNA export from nucleus","positive regulation of nuclease activity","proton-transporting ATP synthase complex, catalytic core F(1)","purine deoxyribonucleotide binding","regulation of forebrain neuron differentiation","RS domain binding","selenocysteine insertion sequence binding","telomerase holoenzyme complex assembly","tRNA transport","mitochondrial ATP synthesis coupled electron transport","positive regulation of blood vessel endothelial cell migration","regulation of protein ubiquitination","ribonucleotide biosynthetic process","negative regulation of cell differentiation","regulation of cell projection organization","regulation of cellular component biogenesis","modification of postsynaptic structure","NADH dehydrogenase (quinone) activity","NADH dehydrogenase (ubiquinone) activity","positive regulation of lamellipodium assembly","regulation of oxidative stress-induced intrinsic apoptotic signaling pathway","positive regulation of apoptotic process","monovalent inorganic cation transport","organelle assembly","positive regulation of endothelial cell migration","RNA-dependent DNA biosynthetic process","telomere maintenance via telomerase","positive regulation of programmed cell death","positive regulation of response to oxidative stress","regulation of nuclease activity","regulation of synapse organization","peptide transport","regulation of leukocyte mediated cytotoxicity","ribose phosphate biosynthetic process","positive regulation of cellular amide metabolic process","regulation of catalytic activity","anatomical structure formation involved in morphogenesis","adenylate cyclase-activating adrenergic receptor signaling pathway involved in heart process","positive regulation of oxidative stress-induced intrinsic apoptotic signaling pathway","protein localization to cytoplasmic stress granule","regulation of barbed-end actin filament capping","regulation of endodeoxyribonuclease activity","Golgi cisterna","viral gene expression","adrenergic receptor signaling pathway","NADH dehydrogenase activity","poly(U) RNA binding","regulation of symbiotic process","nucleoside-triphosphatase activity","purine nucleotide biosynthetic process","RNA polymerase II-specific DNA-binding transcription factor binding","establishment of protein localization to organelle","myeloid leukocyte differentiation","positive regulation of myeloid cell differentiation","response to inorganic substance","ameboidal-type cell migration","viral life cycle","cellular response to interleukin-4","protein localization to chromosome, telomeric region","nucleoside triphosphate metabolic process","movement of cell or subcellular component","regulation of protein stability","leukocyte differentiation","amide transport","macromolecule localization","regulation of gene expression, epigenetic","regulation of DNA replication","regulation of cell cycle process","regulation of synapse structure or activity","negative regulation of cellular amide metabolic process","positive regulation of hemopoiesis","cellular senescence","regulation of immune response","cell migration","positive regulation of kinase activity","miRNA binding","rRNA transcription","protein import into nucleus","purine-containing compound biosynthetic process","regulation of muscle cell differentiation","positive regulation of cell adhesion","tube morphogenesis","regulation of protein polymerization","epithelial cell proliferation","regulation of phagocytosis","Schaffer collateral - CA1 synapse","supramolecular complex","deoxyribonucleotide binding","GDP-dissociation inhibitor binding","large ribosomal subunit rRNA binding","nitric-oxide synthase regulator activity","phosphoserine residue binding","regulation of maintenance of mitotic sister chromatid cohesion","ribosomal small subunit export from nucleus","mitotic cell cycle process","negative regulation of nervous system development","protein transport","cytochrome complex","positive regulation of cyclin-dependent protein kinase activity","response to interleukin-4","positive regulation of protein kinase activity","response to nitrogen compound","cellular response to interferon-gamma","positive regulation of reactive oxygen species metabolic process","import into nucleus","embryo implantation","lamellipodium assembly","negative regulation of actin filament polymerization","positive regulation of myeloid leukocyte differentiation","negative regulation of apoptotic signaling pathway","regulation of protein binding","female gonad development","regulation of intracellular protein transport","endoplasmic reticulum exit site","regulation of histone H3-K4 methylation","regulation of necrotic cell death","regulation of smooth muscle cell differentiation","ruffle membrane","regulation of epithelial cell proliferation","cellular response to nitrogen compound","chromosome organization","regulation of epithelial cell migration","cytoplasmic stress granule","ovarian follicle development","development of primary female sexual characteristics","regulation of cell communication","negative regulation of cell cycle","positive regulation of chromosome segregation","positive regulation of lamellipodium organization","positive regulation of phosphate metabolic process","positive regulation of phosphorus metabolic process","blood vessel endothelial cell migration","cellular response to growth factor stimulus","regulation of phosphate metabolic process","regulation of phosphorus metabolic process","regulation of maintenance of sister chromatid cohesion","Tat protein binding","positive regulation of cell communication","regulation of tumor necrosis factor production","regulation of signaling","modification of synaptic structure","positive regulation of signaling","regulation of leukocyte differentiation","developmental growth","immune response","positive regulation of endocytosis","regulation of tumor necrosis factor superfamily cytokine production","regulation of neuron differentiation","DNA conformation change","regulation of cellular localization","regulation of cytokine production","positive regulation of leukocyte differentiation","regulation of neurogenesis","positive regulation of telomerase activity","regulation of lamellipodium assembly","ribosomal large subunit biogenesis","negative regulation of immune system process","response to growth factor","pyrophosphatase activity","cell motility","localization of cell","neuron development","hydrolase activity, acting on acid anhydrides","hydrolase activity, acting on acid anhydrides, in phosphorus-containing anhydrides","reactive oxygen species biosynthetic process","anion binding","cytoskeleton organization","molecular function regulator","positive regulation of phosphorylation","adenyl nucleotide binding","regulation of cell motility","cellular response to thyroid hormone stimulus","IRES-dependent viral translational initiation","myeloid progenitor cell differentiation","positive regulation of DNA-directed DNA polymerase activity","regulation of DNA-directed DNA polymerase activity","regulation of translational fidelity","small ribosomal subunit rRNA binding","X chromosome","tumor necrosis factor production","regulation of histone methylation","amide binding","neuronal cell body","negative regulation of protein metabolic process","genetic imprinting","tumor necrosis factor superfamily cytokine production","regulation of protein localization","regulation of phosphorylation","enzyme regulator activity","regulation of cellular protein catabolic process","A band","oxidoreductase activity, acting on NAD(P)H, quinone or similar compound as acceptor","positive regulation of protein phosphorylation","regulation of plasma membrane bounded cell projection assembly","synapse organization","secretory granule","cellular response to stress","ATP binding","regulation of cyclin-dependent protein kinase activity","G protein-coupled receptor signaling pathway involved in heart process","mitochondrial respiratory chain complex III","positive regulation of sister chromatid cohesion","pyrimidine nucleotide binding","respiratory chain complex III","T cell mediated immunity","regulation of transport","regulation of nervous system development","regulation of cell projection assembly","ionotropic glutamate receptor binding","sex chromosome","regulation of telomere maintenance","regulation of sister chromatid segregation","regulation of leukocyte mediated immunity","region of cytosol","cAMP-mediated signaling","mitotic cell cycle","oxidoreductase activity, acting on NAD(P)H","multicellular organismal process","neuron projection development","snRNA binding","positive regulation of phagocytosis","dendritic growth cone","dendritic transport","maintenance of mitotic sister chromatid cohesion","protein tag","ribosomal large subunit binding","sequestering of actin monomers","DNA packaging","generation of neurons","double-stranded RNA binding","positive regulation of viral process","neuron death","mitochondrial transport","postsynapse organization","regulatory region nucleic acid binding","regulation of cell migration","male gonad development","regulation of cellular senescence","neurogenesis","regulation of locomotion","lamellipodium organization","adenylate cyclase-activating G protein-coupled receptor signaling pathway","development of primary male sexual characteristics","cation transport","negative regulation of cellular protein metabolic process","positive regulation of immune response","muscle organ development","cellular response to stimulus","leukocyte mediated cytotoxicity","Golgi stack","cyclin-dependent protein kinase holoenzyme complex","regulatory RNA binding","positive regulation of transport","regulation of protein kinase activity","nucleotide biosynthetic process","cell division","maintenance of sister chromatid cohesion","neuron differentiation","intrinsic apoptotic signaling pathway in response to oxidative stress","interaction with symbiont","positive regulation of lymphocyte mediated immunity","glutamatergic synapse","organic substance transport","regulation of lamellipodium organization","nucleoside phosphate biosynthetic process","response to organic cyclic compound","negative regulation of transport","gene silencing","regulation of chromatin organization","negative regulation of protein-containing complex assembly","positive regulation of adaptive immune response based on somatic recombination of immune receptors built from immunoglobulin superfamily domains","T cell activation","dendritic cell cytokine production","messenger ribonucleoprotein complex","positive regulation of smooth muscle cell differentiation","postsynaptic actin cytoskeleton","regulation of vascular associated smooth muscle cell differentiation","response to thyroid hormone","sequence-specific single stranded DNA binding","adenyl ribonucleotide binding","GTP binding","extracellular exosome","vesicle","endothelial cell migration","T cell differentiation","vasculature development","female sex differentiation","maintenance of location in cell","single-stranded RNA binding","mitochondrial nucleoid","nucleoid","leading edge membrane","regulation of protein-containing complex assembly","regulation of blood vessel endothelial cell migration","purine ribonucleoside binding","small molecule binding","central nervous system neuron development","negative regulation of intrinsic apoptotic signaling pathway","regulation of DNA-dependent DNA replication","regulation of telomerase activity","regulation of protein serine/threonine kinase activity","mRNA catabolic process","protein import","regulation of vesicle-mediated transport","protein localization to cell-cell junction","proton-transporting two-sector ATPase complex, catalytic domain","regulation of protein localization to chromosome, telomeric region","regulation of transcription of nucleolar large rRNA by RNA polymerase I","viral translation","circulatory system development","double-stranded DNA binding","negative regulation of myeloid cell differentiation","organelle envelope lumen","regulation of cellular component movement","DNA-binding transcription factor binding","ribonucleoside binding","actin filament-based process","adenylate cyclase-modulating G protein-coupled receptor signaling pathway","positive regulation of adaptive immune response","purine nucleoside binding","maturation of SSU-rRNA","ncRNA transcription","positive regulation of RNA splicing","production of miRNAs involved in gene silencing by miRNA","regulation of cell aging","positive regulation of protein serine/threonine kinase activity","actin cytoskeleton organization","regulation of viral genome replication","negative regulation of cell cycle process","regulation of proteolysis involved in cellular protein catabolic process","protein polymerization","cellular response to organic cyclic compound","regulation of endocytosis","core promoter sequence-specific DNA binding","regulation of supramolecular fiber organization","cellular response to lipid","negative regulation of hemopoiesis","cyclic-nucleotide-mediated signaling","maturation of LSU-rRNA from tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)","modification of postsynaptic actin cytoskeleton","regulation of monocyte differentiation","ribosomal small subunit binding","ribosomal subunit export from nucleus","ribosome localization","angiogenesis","dsRNA processing","positive regulation of telomere maintenance","production of small RNA involved in gene silencing by RNA","positive regulation of cytoskeleton organization","extracellular vesicle","RNA polymerase II transcription regulator complex","sequence-specific double-stranded DNA binding","neuron apoptotic process","nucleoside binding","cellular response to peptide","inner ear auditory receptor cell differentiation","negative regulation of oxidative stress-induced cell death","regulation of neuron death","positive regulation of cell cycle phase transition","growth","transcription regulator complex","guanyl nucleotide binding","guanyl ribonucleotide binding","axo-dendritic protein transport","hydrogen peroxide biosynthetic process","positive regulation of brown fat cell differentiation","positive regulation of cell size","positive regulation of histone H3-K4 methylation","response to hydroperoxide","rRNA-containing ribonucleoprotein complex export from nucleus","regulation of cell adhesion","negative regulation of cellular response to oxidative stress","regulation of chromosome segregation","ligase activity","regulation of histone modification","regulation of peptidyl-serine phosphorylation","transport along microtubule","translational initiation","transmembrane transport","hydrogen peroxide metabolic process","negative regulation of mitochondrion organization","protein kinase inhibitor activity","regulation of vascular associated smooth muscle cell proliferation","vascular associated smooth muscle cell proliferation","distal axon","protein C-terminus binding","transcription regulatory region sequence-specific DNA binding","myeloid leukocyte mediated immunity","protein tyrosine kinase binding","cell cortex","actin filament branch point","actin rod","cellular response to cytochalasin B","cellular response to tumor cell","cofilin-actin rod","cyclin D2-CDK6 complex","distal stomach smooth muscle contraction","DNA polymerase processivity factor complex","epithelial to mesenchymal transition involved in cardiac fibroblast development","ErbB-4 class receptor binding","establishment of cell polarity regulating cell shape","establishment of integrated proviral latency","granzyme B localization to T cell secretory granule","H3K9me3 modified histone binding","inactive sex chromosome","invadopodium assembly","LEM domain binding","lumenal side of Golgi membrane","maintenance of granzyme B location in T cell secretory granule","maintenance of protease location in mast cell secretory granule","maintenance of protease location in T cell secretory granule","maintenance of protein location in mast cell secretory granule","maintenance of protein location in T cell secretory granule","melanocyte adhesion","MH2 domain binding","negative regulation of myeloid progenitor cell differentiation","negative regulation of postsynaptic density organization","negative regulation of telomeric RNA transcription from RNA pol II promoter","negative regulation of transforming growth factor beta activation","negative regulation of unidimensional cell growth","nuclear histone mRNA catabolic process","nuclear subtelomeric heterochromatin","ooplasm","oxidized pyrimidine DNA binding","PCNA complex","phosphoribosylaminoimidazole carboxylase activity","phosphoribosylaminoimidazolesuccinocarboxamide synthase activity","podosome ring","poly(C) RNA binding","positive regulation of cardioblast proliferation","positive regulation of DNA N-glycosylase activity","positive regulation of endodermal cell differentiation","positive regulation of establishment of cell polarity regulating cell shape","positive regulation of invadopodium assembly","positive regulation of protein localization to cell leading edge","positive regulation of pyloric antrum smooth muscle contraction","positive regulation of telomeric RNA transcription from RNA pol II promoter","post-embryonic body morphogenesis","post-embryonic forelimb morphogenesis","presynaptic actin cytoskeleton","protease localization to mast cell secretory granule","protease localization to T cell secretory granule","protein localization to mast cell secretory granule","protein localization to spindle microtubule","protein localization to T cell secretory granule","protein N-linked glycosylation via arginine","pyloric antrum smooth muscle contraction","regulation of D-erythro-sphingosine kinase activity","regulation of eIF2 alpha phosphorylation by dsRNA","regulation of establishment of cell polarity regulating cell shape","regulation of invadopodium assembly","regulation of mRNA stability involved in cellular response to UV","regulation of proton-transporting ATPase activity, rotational mechanism","regulation of pyloric antrum smooth muscle contraction","regulation of telomeric RNA transcription from RNA pol II promoter","regulation of unidimensional cell growth","response to 5-fluorouracil","response to cytochalasin B","response to TNF agonist","RNA localization to chromatin","stem cell factor receptor activity","stomach pylorus smooth muscle contraction","T cell secretory granule organization","telomeric repeat-containing RNA transcription","telomeric repeat-containing RNA transcription by RNA polymerase II","U11/U12 snRNP","ventral surface of cell","regulation of endothelial cell migration","regulation of protein kinase B signaling","chaperone-mediated protein complex assembly","Fc receptor signaling pathway","nucleolar large rRNA transcription by RNA polymerase I","nucleosomal DNA binding","post-embryonic animal morphogenesis","protein refolding","protein kinase A binding","regulation of endothelial cell apoptotic process","negative regulation of binding","regulation of cytoskeleton organization","cytokine production involved in immune response","regulation of actomyosin structure organization","regulation of fibroblast proliferation","regulation of reactive oxygen species biosynthetic process","cation transmembrane transport","histone H3-K4 methylation","megakaryocyte differentiation","necrotic cell death","sister chromatid cohesion","blood vessel morphogenesis","cell junction organization","regulation of cellular catabolic process","negative regulation of protein binding","epidermis development","cellular component morphogenesis","embryonic morphogenesis","negative regulation of mRNA catabolic process","protein phosphorylated amino acid binding","fibroblast proliferation","positive regulation of dendrite development","positive regulation of protein kinase B signaling","negative regulation of release of cytochrome c from mitochondria","postsynaptic cytoskeleton","male sex differentiation","negative regulation of cytoskeleton organization","positive regulation of mitotic cell cycle","regulation of protein catabolic process","locomotion","negative regulation of phosphate metabolic process","negative regulation of phosphorus metabolic process","extracellular organelle","regulation of actin filament bundle assembly","negative regulation of molecular function","chromatin","endothelial cell apoptotic process","hair cell differentiation","pri-miRNA transcription by RNA polymerase II","somatic stem cell population maintenance","inorganic cation transmembrane transport","apoptotic mitochondrial changes","nervous system development","response to peptide","maintenance of location","kinase inhibitor activity","regulation of ATPase activity","regulation of cell cycle arrest","regulation of telomere maintenance via telomere lengthening","release of cytochrome c from mitochondria","transcription cofactor binding","response to abiotic stimulus","dendrite terminus","negative regulation of cellular senescence","proline-rich region binding","regulation of gene expression by genetic imprinting","regulation of macrophage differentiation","respiratory chain complex IV","structural constituent of postsynapse","regulation of establishment of protein localization","G protein-coupled receptor signaling pathway, coupled to cyclic nucleotide second messenger","positive regulation of receptor-mediated endocytosis","oxidation-reduction process","chromosome segregation","purine ribonucleotide metabolic process","nuclear receptor binding","response to stimulus","regulation of receptor-mediated endocytosis","angiogenesis involved in wound healing","dendritic cell migration","positive regulation of mitotic sister chromatid segregation","positive regulation of oxidative stress-induced cell death","positive regulation of viral transcription","regulation of anatomical structure size","negative regulation of neuron death","histone lysine methylation","reproductive process","reproduction","DNA damage response, signal transduction by p53 class mediator","killing of cells of other organism","ribosome binding","regulation of intracellular transport","cellular macromolecule catabolic process","hormone receptor binding","regulation of dendrite morphogenesis","cellular response to peptide hormone stimulus","regulation of peptide transport","dendrite morphogenesis","negative regulation of neuron apoptotic process","RNA catabolic process","regulation of cellular component size","presynapse","positive regulation of dendritic spine development","positive regulation of tumor necrosis factor production","plasma membrane bounded cell projection organization","cytochrome-c oxidase activity","heme-copper terminal oxidase activity","mitochondrial electron transport, NADH to ubiquinone","nitric-oxide synthase binding","oxidoreductase activity, acting on a heme group of donors, oxygen as acceptor","positive regulation of protein localization to cell surface","tau protein binding","chromatin organization","cell cycle phase transition","ribonucleotide metabolic process","in utero embryonic development","positive regulation of tumor necrosis factor superfamily cytokine production","negative regulation of protein modification process","protein modification by small protein conjugation","cardiac muscle tissue development","response to peptide hormone","negative regulation of neurogenesis","dosage compensation by inactivation of X chromosome","myoblast proliferation","oxidoreductase activity, acting on a heme group of donors","pore complex","positive regulation of superoxide anion generation","sperm plasma membrane","positive regulation of cell morphogenesis involved in differentiation","cytoskeletal protein binding","regulation of proteasomal protein catabolic process","cortical cytoskeleton","histone deacetylase binding","regulation of catabolic process","biological adhesion","developmental process involved in reproduction","embryonic organ development","nucleus organization","regulation of nuclear division","chaperone complex","establishment of protein localization to chromosome","G-protein beta/gamma-subunit complex binding","maintenance of protein location in nucleus","negative regulation of nucleocytoplasmic transport","negative regulation of phagocytosis","pericentric heterochromatin","positive regulation of cellular response to oxidative stress","R-SMAD binding","regulation of brown fat cell differentiation","RNA polymerase II core promoter sequence-specific DNA binding","neuron projection morphogenesis","ribose phosphate metabolic process","cis-regulatory region sequence-specific DNA binding","cytoplasmic vesicle","electron transfer activity","negative regulation of RNA catabolic process","epidermal cell differentiation","regulation of nucleocytoplasmic transport","cytoskeleton-dependent intracellular transport","intracellular vesicle","positive regulation of fibroblast proliferation","protein localization to cell surface","regulation of mitotic sister chromatid segregation","dendritic spine development","secretory vesicle","cytoskeleton","dendritic spine","microtubule-based transport","cytoplasmic translational initiation","mRNA 5'-UTR binding","negative regulation of hydrogen peroxide-induced cell death","negative regulation of response to reactive oxygen species","regulation of renal sodium excretion","regulation of neuron apoptotic process","actin binding","positive regulation of cell-substrate adhesion","regulation of mRNA catabolic process","protein kinase B signaling","regulation of lymphocyte mediated immunity","skeletal muscle tissue development","carbohydrate derivative biosynthetic process","leukocyte mediated immunity","regulation of cell morphogenesis involved in differentiation","protein targeting","DNA alkylation","DNA methylation","promoter-specific chromatin binding","adenylate cyclase activator activity","cardiac fibroblast cell development","cardiac fibroblast cell differentiation","cellular response to erythropoietin","cellular response to iron(III) ion","chromosome, subtelomeric region","colon smooth muscle contraction","COPI-coated Golgi to ER transport vesicle","endocytic vesicle lumen","eosinophil fate commitment","epicardium-derived cardiac fibroblast cell development","epicardium-derived cardiac fibroblast cell differentiation","erythropoietin-mediated signaling pathway","establishment of viral latency","FBXO family protein binding","HSP90-CDC37 chaperone complex","IRE1-RACK1-PP2A complex","leading edge cell differentiation","maintenance of mitotic sister chromatid cohesion, telomeric","mast cell secretory granule organization","melanocyte migration","methionine adenosyltransferase activity","methionine adenosyltransferase complex","MHC class I protein complex assembly","mitotic sister chromatid cohesion, telomeric","negative regulation of chromosome condensation","negative regulation of Fc-gamma receptor signaling pathway involved in phagocytosis","negative regulation of maintenance of mitotic sister chromatid cohesion, telomeric","negative regulation of oxidative phosphorylation uncoupler activity","negative regulation of protein localization to chromosome, telomeric region","negative regulation of protein tyrosine phosphatase activity","NF-kappaB complex","PCNA-p21 complex","peptide antigen assembly with MHC class I protein complex","phospholipid-hydroperoxide glutathione peroxidase activity","positive regulation of base-excision repair","positive regulation of colon smooth muscle contraction","positive regulation of cyclic-nucleotide phosphodiesterase activity","positive regulation of cysteine-type endopeptidase activity involved in execution phase of apoptosis","positive regulation of dipeptide transmembrane transport","positive regulation of dipeptide transport","positive regulation of establishment or maintenance of cell polarity regulating cell shape","positive regulation of ferrous iron binding","positive regulation of histone H3-K4 trimethylation","positive regulation of low-density lipoprotein particle receptor biosynthetic process","positive regulation of oligopeptide transport","positive regulation of respiratory burst involved in inflammatory response","positive regulation of telomerase RNA reverse transcriptase activity","positive regulation of transferrin receptor binding","post-embryonic appendage morphogenesis","post-embryonic limb morphogenesis","purine-specific mismatch base pair DNA N-glycosylase activity","regulation of base-excision repair","regulation of colon smooth muscle contraction","regulation of dipeptide transmembrane transport","regulation of dipeptide transport","regulation of DNA N-glycosylase activity","regulation of establishment or maintenance of cell polarity regulating cell shape","regulation of ferrous iron binding","regulation of histone H3-K4 trimethylation","regulation of maintenance of mitotic sister chromatid cohesion, telomeric","regulation of myeloid progenitor cell differentiation","regulation of norepinephrine uptake","regulation of oligopeptide transport","regulation of protein localization to cell leading edge","regulation of transferrin receptor binding","regulation of translation at presynapse","regulation of translation at presynapse, modulating synaptic transmission","response to erythropoietin","response to iron(III) ion","rRNA export from nucleus","sperm mitochondrial sheath","stomach smooth muscle contraction","TFIIH-class transcription factor complex binding","cell projection organization","antimicrobial humoral immune response mediated by antimicrobial peptide","positive regulation of peptidyl-serine phosphorylation","positive regulation of protein polymerization","protein N-terminus binding","regulation of adaptive immune response based on somatic recombination of immune receptors built from immunoglobulin superfamily domains","regulation of dendrite development","oxidoreductase activity","regulation of actin filament organization","p53 binding","purine nucleotide metabolic process","cell morphogenesis","plasma membrane bounded cell projection morphogenesis","epithelial cell differentiation","adrenal gland development","fibroblast growth factor binding","mitotic sister chromatid cohesion","negative regulation of cell aging","postsynaptic cytosol","protein folding chaperone","regulation of histone H3-K9 methylation","regulation of superoxide anion generation","U12-type spliceosomal complex","inorganic ion transmembrane transport","peptidyl-lysine methylation","transcription coregulator activity","neuron spine","positive regulation of supramolecular fiber organization","regulation of anatomical structure morphogenesis","regulation of postsynapse organization","cell projection morphogenesis","regulation of neuron projection development","cell morphogenesis involved in differentiation","heart morphogenesis","cellular response to organonitrogen compound","dendritic spine morphogenesis","forebrain neuron differentiation","receptor tyrosine kinase binding","response to biotic stimulus","cell surface","RNA polymerase II transcription regulatory region sequence-specific DNA binding","actin monomer binding","antigen processing and presentation of exogenous peptide antigen","podosome","renal sodium excretion","regulation of actin cytoskeleton organization","acrosomal vesicle","cell cycle G2/M phase transition","negative regulation of immune effector process","aerobic respiration","regulation of cell morphogenesis","cellular response to ketone","positive regulation of cytokine production","regulation of protein phosphorylation","regulation of protein import into nucleus","smooth muscle cell differentiation","skeletal muscle organ development","actin filament","regulation of protein transport","barbed-end actin filament capping","monocyte differentiation","mononuclear cell differentiation","poly(A) binding","positive regulation of cyclin-dependent protein serine/threonine kinase activity","regulation of hematopoietic progenitor cell differentiation","supramolecular fiber organization","cell cycle arrest","inclusion body","regulation of cyclin-dependent protein serine/threonine kinase activity","transcription repressor complex","protein ubiquitination","growth cone","positive regulation of NIK/NF-kappaB signaling","positive regulation of protein modification by small protein conjugation or removal","regulation of protein localization to nucleus","calyx of Held","mRNA cis splicing, via spliceosome","viral RNA genome replication","intrinsic apoptotic signaling pathway by p53 class mediator","neuroepithelial cell differentiation","positive regulation of cellular component biogenesis","response to lipid","ion transmembrane transport","response to alcohol","cell part morphogenesis","cell activation","regulation of adaptive immune response","regulation of cell size","maturation of LSU-rRNA","myeloid leukocyte cytokine production","positive regulation of cell cycle G2/M phase transition","regulation of vascular endothelial growth factor receptor signaling pathway","sperm head","structural constituent of synapse","cell morphogenesis involved in neuron differentiation","endocytosis","protein-DNA complex subunit organization","regulation of cell-substrate adhesion","defense response to other organism","inner ear development","axo-dendritic transport","blood vessel development","cellular protein modification process","protein modification process","site of polarized growth","ion binding","histone methylation","positive regulation of intracellular protein transport","regulation of mitochondrial membrane potential","regulation of protein import","acetylcholine receptor signaling pathway","androgen receptor binding","cyclin binding","regulation of excretion","regulation of intrinsic apoptotic signaling pathway by p53 class mediator","synapse maturation","GTPase activity","cellular_component","macromolecule methylation","5.8S rRNA binding","antigen processing and presentation of exogenous protein antigen via MHC class Ib, TAP-dependent","aryl hydrocarbon receptor complex","cellular response to actinomycin D","cellular response to rapamycin","cleavage in ITS2 between 5.8S rRNA and LSU-rRNA of tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)","concave side of sperm head","CXCR3 chemokine receptor binding","cyclin D2-CDK4 complex","dinucleotide insertion or deletion binding","DNA polymerase processivity factor activity","eukaryotic translation elongation factor 1 complex","glial limiting end-foot","inactivation of X chromosome by genetic imprinting","invadopodium organization","macrophage migration inhibitory factor signaling pathway","MHC class Ib protein complex","mitotic spindle microtubule","mitotic telomere maintenance via semi-conservative replication","muscle cell projection","muscle cell projection membrane","nascent polypeptide-associated complex","negative regulation of adenylate cyclase-activating adrenergic receptor signaling pathway involved in heart process","negative regulation of histone phosphorylation","negative regulation of lamellipodium assembly","negative regulation of maintenance of mitotic sister chromatid cohesion","negative regulation of maintenance of sister chromatid cohesion","negative regulation of myofibroblast differentiation","peptide antigen assembly with MHC protein complex","peripheral region of growth cone","phosphothreonine residue binding","podosome core","positive regulation of attachment of mitotic spindle microtubules to kinetochore","positive regulation of cytotoxic T cell differentiation","positive regulation of DNA topoisomerase (ATP-hydrolyzing) activity","positive regulation of peptidyl-cysteine S-nitrosylation","positive regulation of small intestine smooth muscle contraction","positive regulation of translational termination","regulation of attachment of mitotic spindle microtubules to kinetochore","regulation of cytotoxic T cell differentiation","regulation of DNA topoisomerase (ATP-hydrolyzing) activity","regulation of endoribonuclease activity","regulation of glucocorticoid mediated signaling pathway","regulation of macrophage migration inhibitory factor signaling pathway","regulation of mitochondrial membrane permeability involved in programmed necrotic cell death","regulation of parathyroid hormone secretion","regulation of primitive erythrocyte differentiation","regulation of telomerase RNA reverse transcriptase activity","response to symbiont","response to symbiotic bacterium","rRNA primary transcript binding","splicing factor binding","subtelomeric heterochromatin assembly","subtelomeric heterochromatin organization","supercoiled DNA binding","translational frameshifting","tRNA export from nucleus","tRNA-containing ribonucleoprotein complex export from nucleus","tubulin complex","establishment of organelle localization","inner ear receptor cell differentiation","negative regulation of phosphorylation","endocrine system development","negative regulation of chromosome organization","cellular response to interleukin-6","negative regulation of endothelial cell apoptotic process","positive regulation of filopodium assembly","positive regulation of mRNA splicing, via spliceosome","replication fork processing","forebrain generation of neurons","membrane depolarization","mitochondrial intermembrane space","mitochondrial transmembrane transport","serine/threonine protein kinase complex","rhythmic process","cell adhesion","morphogenesis of an epithelium","negative regulation of epithelial cell proliferation","regulation of response to DNA damage stimulus","negative regulation of cell development","antioxidant activity","RNA polymerase II cis-regulatory region sequence-specific DNA binding","defense response","establishment or maintenance of cell polarity","perinuclear region of cytoplasm","cellular response to acetylcholine","ion channel inhibitor activity","necroptotic process","positive regulation of osteoclast differentiation","regulation of hydrogen peroxide-induced cell death","regulation of megakaryocyte differentiation","response to acetylcholine","columnar/cuboidal epithelial cell differentiation","helicase activity","activating transcription factor binding","regulation of cytokine production involved in immune response","chromatin assembly","dendrite development","tissue morphogenesis","ion transport","response to other organism","nuclear hormone receptor binding","response to external biotic stimulus","ossification","antigen processing and presentation of exogenous antigen","COP9 signalosome","disordered domain specific binding","mitochondrial outer membrane permeabilization involved in programmed cell death","regulation of transcription by RNA polymerase I","response to interleukin-6","telomeric DNA binding","DNA methylation or demethylation","regulation of protein dephosphorylation","negative regulation of defense response","purine-containing compound metabolic process","positive regulation of cell development","cellular response to leukemia inhibitory factor","positive regulation of mitotic cell cycle phase transition","positive regulation of protein localization to nucleus","protein localization to chromosome","steroid hormone receptor binding","macromolecule modification","centrosome","response to leukemia inhibitory factor","channel inhibitor activity","GTPase complex","heterotrimeric G-protein complex","hyperosmotic response","maturation of SSU-rRNA from tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)","negative regulation of protein dephosphorylation","positive regulation of protein-containing complex disassembly","positive regulation of Rho protein signal transduction","regulation of DNA damage response, signal transduction by p53 class mediator","response to cadmium ion","RNA helicase activity","heart process","innate immune response","cortical actin cytoskeleton","mechanoreceptor differentiation","positive regulation of muscle cell differentiation","signal transduction in response to DNA damage","heparin binding","negative regulation of inflammatory response","response to radiation","cell-substrate junction assembly","histone H3-K9 methylation","negative regulation of actin filament bundle assembly","nuclear DNA replication","poly-purine tract binding","positive regulation of telomere maintenance via telomere lengthening","positive regulation of vascular associated smooth muscle cell proliferation","vascular associated smooth muscle cell differentiation","monovalent inorganic cation homeostasis","ATPase activity","negative regulation of signal transduction","central nervous system neuron differentiation","positive regulation of protein-containing complex assembly","'de novo' IMP biosynthetic process","actin filament branching","actin filament fragmentation","antigen processing and presentation of exogenous peptide antigen via MHC class Ib","cellular response to interleukin-12","cellular response to potassium ion starvation","cellular response to stem cell factor stimulus","corticotropin-releasing hormone receptor 1 binding","cytotoxic T cell differentiation","DNA topoisomerase binding","establishment or maintenance of cell polarity regulating cell shape","extrinsic component of Golgi membrane","glucocorticoid mediated signaling pathway","glucose catabolic process to lactate","glucose catabolic process to lactate via pyruvate","glycolytic fermentation","hypothalamus gonadotrophin-releasing hormone neuron development","hypothalamus gonadotrophin-releasing hormone neuron differentiation","interchromatin granule","Kit signaling pathway","L-lactate dehydrogenase activity","lactate biosynthetic process from pyruvate","leading strand elongation","MHC protein complex assembly","mismatch base pair DNA N-glycosylase activity","mitochondrial permeability transition pore complex","mu-type opioid receptor binding","myofibroblast differentiation","negative regulation of adenylate cyclase-activating adrenergic receptor signaling pathway","negative regulation of branching morphogenesis of a nerve","negative regulation of cell adhesion involved in substrate-bound cell migration","negative regulation of complement-dependent cytotoxicity","negative regulation of forebrain neuron differentiation","negative regulation of intrinsic apoptotic signaling pathway in response to osmotic stress","negative regulation of nuclear-transcribed mRNA catabolic process, nonsense-mediated decay","negative regulation of transcription of nucleolar large rRNA by RNA polymerase I","positive regulation of endodeoxyribonuclease activity","positive regulation of Golgi to plasma membrane protein transport","positive regulation of synapse structural plasticity","positive regulation of translational elongation","positive regulation of translational fidelity","protein localization to secretory granule","protein-disulfide reductase activity","regulation of adenylate cyclase-activating adrenergic receptor signaling pathway involved in heart process","regulation of centromere complex assembly","regulation of myofibroblast differentiation","regulation of nuclear-transcribed mRNA catabolic process, nonsense-mediated decay","regulation of protein tyrosine phosphatase activity","response to actinomycin D","response to rapamycin","response to stem cell factor","S-adenosylmethionine biosynthetic process","storage vacuole","T cell proliferation involved in immune response","telomere maintenance via semi-conservative replication","thrombospondin receptor activity","transcription factor AP-1 complex","type 3 metabotropic glutamate receptor binding","unidimensional cell growth","viral latency","protein phosphorylation","cellular response to steroid hormone stimulus","regulation of ubiquitin-dependent protein catabolic process","regulation of viral life cycle","response to temperature stimulus","muscle cell proliferation","regulation of mitotic nuclear division","iron ion homeostasis","cell death in response to hydrogen peroxide","cellular detoxification","cortical actin cytoskeleton organization","positive regulation of mitochondrial membrane permeability","regulation of intrinsic apoptotic signaling pathway in response to DNA damage","stem cell differentiation","lymphocyte differentiation","regulation of developmental growth","cell-substrate junction organization","T cell differentiation in thymus","regulation of actin filament-based process","regulation of mitotic cell cycle","defense response to virus","receptor-mediated endocytosis","base-excision repair","positive regulation of muscle cell apoptotic process","protein kinase C signaling","regulation of superoxide metabolic process","nuclear outer membrane-endoplasmic reticulum membrane network","chromatin DNA binding","positive regulation of mRNA metabolic process","regulation of epithelial cell apoptotic process","response to heat","hematopoietic progenitor cell differentiation","negative regulation of supramolecular fiber organization","regulation of ossification","negative regulation of cell population proliferation","regulation of stress fiber assembly","antimicrobial humoral response","neuron projection cytoplasm","positive regulation of ATPase activity","positive regulation of G1/S transition of mitotic cell cycle","positive regulation of membrane permeability","positive regulation of substrate adhesion-dependent cell spreading","superoxide anion generation","vasodilation","protein modification by small protein conjugation or removal","regulation of ATP metabolic process","regulation of dendritic spine development","DNA replication","female pregnancy","chromatin remodeling","regulation of cell cycle G2/M phase transition","regulation of muscle cell apoptotic process","ear development","ADP binding","DNA-dependent DNA replication maintenance of fidelity","filamentous actin","one-carbon metabolic process","positive regulation of nucleotide metabolic process","positive regulation of purine nucleotide metabolic process","protein insertion into membrane","bounding membrane of organelle","nucleosome assembly","actin filament bundle assembly","gene silencing by RNA","dendritic spine organization","protein kinase complex","endomembrane system","actin filament capping","cell cycle DNA replication","G-protein alpha-subunit binding","macrophage differentiation","maintenance of protein localization in organelle","programmed necrotic cell death","regulation of renal system process","activation of transmembrane receptor protein tyrosine kinase activity","Barr body","base-excision repair, gap-filling","cell quiescence","chromo shadow domain binding","detection of virus","dipeptide transmembrane transport","dipeptide transport","DNA insertion or deletion binding","eosinophil differentiation","fermentation","hypermethylation of CpG island","lactate dehydrogenase activity","MMXD complex","negative regulation of cholesterol biosynthetic process","negative regulation of cholesterol metabolic process","negative regulation of dendritic spine maintenance","negative regulation of histone H3-K27 methylation","negative regulation of lamellipodium organization","negative regulation of respiratory burst involved in inflammatory response","negative regulation of sister chromatid cohesion","negative regulation of sterol biosynthetic process","negative regulation of transcription from RNA polymerase II promoter in response to endoplasmic reticulum stress","oxidoreductase activity, acting on diphenols and related substances as donors, cytochrome as acceptor","oxidoreductase activity, acting on superoxide radicals as acceptor","parathyroid hormone secretion","pericentric heterochromatin assembly","pericentric heterochromatin organization","porin activity","positive regulation of blood-brain barrier permeability","positive regulation of cellular pH reduction","positive regulation of hydrogen peroxide biosynthetic process","positive regulation of isomerase activity","positive regulation of maintenance of mitotic sister chromatid cohesion","positive regulation of NAD(P)H oxidase activity","protein insertion into mitochondrial outer membrane","protein localization to cell leading edge","protein tyrosine kinase inhibitor activity","regulation of chromosome condensation","regulation of cysteine-type endopeptidase activity involved in execution phase of apoptosis","regulation of intrinsic apoptotic signaling pathway in response to osmotic stress","regulation of isomerase activity","regulation of oxidative phosphorylation uncoupler activity","regulation of small intestine smooth muscle contraction","regulatory region RNA binding","rescue of stalled ribosome","response to selenium ion","rRNA transport","small intestine smooth muscle contraction","superoxide dismutase activity","Tie signaling pathway","ubiquinol-cytochrome-c reductase activity","negative regulation of cellular catabolic process","protein-DNA complex assembly","regulation of actin filament polymerization","cognition","DNA modification","nucleotide metabolic process","negative regulation of cell communication","cytoplasmic vesicle membrane","actin filament bundle organization","cell fate determination","central nervous system neuron axonogenesis","positive regulation of histone methylation","regulation of protein localization to cell surface","cardiac cell development","microvillus","negative regulation of signaling","cell growth","positive regulation of MAPK cascade","chromosome","chromatin assembly or disassembly","placenta development","response to ketone","mitochondrial membrane organization","muscle cell apoptotic process","enzyme linked receptor protein signaling pathway","axon","appendage morphogenesis","limb morphogenesis","positive regulation of mRNA processing","regulation of viral transcription","leukocyte activation","response to organonitrogen compound","protein phosphatase binding","methylated histone binding","methylation-dependent protein binding","negative regulation of viral process","phosphoprotein binding","regulation of mRNA stability","regulation of NIK/NF-kappaB signaling","extracellular region","nucleoside phosphate metabolic process","cellular response to glucocorticoid stimulus","cellular response to toxic substance","positive regulation of DNA-templated transcription, initiation","positive regulation of pri-miRNA transcription by RNA polymerase II","SH2 domain binding","actin filament organization","negative regulation of neuron differentiation","macromolecule catabolic process","heart development","negative regulation of transferase activity","spindle organization","Golgi membrane","tissue homeostasis","neuron projection organization","regulation of release of cytochrome c from mitochondria","regulation of response to reactive oxygen species","spliceosomal complex assembly","methylation","Notch signaling pathway","alkylglycerophosphoethanolamine phosphodiesterase activity","annulate lamellae","BH3 domain binding","cellular response to nicotine","corticotropin-releasing hormone receptor binding","dense fibrillar component","DNA integration","DNA translocase activity","G-quadruplex DNA unwinding","hair follicle placode formation","heparin biosynthetic process","HFE-transferrin receptor complex","histone pre-mRNA 3'end processing complex","lactate biosynthetic process","Lewy body","mast cell cytokine production","MutLalpha complex binding","myeloid dendritic cell cytokine production","negative regulation of macrophage differentiation","negative regulation of MHC class II biosynthetic process","negative regulation of protein export from nucleus","negative regulation of trophoblast cell migration","neuronal ribonucleoprotein granule","norepinephrine uptake","oxidoreductase activity, acting on diphenols and related substances as donors","positive regulation of hydrogen peroxide metabolic process","positive regulation of maintenance of sister chromatid cohesion","positive regulation of myeloid dendritic cell cytokine production","positive regulation of nuclear cell cycle DNA replication","positive regulation of protein localization to nucleolus","positive regulation of respiratory burst","positive regulation of tau-protein kinase activity","positive regulation of vascular associated smooth muscle cell differentiation","primitive erythrocyte differentiation","protein localization to adherens junction","rDNA binding","regulation of cyclic-nucleotide phosphodiesterase activity","regulation of endodermal cell differentiation","regulation of myeloid dendritic cell cytokine production","regulation of protein catabolic process at postsynapse, modulating synaptic transmission","regulation of respiratory burst involved in inflammatory response","response to interleukin-12","response to parathyroid hormone","spongiotrophoblast differentiation","ubiquinone binding","ubiquitin ligase inhibitor activity","sister chromatid segregation","cell-substrate adhesion","nucleobase-containing compound catabolic process","axonal growth cone","cellular response to corticosteroid stimulus","chromosome condensation","histone H3-K9 modification","negative regulation of actin filament depolymerization","negative regulation of intracellular protein transport","postsynaptic signal transduction","regulation of membrane depolarization","spleen development","viral transcription","nuclear-transcribed mRNA catabolic process","second-messenger-mediated signaling","histone binding","G1/S transition of mitotic cell cycle","protein alkylation","protein methylation","chromosome, telomeric region","mRNA transport","positive regulation of intracellular transport","developmental cell growth","sarcomere organization","ion channel regulator activity","NIK/NF-kappaB signaling","positive regulation of response to DNA damage stimulus","cellular homeostasis","protein-DNA complex","regulation of transmembrane transporter activity","binding of sperm to zona pellucida","nucleosome binding","positive regulation of ATP metabolic process","vascular endothelial growth factor receptor signaling pathway","actin filament polymerization","leukocyte migration","contractile actin filament bundle assembly","spindle assembly","stress fiber assembly","regulation of actin polymerization or depolymerization","monovalent inorganic cation transmembrane transporter activity","mitochondrial matrix","mRNA stabilization","positive regulation of cell cycle G1/S phase transition","RNA polymerase II activating transcription factor binding","plasma membrane protein complex","vesicle membrane","peptidyl-amino acid modification","commitment of neuronal cell to specific neuron type in forebrain","DNA hypermethylation","double-stranded methylated DNA binding","endocardium morphogenesis","endoplasmic reticulum quality control compartment","establishment of protein localization to chromatin","fat pad development","intrinsic apoptotic signaling pathway in response to osmotic stress","LIM domain binding","mammalian oogenesis stage","mast cell differentiation","mast cell granule","N6-methyladenosine-containing RNA binding","negative regulation of amyloid precursor protein biosynthetic process","negative regulation of fat cell proliferation","negative regulation of histone H3-K4 methylation","negative regulation of hydrogen peroxide-induced neuron death","negative regulation of monocyte differentiation","negative regulation of respiratory burst","negative regulation of retinoic acid receptor signaling pathway","negative regulation of smoothened signaling pathway involved in dorsal/ventral neural tube patterning","neuron death in response to hydrogen peroxide","oligopeptide transmembrane transport","opsonization","paraspeckles","PML body organization","polysome binding","positive regulation of attachment of spindle microtubules to kinetochore","positive regulation of cell proliferation involved in heart morphogenesis","positive regulation of dendritic cell chemotaxis","positive regulation of epithelial cell proliferation involved in lung morphogenesis","positive regulation of gastro-intestinal system smooth muscle contraction","positive regulation of long-term neuronal synaptic plasticity","positive regulation of vascular associated smooth muscle cell apoptotic process","pre-mRNA 3'-splice site binding","protein hexamerization","protein localization to bicellular tight junction","regulation of cellular pH reduction","regulation of Fc-gamma receptor signaling pathway involved in phagocytosis","regulation of hydrogen peroxide-induced neuron death","regulation of low-density lipoprotein particle receptor biosynthetic process","regulation of transforming growth factor beta activation","regulation of translation at synapse","regulation of translation at synapse, modulating synaptic transmission","respiratory burst involved in inflammatory response","response to glucagon","ribosomal large subunit export from nucleus","telomere assembly","thioredoxin peroxidase activity","TPR domain binding","ubiquitin-protein transferase inhibitor activity","unconventional myosin complex","virion attachment to host cell","response to metal ion","striated muscle cell development","peptidyl-lysine modification","positive regulation of establishment of protein localization"],["CC","CC","CC","CC","MF","MF","CC","CC","BP","BP","BP","BP","BP","BP","CC","CC","CC","MF","BP","BP","CC","CC","MF","CC","BP","BP","BP","BP","MF","BP","BP","MF","BP","BP","BP","CC","CC","CC","MF","BP","BP","CC","BP","MF","BP","BP","CC","CC","CC","MF","BP","BP","CC","BP","BP","BP","BP","CC","MF","BP","BP","BP","BP","BP","MF","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","CC","CC","BP","BP","BP","BP","BP","BP","CC","CC","CC","CC","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","MF","BP","BP","MF","BP","BP","CC","BP","BP","BP","CC","BP","BP","BP","BP","BP","CC","CC","CC","BP","BP","CC","BP","BP","BP","BP","BP","BP","BP","BP","BP","CC","CC","BP","BP","BP","MF","BP","CC","BP","CC","BP","BP","BP","BP","BP","CC","BP","BP","BP","BP","MF","BP","BP","BP","CC","BP","BP","BP","CC","BP","BP","BP","BP","MF","BP","BP","CC","BP","CC","BP","CC","CC","CC","CC","BP","BP","BP","BP","BP","MF","MF","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","CC","BP","BP","BP","CC","BP","MF","BP","BP","CC","BP","BP","CC","BP","BP","BP","BP","CC","BP","BP","BP","CC","BP","BP","BP","BP","BP","CC","BP","BP","BP","BP","BP","BP","BP","BP","CC","BP","BP","BP","BP","CC","BP","BP","BP","BP","MF","CC","BP","CC","MF","BP","BP","BP","BP","CC","BP","BP","CC","CC","CC","MF","BP","BP","BP","BP","BP","BP","MF","BP","CC","CC","MF","BP","CC","BP","MF","BP","BP","CC","BP","CC","BP","MF","MF","BP","BP","BP","BP","CC","BP","BP","BP","BP","CC","BP","BP","BP","BP","BP","BP","CC","CC","BP","CC","BP","CC","BP","MF","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","CC","BP","BP","CC","BP","CC","CC","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","CC","BP","BP","MF","BP","BP","BP","BP","BP","BP","CC","BP","BP","MF","BP","BP","BP","BP","BP","BP","BP","BP","BP","MF","BP","BP","BP","BP","BP","MF","BP","BP","BP","BP","BP","MF","BP","BP","BP","MF","MF","MF","BP","MF","BP","BP","BP","BP","BP","BP","BP","BP","BP","CC","MF","MF","BP","BP","BP","BP","BP","BP","BP","CC","BP","BP","BP","BP","BP","BP","BP","MF","MF","BP","MF","CC","BP","MF","BP","BP","CC","BP","BP","BP","BP","BP","BP","BP","BP","MF","BP","BP","BP","BP","CC","BP","BP","MF","BP","MF","BP","BP","BP","BP","BP","BP","BP","MF","CC","CC","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","MF","BP","MF","BP","BP","BP","BP","MF","MF","BP","BP","BP","BP","MF","BP","BP","MF","MF","MF","MF","MF","BP","BP","CC","BP","BP","MF","MF","BP","MF","BP","BP","BP","BP","BP","MF","CC","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","CC","BP","CC","BP","BP","BP","BP","BP","BP","BP","CC","CC","CC","MF","BP","CC","MF","BP","MF","MF","MF","MF","BP","MF","BP","BP","BP","MF","BP","BP","CC","BP","MF","BP","BP","BP","MF","BP","BP","BP","BP","BP","BP","BP","BP","MF","BP","BP","BP","BP","BP","CC","BP","BP","BP","BP","CC","MF","BP","BP","BP","BP","CC","MF","BP","MF","MF","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","MF","MF","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","CC","BP","BP","MF","MF","BP","MF","BP","MF","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","MF","BP","BP","BP","BP","BP","BP","BP","BP","BP","CC","CC","MF","MF","MF","MF","MF","BP","BP","BP","BP","BP","CC","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","CC","BP","BP","BP","CC","BP","BP","BP","BP","CC","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","MF","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","MF","BP","BP","BP","MF","MF","BP","MF","BP","MF","BP","MF","BP","BP","BP","BP","BP","BP","BP","MF","CC","BP","BP","MF","CC","BP","BP","BP","BP","BP","MF","BP","CC","MF","BP","BP","BP","CC","BP","MF","BP","BP","CC","BP","MF","CC","BP","BP","BP","BP","MF","CC","BP","BP","BP","CC","BP","BP","MF","BP","BP","MF","BP","CC","BP","BP","MF","MF","BP","BP","BP","MF","BP","BP","BP","BP","MF","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","CC","CC","MF","BP","BP","BP","BP","BP","BP","BP","BP","BP","CC","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","CC","BP","CC","BP","BP","MF","MF","MF","CC","CC","BP","BP","BP","BP","BP","MF","CC","CC","CC","BP","BP","MF","MF","BP","BP","BP","BP","BP","BP","BP","BP","BP","CC","BP","BP","BP","BP","MF","BP","CC","BP","MF","MF","BP","BP","BP","MF","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","MF","BP","BP","BP","BP","BP","BP","BP","MF","BP","BP","BP","BP","BP","BP","BP","CC","CC","MF","BP","MF","BP","BP","BP","BP","BP","BP","CC","MF","MF","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","MF","BP","BP","BP","BP","BP","BP","BP","MF","BP","BP","CC","MF","MF","BP","MF","CC","CC","CC","BP","BP","CC","CC","BP","CC","BP","MF","BP","BP","BP","MF","CC","BP","MF","CC","BP","BP","BP","BP","BP","BP","MF","BP","BP","BP","BP","BP","BP","CC","CC","MF","CC","MF","MF","CC","MF","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","CC","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","MF","BP","BP","BP","BP","CC","CC","BP","BP","BP","BP","BP","MF","BP","BP","MF","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","MF","BP","BP","BP","BP","CC","BP","BP","BP","BP","BP","BP","BP","CC","BP","BP","CC","BP","BP","BP","BP","BP","BP","BP","BP","BP","MF","BP","BP","BP","BP","MF","BP","CC","BP","MF","BP","BP","CC","MF","BP","BP","BP","BP","BP","BP","MF","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","MF","BP","BP","MF","BP","BP","BP","BP","BP","BP","BP","CC","BP","BP","BP","MF","MF","BP","MF","MF","BP","MF","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","MF","CC","BP","CC","BP","MF","BP","CC","MF","BP","BP","BP","BP","BP","BP","CC","BP","MF","BP","BP","BP","CC","BP","MF","BP","MF","BP","BP","MF","CC","MF","BP","BP","BP","BP","CC","BP","BP","BP","BP","CC","CC","CC","BP","BP","MF","BP","BP","BP","BP","MF","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","MF","MF","BP","BP","BP","BP","CC","BP","CC","CC","BP","BP","BP","BP","BP","MF","CC","CC","BP","BP","BP","BP","MF","CC","BP","BP","BP","BP","BP","BP","BP","BP","CC","CC","BP","MF","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","MF","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","CC","BP","MF","BP","BP","BP","BP","MF","BP","BP","MF","BP","MF","BP","BP","BP","BP","BP","MF","BP","BP","CC","MF","BP","BP","CC","BP","BP","MF","CC","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","MF","BP","CC","MF","MF","BP","CC","BP","BP","CC","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","CC","BP","BP","BP","BP","MF","BP","BP","BP","BP","CC","BP","CC","BP","CC","BP","BP","BP","CC","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","CC","MF","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","CC","MF","BP","BP","BP","BP","BP","MF","MF","BP","BP","BP","MF","CC","BP","MF","BP","CC","BP","BP","BP","CC","MF","CC","MF","MF","CC","CC","BP","BP","BP","CC","CC","BP","CC","CC","CC","BP","BP","BP","BP","BP","BP","BP","CC","MF","CC","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","MF","MF","BP","BP","MF","BP","BP","BP","CC","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","CC","BP","CC","BP","BP","BP","BP","BP","BP","MF","MF","BP","BP","CC","BP","MF","BP","BP","BP","BP","BP","BP","MF","MF","BP","BP","BP","BP","BP","BP","MF","BP","BP","BP","CC","MF","BP","BP","BP","MF","BP","BP","BP","BP","BP","BP","BP","BP","BP","MF","BP","CC","BP","MF","CC","CC","BP","BP","BP","BP","BP","BP","BP","MF","BP","BP","CC","BP","BP","BP","MF","BP","BP","BP","BP","BP","BP","MF","BP","BP","BP","BP","MF","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","MF","BP","MF","BP","CC","BP","BP","BP","BP","BP","BP","CC","BP","MF","BP","BP","BP","MF","CC","MF","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","MF","BP","BP","BP","BP","BP","BP","BP","BP","BP","CC","BP","BP","MF","CC","MF","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","CC","MF","BP","BP","BP","BP","BP","BP","BP","BP","BP","CC","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","MF","BP","CC","BP","BP","BP","BP","CC","BP","BP","BP","BP","CC","CC","BP","BP","MF","BP","BP","BP","BP","BP","CC","BP","BP","MF","BP","BP","BP","MF","BP","BP","BP","MF","CC","BP","BP","BP","BP","BP","BP","BP","BP","BP","MF","MF","BP","BP","BP","MF","BP","BP","BP","BP","BP","BP","BP","BP","MF","BP","BP","BP","BP","BP","BP","MF","BP","BP","BP","BP","MF","BP","MF","BP","BP","BP","BP","BP","BP","BP","CC","BP","BP","BP","BP","BP","BP","CC","BP","BP","BP","CC","BP","BP","BP","BP","BP","BP","CC","BP","BP","BP","BP","BP","BP","MF","MF","MF","BP","MF","BP","BP","CC","BP","BP","BP","BP","BP","MF","BP","BP","BP","BP","BP","BP","CC","BP","BP","BP","BP","BP","BP","BP","MF","CC","MF","BP","MF","CC","BP","MF","BP","BP","BP","CC","CC","BP","CC","BP","MF","BP","BP","BP","BP","BP","CC","BP","MF","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","MF","BP","BP","BP","BP","BP","BP","BP","BP","MF","MF","BP","BP","BP","CC","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","MF","BP","BP","BP","CC","BP","BP","BP","BP","MF","BP","BP","BP","CC","BP","BP","MF","BP","BP","BP","BP","BP","BP","BP","BP","MF","CC","BP","BP","MF","CC","CC","BP","BP","BP","MF","BP","CC","BP","BP","BP","MF","BP","BP","CC","MF","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","CC","BP","MF","BP","BP","BP","BP","BP","BP","BP","MF","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","MF","MF","MF","CC","BP","BP","BP","BP","BP"],[115,234,198,813,158,1147,3587,5423,5710,5865,627,649,4378,795,46,14122,12366,576,761,5326,12722,78,3419,11142,5512,5414,4512,9877,5346,4455,11039,5447,9152,1494,1038,64,4569,4551,345,9042,10518,125,9716,72,7589,4712,11462,10684,74,13753,6250,6735,1451,5561,17114,854,5104,7143,9318,158,5500,5318,91,5238,2366,116,5936,5725,6063,5530,4666,386,4677,5242,4204,4950,3945,19008,733,2102,3729,154,3661,3470,3865,4594,4595,4595,2065,3711,5834,1613,462,3782,3129,3133,1517,3179,6358,21,3479,266,73,271,271,271,1497,1497,409,2321,2595,2869,3093,1395,2392,1483,1571,2503,146,1943,1794,31,650,1899,3672,439,296,3737,4153,172,3750,213,400,4180,108,3879,405,179,1830,3305,5,5,1290,1294,1295,934,433,196,66,2087,95,127,167,131,253,15,1699,6229,308,308,1971,218,931,535,1397,2730,109,1252,951,1579,676,1028,5977,6449,229,187,149,2496,1960,35,85,1706,707,1067,710,996,20,1077,1077,3351,1917,4418,4310,206,355,306,803,951,5467,3297,3201,3304,12212,3208,215,3312,11652,3211,1871,1787,23,2438,1798,1541,690,2743,324,138,4104,444,2381,73,26,74,3713,931,2031,12,2904,2813,400,111,193,1517,295,51,410,1534,29,4431,6040,413,3366,355,414,967,82,157,157,3587,4571,83,2575,12806,614,160,832,489,1403,626,697,88,374,125,704,267,932,4878,91,16,16,16,35,61,330,1275,17,17,17,953,393,1037,98,182,814,1658,747,39,1660,1845,2321,2324,1665,19,19,409,68,293,6076,1157,41,41,1961,146,20,358,1003,1975,848,72,699,21,2177,200,200,44,44,309,22,573,46,77,1288,2213,1555,1648,2820,1558,1559,379,379,879,380,444,320,79,2832,24,24,24,969,1484,119,1137,81,121,2553,2254,9,9,217,1233,1065,2663,2461,123,395,51,84,2169,3972,1613,127,616,1093,2010,477,178,55,2,2,481,56,292,236,11,420,422,30,424,240,95,1571,715,2455,645,12,12,32,32,192,99,1142,248,507,100,33,440,1328,815,13,64,587,254,34,34,315,591,65,149,1163,1254,150,150,1814,202,66,2315,67,2946,1826,153,756,36,1179,603,1360,68,68,1361,1271,207,155,1097,69,37,1459,608,2967,157,110,110,3,3,3,3,70,332,2874,159,692,71,71,112,113,161,273,73,1486,276,115,74,116,1590,2596,41,76,17,17,797,224,169,77,170,1801,564,78,2109,79,1715,44,814,4,4,4,4,2128,295,126,1167,3745,82,2033,2033,297,46,83,83,128,299,912,84,368,369,240,20,20,131,2574,184,836,86,1565,185,49,447,88,88,135,50,89,137,769,251,51,51,51,5,5,5,5,5,5,5,5,5,22,1788,1121,91,317,318,141,390,23,1607,1805,1045,785,259,1613,1231,1812,95,1817,470,145,201,399,202,2874,97,56,798,98,403,478,25,25,25,6,6,6,6,6,6,6,6,6,6,6,6,6,57,57,206,150,806,807,984,26,26,26,26,26,645,489,816,103,60,60,650,27,27,277,1768,105,157,158,1876,1190,7,7,7,7,7,62,62,28,28,28,218,746,160,284,426,220,108,503,427,221,29,29,109,2101,287,586,1794,2964,163,110,588,289,164,224,65,845,1501,511,30,30,165,165,165,435,937,227,438,113,113,1220,8,8,8,8,8,8,8,680,365,1713,31,31,31,440,1036,114,114,169,68,68,68,68,231,231,115,232,32,32,32,32,69,372,691,1143,235,70,70,118,3472,451,33,33,1148,1148,119,613,1742,1743,9,9,1950,120,3486,34,1956,309,791,1752,122,122,793,242,1067,707,180,977,35,35,74,462,626,799,1663,1663,1267,802,802,125,2730,1366,3519,1078,1467,986,10,10,10,10,10,10,10,10,126,76,391,721,1087,37,128,995,1581,997,254,38,38,1002,190,479,401,1798,1392,80,11,11,11,11,11,132,1906,1103,192,39,39,81,82,261,40,195,833,83,8423,1118,41,84,12,12,12,12,12,12,199,1724,85,85,416,200,200,1421,936,139,42,1835,1033,86,140,140,1131,1036,670,420,7943,141,87,43,43,1039,673,204,587,13,1541,44,89,144,508,2509,45,209,773,512,279,210,147,147,514,14,14,14,14,14,14,14,1455,355,92,2087,212,282,781,149,213,93,47,47,150,438,94,362,2432,95,95,48,48,442,217,217,611,15,15,15,15,15,1174,1579,96,96,1077,365,365,793,219,154,366,49,49,49,49,49,291,705,97,221,221,293,533,222,50,370,621,157,223,16,16,16,16,16,16,537,51,51,51,225,100,159,1498,298,375,299,52,52,377,102,1102,460,379,379,17,17,17,17,17,17,17,723,53,103,163,163,164,164,104,1311,54,54,54,54,54,384,233,1417,105,105,307,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,166,166,18,18,18,18,18,18,55,55,167,557,107,107,107,107,738,56,56,56,56,648,740,740,108,313,835,652,57,57,109,109,109,19,19,171,171,171,395,1863,566,566,110,110,1035,938,58,58,58,58,657,111,2423,400,320,59,59,59,59,59,59,1043,20,20,20,20,20,20,20,662,246,60,854,324,324,114,9692,115,21,21,21,21,21,581,251,116,1573,1574,62,62,62,329,1059,181,117,253,676,182,182,254,413,587,63,63,1581,22,22,22,22,22,22,22,774,415,334,502,64,592,684,258,336,337,23,23,23,23,23,23,188,980,189,123,123,884,1392,1084,512,124,191,24,24,24,24,24,24,24,24,24,24,24,698,345,1090,1939,68,68,193,126,194,1946,69,69,69,127,611,2171,195,195,25,25,25,25,25,270,434,128,128,196,196,196,523,523,351,271,70,70,70,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,1628,129,129,129,129,197,197,806,273,71,354,1107,712,619,26,26,26,26,26,26,26,26,26,714,131,441,200,200,1112,132,717,623,814,277,624,73,73,73,1645,1117,1325,27,27,27,27,361,134,134,134,74,538,135,449,1440,75,75,206,136,633,28,28,28,28,28,28,730,137,76,76,76,638,209,77,139,139,29,29,29,78,78,550,837,937,212,740,1145,213,213,30,30,30,30,30,30,647,647,214,214,1148,215,80,747,3538,3538,216,5513,144,144,81,81,31,31,31,31,31,31,296,23271,297,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,382,82,471,147,147,32,32,32,32,32,83,83,83,83,83,301,1379,567,148,222,387,84,1065,1710,223,666,33,33,33,33,33,33,33,150,150,85,85,151,306,670,1611,1611,152,1613,394,34,34,34,34,34,34,34,87,87,228,396,676,311,88,88,88,88,3722,582,312,35,35,35,35,35,35,35,35,35,35,35,231,879,89,89,89,89,156,156,401,90,36,36,36,36,36,36,36,158,404,1200,235,235,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,1861,159,159,159,159,236,160,92,37,37,37,37,37,237,409,409,93,93,410,502,239,240,38,38,38,38,599,94,94,94,94,163,163,241,699,95,165,39,39,39,39,39,39,39,803,96,96,244,166,167,97,97,246,40,40,40,40,40,40,40,1122,98,169,169,99,99,3929,41,41,41,41,41,41,41,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,250,171,171,336,100,427,1350,337,172,42,42,42,42,101,101,1353,523,523,1464,173,174,174,102,102,928,724,175,175,43,43,1036,932,176,104,104,104,104,104,104,2750,436,44,44,44,44,44,437,259,1260,631,260,179,261,261,106,45,45,45,349,180,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,181,351,352,46,46,46,46,46,46,46,46,46,46,108,445,265,183,183,183,109,109,184,267,47,110,110,110,957,186,270,48,48,48,48,187,360,112,112,112,188,361,273,49,49,49,554,363,1184,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,274,190,364,364],[35,41,38,57,33,62,92,111,113,114,43,43,96,45,18,165,156,39,43,103,157,20,82,147,103,102,93,138,101,92,145,101,131,53,45,17,90,89,28,127,137,19,130,16,109,83,137,131,14,150,94,98,42,87,165,32,82,100,117,16,85,83,13,82,52,14,88,86,89,84,75,21,75,80,70,77,67,170,27,46,64,14,63,61,65,72,72,72,45,63,83,39,21,63,56,56,37,56,86,7,59,16,10,16,16,16,36,36,19,46,49,52,54,34,46,35,36,47,12,40,38,7,22,39,58,18,15,58,62,12,58,13,17,62,10,59,17,12,37,53,4,4,30,30,30,25,17,12,8,39,9,10,11,10,13,5,34,78,14,14,37,12,24,18,30,45,9,28,24,32,20,25,75,79,12,11,10,42,36,6,8,33,20,25,20,24,5,25,25,50,35,60,59,11,14,13,21,23,69,49,48,49,121,48,11,49,117,48,34,33,5,40,33,30,19,43,13,9,56,15,39,7,5,7,52,22,35,4,44,43,14,8,10,29,12,6,14,29,5,58,72,14,48,13,14,22,7,9,9,50,59,7,40,123,17,9,20,15,27,17,18,7,13,8,18,11,21,61,7,4,4,4,5,6,12,25,4,4,4,21,13,22,7,9,19,29,18,5,29,31,36,36,29,4,4,13,6,11,70,23,5,5,32,8,4,12,21,32,19,6,17,4,34,9,9,5,5,11,4,15,5,6,24,34,27,28,40,27,27,12,12,19,12,13,11,6,40,4,4,4,20,26,7,22,6,7,37,34,3,3,9,23,21,38,36,7,12,5,6,33,50,27,7,15,21,31,13,8,5,2,2,13,5,10,9,3,12,12,4,12,9,6,26,16,35,15,3,3,4,4,8,6,21,9,13,6,4,12,23,17,3,5,14,9,4,4,10,14,5,7,21,22,7,7,28,8,5,33,5,39,28,7,16,4,21,14,23,5,5,23,22,8,7,20,5,4,24,14,39,7,6,6,2,2,2,2,5,10,38,7,15,5,5,6,6,7,9,5,24,9,6,5,6,25,35,4,5,3,3,16,8,7,5,7,27,13,5,30,5,26,4,16,2,2,2,2,30,9,6,20,45,5,29,29,9,4,5,5,6,9,17,5,10,10,8,3,3,6,34,7,16,5,24,7,4,11,5,5,6,4,5,6,15,8,4,4,4,2,2,2,2,2,2,2,2,2,3,26,19,5,9,9,6,10,3,24,26,18,15,8,24,20,26,5,26,11,6,7,10,7,36,5,4,15,5,10,11,3,3,3,2,2,2,2,2,2,2,2,2,2,2,2,2,4,4,7,6,15,15,17,3,3,3,3,3,13,11,15,5,4,4,13,3,3,8,25,5,6,6,26,19,2,2,2,2,2,4,4,3,3,3,7,14,6,8,10,7,5,11,10,7,3,3,5,28,8,12,25,36,6,5,12,8,6,7,4,15,22,11,3,3,6,6,6,10,16,7,10,5,5,19,2,2,2,2,2,2,2,13,9,24,3,3,3,10,17,5,5,6,4,4,4,4,7,7,5,7,3,3,3,3,4,9,13,18,7,4,4,5,40,10,3,3,18,18,5,12,24,24,2,2,26,5,40,3,26,8,14,24,5,5,14,7,17,13,6,16,3,3,4,10,12,14,23,23,19,14,14,5,33,20,40,17,21,16,2,2,2,2,2,2,2,2,5,4,9,13,17,3,5,16,22,16,7,3,3,16,6,10,9,24,20,4,2,2,2,2,2,5,25,17,6,3,3,4,4,7,3,6,14,4,79,17,3,4,2,2,2,2,2,2,6,23,4,4,9,6,6,20,15,5,3,24,16,4,5,5,17,16,12,9,75,5,4,3,3,16,12,6,11,2,21,3,4,5,10,30,3,6,13,10,7,6,5,5,10,2,2,2,2,2,2,2,20,8,4,26,6,7,13,5,6,4,3,3,5,9,4,8,29,4,4,3,3,9,6,6,11,2,2,2,2,2,17,21,4,4,16,8,8,13,6,5,8,3,3,3,3,3,7,12,4,6,6,7,10,6,3,8,11,5,6,2,2,2,2,2,2,10,3,3,3,6,4,5,20,7,8,7,3,3,8,4,16,9,8,8,2,2,2,2,2,2,2,12,3,4,5,5,5,5,4,18,3,3,3,3,3,8,6,19,4,4,7,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,5,5,2,2,2,2,2,2,3,3,5,10,4,4,4,4,12,3,3,3,3,11,12,12,4,7,13,11,3,3,4,4,4,2,2,5,5,5,8,23,10,10,4,4,15,14,3,3,3,3,11,4,28,8,7,3,3,3,3,3,3,15,2,2,2,2,2,2,2,11,6,3,13,7,7,4,86,4,2,2,2,2,2,10,6,4,20,20,3,3,3,7,15,5,4,6,11,5,5,6,8,10,3,3,20,2,2,2,2,2,2,2,12,8,7,9,3,10,11,6,7,7,2,2,2,2,2,2,5,14,5,4,4,13,18,15,9,4,5,2,2,2,2,2,2,2,2,2,2,2,11,7,15,23,3,3,5,4,5,23,3,3,3,4,10,25,5,5,2,2,2,2,2,6,8,4,4,5,5,5,9,9,7,6,3,3,3,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,20,4,4,4,4,5,5,12,6,3,7,15,11,10,2,2,2,2,2,2,2,2,2,11,4,8,5,5,15,4,11,10,12,6,10,3,3,3,20,15,17,2,2,2,2,7,4,4,4,3,9,4,8,18,3,3,5,4,10,2,2,2,2,2,2,11,4,3,3,3,10,5,3,4,4,2,2,2,3,3,9,12,13,5,11,15,5,5,2,2,2,2,2,2,10,10,5,5,15,5,3,11,36,36,5,52,4,4,3,3,2,2,2,2,2,2,6,174,6,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,7,3,8,4,4,2,2,2,2,2,3,3,3,3,3,6,17,9,4,5,7,3,14,20,5,10,2,2,2,2,2,2,2,4,4,3,3,4,6,10,19,19,4,19,7,2,2,2,2,2,2,2,3,3,5,7,10,6,3,3,3,3,37,9,6,2,2,2,2,2,2,2,2,2,2,2,5,12,3,3,3,3,4,4,7,3,2,2,2,2,2,2,2,4,7,15,5,5,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,21,4,4,4,4,5,4,3,2,2,2,2,2,5,7,7,3,3,7,8,5,5,2,2,2,2,9,3,3,3,3,4,4,5,10,3,4,2,2,2,2,2,2,2,11,3,3,5,4,4,3,3,5,2,2,2,2,2,2,2,14,3,4,4,3,3,38,2,2,2,2,2,2,2,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,5,4,4,6,3,7,16,6,4,2,2,2,2,3,3,16,8,8,17,4,4,4,3,3,12,10,4,4,2,2,13,12,4,3,3,3,3,3,3,28,7,2,2,2,2,2,7,5,15,9,5,4,5,5,3,2,2,2,6,4,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,4,6,6,2,2,2,2,2,2,2,2,2,2,3,7,5,4,4,4,3,3,4,5,2,3,3,3,12,4,5,2,2,2,2,4,6,3,3,3,4,6,5,2,2,2,8,6,14,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,5,4,6,6],[8.82237543200144e-48,4.20204673042257e-45,2.22683830067703e-43,2.52365824596224e-40,5.61663671456481e-39,1.86299819126449e-37,1.61047212129077e-31,3.55965065330106e-31,1.28389755366044e-30,2.78994235791703e-30,1.14176701703694e-29,4.69076021354044e-29,8.98182272035637e-28,1.53607758873783e-27,2.96417771950121e-27,7.3252461859233e-27,1.11596350617639e-26,1.14659625727026e-26,2.94290767427268e-26,6.56849946121368e-26,7.26161026924923e-26,7.30659336198973e-26,2.65593742427718e-25,3.73075030979742e-25,1.14586385876309e-24,1.27890862586325e-24,1.37369793173302e-24,1.38447340148553e-24,2.22607530858848e-24,2.65222925086016e-24,4.07060664731546e-24,1.01012975555668e-23,3.00203709780501e-23,6.41227924866524e-23,8.93746974358617e-23,1.96571512715383e-22,3.8337424090934e-22,1.32583396191478e-21,2.38286342053543e-21,4.04061241103019e-21,7.38203232915093e-21,5.7146260893078e-20,6.45171704553993e-20,8.40372753023547e-20,4.33295524403966e-17,6.62554674631921e-17,6.74183723133399e-17,1.93001878337032e-16,2.04687874377314e-16,3.07882248437056e-16,4.23027306217885e-15,5.40808548331856e-15,6.48781259340285e-15,1.13924394189982e-14,1.17994499960916e-14,1.87698197980389e-14,2.83856749888489e-14,2.99650482915635e-14,3.59313587159694e-14,3.92426831000783e-14,6.38510451987508e-14,9.35051896798257e-14,1.1443261278846e-13,1.26605227428515e-13,1.28663970671842e-13,1.39728474951112e-13,1.9027169564767e-13,2.1584215570262e-13,2.19960542885436e-13,2.82047906991526e-13,8.12829512716224e-13,8.41675620439155e-13,9.18774557633039e-13,1.32433776300112e-12,1.44305783359252e-12,1.81356904290054e-12,2.3374244322521e-12,3.39448320436109e-12,3.87875030588059e-12,5.98889943735718e-12,6.03503183588631e-12,7.16402571861803e-12,8.6153594181257e-12,8.87636773708214e-12,9.39800180600808e-12,1.12123782868362e-11,1.13321486489467e-11,1.13321486489467e-11,1.25522893897874e-11,1.5602902404844e-11,1.74389479213819e-11,1.92249154959869e-11,2.5146617901627e-11,3.55125241530392e-11,4.68052855999885e-11,4.92080018326894e-11,5.5054265995209e-11,8.69091064782401e-11,8.89809307708138e-11,9.95963891466763e-11,1.00074294336386e-10,1.15290125208863e-10,1.29797412591677e-10,1.51839186143528e-10,1.51839186143528e-10,1.51839186143528e-10,1.56012579885423e-10,1.56012579885423e-10,1.64061490556349e-10,1.71173684717471e-10,1.92982114239333e-10,1.93938603148459e-10,3.04795369886472e-10,3.91805654998847e-10,4.61063851970101e-10,4.86302218921958e-10,5.82479929925776e-10,6.18540653074681e-10,7.65736093158121e-10,1.22869547833738e-09,1.60362038125437e-09,2.11892031674472e-09,2.17578086373149e-09,2.22968082699242e-09,2.46486717034709e-09,3.67551542645809e-09,4.70962288137073e-09,4.78304771184719e-09,4.94966560200064e-09,4.99740003134998e-09,5.44875469922695e-09,5.70909282247591e-09,6.00592709192058e-09,6.3922490406908e-09,6.50469838231103e-09,7.08584338658742e-09,7.2210417751109e-09,7.84219772320671e-09,9.57391409035264e-09,9.75201682361723e-09,1.32687946174435e-08,1.32687946174435e-08,1.44080305289401e-08,1.54463769516935e-08,1.57167231621253e-08,1.82011100244659e-08,1.93123531658702e-08,2.16559198557188e-08,2.60956938499952e-08,3.00700706433459e-08,3.09864610080899e-08,3.12096846536593e-08,4.02605136640141e-08,4.19850889685973e-08,4.42101259499696e-08,5.35426591526502e-08,5.52259017278752e-08,5.66893536325754e-08,6.0960647072511e-08,6.0960647072511e-08,6.58970698478063e-08,7.00733446118769e-08,7.28752227225954e-08,7.56543259104902e-08,8.41168017814284e-08,8.51726655981023e-08,1.03092192053163e-07,1.05367173442412e-07,1.07926472977607e-07,1.09752044190441e-07,1.11623821341785e-07,1.15436669271897e-07,1.17119665449437e-07,1.17421344436322e-07,1.19885239956941e-07,1.27929471628231e-07,1.42063641267712e-07,1.49191261113904e-07,1.76742707669453e-07,1.81626012734434e-07,1.94524699169486e-07,1.97385619853985e-07,2.28415329755593e-07,2.32623856083842e-07,2.4427586049811e-07,2.5106554800781e-07,2.68425609232851e-07,2.76841374400301e-07,2.76841374400301e-07,2.96768991683079e-07,3.1338944002011e-07,3.25905165776126e-07,3.27733597738073e-07,3.38827782504456e-07,3.45259755984884e-07,3.97320750888264e-07,4.05837434746735e-07,4.29343674924267e-07,4.44344986708659e-07,4.60879843096156e-07,4.85690343651072e-07,4.91207177623339e-07,4.92166056758115e-07,5.18077954221202e-07,5.186781644282e-07,5.28166659930362e-07,5.29026097336418e-07,5.32571672414986e-07,5.33213082640749e-07,5.6539277202239e-07,5.72397727754997e-07,6.176693981057e-07,6.48634576458994e-07,6.81087582581658e-07,7.00234785913159e-07,7.17733207609461e-07,7.57232669073229e-07,7.74369267289421e-07,8.49013120982271e-07,9.21675084615668e-07,9.23884327902516e-07,1.01646286715901e-06,1.0994318890465e-06,1.1158499659868e-06,1.13981470494827e-06,1.14835279225513e-06,1.20679078164399e-06,1.26240743678585e-06,1.29789831073049e-06,1.40929749851226e-06,1.42625526944557e-06,1.52109000595782e-06,1.54327019639404e-06,1.54835442415353e-06,1.77120484007445e-06,1.83071061453158e-06,1.90366563495368e-06,1.93231282414187e-06,1.95019604895923e-06,1.95776461461396e-06,2.02429700607429e-06,2.07239089551861e-06,2.08905234486967e-06,2.09027182328466e-06,2.13153021052399e-06,2.13418809679282e-06,2.24535385311032e-06,2.27171347671662e-06,2.27171347671662e-06,2.29689512229728e-06,2.4223975360014e-06,2.4372962769599e-06,2.45825621065819e-06,2.48191607525077e-06,2.59145769476705e-06,2.65593240565366e-06,2.82955106847304e-06,3.02450867164583e-06,3.29891121884256e-06,3.35485617333133e-06,3.43551689828752e-06,3.61507617810916e-06,3.69728651009564e-06,3.71818548653725e-06,3.94491369108526e-06,4.27726070813022e-06,4.30090640177154e-06,4.31747979686398e-06,4.52586052447339e-06,4.53743048430683e-06,4.53743048430683e-06,4.53743048430683e-06,5.14672703305773e-06,5.3159687980011e-06,5.59957910828614e-06,5.76197687554064e-06,5.9000119977277e-06,5.9000119977277e-06,5.9000119977277e-06,6.05339790607087e-06,6.31676113756818e-06,6.52376470005268e-06,7.41358531040598e-06,7.6023598147761e-06,7.74085089280246e-06,8.74269252265128e-06,8.87264222870431e-06,8.91651563065475e-06,8.94524028953335e-06,9.13628946178933e-06,9.15614254157197e-06,9.42140444129875e-06,9.47057950920056e-06,9.50026956731726e-06,9.50026956731726e-06,9.67740632880009e-06,1.00488355424832e-05,1.03006465965077e-05,1.10839306868058e-05,1.13589198553329e-05,1.14704040729443e-05,1.14704040729443e-05,1.15830008465132e-05,1.17071926746467e-05,1.18082305568384e-05,1.26933485464825e-05,1.3106750965995e-05,1.33682948358982e-05,1.37309201395762e-05,1.4004875087393e-05,1.41696454317009e-05,1.45042272568912e-05,1.4825791555466e-05,1.61907818374962e-05,1.61907818374962e-05,1.6333050139433e-05,1.6333050139433e-05,1.69006330236703e-05,1.76272549564976e-05,1.98899981684854e-05,2.03754717043701e-05,2.06262526038115e-05,2.08431661130385e-05,2.08510561505409e-05,2.11893577918031e-05,2.16524246243588e-05,2.17696645369465e-05,2.19189052918267e-05,2.21671192037867e-05,2.22962712361326e-05,2.22962712361326e-05,2.25302832417536e-05,2.28800296421619e-05,2.2956334210884e-05,2.33334347713895e-05,2.38942048946947e-05,2.40155653404945e-05,2.53175425525083e-05,2.53175425525083e-05,2.53175425525083e-05,2.61662057869297e-05,2.63442473355783e-05,2.642621678729e-05,2.70062345872152e-05,2.75664975454693e-05,2.94273292268162e-05,3.01712824992331e-05,3.03991624618679e-05,3.0464027964718e-05,3.0464027964718e-05,3.08225578711862e-05,3.09556669493105e-05,3.17287432504118e-05,3.2340958758411e-05,3.25537299674144e-05,3.27034291733458e-05,3.33790793990085e-05,3.39104361041986e-05,3.39130142462642e-05,3.43308530011998e-05,3.97602092351277e-05,4.00474820221616e-05,4.01580170117843e-05,4.55650837221895e-05,4.61574023842587e-05,4.79597138393535e-05,4.80184865502189e-05,4.86748180967113e-05,4.90572717715459e-05,5.22524928264707e-05,5.22524928264707e-05,5.22750368768639e-05,5.35527841324327e-05,5.69032659388754e-05,5.915678511056e-05,5.9203450920779e-05,6.02485728932349e-05,6.30398018670645e-05,6.31158107576414e-05,6.5942119675592e-05,6.73165178686641e-05,6.78993912322678e-05,6.82402825858395e-05,6.8831727312624e-05,7.24418609186535e-05,7.63425474126693e-05,7.851713208211e-05,7.851713208211e-05,8.18875285822171e-05,8.18875285822171e-05,8.28463828794363e-05,8.54767314527197e-05,8.60118716806185e-05,8.65024471953757e-05,8.88367761709112e-05,9.03917638511863e-05,9.26572389630288e-05,9.36184290444721e-05,9.56524364470556e-05,9.576829997999e-05,0.00010152832426414,0.000102012080666186,0.000102270424544992,0.000103738245036822,0.000104419880573126,0.000104419880573126,0.000106542907056487,0.000109806898667642,0.000109869689147884,0.00011029074699955,0.000110932935309605,0.000114204372927374,0.000114991732112792,0.000114991732112792,0.000115846261336727,0.000117852485249383,0.000118183421243769,0.00012156006712873,0.000126971020883826,0.000128441175653722,0.000129484508395731,0.00013008087382792,0.000131097950039772,0.000131140247039288,0.000134031067100502,0.000135386988116344,0.000136000760052562,0.00013625055940298,0.00013625055940298,0.000137475420513959,0.000138414116779604,0.000139479562057233,0.000141004622212631,0.000143582284114948,0.00014604043229652,0.000146208136073322,0.000146411339042098,0.000147482479774546,0.000149247306302936,0.000152660773307381,0.000153012290243653,0.000153012290243653,0.000156008462229315,0.000156008462229315,0.000156008462229315,0.000156008462229315,0.000156359355856181,0.000163430611732428,0.000164292038771502,0.000165085668621215,0.000165168152114885,0.000167226363691294,0.000167226363691294,0.000168886994695254,0.000177297791309392,0.000178316764640627,0.000178433797954329,0.000190682331570162,0.000192687070757544,0.000193559131974841,0.000195110894165679,0.000203310912587882,0.000204531464408526,0.000211560325164765,0.000217326388545758,0.000219183948581824,0.000230470594501009,0.000236295555842173,0.000236295555842173,0.000237969591139817,0.000238895036550011,0.000240091679659707,0.00024504311824387,0.000248894345517406,0.00024922611224628,0.000252520683094018,0.000260305532020938,0.000269556442843776,0.000276279270340279,0.000276450490626429,0.0002888842220528,0.000300793853764762,0.000310526898620031,0.000310526898620031,0.000310526898620031,0.000310526898620031,0.000314014112730081,0.000316231493125525,0.000319875786120725,0.000322286004172811,0.000325852884094908,0.000328686973961957,0.000330159246695624,0.000330159246695624,0.000332230724150338,0.000343372567921015,0.000347725919475348,0.000347725919475348,0.000348083215475225,0.000348893464341216,0.000358915307794637,0.000367587495755669,0.00037119130962883,0.000379180096714797,0.000379506364047676,0.000389855536606445,0.000389855536606445,0.000394005835559014,0.000397049905595627,0.000401862739501142,0.000403068689760216,0.000409871019456021,0.000411190465611308,0.000415154153009676,0.00043835707460955,0.000439984431010136,0.000455725014828071,0.000455725014828071,0.000462451623812605,0.000473799831968836,0.00048005059737986,0.000499976545927934,0.000503997133027838,0.000510582678014153,0.000511237227361571,0.000511237227361571,0.000511237227361571,0.000515074723430338,0.000515074723430338,0.000515074723430338,0.000515074723430338,0.000515074723430338,0.000515074723430338,0.000515074723430338,0.000515074723430338,0.000515074723430338,0.000521064640776528,0.000521693226057053,0.000523313707807709,0.000531620318486389,0.000532338131865921,0.000544489585007781,0.000582112692229055,0.000583156155607579,0.00059604167421073,0.000599500932975011,0.000600904493580627,0.000611455118100673,0.000622709118313067,0.000627243720693274,0.000631797497998463,0.000632763334133644,0.000636478232744869,0.000647127962421218,0.000663008324005997,0.000666527868796071,0.00067436323086411,0.000679992253136195,0.000694838612371794,0.000700171629419311,0.000706370932593978,0.000711470164882021,0.000730549602672995,0.000736018789019068,0.000745378405451344,0.000749853975702274,0.0007651885239052,0.000765883315864581,0.000765883315864581,0.000765883315864581,0.000768926714136876,0.000768926714136876,0.000768926714136876,0.000768926714136876,0.000768926714136876,0.000768926714136876,0.000768926714136876,0.000768926714136876,0.000768926714136876,0.000768926714136876,0.000768926714136876,0.000768926714136876,0.000768926714136876,0.000781280898853279,0.000781280898853279,0.000785685415292428,0.000805160870061813,0.000814126075057501,0.000824367627090263,0.00084065881649825,0.000861187363773457,0.000861187363773457,0.000861187363773457,0.000861187363773457,0.000861187363773457,0.000887191793645514,0.000920420675554205,0.000921574401246309,0.000933342833591141,0.000948359072707851,0.000948359072707851,0.000951657145134769,0.000963696947321175,0.000963696947321175,0.000969186581353916,0.00100418052795365,0.00101762512692085,0.00102020705837401,0.00105421096279176,0.00105752866860062,0.00106509943088975,0.00107136542994638,0.00107136542994638,0.00107136542994638,0.00107136542994638,0.00107136542994638,0.00107278683249426,0.00107278683249426,0.00107361755276507,0.00107361755276507,0.00107361755276507,0.00109268875075425,0.00112145125893653,0.001124824617411,0.00113692322038512,0.00114040931312443,0.00115192334077103,0.00115447424377514,0.00115489346337468,0.00116057928131967,0.00118247052293186,0.00119114912713406,0.00119114912713406,0.00120298113287407,0.0012142152616771,0.00121555650742266,0.00121985568024958,0.00123189572749405,0.00123456631733914,0.0012374799080394,0.00125297964918981,0.00125554969849497,0.00127034612798614,0.00127688700326855,0.00127793981892133,0.00128026915990229,0.00130356994632679,0.00130617172194327,0.00130965847493241,0.00131648616228699,0.00131648616228699,0.00131724677694812,0.00131724677694812,0.00131724677694812,0.00133267758406256,0.00134821382514487,0.00137934293712327,0.00140234917235541,0.00141220567183093,0.00141220567183093,0.00142092766999018,0.00142168113813581,0.00142168113813581,0.00142168113813581,0.00142168113813581,0.00142168113813581,0.00142168113813581,0.00142168113813581,0.00142665228752574,0.00143234798357734,0.00144360851264073,0.00144981777795047,0.00144981777795047,0.00144981777795047,0.00145042428714187,0.00146623042413597,0.00146845198383254,0.00146845198383254,0.00148851681983105,0.00151428900490005,0.00151428900490005,0.00151428900490005,0.00151428900490005,0.00152421215412575,0.00152421215412575,0.00152633112427194,0.00156221851477072,0.00159132780374719,0.00159132780374719,0.00159132780374719,0.00159132780374719,0.00159848891706027,0.00163118516078689,0.00164417143242278,0.00168062366624363,0.00168068536943175,0.0016858907086484,0.0016858907086484,0.00171005153936704,0.00172715529000327,0.001739411647499,0.0017411948602256,0.0017411948602256,0.00176286321149897,0.00176286321149897,0.00177474846062789,0.00177981664449708,0.00180520582868341,0.00181895085280492,0.00181917174104184,0.00181917174104184,0.0018301261135732,0.00184122186937071,0.00186197015425825,0.00189959243890165,0.0019100494056244,0.00193387500795244,0.00193587355101546,0.00194675700618677,0.00197961409107984,0.00197961409107984,0.00198122881533016,0.00198431490428589,0.00200039586928372,0.00200916273592644,0.00204746133995995,0.00205919453487143,0.00206668898132309,0.00206668898132309,0.00206877049708218,0.00207342157350759,0.00211670037219206,0.00212262662972873,0.00214884123450026,0.00214884123450026,0.00218066698717832,0.00219640646278616,0.00219640646278616,0.00220125236013583,0.00220287347858467,0.0022057791387041,0.00221685612624147,0.00222562107992409,0.00224249038072913,0.00225643334757054,0.00226314270369436,0.00226314270369436,0.00226314270369436,0.00226314270369436,0.00226314270369436,0.00226314270369436,0.00226314270369436,0.00226314270369436,0.00227899519377281,0.00228104496745718,0.00228451312721539,0.00238119319255579,0.00242539314491576,0.00242762794138123,0.00244045166011925,0.00246923140037051,0.00249611182842185,0.00251872606874501,0.00260190140375271,0.00262178269037867,0.00262178269037867,0.00264609430992566,0.00268156808963886,0.00268976814645864,0.00270373807017903,0.00272717307300648,0.00273862392062642,0.00274972095255772,0.00275290698208849,0.00275290698208849,0.00275290698208849,0.00275290698208849,0.00275290698208849,0.00278806995248699,0.00279922152522558,0.00281766014556522,0.00282430261475248,0.00282526121729713,0.00282526121729713,0.00287643643778967,0.00300709025644751,0.00302481894420938,0.00303820786633213,0.00304898130205186,0.00309120847307928,0.00314174161528154,0.00317373367454115,0.00323219437726368,0.00326076238615451,0.00328044932125783,0.00328778495208976,0.00328778495208976,0.00328778495208976,0.00328778495208976,0.00328778495208976,0.00328778495208976,0.00336901205987369,0.00337217915534416,0.00342327176877716,0.00342327176877716,0.00344458978640604,0.00345278737512187,0.00345278737512187,0.00345717547099968,0.00346359208601587,0.00348057308544492,0.003493060002422,0.00353504090172744,0.0035612175374663,0.00357026692728064,0.00358871793938886,0.00358871793938886,0.00363136656116953,0.00366218243974382,0.00366546227293245,0.00366689708668665,0.00367830155639297,0.00369925489700503,0.00372149232923522,0.00373523148939455,0.00373523148939455,0.00376549963423704,0.00379774623604272,0.00380347588798139,0.00381554829474771,0.00386710433896715,0.00398654093234813,0.00398740324066346,0.00403686173901565,0.00404551136714311,0.00407329519199998,0.00418953000638867,0.00424969733900356,0.00427820639797512,0.00429169404671496,0.00430174215001519,0.00435458529481761,0.00437817550749912,0.00441434972719683,0.00441434972719683,0.0044197021518554,0.00449020014754859,0.00449020014754859,0.00449020014754859,0.00449020014754859,0.00449020014754859,0.00449020014754859,0.00449020014754859,0.00449463526437013,0.00449515769013711,0.00454305460275453,0.00457393647436753,0.00458327605640361,0.00461343380811475,0.00467313739456009,0.00467317546096722,0.00468844237455377,0.00472084375560414,0.00480511976696614,0.00480511976696614,0.00480655527444672,0.00481041028220699,0.00490325271149107,0.00504522753616945,0.00507706662760243,0.00509033511804359,0.00509033511804359,0.00509847132464058,0.00509847132464058,0.00509861851880206,0.0051269551401388,0.0051269551401388,0.00513189628872998,0.00515641459299392,0.00515641459299392,0.00515641459299392,0.00515641459299392,0.00515641459299392,0.00525350171643152,0.00525375473114471,0.00528214409383213,0.00528214409383213,0.00529565552586691,0.00529617258168976,0.00529617258168976,0.00529618353401555,0.00535716441830985,0.00536719259678175,0.00538190846012079,0.00540239181920782,0.00540239181920782,0.00540239181920782,0.00540239181920782,0.00540239181920782,0.00545998236314256,0.00546319136469853,0.00547873221980846,0.00559486180214993,0.00559486180214993,0.00566296102968956,0.0056724070075468,0.00571656214077367,0.00571698279711669,0.00573551208618415,0.00577675782728016,0.00581694496694182,0.00584018681739951,0.00586509703217975,0.00586509703217975,0.00586509703217975,0.00586509703217975,0.00586509703217975,0.00586509703217975,0.00596843409207045,0.00604234189522638,0.00604234189522638,0.00604234189522638,0.00609327883775844,0.00609768907303612,0.00613114542902692,0.00616569069228971,0.00619513451243217,0.0062021308765667,0.00630590194056378,0.00637856290478225,0.00637856290478225,0.00639665236941283,0.00653516178884308,0.00655473905587048,0.00656527488226217,0.00659577479242243,0.00659577479242243,0.00661560389569141,0.00661560389569141,0.00661560389569141,0.00661560389569141,0.00661560389569141,0.00661560389569141,0.00661560389569141,0.00662915880607748,0.00672573583458832,0.00676149790824719,0.00679503258959634,0.00679503258959634,0.00696855918036136,0.00696855918036136,0.006992965536328,0.00703385452719694,0.00708394697338517,0.00708394697338517,0.00708394697338517,0.00708394697338517,0.00708394697338517,0.00711415577506481,0.00718609601109089,0.00722882333573821,0.00722961268125503,0.00722961268125503,0.00724600864601047,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00724906297623462,0.00732487040480633,0.00732487040480633,0.00740729862041676,0.00740729862041676,0.00740729862041676,0.00740729862041676,0.00740729862041676,0.00740729862041676,0.00745327895144222,0.00745327895144222,0.0075077094599429,0.00763407113603799,0.00771863454236804,0.00771863454236804,0.00771863454236804,0.00771863454236804,0.00774425747134344,0.00783381080137256,0.00783381080137256,0.00783381080137256,0.00783381080137256,0.00784085502087578,0.00790345991468864,0.00790345991468864,0.00797110224616007,0.00801652045352233,0.00802061402889653,0.00819017496860506,0.00822561801817917,0.00822561801817917,0.00822893543142813,0.00822893543142813,0.00822893543142813,0.00823955158273649,0.00823955158273649,0.00827091713562979,0.00827091713562979,0.00827091713562979,0.00836255917393741,0.00846471972084856,0.00849173703191094,0.00849173703191094,0.00849217904585221,0.00849217904585221,0.00852456862418721,0.008572460842941,0.00862877261854099,0.00862877261854099,0.00862877261854099,0.00862877261854099,0.0086437398473424,0.00876087741001407,0.00893231923738992,0.00898148585366457,0.00899007643614804,0.00904334319934677,0.00904334319934677,0.00904334319934677,0.00904334319934677,0.00904334319934677,0.00904334319934677,0.00910953277446355,0.0091117400323065,0.0091117400323065,0.0091117400323065,0.0091117400323065,0.0091117400323065,0.0091117400323065,0.0091117400323065,0.00911658977573771,0.00925447440451576,0.00946939499548567,0.00957065545674324,0.00958393552037128,0.00958393552037128,0.00960013470893817,0.00972638653307142,0.00989108258586307,0.0100232480264267,0.0100232480264267,0.0100232480264267,0.0100232480264267,0.0100232480264267,0.0100839085174977,0.0101530710439689,0.0101876972621621,0.0102883414676033,0.0103555014696679,0.0103561867049227,0.0103561867049227,0.0103561867049227,0.0103660499646395,0.0103751708135871,0.0104111018249184,0.0104900192109525,0.0105293201950191,0.0105476767023305,0.0106440674577466,0.0106440674577466,0.0107211125851304,0.0107509357902673,0.0107810152318196,0.0108170407878658,0.0108170407878658,0.0108355939645269,0.0109734663649917,0.0109734663649917,0.0109734663649917,0.0109734663649917,0.0109734663649917,0.0109734663649917,0.0109734663649917,0.0110248508516539,0.0110445576085252,0.0111937231273099,0.0112397614680484,0.0112896045359369,0.0113894806944134,0.0114396907757966,0.0115131124656799,0.0115378698021663,0.0117127909915164,0.0119617925260185,0.0119617925260185,0.0119617925260185,0.0119617925260185,0.0119617925260185,0.0119617925260185,0.0121171496360805,0.0122400302172799,0.012375434533946,0.0124259874616422,0.0124259874616422,0.0124865658705626,0.0125396190378213,0.0126271785286632,0.0126507543677567,0.0127693383528709,0.0129031618219853,0.0129876306017457,0.0129876306017457,0.0129876306017457,0.0129876306017457,0.0129876306017457,0.0129876306017457,0.0129876306017457,0.0129876306017457,0.0129876306017457,0.0129876306017457,0.0129876306017457,0.0131385961624431,0.0131819980223674,0.0132208181210815,0.0132334944151413,0.0132978963650794,0.0132978963650794,0.0134459181679702,0.0134740966125472,0.013722988608943,0.0137640831294771,0.0138296876127166,0.0138296876127166,0.0138296876127166,0.0138355723229471,0.0139419836113937,0.0139628244723162,0.0140038835380717,0.0140038835380717,0.0140503912353006,0.0140503912353006,0.0140503912353006,0.0140503912353006,0.0140503912353006,0.0141361266470447,0.0141377515980034,0.0142031568604577,0.0142031568604577,0.014288624920078,0.014288624920078,0.014288624920078,0.0143511562294318,0.0143511562294318,0.0143677224939886,0.0143720092208552,0.0143734397128988,0.0143734397128988,0.0143734397128988,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.0144458734596428,0.01454217829181,0.0145768827126363,0.0145768827126363,0.0145768827126363,0.0145768827126363,0.0145772345032558,0.0145772345032558,0.0147638064893341,0.0148519858898923,0.0149291845280249,0.0149883683763268,0.0150218496126209,0.0150231316735927,0.0151366051172315,0.0151494915579278,0.0151494915579278,0.0151494915579278,0.0151494915579278,0.0151494915579278,0.0151494915579278,0.0151494915579278,0.0151494915579278,0.0151494915579278,0.0153081943931694,0.0153428849354822,0.0154230276797302,0.015466486653987,0.015466486653987,0.0155862670536561,0.0157352229037008,0.0157434193347006,0.0157618513374548,0.0158341658089186,0.0158451993091636,0.0159211290409986,0.0160767661834285,0.0160767661834285,0.0160767661834285,0.0161075348623519,0.0161669900763219,0.0161802358597272,0.0162843551267764,0.0162843551267764,0.0162843551267764,0.0162843551267764,0.0165105863259973,0.0165387214743837,0.0165387214743837,0.0165387214743837,0.0166686533251051,0.0169354580473263,0.0169499396150211,0.0169935615015715,0.0171518379415667,0.0172726338802987,0.0172726338802987,0.0173522686048632,0.0173675075568315,0.0174091019460889,0.0174544118632387,0.0174544118632387,0.0174544118632387,0.0174544118632387,0.0174544118632387,0.0174544118632387,0.0177378792750744,0.0177914523774964,0.0178887265753469,0.0178887265753469,0.0178887265753469,0.0182790186684267,0.0183498816391606,0.0185169476508629,0.018658577586646,0.018658577586646,0.0186590979918391,0.0186590979918391,0.0186590979918391,0.0191573109074616,0.0191573109074616,0.0192366109983309,0.0192407179531623,0.0192689668561265,0.0193846665764037,0.0193962776737577,0.0197342303253351,0.0197379411966404,0.0197379411966404,0.0198978559796643,0.0198978559796643,0.0198978559796643,0.0198978559796643,0.0198978559796643,0.0198978559796643,0.0199251981433933,0.0199251981433933,0.0200954188332531,0.0200954188332531,0.0201494795709251,0.0204571170353316,0.0204745072365346,0.0206239226646813,0.020666213217175,0.020666213217175,0.020823053104626,0.0208437612889704,0.0209397514562406,0.0209397514562406,0.0211513561134094,0.0211513561134094,0.0211701344763341,0.0211701344763341,0.0211701344763341,0.0211701344763341,0.0211701344763341,0.0211701344763341,0.0211943876100775,0.0212053897870244,0.021505904324018,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0215908059583495,0.0217307433140347,0.0218403788674712,0.021903878935417,0.0223871035756686,0.0223871035756686,0.0224753882545067,0.0224753882545067,0.0224753882545067,0.0224753882545067,0.0224753882545067,0.0225415777644284,0.0225415777644284,0.0225415777644284,0.0225415777644284,0.0225415777644284,0.022782974691688,0.0228601596686816,0.0228745934230713,0.0228827883996568,0.0231086021734519,0.0231252237709042,0.0232549528919532,0.0233105732687599,0.02335395505002,0.0235046667458385,0.0237541942935355,0.0238130781509119,0.0238130781509119,0.0238130781509119,0.0238130781509119,0.0238130781509119,0.0238130781509119,0.0238130781509119,0.023894124392847,0.023894124392847,0.0239805022013376,0.0239805022013376,0.0244098116366885,0.0244499685936002,0.024623859572197,0.0248719576165271,0.0248719576165271,0.0249322013312997,0.0251514192085359,0.0251797811199377,0.0251826710079125,0.0251826710079125,0.0251826710079125,0.0251826710079125,0.0251826710079125,0.0251826710079125,0.0251826710079125,0.0254681047349984,0.0254681047349984,0.0255507949208394,0.0257890878868048,0.0259711865902777,0.0261968143399192,0.0262301435471103,0.0262301435471103,0.0262301435471103,0.0262301435471103,0.02629639349579,0.0264735477714334,0.0265558875679864,0.0265836396155846,0.0265836396155846,0.0265836396155846,0.0265836396155846,0.0265836396155846,0.0265836396155846,0.0265836396155846,0.0265836396155846,0.0265836396155846,0.0265836396155846,0.0265836396155846,0.0268316337950831,0.0268402881735434,0.0270043277962299,0.0270043277962299,0.0270043277962299,0.0270043277962299,0.0270891027302561,0.0270891027302561,0.0273563959811583,0.0277906453573426,0.0280154626543151,0.0280154626543151,0.0280154626543151,0.0280154626543151,0.0280154626543151,0.0280154626543151,0.0280154626543151,0.0282081634881494,0.028327311403144,0.0284561012502719,0.0286021992518316,0.0286021992518316,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0286842323115926,0.0287273500118199,0.0287778929864493,0.0287778929864493,0.0287778929864493,0.0287778929864493,0.0290561359459691,0.029354437943826,0.0293996224638722,0.0294776246379113,0.0294776246379113,0.0294776246379113,0.0294776246379113,0.0294776246379113,0.0295146149562657,0.0299970825836586,0.0299970825836586,0.0302222484208145,0.0302222484208145,0.0303388435454134,0.0304194924373924,0.0304452461797001,0.0309174208578073,0.0309696158572186,0.0309696158572186,0.0309696158572186,0.0309696158572186,0.031018162956,0.0310569405870559,0.0310569405870559,0.0310569405870559,0.0310569405870559,0.0311250748156971,0.0311250748156971,0.0313941827814213,0.0316287350010617,0.031903677721915,0.0323397659677883,0.0324909323242414,0.0324909323242414,0.0324909323242414,0.0324909323242414,0.0324909323242414,0.0324909323242414,0.0324909323242414,0.0325985823708397,0.032762436871208,0.032762436871208,0.0328520963861201,0.0329574177100842,0.033581950427879,0.0336331934026357,0.0336331934026357,0.0338471616904718,0.0340410757167629,0.0340410757167629,0.0340410757167629,0.0340410757167629,0.0340410757167629,0.0340410757167629,0.0340410757167629,0.0341296633330337,0.0345159210406855,0.0348516857059266,0.0348516857059266,0.0354105919010551,0.0354105919010551,0.0355279873149461,0.0356195533234608,0.0356195533234608,0.0356195533234608,0.0356195533234608,0.0356195533234608,0.0356195533234608,0.0356195533234608,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0357265217086331,0.0358931222443438,0.0361490195584971,0.0361490195584971,0.0361739758998601,0.0363171765246021,0.0365554752910297,0.0366153845742441,0.0366173810643618,0.0368080482346072,0.0372258779895127,0.0372258779895127,0.0372258779895127,0.0372258779895127,0.0372356439108256,0.0372356439108256,0.0372450916756214,0.0373397520192765,0.0373397520192765,0.0373925149915278,0.0374739901003922,0.0381468484325907,0.0381468484325907,0.0381659615508857,0.0381659615508857,0.0382457870741655,0.038711089802331,0.0388266259199297,0.0388266259199297,0.0388595680626892,0.0388595680626892,0.038907779936331,0.0393081475474591,0.0395133246669108,0.0400620102103778,0.0400620102103778,0.0400620102103778,0.0400620102103778,0.0400620102103778,0.0400620102103778,0.0401586086788511,0.0401646723854739,0.0405201473399286,0.0405201473399286,0.0405201473399286,0.0405201473399286,0.0405201473399286,0.0405795616978378,0.0407710757218526,0.0409365364326801,0.0410005379885452,0.0413367000400589,0.041614960828194,0.0419070724328378,0.0419070724328378,0.0420050334916604,0.0422071450143902,0.0422071450143902,0.0422071450143902,0.0422117805098012,0.0423293541094285,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0427180407075324,0.0430506705452367,0.0431937207726926,0.0436900421552692,0.0439200956229825,0.0439200956229825,0.0439200956229825,0.0439200956229825,0.0439200956229825,0.0439200956229825,0.0439200956229825,0.0439200956229825,0.0439200956229825,0.0439200956229825,0.04399472028728,0.0439996936534209,0.0442361439017385,0.0445140670491006,0.0445140670491006,0.0445140670491006,0.0450069594813136,0.0450069594813136,0.0452561428143917,0.0454292922207489,0.0456585389943611,0.0460307383720275,0.0460307383720275,0.0460307383720275,0.04642348398967,0.0467610344972808,0.0472548676681886,0.0474220201973924,0.0474220201973924,0.0474220201973924,0.0474220201973924,0.0475238428357407,0.0477896992597716,0.0481127353539143,0.0481127353539143,0.0481127353539143,0.0482935535574725,0.0483183603234409,0.0491235409794699,0.0492100894900811,0.0492100894900811,0.0492100894900811,0.0493798349324989,0.0493865303744239,0.0496170978994185,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0496591532537966,0.0497560211149813,0.0498536611194138,0.0499260467465825,0.0499260467465825]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Term<\/th>\n      <th>Ont<\/th>\n      <th>N<\/th>\n      <th>DE<\/th>\n      <th>P.DE<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"targets":5,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 9, 3, \",\", \".\");\n  }"},{"className":"dt-right","targets":[3,4,5]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":["options.columnDefs.0.render"],"jsHooks":[]}</script>
</div>
<p>Figure <a href="#fig:scatterplot-and-ma-plot">7</a> plots the relationship between UMI counts and read counts for all genes and samples. There are a few notable points about this figure:</p>
<ol type="1">
<li>The UMI counts and read counts are very strongly associated.</li>
<li>The log-fold change between the read counts and the UMI counts can be large, indicating the UMI counts are likely removing substantial PCR amplification noise.</li>
<li>The log-fold changes between the read counts and the UMI counts are larger on average for more highly expressed genes. The variation in the log-fold change is larger for the lowly expressed genes.</li>
<li>The hemoglobin genes stand out. These genes are extremely highly expressed, meaning that they likely saturate the 6 bp UMIs and which leads to the strange relationship between the UMI counts and the read counts for these genes.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>x</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log2</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/vector.html'>as.vector</a></span><span class='op'>(</span><span class='fu'>assay</span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"UMI_counts"</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> <span class='fl'>1</span><span class='op'>)</span>
<span class='va'>y</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log2</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/vector.html'>as.vector</a></span><span class='op'>(</span><span class='fu'>assay</span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"read_counts"</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> <span class='fl'>1</span><span class='op'>)</span>
<span class='va'>m</span> <span class='op'>&lt;-</span> <span class='op'>(</span><span class='va'>x</span> <span class='op'>+</span> <span class='va'>y</span><span class='op'>)</span> <span class='op'>/</span> <span class='fl'>2</span>
<span class='va'>d</span> <span class='op'>&lt;-</span> <span class='va'>y</span> <span class='op'>-</span> <span class='va'>x</span>
<span class='va'>g</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/rep.html'>rep</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span>, <span class='fu'><a href='https://rdrr.io/r/base/nrow.html'>ncol</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>)</span>
<span class='co'># NOTE: Only showing unique points for plotting speed.</span>
<span class='va'>i</span> <span class='op'>&lt;-</span> <span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/duplicated.html'>duplicated</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='va'>x</span>, <span class='st'>"."</span>, <span class='va'>y</span><span class='op'>)</span><span class='op'>)</span>

<span class='va'>X</span> <span class='op'>&lt;-</span> <span class='fu'>edgeR</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cpm.html'>cpm</a></span><span class='op'>(</span><span class='fu'>assay</span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"UMI_counts"</span><span class='op'>)</span>, log <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
<span class='va'>Y</span> <span class='op'>&lt;-</span> <span class='fu'>edgeR</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cpm.html'>cpm</a></span><span class='op'>(</span><span class='fu'>assay</span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"read_counts"</span><span class='op'>)</span>, log <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
<span class='va'>M</span> <span class='op'>&lt;-</span> <span class='op'>(</span><span class='va'>X</span> <span class='op'>+</span> <span class='va'>Y</span><span class='op'>)</span> <span class='op'>/</span> <span class='fl'>2</span>
<span class='va'>D</span> <span class='op'>&lt;-</span> <span class='va'>y</span> <span class='op'>-</span> <span class='va'>x</span>
<span class='va'>G</span> <span class='op'>&lt;-</span> <span class='va'>g</span>
<span class='co'># NOTE: Only showing unique points for plotting speed.</span>
<span class='va'>I</span> <span class='op'>&lt;-</span> <span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/duplicated.html'>duplicated</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='va'>x</span>, <span class='st'>"."</span>, <span class='va'>y</span><span class='op'>)</span><span class='op'>)</span>

<span class='fu'><a href='https://rdrr.io/r/graphics/par.html'>par</a></span><span class='op'>(</span>mfrow <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>2</span>, <span class='fl'>2</span><span class='op'>)</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/pkg/limma/man/plotWithHighlights.html'>plotWithHighlights</a></span><span class='op'>(</span>
  x <span class='op'>=</span> <span class='va'>x</span><span class='op'>[</span><span class='va'>i</span><span class='op'>]</span>,
  y <span class='op'>=</span> <span class='va'>y</span><span class='op'>[</span><span class='va'>i</span><span class='op'>]</span>,
  status <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/ifelse.html'>ifelse</a></span><span class='op'>(</span><span class='va'>g</span><span class='op'>[</span><span class='va'>i</span><span class='op'>]</span> <span class='op'>%in%</span> <span class='va'>ribo_set</span>, <span class='fl'>1</span>, <span class='fu'><a href='https://rdrr.io/r/base/ifelse.html'>ifelse</a></span><span class='op'>(</span><span class='va'>g</span><span class='op'>[</span><span class='va'>i</span><span class='op'>]</span> <span class='op'>%in%</span> <span class='va'>mito_set</span>, <span class='op'>-</span><span class='fl'>1</span>, <span class='fl'>0</span><span class='op'>)</span><span class='op'>)</span>,
  legend <span class='op'>=</span> <span class='cn'>FALSE</span>,
  bg.pch <span class='op'>=</span> <span class='st'>"."</span>,
  hl.pch <span class='op'>=</span> <span class='st'>"."</span>,
  hl.col <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"orange"</span>, <span class='st'>"dodgerBlue"</span><span class='op'>)</span>,
  hl.cex <span class='op'>=</span> <span class='fl'>3</span>,
  sub <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='st'>"r = "</span>, <span class='fu'><a href='https://scales.r-lib.org/reference/number.html'>number</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/stats/cor.html'>cor</a></span><span class='op'>(</span><span class='va'>x</span>, <span class='va'>y</span><span class='op'>)</span>, accuracy <span class='op'>=</span> <span class='fl'>0.001</span><span class='op'>)</span><span class='op'>)</span>,
  xlab <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/expression.html'>expression</a></span><span class='op'>(</span><span class='va'>log</span><span class='op'>[</span><span class='fl'>2</span><span class='op'>]</span><span class='op'>(</span><span class='va'>UMIs</span> <span class='op'>+</span> <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span>,
  ylab <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/expression.html'>expression</a></span><span class='op'>(</span><span class='va'>log</span><span class='op'>[</span><span class='fl'>2</span><span class='op'>]</span><span class='op'>(</span><span class='va'>reads</span> <span class='op'>+</span> <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/graphics/abline.html'>abline</a></span><span class='op'>(</span>a <span class='op'>=</span> <span class='fl'>0</span>, b <span class='op'>=</span> <span class='fl'>1</span>, lty <span class='op'>=</span> <span class='fl'>2</span>, col <span class='op'>=</span> <span class='st'>"grey"</span><span class='op'>)</span>

<span class='fu'><a href='https://rdrr.io/pkg/limma/man/plotWithHighlights.html'>plotWithHighlights</a></span><span class='op'>(</span>
  x <span class='op'>=</span> <span class='va'>m</span><span class='op'>[</span><span class='va'>i</span><span class='op'>]</span> <span class='op'>/</span> <span class='fl'>2</span>,
  y <span class='op'>=</span> <span class='va'>d</span><span class='op'>[</span><span class='va'>i</span><span class='op'>]</span>,
  pch <span class='op'>=</span> <span class='st'>"."</span>,
  status <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/ifelse.html'>ifelse</a></span><span class='op'>(</span><span class='va'>g</span><span class='op'>[</span><span class='va'>i</span><span class='op'>]</span> <span class='op'>%in%</span> <span class='va'>ribo_set</span>, <span class='fl'>1</span>, <span class='fu'><a href='https://rdrr.io/r/base/ifelse.html'>ifelse</a></span><span class='op'>(</span><span class='va'>g</span><span class='op'>[</span><span class='va'>i</span><span class='op'>]</span> <span class='op'>%in%</span> <span class='va'>mito_set</span>, <span class='op'>-</span><span class='fl'>1</span>, <span class='fl'>0</span><span class='op'>)</span><span class='op'>)</span>,
  legend <span class='op'>=</span> <span class='cn'>FALSE</span>,
  bg.pch <span class='op'>=</span> <span class='st'>"."</span>,
  hl.pch <span class='op'>=</span> <span class='st'>"."</span>,
  hl.col <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"orange"</span>, <span class='st'>"dodgerBlue"</span><span class='op'>)</span>,
  hl.cex <span class='op'>=</span> <span class='fl'>3</span>,
  xlab <span class='op'>=</span> <span class='st'>"Average log-(count+1)"</span>,
  ylab <span class='op'>=</span> <span class='st'>"logFC (reads vs. UMIs)"</span>,
  ylim <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='op'>-</span><span class='fu'><a href='https://rdrr.io/r/base/Extremes.html'>max</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/MathFun.html'>abs</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='va'>d</span>, <span class='va'>D</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>, <span class='fu'><a href='https://rdrr.io/r/base/Extremes.html'>max</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/MathFun.html'>abs</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='va'>d</span>, <span class='va'>D</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/graphics/abline.html'>abline</a></span><span class='op'>(</span>h <span class='op'>=</span> <span class='fl'>0</span>, lty <span class='op'>=</span> <span class='fl'>2</span>, col <span class='op'>=</span> <span class='st'>"grey"</span><span class='op'>)</span>
<span class='va'>l</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/lowess.html'>lowess</a></span><span class='op'>(</span><span class='va'>m</span>, <span class='va'>d</span>, f <span class='op'>=</span> <span class='fl'>0.3</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/graphics/lines.html'>lines</a></span><span class='op'>(</span><span class='va'>l</span>, col <span class='op'>=</span> <span class='st'>"red"</span>, lwd <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span>

<span class='fu'><a href='https://rdrr.io/pkg/limma/man/plotWithHighlights.html'>plotWithHighlights</a></span><span class='op'>(</span>
  x <span class='op'>=</span> <span class='va'>X</span><span class='op'>[</span><span class='va'>I</span><span class='op'>]</span>,
  y <span class='op'>=</span> <span class='va'>Y</span><span class='op'>[</span><span class='va'>I</span><span class='op'>]</span>,
  status <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/ifelse.html'>ifelse</a></span><span class='op'>(</span><span class='va'>G</span><span class='op'>[</span><span class='va'>I</span><span class='op'>]</span> <span class='op'>%in%</span> <span class='va'>ribo_set</span>, <span class='fl'>1</span>, <span class='fu'><a href='https://rdrr.io/r/base/ifelse.html'>ifelse</a></span><span class='op'>(</span><span class='va'>G</span><span class='op'>[</span><span class='va'>I</span><span class='op'>]</span> <span class='op'>%in%</span> <span class='va'>mito_set</span>, <span class='op'>-</span><span class='fl'>1</span>, <span class='fl'>0</span><span class='op'>)</span><span class='op'>)</span>,
  legend <span class='op'>=</span> <span class='cn'>FALSE</span>,
  bg.pch <span class='op'>=</span> <span class='st'>"."</span>,
  hl.pch <span class='op'>=</span> <span class='st'>"."</span>,
  hl.col <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"orange"</span>, <span class='st'>"dodgerBlue"</span><span class='op'>)</span>,
  hl.cex <span class='op'>=</span> <span class='fl'>3</span>,
  sub <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span>
    <span class='st'>"r = "</span>, 
    <span class='fu'><a href='https://scales.r-lib.org/reference/number.html'>number</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/stats/cor.html'>cor</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/vector.html'>as.vector</a></span><span class='op'>(</span><span class='va'>X</span><span class='op'>)</span>, <span class='fu'><a href='https://rdrr.io/r/base/vector.html'>as.vector</a></span><span class='op'>(</span><span class='va'>Y</span><span class='op'>)</span><span class='op'>)</span>, accuracy <span class='op'>=</span> <span class='fl'>0.001</span><span class='op'>)</span><span class='op'>)</span>,
  xlab <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/expression.html'>expression</a></span><span class='op'>(</span><span class='fu'>logCPM</span><span class='op'>(</span><span class='va'>UMIs</span><span class='op'>)</span><span class='op'>)</span>,
  ylab <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/expression.html'>expression</a></span><span class='op'>(</span><span class='fu'>logCPM</span><span class='op'>(</span><span class='va'>reads</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/graphics/abline.html'>abline</a></span><span class='op'>(</span>a <span class='op'>=</span> <span class='fl'>0</span>, b <span class='op'>=</span> <span class='fl'>1</span>, lty <span class='op'>=</span> <span class='fl'>2</span>, col <span class='op'>=</span> <span class='st'>"grey"</span><span class='op'>)</span>

<span class='fu'><a href='https://rdrr.io/pkg/limma/man/plotWithHighlights.html'>plotWithHighlights</a></span><span class='op'>(</span>
  x <span class='op'>=</span> <span class='va'>M</span><span class='op'>[</span><span class='va'>I</span><span class='op'>]</span>,
  y <span class='op'>=</span> <span class='va'>D</span><span class='op'>[</span><span class='va'>i</span><span class='op'>]</span>,
  status <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/ifelse.html'>ifelse</a></span><span class='op'>(</span><span class='va'>G</span><span class='op'>[</span><span class='va'>I</span><span class='op'>]</span> <span class='op'>%in%</span> <span class='va'>ribo_set</span>, <span class='fl'>1</span>, <span class='fu'><a href='https://rdrr.io/r/base/ifelse.html'>ifelse</a></span><span class='op'>(</span><span class='va'>G</span><span class='op'>[</span><span class='va'>I</span><span class='op'>]</span> <span class='op'>%in%</span> <span class='va'>mito_set</span>, <span class='op'>-</span><span class='fl'>1</span>, <span class='fl'>0</span><span class='op'>)</span><span class='op'>)</span>,
  legend <span class='op'>=</span> <span class='cn'>FALSE</span>,
  bg.pch <span class='op'>=</span> <span class='st'>"."</span>,
  hl.pch <span class='op'>=</span> <span class='st'>"."</span>,
  hl.col <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"orange"</span>, <span class='st'>"dodgerBlue"</span><span class='op'>)</span>,
  hl.cex <span class='op'>=</span> <span class='fl'>3</span>,
  xlab <span class='op'>=</span> <span class='st'>"Average logCPM"</span>,
  ylab <span class='op'>=</span> <span class='st'>"logFC (reads vs. UMIs)"</span>,
  ylim <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='op'>-</span><span class='fu'><a href='https://rdrr.io/r/base/Extremes.html'>max</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/MathFun.html'>abs</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='va'>d</span>, <span class='va'>D</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>, <span class='fu'><a href='https://rdrr.io/r/base/Extremes.html'>max</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/MathFun.html'>abs</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='va'>d</span>, <span class='va'>D</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/graphics/abline.html'>abline</a></span><span class='op'>(</span>h <span class='op'>=</span> <span class='fl'>0</span>, lty <span class='op'>=</span> <span class='fl'>2</span>, col <span class='op'>=</span> <span class='st'>"grey"</span><span class='op'>)</span>
<span class='va'>l</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/lowess.html'>lowess</a></span><span class='op'>(</span><span class='va'>M</span>, <span class='va'>D</span>, f <span class='op'>=</span> <span class='fl'>0.3</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/graphics/lines.html'>lines</a></span><span class='op'>(</span><span class='va'>l</span>, col <span class='op'>=</span> <span class='st'>"red"</span>, lwd <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:scatterplot-and-ma-plot"></span>
<img src="C086_Kinkel.preprocess_files/scatterplot-and-ma-plot-1.png" alt="Scatterplot of the logarithm of the UMI and read counts (top) and logCPM of the UMI and read counts (bottom) for all genes and samples (left) and mean-difference plot of the same data (right). The correlation in the scatterplot is reported below the x-axis label. Genes that are ribosomal protein genes subunits are highlighted in orange and mitochondrial genes in blue. The red line is a trend fitted to the mean-differences." width="624" />
<p class="caption">
Figure 7: Scatterplot of the logarithm of the UMI and read counts (top) and logCPM of the UMI and read counts (bottom) for all genes and samples (left) and mean-difference plot of the same data (right). The correlation in the scatterplot is reported below the x-axis label. Genes that are ribosomal protein genes subunits are highlighted in orange and mitochondrial genes in blue. The red line is a trend fitted to the mean-differences.
</p>
</div>
</div>
<p>Rather than looking at all samples and all genes at once, can also examine the relationship between UMI counts and read counts within a single sample across genes (Figure <a href="#fig:within-sample-cors">8</a>) and within a single gene across samples (Figure <a href="#fig:within-gene-cors">9</a>).</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>X</span> <span class='op'>&lt;-</span> <span class='fu'>edgeR</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cpm.html'>cpm</a></span><span class='op'>(</span><span class='fu'>assay</span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"UMI_counts"</span><span class='op'>)</span>, log <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
<span class='va'>Y</span> <span class='op'>&lt;-</span> <span class='fu'>edgeR</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cpm.html'>cpm</a></span><span class='op'>(</span><span class='fu'>assay</span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"read_counts"</span><span class='op'>)</span>, log <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
<span class='va'>cor_j</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/diag.html'>diag</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/stats/cor.html'>cor</a></span><span class='op'>(</span><span class='va'>X</span>, <span class='va'>Y</span><span class='op'>)</span><span class='op'>)</span>
<span class='co'># NOTE: Splitting up the cor() computations is a hack to improve speed whilst </span>
<span class='co'>#       keeping memory usage somewhat under control.</span>
<span class='va'>b</span> <span class='op'>&lt;-</span> <span class='fl'>1000</span>
<span class='va'>s</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span><span class='op'>(</span><span class='fl'>1</span>, by <span class='op'>=</span> <span class='va'>b</span>, to <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/nrow.html'>nrow</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>e</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/Extremes.html'>pmin</a></span><span class='op'>(</span><span class='va'>s</span> <span class='op'>+</span> <span class='va'>b</span> <span class='op'>-</span> <span class='fl'>1</span>, <span class='fu'><a href='https://rdrr.io/r/base/nrow.html'>nrow</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>cor_i</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/unlist.html'>unlist</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/base/funprog.html'>Map</a></span><span class='op'>(</span>
    f <span class='op'>=</span> <span class='kw'>function</span><span class='op'>(</span><span class='va'>ss</span>, <span class='va'>ee</span> <span class='op'>)</span> <span class='fu'><a href='https://rdrr.io/r/base/diag.html'>diag</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/stats/cor.html'>cor</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/t.html'>t</a></span><span class='op'>(</span><span class='va'>X</span><span class='op'>[</span><span class='va'>ss</span><span class='op'>:</span><span class='va'>ee</span>, <span class='op'>]</span><span class='op'>)</span>, <span class='fu'><a href='https://rdrr.io/r/base/t.html'>t</a></span><span class='op'>(</span><span class='va'>Y</span><span class='op'>[</span><span class='va'>ss</span><span class='op'>:</span><span class='va'>ee</span>, <span class='op'>]</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>,
    ss <span class='op'>=</span> <span class='va'>s</span>,
    ee <span class='op'>=</span> <span class='va'>e</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>These figures show that the UMI counts and read counts are highly correlated across both samples (median = 0.916, median absolute deviation = 0.006) and genes (median = 0.808, median absolute deviation = 0.208).</p>
<aside>
The genes with the lowest correlations include the aforementioned highly-expressed hemoglobin subunits.
</aside>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/r/graphics/par.html'>par</a></span><span class='op'>(</span>mfrow <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>3</span>, <span class='fl'>3</span><span class='op'>)</span>, cex <span class='op'>=</span> <span class='fl'>0.5</span><span class='op'>)</span>
<span class='va'>jj</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/sort.html'>sort</a></span><span class='op'>(</span><span class='va'>cor_j</span><span class='op'>)</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span><span class='op'>:</span><span class='fl'>3</span>, <span class='fl'>17</span><span class='op'>:</span><span class='fl'>18</span>, <span class='fl'>33</span><span class='op'>:</span><span class='fl'>35</span><span class='op'>)</span><span class='op'>]</span><span class='op'>)</span>
<span class='kw'>for</span> <span class='op'>(</span><span class='va'>j</span> <span class='kw'>in</span> <span class='va'>jj</span><span class='op'>)</span> <span class='op'>{</span>
  <span class='va'>g</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span>
  <span class='fu'><a href='https://rdrr.io/r/graphics/plot.default.html'>plot</a></span><span class='op'>(</span>
    x <span class='op'>=</span> <span class='va'>X</span><span class='op'>[</span>, <span class='va'>j</span><span class='op'>]</span>,
    y <span class='op'>=</span> <span class='va'>Y</span><span class='op'>[</span>, <span class='va'>j</span><span class='op'>]</span>,
    pch <span class='op'>=</span> <span class='st'>"."</span>,
    cex <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/ifelse.html'>ifelse</a></span><span class='op'>(</span><span class='va'>g</span> <span class='op'>%in%</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='va'>ribo_set</span>, <span class='va'>mito_set</span><span class='op'>)</span>, <span class='fl'>5</span>, <span class='fl'>1</span><span class='op'>)</span>,
    col <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/ifelse.html'>ifelse</a></span><span class='op'>(</span>
      <span class='va'>g</span> <span class='op'>%in%</span> <span class='va'>ribo_set</span>, <span class='st'>"orange"</span>,
      <span class='fu'><a href='https://rdrr.io/r/base/ifelse.html'>ifelse</a></span><span class='op'>(</span><span class='va'>g</span> <span class='op'>%in%</span> <span class='va'>mito_set</span>, <span class='st'>"dodgerBlue"</span>, <span class='st'>"black"</span><span class='op'>)</span><span class='op'>)</span>,
    sub <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='st'>"r = "</span>, <span class='fu'><a href='https://scales.r-lib.org/reference/number.html'>number</a></span><span class='op'>(</span><span class='va'>cor_j</span><span class='op'>[</span><span class='va'>j</span><span class='op'>]</span>, accuracy <span class='op'>=</span> <span class='fl'>0.001</span><span class='op'>)</span><span class='op'>)</span>,
    xlab <span class='op'>=</span> <span class='st'>"logCPM(UMIs)"</span>,
    ylab <span class='op'>=</span> <span class='st'>"logCPM(reads)"</span>,
    main <span class='op'>=</span> <span class='va'>j</span>,
    xlim <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/range.html'>range</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span>,
    ylim <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/range.html'>range</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>)</span><span class='op'>)</span>
  <span class='fu'><a href='https://rdrr.io/r/graphics/abline.html'>abline</a></span><span class='op'>(</span>a <span class='op'>=</span> <span class='fl'>0</span>, b <span class='op'>=</span> <span class='fl'>1</span>, col <span class='op'>=</span> <span class='st'>"grey"</span>, lty <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span>
<span class='op'>}</span>
<span class='fu'><a href='https://rdrr.io/r/graphics/plot.default.html'>plot</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/stats/density.html'>density</a></span><span class='op'>(</span><span class='va'>cor_j</span>, na.rm <span class='op'>=</span> <span class='cn'>TRUE</span>, from <span class='op'>=</span> <span class='op'>-</span><span class='fl'>1</span>, to <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span>,
  main <span class='op'>=</span> <span class='st'>"All samples"</span>, 
  xlab <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/expression.html'>expression</a></span><span class='op'>(</span><span class='va'>r</span><span class='op'>[</span><span class='va'>p</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:within-sample-cors"></span>
<img src="C086_Kinkel.preprocess_files/within-sample-cors-1.png" alt="Association between UMI counts and read counts for each gene within each sample. The first 8 panels are scatter plots of the UMI counts and read counts for each gene for the samples with the 3 lowest correlations, 2 middlemost correlations, and 3 highest correlations. The correlation in the scatterplot is reported below each panel. Genes that are ribosomal protein genes subunits are highlighted in orange and mitochondrial genes in blue. The final panel is the distribution of these correlations across all samples." width="624" />
<p class="caption">
Figure 8: Association between UMI counts and read counts for each gene within each sample. The first 8 panels are scatter plots of the UMI counts and read counts for each gene for the samples with the 3 lowest correlations, 2 middlemost correlations, and 3 highest correlations. The correlation in the scatterplot is reported below each panel. Genes that are ribosomal protein genes subunits are highlighted in orange and mitochondrial genes in blue. The final panel is the distribution of these correlations across all samples.
</p>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/r/graphics/par.html'>par</a></span><span class='op'>(</span>mfrow <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>3</span>, <span class='fl'>3</span><span class='op'>)</span>, cex <span class='op'>=</span> <span class='fl'>0.5</span><span class='op'>)</span>
<span class='co'># NOTE: There are some NA correlations, so this changes ii.</span>
<span class='va'>ii</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/sort.html'>sort</a></span><span class='op'>(</span><span class='va'>cor_i</span><span class='op'>)</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span><span class='op'>:</span><span class='fl'>3</span>, <span class='fl'>12792</span><span class='op'>:</span><span class='fl'>12793</span>, <span class='fl'>25585</span><span class='op'>:</span><span class='fl'>25587</span><span class='op'>)</span><span class='op'>]</span><span class='op'>)</span>
<span class='kw'>for</span> <span class='op'>(</span><span class='va'>i</span> <span class='kw'>in</span> <span class='va'>ii</span><span class='op'>)</span> <span class='op'>{</span>
  <span class='va'>g</span> <span class='op'>&lt;-</span> <span class='va'>i</span>
  <span class='fu'><a href='https://rdrr.io/r/graphics/plot.default.html'>plot</a></span><span class='op'>(</span>
    x <span class='op'>=</span> <span class='va'>X</span><span class='op'>[</span><span class='va'>i</span>, <span class='op'>]</span>,
    y <span class='op'>=</span> <span class='va'>Y</span><span class='op'>[</span><span class='va'>i</span>, <span class='op'>]</span>,
    pch <span class='op'>=</span> <span class='fl'>16</span>,
    col <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/ifelse.html'>ifelse</a></span><span class='op'>(</span>
      <span class='va'>g</span> <span class='op'>%in%</span> <span class='va'>ribo_set</span>, <span class='st'>"orange"</span>,
      <span class='fu'><a href='https://rdrr.io/r/base/ifelse.html'>ifelse</a></span><span class='op'>(</span><span class='va'>g</span> <span class='op'>%in%</span> <span class='va'>mito_set</span>, <span class='st'>"dodgerBlue"</span>, <span class='st'>"black"</span><span class='op'>)</span><span class='op'>)</span>,
    sub <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='st'>"r = "</span>, <span class='fu'><a href='https://scales.r-lib.org/reference/number.html'>number</a></span><span class='op'>(</span><span class='va'>cor_i</span><span class='op'>[</span><span class='va'>i</span><span class='op'>]</span>, accuracy <span class='op'>=</span> <span class='fl'>0.001</span><span class='op'>)</span><span class='op'>)</span>,
    xlab <span class='op'>=</span> <span class='st'>"logCPM(UMIs)"</span>,
    ylab <span class='op'>=</span> <span class='st'>"logCPM(reads)"</span>,
    main <span class='op'>=</span> <span class='va'>i</span>,
    xlim <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/range.html'>range</a></span><span class='op'>(</span><span class='va'>X</span><span class='op'>)</span>,
    ylim <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/range.html'>range</a></span><span class='op'>(</span><span class='va'>Y</span><span class='op'>)</span><span class='op'>)</span>
  <span class='fu'><a href='https://rdrr.io/r/graphics/abline.html'>abline</a></span><span class='op'>(</span>a <span class='op'>=</span> <span class='fl'>0</span>, b <span class='op'>=</span> <span class='fl'>1</span>, col <span class='op'>=</span> <span class='st'>"grey"</span>, lty <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span>
<span class='op'>}</span>
<span class='fu'><a href='https://rdrr.io/r/graphics/plot.default.html'>plot</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/stats/density.html'>density</a></span><span class='op'>(</span><span class='va'>cor_i</span>, na.rm <span class='op'>=</span> <span class='cn'>TRUE</span>, from <span class='op'>=</span> <span class='op'>-</span><span class='fl'>1</span>, to <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span>,
  main <span class='op'>=</span> <span class='st'>"All genes"</span>, 
  xlab <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/expression.html'>expression</a></span><span class='op'>(</span><span class='va'>r</span><span class='op'>[</span><span class='va'>p</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:within-gene-cors"></span>
<img src="C086_Kinkel.preprocess_files/within-gene-cors-1.png" alt="Association between UMI counts and read counts for each sample within each gene. The first 8 panels are scatter plots of the UMI counts and read counts for each sample for the genes with the 3 lowest correlations, 2 middlemost correlations, and 3 highest correlations. The correlation in the scatterplot is reported below each panel. Genes that are ribosomal protein genes subunits are highlighted in orange and mitochondrial genes in blue. The final panel is the distribution of these correlations across all genes." width="624" />
<p class="caption">
Figure 9: Association between UMI counts and read counts for each sample within each gene. The first 8 panels are scatter plots of the UMI counts and read counts for each sample for the genes with the 3 lowest correlations, 2 middlemost correlations, and 3 highest correlations. The correlation in the scatterplot is reported below each panel. Genes that are ribosomal protein genes subunits are highlighted in orange and mitochondrial genes in blue. The final panel is the distribution of these correlations across all genes.
</p>
</div>
</div>
<h2 id="summary-2">Summary</h2>
<p>The ultimate aim of this analysis is to perform a differential expression (DE) analysis. On the one hand, a DE analysis is more powerful when there are more counts available, which is where the read counts are favoured. On the other hand, a DE analysis may be biased if these counts arise through PCR duplicates, which is where the deduplicated UMI counts are favoured. The choice between using the read counts or the UMI counts in the DE analysis is therefore a trade-off between these two choices.</p>
<p><strong>For now, we defer our choice of read counts vs. UMI counts, but the remainder of this preprocessing report uses read counts</strong>.</p>
<h1 id="quality-control-of-samples">Quality control of samples</h1>
<h2 id="defining-the-quality-control-metrics">Defining the quality control metrics</h2>
<p>Low-quality samples need to be removed to ensure that technical effects do not distort downstream analysis results. We use several quality control (QC) metrics to measure the quality of the samples:</p>
<ul>
<li><code>sum</code>: This measures the library size of the samples, which is the total sum of counts across genes. We want samples to have high library sizes as this means more RNA has been successfully captured during library preparation.</li>
<li><code>detected</code>: This is the number of expressed features<a href="#fn5" class="footnote-ref" id="fnref5" role="doc-noteref"><sup>5</sup></a> in each sample. Samples with few expressed features are likely to be of poor quality, as the diverse transcript population has not been successful captured.</li>
<li><code>subsets_Mt_percent</code>: This measures the proportion of reads which are mapped to mitochondrial RNA. If there is a higher than expected proportion of mitochondrial RNA this is often symptomatic of a sample which is under stress and is therefore of low quality and will not be used for the analysis.</li>
<li><code>subsets_Ribo_percent</code>: This measures the proportion of reads which are mapped to ribosomal protein genes. If there is a higher than expected proportion of ribosomal protein gene expression this is often symptomatic of a cell which is of compromised quality and we may want to exclude it from the analysis.</li>
</ul>
<p>For mini-bulk data, we typically observe library sizes that are in the hundreds of thousands<a href="#fn6" class="footnote-ref" id="fnref6" role="doc-noteref"><sup>6</sup></a> and several thousand expressed genes per cell.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>is_mito</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span> <span class='op'>%in%</span> <span class='va'>mito_set</span>
<span class='fu'><a href='https://rdrr.io/r/base/summary.html'>summary</a></span><span class='op'>(</span><span class='va'>is_mito</span><span class='op'>)</span>
<span class='va'>is_ribo</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span> <span class='op'>%in%</span> <span class='va'>ribo_set</span>
<span class='fu'><a href='https://rdrr.io/r/base/summary.html'>summary</a></span><span class='op'>(</span><span class='va'>is_ribo</span><span class='op'>)</span>
<span class='va'>sce_reads</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scuttle/man/addPerCellQC.html'>addPerCellQC</a></span><span class='op'>(</span>
  <span class='va'>sce</span>,
  subsets <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>Mt <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/which.html'>which</a></span><span class='op'>(</span><span class='va'>is_mito</span><span class='op'>)</span>, Ribo <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/which.html'>which</a></span><span class='op'>(</span><span class='va'>is_ribo</span><span class='op'>)</span><span class='op'>)</span>,
  assay.type <span class='op'>=</span> <span class='st'>"read_counts"</span>,
  use.altexps <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='va'>sce_umis</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scuttle/man/addPerCellQC.html'>addPerCellQC</a></span><span class='op'>(</span>
  <span class='va'>sce</span>,
  subsets <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>Mt <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/which.html'>which</a></span><span class='op'>(</span><span class='va'>is_mito</span><span class='op'>)</span>, Ribo <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/which.html'>which</a></span><span class='op'>(</span><span class='va'>is_ribo</span><span class='op'>)</span><span class='op'>)</span>,
  assay.type <span class='op'>=</span> <span class='st'>"UMI_counts"</span>,
  use.altexps <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>The aim is to remove putative low-quality samples that have low library sizes, low numbers of expressed features, and high mitochondrial proportions. Such samples can interfere with downstream analyses, e.g., by forming distinct clusters that complicate interpretation of the results.</p>
<h2 id="visualizing-the-qc-metrics">Visualizing the QC metrics</h2>
<p>The distributions of these metrics are shown in Figure <a href="#fig:qcplot">10</a> for read and UMI counts.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>p1</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotColData.html'>plotColData</a></span><span class='op'>(</span>
  <span class='va'>sce_reads</span>,
  <span class='st'>"sum"</span>,
  x <span class='op'>=</span> <span class='st'>"genotype.mouse"</span>,
  colour_by <span class='op'>=</span> <span class='st'>"smchd1_genotype_updated"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>scale_y_log10</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>6</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/theme.html'>theme</a></span><span class='op'>(</span>axis.text.x <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/element.html'>element_text</a></span><span class='op'>(</span>angle <span class='op'>=</span> <span class='fl'>45</span>, hjust <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>scale_y_log10</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/annotation_logticks.html'>annotation_logticks</a></span><span class='op'>(</span>
    sides <span class='op'>=</span> <span class='st'>"l"</span>,
    short <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/grid/unit.html'>unit</a></span><span class='op'>(</span><span class='fl'>0.03</span>, <span class='st'>"cm"</span><span class='op'>)</span>,
    mid <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/grid/unit.html'>unit</a></span><span class='op'>(</span><span class='fl'>0.06</span>, <span class='st'>"cm"</span><span class='op'>)</span>,
    long <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/grid/unit.html'>unit</a></span><span class='op'>(</span><span class='fl'>0.09</span>, <span class='st'>"cm"</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>smchd1_genotype_updated_colours</span><span class='op'>)</span>
<span class='va'>p2</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotColData.html'>plotColData</a></span><span class='op'>(</span>
  <span class='va'>sce_reads</span>,
  <span class='st'>"detected"</span>,
  x <span class='op'>=</span> <span class='st'>"genotype.mouse"</span>,
  colour_by <span class='op'>=</span> <span class='st'>"smchd1_genotype_updated"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>6</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/theme.html'>theme</a></span><span class='op'>(</span>axis.text.x <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/element.html'>element_text</a></span><span class='op'>(</span>angle <span class='op'>=</span> <span class='fl'>45</span>, hjust <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>smchd1_genotype_updated_colours</span><span class='op'>)</span>
<span class='va'>p3</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotColData.html'>plotColData</a></span><span class='op'>(</span>
  <span class='va'>sce_reads</span>,
  <span class='st'>"subsets_Mt_percent"</span>,
  x <span class='op'>=</span> <span class='st'>"genotype.mouse"</span>,
  colour_by <span class='op'>=</span> <span class='st'>"smchd1_genotype_updated"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>6</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/theme.html'>theme</a></span><span class='op'>(</span>axis.text.x <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/element.html'>element_text</a></span><span class='op'>(</span>angle <span class='op'>=</span> <span class='fl'>45</span>, hjust <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>smchd1_genotype_updated_colours</span><span class='op'>)</span>
<span class='va'>p4</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotColData.html'>plotColData</a></span><span class='op'>(</span>
  <span class='va'>sce_reads</span>,
  <span class='st'>"subsets_Ribo_percent"</span>,
  x <span class='op'>=</span> <span class='st'>"genotype.mouse"</span>,
  colour_by <span class='op'>=</span> <span class='st'>"smchd1_genotype_updated"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>6</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/theme.html'>theme</a></span><span class='op'>(</span>axis.text.x <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/element.html'>element_text</a></span><span class='op'>(</span>angle <span class='op'>=</span> <span class='fl'>45</span>, hjust <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>smchd1_genotype_updated_colours</span><span class='op'>)</span>
<span class='va'>q1</span> <span class='op'>&lt;-</span> <span class='va'>p1</span> <span class='op'>+</span> <span class='va'>p2</span> <span class='op'>+</span> <span class='va'>p3</span> <span class='op'>+</span> <span class='va'>p4</span> <span class='op'>+</span> <span class='fu'><a href='https://patchwork.data-imaginist.com/reference/plot_layout.html'>plot_layout</a></span><span class='op'>(</span>ncol <span class='op'>=</span> <span class='fl'>4</span><span class='op'>)</span>

<span class='va'>p5</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotColData.html'>plotColData</a></span><span class='op'>(</span>
  <span class='va'>sce_umis</span>,
  <span class='st'>"sum"</span>,
  x <span class='op'>=</span> <span class='st'>"genotype.mouse"</span>,
  colour_by <span class='op'>=</span> <span class='st'>"smchd1_genotype_updated"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>scale_y_log10</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>6</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/theme.html'>theme</a></span><span class='op'>(</span>axis.text.x <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/element.html'>element_text</a></span><span class='op'>(</span>angle <span class='op'>=</span> <span class='fl'>45</span>, hjust <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>scale_y_log10</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/annotation_logticks.html'>annotation_logticks</a></span><span class='op'>(</span>
    sides <span class='op'>=</span> <span class='st'>"l"</span>,
    short <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/grid/unit.html'>unit</a></span><span class='op'>(</span><span class='fl'>0.03</span>, <span class='st'>"cm"</span><span class='op'>)</span>,
    mid <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/grid/unit.html'>unit</a></span><span class='op'>(</span><span class='fl'>0.06</span>, <span class='st'>"cm"</span><span class='op'>)</span>,
    long <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/grid/unit.html'>unit</a></span><span class='op'>(</span><span class='fl'>0.09</span>, <span class='st'>"cm"</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>smchd1_genotype_updated_colours</span><span class='op'>)</span>
<span class='va'>p6</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotColData.html'>plotColData</a></span><span class='op'>(</span>
  <span class='va'>sce_umis</span>,
  <span class='st'>"detected"</span>,
  x <span class='op'>=</span> <span class='st'>"genotype.mouse"</span>,
  colour_by <span class='op'>=</span> <span class='st'>"smchd1_genotype_updated"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>6</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/theme.html'>theme</a></span><span class='op'>(</span>axis.text.x <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/element.html'>element_text</a></span><span class='op'>(</span>angle <span class='op'>=</span> <span class='fl'>45</span>, hjust <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>smchd1_genotype_updated_colours</span><span class='op'>)</span>
<span class='va'>p7</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotColData.html'>plotColData</a></span><span class='op'>(</span>
  <span class='va'>sce_umis</span>,
  <span class='st'>"subsets_Mt_percent"</span>,
  x <span class='op'>=</span> <span class='st'>"genotype.mouse"</span>,
  colour_by <span class='op'>=</span> <span class='st'>"smchd1_genotype_updated"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>6</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/theme.html'>theme</a></span><span class='op'>(</span>axis.text.x <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/element.html'>element_text</a></span><span class='op'>(</span>angle <span class='op'>=</span> <span class='fl'>45</span>, hjust <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>smchd1_genotype_updated_colours</span><span class='op'>)</span>
<span class='va'>p8</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotColData.html'>plotColData</a></span><span class='op'>(</span>
  <span class='va'>sce_umis</span>,
  <span class='st'>"subsets_Ribo_percent"</span>,
  x <span class='op'>=</span> <span class='st'>"genotype.mouse"</span>,
  colour_by <span class='op'>=</span> <span class='st'>"smchd1_genotype_updated"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>6</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/theme.html'>theme</a></span><span class='op'>(</span>axis.text.x <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/element.html'>element_text</a></span><span class='op'>(</span>angle <span class='op'>=</span> <span class='fl'>45</span>, hjust <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>smchd1_genotype_updated_colours</span><span class='op'>)</span>
<span class='va'>q2</span> <span class='op'>&lt;-</span> <span class='va'>p5</span> <span class='op'>+</span> <span class='va'>p6</span> <span class='op'>+</span> <span class='va'>p7</span> <span class='op'>+</span> <span class='va'>p8</span> <span class='op'>+</span> <span class='fu'><a href='https://patchwork.data-imaginist.com/reference/plot_layout.html'>plot_layout</a></span><span class='op'>(</span>ncol <span class='op'>=</span> <span class='fl'>4</span><span class='op'>)</span>

<span class='va'>q1</span> <span class='op'>/</span> <span class='va'>q2</span> <span class='op'>+</span> 
  <span class='fu'><a href='https://patchwork.data-imaginist.com/reference/plot_layout.html'>plot_layout</a></span><span class='op'>(</span>guides <span class='op'>=</span> <span class='st'>"collect"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://patchwork.data-imaginist.com/reference/plot_annotation.html'>plot_annotation</a></span><span class='op'>(</span>
    title <span class='op'>=</span> <span class='st'>"QC metrics"</span>,
    subtitle <span class='op'>=</span> <span class='st'>"Reads counts (top row) and UMI counts (bottom row)"</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:qcplot"></span>
<img src="C086_Kinkel.preprocess_files/qcplot-1.png" alt="Distributions of various QC metrics for all samples in the data set. This includes the library sizes, number of expressed genes, and proportion of reads mapped to mitochondrial genes." width="624" />
<p class="caption">
Figure 10: Distributions of various QC metrics for all samples in the data set. This includes the library sizes, number of expressed genes, and proportion of reads mapped to mitochondrial genes.
</p>
</div>
</div>
<h2 id="identifying-outliers-by-each-metric">Identifying outliers by each metric</h2>
<p>Outliers are defined based on the median absolute deviation (MADs) from the median value of each metric across all samples.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>libsize_drop_reads</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scuttle/man/isOutlier.html'>isOutlier</a></span><span class='op'>(</span>
  metric <span class='op'>=</span> <span class='va'>sce_reads</span><span class='op'>$</span><span class='va'>sum</span>, 
  nmads <span class='op'>=</span> <span class='fl'>3</span>,
  type <span class='op'>=</span> <span class='st'>"lower"</span>, 
  log <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
<span class='va'>libsize_drop_umis</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scuttle/man/isOutlier.html'>isOutlier</a></span><span class='op'>(</span>
  metric <span class='op'>=</span> <span class='va'>sce_umis</span><span class='op'>$</span><span class='va'>sum</span>, 
  nmads <span class='op'>=</span> <span class='fl'>3</span>,
  type <span class='op'>=</span> <span class='st'>"lower"</span>, 
  log <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
<span class='va'>feature_drop_reads</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scuttle/man/isOutlier.html'>isOutlier</a></span><span class='op'>(</span>
  metric <span class='op'>=</span> <span class='va'>sce_reads</span><span class='op'>$</span><span class='va'>detected</span>,
  nmads <span class='op'>=</span> <span class='fl'>3</span>, 
  type <span class='op'>=</span> <span class='st'>"lower"</span>, 
  log <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
<span class='va'>feature_drop_umis</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scuttle/man/isOutlier.html'>isOutlier</a></span><span class='op'>(</span>
  metric <span class='op'>=</span> <span class='va'>sce_umis</span><span class='op'>$</span><span class='va'>detected</span>,
  nmads <span class='op'>=</span> <span class='fl'>3</span>, 
  type <span class='op'>=</span> <span class='st'>"lower"</span>, 
  log <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>The following table summarises the QC cutoffs:</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>libsize_drop_df</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>
  reads_lower <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/attributes.html'>attributes</a></span><span class='op'>(</span><span class='va'>libsize_drop_reads</span><span class='op'>)</span><span class='op'>$</span><span class='va'>thresholds</span><span class='op'>[</span><span class='st'>"lower"</span><span class='op'>]</span>,
  umis_lower <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/attributes.html'>attributes</a></span><span class='op'>(</span><span class='va'>libsize_drop_umis</span><span class='op'>)</span><span class='op'>$</span><span class='va'>thresholds</span><span class='op'>[</span><span class='st'>"lower"</span><span class='op'>]</span><span class='op'>)</span>
<span class='va'>feature_drop_df</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>
  reads_lower <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/attributes.html'>attributes</a></span><span class='op'>(</span><span class='va'>feature_drop_reads</span><span class='op'>)</span><span class='op'>$</span><span class='va'>thresholds</span><span class='op'>[</span><span class='st'>"lower"</span><span class='op'>]</span>,
  umis_lower <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/attributes.html'>attributes</a></span><span class='op'>(</span><span class='va'>feature_drop_umis</span><span class='op'>)</span><span class='op'>$</span><span class='va'>thresholds</span><span class='op'>[</span><span class='st'>"lower"</span><span class='op'>]</span><span class='op'>)</span>
<span class='va'>qc_cutoffs_df</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/limma/man/cbind.html'>cbind</a></span><span class='op'>(</span><span class='va'>libsize_drop_df</span>, <span class='va'>feature_drop_df</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>qc_cutoffs_df</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span>
  <span class='st'>"read counts"</span>, <span class='st'>"umi counts"</span>, <span class='st'>"read features"</span>, <span class='st'>"umi features"</span><span class='op'>)</span>
<span class='va'>qc_cutoffs_df</span> <span class='op'>%&gt;%</span>
  <span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span>caption <span class='op'>=</span> <span class='st'>"QC cutoffs"</span>, digits <span class='op'>=</span> <span class='fl'>0</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<table>
<caption><span id="tab:unnamed-chunk-20">Table 1: </span>QC cutoffs</caption>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">read counts</th>
<th style="text-align: right;">umi counts</th>
<th style="text-align: right;">read features</th>
<th style="text-align: right;">umi features</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">lower</td>
<td style="text-align: right;">757058</td>
<td style="text-align: right;">63235</td>
<td style="text-align: right;">11974</td>
<td style="text-align: right;">11974</td>
</tr>
</tbody>
</table>
</div>
<p><strong>We would typically remove outliers (e.g., small outliers for the library size and the number of expressed features) at this stage.</strong> However, we opt to retain all samples for the time being. The following table highlights samples that would typically exclude as outliers by this criteria:</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>keep</span> <span class='op'>&lt;-</span> <span class='op'>!</span><span class='op'>(</span><span class='va'>libsize_drop_reads</span> <span class='op'>|</span> <span class='va'>libsize_drop_umis</span> <span class='op'>|</span> <span class='va'>feature_drop_reads</span> <span class='op'>|</span> <span class='va'>feature_drop_umis</span><span class='op'>)</span>
<span class='va'>x</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/limma/man/cbind.html'>cbind</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>sample <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span>, keep <span class='op'>=</span> <span class='va'>keep</span><span class='op'>)</span>,
  <span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce_reads</span><span class='op'>)</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"sum"</span>, <span class='st'>"detected"</span><span class='op'>)</span><span class='op'>]</span>,
  <span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce_umis</span><span class='op'>)</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"sum"</span>, <span class='st'>"detected"</span><span class='op'>)</span><span class='op'>]</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span><span class='op'>[</span><span class='fl'>1</span><span class='op'>:</span><span class='fl'>2</span><span class='op'>]</span>,
  <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='st'>"reads_"</span>, <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span><span class='op'>[</span><span class='fl'>3</span><span class='op'>:</span><span class='fl'>4</span><span class='op'>]</span><span class='op'>)</span>,
  <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='st'>"umis_"</span>, <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span><span class='op'>[</span><span class='fl'>5</span><span class='op'>:</span><span class='fl'>6</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/row.names.html'>row.names</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='cn'>NULL</span>
<span class='fu'>DT</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/DT/man/datatable.html'>datatable</a></span><span class='op'>(</span>
  <span class='va'>x</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/order.html'>order</a></span><span class='op'>(</span><span class='va'>keep</span><span class='op'>)</span>, <span class='op'>]</span>,
  caption <span class='op'>=</span> <span class='st'>"Quality control metrics including library size (`sum`) and number of genes detected (`keep`) for each replicate. Each replicate is annotated by whether it would be kept or excluded by the typical quality control procedures (`keep`)."</span>,
  rownames <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div id="htmlwidget-b9ae593cd5fa6121a7f6" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-b9ae593cd5fa6121a7f6">{"x":{"filter":"none","caption":"<caption>Quality control metrics including library size (`sum`) and number of genes detected (`keep`) for each replicate. Each replicate is annotated by whether it would be kept or excluded by the typical quality control procedures (`keep`).<\/caption>","data":[["Del.1282.2","Del.1250.1","Del.1250.2","Del.1253.1","Del.1253.2","Del.1253.3","Het.1264.1","Het.1264.2","Het.1264.3","Het.1248.1","Het.1248.2","Het.1248.3","Del.1250.3","Het.1269.1","Het.1269.2","WT.1271.1","WT.1271.2","WT.1271.3","Het.1272.1","Het.1272.2","Het.1272.3","Del.1266.1","Del.1266.2","Del.1266.3","Het.1269.3","Del.1283.1","Het.1286.1","Het.1286.2","Het.1286.3","Het.1291.1","Het.1291.2","Het.1291.3","Del.1282.1","Del.1282.3","Del.1283.2"],[false,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true],[325500,3057246,2361371,3269790,5050320,3795497,3338054,3809314,5501653,2569723,1667259,2652487,2462512,6069536,4677176,5112634,7236478,6375836,6746558,2131620,5750758,7126330,4268451,3517525,3340891,4933566,4861771,6304428,3815812,8058749,3731110,6249593,3004013,3750452,6734852],[10130,14467,14091,14699,15446,14840,15081,15056,15806,13907,12516,14094,13770,15933,15552,15884,16480,16197,16512,13892,15790,16195,15093,14320,14882,14962,15513,16242,15206,16510,14920,16052,13967,15010,16020],[60010,203074,176314,206944,264722,228389,250085,241552,281671,168991,112262,172917,158837,305832,288885,307399,362018,321182,372375,167906,303735,340742,257046,207484,228998,182702,189971,339636,174598,375732,168316,328669,132731,169475,353703],[10130,14467,14091,14699,15446,14840,15081,15056,15806,13907,12516,14094,13770,15933,15552,15884,16480,16197,16512,13892,15790,16195,15093,14320,14882,14962,15513,16242,15206,16510,14920,16052,13967,15010,16020]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>sample<\/th>\n      <th>keep<\/th>\n      <th>reads_sum<\/th>\n      <th>reads_detected<\/th>\n      <th>umis_sum<\/th>\n      <th>umis_detected<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,5]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<h2 id="summary-3">Summary</h2>
<p>We have opted not to exclude any replicates at this stage. However, we have identified 1 replicate (<code>Del.1282.2</code>) that has unusually small library size (by reads and UMIs) and number of genes detected compared to the other samples. We will keep an eye on this replicate and may later remove it.</p>
<h1 id="examining-gene-level-metrics">Examining gene-level metrics</h1>
<h2 id="inspecting-the-most-highly-expressed-genes">Inspecting the most highly expressed genes</h2>
<p>Figures <a href="#fig:topgenes">11</a> shows the most highly expressed genes in the dataset. Many of these genes are mitochondrial genes and ribosomal protein.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotHighestExprs.html'>plotHighestExprs</a></span><span class='op'>(</span><span class='va'>sce</span>, n <span class='op'>=</span> <span class='fl'>50</span>, exprs_values <span class='op'>=</span> <span class='st'>"read_counts"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span><span class='op'>(</span><span class='st'>"Read counts"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotHighestExprs.html'>plotHighestExprs</a></span><span class='op'>(</span><span class='va'>sce</span>, n <span class='op'>=</span> <span class='fl'>50</span>, exprs_values <span class='op'>=</span> <span class='st'>"UMI_counts"</span><span class='op'>)</span> <span class='op'>+</span> 
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span><span class='op'>(</span><span class='st'>"UMI counts"</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:topgenes"></span>
<img src="C086_Kinkel.preprocess_files/topgenes-1.png" alt="Percentage of total counts assigned to the top 50 most highly-abundant features in the data set. For each feature, each bar represents the percentage assigned to that feature for a single sample, while the circle represents the average across all samples. Bars are coloured by the total number of expressed features in each samples, while circles are coloured according to whether the feature is labelled as a control feature." width="624" />
<p class="caption">
Figure 11: Percentage of total counts assigned to the top 50 most highly-abundant features in the data set. For each feature, each bar represents the percentage assigned to that feature for a single sample, while the circle represents the average across all samples. Bars are coloured by the total number of expressed features in each samples, while circles are coloured according to whether the feature is labelled as a control feature.
</p>
</div>
</div>
<p>Figure <a href="#fig:topgenes-filtered">12</a> shows the most highly expressed after excluding the mitochondrial genes and ribosomal protein.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotHighestExprs.html'>plotHighestExprs</a></span><span class='op'>(</span>
  <span class='va'>sce</span>, 
  drop_features <span class='op'>=</span> <span class='fu'><a href='https://generics.r-lib.org/reference/setops.html'>intersect</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='va'>mito_set</span>, <span class='va'>ribo_set</span><span class='op'>)</span><span class='op'>)</span>,
  n <span class='op'>=</span> <span class='fl'>50</span>,
  exprs_values <span class='op'>=</span> <span class='st'>"read_counts"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span><span class='op'>(</span><span class='st'>"Read counts"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotHighestExprs.html'>plotHighestExprs</a></span><span class='op'>(</span>
    <span class='va'>sce</span>, 
    drop_features <span class='op'>=</span> <span class='fu'><a href='https://generics.r-lib.org/reference/setops.html'>intersect</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='va'>mito_set</span>, <span class='va'>ribo_set</span><span class='op'>)</span><span class='op'>)</span>,
    n <span class='op'>=</span> <span class='fl'>50</span>,
    exprs_values <span class='op'>=</span> <span class='st'>"UMI_counts"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span><span class='op'>(</span><span class='st'>"UMI counts"</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:topgenes-filtered"></span>
<img src="C086_Kinkel.preprocess_files/topgenes-filtered-1.png" alt="Percentage of total counts assigned to the top 50 most highly-abundant features (after excluding mitochondrial genes and ribosomal protein genes) in the data set. For each feature, each bar represents the percentage assigned to that feature for a single sample, while the circle represents the average across all samples. Bars are coloured by the total number of expressed features in each sample, while circles are coloured according to whether the feature is labelled as a control feature." width="624" />
<p class="caption">
Figure 12: Percentage of total counts assigned to the top 50 most highly-abundant features (after excluding mitochondrial genes and ribosomal protein genes) in the data set. For each feature, each bar represents the percentage assigned to that feature for a single sample, while the circle represents the average across all samples. Bars are coloured by the total number of expressed features in each sample, while circles are coloured according to whether the feature is labelled as a control feature.
</p>
</div>
</div>
<h2 id="sex-check-using-xist">Sex check using <em>Xist</em></h2>
<p>Figure <a href="#fig:sex-check">13</a> visualises the expression of <em>Xist</em>, a female-specific gene. This can be used to confirm the reported sex of the sample and to detect sample mix-ups. Figure <a href="#fig:sex-check">13</a> shows that female samples have much higher average expression of <em>Xist</em> than male samples, as we would expect.</p>
<p>There is one replicate, <code>Del.1282.2</code>, that has is nominally male but has <span class="math inline">\(logCPM(UMI_{counts})\)</span> for <em>Xist</em> that is comparable to the female samples. However, we have already seen that this replicate has an unusually small library size (by reads and UMIs) and number of genes detected compared to the other samples, so it already suspect.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'>assay</span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"logCPM_reads"</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'>edgeR</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cpm.html'>cpm</a></span><span class='op'>(</span><span class='fu'>assay</span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"read_counts"</span><span class='op'>)</span>, log <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
<span class='fu'>assay</span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"logCPM_umis"</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'>edgeR</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cpm.html'>cpm</a></span><span class='op'>(</span><span class='fu'>assay</span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"UMI_counts"</span><span class='op'>)</span>, log <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
<span class='va'>p1</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotExpression.html'>plotExpression</a></span><span class='op'>(</span>
  <span class='va'>sce</span>,
  <span class='st'>"Xist"</span>,
  x <span class='op'>=</span> <span class='st'>"sex"</span>,
  exprs_values <span class='op'>=</span> <span class='st'>"read_counts"</span>,
  colour_by <span class='op'>=</span> <span class='st'>"sex"</span><span class='op'>)</span> <span class='op'>+</span> 
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>sex_colours</span>, name <span class='op'>=</span> <span class='st'>"sex"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>fill <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Read counts"</span><span class='op'>)</span>  <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/lims.html'>ylim</a></span><span class='op'>(</span><span class='fl'>0</span>, <span class='cn'>NA</span><span class='op'>)</span>
<span class='va'>p2</span> <span class='op'>&lt;-</span><span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotExpression.html'>plotExpression</a></span><span class='op'>(</span>
  <span class='va'>sce</span>,
  <span class='st'>"Xist"</span>,
  x <span class='op'>=</span> <span class='st'>"sex"</span>,
  exprs_values <span class='op'>=</span> <span class='st'>"logCPM_reads"</span>,
  colour_by <span class='op'>=</span> <span class='st'>"sex"</span><span class='op'>)</span> <span class='op'>+</span> 
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>sex_colours</span>, name <span class='op'>=</span> <span class='st'>"sex"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>fill <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"logCPM(read counts)"</span><span class='op'>)</span>
<span class='va'>p3</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotExpression.html'>plotExpression</a></span><span class='op'>(</span>
  <span class='va'>sce</span>,
  <span class='st'>"Xist"</span>,
  x <span class='op'>=</span> <span class='st'>"sex"</span>,
  exprs_values <span class='op'>=</span> <span class='st'>"UMI_counts"</span>,
  colour_by <span class='op'>=</span> <span class='st'>"sex"</span><span class='op'>)</span> <span class='op'>+</span> 
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>sex_colours</span>, name <span class='op'>=</span> <span class='st'>"sex"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>fill <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"UMI counts"</span><span class='op'>)</span>  <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/lims.html'>ylim</a></span><span class='op'>(</span><span class='fl'>0</span>, <span class='cn'>NA</span><span class='op'>)</span>
<span class='va'>p4</span> <span class='op'>&lt;-</span><span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotExpression.html'>plotExpression</a></span><span class='op'>(</span>
  <span class='va'>sce</span>,
  <span class='st'>"Xist"</span>,
  x <span class='op'>=</span> <span class='st'>"sex"</span>,
  exprs_values <span class='op'>=</span> <span class='st'>"logCPM_umis"</span>,
  colour_by <span class='op'>=</span> <span class='st'>"sex"</span><span class='op'>)</span> <span class='op'>+</span> 
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>sex_colours</span>, name <span class='op'>=</span> <span class='st'>"sex"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>fill <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"logCPM(UMI counts)"</span><span class='op'>)</span>
<span class='va'>p1</span> <span class='op'>+</span> <span class='va'>p2</span> <span class='op'>+</span> <span class='va'>p3</span> <span class='op'>+</span> <span class='va'>p4</span> <span class='op'>+</span> <span class='fu'><a href='https://patchwork.data-imaginist.com/reference/plot_layout.html'>plot_layout</a></span><span class='op'>(</span>guides <span class='op'>=</span> <span class='st'>"collect"</span>, ncol <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:sex-check"></span>
<img src="C086_Kinkel.preprocess_files/sex-check-1.png" alt="Expression of *Xist* by mouse on a 'raw counts' and 'logCPM' scale using the read or UMI counts." width="624" />
<p class="caption">
Figure 13: Expression of <em>Xist</em> by mouse on a ‘raw counts’ and ‘logCPM’ scale using the read or UMI counts.
</p>
</div>
</div>
<h2 id="genotype-check-using-smchd1">Genotype check using <em>Smchd1</em></h2>
<p>Figure <a href="#fig:genotype-check">14</a> visualises the expression of <em>Smchd1</em>, the gene under investigation and which was deleted via a floxed allele in the <code>Del</code> mice. This can be used to confirm the reported genotype of the sample and to detect sample mix-ups. The <code>Del</code> samples have somewhat lower average expression of <em>Smchd1</em> than <code>Het</code> and <code>WT</code> samples on, as we would expect, but the difference is not a striking on-off binary difference. Furthermore, the <code>Del</code> samples have positive counts of <em>Smchd1</em>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>p1</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotExpression.html'>plotExpression</a></span><span class='op'>(</span>
  <span class='va'>sce</span>,
  <span class='st'>"Smchd1"</span>,
  x <span class='op'>=</span> <span class='st'>"genotype.mouse"</span>,
  exprs_values <span class='op'>=</span> <span class='st'>"read_counts"</span>,
  colour_by <span class='op'>=</span> <span class='st'>"smchd1_genotype_updated"</span><span class='op'>)</span> <span class='op'>+</span> 
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>
    values <span class='op'>=</span> <span class='va'>smchd1_genotype_updated_colours</span>,
    name <span class='op'>=</span> <span class='st'>"smchd1_genotype_updated"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>fill <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span> <span class='op'>+</span> 
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Read counts"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_discrete.html'>scale_x_discrete</a></span><span class='op'>(</span>guide <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guide_axis.html'>guide_axis</a></span><span class='op'>(</span>angle <span class='op'>=</span> <span class='fl'>45</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/lims.html'>ylim</a></span><span class='op'>(</span><span class='fl'>0</span>, <span class='cn'>NA</span><span class='op'>)</span>
<span class='va'>p2</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotExpression.html'>plotExpression</a></span><span class='op'>(</span>
    <span class='va'>sce</span>,
    <span class='st'>"Smchd1"</span>,
    x <span class='op'>=</span> <span class='st'>"genotype.mouse"</span>,
    exprs_values <span class='op'>=</span> <span class='st'>"logCPM_reads"</span>,
    colour_by <span class='op'>=</span> <span class='st'>"smchd1_genotype_updated"</span><span class='op'>)</span> <span class='op'>+</span> 
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>
    values <span class='op'>=</span> <span class='va'>smchd1_genotype_updated_colours</span>,
    name <span class='op'>=</span> <span class='st'>"smchd1_genotype_updated"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>fill <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span> <span class='op'>+</span> 
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"logCPM(read counts)"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_discrete.html'>scale_x_discrete</a></span><span class='op'>(</span>guide <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guide_axis.html'>guide_axis</a></span><span class='op'>(</span>angle <span class='op'>=</span> <span class='fl'>45</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>p3</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotExpression.html'>plotExpression</a></span><span class='op'>(</span>
  <span class='va'>sce</span>,
  <span class='st'>"Smchd1"</span>,
  x <span class='op'>=</span> <span class='st'>"genotype.mouse"</span>,
  exprs_values <span class='op'>=</span> <span class='st'>"UMI_counts"</span>,
  colour_by <span class='op'>=</span> <span class='st'>"smchd1_genotype_updated"</span><span class='op'>)</span> <span class='op'>+</span> 
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>
    values <span class='op'>=</span> <span class='va'>smchd1_genotype_updated_colours</span>,
    name <span class='op'>=</span> <span class='st'>"smchd1_genotype_updated"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>fill <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span> <span class='op'>+</span> 
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"UMI counts"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_discrete.html'>scale_x_discrete</a></span><span class='op'>(</span>guide <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guide_axis.html'>guide_axis</a></span><span class='op'>(</span>angle <span class='op'>=</span> <span class='fl'>45</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/lims.html'>ylim</a></span><span class='op'>(</span><span class='fl'>0</span>, <span class='cn'>NA</span><span class='op'>)</span>
<span class='va'>p4</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotExpression.html'>plotExpression</a></span><span class='op'>(</span>
    <span class='va'>sce</span>,
    <span class='st'>"Smchd1"</span>,
    x <span class='op'>=</span> <span class='st'>"genotype.mouse"</span>,
    exprs_values <span class='op'>=</span> <span class='st'>"logCPM_umis"</span>,
    colour_by <span class='op'>=</span> <span class='st'>"smchd1_genotype_updated"</span><span class='op'>)</span> <span class='op'>+</span> 
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>
    values <span class='op'>=</span> <span class='va'>smchd1_genotype_updated_colours</span>,
    name <span class='op'>=</span> <span class='st'>"smchd1_genotype_updated"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>fill <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span> <span class='op'>+</span> 
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"logCPM(UMI counts)"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_discrete.html'>scale_x_discrete</a></span><span class='op'>(</span>guide <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guide_axis.html'>guide_axis</a></span><span class='op'>(</span>angle <span class='op'>=</span> <span class='fl'>45</span><span class='op'>)</span><span class='op'>)</span>

<span class='va'>p1</span> <span class='op'>+</span> <span class='va'>p2</span> <span class='op'>+</span> <span class='va'>p3</span> <span class='op'>+</span> <span class='va'>p4</span> <span class='op'>+</span> <span class='fu'><a href='https://patchwork.data-imaginist.com/reference/plot_layout.html'>plot_layout</a></span><span class='op'>(</span>guides <span class='op'>=</span> <span class='st'>"collect"</span>, ncol <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:genotype-check"></span>
<img src="C086_Kinkel.preprocess_files/genotype-check-1.png" alt="Expression of *Smchd1* by mouse on a 'raw counts' and 'logCPM' scale using the read or UMI counts." width="624" />
<p class="caption">
Figure 14: Expression of <em>Smchd1</em> by mouse on a ‘raw counts’ and ‘logCPM’ scale using the read or UMI counts.
</p>
</div>
</div>
<h1 id="concluding-remarks">Concluding remarks</h1>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='co'># Remove some unneeded assays</span>
<span class='fu'>assay</span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"logCPM_reads"</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='cn'>NULL</span>
<span class='fu'>assay</span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"logCPM_umis"</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='cn'>NULL</span>

<span class='co'># Save</span>
<span class='fu'><a href='https://rdrr.io/r/base/readRDS.html'>saveRDS</a></span><span class='op'>(</span>
  <span class='va'>sce</span>,
  <span class='fu'><a href='https://here.r-lib.org//reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"data"</span>, <span class='st'>"SCEs"</span>, <span class='st'>"C086_Kinkel.preprocessed.SCE.rds"</span><span class='op'>)</span>,
  compress <span class='op'>=</span> <span class='st'>"xz"</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>The processed <em>SingleCellExperiment</em> object is available (see <a href="../data/SCEs/C086_Kinkel.preprocessed.SCE.rds"><code>data/SCEs/C086_Kinkel.preprocessed.SCE.rds</code></a>). This will be used in downstream analyses, i.e. differential expression analysis.</p>
<h1 id="additional-information" class="appendix">Additional information</h1>
<p>The following are available on request:</p>
<ul>
<li>Full CSV tables of any data presented.</li>
<li>PDF/PNG files of any static plots.</li>
</ul>
<h1 id="session-info" class="appendix">Session info</h1>
<summary>The analysis and this document were prepared using the following software (click triangle to expand)</summary>
<details>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'>sessioninfo</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/sessioninfo/man/session_info.html'>session_info</a></span><span class='op'>(</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<pre><code>─ Session info ─────────────────────────────────────────────────────
 setting  value                       
 version  R version 4.0.3 (2020-10-10)
 os       CentOS Linux 7 (Core)       
 system   x86_64, linux-gnu           
 ui       X11                         
 language (EN)                        
 collate  en_US.UTF-8                 
 ctype    en_US.UTF-8                 
 tz       Australia/Melbourne         
 date     2021-03-31                  

─ Packages ─────────────────────────────────────────────────────────
 ! package                            * version  date       lib
 P AnnotationDbi                      * 1.52.0   2020-10-27 [?]
 P AnnotationFilter                   * 1.14.0   2020-10-27 [?]
 P askpass                              1.1      2019-01-13 [?]
 P assertthat                           0.2.1    2019-03-21 [?]
 P backports                            1.2.1    2020-12-09 [?]
 P beachmat                             2.6.4    2020-12-20 [?]
 P beeswarm                             0.2.3    2016-04-25 [?]
 P Biobase                            * 2.50.0   2020-10-27 [?]
 P BiocFileCache                        1.14.0   2020-10-27 [?]
 P BiocGenerics                       * 0.36.0   2020-10-27 [?]
 P BiocManager                          1.30.10  2019-11-16 [?]
 P BiocNeighbors                        1.8.2    2020-12-07 [?]
 P BiocParallel                       * 1.24.1   2020-11-06 [?]
 P BiocSingular                         1.6.0    2020-10-27 [?]
 P BiocStyle                            2.18.1   2020-11-24 [?]
 P biomaRt                              2.46.3   2021-02-09 [?]
 P Biostrings                           2.58.0   2020-10-27 [?]
 P bit                                  4.0.4    2020-08-04 [?]
 P bit64                                4.0.5    2020-08-30 [?]
 P bitops                               1.0-6    2013-08-17 [?]
 P blob                                 1.2.1    2020-01-20 [?]
 P cachem                               1.0.4    2021-02-13 [?]
 P cellranger                           1.1.0    2016-07-27 [?]
 P checkmate                            2.0.0    2020-02-06 [?]
 P cli                                  2.3.1    2021-02-23 [?]
 P colorspace                           2.0-0    2020-11-11 [?]
 P commonmark                           1.7      2018-12-01 [?]
 P cowplot                            * 1.1.1    2020-12-30 [?]
 P crayon                               1.4.1    2021-02-08 [?]
 P crosstalk                            1.1.1    2021-01-12 [?]
 P curl                                 4.3      2019-12-02 [?]
 P DBI                                  1.1.1    2021-01-15 [?]
 P dbplyr                               2.1.0    2021-02-03 [?]
 P DelayedArray                         0.16.1   2021-01-22 [?]
 P DelayedMatrixStats                   1.12.3   2021-02-03 [?]
 P DEoptimR                             1.0-8    2016-11-19 [?]
 P digest                               0.6.27   2020-10-24 [?]
 P distill                              1.2      2021-01-13 [?]
 P downlit                              0.2.1    2020-11-04 [?]
 P dplyr                              * 1.0.4    2021-02-02 [?]
 P DT                                   0.17     2021-01-06 [?]
 P edgeR                                3.32.1   2021-01-14 [?]
 P ellipsis                             0.3.1    2020-05-15 [?]
 P EnsDb.Mmusculus.v79                * 2.99.0   2020-07-15 [?]
 P ensembldb                          * 2.14.0   2020-10-27 [?]
 P evaluate                             0.14     2019-05-28 [?]
 P fansi                                0.4.2    2021-01-15 [?]
 P farver                               2.0.3    2020-01-16 [?]
 P fastmap                              1.1.0    2021-01-25 [?]
 P generics                             0.1.0    2020-10-31 [?]
 P GenomeInfoDb                       * 1.26.2   2020-12-08 [?]
 P GenomeInfoDbData                     1.2.4    2020-10-20 [?]
 P GenomicAlignments                    1.26.0   2020-10-27 [?]
 P GenomicFeatures                    * 1.42.1   2020-11-12 [?]
 P GenomicRanges                      * 1.42.0   2020-10-27 [?]
 P GGally                               2.1.0    2021-01-06 [?]
 P ggbeeswarm                           0.6.0    2017-08-07 [?]
 P ggplot2                            * 3.3.3    2020-12-30 [?]
 P glue                                 1.4.2    2020-08-27 [?]
 P GO.db                              * 3.12.1   2020-11-04 [?]
 P graph                                1.68.0   2020-10-27 [?]
 P gridExtra                            2.3      2017-09-09 [?]
 P gt                                 * 0.2.2    2020-08-05 [?]
 P gtable                               0.3.0    2019-03-25 [?]
 P here                               * 1.0.1    2020-12-13 [?]
 P highr                                0.8      2019-03-20 [?]
 P hms                                  1.0.0    2021-01-13 [?]
 P htmltools                            0.5.1.1  2021-01-22 [?]
 P htmlwidgets                          1.5.3    2020-12-10 [?]
 P httr                                 1.4.2    2020-07-20 [?]
 P IRanges                            * 2.24.1   2020-12-12 [?]
 P irlba                                2.3.3    2019-02-05 [?]
 P janitor                            * 2.1.0    2021-01-05 [?]
 P jsonlite                             1.7.2    2020-12-09 [?]
 P knitr                                1.31     2021-01-27 [?]
 P labeling                             0.4.2    2020-10-20 [?]
 P lattice                              0.20-41  2020-04-02 [3]
 P lazyeval                             0.2.2    2019-03-15 [?]
 P lifecycle                            1.0.0    2021-02-15 [?]
 P limma                              * 3.46.0   2020-10-27 [?]
 P locfit                               1.5-9.4  2020-03-25 [?]
 P lubridate                            1.7.9.2  2020-11-13 [?]
 P magrittr                             2.0.1    2020-11-17 [?]
 P Matrix                               1.2-18   2019-11-27 [3]
 P MatrixGenerics                     * 1.2.1    2021-01-30 [?]
 P matrixStats                        * 0.58.0   2021-01-29 [?]
 P mclust                               5.4.7    2020-11-20 [?]
 P memoise                              2.0.0    2021-01-26 [?]
 P msigdbr                            * 7.2.1    2020-10-02 [?]
 P munsell                              0.5.0    2018-06-12 [?]
 P Mus.musculus                       * 1.3.1    2020-07-15 [?]
 P openssl                              1.4.3    2020-09-18 [?]
 P org.Hs.eg.db                         3.12.0   2020-10-20 [?]
 P org.Mm.eg.db                       * 3.12.0   2020-10-20 [?]
 P OrganismDbi                        * 1.32.0   2020-10-27 [?]
 P patchwork                          * 1.1.1    2020-12-17 [?]
 P pillar                               1.5.0    2021-02-22 [?]
 P pkgconfig                            2.0.3    2019-09-22 [?]
 P plyr                                 1.8.6    2020-03-03 [?]
 P prettyunits                          1.1.1    2020-01-24 [?]
 P progress                             1.2.2    2019-05-16 [?]
 P ProtGenerics                         1.22.0   2020-10-27 [?]
 P purrr                                0.3.4    2020-04-17 [?]
 P R6                                   2.5.0    2020-10-28 [?]
 P rappdirs                             0.3.3    2021-01-31 [?]
 P RBGL                                 1.66.0   2020-10-27 [?]
 P RColorBrewer                         1.1-2    2014-12-07 [?]
 P Rcpp                                 1.0.6    2021-01-15 [?]
 P RCurl                                1.98-1.2 2020-04-18 [?]
 P readxl                             * 1.3.1    2019-03-13 [?]
 P rematch                              1.0.1    2016-04-21 [?]
 P reshape                              0.8.8    2018-10-23 [?]
 P Rhtslib                              1.22.0   2020-10-27 [?]
 P rlang                                0.4.10   2020-12-30 [?]
 P rmarkdown                          * 2.7      2021-02-19 [?]
 P robustbase                           0.93-7   2021-01-04 [?]
 P rprojroot                            2.0.2    2020-11-15 [?]
 P Rsamtools                            2.6.0    2020-10-27 [?]
 P RSQLite                              2.2.3    2021-01-24 [?]
 P rsvd                                 1.0.3    2020-02-17 [?]
 P rtracklayer                          1.50.0   2020-10-27 [?]
 P S4Vectors                          * 0.28.1   2020-12-09 [?]
 P sass                                 0.3.1    2021-01-24 [?]
 P scales                             * 1.1.1    2020-05-11 [?]
 P scater                             * 1.18.5   2021-02-16 [?]
 P scPipe                             * 1.12.0   2020-10-27 [?]
 P scuttle                              1.0.4    2020-12-17 [?]
 P sessioninfo                          1.1.1    2018-11-05 [?]
 P SingleCellExperiment               * 1.12.0   2020-10-27 [?]
 P snakecase                            0.11.0   2019-05-25 [?]
 P sparseMatrixStats                    1.2.1    2021-02-02 [?]
 P stringi                              1.5.3    2020-09-09 [?]
 P stringr                              1.4.0    2019-02-10 [?]
 P SummarizedExperiment               * 1.20.0   2020-10-27 [?]
 P tibble                               3.0.6    2021-01-29 [?]
 P tidyr                              * 1.1.2    2020-08-27 [?]
 P tidyselect                           1.1.0    2020-05-11 [?]
 P TxDb.Mmusculus.UCSC.mm10.knownGene * 3.10.0   2020-07-15 [?]
 P utf8                                 1.1.4    2018-05-24 [?]
 P vctrs                                0.3.6    2020-12-17 [?]
 P vipor                                0.4.5    2017-03-22 [?]
 P viridis                              0.5.1    2018-03-29 [?]
 P viridisLite                          0.3.0    2018-02-01 [?]
 P withr                                2.4.1    2021-01-26 [?]
 P xfun                                 0.21     2021-02-10 [?]
 P XML                                  3.99-0.5 2020-07-23 [?]
 P xml2                                 1.3.2    2020-04-23 [?]
 P XVector                              0.30.0   2020-10-27 [?]
 P yaml                                 2.2.1    2020-02-01 [?]
 P zlibbioc                             1.36.0   2020-10-27 [?]
 source        
 Bioconductor  
 Bioconductor  
 CRAN (R 4.0.0)
 CRAN (R 4.0.0)
 CRAN (R 4.0.3)
 Bioconductor  
 CRAN (R 4.0.0)
 Bioconductor  
 Bioconductor  
 Bioconductor  
 CRAN (R 4.0.0)
 Bioconductor  
 Bioconductor  
 Bioconductor  
 Bioconductor  
 Bioconductor  
 Bioconductor  
 CRAN (R 4.0.0)
 CRAN (R 4.0.0)
 CRAN (R 4.0.0)
 CRAN (R 4.0.0)
 CRAN (R 4.0.3)
 CRAN (R 4.0.0)
 CRAN (R 4.0.0)
 CRAN (R 4.0.3)
 CRAN (R 4.0.3)
 CRAN (R 4.0.0)
 CRAN (R 4.0.3)
 CRAN (R 4.0.3)
 CRAN (R 4.0.3)
 CRAN (R 4.0.0)
 CRAN (R 4.0.3)
 CRAN (R 4.0.3)
 Bioconductor  
 Bioconductor  
 CRAN (R 4.0.0)
 CRAN (R 4.0.2)
 CRAN (R 4.0.3)
 CRAN (R 4.0.3)
 CRAN (R 4.0.3)
 CRAN (R 4.0.3)
 Bioconductor  
 CRAN (R 4.0.0)
 Bioconductor  
 Bioconductor  
 CRAN (R 4.0.0)
 CRAN (R 4.0.3)
 CRAN (R 4.0.0)
 CRAN (R 4.0.3)
 CRAN (R 4.0.3)
 Bioconductor  
 Bioconductor  
 Bioconductor  
 Bioconductor  
 Bioconductor  
 CRAN (R 4.0.3)
 CRAN (R 4.0.0)
 CRAN (R 4.0.3)
 CRAN (R 4.0.0)
 Bioconductor  
 Bioconductor  
 CRAN (R 4.0.0)
 CRAN (R 4.0.3)
 CRAN (R 4.0.0)
 CRAN (R 4.0.3)
 CRAN (R 4.0.0)
 CRAN (R 4.0.3)
 CRAN (R 4.0.3)
 CRAN (R 4.0.3)
 CRAN (R 4.0.0)
 Bioconductor  
 CRAN (R 4.0.0)
 CRAN (R 4.0.3)
 CRAN (R 4.0.3)
 CRAN (R 4.0.3)
 CRAN (R 4.0.0)
 CRAN (R 4.0.3)
 CRAN (R 4.0.0)
 CRAN (R 4.0.3)
 Bioconductor  
 CRAN (R 4.0.0)
 CRAN (R 4.0.3)
 CRAN (R 4.0.3)
 CRAN (R 4.0.3)
 Bioconductor  
 CRAN (R 4.0.3)
 CRAN (R 4.0.3)
 CRAN (R 4.0.3)
 CRAN (R 4.0.2)
 CRAN (R 4.0.0)
 Bioconductor  
 CRAN (R 4.0.2)
 Bioconductor  
 Bioconductor  
 Bioconductor  
 CRAN (R 4.0.3)
 CRAN (R 4.0.3)
 CRAN (R 4.0.0)
 CRAN (R 4.0.0)
 CRAN (R 4.0.0)
 CRAN (R 4.0.0)
 Bioconductor  
 CRAN (R 4.0.0)
 CRAN (R 4.0.2)
 CRAN (R 4.0.3)
 Bioconductor  
 CRAN (R 4.0.0)
 CRAN (R 4.0.3)
 CRAN (R 4.0.0)
 CRAN (R 4.0.0)
 CRAN (R 4.0.0)
 CRAN (R 4.0.0)
 Bioconductor  
 CRAN (R 4.0.3)
 CRAN (R 4.0.3)
 CRAN (R 4.0.3)
 CRAN (R 4.0.3)
 Bioconductor  
 CRAN (R 4.0.3)
 CRAN (R 4.0.0)
 Bioconductor  
 Bioconductor  
 CRAN (R 4.0.3)
 CRAN (R 4.0.0)
 Bioconductor  
 Bioconductor  
 Bioconductor  
 CRAN (R 4.0.0)
 Bioconductor  
 CRAN (R 4.0.0)
 Bioconductor  
 CRAN (R 4.0.0)
 CRAN (R 4.0.0)
 Bioconductor  
 CRAN (R 4.0.3)
 CRAN (R 4.0.0)
 CRAN (R 4.0.0)
 Bioconductor  
 CRAN (R 4.0.0)
 CRAN (R 4.0.3)
 CRAN (R 4.0.0)
 CRAN (R 4.0.0)
 CRAN (R 4.0.0)
 CRAN (R 4.0.3)
 CRAN (R 4.0.3)
 CRAN (R 4.0.0)
 CRAN (R 4.0.0)
 Bioconductor  
 CRAN (R 4.0.0)
 Bioconductor  

[1] /stornext/Projects/score/Analyses/C086_Kinkel/renv/library/R-4.0/x86_64-pc-linux-gnu
[2] /tmp/RtmpuiV8yk/renv-system-library
[3] /stornext/System/data/apps/R/R-4.0.3/lib64/R/library

 P ── Loaded and on-disk path mismatch.</code></pre>
</div>
</details>
<div class="sourceCode" id="cb2"><pre class="sourceCode r distill-force-highlighting-css"><code class="sourceCode r"></code></pre></div>
<section class="footnotes" role="doc-endnotes">
<hr />
<ol>
<li id="fn1" role="doc-endnote"><p>In the terminology of Bioconductor this is called a <em>SingleCellExperiment</em> object.<a href="#fnref1" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn2" role="doc-endnote"><p>Some care is taken to account for missing and duplicate gene symbols; missing symbols are replaced with the Ensembl identifier and duplicated symbols are concatenated with the (unique) Ensembl identifiers.<a href="#fnref2" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn3" role="doc-endnote"><p>I.e. the ‘library size’ when using read counts<a href="#fnref3" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn4" role="doc-endnote"><p>The choice of a 6 bp UMI in the mini-bulk protocol is because it is adapted from SCORE’s CEL-Seq2 single-cell RNA-sequencing protocol. With scRNA-seq data it is very rare to observe more than <span class="math inline">\(4^k = 4^6 = 4096\)</span> unique molecules for any one gene.<a href="#fnref4" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn5" role="doc-endnote"><p>The number of expressed genes refers to the number of genes which have non-zero counts (ie. they have been identified in the sample at least once)<a href="#fnref5" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn6" role="doc-endnote"><p>This is consistent with the use of UMI counts rather than read counts, as each transcript molecule can only produce one UMI count but can yield many reads after fragmentation.<a href="#fnref6" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
</ol>
</section>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom">
<h3 id="references">References</h3>
<div id="references-listing"></div>
</div>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<!--/radix_placeholder_navigation_after_body-->

</body>

</html>
