---
title: "Multi-sample comparisons of the Kinkel (C086) Smchd1 LSK mini-bulk data set"
description: "Read-based analysis"
author:
  - name: Peter Hickey
    url: https://peterhickey.org
    affiliation: WEHI SCORE
    affiliation_url: https://www.wehi.edu.au/people/shalin-naik/3310/score
date: "`r Sys.Date()`"
editor_options:
  chunk_output_type: console
bibliography: ref.bib
---

# Summary

This report analyses the read counts of the `Del` and `Het` samples only.
The report begins with some general [Setup] and is followed by various analyses denoted by a prefix (`y`, `ym`, `yf`):

  - `y`: [Analysis of aggregated technical replicates] with Remove Unwanted Variation (RUV) normalization
  - `ym`: [Male-only: analysis of aggregated technical replicates] using RUV normalization
  - `yf`: [Female-only: analysis of aggregated technical replicates] using RUV normalization
  
# Setup

```{r}
library(SingleCellExperiment)
library(here)
library(edgeR)
library(ggplot2)
library(cowplot)
library(patchwork)
library(ggrepel)
library(magrittr)
library(Glimma)
library(EDASeq)
library(RUVSeq)
library(pheatmap)

source(here("code/helper_functions.R"))
```

## Load data

```{r}
sce <- readRDS(here("data", "SCEs", "C086_Kinkel.preprocessed.SCE.rds"))
# NOTE: Reorder samples for plots (e.g., RLE plots).
sce <- sce[, order(sce$smchd1_genotype_updated, sce$sex, sce$mouse_number)]

# Restrict analysis to read counts
assays(sce) <- list(counts = assay(sce, "read_counts"))

outdir <- here("output", "DEGs", "reads")
dir.create(outdir, recursive = TRUE)
```

## Gene filtering

We filter the genes to only retain protein coding genes.

```{r}
sce <- sce[any(grepl("protein_coding", rowData(sce)$ENSEMBL.GENEBIOTYPE)), ]
```

## Sample filtering

We filter out the `WT` samples as these are not useful for DE analysis since there is only 1 biological replicate.

```{r}
sce <- sce[, sce$smchd1_genotype_updated != "WT"]
```

```{r}
colData(sce) <- droplevels(colData(sce))
# Some useful colours
sex_colours <- setNames(
  unique(sce$sex_colours),
  unique(names(sce$sex_colours)))
smchd1_genotype_updated_colours <- setNames(
  unique(sce$smchd1_genotype_updated_colours),
  unique(names(sce$smchd1_genotype_updated_colours)))
mouse_number_colours <- setNames(
  unique(sce$mouse_number_colours),
  unique(names(sce$mouse_number_colours)))
```

## Create DGEList objects

```{r}
x <- DGEList(
  counts = as.matrix(counts(sce)),
  samples = colData(sce),
  group = factor(sce$smchd1_genotype_updated),
  genes = flattenDF(rowData(sce)))
x <- x[rowSums(x$counts) > 0, ]
```

```{r}
y <- sumTechReps(x, x$samples$genotype.mouse)
ym <- y[, y$samples$sex == "M"]
yf <- y[, y$samples$sex == "F"]
```

### Gene filtering

For each analysis, shown are the number of genes that have sufficient counts to test for DE (`TRUE`) or not (`FALSE`).

#### `y`

```{r}
y_keep_exprs <- filterByExpr(y, group = y$samples$group, min.count = 5)
y <- y[y_keep_exprs, , keep.lib.sizes = FALSE]
table(y_keep_exprs)
```

#### `ym`

```{r}
ym_keep_exprs <- filterByExpr(ym, group = ym$samples$group, min.count = 5)
ym <- ym[ym_keep_exprs, , keep.lib.sizes = FALSE]
table(ym_keep_exprs)
```

#### `yf`

```{r}
yf_keep_exprs <- filterByExpr(yf, group = yf$samples$group, min.count = 5)
yf <- yf[yf_keep_exprs, , keep.lib.sizes = FALSE]
table(yf_keep_exprs)
```

### Normalization

Normalization with upper-quartile (UQ) normalization.

```{r}
x <- calcNormFactors(x, method = "upperquartile")
y <- calcNormFactors(y, method = "upperquartile")
ym <- calcNormFactors(ym, method = "upperquartile")
yf <- calcNormFactors(yf, method = "upperquartile")
```

## Aggregation of technical replicates

We have up to 3 technical replicates from each biological replicate.
Ideally, the technical replicates are very similar to one another when view on an MDS plot.
However, Figure \@ref(fig:mds) shows that:

- Some of the technical replicates are more variable than we would like
- Those 'outlier' samples have smaller library sizes

```{r mds, fig.cap = "MDS plots of individual technical replicates coloured by various experimental factors."}
x_mds <- plotMDS(x, plot = FALSE)

x_df <- cbind(data.frame(x = x_mds$x, y = x_mds$y), x$samples)
rownames(x_df) <- colnames(x)

p1 <- ggplot(
  aes(x = x, y = y, colour = smchd1_genotype_updated, label = rownames(x_df)),
  data = x_df) +
  geom_point() +
  geom_text_repel(size = 2) +
  theme_cowplot(font_size = 8) +
  xlab("Leading logFC dim 1") +
  ylab("Leading logFC dim 2") +
  scale_colour_manual(values = smchd1_genotype_updated_colours)
p2 <- ggplot(
  aes(x = x, y = y, colour = sex, label = rownames(x_df)),
  data = x_df) +
  geom_point() +
  geom_text_repel(size = 2) +
  theme_cowplot(font_size = 8) +
  xlab("Leading logFC dim 1") +
  ylab("Leading logFC dim 2") +
  scale_colour_manual(values = sex_colours)
p3 <- ggplot(
  aes(x = x, y = y, colour = mouse_number, label = rownames(x_df)),
  data = x_df) +
  geom_point() +
  geom_text_repel(size = 2) +
  theme_cowplot(font_size = 8) +
  xlab("Leading logFC dim 1") +
  ylab("Leading logFC dim 2") +
  scale_colour_manual(values = mouse_number_colours)
p4 <- ggplot(
  aes(x = x, y = y, colour = log10(lib.size), label = rownames(x_df)),
  data = x_df) +
  geom_point() +
  geom_text_repel(size = 2) +
  theme_cowplot(font_size = 8) +
  scale_colour_viridis_c() +
  xlab("Leading logFC dim 1") +
  ylab("Leading logFC dim 2")

p1 + p2 + p3 + p4 + plot_layout(ncol = 2)
```

Although we could analyse the data at the level of technical replicates, this is more complicated.
It will make our life easier by aggregating these technical replicates so that we have 1 sample per biological replicate.
To aggregate technical replicates we simply sum their counts.

Figure \@ref(fig:mds-y) shows the MDS plot after aggregating the technical replicates.
We still observe that:

- Samples cluster more strongly by sex than by genotype
- Some of the variation is associated with library size

These observations (and others) prompt us to consider further normalization of the data.

```{r mds-y, fig.cap = "MDS plots of aggregated technical replicates coloured by various experimental factors."}
y_mds <- plotMDS(y, plot = FALSE)

y_df <- cbind(data.frame(x = y_mds$x, y = y_mds$y), y$samples)
rownames(y_df) <- colnames(y)

p1 <- ggplot(
  aes(x = x, y = y, colour = smchd1_genotype_updated, label = rownames(y_df)),
  data = y_df) +
  geom_point() +
  geom_text_repel(size = 2) +
  theme_cowplot(font_size = 8) +
  xlab("Leading logFC dim 1") +
  ylab("Leading logFC dim 2") +
  scale_colour_manual(values = smchd1_genotype_updated_colours)
p2 <- ggplot(
  aes(x = x, y = y, colour = sex, label = rownames(y_df)),
  data = y_df) +
  geom_point() +
  geom_text_repel(size = 2) +
  theme_cowplot(font_size = 8) +
  xlab("Leading logFC dim 1") +
  ylab("Leading logFC dim 2") +
  scale_colour_manual(values = sex_colours)
p3 <- ggplot(
  aes(x = x, y = y, colour = mouse_number, label = rownames(y_df)),
  data = y_df) +
  geom_point() +
  geom_text_repel(size = 2) +
  theme_cowplot(font_size = 8) +
  xlab("Leading logFC dim 1") +
  ylab("Leading logFC dim 2") +
  scale_colour_manual(values = mouse_number_colours)
p4 <- ggplot(
  aes(x = x, y = y, colour = log10(lib.size), label = rownames(y_df)),
  data = y_df) +
  geom_point() +
  geom_text_repel(size = 2) +
  theme_cowplot(font_size = 8) +
  scale_colour_viridis_c() +
  xlab("Leading logFC dim 1") +
  ylab("Leading logFC dim 2")

p1 + p2 + p3 + p4 + plot_layout(ncol = 2)
```

## Further normalization

```{r, results = "hide"}
design <- model.matrix(
  ~0 + smchd1_genotype_updated + sex,
  y$samples)
colnames(design) <- sub(
  "smchd1_genotype_updated|sex",
  "",
  colnames(design))

y <- estimateDisp(y, design)

dgeglm <- glmFit(y, design)
contrasts <- makeContrasts(Het - Del, levels = design)
dgelrt <- glmLRT(dgeglm, contrast = contrasts)
tt <- topTags(dgelrt, n = Inf)
summary(decideTests(dgelrt))
```

The MDS plots in Figure \@ref(fig:mds-y) strongly suggest that we will not detect many differentially expressed genes (DEGs) using a standard analysis.
This is borne out in Figure \@ref(fig:vanilla-de), which shows the results of a standard `r BiocStyle::Biocpkg("edgeR")` analysis of the data (there is `r scales::number(sum(decideTests(dgelrt) != 0))` DEG ($FDR < 0.05$)).

```{r vanilla-de, fig.cap = "MD plot highlighting DE genes (left) and Smchd1 (right)."}
par(mfrow = c(1, 2))
plotMD(dgelrt)
abline(h = 0, lty = 2, col = "dodgerBlue")
plotMD(
  dgelrt,
  status = ifelse(rownames(dgelrt) == "Smchd1", "Smchd1", "Other"),
  hl.col = "orange")
abline(h = 0, lty = 2, col = "dodgerBlue")
```

To further normalize the data, we use [**RUVs**](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4404308/).

```{r}
differences <- makeGroups(
  interaction(y$samples$smchd1_genotype_updated, y$samples$sex))
ruv <- RUVs(
    x = betweenLaneNormalization(y$counts, "upper"),
    cIdx = rownames(y),
    k = 3,
    scIdx = differences)

design <- model.matrix(
  ~0 + smchd1_genotype_updated + sex + ruv$W,
  y$samples)
colnames(design) <- sub(
  "ruv\\$|smchd1_genotype_updated|sex",
  "",
  colnames(design))
```

Figure \@ref(fig:ruv-mds) shows that after applying RUVs samples cluster by combination of genotype and sex, which is what we want to see.

**Therefore, we opt to further normalize the data using RUVs in each DE analysis^[Details of RUVs parameters are given separately for each analysis.]**

```{r ruv-mds, fig.cap = "MDS plots of aggregated technical replicates following RUV coloured by various experimental factors."}
y_mds <- plotMDS(cpm(ruv$normalizedCounts, log = TRUE), plot = FALSE)

y_df <- cbind(data.frame(x = y_mds$x, y = y_mds$y), y$samples)
rownames(y_df) <- colnames(y)

p1 <- ggplot(
  aes(x = x, y = y, colour = smchd1_genotype_updated, label = rownames(y_df)),
  data = y_df) +
  geom_point() +
  geom_text_repel(size = 2) +
  theme_cowplot(font_size = 8) +
  xlab("Leading logFC dim 1") +
  ylab("Leading logFC dim 2") +
  scale_colour_manual(values = smchd1_genotype_updated_colours)
p2 <- ggplot(
  aes(x = x, y = y, colour = sex, label = rownames(y_df)),
  data = y_df) +
  geom_point() +
  geom_text_repel(size = 2) +
  theme_cowplot(font_size = 8) +
  xlab("Leading logFC dim 1") +
  ylab("Leading logFC dim 2") +
  scale_colour_manual(values = sex_colours)
p3 <- ggplot(
  aes(x = x, y = y, colour = mouse_number, label = rownames(y_df)),
  data = y_df) +
  geom_point() +
  geom_text_repel(size = 2) +
  theme_cowplot(font_size = 8) +
  xlab("Leading logFC dim 1") +
  ylab("Leading logFC dim 2") +
  scale_colour_manual(values = mouse_number_colours)
p4 <- ggplot(
  aes(x = x, y = y, colour = log10(lib.size), label = rownames(y_df)),
  data = y_df) +
  geom_point() +
  geom_text_repel(size = 2) +
  theme_cowplot(font_size = 8) +
  scale_colour_viridis_c() +
  xlab("Leading logFC dim 1") +
  ylab("Leading logFC dim 2")

p1 + p2 + p3 + p4 + plot_layout(ncol = 2)
```

## Gene sets

Marnie and Sarah interested in the following gene sets:

- Non-directional gene sets (i.e. no logFC available from previous data)
  - `X-linked`: Anything with `ENSEMBL.SEQNAME` set as '`X`').
  - `Protocadherin`: Genes with symbol of the form *Pcdh#*.
  - `Prader-Willi`: Imprinted genes from the Prader-Willi cluster (*Ndn*, *Mkrn3*, *Peg12*)
- Directional gene sets (i.e. logFC available from previous data). These are the DEGs from a "previous RNA-seq experiment on male and female Smchd1-het and -del B cells from aged mice (>12 months old). [Sarah] used SeqMonk/EdgeR to compare Smchd1-del to Smchd1-het, keeping males and females separate" (email from Sarah). These are available as Excel files in [`data/sarahs_gene_lists/`](../data/sarahs_gene_lists)
  - `Male: Het vs. Del`: DEGs from the comparison of male samples.
  - `Female: Het vs. Del`: DEGs from the comparison of female samples.

```{r}
male_het_vs_del <- readxl::read_excel(
  here("data", "sarahs_gene_lists", "MaleHetvDel_EdgeR.xlsx"))
male_het_vs_del <- as.data.frame(male_het_vs_del)
female_het_vs_del <- readxl::read_excel(
  here("data", "sarahs_gene_lists", "Bcell_female_HetvDel_EdgeR_mRNA.xlsx"))
female_het_vs_del <- as.data.frame(female_het_vs_del)
```

# Analysis of aggregated technical replicates

```{r}
y_differences <- makeGroups(
  interaction(y$samples$smchd1_genotype_updated, y$samples$sex))
y_ruv <- RUVs(
    x = betweenLaneNormalization(y$counts, "upper"),
    cIdx = rownames(y),
    k = 3,
    scIdx = y_differences)

y_design <- model.matrix(
  ~0 + smchd1_genotype_updated + sex + y_ruv$W,
  y$samples)
colnames(y_design) <- sub(
  "ruv\\$|smchd1_genotype_updated|sex",
  "",
  colnames(y_design))
```

## MDS

```{r, fig.cap = "MDS plots of aggregated technical replicates following RUV coloured by various experimental factors."}
y_mds <- plotMDS(cpm(y_ruv$normalizedCounts, log = TRUE), plot = FALSE)

y_df <- cbind(data.frame(x = y_mds$x, y = y_mds$y), y$samples)
rownames(y_df) <- colnames(y)

p1 <- ggplot(
  aes(x = x, y = y, colour = smchd1_genotype_updated, label = rownames(y_df)),
  data = y_df) +
  geom_point() +
  geom_text_repel(size = 2) +
  theme_cowplot(font_size = 8) +
  xlab("Leading logFC dim 1") +
  ylab("Leading logFC dim 2") +
  scale_colour_manual(values = smchd1_genotype_updated_colours)
p2 <- ggplot(
  aes(x = x, y = y, colour = sex, label = rownames(y_df)),
  data = y_df) +
  geom_point() +
  geom_text_repel(size = 2) +
  theme_cowplot(font_size = 8) +
  xlab("Leading logFC dim 1") +
  ylab("Leading logFC dim 2") +
  scale_colour_manual(values = sex_colours)
p3 <- ggplot(
  aes(x = x, y = y, colour = mouse_number, label = rownames(y_df)),
  data = y_df) +
  geom_point() +
  geom_text_repel(size = 2) +
  theme_cowplot(font_size = 8) +
  xlab("Leading logFC dim 1") +
  ylab("Leading logFC dim 2") +
  scale_colour_manual(values = mouse_number_colours)
p4 <- ggplot(
  aes(x = x, y = y, colour = log10(lib.size), label = rownames(y_df)),
  data = y_df) +
  geom_point() +
  geom_text_repel(size = 2) +
  theme_cowplot(font_size = 8) +
  scale_colour_viridis_c() +
  xlab("Leading logFC dim 1") +
  ylab("Leading logFC dim 2")

p1 + p2 + p3 + p4 + plot_layout(ncol = 2)
```

## DE analysis

Fit a model using `r BiocStyle::Biocpkg("edgeR")` with a term for `smchd1_genotype_updated` and `sex`.

```{r, fig.show = "hide"}
y <- estimateDisp(y, y_design)
par(mfrow = c(1, 1))
plotBCV(y)
```

```{r, fig.asp = 1}
y_dgeglm <- glmFit(y, y_design)
y_contrast <- makeContrasts(Het - Del, levels = y_design)
y_dgelrt <- glmLRT(y_dgeglm, contrast = y_contrast)
y_dt <- decideTests(y_dgelrt)

summary(y_dt) %>%
  knitr::kable(caption = "Number of DEGs (FDR < 0.05)")
```

```{r, fig.cap = "MD plot highlighting DEGs (red).", fig.asp = 1}
par(mfrow = c(1, 1))
plotMD(y_dgelrt)
```

An interactive `r BiocStyle::Biocpkg("Glimma")` MD plot of the differential expression results is available from [`output/DEGs/reads/glimma-plots/aggregated_tech_reps.md-plot.html`](../output/DEGs/reads/glimma-plots/aggregated_tech_reps.md-plot.html).

```{r}
y_tt <- topTags(y_dgelrt, n = Inf, sort.by = "none")
glMDPlot(
  x = y_dgelrt,
  counts = y,
  anno = y$genes,
  display.columns = c("ENSEMBL.SYMBOL"),
  groups = y$samples$group,
  samples = colnames(y),
  status = y_dt,
  transform = TRUE,
  sample.cols = y$samples$smchd1_genotype_updated_colours,
  path = outdir,
  html = "aggregated_tech_reps.md-plot",
  launch = FALSE)
```

```{r}
y_tt <- topTags(y_dgelrt, p.value = 0.05, n = Inf)
if (nrow(y_tt)) {
  y_tt$table[, c("logFC", "logCPM", "LR", "PValue", "FDR")] %>%
    DT::datatable(caption = "DEGs (FDR < 0.05)") %>%
    DT::formatSignif(
      c("logFC", "logCPM", "LR", "PValue", "FDR"),
      digits = 3)
}
```

CSV file of results available from [`output/DEGs/reads/aggregated_tech_reps.DEGs.csv.gz`](../output/DEGs/reads/aggregated_tech_reps.DEGs.csv.gz).

```{r}
gzout <- gzfile(
  description = file.path(outdir, "aggregated_tech_reps.DEGs.csv.gz"),
  open = "wb")
write.csv(
  topTags(y_dgelrt, n = Inf),
  gzout,
  # NOTE: quote = TRUE needed because some fields contain commas.
  quote = TRUE,
  row.names = FALSE)
close(gzout)
```

Figure \@ref(fig:heatmap-y) is a heatmap showing the expression of the DEGs.

```{r heatmap-y, fig.asp = 1.5, fig.cap = "Heatmap of upper-quartile-normalized logCPM values for the DEGs."}
pheatmap(
  cpm(y, log = TRUE)[rownames(y_tt), ],
  cluster_rows = TRUE,
  cluster_cols = FALSE,
  color = hcl.colors(101, "Blue-Red 3"),
  breaks = seq(-3, 3, length.out = 101),
  scale = "row",
  annotation_col = data.frame(
    genotype = y$samples$smchd1_genotype_updated,
    sex = y$samples$sex,
    row.names = colnames(y)),
  annotation_colors = list(
    genotype = smchd1_genotype_updated_colours,
    sex = sex_colours),
  main = "Male and female samples",
  fontsize = 9)
```

## Gene set tests

```{r}
y_X <- y$genes$ENSEMBL.SEQNAME == "X"
y_protocadherin <- grepl("Pcdh", rownames(y))
y_pw <- rownames(y) %in% c("Ndn", "Mkrn3", "Peg12")
y_male <- male_het_vs_del[, c("Probe", "Differential expression")]
y_female <- female_het_vs_del[, c("Probe", "Differential expression")]

y_index <- list(
  `X-linked` = y_X,
  Protocadherin = y_protocadherin,
  `Prader-Willi` = y_pw,
  `Male: Het vs. Del` = y_male,
  `Female: Het vs. Del` = y_female)
```

```{r}
fry(y, index = y_index, design = y_design, contrast = y_contrast) %>%
    knitr::kable(
      caption = "Results of applying `fry()` to the RNA-seq differential expression results using the supplied gene sets. A significant 'Up' P-value means that the differential expression results found in the RNA-seq data are positively correlated with the expression signature from the corresponding gene set. Conversely, a significant 'Down' P-value means that the differential expression log-fold-changes are negatively correlated with the expression signature from the corresponding gene set. A significant 'Mixed' P-value means that the genes in the set tend to be differentially expressed without regard for direction.")
```

```{r, fig.asp = 5 / 2, fig.cap = "MD-plot and barcode plot of DEGs in supplied gene sets. For each set, an enrichment line that shows the relative enrichment of the vertical bars in each part of the plot is displayed. Directional gene sets have (1) MD plot points coloured according to the statistical significance of the gene **in the gene set** (2) barcode plot 'weights' given by the logFC of the gene **in the gene set**."}
par(mfrow = c(length(y_index), 2))
for (n in names(y_index)) {
  if (is.data.frame(y_index[[n]])) {
    y_status <- rep(0, nrow(y_dgelrt))
    y_status[rownames(y) %in% y_index[[n]][, 1]] <-
      sign(y_index[[n]][y_index[[n]][, 1] %in% rownames(y), 2])
    plotMD(
      y_dgelrt,
      status = y_status,
      hl.col = c("red", "blue"),
      legend = "bottomright",
      main = n)
    abline(h = 0, col = "darkgrey")
  } else {
    y_status <- rep("Other", nrow(y_dgelrt))
    y_status[y_index[[n]]] <- n
    plotMD(
      y_dgelrt,
      status = y_status,
      values = n,
      hl.col = "red",
      legend = "bottomright",
      main = n)
    abline(h = 0, col = "darkgrey")
  }
  
  if (is.data.frame(y_index[[n]])) {
    barcodeplot(
      y_dgelrt$table$logFC,
      rownames(y) %in% y_index[[n]][, 1],
      gene.weights = y_index[[n]][
        y_index[[n]][, 1] %in% rownames(y), 2],
      labels = c("Down in Del", "Up in Del"))
  } else {
    barcodeplot(
      y_dgelrt$table$logFC,
      y_index[[n]],
      labels = c("Down in Del", "Up in Del"))
  }
}
```

# Male-only: analysis of aggregated technical replicates

```{r}
ym_differences <- makeGroups(ym$samples$smchd1_genotype_updated)
ym_ruv <- RUVs(
    x = betweenLaneNormalization(ym$counts, "upper"),
    cIdx = rownames(ym),
    k = 1,
    scIdx = ym_differences)

ym_design <- model.matrix(
  ~0 + smchd1_genotype_updated + ym_ruv$W,
  ym$samples)
colnames(ym_design) <- sub(
  "ruv\\$|smchd1_genotype_updated",
  "",
  colnames(ym_design))
```

## MDS

```{r, fig.cap = "MDS plots of aggregated technical replicates following RUV coloured by various experimental factors."}
ym_mds <- plotMDS(cpm(ym_ruv$normalizedCounts, log = TRUE), plot = FALSE)

ym_df <- cbind(data.frame(x = ym_mds$x, y = ym_mds$y), ym$samples)
rownames(ym_df) <- colnames(ym)

p1 <- ggplot(
  aes(x = x, y = y, colour = smchd1_genotype_updated, label = rownames(ym_df)),
  data = ym_df) +
  geom_point() +
  geom_text_repel(size = 2) +
  theme_cowplot(font_size = 8) +
  xlab("Leading logFC dim 1") +
  ylab("Leading logFC dim 2") +
  scale_colour_manual(values = smchd1_genotype_updated_colours)
p2 <- ggplot(
  aes(x = x, y = y, colour = sex, label = rownames(ym_df)),
  data = ym_df) +
  geom_point() +
  geom_text_repel(size = 2) +
  theme_cowplot(font_size = 8) +
  xlab("Leading logFC dim 1") +
  ylab("Leading logFC dim 2") +
  scale_colour_manual(values = sex_colours)
p3 <- ggplot(
  aes(x = x, y = y, colour = mouse_number, label = rownames(ym_df)),
  data = ym_df) +
  geom_point() +
  geom_text_repel(size = 2) +
  theme_cowplot(font_size = 8) +
  xlab("Leading logFC dim 1") +
  ylab("Leading logFC dim 2") +
  scale_colour_manual(values = mouse_number_colours)
p4 <- ggplot(
  aes(x = x, y = y, colour = log10(lib.size), label = rownames(ym_df)),
  data = ym_df) +
  geom_point() +
  geom_text_repel(size = 2) +
  theme_cowplot(font_size = 8) +
  scale_colour_viridis_c() +
  xlab("Leading logFC dim 1") +
  ylab("Leading logFC dim 2")

p1 + p2 + p3 + p4 + plot_layout(ncol = 2)
```

## DE analysis

Fit a model using `r BiocStyle::Biocpkg("edgeR")` with a term for `smchd1_genotype_updated`.

```{r, fig.show = "hide"}
ym <- estimateDisp(ym, ym_design)
par(mfrow = c(1, 1))
plotBCV(ym)
```

```{r, fig.asp = 1}
ym_dgeglm <- glmFit(ym, ym_design)
ym_contrast <- makeContrasts(Het - Del, levels = ym_design)
ym_dgelrt <- glmLRT(ym_dgeglm, contrast = ym_contrast)
ym_dt <- decideTests(ym_dgelrt)

summary(ym_dt) %>%
  knitr::kable(caption = "Number of DEGs (FDR < 0.05)")
```

```{r, fig.cap = "MD plot highlighting DEGs (red).", fig.asp = 1}
par(mfrow = c(1, 1))
plotMD(ym_dgelrt)
```

An interactive `r BiocStyle::Biocpkg("Glimma")` MD plot of the differential expression results is available from [`output/DEGs/reads/glimma-plots/aggregated_tech_reps.md-plot.html`](../output/DEGs/reads/glimma-plots/aggregated_tech_reps.md-plot.html).

```{r}
ym_tt <- topTags(ym_dgelrt, n = Inf, sort.by = "none")
glMDPlot(
  x = ym_dgelrt,
  counts = ym,
  anno = ym$genes,
  display.columns = c("ENSEMBL.SYMBOL"),
  groups = ym$samples$group,
  samples = colnames(ym),
  status = ym_dt,
  transform = TRUE,
  sample.cols = ym$samples$smchd1_genotype_updated_colours,
  path = outdir,
  html = "aggregated_tech_reps.males.md-plot",
  launch = FALSE)
```

```{r}
ym_tt <- topTags(ym_dgelrt, p.value = 0.05, n = Inf)
if (nrow(ym_tt)) {
  ym_tt$table[, c("logFC", "logCPM", "LR", "PValue", "FDR")] %>%
    DT::datatable(caption = "DEGs (FDR < 0.05)") %>%
    DT::formatSignif(
      c("logFC", "logCPM", "LR", "PValue", "FDR"),
      digits = 3)
}
```

CSV file of results available from [`output/DEGs/reads/aggregated_tech_reps.males.DEGs.csv.gz`](../output/DEGs/reads/aggregated_tech_reps.males.DEGs.csv.gz).

```{r}
gzout <- gzfile(
  description = file.path(outdir, "aggregated_tech_reps.males.DEGs.csv.gz"),
  open = "wb")
write.csv(
  topTags(ym_dgelrt, n = Inf),
  gzout,
  # NOTE: quote = TRUE needed because some fields contain commas.
  quote = TRUE,
  row.names = FALSE)
close(gzout)
```

Figure \@ref(fig:heatmap-ym) is a heatmap showing the expression of the DEGs.

```{r heatmap-ym, fig.asp = 1.5, fig.cap = "Heatmap of upper-quartile-normalzied logCPM values for the DEGs."}
pheatmap(
  cpm(ym, log = TRUE)[rownames(ym_tt), ],
  cluster_rows = TRUE,
  cluster_cols = FALSE,
  color = hcl.colors(101, "Blue-Red 3"),
  breaks = seq(-3, 3, length.out = 101),
  scale = "row",
  annotation_col = data.frame(
    genotype = ym$samples$smchd1_genotype_updated,
    sex = ym$samples$sex,
    row.names = colnames(ym)),
  annotation_colors = list(
    genotype = smchd1_genotype_updated_colours,
    sex = sex_colours),
  main = "Male samples",
  fontsize = 9)
```

## Gene set tests

```{r}
ym_X <- ym$genes$ENSEMBL.SEQNAME == "X"
ym_protocadherin <- grepl("Pcdh", rownames(ym))
ym_pw <- rownames(ym) %in% c("Ndn", "Mkrn3", "Peg12")
ym_male <- male_het_vs_del[, c("Probe", "Differential expression")]
ym_female <- female_het_vs_del[, c("Probe", "Differential expression")]

ym_index <- list(
  `X-linked` = ym_X,
  Protocadherin = ym_protocadherin,
  `Prader-Willi` = ym_pw,
  `Male: Het vs. Del` = ym_male,
  `Female: Het vs. Del` = ym_female)
```

```{r}
fry(ym, index = ym_index, design = ym_design, contrast = ym_contrast) %>%
    knitr::kable(
      caption = "Results of applying `fry()` to the RNA-seq differential expression results using the supplied gene sets. A significant 'Up' P-value means that the differential expression results found in the RNA-seq data are positively correlated with the expression signature from the corresponding gene set. Conversely, a significant 'Down' P-value means that the differential expression log-fold-changes are negatively correlated with the expression signature from the corresponding gene set. A significant 'Mixed' P-value means that the genes in the set tend to be differentially expressed without regard for direction.")
```

```{r, fig.asp = 5 / 2, fig.cap = "MD-plot and barcode plot of DEGs in supplied gene sets. For each set, an enrichment line that shows the relative enrichment of the vertical bars in each part of the plot is displayed. Directional gene sets have (1) MD plot points coloured according to the statistical significance of the gene **in the gene set** (2) barcode plot 'weights' given by the logFC of the gene **in the gene set**."}
par(mfrow = c(length(ym_index), 2))
for (n in names(ym_index)) {
  if (is.data.frame(ym_index[[n]])) {
    ym_status <- rep(0, nrow(ym_dgelrt))
    ym_status[rownames(ym) %in% ym_index[[n]][, 1]] <-
      sign(ym_index[[n]][ym_index[[n]][, 1] %in% rownames(ym), 2])
    plotMD(
      ym_dgelrt,
      status = ym_status,
      hl.col = c("red", "blue"),
      legend = "bottomright",
      main = n)
    abline(h = 0, col = "darkgrey")
  } else {
    ym_status <- rep("Other", nrow(ym_dgelrt))
    ym_status[ym_index[[n]]] <- n
    plotMD(
      ym_dgelrt,
      status = ym_status,
      values = n,
      hl.col = "red",
      legend = "bottomright",
      main = n)
    abline(h = 0, col = "darkgrey")
  }
  
  if (is.data.frame(ym_index[[n]])) {
    barcodeplot(
      ym_dgelrt$table$logFC,
      rownames(ym) %in% ym_index[[n]][, 1],
      gene.weights = ym_index[[n]][
        ym_index[[n]][, 1] %in% rownames(ym), 2],
      labels = c("Down in Del", "Up in Del"))
  } else {
    barcodeplot(
      ym_dgelrt$table$logFC,
      ym_index[[n]],
      labels = c("Down in Del", "Up in Del"))
  }
}
```


# Female-only: analysis of aggregated technical replicates

```{r}
yf_differences <- makeGroups(yf$samples$smchd1_genotype_updated)
yf_ruv <- RUVs(
    x = betweenLaneNormalization(yf$counts, "upper"),
    cIdx = rownames(yf),
    k = 1,
    scIdx = yf_differences)

yf_design <- model.matrix(
  ~0 + smchd1_genotype_updated + yf_ruv$W,
  yf$samples)
colnames(yf_design) <- sub(
  "ruv\\$|smchd1_genotype_updated",
  "",
  colnames(yf_design))
```

## MDS

```{r, fig.cap = "MDS plots of aggregated technical replicates following RUV coloured by various experimental factors."}
yf_mds <- plotMDS(cpm(yf_ruv$normalizedCounts, log = TRUE), plot = FALSE)

yf_df <- cbind(data.frame(x = yf_mds$x, y = yf_mds$y), yf$samples)
rownames(yf_df) <- colnames(yf)

p1 <- ggplot(
  aes(x = x, y = y, colour = smchd1_genotype_updated, label = rownames(yf_df)),
  data = yf_df) +
  geom_point() +
  geom_text_repel(size = 2) +
  theme_cowplot(font_size = 8) +
  xlab("Leading logFC dim 1") +
  ylab("Leading logFC dim 2") +
  scale_colour_manual(values = smchd1_genotype_updated_colours)
p2 <- ggplot(
  aes(x = x, y = y, colour = sex, label = rownames(yf_df)),
  data = yf_df) +
  geom_point() +
  geom_text_repel(size = 2) +
  theme_cowplot(font_size = 8) +
  xlab("Leading logFC dim 1") +
  ylab("Leading logFC dim 2") +
  scale_colour_manual(values = sex_colours)
p3 <- ggplot(
  aes(x = x, y = y, colour = mouse_number, label = rownames(yf_df)),
  data = yf_df) +
  geom_point() +
  geom_text_repel(size = 2) +
  theme_cowplot(font_size = 8) +
  xlab("Leading logFC dim 1") +
  ylab("Leading logFC dim 2") +
  scale_colour_manual(values = mouse_number_colours)
p4 <- ggplot(
  aes(x = x, y = y, colour = log10(lib.size), label = rownames(yf_df)),
  data = yf_df) +
  geom_point() +
  geom_text_repel(size = 2) +
  theme_cowplot(font_size = 8) +
  scale_colour_viridis_c() +
  xlab("Leading logFC dim 1") +
  ylab("Leading logFC dim 2")

p1 + p2 + p3 + p4 + plot_layout(ncol = 2)
```

## DE analysis

Fit a model using `r BiocStyle::Biocpkg("edgeR")` with a term for `smchd1_genotype_updated`.

```{r, fig.show = "hide"}
yf <- estimateDisp(yf, yf_design)
par(mfrow = c(1, 1))
plotBCV(yf)
```

```{r, fig.asp = 1}
yf_dgeglm <- glmFit(yf, yf_design)
yf_contrast <- makeContrasts(Het - Del, levels = yf_design)
yf_dgelrt <- glmLRT(yf_dgeglm, contrast = yf_contrast)
yf_dt <- decideTests(yf_dgelrt)

summary(yf_dt) %>%
  knitr::kable(caption = "Number of DEGs (FDR < 0.05)")
```

```{r, fig.cap = "MD plot highlighting DEGs (red).", fig.asp = 1}
par(mfrow = c(1, 1))
plotMD(yf_dgelrt)
```

An interactive `r BiocStyle::Biocpkg("Glimma")` MD plot of the differential expression results is available from [`output/DEGs/reads/glimma-plots/aggregated_tech_reps.md-plot.html`](../output/DEGs/reads/glimma-plots/aggregated_tech_reps.md-plot.html).

```{r}
yf_tt <- topTags(yf_dgelrt, n = Inf, sort.by = "none")
glMDPlot(
  x = yf_dgelrt,
  counts = yf,
  anno = yf$genes,
  display.columns = c("ENSEMBL.SYMBOL"),
  groups = yf$samples$group,
  samples = colnames(yf),
  status = yf_dt,
  transform = TRUE,
  sample.cols = yf$samples$smchd1_genotype_updated_colours,
  path = outdir,
  html = "aggregated_tech_reps.females.md-plot",
  launch = FALSE)
```

```{r}
yf_tt <- topTags(yf_dgelrt, p.value = 0.05, n = Inf)
if (nrow(yf_tt)) {
  yf_tt$table[, c("logFC", "logCPM", "LR", "PValue", "FDR")] %>%
    DT::datatable(caption = "DEGs (FDR < 0.05)") %>%
    DT::formatSignif(
      c("logFC", "logCPM", "LR", "PValue", "FDR"),
      digits = 3)
}
```

CSV file of results available from [`output/DEGs/reads/aggregated_tech_reps.females.DEGs.csv.gz`](../output/DEGs/reads/aggregated_tech_reps.females.DEGs.csv.gz).

```{r}
gzout <- gzfile(
  description = file.path(outdir, "aggregated_tech_reps.females.DEGs.csv.gz"),
  open = "wb")
write.csv(
  topTags(yf_dgelrt, n = Inf),
  gzout,
  # NOTE: quote = TRUE needed because some fields contain commas.
  quote = TRUE,
  row.names = FALSE)
close(gzout)
```

Figure \@ref(fig:heatmap-yf) is a heatmap showing the expression of the DEGs.

```{r heatmap-yf, fig.asp = 1.5, fig.cap = "Heatmap of upper-quartile-normalzied logCPM values for the DEGs."}
pheatmap(
  cpm(yf, log = TRUE)[rownames(yf_tt), ],
  cluster_rows = TRUE,
  cluster_cols = FALSE,
  color = hcl.colors(101, "Blue-Red 3"),
  breaks = seq(-3, 3, length.out = 101),
  scale = "row",
  annotation_col = data.frame(
    genotype = yf$samples$smchd1_genotype_updated,
    sex = yf$samples$sex,
    row.names = colnames(yf)),
  annotation_colors = list(
    genotype = smchd1_genotype_updated_colours,
    sex = sex_colours),
  main = "Female samples",
  fontsize = 9)
```

## Gene set tests

```{r}
yf_X <- yf$genes$ENSEMBL.SEQNAME == "X"
yf_protocadherin <- grepl("Pcdh", rownames(yf))
yf_pw <- rownames(yf) %in% c("Ndn", "Mkrn3", "Peg12")
yf_male <- male_het_vs_del[, c("Probe", "Differential expression")]
yf_female <- female_het_vs_del[, c("Probe", "Differential expression")]

yf_index <- list(
  `X-linked` = yf_X,
  Protocadherin = yf_protocadherin,
  `Prader-Willi` = yf_pw,
  `Male: Het vs. Del` = yf_male,
  `Female: Het vs. Del` = yf_female)
```

```{r}
fry(yf, index = yf_index, design = yf_design, contrast = yf_contrast) %>%
    knitr::kable(
      caption = "Results of applying `fry()` to the RNA-seq differential expression results using the supplied gene sets. A significant 'Up' P-value means that the differential expression results found in the RNA-seq data are positively correlated with the expression signature from the corresponding gene set. Conversely, a significant 'Down' P-value means that the differential expression log-fold-changes are negatively correlated with the expression signature from the corresponding gene set. A significant 'Mixed' P-value means that the genes in the set tend to be differentially expressed without regard for direction.")
```

```{r, fig.asp = 5 / 2, fig.cap = "MD-plot and barcode plot of DEGs in supplied gene sets. For each set, an enrichment line that shows the relative enrichment of the vertical bars in each part of the plot is displayed. Directional gene sets have (1) MD plot points coloured according to the statistical significance of the gene **in the gene set** (2) barcode plot 'weights' given by the logFC of the gene **in the gene set**."}
par(mfrow = c(length(yf_index), 2))
for (n in names(yf_index)) {
  if (is.data.frame(yf_index[[n]])) {
    yf_status <- rep(0, nrow(yf_dgelrt))
    yf_status[rownames(yf) %in% yf_index[[n]][, 1]] <-
      sign(yf_index[[n]][yf_index[[n]][, 1] %in% rownames(yf), 2])
    plotMD(
      yf_dgelrt,
      status = yf_status,
      hl.col = c("red", "blue"),
      legend = "bottomright",
      main = n)
    abline(h = 0, col = "darkgrey")
  } else {
    yf_status <- rep("Other", nrow(yf_dgelrt))
    yf_status[yf_index[[n]]] <- n
    plotMD(
      yf_dgelrt,
      status = yf_status,
      values = n,
      hl.col = "red",
      legend = "bottomright",
      main = n)
    abline(h = 0, col = "darkgrey")
  }
  
  if (is.data.frame(yf_index[[n]])) {
    barcodeplot(
      yf_dgelrt$table$logFC,
      rownames(yf) %in% yf_index[[n]][, 1],
      gene.weights = yf_index[[n]][
        yf_index[[n]][, 1] %in% rownames(yf), 2],
      labels = c("Down in Del", "Up in Del"))
  } else {
    barcodeplot(
      yf_dgelrt$table$logFC,
      yf_index[[n]],
      labels = c("Down in Del", "Up in Del"))
  }
}
```

# Session info {.appendix}

<summary>The analysis and this document were prepared using the following software (click triangle to expand)</summary>
<details>

```{r}
sessioninfo::session_info()
```

</details>
